{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "23a7f7d2-2e98-45be-b488-688489353f1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a29f3799-2118-4ac8-b80b-4c0ff7748a4e",
   "metadata": {},
   "source": [
    "# hMSC ATAC-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "0444d201-284c-4343-bd23-243324430ab7",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_atac = pd.read_csv('/mnt/10T-backup-mingju/ifar/ATAC_seq/hMSC/hMSC_Day1/atac_pipeline_results/idr.optimal_peak.narrowPeak_w_header',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4d9d72a8-f8cc-465b-a0aa-4ee95e8a4f2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521173</td>\n",
       "      <td>22521671</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>30.15064</td>\n",
       "      <td>3810.73267</td>\n",
       "      <td>3801.84375</td>\n",
       "      <td>305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633719</td>\n",
       "      <td>634669</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>5.18915</td>\n",
       "      <td>3772.65210</td>\n",
       "      <td>3764.00610</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>632136</td>\n",
       "      <td>633008</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>4.69070</td>\n",
       "      <td>3083.66895</td>\n",
       "      <td>3076.10742</td>\n",
       "      <td>428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>4.60007</td>\n",
       "      <td>2907.49854</td>\n",
       "      <td>2900.17773</td>\n",
       "      <td>616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521173</td>\n",
       "      <td>22521671</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>22.59163</td>\n",
       "      <td>2536.15381</td>\n",
       "      <td>2529.12964</td>\n",
       "      <td>199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202010</th>\n",
       "      <td>chr4</td>\n",
       "      <td>173243168</td>\n",
       "      <td>173243818</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.79579</td>\n",
       "      <td>16.95501</td>\n",
       "      <td>14.97065</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202011</th>\n",
       "      <td>chr1</td>\n",
       "      <td>170318562</td>\n",
       "      <td>170319107</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.79579</td>\n",
       "      <td>16.95501</td>\n",
       "      <td>14.97065</td>\n",
       "      <td>187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202012</th>\n",
       "      <td>chr12</td>\n",
       "      <td>79451933</td>\n",
       "      <td>79452511</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.65382</td>\n",
       "      <td>16.95480</td>\n",
       "      <td>14.97045</td>\n",
       "      <td>464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202013</th>\n",
       "      <td>chr1</td>\n",
       "      <td>214524921</td>\n",
       "      <td>214525302</td>\n",
       "      <td>.</td>\n",
       "      <td>690</td>\n",
       "      <td>.</td>\n",
       "      <td>3.08018</td>\n",
       "      <td>16.95387</td>\n",
       "      <td>14.96954</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202014</th>\n",
       "      <td>chr4</td>\n",
       "      <td>68465850</td>\n",
       "      <td>68467032</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.21736</td>\n",
       "      <td>16.95297</td>\n",
       "      <td>14.96865</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>202015 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0       chr17   22521173   22521671    .   1000      .     30.15064   \n",
       "1        chr1     633719     634669    .   1000      .      5.18915   \n",
       "2        chr1     632136     633008    .   1000      .      4.69070   \n",
       "3        chr1     629079     630024    .   1000      .      4.60007   \n",
       "4       chr17   22521173   22521671    .   1000      .     22.59163   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "202010   chr4  173243168  173243818    .   1000      .      3.79579   \n",
       "202011   chr1  170318562  170319107    .   1000      .      3.79579   \n",
       "202012  chr12   79451933   79452511    .   1000      .      3.65382   \n",
       "202013   chr1  214524921  214525302    .    690      .      3.08018   \n",
       "202014   chr4   68465850   68467032    .   1000      .      3.21736   \n",
       "\n",
       "            pValue      qValue  peak  \n",
       "0       3810.73267  3801.84375   305  \n",
       "1       3772.65210  3764.00610   291  \n",
       "2       3083.66895  3076.10742   428  \n",
       "3       2907.49854  2900.17773   616  \n",
       "4       2536.15381  2529.12964   199  \n",
       "...            ...         ...   ...  \n",
       "202010    16.95501    14.97065    57  \n",
       "202011    16.95501    14.97065   187  \n",
       "202012    16.95480    14.97045   464  \n",
       "202013    16.95387    14.96954    49  \n",
       "202014    16.95297    14.96865   220  \n",
       "\n",
       "[202015 rows x 10 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "4ada0913-19b6-4530-9759-df5e068706c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_atac['index'] = hMSC_atac['Chr'] + \":\" + hMSC_atac['Start'].astype(str) + \"-\" + hMSC_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "b47a673d-73d6-4efe-8a27-cbbea8e45e9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_atac['ATAC_hMSC'] = \"ATAC-seq:\" + hMSC_atac['Chr'] + \":\" + hMSC_atac['Start'].astype(str) + \"-\" + hMSC_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a959e645-7cd5-416a-8aa8-3a4030e4fa18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521173</td>\n",
       "      <td>22521671</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>30.15064</td>\n",
       "      <td>3810.73267</td>\n",
       "      <td>3801.84375</td>\n",
       "      <td>305</td>\n",
       "      <td>chr17:22521173-22521671</td>\n",
       "      <td>ATAC-seq:chr17:22521173-22521671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633719</td>\n",
       "      <td>634669</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>5.18915</td>\n",
       "      <td>3772.65210</td>\n",
       "      <td>3764.00610</td>\n",
       "      <td>291</td>\n",
       "      <td>chr1:633719-634669</td>\n",
       "      <td>ATAC-seq:chr1:633719-634669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>632136</td>\n",
       "      <td>633008</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>4.69070</td>\n",
       "      <td>3083.66895</td>\n",
       "      <td>3076.10742</td>\n",
       "      <td>428</td>\n",
       "      <td>chr1:632136-633008</td>\n",
       "      <td>ATAC-seq:chr1:632136-633008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>4.60007</td>\n",
       "      <td>2907.49854</td>\n",
       "      <td>2900.17773</td>\n",
       "      <td>616</td>\n",
       "      <td>chr1:629079-630024</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521173</td>\n",
       "      <td>22521671</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>22.59163</td>\n",
       "      <td>2536.15381</td>\n",
       "      <td>2529.12964</td>\n",
       "      <td>199</td>\n",
       "      <td>chr17:22521173-22521671</td>\n",
       "      <td>ATAC-seq:chr17:22521173-22521671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202010</th>\n",
       "      <td>chr4</td>\n",
       "      <td>173243168</td>\n",
       "      <td>173243818</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.79579</td>\n",
       "      <td>16.95501</td>\n",
       "      <td>14.97065</td>\n",
       "      <td>57</td>\n",
       "      <td>chr4:173243168-173243818</td>\n",
       "      <td>ATAC-seq:chr4:173243168-173243818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202011</th>\n",
       "      <td>chr1</td>\n",
       "      <td>170318562</td>\n",
       "      <td>170319107</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.79579</td>\n",
       "      <td>16.95501</td>\n",
       "      <td>14.97065</td>\n",
       "      <td>187</td>\n",
       "      <td>chr1:170318562-170319107</td>\n",
       "      <td>ATAC-seq:chr1:170318562-170319107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202012</th>\n",
       "      <td>chr12</td>\n",
       "      <td>79451933</td>\n",
       "      <td>79452511</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.65382</td>\n",
       "      <td>16.95480</td>\n",
       "      <td>14.97045</td>\n",
       "      <td>464</td>\n",
       "      <td>chr12:79451933-79452511</td>\n",
       "      <td>ATAC-seq:chr12:79451933-79452511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202013</th>\n",
       "      <td>chr1</td>\n",
       "      <td>214524921</td>\n",
       "      <td>214525302</td>\n",
       "      <td>.</td>\n",
       "      <td>690</td>\n",
       "      <td>.</td>\n",
       "      <td>3.08018</td>\n",
       "      <td>16.95387</td>\n",
       "      <td>14.96954</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:214524921-214525302</td>\n",
       "      <td>ATAC-seq:chr1:214524921-214525302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202014</th>\n",
       "      <td>chr4</td>\n",
       "      <td>68465850</td>\n",
       "      <td>68467032</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.21736</td>\n",
       "      <td>16.95297</td>\n",
       "      <td>14.96865</td>\n",
       "      <td>220</td>\n",
       "      <td>chr4:68465850-68467032</td>\n",
       "      <td>ATAC-seq:chr4:68465850-68467032</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>202015 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0       chr17   22521173   22521671    .   1000      .     30.15064   \n",
       "1        chr1     633719     634669    .   1000      .      5.18915   \n",
       "2        chr1     632136     633008    .   1000      .      4.69070   \n",
       "3        chr1     629079     630024    .   1000      .      4.60007   \n",
       "4       chr17   22521173   22521671    .   1000      .     22.59163   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "202010   chr4  173243168  173243818    .   1000      .      3.79579   \n",
       "202011   chr1  170318562  170319107    .   1000      .      3.79579   \n",
       "202012  chr12   79451933   79452511    .   1000      .      3.65382   \n",
       "202013   chr1  214524921  214525302    .    690      .      3.08018   \n",
       "202014   chr4   68465850   68467032    .   1000      .      3.21736   \n",
       "\n",
       "            pValue      qValue  peak                     index  \\\n",
       "0       3810.73267  3801.84375   305   chr17:22521173-22521671   \n",
       "1       3772.65210  3764.00610   291        chr1:633719-634669   \n",
       "2       3083.66895  3076.10742   428        chr1:632136-633008   \n",
       "3       2907.49854  2900.17773   616        chr1:629079-630024   \n",
       "4       2536.15381  2529.12964   199   chr17:22521173-22521671   \n",
       "...            ...         ...   ...                       ...   \n",
       "202010    16.95501    14.97065    57  chr4:173243168-173243818   \n",
       "202011    16.95501    14.97065   187  chr1:170318562-170319107   \n",
       "202012    16.95480    14.97045   464   chr12:79451933-79452511   \n",
       "202013    16.95387    14.96954    49  chr1:214524921-214525302   \n",
       "202014    16.95297    14.96865   220    chr4:68465850-68467032   \n",
       "\n",
       "                                ATAC_hMSC  \n",
       "0        ATAC-seq:chr17:22521173-22521671  \n",
       "1             ATAC-seq:chr1:633719-634669  \n",
       "2             ATAC-seq:chr1:632136-633008  \n",
       "3             ATAC-seq:chr1:629079-630024  \n",
       "4        ATAC-seq:chr17:22521173-22521671  \n",
       "...                                   ...  \n",
       "202010  ATAC-seq:chr4:173243168-173243818  \n",
       "202011  ATAC-seq:chr1:170318562-170319107  \n",
       "202012   ATAC-seq:chr12:79451933-79452511  \n",
       "202013  ATAC-seq:chr1:214524921-214525302  \n",
       "202014    ATAC-seq:chr4:68465850-68467032  \n",
       "\n",
       "[202015 rows x 12 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "b487e865-8aa9-4ee8-845a-c13884bcbaa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_atac_uniq = hMSC_atac.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "36b1f4fa-0cfc-463c-baf0-fd21451af271",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521173</td>\n",
       "      <td>22521671</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>30.15064</td>\n",
       "      <td>3810.73267</td>\n",
       "      <td>3801.84375</td>\n",
       "      <td>305</td>\n",
       "      <td>chr17:22521173-22521671</td>\n",
       "      <td>ATAC-seq:chr17:22521173-22521671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633719</td>\n",
       "      <td>634669</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>5.18915</td>\n",
       "      <td>3772.65210</td>\n",
       "      <td>3764.00610</td>\n",
       "      <td>291</td>\n",
       "      <td>chr1:633719-634669</td>\n",
       "      <td>ATAC-seq:chr1:633719-634669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>632136</td>\n",
       "      <td>633008</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>4.69070</td>\n",
       "      <td>3083.66895</td>\n",
       "      <td>3076.10742</td>\n",
       "      <td>428</td>\n",
       "      <td>chr1:632136-633008</td>\n",
       "      <td>ATAC-seq:chr1:632136-633008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>4.60007</td>\n",
       "      <td>2907.49854</td>\n",
       "      <td>2900.17773</td>\n",
       "      <td>616</td>\n",
       "      <td>chr1:629079-630024</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr5</td>\n",
       "      <td>134926689</td>\n",
       "      <td>134927408</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>18.85654</td>\n",
       "      <td>2249.87158</td>\n",
       "      <td>2243.02393</td>\n",
       "      <td>180</td>\n",
       "      <td>chr5:134926689-134927408</td>\n",
       "      <td>ATAC-seq:chr5:134926689-134927408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200937</th>\n",
       "      <td>chr6</td>\n",
       "      <td>140344238</td>\n",
       "      <td>140344442</td>\n",
       "      <td>.</td>\n",
       "      <td>780</td>\n",
       "      <td>.</td>\n",
       "      <td>3.97126</td>\n",
       "      <td>17.30639</td>\n",
       "      <td>15.31421</td>\n",
       "      <td>157</td>\n",
       "      <td>chr6:140344238-140344442</td>\n",
       "      <td>ATAC-seq:chr6:140344238-140344442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200947</th>\n",
       "      <td>chr5</td>\n",
       "      <td>56615575</td>\n",
       "      <td>56615997</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.97126</td>\n",
       "      <td>17.30639</td>\n",
       "      <td>15.31421</td>\n",
       "      <td>330</td>\n",
       "      <td>chr5:56615575-56615997</td>\n",
       "      <td>ATAC-seq:chr5:56615575-56615997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201680</th>\n",
       "      <td>chr1</td>\n",
       "      <td>23531600</td>\n",
       "      <td>23531724</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.09943</td>\n",
       "      <td>17.10294</td>\n",
       "      <td>15.11652</td>\n",
       "      <td>62</td>\n",
       "      <td>chr1:23531600-23531724</td>\n",
       "      <td>ATAC-seq:chr1:23531600-23531724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201730</th>\n",
       "      <td>chr20</td>\n",
       "      <td>9500479</td>\n",
       "      <td>9500608</td>\n",
       "      <td>.</td>\n",
       "      <td>545</td>\n",
       "      <td>.</td>\n",
       "      <td>3.58944</td>\n",
       "      <td>17.08108</td>\n",
       "      <td>15.09497</td>\n",
       "      <td>64</td>\n",
       "      <td>chr20:9500479-9500608</td>\n",
       "      <td>ATAC-seq:chr20:9500479-9500608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201992</th>\n",
       "      <td>chr11</td>\n",
       "      <td>78417605</td>\n",
       "      <td>78417682</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>2.35561</td>\n",
       "      <td>16.96461</td>\n",
       "      <td>14.98013</td>\n",
       "      <td>35</td>\n",
       "      <td>chr11:78417605-78417682</td>\n",
       "      <td>ATAC-seq:chr11:78417605-78417682</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>81442 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0       chr17   22521173   22521671    .   1000      .     30.15064   \n",
       "1        chr1     633719     634669    .   1000      .      5.18915   \n",
       "2        chr1     632136     633008    .   1000      .      4.69070   \n",
       "3        chr1     629079     630024    .   1000      .      4.60007   \n",
       "8        chr5  134926689  134927408    .   1000      .     18.85654   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "200937   chr6  140344238  140344442    .    780      .      3.97126   \n",
       "200947   chr5   56615575   56615997    .   1000      .      3.97126   \n",
       "201680   chr1   23531600   23531724    .   1000      .      3.09943   \n",
       "201730  chr20    9500479    9500608    .    545      .      3.58944   \n",
       "201992  chr11   78417605   78417682    .   1000      .      2.35561   \n",
       "\n",
       "            pValue      qValue  peak                     index  \\\n",
       "0       3810.73267  3801.84375   305   chr17:22521173-22521671   \n",
       "1       3772.65210  3764.00610   291        chr1:633719-634669   \n",
       "2       3083.66895  3076.10742   428        chr1:632136-633008   \n",
       "3       2907.49854  2900.17773   616        chr1:629079-630024   \n",
       "8       2249.87158  2243.02393   180  chr5:134926689-134927408   \n",
       "...            ...         ...   ...                       ...   \n",
       "200937    17.30639    15.31421   157  chr6:140344238-140344442   \n",
       "200947    17.30639    15.31421   330    chr5:56615575-56615997   \n",
       "201680    17.10294    15.11652    62    chr1:23531600-23531724   \n",
       "201730    17.08108    15.09497    64     chr20:9500479-9500608   \n",
       "201992    16.96461    14.98013    35   chr11:78417605-78417682   \n",
       "\n",
       "                                ATAC_hMSC  \n",
       "0        ATAC-seq:chr17:22521173-22521671  \n",
       "1             ATAC-seq:chr1:633719-634669  \n",
       "2             ATAC-seq:chr1:632136-633008  \n",
       "3             ATAC-seq:chr1:629079-630024  \n",
       "8       ATAC-seq:chr5:134926689-134927408  \n",
       "...                                   ...  \n",
       "200937  ATAC-seq:chr6:140344238-140344442  \n",
       "200947    ATAC-seq:chr5:56615575-56615997  \n",
       "201680    ATAC-seq:chr1:23531600-23531724  \n",
       "201730     ATAC-seq:chr20:9500479-9500608  \n",
       "201992   ATAC-seq:chr11:78417605-78417682  \n",
       "\n",
       "[81442 rows x 12 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_atac_uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6537331d-3c8a-46c0-828f-d5cb5edcd06e",
   "metadata": {},
   "source": [
    "# hMSC DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "a4a0ed79-75f5-4e2f-90ea-783256335851",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_DNase = pd.read_csv('/mnt/10T-backup-mingju/ifar/DNase_seq/hMSC/ENCFF684GJJ_w_header.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "d1431f3c-7a81-4fda-a912-8329de7365b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.121317</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.090988</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.463211</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778660</td>\n",
       "      <td>778860</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>13.645400</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>779086</td>\n",
       "      <td>779280</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.193005</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26624627</td>\n",
       "      <td>26624680</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.052387</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314443</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26651123</td>\n",
       "      <td>26651270</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.077202</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663634</td>\n",
       "      <td>26663820</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.093745</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26669995</td>\n",
       "      <td>26670117</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.088231</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26670860</td>\n",
       "      <td>26671520</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.366709</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>314447 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0       chr1    181380    181580    .      0      .     0.121317      -1   \n",
       "1       chr1    191406    191561    .      0      .     0.090988      -1   \n",
       "2       chr1    268011    268060    .      0      .     0.463211      -1   \n",
       "3       chr1    778660    778860    .      0      .    13.645400      -1   \n",
       "4       chr1    779086    779280    .      0      .     0.193005      -1   \n",
       "...      ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "314442  chrY  26624627  26624680    .      0      .     0.052387      -1   \n",
       "314443  chrY  26651123  26651270    .      0      .     0.077202      -1   \n",
       "314444  chrY  26663634  26663820    .      0      .     0.093745      -1   \n",
       "314445  chrY  26669995  26670117    .      0      .     0.088231      -1   \n",
       "314446  chrY  26670860  26671520    .      0      .     0.366709      -1   \n",
       "\n",
       "        qValue  peak  \n",
       "0           -1    75  \n",
       "1           -1    75  \n",
       "2           -1    75  \n",
       "3           -1    75  \n",
       "4           -1    75  \n",
       "...        ...   ...  \n",
       "314442      -1    75  \n",
       "314443      -1    75  \n",
       "314444      -1    75  \n",
       "314445      -1    75  \n",
       "314446      -1    75  \n",
       "\n",
       "[314447 rows x 10 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "f7f2d258-eb5a-4f7b-a1fc-91482002c5a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_DNase['index'] = hMSC_DNase['Chr'] + \":\" + hMSC_DNase['Start'].astype(str) + \"-\" + hMSC_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "a99223c6-f504-42ce-830e-62fdac815009",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_DNase['DNase_hMSC'] = \"DNase-seq:\" + hMSC_DNase['Chr'] + \":\" + hMSC_DNase['Start'].astype(str) + \"-\" + hMSC_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "390bd32f-5930-44f0-9a61-c518c16855b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.121317</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181380-181580</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.090988</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:191406-191561</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.463211</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:268011-268060</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778660</td>\n",
       "      <td>778860</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>13.645400</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:778660-778860</td>\n",
       "      <td>DNase-seq:chr1:778660-778860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>779086</td>\n",
       "      <td>779280</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.193005</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:779086-779280</td>\n",
       "      <td>DNase-seq:chr1:779086-779280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26624627</td>\n",
       "      <td>26624680</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.052387</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26624627-26624680</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314443</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26651123</td>\n",
       "      <td>26651270</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.077202</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26651123-26651270</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663634</td>\n",
       "      <td>26663820</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.093745</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26663634-26663820</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26669995</td>\n",
       "      <td>26670117</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.088231</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26669995-26670117</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26670860</td>\n",
       "      <td>26671520</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.366709</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26670860-26671520</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>314447 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0       chr1    181380    181580    .      0      .     0.121317      -1   \n",
       "1       chr1    191406    191561    .      0      .     0.090988      -1   \n",
       "2       chr1    268011    268060    .      0      .     0.463211      -1   \n",
       "3       chr1    778660    778860    .      0      .    13.645400      -1   \n",
       "4       chr1    779086    779280    .      0      .     0.193005      -1   \n",
       "...      ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "314442  chrY  26624627  26624680    .      0      .     0.052387      -1   \n",
       "314443  chrY  26651123  26651270    .      0      .     0.077202      -1   \n",
       "314444  chrY  26663634  26663820    .      0      .     0.093745      -1   \n",
       "314445  chrY  26669995  26670117    .      0      .     0.088231      -1   \n",
       "314446  chrY  26670860  26671520    .      0      .     0.366709      -1   \n",
       "\n",
       "        qValue  peak                   index                        DNase_hMSC  \n",
       "0           -1    75      chr1:181380-181580      DNase-seq:chr1:181380-181580  \n",
       "1           -1    75      chr1:191406-191561      DNase-seq:chr1:191406-191561  \n",
       "2           -1    75      chr1:268011-268060      DNase-seq:chr1:268011-268060  \n",
       "3           -1    75      chr1:778660-778860      DNase-seq:chr1:778660-778860  \n",
       "4           -1    75      chr1:779086-779280      DNase-seq:chr1:779086-779280  \n",
       "...        ...   ...                     ...                               ...  \n",
       "314442      -1    75  chrY:26624627-26624680  DNase-seq:chrY:26624627-26624680  \n",
       "314443      -1    75  chrY:26651123-26651270  DNase-seq:chrY:26651123-26651270  \n",
       "314444      -1    75  chrY:26663634-26663820  DNase-seq:chrY:26663634-26663820  \n",
       "314445      -1    75  chrY:26669995-26670117  DNase-seq:chrY:26669995-26670117  \n",
       "314446      -1    75  chrY:26670860-26671520  DNase-seq:chrY:26670860-26671520  \n",
       "\n",
       "[314447 rows x 12 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "118add04-3a86-4a9c-a938-e4e1c1923012",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_DNase_uniq = hMSC_DNase.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "bcd28bc9-2d1d-4792-9c5c-ce380db5e243",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.121317</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181380-181580</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.090988</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:191406-191561</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.463211</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:268011-268060</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778660</td>\n",
       "      <td>778860</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>13.645400</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:778660-778860</td>\n",
       "      <td>DNase-seq:chr1:778660-778860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>779086</td>\n",
       "      <td>779280</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.193005</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:779086-779280</td>\n",
       "      <td>DNase-seq:chr1:779086-779280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314442</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26624627</td>\n",
       "      <td>26624680</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.052387</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26624627-26624680</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314443</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26651123</td>\n",
       "      <td>26651270</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.077202</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26651123-26651270</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314444</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663634</td>\n",
       "      <td>26663820</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.093745</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26663634-26663820</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314445</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26669995</td>\n",
       "      <td>26670117</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.088231</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26669995-26670117</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>314446</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26670860</td>\n",
       "      <td>26671520</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.366709</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26670860-26671520</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>314447 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0       chr1    181380    181580    .      0      .     0.121317      -1   \n",
       "1       chr1    191406    191561    .      0      .     0.090988      -1   \n",
       "2       chr1    268011    268060    .      0      .     0.463211      -1   \n",
       "3       chr1    778660    778860    .      0      .    13.645400      -1   \n",
       "4       chr1    779086    779280    .      0      .     0.193005      -1   \n",
       "...      ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "314442  chrY  26624627  26624680    .      0      .     0.052387      -1   \n",
       "314443  chrY  26651123  26651270    .      0      .     0.077202      -1   \n",
       "314444  chrY  26663634  26663820    .      0      .     0.093745      -1   \n",
       "314445  chrY  26669995  26670117    .      0      .     0.088231      -1   \n",
       "314446  chrY  26670860  26671520    .      0      .     0.366709      -1   \n",
       "\n",
       "        qValue  peak                   index                        DNase_hMSC  \n",
       "0           -1    75      chr1:181380-181580      DNase-seq:chr1:181380-181580  \n",
       "1           -1    75      chr1:191406-191561      DNase-seq:chr1:191406-191561  \n",
       "2           -1    75      chr1:268011-268060      DNase-seq:chr1:268011-268060  \n",
       "3           -1    75      chr1:778660-778860      DNase-seq:chr1:778660-778860  \n",
       "4           -1    75      chr1:779086-779280      DNase-seq:chr1:779086-779280  \n",
       "...        ...   ...                     ...                               ...  \n",
       "314442      -1    75  chrY:26624627-26624680  DNase-seq:chrY:26624627-26624680  \n",
       "314443      -1    75  chrY:26651123-26651270  DNase-seq:chrY:26651123-26651270  \n",
       "314444      -1    75  chrY:26663634-26663820  DNase-seq:chrY:26663634-26663820  \n",
       "314445      -1    75  chrY:26669995-26670117  DNase-seq:chrY:26669995-26670117  \n",
       "314446      -1    75  chrY:26670860-26671520  DNase-seq:chrY:26670860-26671520  \n",
       "\n",
       "[314447 rows x 12 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_DNase_uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "38dd58db-1c4f-4810-8458-037b15ea7ba7",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase = pd.merge(hMSC_atac_uniq[['index','Chr','Start','End','ATAC_hMSC']], hMSC_DNase_uniq[['index','DNase_hMSC']], on='index', how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "eed0c8d6-2f54-4b61-be73-65299bc61f30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100000280-100000580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100000280-100000580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100004493-100004543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100004493-100004543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100005020-100005460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100005020-100005460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100006274-100006634</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100006274.0</td>\n",
       "      <td>100006634.0</td>\n",
       "      <td>ATAC-seq:chr10:100006274-100006634</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100006280-100006560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006280-100006560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395883</th>\n",
       "      <td>chrY:9122460-9122526</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9122460-9122526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395884</th>\n",
       "      <td>chrY:9253660-9253820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9253660-9253820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395885</th>\n",
       "      <td>chrY:9297268-9297360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9297268-9297360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395886</th>\n",
       "      <td>chrY:9589288-9589440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9589288-9589440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395887</th>\n",
       "      <td>chrY:9922788-9922925</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9922788-9922925</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>395888 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index    Chr        Start          End  \\\n",
       "0       chr10:100000280-100000580    NaN          NaN          NaN   \n",
       "1       chr10:100004493-100004543    NaN          NaN          NaN   \n",
       "2       chr10:100005020-100005460    NaN          NaN          NaN   \n",
       "3       chr10:100006274-100006634  chr10  100006274.0  100006634.0   \n",
       "4       chr10:100006280-100006560    NaN          NaN          NaN   \n",
       "...                           ...    ...          ...          ...   \n",
       "395883       chrY:9122460-9122526    NaN          NaN          NaN   \n",
       "395884       chrY:9253660-9253820    NaN          NaN          NaN   \n",
       "395885       chrY:9297268-9297360    NaN          NaN          NaN   \n",
       "395886       chrY:9589288-9589440    NaN          NaN          NaN   \n",
       "395887       chrY:9922788-9922925    NaN          NaN          NaN   \n",
       "\n",
       "                                 ATAC_hMSC  \\\n",
       "0                                      NaN   \n",
       "1                                      NaN   \n",
       "2                                      NaN   \n",
       "3       ATAC-seq:chr10:100006274-100006634   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "395883                                 NaN   \n",
       "395884                                 NaN   \n",
       "395885                                 NaN   \n",
       "395886                                 NaN   \n",
       "395887                                 NaN   \n",
       "\n",
       "                                 DNase_hMSC  \n",
       "0       DNase-seq:chr10:100000280-100000580  \n",
       "1       DNase-seq:chr10:100004493-100004543  \n",
       "2       DNase-seq:chr10:100005020-100005460  \n",
       "3                                       NaN  \n",
       "4       DNase-seq:chr10:100006280-100006560  \n",
       "...                                     ...  \n",
       "395883       DNase-seq:chrY:9122460-9122526  \n",
       "395884       DNase-seq:chrY:9253660-9253820  \n",
       "395885       DNase-seq:chrY:9297268-9297360  \n",
       "395886       DNase-seq:chrY:9589288-9589440  \n",
       "395887       DNase-seq:chrY:9922788-9922925  \n",
       "\n",
       "[395888 rows x 6 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "a9c9b6be-5b84-4c50-9956-fc78965b93ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[\"OpenChromatin_hMSC\"] = merge_atac_dnase.apply(\n",
    "    lambda row: f\"{row['ATAC_hMSC']};{row['DNase_hMSC']}\" if pd.notna(row['ATAC_hMSC']) and pd.notna(row['DNase_hMSC']) and row['DNase_hMSC'] != \"\"\n",
    "                else row['ATAC_hMSC'] if pd.notna(row['ATAC_hMSC'])\n",
    "                else row['DNase_hMSC'] if pd.notna(row['DNase_hMSC']) and row['DNase_hMSC'] != \"\"\n",
    "                else \"\", \n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "f664675c-83a1-4733-b731-8cd4733396a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100000280-100000580</td>\n",
       "      <td>10</td>\n",
       "      <td>100000280</td>\n",
       "      <td>100000580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100000280-100000580</td>\n",
       "      <td>DNase-seq:chr10:100000280-100000580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100004493-100004543</td>\n",
       "      <td>10</td>\n",
       "      <td>100004493</td>\n",
       "      <td>100004543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100004493-100004543</td>\n",
       "      <td>DNase-seq:chr10:100004493-100004543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100005020-100005460</td>\n",
       "      <td>10</td>\n",
       "      <td>100005020</td>\n",
       "      <td>100005460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100005020-100005460</td>\n",
       "      <td>DNase-seq:chr10:100005020-100005460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100006274-100006634</td>\n",
       "      <td>10</td>\n",
       "      <td>100006274</td>\n",
       "      <td>100006634</td>\n",
       "      <td>ATAC-seq:chr10:100006274-100006634</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100006274-100006634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100006280-100006560</td>\n",
       "      <td>10</td>\n",
       "      <td>100006280</td>\n",
       "      <td>100006560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006280-100006560</td>\n",
       "      <td>DNase-seq:chr10:100006280-100006560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395883</th>\n",
       "      <td>chrY:9122460-9122526</td>\n",
       "      <td>Y</td>\n",
       "      <td>9122460</td>\n",
       "      <td>9122526</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9122460-9122526</td>\n",
       "      <td>DNase-seq:chrY:9122460-9122526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395884</th>\n",
       "      <td>chrY:9253660-9253820</td>\n",
       "      <td>Y</td>\n",
       "      <td>9253660</td>\n",
       "      <td>9253820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9253660-9253820</td>\n",
       "      <td>DNase-seq:chrY:9253660-9253820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395885</th>\n",
       "      <td>chrY:9297268-9297360</td>\n",
       "      <td>Y</td>\n",
       "      <td>9297268</td>\n",
       "      <td>9297360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9297268-9297360</td>\n",
       "      <td>DNase-seq:chrY:9297268-9297360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395886</th>\n",
       "      <td>chrY:9589288-9589440</td>\n",
       "      <td>Y</td>\n",
       "      <td>9589288</td>\n",
       "      <td>9589440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9589288-9589440</td>\n",
       "      <td>DNase-seq:chrY:9589288-9589440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395887</th>\n",
       "      <td>chrY:9922788-9922925</td>\n",
       "      <td>Y</td>\n",
       "      <td>9922788</td>\n",
       "      <td>9922925</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9922788-9922925</td>\n",
       "      <td>DNase-seq:chrY:9922788-9922925</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>395888 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index Chr      Start        End  \\\n",
       "0       chr10:100000280-100000580  10  100000280  100000580   \n",
       "1       chr10:100004493-100004543  10  100004493  100004543   \n",
       "2       chr10:100005020-100005460  10  100005020  100005460   \n",
       "3       chr10:100006274-100006634  10  100006274  100006634   \n",
       "4       chr10:100006280-100006560  10  100006280  100006560   \n",
       "...                           ...  ..        ...        ...   \n",
       "395883       chrY:9122460-9122526   Y    9122460    9122526   \n",
       "395884       chrY:9253660-9253820   Y    9253660    9253820   \n",
       "395885       chrY:9297268-9297360   Y    9297268    9297360   \n",
       "395886       chrY:9589288-9589440   Y    9589288    9589440   \n",
       "395887       chrY:9922788-9922925   Y    9922788    9922925   \n",
       "\n",
       "                                 ATAC_hMSC  \\\n",
       "0                                      NaN   \n",
       "1                                      NaN   \n",
       "2                                      NaN   \n",
       "3       ATAC-seq:chr10:100006274-100006634   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "395883                                 NaN   \n",
       "395884                                 NaN   \n",
       "395885                                 NaN   \n",
       "395886                                 NaN   \n",
       "395887                                 NaN   \n",
       "\n",
       "                                 DNase_hMSC  \\\n",
       "0       DNase-seq:chr10:100000280-100000580   \n",
       "1       DNase-seq:chr10:100004493-100004543   \n",
       "2       DNase-seq:chr10:100005020-100005460   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100006280-100006560   \n",
       "...                                     ...   \n",
       "395883       DNase-seq:chrY:9122460-9122526   \n",
       "395884       DNase-seq:chrY:9253660-9253820   \n",
       "395885       DNase-seq:chrY:9297268-9297360   \n",
       "395886       DNase-seq:chrY:9589288-9589440   \n",
       "395887       DNase-seq:chrY:9922788-9922925   \n",
       "\n",
       "                         OpenChromatin_hMSC  \n",
       "0       DNase-seq:chr10:100000280-100000580  \n",
       "1       DNase-seq:chr10:100004493-100004543  \n",
       "2       DNase-seq:chr10:100005020-100005460  \n",
       "3        ATAC-seq:chr10:100006274-100006634  \n",
       "4       DNase-seq:chr10:100006280-100006560  \n",
       "...                                     ...  \n",
       "395883       DNase-seq:chrY:9122460-9122526  \n",
       "395884       DNase-seq:chrY:9253660-9253820  \n",
       "395885       DNase-seq:chrY:9297268-9297360  \n",
       "395886       DNase-seq:chrY:9589288-9589440  \n",
       "395887       DNase-seq:chrY:9922788-9922925  \n",
       "\n",
       "[395888 rows x 7 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "a243595e-64d5-41d6-986d-716aa8b71758",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_index(index_str):\n",
    "    if pd.isna(index_str) or index_str == \"NaN\":\n",
    "        return pd.NA, pd.NA, pd.NA\n",
    "    chr_part, range_part = index_str.split(\":\")\n",
    "    chr_value = chr_part.replace(\"chr\", \"\")\n",
    "    start, end = range_part.split(\"-\")\n",
    "    return chr_value, pd.to_numeric(start), pd.to_numeric(end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "9367a17f-4410-4afa-9460-3a9d80d175ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[['Chr', 'Start', 'End']] = merge_atac_dnase['index'].apply(lambda x: pd.Series(parse_index(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "ae373cf5-1cd5-4c34-a158-988fcc95b7af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100000280-100000580</td>\n",
       "      <td>10</td>\n",
       "      <td>100000280</td>\n",
       "      <td>100000580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100000280-100000580</td>\n",
       "      <td>DNase-seq:chr10:100000280-100000580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100004493-100004543</td>\n",
       "      <td>10</td>\n",
       "      <td>100004493</td>\n",
       "      <td>100004543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100004493-100004543</td>\n",
       "      <td>DNase-seq:chr10:100004493-100004543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100005020-100005460</td>\n",
       "      <td>10</td>\n",
       "      <td>100005020</td>\n",
       "      <td>100005460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100005020-100005460</td>\n",
       "      <td>DNase-seq:chr10:100005020-100005460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100006274-100006634</td>\n",
       "      <td>10</td>\n",
       "      <td>100006274</td>\n",
       "      <td>100006634</td>\n",
       "      <td>ATAC-seq:chr10:100006274-100006634</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100006274-100006634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100006280-100006560</td>\n",
       "      <td>10</td>\n",
       "      <td>100006280</td>\n",
       "      <td>100006560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006280-100006560</td>\n",
       "      <td>DNase-seq:chr10:100006280-100006560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395883</th>\n",
       "      <td>chrY:9122460-9122526</td>\n",
       "      <td>Y</td>\n",
       "      <td>9122460</td>\n",
       "      <td>9122526</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9122460-9122526</td>\n",
       "      <td>DNase-seq:chrY:9122460-9122526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395884</th>\n",
       "      <td>chrY:9253660-9253820</td>\n",
       "      <td>Y</td>\n",
       "      <td>9253660</td>\n",
       "      <td>9253820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9253660-9253820</td>\n",
       "      <td>DNase-seq:chrY:9253660-9253820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395885</th>\n",
       "      <td>chrY:9297268-9297360</td>\n",
       "      <td>Y</td>\n",
       "      <td>9297268</td>\n",
       "      <td>9297360</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9297268-9297360</td>\n",
       "      <td>DNase-seq:chrY:9297268-9297360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395886</th>\n",
       "      <td>chrY:9589288-9589440</td>\n",
       "      <td>Y</td>\n",
       "      <td>9589288</td>\n",
       "      <td>9589440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9589288-9589440</td>\n",
       "      <td>DNase-seq:chrY:9589288-9589440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395887</th>\n",
       "      <td>chrY:9922788-9922925</td>\n",
       "      <td>Y</td>\n",
       "      <td>9922788</td>\n",
       "      <td>9922925</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9922788-9922925</td>\n",
       "      <td>DNase-seq:chrY:9922788-9922925</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>395888 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index Chr      Start        End  \\\n",
       "0       chr10:100000280-100000580  10  100000280  100000580   \n",
       "1       chr10:100004493-100004543  10  100004493  100004543   \n",
       "2       chr10:100005020-100005460  10  100005020  100005460   \n",
       "3       chr10:100006274-100006634  10  100006274  100006634   \n",
       "4       chr10:100006280-100006560  10  100006280  100006560   \n",
       "...                           ...  ..        ...        ...   \n",
       "395883       chrY:9122460-9122526   Y    9122460    9122526   \n",
       "395884       chrY:9253660-9253820   Y    9253660    9253820   \n",
       "395885       chrY:9297268-9297360   Y    9297268    9297360   \n",
       "395886       chrY:9589288-9589440   Y    9589288    9589440   \n",
       "395887       chrY:9922788-9922925   Y    9922788    9922925   \n",
       "\n",
       "                                 ATAC_hMSC  \\\n",
       "0                                      NaN   \n",
       "1                                      NaN   \n",
       "2                                      NaN   \n",
       "3       ATAC-seq:chr10:100006274-100006634   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "395883                                 NaN   \n",
       "395884                                 NaN   \n",
       "395885                                 NaN   \n",
       "395886                                 NaN   \n",
       "395887                                 NaN   \n",
       "\n",
       "                                 DNase_hMSC  \\\n",
       "0       DNase-seq:chr10:100000280-100000580   \n",
       "1       DNase-seq:chr10:100004493-100004543   \n",
       "2       DNase-seq:chr10:100005020-100005460   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100006280-100006560   \n",
       "...                                     ...   \n",
       "395883       DNase-seq:chrY:9122460-9122526   \n",
       "395884       DNase-seq:chrY:9253660-9253820   \n",
       "395885       DNase-seq:chrY:9297268-9297360   \n",
       "395886       DNase-seq:chrY:9589288-9589440   \n",
       "395887       DNase-seq:chrY:9922788-9922925   \n",
       "\n",
       "                         OpenChromatin_hMSC  \n",
       "0       DNase-seq:chr10:100000280-100000580  \n",
       "1       DNase-seq:chr10:100004493-100004543  \n",
       "2       DNase-seq:chr10:100005020-100005460  \n",
       "3        ATAC-seq:chr10:100006274-100006634  \n",
       "4       DNase-seq:chr10:100006280-100006560  \n",
       "...                                     ...  \n",
       "395883       DNase-seq:chrY:9122460-9122526  \n",
       "395884       DNase-seq:chrY:9253660-9253820  \n",
       "395885       DNase-seq:chrY:9297268-9297360  \n",
       "395886       DNase-seq:chrY:9589288-9589440  \n",
       "395887       DNase-seq:chrY:9922788-9922925  \n",
       "\n",
       "[395888 rows x 7 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "9a58dab4-e981-4bd8-8f9d-7cfc07d5a5cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the chromosome order: 1 to 22, X, Y\n",
    "chr_order = [str(i) for i in range(1, 23)] + [\"X\", \"Y\"]\n",
    "\n",
    "# Convert Chr to categorical with the specified order\n",
    "merge_atac_dnase[\"Chr\"] = pd.Categorical(merge_atac_dnase[\"Chr\"], categories=chr_order, ordered=True)\n",
    "\n",
    "# Sort by Chr and Start\n",
    "merge_atac_dnase_sorted = merge_atac_dnase.sort_values(by=[\"Chr\", \"Start\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "ea650c3c-46eb-401b-8b54-f7d0d60cedec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:181380-181580</td>\n",
       "      <td>1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:191406-191561</td>\n",
       "      <td>1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:268011-268060</td>\n",
       "      <td>1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:629079-630024</td>\n",
       "      <td>1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:630219-630484</td>\n",
       "      <td>1</td>\n",
       "      <td>630219</td>\n",
       "      <td>630484</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395883</th>\n",
       "      <td>chrY:26624627-26624680</td>\n",
       "      <td>Y</td>\n",
       "      <td>26624627</td>\n",
       "      <td>26624680</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395884</th>\n",
       "      <td>chrY:26651123-26651270</td>\n",
       "      <td>Y</td>\n",
       "      <td>26651123</td>\n",
       "      <td>26651270</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395885</th>\n",
       "      <td>chrY:26663634-26663820</td>\n",
       "      <td>Y</td>\n",
       "      <td>26663634</td>\n",
       "      <td>26663820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395886</th>\n",
       "      <td>chrY:26669995-26670117</td>\n",
       "      <td>Y</td>\n",
       "      <td>26669995</td>\n",
       "      <td>26670117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395887</th>\n",
       "      <td>chrY:26670860-26671520</td>\n",
       "      <td>Y</td>\n",
       "      <td>26670860</td>\n",
       "      <td>26671520</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>395888 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End  \\\n",
       "0           chr1:181380-181580   1    181380    181580   \n",
       "1           chr1:191406-191561   1    191406    191561   \n",
       "2           chr1:268011-268060   1    268011    268060   \n",
       "3           chr1:629079-630024   1    629079    630024   \n",
       "4           chr1:630219-630484   1    630219    630484   \n",
       "...                        ...  ..       ...       ...   \n",
       "395883  chrY:26624627-26624680   Y  26624627  26624680   \n",
       "395884  chrY:26651123-26651270   Y  26651123  26651270   \n",
       "395885  chrY:26663634-26663820   Y  26663634  26663820   \n",
       "395886  chrY:26669995-26670117   Y  26669995  26670117   \n",
       "395887  chrY:26670860-26671520   Y  26670860  26671520   \n",
       "\n",
       "                          ATAC_hMSC                        DNase_hMSC  \\\n",
       "0                               NaN      DNase-seq:chr1:181380-181580   \n",
       "1                               NaN      DNase-seq:chr1:191406-191561   \n",
       "2                               NaN      DNase-seq:chr1:268011-268060   \n",
       "3       ATAC-seq:chr1:629079-630024                               NaN   \n",
       "4       ATAC-seq:chr1:630219-630484                               NaN   \n",
       "...                             ...                               ...   \n",
       "395883                          NaN  DNase-seq:chrY:26624627-26624680   \n",
       "395884                          NaN  DNase-seq:chrY:26651123-26651270   \n",
       "395885                          NaN  DNase-seq:chrY:26663634-26663820   \n",
       "395886                          NaN  DNase-seq:chrY:26669995-26670117   \n",
       "395887                          NaN  DNase-seq:chrY:26670860-26671520   \n",
       "\n",
       "                      OpenChromatin_hMSC  \n",
       "0           DNase-seq:chr1:181380-181580  \n",
       "1           DNase-seq:chr1:191406-191561  \n",
       "2           DNase-seq:chr1:268011-268060  \n",
       "3            ATAC-seq:chr1:629079-630024  \n",
       "4            ATAC-seq:chr1:630219-630484  \n",
       "...                                  ...  \n",
       "395883  DNase-seq:chrY:26624627-26624680  \n",
       "395884  DNase-seq:chrY:26651123-26651270  \n",
       "395885  DNase-seq:chrY:26663634-26663820  \n",
       "395886  DNase-seq:chrY:26669995-26670117  \n",
       "395887  DNase-seq:chrY:26670860-26671520  \n",
       "\n",
       "[395888 rows x 7 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "5743febe-5606-483e-a92f-025632971f88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>150987</th>\n",
       "      <td>chr1:181380-181580</td>\n",
       "      <td>1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152254</th>\n",
       "      <td>chr1:191406-191561</td>\n",
       "      <td>1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163325</th>\n",
       "      <td>chr1:268011-268060</td>\n",
       "      <td>1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169708</th>\n",
       "      <td>chr1:629079-630024</td>\n",
       "      <td>1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169713</th>\n",
       "      <td>chr1:630219-630484</td>\n",
       "      <td>1</td>\n",
       "      <td>630219</td>\n",
       "      <td>630484</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175999</th>\n",
       "      <td>chr1:999886-1000314</td>\n",
       "      <td>1</td>\n",
       "      <td>999886</td>\n",
       "      <td>1000314</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140679</th>\n",
       "      <td>chr1:1000060-1000260</td>\n",
       "      <td>1</td>\n",
       "      <td>1000060</td>\n",
       "      <td>1000260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1000060-1000260</td>\n",
       "      <td>DNase-seq:chr1:1000060-1000260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140686</th>\n",
       "      <td>chr1:1000382-1000639</td>\n",
       "      <td>1</td>\n",
       "      <td>1000382</td>\n",
       "      <td>1000639</td>\n",
       "      <td>ATAC-seq:chr1:1000382-1000639</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1000382-1000639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140690</th>\n",
       "      <td>chr1:1000400-1000640</td>\n",
       "      <td>1</td>\n",
       "      <td>1000400</td>\n",
       "      <td>1000640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1000400-1000640</td>\n",
       "      <td>DNase-seq:chr1:1000400-1000640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140694</th>\n",
       "      <td>chr1:1000720-1001000</td>\n",
       "      <td>1</td>\n",
       "      <td>1000720</td>\n",
       "      <td>1001000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       index Chr    Start      End  \\\n",
       "150987    chr1:181380-181580   1   181380   181580   \n",
       "152254    chr1:191406-191561   1   191406   191561   \n",
       "163325    chr1:268011-268060   1   268011   268060   \n",
       "169708    chr1:629079-630024   1   629079   630024   \n",
       "169713    chr1:630219-630484   1   630219   630484   \n",
       "...                      ...  ..      ...      ...   \n",
       "175999   chr1:999886-1000314   1   999886  1000314   \n",
       "140679  chr1:1000060-1000260   1  1000060  1000260   \n",
       "140686  chr1:1000382-1000639   1  1000382  1000639   \n",
       "140690  chr1:1000400-1000640   1  1000400  1000640   \n",
       "140694  chr1:1000720-1001000   1  1000720  1001000   \n",
       "\n",
       "                            ATAC_hMSC                      DNase_hMSC  \\\n",
       "150987                            NaN    DNase-seq:chr1:181380-181580   \n",
       "152254                            NaN    DNase-seq:chr1:191406-191561   \n",
       "163325                            NaN    DNase-seq:chr1:268011-268060   \n",
       "169708    ATAC-seq:chr1:629079-630024                             NaN   \n",
       "169713    ATAC-seq:chr1:630219-630484                             NaN   \n",
       "...                               ...                             ...   \n",
       "175999   ATAC-seq:chr1:999886-1000314                             NaN   \n",
       "140679                            NaN  DNase-seq:chr1:1000060-1000260   \n",
       "140686  ATAC-seq:chr1:1000382-1000639                             NaN   \n",
       "140690                            NaN  DNase-seq:chr1:1000400-1000640   \n",
       "140694                            NaN  DNase-seq:chr1:1000720-1001000   \n",
       "\n",
       "                    OpenChromatin_hMSC  \n",
       "150987    DNase-seq:chr1:181380-181580  \n",
       "152254    DNase-seq:chr1:191406-191561  \n",
       "163325    DNase-seq:chr1:268011-268060  \n",
       "169708     ATAC-seq:chr1:629079-630024  \n",
       "169713     ATAC-seq:chr1:630219-630484  \n",
       "...                                ...  \n",
       "175999    ATAC-seq:chr1:999886-1000314  \n",
       "140679  DNase-seq:chr1:1000060-1000260  \n",
       "140686   ATAC-seq:chr1:1000382-1000639  \n",
       "140690  DNase-seq:chr1:1000400-1000640  \n",
       "140694  DNase-seq:chr1:1000720-1001000  \n",
       "\n",
       "[100 rows x 7 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.head(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "2f08790f-b04c-4444-ad34-a91ca893ace2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>49178</th>\n",
       "      <td>chr12:48853300-48853520</td>\n",
       "      <td>12</td>\n",
       "      <td>48853300</td>\n",
       "      <td>48853520</td>\n",
       "      <td>ATAC-seq:chr12:48853300-48853520</td>\n",
       "      <td>DNase-seq:chr12:48853300-48853520</td>\n",
       "      <td>ATAC-seq:chr12:48853300-48853520;DNase-seq:chr...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End  \\\n",
       "49178  chr12:48853300-48853520  12  48853300  48853520   \n",
       "\n",
       "                              ATAC_hMSC                         DNase_hMSC  \\\n",
       "49178  ATAC-seq:chr12:48853300-48853520  DNase-seq:chr12:48853300-48853520   \n",
       "\n",
       "                                      OpenChromatin_hMSC  \n",
       "49178  ATAC-seq:chr12:48853300-48853520;DNase-seq:chr...  "
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted[merge_atac_dnase_sorted['ATAC_hMSC'].notna()&merge_atac_dnase_sorted['DNase_hMSC'].notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "6e472aa7-b594-416b-9ab3-c8ddad47c265",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_hMSC</th>\n",
       "      <th>DNase_hMSC</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>150987</th>\n",
       "      <td>chr1:181380-181580</td>\n",
       "      <td>1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152254</th>\n",
       "      <td>chr1:191406-191561</td>\n",
       "      <td>1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163325</th>\n",
       "      <td>chr1:268011-268060</td>\n",
       "      <td>1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169708</th>\n",
       "      <td>chr1:629079-630024</td>\n",
       "      <td>1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169713</th>\n",
       "      <td>chr1:630219-630484</td>\n",
       "      <td>1</td>\n",
       "      <td>630219</td>\n",
       "      <td>630484</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395717</th>\n",
       "      <td>chrY:26624627-26624680</td>\n",
       "      <td>Y</td>\n",
       "      <td>26624627</td>\n",
       "      <td>26624680</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395718</th>\n",
       "      <td>chrY:26651123-26651270</td>\n",
       "      <td>Y</td>\n",
       "      <td>26651123</td>\n",
       "      <td>26651270</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395719</th>\n",
       "      <td>chrY:26663634-26663820</td>\n",
       "      <td>Y</td>\n",
       "      <td>26663634</td>\n",
       "      <td>26663820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395720</th>\n",
       "      <td>chrY:26669995-26670117</td>\n",
       "      <td>Y</td>\n",
       "      <td>26669995</td>\n",
       "      <td>26670117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395721</th>\n",
       "      <td>chrY:26670860-26671520</td>\n",
       "      <td>Y</td>\n",
       "      <td>26670860</td>\n",
       "      <td>26671520</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>395888 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End  \\\n",
       "150987      chr1:181380-181580   1    181380    181580   \n",
       "152254      chr1:191406-191561   1    191406    191561   \n",
       "163325      chr1:268011-268060   1    268011    268060   \n",
       "169708      chr1:629079-630024   1    629079    630024   \n",
       "169713      chr1:630219-630484   1    630219    630484   \n",
       "...                        ...  ..       ...       ...   \n",
       "395717  chrY:26624627-26624680   Y  26624627  26624680   \n",
       "395718  chrY:26651123-26651270   Y  26651123  26651270   \n",
       "395719  chrY:26663634-26663820   Y  26663634  26663820   \n",
       "395720  chrY:26669995-26670117   Y  26669995  26670117   \n",
       "395721  chrY:26670860-26671520   Y  26670860  26671520   \n",
       "\n",
       "                          ATAC_hMSC                        DNase_hMSC  \\\n",
       "150987                          NaN      DNase-seq:chr1:181380-181580   \n",
       "152254                          NaN      DNase-seq:chr1:191406-191561   \n",
       "163325                          NaN      DNase-seq:chr1:268011-268060   \n",
       "169708  ATAC-seq:chr1:629079-630024                               NaN   \n",
       "169713  ATAC-seq:chr1:630219-630484                               NaN   \n",
       "...                             ...                               ...   \n",
       "395717                          NaN  DNase-seq:chrY:26624627-26624680   \n",
       "395718                          NaN  DNase-seq:chrY:26651123-26651270   \n",
       "395719                          NaN  DNase-seq:chrY:26663634-26663820   \n",
       "395720                          NaN  DNase-seq:chrY:26669995-26670117   \n",
       "395721                          NaN  DNase-seq:chrY:26670860-26671520   \n",
       "\n",
       "                      OpenChromatin_hMSC  \n",
       "150987      DNase-seq:chr1:181380-181580  \n",
       "152254      DNase-seq:chr1:191406-191561  \n",
       "163325      DNase-seq:chr1:268011-268060  \n",
       "169708       ATAC-seq:chr1:629079-630024  \n",
       "169713       ATAC-seq:chr1:630219-630484  \n",
       "...                                  ...  \n",
       "395717  DNase-seq:chrY:26624627-26624680  \n",
       "395718  DNase-seq:chrY:26651123-26651270  \n",
       "395719  DNase-seq:chrY:26663634-26663820  \n",
       "395720  DNase-seq:chrY:26669995-26670117  \n",
       "395721  DNase-seq:chrY:26670860-26671520  \n",
       "\n",
       "[395888 rows x 7 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "70d2b8f7-5097-4a33-a80d-97ccf4b65369",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_merge_atac_dnase_sorted_bed = merge_atac_dnase_sorted[[\"Chr\",\"Start\",\"End\",\"OpenChromatin_hMSC\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "cb43fd4d-d539-48af-9397-de97ee407b8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>150987</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152254</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163325</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169708</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169713</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630219</td>\n",
       "      <td>630484</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395717</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26624627</td>\n",
       "      <td>26624680</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395718</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26651123</td>\n",
       "      <td>26651270</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395719</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663634</td>\n",
       "      <td>26663820</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395720</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26669995</td>\n",
       "      <td>26670117</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395721</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26670860</td>\n",
       "      <td>26671520</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>395888 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End                OpenChromatin_hMSC\n",
       "150987  chr1    181380    181580      DNase-seq:chr1:181380-181580\n",
       "152254  chr1    191406    191561      DNase-seq:chr1:191406-191561\n",
       "163325  chr1    268011    268060      DNase-seq:chr1:268011-268060\n",
       "169708  chr1    629079    630024       ATAC-seq:chr1:629079-630024\n",
       "169713  chr1    630219    630484       ATAC-seq:chr1:630219-630484\n",
       "...      ...       ...       ...                               ...\n",
       "395717  chrY  26624627  26624680  DNase-seq:chrY:26624627-26624680\n",
       "395718  chrY  26651123  26651270  DNase-seq:chrY:26651123-26651270\n",
       "395719  chrY  26663634  26663820  DNase-seq:chrY:26663634-26663820\n",
       "395720  chrY  26669995  26670117  DNase-seq:chrY:26669995-26670117\n",
       "395721  chrY  26670860  26671520  DNase-seq:chrY:26670860-26671520\n",
       "\n",
       "[395888 rows x 4 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "298a4b03-0774-44a3-9c4c-2d0a5e04a25c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2558907/2250446228.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  hMSC_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + hMSC_merge_atac_dnase_sorted_bed['Chr'].astype(str)\n"
     ]
    }
   ],
   "source": [
    "hMSC_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + hMSC_merge_atac_dnase_sorted_bed['Chr'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "1ca0b26a-0d6c-473c-8b62-56054a166587",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>150987</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181380</td>\n",
       "      <td>181580</td>\n",
       "      <td>DNase-seq:chr1:181380-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152254</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191406</td>\n",
       "      <td>191561</td>\n",
       "      <td>DNase-seq:chr1:191406-191561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163325</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268060</td>\n",
       "      <td>DNase-seq:chr1:268011-268060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169708</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629079</td>\n",
       "      <td>630024</td>\n",
       "      <td>ATAC-seq:chr1:629079-630024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169713</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630219</td>\n",
       "      <td>630484</td>\n",
       "      <td>ATAC-seq:chr1:630219-630484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395717</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26624627</td>\n",
       "      <td>26624680</td>\n",
       "      <td>DNase-seq:chrY:26624627-26624680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395718</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26651123</td>\n",
       "      <td>26651270</td>\n",
       "      <td>DNase-seq:chrY:26651123-26651270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395719</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663634</td>\n",
       "      <td>26663820</td>\n",
       "      <td>DNase-seq:chrY:26663634-26663820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395720</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26669995</td>\n",
       "      <td>26670117</td>\n",
       "      <td>DNase-seq:chrY:26669995-26670117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395721</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26670860</td>\n",
       "      <td>26671520</td>\n",
       "      <td>DNase-seq:chrY:26670860-26671520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>395888 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End                OpenChromatin_hMSC\n",
       "150987  chr1    181380    181580      DNase-seq:chr1:181380-181580\n",
       "152254  chr1    191406    191561      DNase-seq:chr1:191406-191561\n",
       "163325  chr1    268011    268060      DNase-seq:chr1:268011-268060\n",
       "169708  chr1    629079    630024       ATAC-seq:chr1:629079-630024\n",
       "169713  chr1    630219    630484       ATAC-seq:chr1:630219-630484\n",
       "...      ...       ...       ...                               ...\n",
       "395717  chrY  26624627  26624680  DNase-seq:chrY:26624627-26624680\n",
       "395718  chrY  26651123  26651270  DNase-seq:chrY:26651123-26651270\n",
       "395719  chrY  26663634  26663820  DNase-seq:chrY:26663634-26663820\n",
       "395720  chrY  26669995  26670117  DNase-seq:chrY:26669995-26670117\n",
       "395721  chrY  26670860  26671520  DNase-seq:chrY:26670860-26671520\n",
       "\n",
       "[395888 rows x 4 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "96f1f652-46bd-4e89-a9f6-6e7421b59c60",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_merge_atac_dnase_sorted_bed.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/hMSC/hg38_OpenChromatin_hMSC.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb553935-ac7f-4f70-9307-041aec35b3be",
   "metadata": {},
   "source": [
    "### promoter table overlap with openchromatin "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "3b9a02b2-ca3d-4686-9901-420de1720a8f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/promoter_regions_pcg_250bp_sorted.bed -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/hMSC/hg38_OpenChromatin_hMSC.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3fd31f1-2423-41ea-b4f9-7b244cd80535",
   "metadata": {},
   "source": [
    "### remove duplicated rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "f53ae2b1-85be-4e39-86d1-5fa0dc43fb92",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'chr_open','start_open','end_open','openChromatin_hMSC','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "5f46f4d7-96c6-4b1a-b74d-b29cb1593c1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "3666029e-908b-4ee6-802b-bd23f7877331",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924160</td>\n",
       "      <td>924202</td>\n",
       "      <td>DNase-seq:chr1:924160-924202</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299287</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299288</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299289</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299290</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299291</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>299292 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "299287  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "299288  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "299289  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "299290  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "299291  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923760    923940   \n",
       "4                            5.0     chr1      924160    924202   \n",
       "...                          ...      ...         ...       ...   \n",
       "299287                       5.0        .          -1        -1   \n",
       "299288                       1.0        .          -1        -1   \n",
       "299289                       2.0        .          -1        -1   \n",
       "299290                       1.0        .          -1        -1   \n",
       "299291                       2.0        .          -1        -1   \n",
       "\n",
       "                  openChromatin_hMSC  overlap  \n",
       "0                                  .        0  \n",
       "1                                  .        0  \n",
       "2                                  .        0  \n",
       "3       DNase-seq:chr1:923760-923940      180  \n",
       "4       DNase-seq:chr1:924160-924202       13  \n",
       "...                              ...      ...  \n",
       "299287                             .        0  \n",
       "299288                             .        0  \n",
       "299289                             .        0  \n",
       "299290                             .        0  \n",
       "299291                             .        0  \n",
       "\n",
       "[299292 rows x 13 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "449c8781-8235-4b95-a974-533f204d70c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924160</td>\n",
       "      <td>924202</td>\n",
       "      <td>DNase-seq:chr1:924160-924202</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924160</td>\n",
       "      <td>924202</td>\n",
       "      <td>DNase-seq:chr1:924160-924202</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924750</td>\n",
       "      <td>925302</td>\n",
       "      <td>ATAC-seq:chr1:924750-925302</td>\n",
       "      <td>402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924760</td>\n",
       "      <td>925000</td>\n",
       "      <td>DNase-seq:chr1:924760-925000</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940206</td>\n",
       "      <td>940767</td>\n",
       "      <td>ATAC-seq:chr1:940206-940767</td>\n",
       "      <td>390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940320</td>\n",
       "      <td>940600</td>\n",
       "      <td>DNase-seq:chr1:940320-940600</td>\n",
       "      <td>276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940860</td>\n",
       "      <td>941060</td>\n",
       "      <td>DNase-seq:chr1:940860-941060</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941640</td>\n",
       "      <td>941880</td>\n",
       "      <td>DNase-seq:chr1:941640-941880</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941643</td>\n",
       "      <td>942046</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941643</td>\n",
       "      <td>942046</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958614</td>\n",
       "      <td>958789</td>\n",
       "      <td>DNase-seq:chr1:958614-958789</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959185</td>\n",
       "      <td>959484</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959200</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959200-959440</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959185</td>\n",
       "      <td>959484</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959200</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959200-959440</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959185</td>\n",
       "      <td>959484</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959200</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959200-959440</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960460</td>\n",
       "      <td>960640</td>\n",
       "      <td>DNase-seq:chr1:960460-960640</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960477</td>\n",
       "      <td>960665</td>\n",
       "      <td>ATAC-seq:chr1:960477-960665</td>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>961371</td>\n",
       "      <td>961466</td>\n",
       "      <td>DNase-seq:chr1:961371-961466</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966118</td>\n",
       "      <td>966280</td>\n",
       "      <td>DNase-seq:chr1:966118-966280</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966362</td>\n",
       "      <td>966465</td>\n",
       "      <td>DNase-seq:chr1:966362-966465</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966700</td>\n",
       "      <td>966841</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966118</td>\n",
       "      <td>966280</td>\n",
       "      <td>DNase-seq:chr1:966118-966280</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966362</td>\n",
       "      <td>966465</td>\n",
       "      <td>DNase-seq:chr1:966362-966465</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966700</td>\n",
       "      <td>966841</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966118</td>\n",
       "      <td>966280</td>\n",
       "      <td>DNase-seq:chr1:966118-966280</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966362</td>\n",
       "      <td>966465</td>\n",
       "      <td>DNase-seq:chr1:966362-966465</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966700</td>\n",
       "      <td>966841</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom   start     end       Transcript     TSS     Gene strand  \\\n",
       "0   chr1   65169   65669  ENST00000641515   65419    OR4F5      +   \n",
       "1   chr1  451428  451928  ENST00000426406  451678   OR4F29      -   \n",
       "2   chr1  686404  686904  ENST00000332831  686654   OR4F16      -   \n",
       "3   chr1  923673  924173  ENST00000616016  923923   SAMD11      +   \n",
       "4   chr1  923673  924173  ENST00000616016  923923   SAMD11      +   \n",
       "5   chr1  923673  924173  ENST00000618323  923923   SAMD11      +   \n",
       "6   chr1  923673  924173  ENST00000618323  923923   SAMD11      +   \n",
       "7   chr1  924900  925400  ENST00000437963  925150   SAMD11      +   \n",
       "8   chr1  924900  925400  ENST00000437963  925150   SAMD11      +   \n",
       "9   chr1  925481  925981  ENST00000342066  925731   SAMD11      +   \n",
       "10  chr1  925692  926192  ENST00000616125  925942   SAMD11      +   \n",
       "11  chr1  925692  926192  ENST00000617307  925942   SAMD11      +   \n",
       "12  chr1  925692  926192  ENST00000618181  925942   SAMD11      +   \n",
       "13  chr1  925692  926192  ENST00000618779  925942   SAMD11      +   \n",
       "14  chr1  925692  926192  ENST00000622503  925942   SAMD11      +   \n",
       "15  chr1  930062  930562  ENST00000341065  930312   SAMD11      +   \n",
       "16  chr1  939025  939525  ENST00000455979  939275   SAMD11      +   \n",
       "17  chr1  940096  940596  ENST00000478729  940346   SAMD11      +   \n",
       "18  chr1  940096  940596  ENST00000478729  940346   SAMD11      +   \n",
       "19  chr1  940826  941326  ENST00000474461  941076   SAMD11      +   \n",
       "20  chr1  941853  942353  ENST00000466827  942103   SAMD11      +   \n",
       "21  chr1  941853  942353  ENST00000466827  942103   SAMD11      +   \n",
       "22  chr1  941916  942416  ENST00000464948  942166   SAMD11      +   \n",
       "23  chr1  945312  945812  ENST00000496938  945562    NOC2L      -   \n",
       "24  chr1  946810  947310  ENST00000483767  947060    NOC2L      -   \n",
       "25  chr1  958208  958708  ENST00000477976  958458    NOC2L      -   \n",
       "26  chr1  959006  959506  ENST00000327044  959256    NOC2L      -   \n",
       "27  chr1  959006  959506  ENST00000327044  959256    NOC2L      -   \n",
       "28  chr1  959006  959506  ENST00000469563  959256    NOC2L      -   \n",
       "29  chr1  959006  959506  ENST00000469563  959256    NOC2L      -   \n",
       "30  chr1  959059  959559  ENST00000487214  959309    NOC2L      -   \n",
       "31  chr1  959059  959559  ENST00000487214  959309    NOC2L      -   \n",
       "32  chr1  960334  960834  ENST00000338591  960584   KLHL17      +   \n",
       "33  chr1  960334  960834  ENST00000338591  960584   KLHL17      +   \n",
       "34  chr1  961199  961699  ENST00000463212  961449   KLHL17      +   \n",
       "35  chr1  962477  962977  ENST00000466300  962727   KLHL17      +   \n",
       "36  chr1  963302  963802  ENST00000481067  963552   KLHL17      +   \n",
       "37  chr1  966232  966732  ENST00000379410  966482  PLEKHN1      +   \n",
       "38  chr1  966232  966732  ENST00000379410  966482  PLEKHN1      +   \n",
       "39  chr1  966232  966732  ENST00000379410  966482  PLEKHN1      +   \n",
       "40  chr1  966252  966752  ENST00000379407  966502  PLEKHN1      +   \n",
       "41  chr1  966252  966752  ENST00000379407  966502  PLEKHN1      +   \n",
       "42  chr1  966252  966752  ENST00000379407  966502  PLEKHN1      +   \n",
       "43  chr1  966252  966752  ENST00000379409  966502  PLEKHN1      +   \n",
       "44  chr1  966252  966752  ENST00000379409  966502  PLEKHN1      +   \n",
       "45  chr1  966252  966752  ENST00000379409  966502  PLEKHN1      +   \n",
       "46  chr1  970625  971125  ENST00000480267  970875  PLEKHN1      +   \n",
       "47  chr1  973262  973762  ENST00000491024  973512  PLEKHN1      +   \n",
       "48  chr1  976391  976891  ENST00000479361  976641    PERM1      -   \n",
       "49  chr1  981843  982343  ENST00000341290  982093    PERM1      -   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923760    923940   \n",
       "4                        5.0     chr1      924160    924202   \n",
       "5                        5.0     chr1      923760    923940   \n",
       "6                        5.0     chr1      924160    924202   \n",
       "7                        5.0     chr1      924750    925302   \n",
       "8                        5.0     chr1      924760    925000   \n",
       "9                        5.0     chr1      925610    925786   \n",
       "10                       5.0     chr1      925610    925786   \n",
       "11                       5.0     chr1      925610    925786   \n",
       "12                       5.0     chr1      925610    925786   \n",
       "13                       5.0     chr1      925610    925786   \n",
       "14                       5.0     chr1      925610    925786   \n",
       "15                       5.0        .          -1        -1   \n",
       "16                       2.0        .          -1        -1   \n",
       "17                       5.0     chr1      940206    940767   \n",
       "18                       5.0     chr1      940320    940600   \n",
       "19                       5.0     chr1      940860    941060   \n",
       "20                       3.0     chr1      941640    941880   \n",
       "21                       3.0     chr1      941643    942046   \n",
       "22                       2.0     chr1      941643    942046   \n",
       "23                       5.0        .          -1        -1   \n",
       "24                       2.0        .          -1        -1   \n",
       "25                       5.0     chr1      958614    958789   \n",
       "26                       1.0     chr1      959185    959484   \n",
       "27                       1.0     chr1      959200    959440   \n",
       "28                       2.0     chr1      959185    959484   \n",
       "29                       2.0     chr1      959200    959440   \n",
       "30                       3.0     chr1      959185    959484   \n",
       "31                       3.0     chr1      959200    959440   \n",
       "32                       1.0     chr1      960460    960640   \n",
       "33                       1.0     chr1      960477    960665   \n",
       "34                       1.0     chr1      961371    961466   \n",
       "35                       3.0        .          -1        -1   \n",
       "36                       2.0        .          -1        -1   \n",
       "37                       1.0     chr1      966118    966280   \n",
       "38                       1.0     chr1      966362    966465   \n",
       "39                       1.0     chr1      966700    966841   \n",
       "40                       1.0     chr1      966118    966280   \n",
       "41                       1.0     chr1      966362    966465   \n",
       "42                       1.0     chr1      966700    966841   \n",
       "43                       2.0     chr1      966118    966280   \n",
       "44                       2.0     chr1      966362    966465   \n",
       "45                       2.0     chr1      966700    966841   \n",
       "46                       3.0        .          -1        -1   \n",
       "47                       3.0        .          -1        -1   \n",
       "48                       1.0        .          -1        -1   \n",
       "49                       2.0        .          -1        -1   \n",
       "\n",
       "              openChromatin_hMSC  overlap  \n",
       "0                              .        0  \n",
       "1                              .        0  \n",
       "2                              .        0  \n",
       "3   DNase-seq:chr1:923760-923940      180  \n",
       "4   DNase-seq:chr1:924160-924202       13  \n",
       "5   DNase-seq:chr1:923760-923940      180  \n",
       "6   DNase-seq:chr1:924160-924202       13  \n",
       "7    ATAC-seq:chr1:924750-925302      402  \n",
       "8   DNase-seq:chr1:924760-925000      100  \n",
       "9   DNase-seq:chr1:925610-925786      176  \n",
       "10  DNase-seq:chr1:925610-925786       94  \n",
       "11  DNase-seq:chr1:925610-925786       94  \n",
       "12  DNase-seq:chr1:925610-925786       94  \n",
       "13  DNase-seq:chr1:925610-925786       94  \n",
       "14  DNase-seq:chr1:925610-925786       94  \n",
       "15                             .        0  \n",
       "16                             .        0  \n",
       "17   ATAC-seq:chr1:940206-940767      390  \n",
       "18  DNase-seq:chr1:940320-940600      276  \n",
       "19  DNase-seq:chr1:940860-941060      200  \n",
       "20  DNase-seq:chr1:941640-941880       27  \n",
       "21   ATAC-seq:chr1:941643-942046      193  \n",
       "22   ATAC-seq:chr1:941643-942046      130  \n",
       "23                             .        0  \n",
       "24                             .        0  \n",
       "25  DNase-seq:chr1:958614-958789       94  \n",
       "26   ATAC-seq:chr1:959185-959484      299  \n",
       "27  DNase-seq:chr1:959200-959440      240  \n",
       "28   ATAC-seq:chr1:959185-959484      299  \n",
       "29  DNase-seq:chr1:959200-959440      240  \n",
       "30   ATAC-seq:chr1:959185-959484      299  \n",
       "31  DNase-seq:chr1:959200-959440      240  \n",
       "32  DNase-seq:chr1:960460-960640      180  \n",
       "33   ATAC-seq:chr1:960477-960665      188  \n",
       "34  DNase-seq:chr1:961371-961466       95  \n",
       "35                             .        0  \n",
       "36                             .        0  \n",
       "37  DNase-seq:chr1:966118-966280       48  \n",
       "38  DNase-seq:chr1:966362-966465      103  \n",
       "39   ATAC-seq:chr1:966700-966841       32  \n",
       "40  DNase-seq:chr1:966118-966280       28  \n",
       "41  DNase-seq:chr1:966362-966465      103  \n",
       "42   ATAC-seq:chr1:966700-966841       52  \n",
       "43  DNase-seq:chr1:966118-966280       28  \n",
       "44  DNase-seq:chr1:966362-966465      103  \n",
       "45   ATAC-seq:chr1:966700-966841       52  \n",
       "46                             .        0  \n",
       "47                             .        0  \n",
       "48                             .        0  \n",
       "49                             .        0  "
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "c87fb1ef-12a8-43f4-b3fd-4af9a1b4982d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "ad368f03-a970-4982-b1ba-5a35b74079f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924160</td>\n",
       "      <td>924202</td>\n",
       "      <td>DNase-seq:chr1:924160-924202</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299287</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299288</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299289</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299290</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>299291</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>299292 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "299287  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "299288  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "299289  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "299290  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "299291  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923760    923940   \n",
       "4                            5.0     chr1      924160    924202   \n",
       "...                          ...      ...         ...       ...   \n",
       "299287                       5.0        .          -1        -1   \n",
       "299288                       1.0        .          -1        -1   \n",
       "299289                       2.0        .          -1        -1   \n",
       "299290                       1.0        .          -1        -1   \n",
       "299291                       2.0        .          -1        -1   \n",
       "\n",
       "                  openChromatin_hMSC  overlap  \n",
       "0                                  .        0  \n",
       "1                                  .        0  \n",
       "2                                  .        0  \n",
       "3       DNase-seq:chr1:923760-923940      180  \n",
       "4       DNase-seq:chr1:924160-924202       13  \n",
       "...                              ...      ...  \n",
       "299287                             .        0  \n",
       "299288                             .        0  \n",
       "299289                             .        0  \n",
       "299290                             .        0  \n",
       "299291                             .        0  \n",
       "\n",
       "[299292 rows x 13 columns]"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "b8e5b8f5-db77-4d35-baf9-772c5bd2fcc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924160</td>\n",
       "      <td>924202</td>\n",
       "      <td>DNase-seq:chr1:924160-924202</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924160</td>\n",
       "      <td>924202</td>\n",
       "      <td>DNase-seq:chr1:924160-924202</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924750</td>\n",
       "      <td>925302</td>\n",
       "      <td>ATAC-seq:chr1:924750-925302</td>\n",
       "      <td>402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924760</td>\n",
       "      <td>925000</td>\n",
       "      <td>DNase-seq:chr1:924760-925000</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>925610</td>\n",
       "      <td>925786</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940206</td>\n",
       "      <td>940767</td>\n",
       "      <td>ATAC-seq:chr1:940206-940767</td>\n",
       "      <td>390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940320</td>\n",
       "      <td>940600</td>\n",
       "      <td>DNase-seq:chr1:940320-940600</td>\n",
       "      <td>276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940860</td>\n",
       "      <td>941060</td>\n",
       "      <td>DNase-seq:chr1:940860-941060</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941640</td>\n",
       "      <td>941880</td>\n",
       "      <td>DNase-seq:chr1:941640-941880</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941643</td>\n",
       "      <td>942046</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941643</td>\n",
       "      <td>942046</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958614</td>\n",
       "      <td>958789</td>\n",
       "      <td>DNase-seq:chr1:958614-958789</td>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959185</td>\n",
       "      <td>959484</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959200</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959200-959440</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959185</td>\n",
       "      <td>959484</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959200</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959200-959440</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959185</td>\n",
       "      <td>959484</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959200</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959200-959440</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960460</td>\n",
       "      <td>960640</td>\n",
       "      <td>DNase-seq:chr1:960460-960640</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960477</td>\n",
       "      <td>960665</td>\n",
       "      <td>ATAC-seq:chr1:960477-960665</td>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>961371</td>\n",
       "      <td>961466</td>\n",
       "      <td>DNase-seq:chr1:961371-961466</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966118</td>\n",
       "      <td>966280</td>\n",
       "      <td>DNase-seq:chr1:966118-966280</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966362</td>\n",
       "      <td>966465</td>\n",
       "      <td>DNase-seq:chr1:966362-966465</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966700</td>\n",
       "      <td>966841</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966118</td>\n",
       "      <td>966280</td>\n",
       "      <td>DNase-seq:chr1:966118-966280</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966362</td>\n",
       "      <td>966465</td>\n",
       "      <td>DNase-seq:chr1:966362-966465</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966700</td>\n",
       "      <td>966841</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966118</td>\n",
       "      <td>966280</td>\n",
       "      <td>DNase-seq:chr1:966118-966280</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966362</td>\n",
       "      <td>966465</td>\n",
       "      <td>DNase-seq:chr1:966362-966465</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966700</td>\n",
       "      <td>966841</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom   start     end       Transcript     TSS     Gene strand  \\\n",
       "0   chr1   65169   65669  ENST00000641515   65419    OR4F5      +   \n",
       "1   chr1  451428  451928  ENST00000426406  451678   OR4F29      -   \n",
       "2   chr1  686404  686904  ENST00000332831  686654   OR4F16      -   \n",
       "3   chr1  923673  924173  ENST00000616016  923923   SAMD11      +   \n",
       "4   chr1  923673  924173  ENST00000616016  923923   SAMD11      +   \n",
       "5   chr1  923673  924173  ENST00000618323  923923   SAMD11      +   \n",
       "6   chr1  923673  924173  ENST00000618323  923923   SAMD11      +   \n",
       "7   chr1  924900  925400  ENST00000437963  925150   SAMD11      +   \n",
       "8   chr1  924900  925400  ENST00000437963  925150   SAMD11      +   \n",
       "9   chr1  925481  925981  ENST00000342066  925731   SAMD11      +   \n",
       "10  chr1  925692  926192  ENST00000616125  925942   SAMD11      +   \n",
       "11  chr1  925692  926192  ENST00000617307  925942   SAMD11      +   \n",
       "12  chr1  925692  926192  ENST00000618181  925942   SAMD11      +   \n",
       "13  chr1  925692  926192  ENST00000618779  925942   SAMD11      +   \n",
       "14  chr1  925692  926192  ENST00000622503  925942   SAMD11      +   \n",
       "15  chr1  930062  930562  ENST00000341065  930312   SAMD11      +   \n",
       "16  chr1  939025  939525  ENST00000455979  939275   SAMD11      +   \n",
       "17  chr1  940096  940596  ENST00000478729  940346   SAMD11      +   \n",
       "18  chr1  940096  940596  ENST00000478729  940346   SAMD11      +   \n",
       "19  chr1  940826  941326  ENST00000474461  941076   SAMD11      +   \n",
       "20  chr1  941853  942353  ENST00000466827  942103   SAMD11      +   \n",
       "21  chr1  941853  942353  ENST00000466827  942103   SAMD11      +   \n",
       "22  chr1  941916  942416  ENST00000464948  942166   SAMD11      +   \n",
       "23  chr1  945312  945812  ENST00000496938  945562    NOC2L      -   \n",
       "24  chr1  946810  947310  ENST00000483767  947060    NOC2L      -   \n",
       "25  chr1  958208  958708  ENST00000477976  958458    NOC2L      -   \n",
       "26  chr1  959006  959506  ENST00000327044  959256    NOC2L      -   \n",
       "27  chr1  959006  959506  ENST00000327044  959256    NOC2L      -   \n",
       "28  chr1  959006  959506  ENST00000469563  959256    NOC2L      -   \n",
       "29  chr1  959006  959506  ENST00000469563  959256    NOC2L      -   \n",
       "30  chr1  959059  959559  ENST00000487214  959309    NOC2L      -   \n",
       "31  chr1  959059  959559  ENST00000487214  959309    NOC2L      -   \n",
       "32  chr1  960334  960834  ENST00000338591  960584   KLHL17      +   \n",
       "33  chr1  960334  960834  ENST00000338591  960584   KLHL17      +   \n",
       "34  chr1  961199  961699  ENST00000463212  961449   KLHL17      +   \n",
       "35  chr1  962477  962977  ENST00000466300  962727   KLHL17      +   \n",
       "36  chr1  963302  963802  ENST00000481067  963552   KLHL17      +   \n",
       "37  chr1  966232  966732  ENST00000379410  966482  PLEKHN1      +   \n",
       "38  chr1  966232  966732  ENST00000379410  966482  PLEKHN1      +   \n",
       "39  chr1  966232  966732  ENST00000379410  966482  PLEKHN1      +   \n",
       "40  chr1  966252  966752  ENST00000379407  966502  PLEKHN1      +   \n",
       "41  chr1  966252  966752  ENST00000379407  966502  PLEKHN1      +   \n",
       "42  chr1  966252  966752  ENST00000379407  966502  PLEKHN1      +   \n",
       "43  chr1  966252  966752  ENST00000379409  966502  PLEKHN1      +   \n",
       "44  chr1  966252  966752  ENST00000379409  966502  PLEKHN1      +   \n",
       "45  chr1  966252  966752  ENST00000379409  966502  PLEKHN1      +   \n",
       "46  chr1  970625  971125  ENST00000480267  970875  PLEKHN1      +   \n",
       "47  chr1  973262  973762  ENST00000491024  973512  PLEKHN1      +   \n",
       "48  chr1  976391  976891  ENST00000479361  976641    PERM1      -   \n",
       "49  chr1  981843  982343  ENST00000341290  982093    PERM1      -   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923760    923940   \n",
       "4                        5.0     chr1      924160    924202   \n",
       "5                        5.0     chr1      923760    923940   \n",
       "6                        5.0     chr1      924160    924202   \n",
       "7                        5.0     chr1      924750    925302   \n",
       "8                        5.0     chr1      924760    925000   \n",
       "9                        5.0     chr1      925610    925786   \n",
       "10                       5.0     chr1      925610    925786   \n",
       "11                       5.0     chr1      925610    925786   \n",
       "12                       5.0     chr1      925610    925786   \n",
       "13                       5.0     chr1      925610    925786   \n",
       "14                       5.0     chr1      925610    925786   \n",
       "15                       5.0        .          -1        -1   \n",
       "16                       2.0        .          -1        -1   \n",
       "17                       5.0     chr1      940206    940767   \n",
       "18                       5.0     chr1      940320    940600   \n",
       "19                       5.0     chr1      940860    941060   \n",
       "20                       3.0     chr1      941640    941880   \n",
       "21                       3.0     chr1      941643    942046   \n",
       "22                       2.0     chr1      941643    942046   \n",
       "23                       5.0        .          -1        -1   \n",
       "24                       2.0        .          -1        -1   \n",
       "25                       5.0     chr1      958614    958789   \n",
       "26                       1.0     chr1      959185    959484   \n",
       "27                       1.0     chr1      959200    959440   \n",
       "28                       2.0     chr1      959185    959484   \n",
       "29                       2.0     chr1      959200    959440   \n",
       "30                       3.0     chr1      959185    959484   \n",
       "31                       3.0     chr1      959200    959440   \n",
       "32                       1.0     chr1      960460    960640   \n",
       "33                       1.0     chr1      960477    960665   \n",
       "34                       1.0     chr1      961371    961466   \n",
       "35                       3.0        .          -1        -1   \n",
       "36                       2.0        .          -1        -1   \n",
       "37                       1.0     chr1      966118    966280   \n",
       "38                       1.0     chr1      966362    966465   \n",
       "39                       1.0     chr1      966700    966841   \n",
       "40                       1.0     chr1      966118    966280   \n",
       "41                       1.0     chr1      966362    966465   \n",
       "42                       1.0     chr1      966700    966841   \n",
       "43                       2.0     chr1      966118    966280   \n",
       "44                       2.0     chr1      966362    966465   \n",
       "45                       2.0     chr1      966700    966841   \n",
       "46                       3.0        .          -1        -1   \n",
       "47                       3.0        .          -1        -1   \n",
       "48                       1.0        .          -1        -1   \n",
       "49                       2.0        .          -1        -1   \n",
       "\n",
       "              openChromatin_hMSC  overlap  \n",
       "0                              .        0  \n",
       "1                              .        0  \n",
       "2                              .        0  \n",
       "3   DNase-seq:chr1:923760-923940      180  \n",
       "4   DNase-seq:chr1:924160-924202       13  \n",
       "5   DNase-seq:chr1:923760-923940      180  \n",
       "6   DNase-seq:chr1:924160-924202       13  \n",
       "7    ATAC-seq:chr1:924750-925302      402  \n",
       "8   DNase-seq:chr1:924760-925000      100  \n",
       "9   DNase-seq:chr1:925610-925786      176  \n",
       "10  DNase-seq:chr1:925610-925786       94  \n",
       "11  DNase-seq:chr1:925610-925786       94  \n",
       "12  DNase-seq:chr1:925610-925786       94  \n",
       "13  DNase-seq:chr1:925610-925786       94  \n",
       "14  DNase-seq:chr1:925610-925786       94  \n",
       "15                             .        0  \n",
       "16                             .        0  \n",
       "17   ATAC-seq:chr1:940206-940767      390  \n",
       "18  DNase-seq:chr1:940320-940600      276  \n",
       "19  DNase-seq:chr1:940860-941060      200  \n",
       "20  DNase-seq:chr1:941640-941880       27  \n",
       "21   ATAC-seq:chr1:941643-942046      193  \n",
       "22   ATAC-seq:chr1:941643-942046      130  \n",
       "23                             .        0  \n",
       "24                             .        0  \n",
       "25  DNase-seq:chr1:958614-958789       94  \n",
       "26   ATAC-seq:chr1:959185-959484      299  \n",
       "27  DNase-seq:chr1:959200-959440      240  \n",
       "28   ATAC-seq:chr1:959185-959484      299  \n",
       "29  DNase-seq:chr1:959200-959440      240  \n",
       "30   ATAC-seq:chr1:959185-959484      299  \n",
       "31  DNase-seq:chr1:959200-959440      240  \n",
       "32  DNase-seq:chr1:960460-960640      180  \n",
       "33   ATAC-seq:chr1:960477-960665      188  \n",
       "34  DNase-seq:chr1:961371-961466       95  \n",
       "35                             .        0  \n",
       "36                             .        0  \n",
       "37  DNase-seq:chr1:966118-966280       48  \n",
       "38  DNase-seq:chr1:966362-966465      103  \n",
       "39   ATAC-seq:chr1:966700-966841       32  \n",
       "40  DNase-seq:chr1:966118-966280       28  \n",
       "41  DNase-seq:chr1:966362-966465      103  \n",
       "42   ATAC-seq:chr1:966700-966841       52  \n",
       "43  DNase-seq:chr1:966118-966280       28  \n",
       "44  DNase-seq:chr1:966362-966465      103  \n",
       "45   ATAC-seq:chr1:966700-966841       52  \n",
       "46                             .        0  \n",
       "47                             .        0  \n",
       "48                             .        0  \n",
       "49                             .        0  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "67947a08-1e02-49ce-83b1-a167901ae8a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to transcript_support_level)\n",
    "group_cols = promoter_like_open.columns[:8]\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "dccec6f3-8b09-454d-90a9-7dd12899d7e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open.groupby(list(group_cols), dropna=False)['openChromatin_hMSC'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "4558aa29-a1bb-4237-a26a-ad8dec8070b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "170552                       5.0   \n",
       "170553                       1.0   \n",
       "170554                       2.0   \n",
       "170555                       1.0   \n",
       "170556                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC  \n",
       "0                                                       .  \n",
       "1                                                       .  \n",
       "2                                                       .  \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...  \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 9 columns]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "f1a85a30-c934-46c4-acd6-62af217bd97e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:924750-925302;DNase-seq:chr1:924...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940206-940767;DNase-seq:chr1:940...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:940860-941060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046;DNase-seq:chr1:941...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:958614-958789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960477-960665;DNase-seq:chr1:960...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>DNase-seq:chr1:961371-961466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:1000382-1000639;ATAC-seq:chr1:99...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1013101-1013955;DNase-seq:chr1:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019347-1020347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019347-1020347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1033850-1033983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1033850-1033983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                   openChromatin_hMSC  \n",
       "0                                                   .  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3   DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...  \n",
       "4   DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...  \n",
       "5   ATAC-seq:chr1:924750-925302;DNase-seq:chr1:924...  \n",
       "6                        DNase-seq:chr1:925610-925786  \n",
       "7                        DNase-seq:chr1:925610-925786  \n",
       "8                        DNase-seq:chr1:925610-925786  \n",
       "9                        DNase-seq:chr1:925610-925786  \n",
       "10                       DNase-seq:chr1:925610-925786  \n",
       "11                       DNase-seq:chr1:925610-925786  \n",
       "12                                                  .  \n",
       "13                                                  .  \n",
       "14  ATAC-seq:chr1:940206-940767;DNase-seq:chr1:940...  \n",
       "15                       DNase-seq:chr1:940860-941060  \n",
       "16  ATAC-seq:chr1:941643-942046;DNase-seq:chr1:941...  \n",
       "17                        ATAC-seq:chr1:941643-942046  \n",
       "18                                                  .  \n",
       "19                                                  .  \n",
       "20                       DNase-seq:chr1:958614-958789  \n",
       "21  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...  \n",
       "22  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...  \n",
       "23  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...  \n",
       "24  ATAC-seq:chr1:960477-960665;DNase-seq:chr1:960...  \n",
       "25                       DNase-seq:chr1:961371-961466  \n",
       "26                                                  .  \n",
       "27                                                  .  \n",
       "28  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...  \n",
       "29  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...  \n",
       "30  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...  \n",
       "31                                                  .  \n",
       "32                                                  .  \n",
       "33                                                  .  \n",
       "34                                                  .  \n",
       "35                                                  .  \n",
       "36                                                  .  \n",
       "37  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...  \n",
       "38  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...  \n",
       "39  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...  \n",
       "40  ATAC-seq:chr1:1000382-1000639;ATAC-seq:chr1:99...  \n",
       "41                     DNase-seq:chr1:1000720-1001000  \n",
       "42                     DNase-seq:chr1:1000720-1001000  \n",
       "43  ATAC-seq:chr1:1013101-1013955;DNase-seq:chr1:1...  \n",
       "44                      ATAC-seq:chr1:1019347-1020347  \n",
       "45                      ATAC-seq:chr1:1019347-1020347  \n",
       "46                     DNase-seq:chr1:1033850-1033983  \n",
       "47                     DNase-seq:chr1:1033850-1033983  \n",
       "48                                                  .  \n",
       "49                                                  .  "
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "70759c9e-2a8a-470b-af36-c276ffb07909",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin.bed_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51e718c2-56ec-4903-8f6b-cf50fdaff699",
   "metadata": {},
   "source": [
    "## promoter table overlap with chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "06ae8b61-4fca-4708-b3c6-da6389dc21aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"ChromHMM_hMSC\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "b5361981-a1cb-41ec-bdea-c4116e9226a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E026_25_hMSC/hg38_chromHMM_E026_25.txt',\n",
    "                          sep='\\t',\n",
    "                          header=None,\n",
    "                          names=cols\n",
    "                          )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "e87a09c4-0ed6-4cb7-98ea-6112acd69466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>10400</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10400</td>\n",
       "      <td>10600</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916680</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155929536</td>\n",
       "      <td>155930136</td>\n",
       "      <td>20_ZNF/Rpts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916681</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930136</td>\n",
       "      <td>155930336</td>\n",
       "      <td>21_Het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916682</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930336</td>\n",
       "      <td>156030735</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916683</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916684</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>916685 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end ChromHMM_hMSC\n",
       "0       chr1      10000      10400      25_Quies\n",
       "1       chr1      10400      10600      19_DNase\n",
       "2       chr1      10600      96080      25_Quies\n",
       "3       chr1      96276      96476      25_Quies\n",
       "4       chr1      97276     177200      25_Quies\n",
       "...      ...        ...        ...           ...\n",
       "916680  chrX  155929536  155930136   20_ZNF/Rpts\n",
       "916681  chrX  155930136  155930336        21_Het\n",
       "916682  chrX  155930336  156030735      25_Quies\n",
       "916683  chrY    1086142    1086334      25_Quies\n",
       "916684  chrY    1198191    1198391      25_Quies\n",
       "\n",
       "[916685 rows x 4 columns]"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "1fcc068f-3f0d-4f23-8378-fcc0e7ae08a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_chromHMM['ChromHMM_hMSC'] = hMSC_chromHMM['Chrom'] + \":\" + hMSC_chromHMM['start'].astype('str') + \":\" + hMSC_chromHMM['end'].astype('str') + \":\" + hMSC_chromHMM['ChromHMM_hMSC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "b7e24ff8-e263-44c6-b30e-dce9e991e152",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>10400</td>\n",
       "      <td>chr1:10000:10400:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10400</td>\n",
       "      <td>10600</td>\n",
       "      <td>chr1:10400:10600:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>chr1:96276:96476:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>chr1:97276:177200:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916680</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155929536</td>\n",
       "      <td>155930136</td>\n",
       "      <td>chrX:155929536:155930136:20_ZNF/Rpts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916681</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930136</td>\n",
       "      <td>155930336</td>\n",
       "      <td>chrX:155930136:155930336:21_Het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916682</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930336</td>\n",
       "      <td>156030735</td>\n",
       "      <td>chrX:155930336:156030735:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916683</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>chrY:1086142:1086334:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916684</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>chrY:1198191:1198391:25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>916685 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end                         ChromHMM_hMSC\n",
       "0       chr1      10000      10400             chr1:10000:10400:25_Quies\n",
       "1       chr1      10400      10600             chr1:10400:10600:19_DNase\n",
       "2       chr1      10600      96080             chr1:10600:96080:25_Quies\n",
       "3       chr1      96276      96476             chr1:96276:96476:25_Quies\n",
       "4       chr1      97276     177200            chr1:97276:177200:25_Quies\n",
       "...      ...        ...        ...                                   ...\n",
       "916680  chrX  155929536  155930136  chrX:155929536:155930136:20_ZNF/Rpts\n",
       "916681  chrX  155930136  155930336       chrX:155930136:155930336:21_Het\n",
       "916682  chrX  155930336  156030735     chrX:155930336:156030735:25_Quies\n",
       "916683  chrY    1086142    1086334         chrY:1086142:1086334:25_Quies\n",
       "916684  chrY    1198191    1198391         chrY:1198191:1198391:25_Quies\n",
       "\n",
       "[916685 rows x 4 columns]"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hMSC_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "5efa4563-d041-49bd-95e8-72c56adcd6b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "hMSC_chromHMM.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E026_25_hMSC/hg38_chromHMM_hMSC.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "89b342b8-c2c6-4e67-bd2c-e4e88eef71d4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin.bed_uniq -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E026_25_hMSC/hg38_chromHMM_hMSC.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin_chromHMM.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "c5e71ecf-dc77-4f08-9ec0-3e1c12adadef",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'openChromatin_hMSC','chr_HMM','start_HMM','end_HMM','chromHMM_hMSC','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "75f9f5e3-77f8-4174-bc66-67464437ffa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin_chromHMM.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "e1ddb9b9-2d80-4103-9525-b432d43ff309",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chr_HMM</th>\n",
       "      <th>start_HMM</th>\n",
       "      <th>end_HMM</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>348136</td>\n",
       "      <td>501536</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>634820</td>\n",
       "      <td>733820</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923620</td>\n",
       "      <td>923820</td>\n",
       "      <td>chr1:923620:923820:2_PromU</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923820</td>\n",
       "      <td>924620</td>\n",
       "      <td>chr1:923820:924620:22_PromP</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287373</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287374</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287375</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287376</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287377</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>287378 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "287373  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "287374  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "287375  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "287376  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "287377  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "287373                       5.0   \n",
       "287374                       1.0   \n",
       "287375                       2.0   \n",
       "287376                       1.0   \n",
       "287377                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC chr_HMM  start_HMM  \\\n",
       "0                                                       .    chr1      10600   \n",
       "1                                                       .    chr1     348136   \n",
       "2                                                       .    chr1     634820   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...    chr1     923620   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...    chr1     923820   \n",
       "...                                                   ...     ...        ...   \n",
       "287373                                                  .       .         -1   \n",
       "287374                                                  .       .         -1   \n",
       "287375                                                  .       .         -1   \n",
       "287376                                                  .       .         -1   \n",
       "287377                                                  .       .         -1   \n",
       "\n",
       "        end_HMM                chromHMM_hMSC  overlap  \n",
       "0         96080    chr1:10600:96080:25_Quies      500  \n",
       "1        501536  chr1:348136:501536:25_Quies      500  \n",
       "2        733820  chr1:634820:733820:25_Quies      500  \n",
       "3        923820   chr1:923620:923820:2_PromU      147  \n",
       "4        924620  chr1:923820:924620:22_PromP      353  \n",
       "...         ...                          ...      ...  \n",
       "287373       -1                            .        0  \n",
       "287374       -1                            .        0  \n",
       "287375       -1                            .        0  \n",
       "287376       -1                            .        0  \n",
       "287377       -1                            .        0  \n",
       "\n",
       "[287378 rows x 14 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "3f9725c9-7ae4-4bd2-8bea-71be49de0499",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "07e9feff-7bb2-45f3-90fd-0548cc9c6104",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chr_HMM</th>\n",
       "      <th>start_HMM</th>\n",
       "      <th>end_HMM</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>348136</td>\n",
       "      <td>501536</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>634820</td>\n",
       "      <td>733820</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923620</td>\n",
       "      <td>923820</td>\n",
       "      <td>chr1:923620:923820:2_PromU</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923820</td>\n",
       "      <td>924620</td>\n",
       "      <td>chr1:923820:924620:22_PromP</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287373</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287374</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287375</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287376</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287377</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>287378 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "287373  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "287374  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "287375  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "287376  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "287377  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "287373                       5.0   \n",
       "287374                       1.0   \n",
       "287375                       2.0   \n",
       "287376                       1.0   \n",
       "287377                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC chr_HMM  start_HMM  \\\n",
       "0                                                       .    chr1      10600   \n",
       "1                                                       .    chr1     348136   \n",
       "2                                                       .    chr1     634820   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...    chr1     923620   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...    chr1     923820   \n",
       "...                                                   ...     ...        ...   \n",
       "287373                                                  .       .         -1   \n",
       "287374                                                  .       .         -1   \n",
       "287375                                                  .       .         -1   \n",
       "287376                                                  .       .         -1   \n",
       "287377                                                  .       .         -1   \n",
       "\n",
       "        end_HMM                chromHMM_hMSC  overlap  \n",
       "0         96080    chr1:10600:96080:25_Quies      500  \n",
       "1        501536  chr1:348136:501536:25_Quies      500  \n",
       "2        733820  chr1:634820:733820:25_Quies      500  \n",
       "3        923820   chr1:923620:923820:2_PromU      147  \n",
       "4        924620  chr1:923820:924620:22_PromP      353  \n",
       "...         ...                          ...      ...  \n",
       "287373       -1                            .        0  \n",
       "287374       -1                            .        0  \n",
       "287375       -1                            .        0  \n",
       "287376       -1                            .        0  \n",
       "287377       -1                            .        0  \n",
       "\n",
       "[287378 rows x 14 columns]"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "eecc0fdc-a318-4f06-8214-010457b6b75c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level', 'openChromatin_hMSC'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to openChromatin)\n",
    "group_cols = promoter_like_open.columns[:9]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "1ee55dbf-6dd4-41e9-9e4d-99d7bae31daf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open.groupby(list(group_cols), dropna=False)['chromHMM_hMSC'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "24ff1d6d-f8cc-4742-b862-0fdb6cc0d39c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "170552                       5.0   \n",
       "170553                       1.0   \n",
       "170554                       2.0   \n",
       "170555                       1.0   \n",
       "170556                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC  \\\n",
       "0                                                       .   \n",
       "1                                                       .   \n",
       "2                                                       .   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "...                                                   ...   \n",
       "170552                                                  .   \n",
       "170553                                                  .   \n",
       "170554                                                  .   \n",
       "170555                                                  .   \n",
       "170556                                                  .   \n",
       "\n",
       "                                            chromHMM_hMSC  \n",
       "0                               chr1:10600:96080:25_Quies  \n",
       "1                             chr1:348136:501536:25_Quies  \n",
       "2                             chr1:634820:733820:25_Quies  \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 10 columns]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "38e59480-fd89-428d-ac23-fcf67893c731",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:924750-925302;DNase-seq:chr1:924...</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:927020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:927020:935020:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938420:939820:17_EnhW2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940206-940767;DNase-seq:chr1:940...</td>\n",
       "      <td>chr1:939820:940420:16_EnhW1;chr1:940420:941620...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:940860-941060</td>\n",
       "      <td>chr1:940420:941620:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046;DNase-seq:chr1:941...</td>\n",
       "      <td>chr1:941620:942020:16_EnhW1;chr1:942020:942820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046</td>\n",
       "      <td>chr1:941620:942020:16_EnhW1;chr1:942020:942820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945020:949820:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945020:949820:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:958614-958789</td>\n",
       "      <td>chr1:957620:958820:4_PromD2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958820:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958820:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958820:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960477-960665;DNase-seq:chr1:960...</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961020:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>DNase-seq:chr1:961371-961466</td>\n",
       "      <td>chr1:961020:961220:22_PromP;chr1:961220:961820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:963820:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:963820:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "      <td>chr1:966020:967420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "      <td>chr1:966020:967420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "      <td>chr1:966020:967420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:970220:972020:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972020:975820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:976220:976620:2_PromU;chr1:976620:979820:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:979820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:979820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:979820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999220:99982...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999820:10002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999820:10002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:1000382-1000639;ATAC-seq:chr1:99...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999820:10002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1013101-1013955;DNase-seq:chr1:1...</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10144...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019347-1020347</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019347-1020347</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1033850-1033983</td>\n",
       "      <td>chr1:1033620:1034620:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1033850-1033983</td>\n",
       "      <td>chr1:1033620:1034620:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033620:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1045020:1051620:8_TxWk</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                   openChromatin_hMSC  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3   DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4   DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "5   ATAC-seq:chr1:924750-925302;DNase-seq:chr1:924...   \n",
       "6                        DNase-seq:chr1:925610-925786   \n",
       "7                        DNase-seq:chr1:925610-925786   \n",
       "8                        DNase-seq:chr1:925610-925786   \n",
       "9                        DNase-seq:chr1:925610-925786   \n",
       "10                       DNase-seq:chr1:925610-925786   \n",
       "11                       DNase-seq:chr1:925610-925786   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14  ATAC-seq:chr1:940206-940767;DNase-seq:chr1:940...   \n",
       "15                       DNase-seq:chr1:940860-941060   \n",
       "16  ATAC-seq:chr1:941643-942046;DNase-seq:chr1:941...   \n",
       "17                        ATAC-seq:chr1:941643-942046   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                       DNase-seq:chr1:958614-958789   \n",
       "21  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...   \n",
       "24  ATAC-seq:chr1:960477-960665;DNase-seq:chr1:960...   \n",
       "25                       DNase-seq:chr1:961371-961466   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...   \n",
       "29  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...   \n",
       "30  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                                                  .   \n",
       "34                                                  .   \n",
       "35                                                  .   \n",
       "36                                                  .   \n",
       "37  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:1000382-1000639;ATAC-seq:chr1:99...   \n",
       "41                     DNase-seq:chr1:1000720-1001000   \n",
       "42                     DNase-seq:chr1:1000720-1001000   \n",
       "43  ATAC-seq:chr1:1013101-1013955;DNase-seq:chr1:1...   \n",
       "44                      ATAC-seq:chr1:1019347-1020347   \n",
       "45                      ATAC-seq:chr1:1019347-1020347   \n",
       "46                     DNase-seq:chr1:1033850-1033983   \n",
       "47                     DNase-seq:chr1:1033850-1033983   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                        chromHMM_hMSC  \n",
       "0                           chr1:10600:96080:25_Quies  \n",
       "1                         chr1:348136:501536:25_Quies  \n",
       "2                         chr1:634820:733820:25_Quies  \n",
       "3   chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "4   chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...  \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:927020...  \n",
       "7                       chr1:925620:927020:23_PromBiv  \n",
       "8                       chr1:925620:927020:23_PromBiv  \n",
       "9                       chr1:925620:927020:23_PromBiv  \n",
       "10                      chr1:925620:927020:23_PromBiv  \n",
       "11                      chr1:925620:927020:23_PromBiv  \n",
       "12                        chr1:927020:935020:25_Quies  \n",
       "13                        chr1:938420:939820:17_EnhW2  \n",
       "14  chr1:939820:940420:16_EnhW1;chr1:940420:941620...  \n",
       "15                         chr1:940420:941620:2_PromU  \n",
       "16  chr1:941620:942020:16_EnhW1;chr1:942020:942820...  \n",
       "17  chr1:941620:942020:16_EnhW1;chr1:942020:942820...  \n",
       "18                          chr1:945020:949820:7_Tx3'  \n",
       "19                          chr1:945020:949820:7_Tx3'  \n",
       "20                        chr1:957620:958820:4_PromD2  \n",
       "21  chr1:958820:959220:3_PromD1;chr1:959220:959420...  \n",
       "22  chr1:958820:959220:3_PromD1;chr1:959220:959420...  \n",
       "23  chr1:958820:959220:3_PromD1;chr1:959220:959420...  \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961020:...  \n",
       "25  chr1:961020:961220:22_PromP;chr1:961220:961820...  \n",
       "26                          chr1:962420:963820:7_Tx3'  \n",
       "27                          chr1:962420:963820:7_Tx3'  \n",
       "28                         chr1:966020:967420:2_PromU  \n",
       "29                         chr1:966020:967420:2_PromU  \n",
       "30                         chr1:966020:967420:2_PromU  \n",
       "31                       chr1:970220:972020:24_ReprPC  \n",
       "32                        chr1:972020:975820:25_Quies  \n",
       "33  chr1:976220:976620:2_PromU;chr1:976620:979820:...  \n",
       "34                       chr1:979820:984220:24_ReprPC  \n",
       "35                       chr1:979820:984220:24_ReprPC  \n",
       "36                       chr1:979820:984220:24_ReprPC  \n",
       "37  chr1:1000220:1001420:2_PromU;chr1:999220:99982...  \n",
       "38  chr1:1000220:1001420:2_PromU;chr1:999820:10002...  \n",
       "39  chr1:1000220:1001420:2_PromU;chr1:999820:10002...  \n",
       "40  chr1:1000220:1001420:2_PromU;chr1:999820:10002...  \n",
       "41                       chr1:1000220:1001420:2_PromU  \n",
       "42                       chr1:1000220:1001420:2_PromU  \n",
       "43  chr1:1013220:1013420:1_TssA;chr1:1013420:10144...  \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "46                     chr1:1033620:1034620:24_ReprPC  \n",
       "47                     chr1:1033620:1034620:24_ReprPC  \n",
       "48  chr1:1033620:1034620:24_ReprPC;chr1:1034620:10...  \n",
       "49                        chr1:1045020:1051620:8_TxWk  "
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "4687dbce-a028-4937-a1f0-82ef46fe4945",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin_chromHMM_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48e46e4c-4a02-4286-b18d-b9843cbce955",
   "metadata": {},
   "source": [
    "## promoter overlap with Hi-C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "c641047c-e227-441b-a240-966c2e6f4463",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin_chromHMM_uniq -b /mnt/10T-backup-mingju/ifar/Hi-C/hMSC/hMSC_merge_hic.txt_sorted_lexicographical_addchr -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin_chromHMM_HiC.bed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "37b3a092-3e46-41aa-97ff-18431186071b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'openChromatin_hMSC','chromHMM_hMSC','chr_hic','start_hic','end_hic','HiC_hMSC','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "d89d3ca3-a38f-41f4-9b86-844ee9a210e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin_chromHMM_HiC.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "162b7fbc-df46-4875-991c-10a094cd253b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>chr_hic</th>\n",
       "      <th>start_hic</th>\n",
       "      <th>end_hic</th>\n",
       "      <th>HiC_hMSC</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>chr1</td>\n",
       "      <td>64000</td>\n",
       "      <td>66000</td>\n",
       "      <td>P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>922000</td>\n",
       "      <td>924000</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>922000</td>\n",
       "      <td>924000</td>\n",
       "      <td>P:1:922000:924000;D:1:910000:912000;HOMER:read...</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6260633</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6260634</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6260635</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6260636</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6260637</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6260638 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0        chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1        chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2        chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3        chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4        chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...       ...       ...       ...              ...       ...     ...    ...   \n",
       "6260633  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "6260634  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "6260635  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "6260636  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "6260637  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "         transcript_support_level  \\\n",
       "0                             NaN   \n",
       "1                             NaN   \n",
       "2                             NaN   \n",
       "3                             5.0   \n",
       "4                             5.0   \n",
       "...                           ...   \n",
       "6260633                       5.0   \n",
       "6260634                       1.0   \n",
       "6260635                       2.0   \n",
       "6260636                       1.0   \n",
       "6260637                       2.0   \n",
       "\n",
       "                                        openChromatin_hMSC  \\\n",
       "0                                                        .   \n",
       "1                                                        .   \n",
       "2                                                        .   \n",
       "3        DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4        DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "...                                                    ...   \n",
       "6260633                                                  .   \n",
       "6260634                                                  .   \n",
       "6260635                                                  .   \n",
       "6260636                                                  .   \n",
       "6260637                                                  .   \n",
       "\n",
       "                                             chromHMM_hMSC chr_hic  start_hic  \\\n",
       "0                                chr1:10600:96080:25_Quies    chr1      64000   \n",
       "1                              chr1:348136:501536:25_Quies       .         -1   \n",
       "2                              chr1:634820:733820:25_Quies       .         -1   \n",
       "3        chr1:923620:923820:2_PromU;chr1:923820:924620:...    chr1     922000   \n",
       "4        chr1:923620:923820:2_PromU;chr1:923820:924620:...    chr1     922000   \n",
       "...                                                    ...     ...        ...   \n",
       "6260633                                                  .       .         -1   \n",
       "6260634                                                  .       .         -1   \n",
       "6260635                                                  .       .         -1   \n",
       "6260636                                                  .       .         -1   \n",
       "6260637                                                  .       .         -1   \n",
       "\n",
       "         end_hic                                           HiC_hMSC  overlap  \n",
       "0          66000  P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...      500  \n",
       "1             -1                                                  .        0  \n",
       "2             -1                                                  .        0  \n",
       "3         924000  P:1:922000:924000;D:1:906000:908000;HOMER:read...      327  \n",
       "4         924000  P:1:922000:924000;D:1:910000:912000;HOMER:read...      327  \n",
       "...          ...                                                ...      ...  \n",
       "6260633       -1                                                  .        0  \n",
       "6260634       -1                                                  .        0  \n",
       "6260635       -1                                                  .        0  \n",
       "6260636       -1                                                  .        0  \n",
       "6260637       -1                                                  .        0  \n",
       "\n",
       "[6260638 rows x 15 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "cf2857fe-7286-4625-9f30-b1b9788f701e",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open_chromHMM.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "956e233f-312f-4751-91bb-fafbd06a580b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6260638"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoter_like_open_chromHMM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "4799dd9d-e93c-4de9-b232-d5333977fe54",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level', 'openChromatin_hMSC', 'chromHMM_hMSC'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to chromHMM_myotube)\n",
    "group_cols = promoter_like_open_chromHMM.columns[:10]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "8d8e279f-5444-4bfb-9eb0-89011bf8a0fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open_chromHMM.groupby(list(group_cols), dropna=False)['HiC_hMSC'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "e1617945-fc63-47a7-887b-dafce9f38cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>HiC_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "170552                       5.0   \n",
       "170553                       1.0   \n",
       "170554                       2.0   \n",
       "170555                       1.0   \n",
       "170556                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC  \\\n",
       "0                                                       .   \n",
       "1                                                       .   \n",
       "2                                                       .   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "...                                                   ...   \n",
       "170552                                                  .   \n",
       "170553                                                  .   \n",
       "170554                                                  .   \n",
       "170555                                                  .   \n",
       "170556                                                  .   \n",
       "\n",
       "                                            chromHMM_hMSC  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170552                                                  .   \n",
       "170553                                                  .   \n",
       "170554                                                  .   \n",
       "170555                                                  .   \n",
       "170556                                                  .   \n",
       "\n",
       "                                                 HiC_hMSC  \n",
       "0       P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...  \n",
       "1                                                       .  \n",
       "2                                                       .  \n",
       "3       P:1:922000:924000;D:1:906000:908000;HOMER:read...  \n",
       "4       P:1:922000:924000;D:1:906000:908000;HOMER:read...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 11 columns]"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "4123bfc4-2464-4621-a25a-f9bfc4e5eab0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>HiC_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:924750-925302;DNase-seq:chr1:924...</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "      <td>P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:927020...</td>\n",
       "      <td>P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:925610-925786</td>\n",
       "      <td>chr1:925620:927020:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:927020:935020:25_Quies</td>\n",
       "      <td>P:1:930000:932000;D:1:916000:918000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938420:939820:17_EnhW2</td>\n",
       "      <td>P:1:938000:940000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940206-940767;DNase-seq:chr1:940...</td>\n",
       "      <td>chr1:939820:940420:16_EnhW1;chr1:940420:941620...</td>\n",
       "      <td>P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:940860-941060</td>\n",
       "      <td>chr1:940420:941620:2_PromU</td>\n",
       "      <td>P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046;DNase-seq:chr1:941...</td>\n",
       "      <td>chr1:941620:942020:16_EnhW1;chr1:942020:942820...</td>\n",
       "      <td>P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:941643-942046</td>\n",
       "      <td>chr1:941620:942020:16_EnhW1;chr1:942020:942820...</td>\n",
       "      <td>P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945020:949820:7_Tx3'</td>\n",
       "      <td>P:1:944000:946000;D:1:924000:926000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945020:949820:7_Tx3'</td>\n",
       "      <td>P:1:946000:948000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:958614-958789</td>\n",
       "      <td>chr1:957620:958820:4_PromD2</td>\n",
       "      <td>P:1:958000:960000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958820:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958820:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958820:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:906000:908000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960477-960665;DNase-seq:chr1:960...</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961020:...</td>\n",
       "      <td>P:1:960000:962000;D:1:906000:908000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>DNase-seq:chr1:961371-961466</td>\n",
       "      <td>chr1:961020:961220:22_PromP;chr1:961220:961820...</td>\n",
       "      <td>P:1:960000:962000;D:1:906000:908000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:963820:7_Tx3'</td>\n",
       "      <td>P:1:962000:964000;D:1:940000:942000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:963820:7_Tx3'</td>\n",
       "      <td>P:1:962000:964000;D:1:940000:942000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "      <td>chr1:966020:967420:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:906000:908000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "      <td>chr1:966020:967420:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:906000:908000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...</td>\n",
       "      <td>chr1:966020:967420:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:906000:908000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:970220:972020:24_ReprPC</td>\n",
       "      <td>P:1:970000:972000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972020:975820:25_Quies</td>\n",
       "      <td>P:1:972000:974000;D:1:960000:962000;HOMER:read...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:976220:976620:2_PromU;chr1:976620:979820:...</td>\n",
       "      <td>P:1:976000:978000;D:1:970000:972000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:979820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1030000:1032000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:979820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1030000:1032000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:979820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1030000:1032000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999220:99982...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999820:10002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999820:10002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:1000382-1000639;ATAC-seq:chr1:99...</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU;chr1:999820:10002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:1000720-1001000</td>\n",
       "      <td>chr1:1000220:1001420:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1013101-1013955;DNase-seq:chr1:1...</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10144...</td>\n",
       "      <td>P:1:1012000:1014000;D:1:1000000:1002000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019347-1020347</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019347-1020347</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1033850-1033983</td>\n",
       "      <td>chr1:1033620:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1010000:1012000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1033850-1033983</td>\n",
       "      <td>chr1:1033620:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1010000:1012000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033620:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "      <td>P:1:1034000:1036000;D:1:1014000:1016000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1045020:1051620:8_TxWk</td>\n",
       "      <td>P:1:1044000:1046000;D:1:1018000:1020000;HOMER:...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                   openChromatin_hMSC  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3   DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4   DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "5   ATAC-seq:chr1:924750-925302;DNase-seq:chr1:924...   \n",
       "6                        DNase-seq:chr1:925610-925786   \n",
       "7                        DNase-seq:chr1:925610-925786   \n",
       "8                        DNase-seq:chr1:925610-925786   \n",
       "9                        DNase-seq:chr1:925610-925786   \n",
       "10                       DNase-seq:chr1:925610-925786   \n",
       "11                       DNase-seq:chr1:925610-925786   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14  ATAC-seq:chr1:940206-940767;DNase-seq:chr1:940...   \n",
       "15                       DNase-seq:chr1:940860-941060   \n",
       "16  ATAC-seq:chr1:941643-942046;DNase-seq:chr1:941...   \n",
       "17                        ATAC-seq:chr1:941643-942046   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                       DNase-seq:chr1:958614-958789   \n",
       "21  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:959185-959484;DNase-seq:chr1:959...   \n",
       "24  ATAC-seq:chr1:960477-960665;DNase-seq:chr1:960...   \n",
       "25                       DNase-seq:chr1:961371-961466   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...   \n",
       "29  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...   \n",
       "30  ATAC-seq:chr1:966700-966841;DNase-seq:chr1:966...   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                                                  .   \n",
       "34                                                  .   \n",
       "35                                                  .   \n",
       "36                                                  .   \n",
       "37  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999886-1000314;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:1000382-1000639;ATAC-seq:chr1:99...   \n",
       "41                     DNase-seq:chr1:1000720-1001000   \n",
       "42                     DNase-seq:chr1:1000720-1001000   \n",
       "43  ATAC-seq:chr1:1013101-1013955;DNase-seq:chr1:1...   \n",
       "44                      ATAC-seq:chr1:1019347-1020347   \n",
       "45                      ATAC-seq:chr1:1019347-1020347   \n",
       "46                     DNase-seq:chr1:1033850-1033983   \n",
       "47                     DNase-seq:chr1:1033850-1033983   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                        chromHMM_hMSC  \\\n",
       "0                           chr1:10600:96080:25_Quies   \n",
       "1                         chr1:348136:501536:25_Quies   \n",
       "2                         chr1:634820:733820:25_Quies   \n",
       "3   chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "4   chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...   \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:927020...   \n",
       "7                       chr1:925620:927020:23_PromBiv   \n",
       "8                       chr1:925620:927020:23_PromBiv   \n",
       "9                       chr1:925620:927020:23_PromBiv   \n",
       "10                      chr1:925620:927020:23_PromBiv   \n",
       "11                      chr1:925620:927020:23_PromBiv   \n",
       "12                        chr1:927020:935020:25_Quies   \n",
       "13                        chr1:938420:939820:17_EnhW2   \n",
       "14  chr1:939820:940420:16_EnhW1;chr1:940420:941620...   \n",
       "15                         chr1:940420:941620:2_PromU   \n",
       "16  chr1:941620:942020:16_EnhW1;chr1:942020:942820...   \n",
       "17  chr1:941620:942020:16_EnhW1;chr1:942020:942820...   \n",
       "18                          chr1:945020:949820:7_Tx3'   \n",
       "19                          chr1:945020:949820:7_Tx3'   \n",
       "20                        chr1:957620:958820:4_PromD2   \n",
       "21  chr1:958820:959220:3_PromD1;chr1:959220:959420...   \n",
       "22  chr1:958820:959220:3_PromD1;chr1:959220:959420...   \n",
       "23  chr1:958820:959220:3_PromD1;chr1:959220:959420...   \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961020:...   \n",
       "25  chr1:961020:961220:22_PromP;chr1:961220:961820...   \n",
       "26                          chr1:962420:963820:7_Tx3'   \n",
       "27                          chr1:962420:963820:7_Tx3'   \n",
       "28                         chr1:966020:967420:2_PromU   \n",
       "29                         chr1:966020:967420:2_PromU   \n",
       "30                         chr1:966020:967420:2_PromU   \n",
       "31                       chr1:970220:972020:24_ReprPC   \n",
       "32                        chr1:972020:975820:25_Quies   \n",
       "33  chr1:976220:976620:2_PromU;chr1:976620:979820:...   \n",
       "34                       chr1:979820:984220:24_ReprPC   \n",
       "35                       chr1:979820:984220:24_ReprPC   \n",
       "36                       chr1:979820:984220:24_ReprPC   \n",
       "37  chr1:1000220:1001420:2_PromU;chr1:999220:99982...   \n",
       "38  chr1:1000220:1001420:2_PromU;chr1:999820:10002...   \n",
       "39  chr1:1000220:1001420:2_PromU;chr1:999820:10002...   \n",
       "40  chr1:1000220:1001420:2_PromU;chr1:999820:10002...   \n",
       "41                       chr1:1000220:1001420:2_PromU   \n",
       "42                       chr1:1000220:1001420:2_PromU   \n",
       "43  chr1:1013220:1013420:1_TssA;chr1:1013420:10144...   \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "46                     chr1:1033620:1034620:24_ReprPC   \n",
       "47                     chr1:1033620:1034620:24_ReprPC   \n",
       "48  chr1:1033620:1034620:24_ReprPC;chr1:1034620:10...   \n",
       "49                        chr1:1045020:1051620:8_TxWk   \n",
       "\n",
       "                                             HiC_hMSC  \n",
       "0   P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3   P:1:922000:924000;D:1:906000:908000;HOMER:read...  \n",
       "4   P:1:922000:924000;D:1:906000:908000;HOMER:read...  \n",
       "5   P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...  \n",
       "6   P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...  \n",
       "7   P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...  \n",
       "8   P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...  \n",
       "9   P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...  \n",
       "10  P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...  \n",
       "11  P:1:924000:926000;D:1:892000:894000;GOTHIC:rea...  \n",
       "12  P:1:930000:932000;D:1:916000:918000;HOMER:read...  \n",
       "13  P:1:938000:940000;D:1:906000:908000;HOMER:read...  \n",
       "14  P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...  \n",
       "15  P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...  \n",
       "16  P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...  \n",
       "17  P:1:940000:942000;D:1:888000:890000;GOTHIC:rea...  \n",
       "18  P:1:944000:946000;D:1:924000:926000;HOMER:read...  \n",
       "19  P:1:946000:948000;D:1:906000:908000;HOMER:read...  \n",
       "20  P:1:958000:960000;D:1:906000:908000;HOMER:read...  \n",
       "21  P:1:958000:960000;D:1:906000:908000;HOMER:read...  \n",
       "22  P:1:958000:960000;D:1:906000:908000;HOMER:read...  \n",
       "23  P:1:958000:960000;D:1:906000:908000;HOMER:read...  \n",
       "24  P:1:960000:962000;D:1:906000:908000;GOTHIC:rea...  \n",
       "25  P:1:960000:962000;D:1:906000:908000;GOTHIC:rea...  \n",
       "26  P:1:962000:964000;D:1:940000:942000;HOMER:read...  \n",
       "27  P:1:962000:964000;D:1:940000:942000;HOMER:read...  \n",
       "28  P:1:966000:968000;D:1:906000:908000;GOTHIC:rea...  \n",
       "29  P:1:966000:968000;D:1:906000:908000;GOTHIC:rea...  \n",
       "30  P:1:966000:968000;D:1:906000:908000;GOTHIC:rea...  \n",
       "31  P:1:970000:972000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "32  P:1:972000:974000;D:1:960000:962000;HOMER:read...  \n",
       "33  P:1:976000:978000;D:1:970000:972000;GOTHIC:rea...  \n",
       "34  P:1:980000:982000;D:1:1030000:1032000;GOTHIC:r...  \n",
       "35  P:1:980000:982000;D:1:1030000:1032000;GOTHIC:r...  \n",
       "36  P:1:980000:982000;D:1:1030000:1032000;GOTHIC:r...  \n",
       "37  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "38  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "39  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "40  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "41  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "42  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "43  P:1:1012000:1014000;D:1:1000000:1002000;HOMER:...  \n",
       "44  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "45  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "46  P:1:1032000:1034000;D:1:1010000:1012000;GOTHIC...  \n",
       "47  P:1:1032000:1034000;D:1:1010000:1012000;GOTHIC...  \n",
       "48  P:1:1034000:1036000;D:1:1014000:1016000;HOMER:...  \n",
       "49  P:1:1044000:1046000;D:1:1018000:1020000;HOMER:...  "
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "1f5748b3-cc82-4be4-a65c-348cd6d7afbe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju\n"
     ]
    }
   ],
   "source": [
    "%cd ~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "7a57f452-5a9e-4929-9b7a-2d00c22bd30c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju/PreprocessDB/promoter_table\n"
     ]
    }
   ],
   "source": [
    "%cd ./PreprocessDB/promoter_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "80d87820-4296-4386-bb10-a57feead85f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mingju/PreprocessDB/promoter_table'"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecf543b8-b361-44f2-aec0-62e86987818c",
   "metadata": {},
   "source": [
    "# add TPM from RNA-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "8a643846-bbc5-48bf-a9f9-2075f41f2ce7",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/hMSC/day1/nf_results/star_rsem/rsem.merged.gene_tpm.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "20856a02-61d6-49d2-b701-46a7d86a184f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>hMSC_day1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>11.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>75.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>3.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>8.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78927</th>\n",
       "      <td>ENSG00000310553</td>\n",
       "      <td>ENST00000850732</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78928</th>\n",
       "      <td>ENSG00000310554</td>\n",
       "      <td>ENST00000850733</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78929</th>\n",
       "      <td>ENSG00000310555</td>\n",
       "      <td>ENST00000850734</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78930</th>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>ENST00000850735,ENST00000850736</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78931</th>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>ENST00000850818,ENST00000850819,ENST0000085082...</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78932 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)  \\\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...   \n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971   \n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...   \n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...   \n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...   \n",
       "...                ...                                                ...   \n",
       "78927  ENSG00000310553                                    ENST00000850732   \n",
       "78928  ENSG00000310554                                    ENST00000850733   \n",
       "78929  ENSG00000310555                                    ENST00000850734   \n",
       "78930  ENSG00000310556                    ENST00000850735,ENST00000850736   \n",
       "78931  ENSG00000310557  ENST00000850818,ENST00000850819,ENST0000085082...   \n",
       "\n",
       "       hMSC_day1  \n",
       "0          11.30  \n",
       "1           0.00  \n",
       "2          75.83  \n",
       "3           3.09  \n",
       "4           8.02  \n",
       "...          ...  \n",
       "78927       0.00  \n",
       "78928       0.00  \n",
       "78929       0.00  \n",
       "78930       0.00  \n",
       "78931       0.00  \n",
       "\n",
       "[78932 rows x 3 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "f873299d-3aaa-4b63-a570-53d2de783931",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map =pd.read_csv('./geneid2name.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "c1fc8990-3aa6-45cc-8bc6-0a2b7bf6c75b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000142611</td>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000157911</td>\n",
       "      <td>PEX10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000142655</td>\n",
       "      <td>PEX14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000149527</td>\n",
       "      <td>PLCH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000171621</td>\n",
       "      <td>SPSB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19469</th>\n",
       "      <td>ENSG00000198886</td>\n",
       "      <td>MT-ND4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19470</th>\n",
       "      <td>ENSG00000198786</td>\n",
       "      <td>MT-ND5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19471</th>\n",
       "      <td>ENSG00000198695</td>\n",
       "      <td>MT-ND6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19472</th>\n",
       "      <td>ENSG00000198727</td>\n",
       "      <td>MT-CYB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19473</th>\n",
       "      <td>ENSG00000274847</td>\n",
       "      <td>MAFIP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19474 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id gene_name\n",
       "0      ENSG00000142611    PRDM16\n",
       "1      ENSG00000157911     PEX10\n",
       "2      ENSG00000142655     PEX14\n",
       "3      ENSG00000149527     PLCH2\n",
       "4      ENSG00000171621     SPSB1\n",
       "...                ...       ...\n",
       "19469  ENSG00000198886    MT-ND4\n",
       "19470  ENSG00000198786    MT-ND5\n",
       "19471  ENSG00000198695    MT-ND6\n",
       "19472  ENSG00000198727    MT-CYB\n",
       "19473  ENSG00000274847     MAFIP\n",
       "\n",
       "[19474 rows x 2 columns]"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "bdbcd868-eab4-492b-a43a-a37e033c6e89",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm.merge(\n",
    "    gene_map,\n",
    "    on=\"gene_id\",\n",
    "    how=\"left\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "c10052b0-78bc-4851-8959-72d5fc46a14e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>hMSC_day1</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>11.30</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>75.83</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>3.09</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>8.02</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78927</th>\n",
       "      <td>ENSG00000310553</td>\n",
       "      <td>ENST00000850732</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78928</th>\n",
       "      <td>ENSG00000310554</td>\n",
       "      <td>ENST00000850733</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78929</th>\n",
       "      <td>ENSG00000310555</td>\n",
       "      <td>ENST00000850734</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78930</th>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>ENST00000850735,ENST00000850736</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78931</th>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>ENST00000850818,ENST00000850819,ENST0000085082...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78932 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)  \\\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...   \n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971   \n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...   \n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...   \n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...   \n",
       "...                ...                                                ...   \n",
       "78927  ENSG00000310553                                    ENST00000850732   \n",
       "78928  ENSG00000310554                                    ENST00000850733   \n",
       "78929  ENSG00000310555                                    ENST00000850734   \n",
       "78930  ENSG00000310556                    ENST00000850735,ENST00000850736   \n",
       "78931  ENSG00000310557  ENST00000850818,ENST00000850819,ENST0000085082...   \n",
       "\n",
       "       hMSC_day1 gene_name  \n",
       "0          11.30    TSPAN6  \n",
       "1           0.00      TNMD  \n",
       "2          75.83      DPM1  \n",
       "3           3.09     SCYL3  \n",
       "4           8.02     FIRRM  \n",
       "...          ...       ...  \n",
       "78927       0.00       NaN  \n",
       "78928       0.00       NaN  \n",
       "78929       0.00       NaN  \n",
       "78930       0.00       NaN  \n",
       "78931       0.00       NaN  \n",
       "\n",
       "[78932 rows x 4 columns]"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "c7f394c4-1adb-4076-a563-a5a12adf2715",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[gene_tpm[\"gene_name\"].notna()].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "7f216c85-c873-4632-b7d1-4d4b0ba39eb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm.rename(columns={'hMSC_day1':\"GeneTPM_hMSC\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "082a4e96-d8bc-449c-9b7a-dc5bf78477a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[['GeneTPM_hMSC','gene_name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "1368f749-a25d-4a9e-8b91-c7208e6deb8c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneTPM_hMSC</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11.30</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>75.83</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3.09</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.02</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19469</th>\n",
       "      <td>0.00</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19470</th>\n",
       "      <td>4.34</td>\n",
       "      <td>MSL3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19471</th>\n",
       "      <td>0.00</td>\n",
       "      <td>DUSP13B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19472</th>\n",
       "      <td>0.00</td>\n",
       "      <td>DUSP13A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19473</th>\n",
       "      <td>133.36</td>\n",
       "      <td>CAST</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19474 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneTPM_hMSC gene_name\n",
       "0             11.30    TSPAN6\n",
       "1              0.00      TNMD\n",
       "2             75.83      DPM1\n",
       "3              3.09     SCYL3\n",
       "4              8.02     FIRRM\n",
       "...             ...       ...\n",
       "19469          0.00      IL9R\n",
       "19470          4.34     MSL3B\n",
       "19471          0.00   DUSP13B\n",
       "19472          0.00   DUSP13A\n",
       "19473        133.36      CAST\n",
       "\n",
       "[19474 rows x 2 columns]"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "3428527d-0c5a-4ae1-a164-645ca7573399",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df = collapsed_df.merge(\n",
    "    gene_tpm,\n",
    "    left_on=\"Gene\",\n",
    "    right_on=\"gene_name\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['gene_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "2fe985b9-7364-4b32-8316-c4c41954e728",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>HiC_hMSC</th>\n",
       "      <th>GeneTPM_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "170934                       2.0   \n",
       "170935                       1.0   \n",
       "170936                       1.0   \n",
       "170937                       2.0   \n",
       "170938                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC  \\\n",
       "0                                                       .   \n",
       "1                                                       .   \n",
       "2                                                       .   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                            chromHMM_hMSC  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                                 HiC_hMSC  GeneTPM_hMSC  \n",
       "0       P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...          0.00  \n",
       "1                                                       .          0.00  \n",
       "2                                                       .          0.00  \n",
       "3       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16  \n",
       "4       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16  \n",
       "...                                                   ...           ...  \n",
       "170934                                                  .          4.61  \n",
       "170935                                                  .          4.61  \n",
       "170936                                                  .          4.61  \n",
       "170937                                                  .          4.61  \n",
       "170938                                                  .          4.61  \n",
       "\n",
       "[170939 rows x 12 columns]"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "b90d2d51-4cee-41bf-a7e6-724f6fba9dbd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneTPM_hMSC\n",
       "False    170939\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df['GeneTPM_hMSC'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "a7e5e241-7517-45a4-bccc-f4ab8bc75d53",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/hMSC/day1/nf_results/star_rsem/rsem.merged.transcript_tpm.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "0bb88658-ff3c-4dcd-9eaa-28ab6cde415c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>hMSC_day1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000373020</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>11.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000494424</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000496771</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000612152</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000373031</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387939</th>\n",
       "      <td>ENST00000850736</td>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387940</th>\n",
       "      <td>ENST00000850818</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387941</th>\n",
       "      <td>ENST00000850819</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387942</th>\n",
       "      <td>ENST00000850820</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387943</th>\n",
       "      <td>ENST00000850821</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>387944 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          transcript_id          gene_id  hMSC_day1\n",
       "0       ENST00000373020  ENSG00000000003       11.3\n",
       "1       ENST00000494424  ENSG00000000003        0.0\n",
       "2       ENST00000496771  ENSG00000000003        0.0\n",
       "3       ENST00000612152  ENSG00000000003        0.0\n",
       "4       ENST00000373031  ENSG00000000005        0.0\n",
       "...                 ...              ...        ...\n",
       "387939  ENST00000850736  ENSG00000310556        0.0\n",
       "387940  ENST00000850818  ENSG00000310557        0.0\n",
       "387941  ENST00000850819  ENSG00000310557        0.0\n",
       "387942  ENST00000850820  ENSG00000310557        0.0\n",
       "387943  ENST00000850821  ENSG00000310557        0.0\n",
       "\n",
       "[387944 rows x 3 columns]"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tx_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "8eb811f2-0e3b-4f78-a791-793ac0b104cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = tx_tpm.rename(columns={'hMSC_day1':\"TranscriptTPM_hMSC\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "bda4f8b6-aaf2-4bdf-a01b-77ea2f127df9",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df['Transcript'] = collapsed_df['Transcript'].str.split(\".\").str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "531da245-0367-4067-887e-c57845a18352",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>HiC_hMSC</th>\n",
       "      <th>GeneTPM_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "170934                       2.0   \n",
       "170935                       1.0   \n",
       "170936                       1.0   \n",
       "170937                       2.0   \n",
       "170938                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC  \\\n",
       "0                                                       .   \n",
       "1                                                       .   \n",
       "2                                                       .   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                            chromHMM_hMSC  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                                 HiC_hMSC  GeneTPM_hMSC  \n",
       "0       P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...          0.00  \n",
       "1                                                       .          0.00  \n",
       "2                                                       .          0.00  \n",
       "3       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16  \n",
       "4       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16  \n",
       "...                                                   ...           ...  \n",
       "170934                                                  .          4.61  \n",
       "170935                                                  .          4.61  \n",
       "170936                                                  .          4.61  \n",
       "170937                                                  .          4.61  \n",
       "170938                                                  .          4.61  \n",
       "\n",
       "[170939 rows x 12 columns]"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "e6f054b6-3040-4e48-85d1-e9b53eeb790f",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df = collapsed_df.merge(\n",
    "    tx_tpm,\n",
    "    left_on=\"Transcript\",\n",
    "    right_on=\"transcript_id\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['transcript_id','gene_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "7aea43e8-d31f-4873-a953-e2cbbaf463f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>HiC_hMSC</th>\n",
       "      <th>GeneTPM_hMSC</th>\n",
       "      <th>TranscriptTPM_hMSC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "      <td>2.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                            NaN   \n",
       "1                            NaN   \n",
       "2                            NaN   \n",
       "3                            5.0   \n",
       "4                            5.0   \n",
       "...                          ...   \n",
       "170934                       2.0   \n",
       "170935                       1.0   \n",
       "170936                       1.0   \n",
       "170937                       2.0   \n",
       "170938                       2.0   \n",
       "\n",
       "                                       openChromatin_hMSC  \\\n",
       "0                                                       .   \n",
       "1                                                       .   \n",
       "2                                                       .   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                            chromHMM_hMSC  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                                 HiC_hMSC  GeneTPM_hMSC  \\\n",
       "0       P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...          0.00   \n",
       "1                                                       .          0.00   \n",
       "2                                                       .          0.00   \n",
       "3       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16   \n",
       "4       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16   \n",
       "...                                                   ...           ...   \n",
       "170934                                                  .          4.61   \n",
       "170935                                                  .          4.61   \n",
       "170936                                                  .          4.61   \n",
       "170937                                                  .          4.61   \n",
       "170938                                                  .          4.61   \n",
       "\n",
       "        TranscriptTPM_hMSC  \n",
       "0                     0.00  \n",
       "1                     0.00  \n",
       "2                     0.00  \n",
       "3                     0.00  \n",
       "4                     2.23  \n",
       "...                    ...  \n",
       "170934                0.00  \n",
       "170935                0.03  \n",
       "170936                0.03  \n",
       "170937                0.00  \n",
       "170938                0.00  \n",
       "\n",
       "[170939 rows x 13 columns]"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "e483d2e3-4c88-43de-b2ba-6d090686ed66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TranscriptTPM_hMSC\n",
       "False    170939\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df['TranscriptTPM_hMSC'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f80d806-da69-4112-a3c3-48b3194fa03e",
   "metadata": {},
   "source": [
    "# update the promoter evidence Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "294a851c-f161-4064-a5a8-2865b1780689",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df[\"transcript_support_level\"] = (\n",
    "    collapsed_df[\"transcript_support_level\"]\n",
    "        .fillna(0)\n",
    "        .astype(int)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "eb83069e-df7b-40c2-b1d1-029b1151cdc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def classify_promoter_type(row):\n",
    "    score = 0\n",
    "    sources = []  # track where score comes from\n",
    "\n",
    "    # Open chromatin evidence (Essential)\n",
    "    open_chrom = row['openChromatin_hMSC']\n",
    "    if isinstance(open_chrom, str) and open_chrom != \".\":\n",
    "        if 'ATAC' in open_chrom and 'DNase' in open_chrom:\n",
    "            score += 3\n",
    "            sources.append(\"OpenChromatin:ATAC+DNase(+3)\")\n",
    "        else:\n",
    "            score += 2\n",
    "            sources.append(\"OpenChromatin:ATAC/DNase(+2)\")\n",
    "\n",
    "    # ChromHMM evidence (Essential)\n",
    "    chromhmm = row['chromHMM_hMSC']\n",
    "    if isinstance(chromhmm, str) and chromhmm != \".\":\n",
    "        active_states = ['TssA', 'PromU', 'PromP', '1_Tss', '2_Prom']\n",
    "        if any(state in chromhmm for state in active_states):\n",
    "            score += 2\n",
    "            sources.append(\"ChromHMM:ActivePromoter(+2)\")\n",
    "\n",
    "    # Expression evidence\n",
    "    transcript_tpm = row.get('TranscriptTPM_hMSC', 0)\n",
    "    gene_tpm = row.get('GeneTPM_hMSC', 0)\n",
    "\n",
    "    if transcript_tpm > 1:\n",
    "        score += 2\n",
    "        sources.append(\"TranscriptTPM>1(+2)\")\n",
    "    elif transcript_tpm > 0:\n",
    "        score += 1\n",
    "        sources.append(\"TranscriptTPM>0(+1)\")\n",
    "    elif gene_tpm > 1:\n",
    "        score += 1\n",
    "        sources.append(\"GeneTPM>1(+1)\")\n",
    "\n",
    "    # Transcript support level\n",
    "    tsl = str(row['transcript_support_level'])\n",
    "    if tsl in ['1', '2']:\n",
    "        score += 1\n",
    "        sources.append(\"TSL1-2(+1)\")\n",
    "\n",
    "    # Optional Hi-C\n",
    "    # hic = row.get('HiC_hMSC')\n",
    "    # if isinstance(hic, str) and hic != '.':\n",
    "    #     score += 1\n",
    "    #     sources.append(\"HiC(+1)\")\n",
    "\n",
    "    # Final classification\n",
    "    if score >= 6:\n",
    "        promoter_type = 'very_high'\n",
    "    elif score >= 4:\n",
    "        promoter_type = 'high'\n",
    "    elif score >= 2:\n",
    "        promoter_type = 'moderate'\n",
    "    else:\n",
    "        promoter_type = 'low'\n",
    "\n",
    "    return promoter_type, score, \"; \".join(sources)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "70b02bca-4278-481d-83dd-8a830487b7ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df[[\"Type\", \"score\", \"score_source\"]] = (\n",
    "    collapsed_df\n",
    "        .apply(classify_promoter_type, axis=1, result_type=\"expand\")\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "c09bb180-bcf7-47eb-9b86-d19bb6b4fc4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_hMSC</th>\n",
       "      <th>chromHMM_hMSC</th>\n",
       "      <th>HiC_hMSC</th>\n",
       "      <th>GeneTPM_hMSC</th>\n",
       "      <th>TranscriptTPM_hMSC</th>\n",
       "      <th>Type</th>\n",
       "      <th>score</th>\n",
       "      <th>score_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>5</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:906000:908000;HOMER:read...</td>\n",
       "      <td>10.16</td>\n",
       "      <td>2.23</td>\n",
       "      <td>very_high</td>\n",
       "      <td>6</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.00</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>GeneTPM&gt;1(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.03</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>TranscriptTPM&gt;0(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.03</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>TranscriptTPM&gt;0(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.00</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>GeneTPM&gt;1(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.61</td>\n",
       "      <td>0.00</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>GeneTPM&gt;1(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level  \\\n",
       "0                              0   \n",
       "1                              0   \n",
       "2                              0   \n",
       "3                              5   \n",
       "4                              5   \n",
       "...                          ...   \n",
       "170934                         2   \n",
       "170935                         1   \n",
       "170936                         1   \n",
       "170937                         2   \n",
       "170938                         2   \n",
       "\n",
       "                                       openChromatin_hMSC  \\\n",
       "0                                                       .   \n",
       "1                                                       .   \n",
       "2                                                       .   \n",
       "3       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "4       DNase-seq:chr1:923760-923940;DNase-seq:chr1:92...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                            chromHMM_hMSC  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                                 HiC_hMSC  GeneTPM_hMSC  \\\n",
       "0       P:1:64000:66000;D:1:62000:64000;HOMER:readcoun...          0.00   \n",
       "1                                                       .          0.00   \n",
       "2                                                       .          0.00   \n",
       "3       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16   \n",
       "4       P:1:922000:924000;D:1:906000:908000;HOMER:read...         10.16   \n",
       "...                                                   ...           ...   \n",
       "170934                                                  .          4.61   \n",
       "170935                                                  .          4.61   \n",
       "170936                                                  .          4.61   \n",
       "170937                                                  .          4.61   \n",
       "170938                                                  .          4.61   \n",
       "\n",
       "        TranscriptTPM_hMSC       Type  score  \\\n",
       "0                     0.00        low      0   \n",
       "1                     0.00        low      0   \n",
       "2                     0.00        low      0   \n",
       "3                     0.00       high      5   \n",
       "4                     2.23  very_high      6   \n",
       "...                    ...        ...    ...   \n",
       "170934                0.00   moderate      2   \n",
       "170935                0.03   moderate      2   \n",
       "170936                0.03   moderate      2   \n",
       "170937                0.00   moderate      2   \n",
       "170938                0.00   moderate      2   \n",
       "\n",
       "                                             score_source  \n",
       "0                                                          \n",
       "1                                                          \n",
       "2                                                          \n",
       "3       OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "4       OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "...                                                   ...  \n",
       "170934                          GeneTPM>1(+1); TSL1-2(+1)  \n",
       "170935                    TranscriptTPM>0(+1); TSL1-2(+1)  \n",
       "170936                    TranscriptTPM>0(+1); TSL1-2(+1)  \n",
       "170937                          GeneTPM>1(+1); TSL1-2(+1)  \n",
       "170938                          GeneTPM>1(+1); TSL1-2(+1)  \n",
       "\n",
       "[170939 rows x 16 columns]"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "1f6223fb-d6ca-4185-bee6-831ea759c33b",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/hMSC/promoter_like_regions_hMSC_openchromatin_chromHMM_HiC_level.tsv',sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "cbb225f2-fa98-49ac-b82e-d58f7874eaf6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Type\n",
      "very_high    41.688556\n",
      "low          27.344842\n",
      "moderate     17.991213\n",
      "high         12.975389\n",
      "Name: proportion, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "type_percentages = collapsed_df['Type'].value_counts(normalize=True) * 100\n",
    "print(type_percentages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "6b74bbe4-f81c-46ba-948b-fa5460e93be0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAtIAAAIoCAYAAABEYwzVAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjgsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvwVt1zgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAXzNJREFUeJzt3QeYVNX9//EvvXcRAopSBGlSjBQLIEUIJUD8oaAI0oz0KCgoigGRohSBUAQx0hGMEEFQ+UnXQBKaKKBSBCSEJr23//M5/+fOb2Z2dtm9u2x9v55nn9m5c+bO3Z27s5858z3npLtx48YNAwAAABAn6ePWHAAAAABBGgAAAPCJHmkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAEsQnn3xiZcqUCXxVrFjRHnroIXvmmWfsvffes+PHj0e5z/jx413buLhw4YK734YNG+J0v0iPVbduXfvjH/9oCWnx4sX24YcfRrxNj6/jSM7+8Y9/2B/+8AerXLmyO97//d//jdJGz2nwcx3dV3L8Wf/9739b79697ZFHHrEKFSrY/fffb61bt7Y5c+bY+fPnE/T5To70d6PnJq5/PwAiyxjNdgDwZdiwYVaiRAm7evWqC88bN260qVOn2gcffGBjxoyxBx98MNC2VatWLtDENUj/5S9/sR49elj16tVjfT8/j+XHkiVL7KeffrJnn302ym0fffSRFS5c2JKrGzdu2J/+9Ce7++67bdKkSZYtWzYrXrx4lHZvvPGGnT17NnB91apVrr333HuS2886btw4mzBhglWpUsWF6WLFirnzafPmze6c+vnnn+3VV19NsOc7OSpfvrw7D0uVKpXUhwKkCgRpAAnqnnvucb3RnoYNG7qQ8dRTT7nw++WXX9ptt90WCFq3OmwpKCkQJsZj3Yx6eZOzI0eO2MmTJ61+/fpWs2bNaNuFh7A9e/ZEfO6Tk2XLlrkQ/T//8z82ZMgQS5cuXeC22rVrW+fOnW3Lli2WWl25csX9zDlz5kz25yGQklDaAeCWK1KkiPXr18/OnTtn8+bNi7HcQqUFKh1Qb/N9991nderUsZ49e7pA/MsvvwQCnnoQvRKC/v37h+zv+++/t169etkDDzxgDRo0iPaxPMuXL7dmzZq5EFivXj2bMWNGxLIVPX5MH5PruNU7e/DgwZASB0+kcocff/zRunbt6o5Vj9+8eXNbuHBhxMdR76d69R9++GGrWrWqe4PihdjYlDS0b9/e9cZWqlTJlTPoWIOfi1q1arnvR44c6R5PpS9+LFq0yN1fPb3h9LypV/Tw4cOB31nTpk3d8T3xxBPuOdcnB++++65du3Yt5L6XL1+2iRMnWqNGjVxZRo0aNeyVV16xX3/99abHpPvlyZPHXnvttZAQ7VHA1O/VM3v2bHv66afd+abgqfNDn6wokHpu9nzH9njVbvjw4a4USs+NHve7775zv3/v3PZzvuh50H71+1Tbffv2RVvasW3bNnv++eetWrVqrm2LFi1s6dKlIW30NzhixAh3XGqjtioD0nkJpFX0SANIFOr1y5AhgwtM0VFQVc3yb3/7W3vrrbcsd+7cLnCtXbvWBZjbb7/d3n//fdd7qJ5FlWtI/vz5Q/aj4N24cWMXFm9W97pjxw4bOnSo6y1XT7lqXvXYerxOnTrF6WdUycPrr79uBw4ccIHxZhSCdYwFChSwAQMGWL58+ezTTz914enYsWPWpUuXkPajR492AVrHp9IKBV6FKgUe/W6j889//tM6duxopUuXdvfNnDmzzZ071wUn7VO/K/0u7733Xvd78MKt2vmh/enYFEYV3D0q91FZgXq8CxUqFNh+9OhRe+GFF+y5555zb4C8UpHTp0/bwIEDXZvr169bt27dXKmQnhf9HhRg9Qbg22+/tb/97W+WNWvWaHvaFUB1XPp0Ijb279/vfgd33HGHZcqUyXbu3GmTJ092z5lKWG72fMfleBWu9RzqvFbY3rVrl3segstn/J4vehMwaNAgS58+vbuf2oVbv369e2yF+D//+c+WK1cudzx6Ti5evOjCsujn1uOp/Kds2bIuWOv3qk8xgLSKIA0gUWTPnt3941eoiY56ki9dumQvv/yyC3Ue9QZ61JspKtOI7iNq9aYpkMWGjkc9d97jKfCrx1A9iSpHiW3w8koeFP4VQGPz8bnClwK7esB/85vfBB5fAVJlCApNCjXB+1dA9SgcKdSoNzGmxxs1apQ7rpkzZ1qOHDnctkcffdT9ntTD+Lvf/c79PhV0RccSn4//9fM/+eSTbpCpQqICnKisR7/vtm3bhrRXENPvW58GiHqGdR4o7Cvg6RMNlWboDZWC6GOPPRa4r543vanSpwZ6viI5dOiQu1Qoji0dd3Ao1pu7vHnzuhpqBVf1bsf0fMf2eBWa1aOrENynTx/XRj3TelP34osvxut8UQ246sJvRkFbZTnTp0+3jBn/fyxQL/aJEydcGNd5onNNnzDo2ILrwfWJEZCWUdoBIFEHs8VEvVzq/VMvnz6uVk+fH8HB5WYUIIJDu6gnUr2BCva3knoCVTrghSJPy5YtA4PggoWXWnhlBP/5z3+ifQz1yG/dutXVqnshWtSD/fvf/97++9//xro8JC7atGnjLufPnx/Yph5q9YqrLCGYjssL0cHPgQLsv/71L3d95cqVLrTqDYACv/elc6ZgwYKu1z0hbd++3fXYq8RIj6E3cCpPUrmJBiXeTGyP17vUm5lger68UOv3fInN34HKPfT8e29Wg49VpT76tGDv3r3uNpVzrFmzxr2ZU2mIequBtI4eaQCJQoFOPY8KUtFRD5qmElP5xuDBg9197rzzTldqoPre2FIJSGx5Ax8jbbvVH1lr/wpV0R1/+OOrRzSYV3oRU6BRb6XewMTlcRKCfocqpVAph0o2NLOFynr0vEZqG90279g0A4x+FtUaR6Le0+h4wTO8xj06emOiOmXNWKIe6KJFi1qWLFlcSYaOPzYBMrbH6/184b8Dhejw5zuu50uktuG8Ug99MqGvmI5V9eX65EJlH6oX1+9Enx7oEyTN9AKkRQRpAIlCda/qzdMApZjoI3R9qa0GXKkcQTXMChpNmjRJ8OOKVDPqbfOCjAKDNygstuEtNrR/9fiF88pfVAoTX+oV1cfyt/pxImnXrp39/e9/t6+++sqVOehYgst04vIc6Bj1vd5kRRLc2x4paOoN3Ndffx2YxSUmmjtbb+JUlqEQ7VGddGzF9ni9n08/b3DduHqEI72RisvzGGlQZaTjFI1N8AbmhvOmQFR5lkqm9KXjVe+0yobUc//555/f9LGA1IjSDgC3nHr43n77bVe/qTrO2FDpgQY/aUCXeGUWsemFjQv1lIYHJNWsKuh49dhemPrhhx9C2q1YsSLK/nR8sT02fUyvj+u9GSw8Cp8KewkxTZnCj36Pmpkk+LhUNqGBY+phjDRXdEJQb6wGG6r3UoM4VYKg4wmn2VwUtsOfA70B8MpAVIurYKnjVolB+Ffw/NWRaODfqVOn3NR3kUqMdAzr1q0LCaDBgy11n+AylZs937E9Xu/nC58h44svvgjUrN/K80XHod5k/Q1EOk59aUaTcHpjq0GIenOr0g+9QQHSInqkASQoBVP1JisEaNCePs7XwCoFYw2WCp9hI5gGlykoKITo43gNONPsBuIt5KJ/6gq2Cl4KFhr0pV61uAwkC++t1MwXmiVBH4UrXKrnsm/fvoGeS4UJhU29GdDPpp5V9VpqRoZw6vnUoDqtlKcgqVAW3dzK3bt3d7W06rnV9/pZFDjVe//SSy+FDByLDw1a06wdehxdqg5dx6fnSoPJYtNz6ZceU7M/6DGiGwyonlbNFqFBgQp1q1evdqFVddYaaCgKbPrdqExEpT6aJk8/h2q8Va+rGuvoelS9GmTNMKFBjaoJ1oA/b0EWlWxoWkaVoqhUQeea9q3fmwY76pMInZsq1Yjt8x3b41WNvurB//rXv7q/Ec3aoedF1/X8Bz83t+p80WBDDXbU7CJ6s6Oecb3p2L17t3sD6w1Y1Mwu+ttUbb4eW7crxOvNUlwG5QKpCUEaQILyZjtQaFDgLFmypPsnrX/CMYVo0UAshVh9pK6PsNV7qaCiqdCC5/jVFG4KtQrACjn656/5cv3QY6pnTY+pQWQK1voZgmcmUMDR1Gdvvvmm6yFXL6SCkqZmU1AKppCjIKT5ns+cOeN6MsN7soN7AxXgFGa92lv9vjTNmDflWEJQOY1qz/Uz6mdTL6kGWOr3qsFwt5KmutPvS4P2oquj1RsY/S5Vo6uwq5CmcgFNYxj8HOh4NWOFwtuUKVPcNvWoq1c3ptp7j1YzVEhWuZCeH/UYq2xHYbZDhw5uphHRc6Dfleay1jEo6Cvs6pwIn2Iuuuc7Lser51u/g48//tg9Tzon9dgK8fobutXni8L7ggUL3DmuMiq9YdDPrH0HD4JUO30Ko9k99AZEgVszeui5AtKqdDduNoweAACfFLz0hkdBUlO1hVNvrWrNWdQj1KZNm1yPvGbIiFRXDiB5oEcaAJDgND+yFiBRL7N6WL1VExGVPoXR1HUqDVEPuXq09cZDPfhxmcoRQOIjSAMAEpzqbtWrWq5cOVd2cyvrsFM61f0rTKsMRIMeVfOvNx6q0fZmjAGQPFHaAQAAAPjA9HcAAACADwRpAAAAwAeCNAAAAOADgw0TmUZma8ZBzbELAACA5OfKlStukLQWHIoJQTqRKUQzdTcAAEDyFdusRpBOZF5PdHRLBgMAACBpbdu2LVbtqJEGAAAAfCBIAwAAAARpAAAAIHHQIw0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0bolz585ZrVq1rEyZMrZt27bA9q+//tr69Olj9evXd7cNHjw4Vvv75JNPXPtIX506dQq0O3bsmHXu3NmqVq1qbdq0sX379oXs5+TJk1azZk37/vvvE/CnBQAAaVHGpD4ApE4TJ060a9euRdm+Zs0a27Fjhz3wwAN26tSpWO+vTp069tFHH4Vs+/nnn61fv34usHuGDh3qHnfcuHE2depU69+/v82dOzdw+7vvvutCfPny5X3/bAAAAEKQRoLbvXu3zZkzx4XcN954I+Q2bXvllVfc9xs2bIj1PvPnz+++gq1du9YyZMhgjRs3DunxVoC+7777LFeuXPbEE0/Y+fPnLXv27LZz505btmyZ+wIAAIgvSjuQ4N566y1r3bq1FS9ePOoJlz7hTrklS5ZYjRo1rGDBgoFtV65csaxZs7rvvUtt846re/fuUQI5AACAHwRpJKjPP//c9fwqsN5KqrtWaUfTpk1DtlesWNH1hqtsZPbs2VasWDHLkyePffbZZ3bixAl76qmnbulxAQCAtIPSDiSYCxcu2PDhw+3FF1+0nDlz3tLfrHqjs2TJYo899liU0pHnnnvO1UWrtGP8+PHuuN555x0bNmyYZczIKQ8AABIGqQIJZtKkSVagQAH7wx/+cEt/q9evX7elS5e6AYjhgb1cuXK2cuVKO3DggBUtWtSF7TFjxriaac3WodsU9jW7x6OPPmqDBg2yHDly3NLjBQAAqROlHUgQBw8etA8++MB69eplZ8+etdOnT7tBfqJLTYeXUDRI8ciRI9asWbOIt2fKlMlKlCjhQrQCtXqn1VN9/Phx11veo0cPW758ue3du9eFfwAAAD/okUaC+OWXX9ygPpVVhGvXrp1VqlTJ5s+fnyCPtXjxYle2Ubt27Zu2VTnHM88843qnv/rqK8ucOXMggLds2dLNTw0AAOAHQRoJomzZsjZjxoyQbZovWkFW5RMaBJgQLl++7HqTVRutUBwTTYWnYxg9enRgm8K+5pnWtHlejzkAAIAfBGkkiNy5c1v16tUj3qbFT7wFUFQC4q10qEGA+/fvdzN9SKNGjQL3adCggRUpUsSmT58esq/Vq1e7spHoyjo8V69eddPdqaTDmwavcuXKrr565MiRVq1aNTe7R/Ac1AAAAHFBkEaiUn2ztyCLt6iKvuSHH34IbFevsUJvpLIOzRsdXWj3zJo1y2677baQcK6BkKNGjbK3337blZlosGK3bt0S6CcDAABpTbobN27cSOqDSEu83tiEKnUAAABA0uQ1Zu0AAAAAfCBIAwAAAD4QpFOY69epxElLeL4BAEi+GGyYwqRPn84mzP3aDh45ldSHglus6O15rHubh/g9AwCQTCV5kNZiGf/85z8j3qb5f5s0aRKY9kxLPe/evdsKFy5szz77rD399NNR7jNt2jSbPXu2HT161EqXLm0vv/xylBketPKeZm744osv3LzEuv311193i3YE08p3Q4YMsY0bN1q2bNncsfTt2zcwnVpSUYj++eCJJD0GAACAtC7Jg/Qbb7zhgm0wzR385ZdfWs2aNd31zZs3u2nKmjdvbv3797dNmza5gKsFOVq1ahUSohW2X3jhBStXrpwtWLDAunTp4i7LlCkTaNenTx/7/vvvXXjOmTOnjRs3zjp06GCffvppICRrruL27du7uYx1+6+//uoWFzl58qSbhxgAAABpW5IH6VKlSkXZpqD70EMPWf78+d31CRMmuGA8dOhQd71GjRp26NAhGzt2rD3++OOWPn1617M8adIktxx1p06dXDstuqGFOyZPnuwCtmzdutVWrVplU6ZMCSwxrZ5rLQCycOFCa9Omjds2b948F6YXLVoUOA6thqce6a5du1rJkiUT6TcEAACA5CjZDTZUb/Mvv/wSWLlOAXn9+vWBEg+Pblf5xvbt2wP3O3PmjDVt2jTQRsFXK9epLMSbLlvfaxW+WrVqBdqp17lq1aruNs+aNWtcj7gXoqVhw4auFzy4HQAAANKmJO+RDrdkyRJXj1yvXj13XUtIX7lyxUqUKBGxJ1s10xUqVHCXEt5OPcfnzp2zw4cPu9pqtStevLilS5cuyv7WrVsXuK526u0OphBdrFixwGP5pVB//vz5ON9Px6zfDdIWLaXOukkAACQe/d8Nz4rJPkhfvXrVPv/8cxeis2fP7radOvX/Z6dQL3Iw77p3u8owFHTDBwLmyZPHXaq2WUFa7XLlyhXlsbU/b1/e/sIfM1I7P/TGYMeOHXG+n0K0SlyQtmjQq8I0AABIPMqVKSpIf/3113b8+PGQ8gxPdO8KgrdHauP15N2sXUzbw/cXm3YxyZQpU8Ta8JuJ7+MiZdInKPRIAwCQeHbt2hWrdhmTW1lH3rx57eGHH47SoxzeC6weY/F6jXV56dIl95UlS5Yo7bz9qJ0GKoYL74HW9959g6kOO74DDRWIvR534GYo5wEAIHHFtvMy2Qw2vHjxon311VfWqFEj12PrUU2yru/ZsyfiOwUv1HqX4fXLup4jRw4rVKhQoJ0+Kg/v4dP+ggOyvg/flwY+qmabGTsAAACQbIL0ihUr3KBAb7aO4PoUTXe3bNmyKL3XBQsWDNQMa9YN1T4vXbo00ObatWvufprmzntnoe/V07x27dpAO/VQa9YPbzo80awemi3kxIn/W/hk+fLlLkwHtwMAAEDalGxKOxYvXuymobv//vuj3Na9e3dr27atvfbaay5oK/RqkZXBgwe7OaS9wK35nTVftKas8xZkOXDggFsh0VOpUiWrU6eODRgwwC3uogVZNB+1VjVs2bJloF3r1q1t1qxZbiEYfal2e/jw4e7x6ZEGAABAsgjSqn9WD7FWEoxUk1KlShWbOHGiC8RaIEWzbyhUB69qKB07dnQlGzNnzrRjx465hVa08ErwqoYyatQoGzFihA0aNMjNoKElwsePHx8y44dqpLXColZQ7Nmzp7tNgyC1IAsAAACQ7gbTASSqbdu2ucuKFSv63serY5fazwf/r+QEqdPdRfPZ0N6Nk/owAABIc7bFMq8lmxppAAAAICUhSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAJCSg/SCBQvs97//vVWsWNFq1qxpzz//fMjtq1evthYtWrjbGzRoYLNnz464n2nTplndunVdu8cff9w2bNgQpc3Zs2dt4MCBVr16datSpYp7rIMHD0Zpt3fvXuvUqZNVrlzZHdOQIUPs4sWLCfhTAwAAIKVKFkF6/PjxNnz4cGvWrJkLwoMHD7bbb789cPvmzZutW7duVq5cOZs6daq1bNnShVqF72C675gxY+zpp5+2KVOm2F133WVdunSxH374IaRdnz59bMWKFfb666+79keOHLEOHTqEhOTTp09b+/bt7dy5czZu3Djr16+fLV682F577bVE+I0AAAAgucuY1Aewe/dumzRpkgu+Dz/8cGC7ep09EyZMcCF66NCh7nqNGjXs0KFDNnbsWNfrnD59ert8+bLbT7t27VwvslSrVs2F88mTJ7vALFu3brVVq1a5x6tdu7bbVrp0afd4CxcutDZt2rht8+bNc2F60aJFlj9/frctQ4YM1rdvX+vatauVLFkyEX9LAAAASG6SvEf6k08+sTvvvDMkRAdTQF6/fr01adIkZLsC8tGjR2379u3u+qZNm+zMmTPWtGnTQBsF38aNG7uykBs3brht+j537txWq1atQLsiRYpY1apV3W2eNWvWuHIOL0RLw4YNLXPmzCHtAAAAkDYleY+0eojVI6xe51mzZrkwrJrkAQMGWNmyZW3//v125coVK1GiRMj9SpUqFejRrlChgruU8HbqOVZ5xuHDh61w4cKuXfHixS1dunRR9rdu3brAdbVTb3cwhehixYoFHssvhfrz58/H+X465mzZssXrsZHyXLhwIfBGEAAA3Hr6vxueFZNlkFav8vfff28//fSTDRo0yDJlymR/+ctfXM3yl19+aadOnXLt1IsczLvu3a4yDAXdrFmzhrTLkyePuzx58qQL0mqXK1euKMeh/Xn78vYX/piR2vmhNwY7duyI8/0UolXigrRFg14VpgEAQOJRrkz2QdrrndWAw3vuucdtK1++vNWrV88++ugjV3Ih0b0rCN4eqY3Xk3ezdjFtD99fbNrFRG8WvB71uIjv4yJl0ico9EgDAJB4du3aFat2SR6k1WN82223BUK0aMYOlWjoh3j00UfdtvBeYPUYi9drrMtLly65ryxZskRp5/VMq50GKoYL74HW9959g6n0JL4DDRWIs2fPHq99IO2gnAcAgMQV287LJB9sGF0oVQ+cZuNQTbJ6cPfs2RPxnYJ3f+8yvH5Z13PkyGGFChUKtNNH5eE9fNpf8LHo+/B9aeCjaraZsQMAAABJHqTr1Kljx44dsx9//DGwTQMDFZzLlCnj6lM03d2yZctC7rdkyRIrWLBgoGZYJSCqfV66dGmgzbVr19z9NM2d985C36unee3atYF26qHWrB/edHiiWT00W8iJEycC25YvX+7CdHA7AAAApE1JXtqh+ZtVE92zZ0/r3bu3C86awUPTzj3xxBOuTffu3a1t27ZuMRRNe6fQq8VYtHCLeq1F99P8zpovWvdVwFabAwcO2OjRowOPV6lSJRfeNStI//79LWfOnG4+6qJFi7qFXjytW7d2s4hoIRh9HT9+PLBoDD3SAAAASHcjGYxiUkjVYiuan/nq1av2wAMP2CuvvBIylZ1uUyBWuYVm39CsHlrBMJh+FK1uqOXD1cutafVeeukl16MdvkT4iBEj7IsvvnAzaGipcK1yqDAdTCUgWkFx48aNbjYQzVGtBVnCZwaJi23btrlLLWHu16tjl9rPB/+vpxyp091F89nQ3o2T+jAAAEhztsUyryWLIJ2WEKQRWwRpAACSd15L8hppAAAAICUiSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAJASg/Qnn3xiZcqUifI1cuTIkHarV6+2Fi1aWMWKFa1BgwY2e/bsiPubNm2a1a1b17V7/PHHbcOGDVHanD171gYOHGjVq1e3KlWq2PPPP28HDx6M0m7v3r3WqVMnq1y5stWsWdOGDBliFy9eTMCfHgAAAClVRksm3n//fcuVK1fgeqFChQLfb9682bp162bNmze3/v3726ZNm1yozZw5s7Vq1SokRI8ZM8ZeeOEFK1eunC1YsMC6dOniLhXOPX369LHvv//eXn/9dcuZM6eNGzfOOnToYJ9++qllzZrVtTl9+rS1b9/eihQp4m7/9ddfbdiwYXby5MkoIR8AAABpT7IJ0uXLl7f8+fNHvG3ChAkuGA8dOtRdr1Gjhh06dMjGjh3rep3Tp09vly9ftkmTJlm7du1cL7JUq1bNmjVrZpMnT3YBW7Zu3WqrVq2yKVOmWO3atd220qVLu17uhQsXWps2bdy2efPmuTC9aNGiwHFlyJDB+vbta127drWSJUsmyu8FAAAAyVOSl3bcjALy+vXrrUmTJiHbFZCPHj1q27dvd9fVS33mzBlr2rRpoI2Cb+PGjV1ZyI0bN9w2fZ87d26rVatWoJ16natWrepu86xZs8aVcwSH+4YNG7pe8OB2AAAASJuSTY+0AvCJEydcqH3iiSesc+fOLgjv37/frly5YiVKlAhpX6pUKXe5e/duq1ChgruU8HbqOT537pwdPnzYChcu7NoVL17c0qVLF2V/69atC1xXO/V2B1OILlasWOCx/FKoP3/+fJzvp2POli1bvB4bKc+FCxcCbwQBAMCtp/+74VkxWQbpggULWs+ePa1SpUrugFesWGHvvvuuC74aEHjq1CnXTr3Iwbzr3u0qw1DQ9WqcPXny5HGXqm1WkFa74Frs4P15+/L2F/6Ykdr5oTcGO3bsiPP9FKJV4oK0RYNeFaYBAEDiUa5M9kH6kUcecV+ehx9+2LJkyWLTp093s2l4ontXELw9UhuvJ+9m7WLaHr6/2LSLSaZMmQI96nER38dFyqRPUOiRBgAg8ezatStW7ZI8SEfyu9/9zj744APXa1u0aFG3LbwXWD3G4vUa6/LSpUvuS0E8vJ3XM612GqgYLrwHWt979w2mOuz4DjRUIM6ePXu89oG0g3IeAAASV2w7L5P9YEPVJKsHd8+ePRHfKXih1rsMr1/W9Rw5cgSm01M7fVQe3sOn/QUHZH0fvi8NfFTNNjN2AAAAIFkG6aVLl7qBhqoHVn2KprtbtmxZSJslS5a4+mqvZlizbqj2Wff1XLt2zd1P09x57yz0vXqa165dG2inHmrN+uFNhyea1UOzhWgApGf58uUuTAe3AwAAQNqU5KUdmvNZQVlzOctXX31l8+fPd/NBKyhL9+7drW3btvbaa6+5ae8UerXIyuDBg90c0qLArfmdNV+0pqzzFmQ5cOCAjR49OvB4GtRYp04dGzBggFvcRQuyaD5qlZC0bNky0K5169Y2a9YstxCMvo4fP27Dhw93j0+PNAAASEjq+Fu8eLFbME7lrHfeeadb20J5xMs6wYvLhVMH4e233x7xNq3o/Oqrr9p3331nx44dc+WlmvGsV69edt999wXa6TZv4Ts9lnLPXXfdFbhdEzeo/FaL6Gn9DySDIK2BVB9//LH997//tevXr9vdd9/tnuxnnnkm0EbLeE+cONEFYi2Qotk3FKqDVzWUjh07upKNmTNnupNB4VwLr4SfeKNGjbIRI0bYoEGD3AwaWip8/PjxITN+qEZaAx61gqJmFdFtmqJPC7IAAAAkpL/+9a9uCuCXX37ZChQoYBs2bLC33nrLdQj269fPtfnoo4+i3E+3aSxNdCFalHU0fkx55je/+Y0b76WMoxWcP/nkE5fFRAvf6dN8reg8depUF6rnzp0b2I9mVatfvz4hOki6G0wHkKi2bdvmLitWrOh7H6+OXWo/H/y/khOkTncXzWdDezdO6sMAACSCX3/9NcoKz8OGDXNB9t///nfEqdh++eUXq1evnr300ktu/Y240Bob6kjs0aNHYJY0XVeAVi+1VoLWuh6bN292Pdg7d+50wVs959GtRJ0W81qyrJEGAABISyKF07Jly7rZyFRSEYnGi2kMWPCqzrGlcKxe6qtXr4b0XHufznuX2ibqHVepbVoI0XFBkAYAAEiGNm7caHnz5nWlHpF89tln9sADD7iS19hQCa2C85EjR1z9s2qvmzdvHrhdva9z5sxxNdqzZ892M6dp+mA9jiZfeOqppxLsZ0stkrxGGgAAAFFLC1S/rF5gzWQWTqUWP/74o5t4IbY0ucLkyZPd9wrnGkemQY3B9dbPPfecKyfRTGgaP6aVdd955x1XZpIxI7ExHD3SAAAAycjRo0fdjBrqIe7SpUvENprhQ+tsNGzYMNb7VY+yJniYNGmSq4NWaNYsIR7NeLZy5UpXB/31119bzZo1XfBWW32v2/R4999/v5t84dy5c5bWEaQBAACSCc2oofCsGmUFXoXlcJonQutmPPLII670I7a0OJ3Ced26dd1saJolRDN0BNPjlShRwtVPa8YQ9U6rp1rTAL/44otucKLW1dDidpMmTbK0jiANAACQDGhgodbE0BS+mqs5X7580dZO/+c//3FrW/il+mgNZty3b1+0bVTOoemItdbGli1b3MwhekwNONTaG998842ldQRpAACAJKZBgL1793a1zwrRCq/RUVmHZt149NFHfT+eZuP49ttvQ2qkg6m0Y8eOHSGlJbqP5pmW8+fP+37s1ISqcQAAgCSmQYOqQdac0BcvXnQ9wJ5SpUq5lZi9wP3FF1+4hVG0EEskDRo0cGUbWnTFW8hFofnBBx90q0arBnvevHm2f//+iIMV9Ria7k4lHd40eJUrV3azfowcOdKqVavmZvdo3Ji1DgjSAAAASWzdunXuUjNkhJsxY4ZbLMVrp6noYpo7Wr3GCr3BQfzLL7904fj06dMuTKtWWgMP77333ij3nzVrlt12223WqFGjwDbN8qGVod9++22bP3++1alTx7p162ZpHSsbJjJWNkRssbIhAABJg5UNAQAAgFuIwYYAAACADwRpAACQJly/fiOpDwGp7PlmsCEAAEgT0qdPZxPmfm0Hj5xK6kPBLVb09jzWvc1Dt/phCNIAACDtUIj++eCJpD4MpBKUdgAAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAABAYgXpsmXL2rfffhvxtu+++87dDgAAAKRmvoL0jRs3or3t+vXrli5duvgcEwAAAJD2Sju+//57y5UrV0LvFgAAAEhWMsa24fTp023GjBnue/U4d+/e3TJnzhzS5tKlS3b8+HFr2LBhwh8pAAAAkBKDdIECBeyee+5x3x88eNDuvPNOy507d0gbBevSpUtbu3btEv5IAQAAgJQYpJs2beq+5JlnnrE///nPVrJkyVt5bAAAAEDKD9LBZs6cmfBHAgAAAKT2IO3N3LFt2zZX5qHa6HAtWrSI77EBAAAAqStI792717p27Wr79u2LOBWeBiMSpAEAAJCa+QrSgwcPtsuXL9uYMWOsTJkyUWbvAAAAAFI7X0Faqxq++eab1qhRo4Q/IgAAACC1LsiSPXt2y5kzZ8IfDQAAAJCag/Qf/vAHW7JkSYIfzLlz56xWrVquXEQDGYOtXr3a1V1XrFjRGjRoYLNnz464j2nTplndunVdu8cff9w2bNgQpc3Zs2dt4MCBVr16datSpYo9//zzbtBkpFrwTp06WeXKla1mzZo2ZMgQu3jxYgL+xAAAAEhTpR1adOWzzz5zAVShNW/evFHaPPbYY3He78SJE+3atWtRtm/evNm6detmzZs3t/79+9umTZtcqFVtdqtWrUJCtOq2X3jhBStXrpwtWLDAunTp4i4Vzj19+vRxS5m//vrrrmd93Lhx1qFDB/v0008ta9asrs3p06etffv2VqRIEXf7r7/+asOGDbOTJ0/ayJEj4/yzAQAAIHXxFaQVROWXX36xVatWRZy1Y8eOHXHa5+7du23OnDnWr18/e+ONN0JumzBhggvGQ4cOdddr1Khhhw4dsrFjx7pe5/Tp07vBj5MmTXKrKqoXWapVq2bNmjWzyZMnu4AtW7dudcc8ZcoUq127duCNgXq5Fy5caG3atHHb5s2b58L0okWLLH/+/G5bhgwZrG/fvm7GEhajAQAASNt8BekZM2Yk+IG89dZb1rp1aytevHjIdgXk9evXuwAbTAF5/vz5tn37dqtQoYLrpT5z5kxg9UUv+DZu3Ng++OADN02fAr5KRLS0uUpIPOp1rlq1qrvNC9Jr1qxx5RxeiJaGDRvaq6++6toRpAEAANI2X0FaPb0J6fPPP7edO3e6EgqVXATbv3+/XblyxUqUKBGyvVSpUoGebAVpXUp4OwVe1V4fPnzYChcu7NoprCtUh+9v3bp1getqp97uYColKVasWOCx/FKoP3/+fJzvp2POli1bvB4bKc+FCxciztcOAIg9/oemTRd8/g/1OmBv2cqGCfkDDh8+3F588cWIM4GcOnXKXaoXOZh33btdZRgKul6NsydPnjzuUrXNCtJqlytXriiPo/15+/L2F/6Ykdr5oTcGcS19EYVolbggbdGgV/2dAAD8439o2rQ3Hv9DY7NOiq8grTrkmCjBT58+PVb7Ul1zgQIF3EwgN9vnzbZHauO9C7lZu5i2h+8vNu1ikilTpkCPelzE93GRMukTFHqkASB++B+aNhX3+T90165dsWrnK0hHOiD1+Cr1q6b47rvvjtV+NOWc6pc1mFBT0olX8qBLlWR4PcrhvcDqMRav11iXly5dcl9ZsmSJ0s7bj9ppoGK48B5ofe/dN5jqsONbH60/Zs3FDcQG5TwAACTu/9DYvvHyFaRnzpwZcbuCtKap69GjR6z2o1k/VObw3HPPRez1rlSpks2aNcv14O7ZsydkgKD3TsELtd6l6peDyx90PUeOHFaoUKFAu2+++SZKz7L2FxyQ9X14LbQGPqpmO7x2GgAAAGmPrwVZYuo+19Rz77zzTqzaly1b1s0AEvz1yiuvuNsGDRrkpsFTfYqmu1u2bFnIfbUgTMGCBQOhWbNuqPZ56dKlgTaak1r30zR3XmjW9+ppXrt2baCdeqg164c3HZ4otGu2kBMnTgS2LV++3IXp4HYAAABImxJ8sGHRokXtp59+ilVblU9odcFIypcv776ke/fu1rZtW3vttdfctHcKvVpkZfDgwW4OaVHg1vzOmi9a5SXegiwHDhyw0aNHB/arXu46derYgAED3OIuGuCo+ah13C1btgy001R86g1XD7u+jh8/7gZF6vGZ+g4AAAAJHqS//PJLu/322xN0n1rGW6seKhBrgRTNvqFQHbyqoXTs2NGVbKj05NixY26hFS28EryqoYwaNcpGjBjher1VWqIwP378+JAZPxTyNWBSKyj27NnT3aY5qsPnswYAAEDalO6Gj6GMXvlFMJU8/Pjjj67W+KWXXnKhFlFt27bNXVasWNH3r+fVsUvt54P/V3KC1OnuovlsaO/GSX0YAJCq8D80bbg7nv9DY5vXfPVIb9iwIco2zZSh8ggNHFT5AwAAAJCa+QrSK1asSPgjAQAAANLqrB0AAABAWuF7sKEWYPnwww8DU8Tly5fPHnzwQWvfvn1g8RMAAAAgtfLVI3348GG3pPfkyZPdSn9FihRxl5pZQ1PI6XYAAAAgNfPVI61p6C5evGjz58+3++67L7D922+/DczlrDmXAQAAgNTKV4+0VgX805/+FBKiRdd79epla9asSajjAwAAAFJPkFYZh6a6i+SOO+5wtwMAAACpma8grbC8atWqiLepN1q3AwAAAKmZrxppDTTUMttaFLFFixZWsGBBO3r0qH366ac2a9Ys69OnT8IfKQAAAJDSg3Tnzp3twIEDLjTPnj07sF3B+oknnrBOnTol5DECAAAAqSNIp0uXzgYPHmzPPvusWy5cc0rnzZvXatSoYcWLF0/4owQAAABSao30qVOnrGfPnrZy5crAthIlSlibNm3clHe6/Pnnn10bLdACAAAApGaxDtILFiywnTt32iOPPBJtG932448/hpR7AAAAAGk6SC9dutRatWplGTNGXw2i29RmxYoVCXV8AAAAQMoO0nv37rWKFSvetF358uVdiQcAAACQmsU6SF+7di3G3miP2ly9ejW+xwUAAACkjiCtuaJ37dp103Y//fST3XbbbfE9LgAAACB1BOlq1arZnDlz7MqVK9G20W1z58616tWrJ9TxAQAAACk7SLdv397VSffo0cMOHz4c5XZt6969u2uj+aUBAACA1CzWC7Lce++9NnDgQBs0aJDVq1fPKlSoYEWLFnW3HTx40L777ju3suGf//xnK1OmzK08ZgAAACBlrWyo5b/vuecee++999yKhlu2bHHbs2XL5uaQ/uMf/2iVK1e+VccKAAAApNwlwqtUqWKTJ0+269evB1YwzJcvn6VPH+sqEQAAACDtBWmPgnOBAgUS9mgAAACAFIJuZAAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAKKxb98+GzhwoDVv3tzKlStnTZs2jdLm6tWrNmXKFGvUqJFVqlTJ6tata0OGDLHTp0/f9Pd65coVGzVqlD388MPuvs8884zt3LkzpM2xY8esc+fOVrVqVWvTpo07pmAnT560mjVr2vfff8/zCCQygjQAANH46aefbPXq1XbXXXdZyZIlI7aZMGGCjR071lq0aGHvvfeederUyRYtWmQvv/zyTX+vw4YNs9mzZ1uvXr1s4sSJljFjRnv22Wft6NGjgTZDhw61a9eu2bhx4yxz5szWv3//kH28++67Vr9+fStfvjzPI5BSlggHACC1U++yQqoowH733XdR2ixZssT1VD///PPueo0aNez8+fM2evRod5k9e/aI+z58+LDNmzfPBgwYYE888YTbpl7pevXq2fTp061v375u29dff21Tp061++67z3LlyuXaevtV7/WyZcvcF4DER480AADR/ZNMf/N/kyrtUMANljt3brtx44b7is66detcT3OTJk0C23LmzOnCu3rBg8s/smbN6r73LrVN3nrrLevevbvlz5+f5xBIAgRpAADi4cknn7S///3v9s0339i5c+ds27Zt9sEHH1jLli0tR44c0d5v9+7ddtttt1nevHlDtquEZO/evXb9+nV3vWLFijZnzhw7deqUKwMpVqyY5cmTxz777DM7ceKEPfXUUzx/QBKhtAMAgHhQSYd6pTt27BjogX7sscds8ODBMd5PgxHDe7JFIVk9zirfUA91v3797LnnnrO5c+e69uPHj7cLFy7YO++842qsVVcNIGnQIw0AQDzMmjXLPvzwQ1dDre/ffPNN27Rpk7322ms3vW+6dOmibAsvB9FsIStXrnR10KqX1gwdkydPdjXT+l63NWzY0O6//35XV61ecQCJgyANAIBPKq0YMWKEm3VDs2088MADbjDgoEGD3MwdMU1JpzrqSFPkaVumTJlCBinqeokSJSxLlix24MAB1zutnurjx4/biy++aD169LDly5e7kpBJkybxfAKJhCANAIBPCrWXL1+2smXLhmz3ru/fvz/a+6oWWkFY80CH104XL1482oGOKufQfNNFixa1LVu2uCnxmjVr5gYcqi5btdoAEgdBGgAAn4oUKeIuw3uevWnyFHajo0VYFJaDp65TWcaKFSusdu3aEe+j0o4dO3ZYly5dAttUT63ZP0R11QASDyMUAACIhgb1eVPRHTx40M6ePWuff/65u16tWjU364bqk7UgiwYcVqhQwfbs2eMGBFapUsVd9zRo0MAFb80RLYUKFbLWrVvbyJEj3YBB3abZPqR9+/ZRjkX713R3KunwpsGrXLmym91D+9DxaHaPxo0b83wCiYQgDQBANFR60bt375Bt3vUZM2ZY9erV3cqDqkueP3++W31Q4VqzdqhdcHmGeo29Ke08GqCoWmitTnjmzBm3IIuCdsGCBaMciwYyat9aitxToEABt8T422+/7R6/Tp061q1bN55PIJGkuxHTbPFIcJpf1JsX1K9Xxy61nw+eSMCjQnJ0d9F8NrQ3PUsAkJD4H5o23B3P/6GxzWvUSAMAAAA+JHmQXrt2rbVt29Zq1Kjhasnq1avnRiTrI65gqlFr0aKFe2egOjOt7hTJtGnT3PKqavf444/bhg0borRRjdvAgQPdR3KqYdNk+qp9C6dphDp16uRq0DRX55AhQ+zixYsJ+NMDAAAgpUryIK0lTxVmNYG9QnCHDh3c3JvBNWmbN292NV+alH7q1Klueh+F2gULFoTsS/cfM2aMPf300zZlyhS766673MjmH374IaRdnz593Kjo119/3bU/cuSIe9zgkKx5PDXYQyOoVfOmwR2LFy+O1QT7AIDYu36dCsO0hOcbqUmSDzZs2rSp+/Kol1hzYirkHj582I1qnjBhggvRGtAh6r0+dOiQGyWtXmcN5tA8nhrs0a5dO9eLLBrBrLk1tQKUArNs3brVVq1a5YK2N71Q6dKlXS/3woULrU2bNm7bvHnzXJhWqNfcnJIhQwa3alTXrl3d/J8AgPhLnz6dTZj7tR08copfZypX9PY81r3NQ0l9GEDqCdKR5M2bNzDVjwLy+vXrXYANpoCsEcrbt293JSFajlXlIMGhXMFX0wBpOiGNqdRSrCoR0WpStWrVCrTTlENVq1Z1t3lBes2aNa6cwwvRoimOXn31VdeOIA0ACUchmkHUAFKaZBOkNS2QgvOuXbtcD/Sjjz7qJrLXdU02r6VRg5UqVSqwApSCtC4lvJ0Cr8oz1LtduHDhwIpRCtXh+1u3bl3gutqptzuYesqLFSsWeCy/FOr9TJqvY86WLVu8Hhspcx5bJtdBasXrWtqUFK9rnGtp0wWf55rXAZtigrSCs8KuPPLIIzZ69OhADbWoFzmYd927XWUYCrreJPWePHnyuEstwaogrXa5cuWK8vjan7cvb3/hjxmpnR96Y6CVqeJKIVolLkhbNOhVLwRAasTrWtqUFK9rnGtp0954nGvKlSkmSKtmWb206oGeOHGim0njr3/9a+D26N4VBG+P1MZ7F3KzdjFtD99fbNrFJFOmTIEe9biI7+MiZdInKPRII7XidS1tSorXNc61tKm4z3NNeTQ2kk2Qvvfee92lapXV66qyiuXLlwcCZ3gvsHqMxes11uWlS5fcV5YsWaK083qm1U4DFcOF90Dre+++wVSHHd/6aP0xayUrIDYo5wGQ2vC6huR+rsX2jVeST38XSdmyZd1Awf3797uaZPXg7tmzJ+I7BS/Uepfh9cu6niNHDjf7h9dO3fzh7060v+CArO/D96WBjzomBhoCAAAgWQZpzRutwYd33HGHq0/RdHfLli0LabNkyRIrWLBgoGZYPdmqfV66dGmgjfah+2maO++dhb5XT7MWgvGoh1qzfnjT4Ylm9dBsISdO/N9S3OohV5gObgcAAIC0KclLO3r06OFm3ShTpowbKLhz5057//333fX69eu7Nt27d3erH2oxFE17p9CrxVgGDx7s5pAWBW7N76z5ojVlnQK22hw4cCAwcFEqVapkderUsQEDBlj//v0tZ86cbj5qzRCihV48rVu3tlmzZrmFYPR1/PhxGz58uHt8eqQBAACQ5EH6vvvuc73IGmyocgsF2ieeeMItquKNltTKhxqAqECsBVI0+4ZCdatWrUL21bFjR7ePmTNn2rFjx9xCK9qvQnmwUaNG2YgRI2zQoEFuBg0tAjN+/PiQGT9UIz19+nS3gmLPnj3dbZqjOnw+awAAAKRNSR6kn3vuOfd1MyqnuFlJhco3Onfu7L5iol5oLUmur5uN9NSy4wAAAECKqJEGAAAAkjuCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpACnOvn37bODAgda8eXMrV66cNW3aNEqbDz74wFq0aGG//e1vrXLlytasWTObNWuW3bhx46b7v3Llio0aNcoefvhhq1Spkj3zzDO2c+fOkDbHjh2zzp07W9WqVa1NmzbumIKdPHnSatasad9//30C/MQAgOSIIA0gxfnpp59s9erVdtddd1nJkiUjtjlz5owL2ArEEydOtDp16tiQIUPsvffeu+n+hw0bZrNnz7ZevXq5+2bMmNGeffZZO3r0aKDN0KFD7dq1azZu3DjLnDmz9e/fP2Qf7777rtWvX9/Kly+fAD8xACA5ypjUBwAAcVW3bl0XUkUB9rvvvovSpnfv3iHXH3zwQfvPf/5jCxcutOeffz7afR8+fNjmzZtnAwYMsCeeeMJtU690vXr1bPr06da3b1+37euvv7apU6fafffdZ7ly5XJtz58/b9mzZ3e918uWLXNfAIDUix5pAClO+vT+Xrry5cvnyjZism7dOtfT3KRJk8C2nDlzuvCuXnCP9pM1a1b3vXfp7futt96y7t27W/78+X0dJwAgZSBIA0jVrl69aufOnbNVq1bZokWLrF27djG23717t912222WN2/ekO0qIdm7d69dv37dXa9YsaLNmTPHTp065cpAihUrZnny5LHPPvvMTpw4YU899dQt/bkAAEmP0g4AqZYGAD722GOB6127dnW1zjE5ffq0K9UIp5CsHmeVb6iHul+/fvbcc8/Z3LlzXfvx48fbhQsX7J133nE11qqrBgCkbrzSA0i1fvOb39jHH3/swu+//vUvV9OsshANIoxJunTpomwLn+1Ds4WsXLnSDhw4YEWLFrUsWbLYmDFjXM20ZuvQbcOHD3ezezz66KM2aNAgy5EjR4L/jACApENpB4BUS7NpqASjevXq1qNHDzcAcfLkySGzb4TLnTu365UOp22ZMmVygwk9ul6iRAkXohWo1Tutnurjx4/biy++6B5z+fLlriRk0qRJt+znBAAkDYI0gDRDU9FpIOHBgwejbaNaaAVhzQMdXjtdvHjxaAc6qpxD802rd3rLli0uxGvuag04bNmypX3zzTcJ/vMAAJIWQRpAmrFx40ZXtnHHHXdE20aLsCgsB09dp8GKK1assNq1a0e8j6bC27Fjh3Xp0iWwTfXUCu2i0hIAQOpDjTSAFEeD+ryp6NS7fPbsWfv888/d9WrVqrmSC4Xa3//+927RFs3csX79eps5c6Y9+eSTblYOT4MGDaxIkSJujmgpVKiQtW7d2kaOHOkGDOo2rZIo7du3j3Is2remu1NJhzcNnlZS1Owe2oeOR7N7NG7cOFF+NwCAxEOQBpDiqPQifMEV7/qMGTOsSpUqrgzjww8/dAusKOBqejoN+NOy4cHUa+xNaefRIi+qhdbqhFohUQuyKGgXLFgwyrFo2XEF80aNGgW2FShQwK2o+Pbbb9v8+fPdqordunVL4N8CACCpEaQBpDgqzfjhhx9ibKOa5dhQyUY41TdrBUNvFcOYaDq9SFPqaSVEfQEAUi9qpAEAAAAfCNIAIrp+PXTeZKRuPN8AEHeUdgCIKH36dDZh7td28MgpfkOpXNHb81j3Ng8l9WEAQIpDkAYQLYXonw+e4DcEAEAElHYAAAAAPhCkAQAAAB8I0gAAAIAPBGkAAAAgJQbpZcuWuRW/ateu7ZbVbdasmVtON3ylMS0HrBXJKlas6Jb0nT17dsT9TZs2zerWrevaPf7447Zhw4YobbSc8MCBA6169epuBbTnn3/eLTMcbu/evdapUyd3XDVr1rQhQ4bYxYsXE/CnBwAAQEqV5EH6r3/9q1tF7OWXX7bJkydb/fr17a233rJ33nkn0Gbz5s0ubJcrV86mTp1qLVu2dKF2wYIFUUL0mDFj7Omnn7YpU6bYXXfdZV26dImyAlqfPn3camavv/66a3/kyBHr0KFDSEg+ffq0tW/f3s6dO2fjxo2zfv362eLFi+21115LhN8KAAAAkrskn/5O4Tl//vyB6zVq1LDz58+7HucXXnjBhewJEya4ED106NBAm0OHDtnYsWNdr3P69Ont8uXLNmnSJGvXrp3rRZZq1aq5Hm49hgKzbN261VatWuWCtnrBpXTp0q6Xe+HChdamTRu3bd68eS5ML1q0KHB8GTJkcEsGd+3a1UqWLJnovysAAAAkH0neIx0coj1ly5a1S5cu2cmTJ11AXr9+vTVp0iSkjQLy0aNHbfv27e76pk2b7MyZM9a0adNAGwXfxo0bu7KQGzf+/ypt+j537txWq1atQLsiRYpY1apV3W2eNWvWuHKO4ONr2LChC/bB7QAAAJA2JXmQjmTjxo2WN29eK1CggO3fv9+uXLliJUqUCGlTqlQpd7l79+6Qy/B26jlWecbhw4cD7YoXL27p0qWLsj9vH1678F5nhehixYqFtAMAAEDalOSlHeG2bdtmn3zyiXXv3t31KJ869f+XJ1YvcjDvune7yjAUdLNmzRrSLk+ePO5SvduFCxd27XLlyhXlcbU/b1/e/sIfM1I7P9Q7rvKVuFL4z5YtW7weGynPhQsXAp+oJBbOtbSJcw2ca0htLvj8H6r7hHe6JvsgrVKNXr16uRk3NEgwWHQ/TPD2SG28X97N2sW0PXx/sWkXE/Ww79ixI873U4hWrTjSFs0eoxeCxMS5ljZxroFzDanN3nj8D1UHbYoJ0qpvVnhWj7IGDWbKlCmkRzm8F1g9xuL1GutSddX6ypIlS5R23n7UTgMVw4X3QOt7777hxxnfgYb62bzSlLiIb4BHyqRSpKTokUbaw7kGzjWkNsV9/g/dtWtXrNoliyCt8KuZMI4dO2YfffSR5cuXL3CbapIVPPfs2RMyQND7Ab1Q612qfjm411bXc+TIYYUKFQq0++abb6L0LGt/wQFZ34fXQmvgo2q2NVNIfOhxs2fPHq99IO2gnAeca0hteF1Dcj/XYtuhlOSDDa9evWq9e/e2nTt32vvvv29FixaN0q2u6e60cEuwJUuWWMGCBQOhWbNuqPZ56dKlgTbXrl1z99M0d94vRN+rp3nt2rWBduqh1qwf3nR4otCu2UJOnDgR2LZ8+XIXpoPbAQAAIG1K8h7pwYMH28qVK+2ll15yC6Js2bIlcJvKH3LmzOkGHrZt29YthqJp7xR6tRiL7qs5pL3ArV5tzRetKesUsNXmwIEDNnr06MA+K1WqZHXq1LEBAwZY//793f41H7UCvBZ68bRu3dpmzZrlFoLR1/Hjx2348OHu8ZlDGgAAAEkepNetW+cug1cy9MyYMSOwjPfEiRNdINYCKZp9Q6G6VatWIe07duzoSjZmzpzpykS00IoWXilTpkxIu1GjRtmIESNs0KBBbuCfHmP8+PEhM36oRnr69OluBcWePXu62zRHtRZkAQAAAJI8SGup7thQOcXNSipUvtG5c2f3FRP1Qr/55pvu62YF6lp2HAAAAEh2NdIAAABASkSQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAICUG6X379tnAgQOtefPmVq5cOWvatGnEdqtXr7YWLVpYxYoVrUGDBjZ79uyI7aZNm2Z169Z17R5//HHbsGFDlDZnz551j1m9enWrUqWKPf/883bw4MEo7fbu3WudOnWyypUrW82aNW3IkCF28eLFBPipAQAAkNIleZD+6aefXEi+6667rGTJkhHbbN682bp16+aC9tSpU61ly5Yu1C5YsCBKiB4zZow9/fTTNmXKFLfPLl262A8//BDSrk+fPrZixQp7/fXXXfsjR45Yhw4dQkLy6dOnrX379nbu3DkbN26c9evXzxYvXmyvvfbaLfpNAAAAICXJmNQHoN7j+vXru+/79+9v3333XZQ2EyZMcCF66NCh7nqNGjXs0KFDNnbsWNfrnD59ert8+bJNmjTJ2rVr53qRpVq1atasWTObPHmyC8yydetWW7VqlQvatWvXdttKly7terkXLlxobdq0cdvmzZvnwvSiRYssf/78bluGDBmsb9++1rVr12hDPwAAANKGJO+RVgiOiQLy+vXrrUmTJiHbFZCPHj1q27dvd9c3bdpkZ86cCSkNUfBt3Lix6/G+ceOG26bvc+fObbVq1Qq0K1KkiFWtWtXd5lmzZo0r5/BCtDRs2NAyZ84c0g4AAABpU5IH6ZvZv3+/XblyxUqUKBGyvVSpUu5y9+7dIZfh7dRzrPKMw4cPB9oVL17c0qVLF2V/3j68duG9zgrRxYoVC2kHAACAtCnJSztu5tSpU+5SvcjBvOve7SrDUNDNmjVrSLs8efK4y5MnT1rhwoVdu1y5ckV5HO3P25e3v/DHjNTOD/WOnz9/Ps73U/jPli1bvB4bKc+FCxcCn6gkFs61tIlzDZxrSG0u+PwfqvuEd7qmyCDtie6HCd4eqY33y7tZu5i2h+8vNu1ioh72HTt2xPl+CtGqFUfaotlj9EKQmDjX0ibONXCuIbXZG4//oeqgTfFB2utRDu8FVo+xeL3Gurx06ZL7ypIlS5R23n7UTgMVw4X3QOt7777BVIcd34GGmTJlCpSmxEV8AzxSJpUiJUWPNNIezjVwriG1Ke7zf+iuXbti1S7ZB2nVJCt47tmzJ2SAoPcDeqHWu1T9cnCvra7nyJHDChUqFGj3zTffROlZ1v6CA7K+D6+F1sBH1WxrppD40ONmz549XvtA2kE5DzjXkNrwuobkfq7FtkMp2Q82VLe6prtbtmxZyPYlS5ZYwYIFA6FZs26o9nnp0qWBNteuXXP30zR33i9E36unee3atYF26qHWrB/edHii0K7ZQk6cOBHYtnz5chemg9sBAAAgbUryHmnVrXjTyWl1Qa06+Pnnnwfmgdb0c927d7e2bdu6xVA07Z1CrxZjGTx4cGD6PAVuze+s+aJ1HwVstTlw4ICNHj068HiVKlWyOnXq2IABA9y81Tlz5nTzURctWtQt9OJp3bq1zZo1yy0Eo6/jx4/b8OHD3eMzhzQAAACSPEgroPbu3Ttkm3d9xowZgWW8J06c6AKxFkjR7BsK1a1atQq5X8eOHV3JxsyZM+3YsWNuoRUtvFKmTJmQdqNGjbIRI0bYoEGD3MA/Pcb48eNDZvxQjfT06dPdCoo9e/Z0t2mOai3IAgAAACR5kL7jjjuiLOEdicopblZSofKNzp07u6+YqBf6zTffdF83K1DXsuMAAABAiquRBgAAAJIjgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAIAgDQAAACQOeqQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQfom9u7da506dbLKlStbzZo1bciQIXbx4kU/v2sAAACkIhmT+gCSs9OnT1v79u2tSJEiNm7cOPv1119t2LBhdvLkSRs5cmRSHx4AAACSEEE6BvPmzXNhetGiRZY/f363LUOGDNa3b1/r2rWrlSxZMrGeJwAAACQzlHbEYM2aNa6cwwvR0rBhQ8ucObOtXr06MZ4fAAAAJFME6Rjs3r07Sq+zQnSxYsXcbQAAAEi7KO2Igco6cufOHWW7tp06dcrXL/zKlSt248YN+/bbb33dP126dNakWkG7dr2Ar/sj5ciQPr1t27bNnS9JgXMt7eBcA+caUpsM8fwfqrym/4M3Q5D2QU9KbH65kXj383t/yZ0zq+/7IuWJz7kSX5xraQvnGjjXkNqki0deI0jHk3qe1Ssd7syZM74HGlapUiW+hwUAAIBkgBrpGCgsh9dCX7582fbv38+MHQAAAGkcQToGtWrVsvXr19uJEycC25YvX+7CdO3atRPj+QEAAEAyle5GUo1kSgFU1tG0aVMrWrSodevWzY4fP27Dhw+3hx9+mAVZAAAA0jiCdCyWCNey4Bs3brSsWbO6YK0FWfQ9AAAA0i6CNAAAAOADNdIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKSRIv3yyy9WpkwZ+/zzz2Ns98knn7h2v/76a5z27/d+SB769+/vFk+KSd26dW3w4MFx3rff+yFt0nmo8zEx7dixw8aPH28XLlxI1MdF4tHzW6VKFX7lyQBBGqlanTp17KOPPrLcuXMn9aEgmfnLX/5iHTt2TOrDAG5JkNb5TZAGbr2MifAYQMDFixcTdXn1/Pnzuy8gXLly5filIMVI7NdOALFDjzQiljUoZBw7dixk+8mTJ61ChQo2e/Zsd33z5s3Wrl07q1y5st1///3Wp08fO378eJTyC+3vtddes+rVq9v//M//2IwZM9x9zp49G7L/vXv3uvZfffVVrJ+VS5cuuY/ZH3jgAXv44YdtxIgRdvXq1RhLNP773//aH//4R7vvvvvskUcesffff9/tQx/Zhzt06JB17tzZHe9jjz1mixYt4oxJQTZs2GAtWrRwz5/Ove+++y7GEo158+bZo48+apUqVbL27dvb1q1bA+dwuFmzZrm2Ove7detGGVAKKvlZu3atNWvWzL0GPPXUU3bgwAH3+vanP/3JqlatavXr17elS5eG3FefbP3ud79zr4H6pGvMmDEhrzWyadMm+8Mf/mAVK1Z0j7N69eqIx+H3tVNWrVplHTp0sJo1a7pjbdWqla1ZsyZwX93nlVdecd+rjfYT/Nqm17++ffu6fernf/rpp0P+LpAy/ec//7FevXrZb3/7W/f6pfNr27ZtgdsnTJjgzlvPjRs33DmgL33vqV27tk2cODHRjz8lI0gjCgXGjBkzRqk//vLLL90fnP6Z6B/BM888Y7ly5XL/UN588033R9u1a9co+xs9erSlS5fORo0aZS+99JI1b97crl+/bkuWLAlp97e//c0KFizo/pBj691337X06dO7yyeffNI++OADW7BgQbTtdfwKPfroUyHqz3/+swvu0YV3Ha8Cul6E7r33XvePeNeuXZw1KcDRo0dtyJAh1qlTJ3eOqkevR48eduXKlYjtdQ688cYb9tBDD7mPxXWpgBPJihUrbOXKlTZw4EAbMGCA/fOf/3R/A0gZ58XIkSPda5UuFVr1d/7iiy/aPffc42pPy5cv77YdPHjQ3WfmzJnuua5Ro4ZNmjTJWrdubdOmTXPbgvercy1z5szu9UjfDxo0yG0PFp/XTtHx6g3c22+/7Y5VYfq5555zbxpFYcnblzoJ9AZA57OcOnXKvXHYuXOnvf766+7+2bJlc28ag4M8UhZ1SrVt29a9IdI5qfPl8uXLLkzv3r3btVFnkzqG9KZRfvjhB3c/ff34449um27TGy21RexR2oEocubM6cKsgq7+OD26rn8kKpXQO1/1zOgFWi/0on9C6uVRL0xwGFbvdnjIaNiwoQvO+ock165dc7296j1UiI8t9aiox0YUfL755hv74osvrE2bNhHbq+fm+++/d73qeucuekdeq1Yty5s3b5T26q3Rl+hdvnqD9IaiVKlSnDnJnEKDeo11XkqWLFlcT556mb3nPpgCks5vhW/RpxX6xMMLIeFvyNReoUn27dvngpXeIOqNHZL3eTFnzhwrWbKku37kyBH3+tSlSxfr3r2726Ye5eXLl9v//u//utdAvZFu1KiRe6PlnRt63VMQVmi98847bfr06W7blClTAmMy1DGgQB1MISc+r53Br8k63/T6pTf38+fPd9/r9blYsWLudr0hCC5t0zGePn3adTYUKFAg0GvdoEEDd/6+/PLLCfzbRmLQpxDqkV68eHHg9U7Pqz6JmDp1qg0fPtz9/9Lr1b/+9S93vv773/9255dey7RNn1zoUm3UFrHHKz4iatKkiW3ZssX9cYp6VfRHphd7DWDRR5j6x6IArI839VW8eHH3jyP44ySJ1MP8xBNP2Lfffms//fRTIODqMR5//PE4PSPqLQ6mgKt31NHRsemfXHCQ0hsH/QO62f7V7je/+U2M+0fycfvttwf+qYgXnA4fPhylrc5jfUoRXt5Tr169iPtWj40Xor3zTj3d9OqljPPCOxfk7rvvdpcPPvhgYJteIxRA9be+Z88eO3HihDVu3DjKa6RCyMaNG911vUHT60jwwGa9fuh1w5MQr506pn79+rkwryCksLxu3TpXGnczX3/9tTvGPHnyBB5bb/z0ehj+2Eg5FIr1Whf8epcjRw73yYVu8zoS9AZR/8dFl3od05c+UfO2qXMq+LUNN0ePNCLSH6D+ED/77DPXU6N6wUyZMrnaQfVo6J/AsGHD3Fc4fXwULNJgP/3x6p/Hxx9/7Or5dKkXc22LC308GkzHqI+0oqPep0jH4/XOxHf/SD7CZ2rRcyfqZQ6nGnqFivBzI7rzIi77RvIS3XMX/reuMKHnUz3Yctttt4XcruAr3u3qCLjrrruiPF7wORTf1071QKsH/MyZM+5TQT2eSjPGjRsX5b6R6A2BOkgUvsN5vdhIeXRehZ+fom3e+en939X/dNEbQH0CrDeDf//73902hW69QUTcEKQRkd69egNuvCCt2jv1rqgHQx9JasCe2oTLly9fyHXv48twGiSjGj5NQaaPNBOjxlS9UZHmhqYnMW1TYFFJUfi5wXkBr+Qr/Fzwap/Vu+sF60jnS/A2hfX4vHaqhGj79u2u1CT4/qr/jw0dq3qye/fuHeU2eiFTLj2v+uQknCYM8M5PL0hPnjzZ/vGPf7jzUgNdvTdY2rZ//37qo30gSCNaemequmWNcFcvhgamSPbs2d1oc/3h6qMiv1q2bOlqDDWgS8FdH3feajpevXv3PtYSDbbQQJ1INdJIGzJkyGBly5Z1Aw418MqjGlmkbfqUTG+0li1b5gZie9S5oKDrhRF9JD537lzXW+z1bqvkInh2ovi+dnqfeHi96KIBkRrA6JWoBN8e/umZylc+/fRTV9qiY0HqoHNQY4NUK++N3zl//rwbEK1Plz1awEUdBpqVQzXR3qczpUuXdtt0G4u8xB1BGtHSi67+gbz66quBAYgeDUpR4NB0UQrc+oNU7Z4G+2n6p+hqjoNp36pB1ewgmnFDH1HeahpUqI81Fd41Sl/HrcEYXk8R0i59ZK4ZXTR4VW/q1PPnfeTJAMK0/SZLgxD1iZlesxRMdG6onEKvdRq4JXo91CBGfYKnL71hV+dD+Bv0+Lx2lihRwgoXLuwGLKrMQzXXOg590hbMqwHXoGr1XGv+aQWnZ5991g1I04BFzehQpEgR9ymM6rsLFSrkbkfKo/Pmww8/dJ906LzSmyR92qs3XjoXPSrXVIeBaqI1c4xHnUqamUZvBnmDFXcMNkS09O5UgUJ1xRrVrV5jj6Zc0j8NvetVjbOmX9I7Wr1gR6oTjI72K94cqbeawrKOU1PZaZogfekfY7Vq1aLUSCJt0Zs6TYeoXkQFan0S483SEDxgDGmPgqemstNgveeff94FVM3GETwPucKs3pSrzEKlE/pery9eLXVCvHaq/ELhXJd6jLFjx7rj0etXMA1C7Nmzp+t91sxI3nR4Kh3RdHgKU5r6T2V1qtVWr7ZCFFImvT5phiJ1Euk1TJ1E+v+tNRuCB9aK90ls8BR33veRZjPCzaW7ETwTN5DI1Duj2RLUS5JU9PGn5sbWP6NIA4CQdmmaMPVQq+TjjjvuSOrDAQAkM5R2IEloMngFaNUZer1+iUU9MvpYVLWP+vhVdY0a8a6FCpB2aWU7ze2ruaT1EaimA9PAHPVUE6IBAJEQpJEk9FGjavM0/U743NH6kERTREVH9arxqVlViYo+dtUKYaIyj/feey9eAyeR8umjUK3spemh9AZLH4NrFU4tpwwAQCSUdiDZ0QwaGggT02wfWqkJAAAgKRGkkexouqiYVulSTyEftQMAgKRGkAYAAAB8YPo7AAAAwAeCNAAAAOADQRoAAADwgenvACAN0pLRsaHV0WJathoA0jIGGwJAGrRly5aQ61qmWlNPTp8+PWR7qVKlWCIdAKJBjzQApEGVK1cOuZ4/f3630FH4dgBA9KiRBgBE8eqrr1q1atXswoULUW7TgklNmjQJKRMZPHiwzZs3zxo2bGgVKlSwxo0bu1Uiwx09etQGDhxotWrVcu3q1q3rlma/evUqzwKAFIceaQBAxLD8t7/9zZYsWWKtWrUKbN+1a5crAVEYDrZixQq3vVevXpYtWzabM2eOvfjii5YhQwZr1KhRIERrX+r57t69uxUrVsw2b95skyZNsoMHD9qwYcN4JgCkKARpAEAU9957r+uRnj17dkiQnjVrlquZbtGiRUj7EydO2Mcff2y33Xabu167dm1r2rSpjR49OhCkx48fb6dOnXI91UWKFHHbatasaVmzZrURI0ZYp06dXE02AKQUlHYAAKLtld6xY4dt3LjRXT979qz9/e9/t5YtW1qOHDlC2ioQeyFa1BOt8o59+/bZf//7X7dt1apVbgaQ22+/3ZVyeF8q85B//vOfPBMAUhR6pAEAEdWrV8+KFi3qyjTuv/9+++STT1zN9FNPPRWlbXCIDt928uRJK1y4sB0/ftxWrlxp5cuXj/h46tUGgJSEIA0AiEi1zE8//bSNGTPG+vXr5wK1ep5LlCgRpe2xY8ei3ZY3b153mS9fPjcw8U9/+lPEx1NPNQCkJARpAEC0VB+tWTX69u1re/fudZeR/OMf/3DB2euFvnbtmi1dutQNKFRvtNSpU8dWr17ttuXJk4ffOoAUjyANAIhW7ty5rXnz5jZ37lxX5qHp6iJRb3P79u2tW7dugVk79uzZ43qzPZrR45tvvrHWrVvbM888Y8WLF7fLly/bL7/8YmvWrLFBgwYFQjcApAQEaQBAjDRoUEFaAVjlHpEoYGvGjXfffdcOHTpkd955p40cOdLdN7h0QzN7aBXFadOm2eHDh92gRQX0Rx55xIV2AEhJWCIcABCj4cOHuyCtWTfU8xxOdc+qpQ6fWxoAUjt6pAEAEW3ZssV+/vlnV6bx5JNPRgzRAJCWEaQBABEpPKveWYMEo5tpAwDSMko7AAAAAB9Y2RAAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAAsLj7fzodEhIAildiAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 800x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Set plot style\n",
    "sns.set(style=\"whitegrid\")\n",
    "\n",
    "# Calculate counts and percentages\n",
    "counts = collapsed_df['Type'].value_counts()\n",
    "total = counts.sum()\n",
    "percentages = (counts / total * 100).round(2)\n",
    "\n",
    "# Desired order\n",
    "type_order = ['very_high', 'high', 'moderate', 'low']\n",
    "\n",
    "# Plot\n",
    "plt.figure(figsize=(8, 6))\n",
    "ax = sns.countplot(data=collapsed_df, x='Type', order=type_order)\n",
    "\n",
    "# Annotate each bar\n",
    "for p in ax.patches:\n",
    "    height = p.get_height()\n",
    "    label = p.get_x() + p.get_width() / 2\n",
    "    category = p.get_x() + p.get_width() / 2\n",
    "    type_name = p.get_x() + p.get_width() / 2\n",
    "    percentage = (height / total) * 100\n",
    "    ax.annotate(f'{percentage:.1f}%', (p.get_x() + p.get_width() / 2, height),\n",
    "                ha='center', va='bottom', fontsize=11)\n",
    "\n",
    "# Labels\n",
    "plt.title('Distribution of Type Categories')\n",
    "plt.xlabel('Type')\n",
    "plt.ylabel('Count')\n",
    "plt.ylim(0, counts.max() * 1.1)  # Add some space above bars\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf5323fc-ceaa-432f-90d1-87e0b7c97ddf",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "browser_data",
   "language": "python",
   "name": "browser_data"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
