{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "23a7f7d2-2e98-45be-b488-688489353f1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a29f3799-2118-4ac8-b80b-4c0ff7748a4e",
   "metadata": {},
   "source": [
    "# MB ATAC-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "87bef440-6063-4668-abc1-ce2dddec0870",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju\n"
     ]
    }
   ],
   "source": [
    "%cd ~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "1ed15ca4-47b6-4216-8403-5e94b5966986",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju/PreprocessDB/promoter_table\n"
     ]
    }
   ],
   "source": [
    "%cd ./PreprocessDB/promoter_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "57b0b52d-5c3f-4d1a-8d69-762a67a23267",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mingju/PreprocessDB/promoter_table'"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0444d201-284c-4343-bd23-243324430ab7",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_atac = pd.read_csv('/mnt/10T-backup-mingju/ifar/ATAC_seq/MB/atac_output/idr.optimal_peak.narrowPeak_w_header',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4d9d72a8-f8cc-465b-a0aa-4ee95e8a4f2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629232</td>\n",
       "      <td>630132</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>8.03966</td>\n",
       "      <td>5807.30420</td>\n",
       "      <td>5799.29053</td>\n",
       "      <td>471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629232</td>\n",
       "      <td>630132</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.56924</td>\n",
       "      <td>5235.43115</td>\n",
       "      <td>5228.03027</td>\n",
       "      <td>665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633820</td>\n",
       "      <td>634277</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.97965</td>\n",
       "      <td>4538.55518</td>\n",
       "      <td>4531.56543</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633820</td>\n",
       "      <td>634277</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.06016</td>\n",
       "      <td>3501.44189</td>\n",
       "      <td>3494.67920</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521223</td>\n",
       "      <td>22521670</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.38412</td>\n",
       "      <td>3410.55688</td>\n",
       "      <td>3403.83081</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152796</th>\n",
       "      <td>chr11</td>\n",
       "      <td>112961786</td>\n",
       "      <td>112963648</td>\n",
       "      <td>.</td>\n",
       "      <td>827</td>\n",
       "      <td>.</td>\n",
       "      <td>1.91464</td>\n",
       "      <td>3.05557</td>\n",
       "      <td>1.40385</td>\n",
       "      <td>308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152797</th>\n",
       "      <td>chr7</td>\n",
       "      <td>13551221</td>\n",
       "      <td>13552318</td>\n",
       "      <td>.</td>\n",
       "      <td>737</td>\n",
       "      <td>.</td>\n",
       "      <td>2.55754</td>\n",
       "      <td>3.05534</td>\n",
       "      <td>1.40367</td>\n",
       "      <td>280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152798</th>\n",
       "      <td>chr4</td>\n",
       "      <td>99300712</td>\n",
       "      <td>99302001</td>\n",
       "      <td>.</td>\n",
       "      <td>840</td>\n",
       "      <td>.</td>\n",
       "      <td>2.55754</td>\n",
       "      <td>3.05534</td>\n",
       "      <td>1.40367</td>\n",
       "      <td>56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152799</th>\n",
       "      <td>chr16</td>\n",
       "      <td>113146</td>\n",
       "      <td>114212</td>\n",
       "      <td>.</td>\n",
       "      <td>891</td>\n",
       "      <td>.</td>\n",
       "      <td>2.55754</td>\n",
       "      <td>3.05534</td>\n",
       "      <td>1.40367</td>\n",
       "      <td>813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152800</th>\n",
       "      <td>chr14</td>\n",
       "      <td>77027859</td>\n",
       "      <td>77029044</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>1.77600</td>\n",
       "      <td>3.05499</td>\n",
       "      <td>1.40359</td>\n",
       "      <td>381</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>152801 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629232     630132    .   1000      .      8.03966   \n",
       "1        chr1     629232     630132    .   1000      .      7.56924   \n",
       "2        chr1     633820     634277    .   1000      .      6.97965   \n",
       "3        chr1     633820     634277    .   1000      .      6.06016   \n",
       "4       chr17   22521223   22521670    .   1000      .     32.38412   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "152796  chr11  112961786  112963648    .    827      .      1.91464   \n",
       "152797   chr7   13551221   13552318    .    737      .      2.55754   \n",
       "152798   chr4   99300712   99302001    .    840      .      2.55754   \n",
       "152799  chr16     113146     114212    .    891      .      2.55754   \n",
       "152800  chr14   77027859   77029044    .   1000      .      1.77600   \n",
       "\n",
       "            pValue      qValue  peak  \n",
       "0       5807.30420  5799.29053   471  \n",
       "1       5235.43115  5228.03027   665  \n",
       "2       4538.55518  4531.56543   120  \n",
       "3       3501.44189  3494.67920   299  \n",
       "4       3410.55688  3403.83081   129  \n",
       "...            ...         ...   ...  \n",
       "152796     3.05557     1.40385   308  \n",
       "152797     3.05534     1.40367   280  \n",
       "152798     3.05534     1.40367    56  \n",
       "152799     3.05534     1.40367   813  \n",
       "152800     3.05499     1.40359   381  \n",
       "\n",
       "[152801 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4ada0913-19b6-4530-9759-df5e068706c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_atac['index'] = MB_atac['Chr'] + \":\" + MB_atac['Start'].astype(str) + \"-\" + MB_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b47a673d-73d6-4efe-8a27-cbbea8e45e9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_atac['ATAC_MB'] = \"ATAC-seq:\" + MB_atac['Chr'] + \":\" + MB_atac['Start'].astype(str) + \"-\" + MB_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a959e645-7cd5-416a-8aa8-3a4030e4fa18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>ATAC_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629232</td>\n",
       "      <td>630132</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>8.03966</td>\n",
       "      <td>5807.30420</td>\n",
       "      <td>5799.29053</td>\n",
       "      <td>471</td>\n",
       "      <td>chr1:629232-630132</td>\n",
       "      <td>ATAC-seq:chr1:629232-630132</td>\n",
       "      <td>ATAC-seq:chr1:629232-630132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629232</td>\n",
       "      <td>630132</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.56924</td>\n",
       "      <td>5235.43115</td>\n",
       "      <td>5228.03027</td>\n",
       "      <td>665</td>\n",
       "      <td>chr1:629232-630132</td>\n",
       "      <td>ATAC-seq:chr1:629232-630132</td>\n",
       "      <td>ATAC-seq:chr1:629232-630132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633820</td>\n",
       "      <td>634277</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.97965</td>\n",
       "      <td>4538.55518</td>\n",
       "      <td>4531.56543</td>\n",
       "      <td>120</td>\n",
       "      <td>chr1:633820-634277</td>\n",
       "      <td>ATAC-seq:chr1:633820-634277</td>\n",
       "      <td>ATAC-seq:chr1:633820-634277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633820</td>\n",
       "      <td>634277</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.06016</td>\n",
       "      <td>3501.44189</td>\n",
       "      <td>3494.67920</td>\n",
       "      <td>299</td>\n",
       "      <td>chr1:633820-634277</td>\n",
       "      <td>ATAC-seq:chr1:633820-634277</td>\n",
       "      <td>ATAC-seq:chr1:633820-634277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521223</td>\n",
       "      <td>22521670</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.38412</td>\n",
       "      <td>3410.55688</td>\n",
       "      <td>3403.83081</td>\n",
       "      <td>129</td>\n",
       "      <td>chr17:22521223-22521670</td>\n",
       "      <td>ATAC-seq:chr17:22521223-22521670</td>\n",
       "      <td>ATAC-seq:chr17:22521223-22521670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152796</th>\n",
       "      <td>chr11</td>\n",
       "      <td>112961786</td>\n",
       "      <td>112963648</td>\n",
       "      <td>.</td>\n",
       "      <td>827</td>\n",
       "      <td>.</td>\n",
       "      <td>1.91464</td>\n",
       "      <td>3.05557</td>\n",
       "      <td>1.40385</td>\n",
       "      <td>308</td>\n",
       "      <td>chr11:112961786-112963648</td>\n",
       "      <td>ATAC-seq:chr11:112961786-112963648</td>\n",
       "      <td>ATAC-seq:chr11:112961786-112963648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152797</th>\n",
       "      <td>chr7</td>\n",
       "      <td>13551221</td>\n",
       "      <td>13552318</td>\n",
       "      <td>.</td>\n",
       "      <td>737</td>\n",
       "      <td>.</td>\n",
       "      <td>2.55754</td>\n",
       "      <td>3.05534</td>\n",
       "      <td>1.40367</td>\n",
       "      <td>280</td>\n",
       "      <td>chr7:13551221-13552318</td>\n",
       "      <td>ATAC-seq:chr7:13551221-13552318</td>\n",
       "      <td>ATAC-seq:chr7:13551221-13552318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152798</th>\n",
       "      <td>chr4</td>\n",
       "      <td>99300712</td>\n",
       "      <td>99302001</td>\n",
       "      <td>.</td>\n",
       "      <td>840</td>\n",
       "      <td>.</td>\n",
       "      <td>2.55754</td>\n",
       "      <td>3.05534</td>\n",
       "      <td>1.40367</td>\n",
       "      <td>56</td>\n",
       "      <td>chr4:99300712-99302001</td>\n",
       "      <td>ATAC-seq:chr4:99300712-99302001</td>\n",
       "      <td>ATAC-seq:chr4:99300712-99302001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152799</th>\n",
       "      <td>chr16</td>\n",
       "      <td>113146</td>\n",
       "      <td>114212</td>\n",
       "      <td>.</td>\n",
       "      <td>891</td>\n",
       "      <td>.</td>\n",
       "      <td>2.55754</td>\n",
       "      <td>3.05534</td>\n",
       "      <td>1.40367</td>\n",
       "      <td>813</td>\n",
       "      <td>chr16:113146-114212</td>\n",
       "      <td>ATAC-seq:chr16:113146-114212</td>\n",
       "      <td>ATAC-seq:chr16:113146-114212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152800</th>\n",
       "      <td>chr14</td>\n",
       "      <td>77027859</td>\n",
       "      <td>77029044</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>1.77600</td>\n",
       "      <td>3.05499</td>\n",
       "      <td>1.40359</td>\n",
       "      <td>381</td>\n",
       "      <td>chr14:77027859-77029044</td>\n",
       "      <td>ATAC-seq:chr14:77027859-77029044</td>\n",
       "      <td>ATAC-seq:chr14:77027859-77029044</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>152801 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629232     630132    .   1000      .      8.03966   \n",
       "1        chr1     629232     630132    .   1000      .      7.56924   \n",
       "2        chr1     633820     634277    .   1000      .      6.97965   \n",
       "3        chr1     633820     634277    .   1000      .      6.06016   \n",
       "4       chr17   22521223   22521670    .   1000      .     32.38412   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "152796  chr11  112961786  112963648    .    827      .      1.91464   \n",
       "152797   chr7   13551221   13552318    .    737      .      2.55754   \n",
       "152798   chr4   99300712   99302001    .    840      .      2.55754   \n",
       "152799  chr16     113146     114212    .    891      .      2.55754   \n",
       "152800  chr14   77027859   77029044    .   1000      .      1.77600   \n",
       "\n",
       "            pValue      qValue  peak                      index  \\\n",
       "0       5807.30420  5799.29053   471         chr1:629232-630132   \n",
       "1       5235.43115  5228.03027   665         chr1:629232-630132   \n",
       "2       4538.55518  4531.56543   120         chr1:633820-634277   \n",
       "3       3501.44189  3494.67920   299         chr1:633820-634277   \n",
       "4       3410.55688  3403.83081   129    chr17:22521223-22521670   \n",
       "...            ...         ...   ...                        ...   \n",
       "152796     3.05557     1.40385   308  chr11:112961786-112963648   \n",
       "152797     3.05534     1.40367   280     chr7:13551221-13552318   \n",
       "152798     3.05534     1.40367    56     chr4:99300712-99302001   \n",
       "152799     3.05534     1.40367   813        chr16:113146-114212   \n",
       "152800     3.05499     1.40359   381    chr14:77027859-77029044   \n",
       "\n",
       "                                   ATAC_MT                             ATAC_MB  \n",
       "0              ATAC-seq:chr1:629232-630132         ATAC-seq:chr1:629232-630132  \n",
       "1              ATAC-seq:chr1:629232-630132         ATAC-seq:chr1:629232-630132  \n",
       "2              ATAC-seq:chr1:633820-634277         ATAC-seq:chr1:633820-634277  \n",
       "3              ATAC-seq:chr1:633820-634277         ATAC-seq:chr1:633820-634277  \n",
       "4         ATAC-seq:chr17:22521223-22521670    ATAC-seq:chr17:22521223-22521670  \n",
       "...                                    ...                                 ...  \n",
       "152796  ATAC-seq:chr11:112961786-112963648  ATAC-seq:chr11:112961786-112963648  \n",
       "152797     ATAC-seq:chr7:13551221-13552318     ATAC-seq:chr7:13551221-13552318  \n",
       "152798     ATAC-seq:chr4:99300712-99302001     ATAC-seq:chr4:99300712-99302001  \n",
       "152799        ATAC-seq:chr16:113146-114212        ATAC-seq:chr16:113146-114212  \n",
       "152800    ATAC-seq:chr14:77027859-77029044    ATAC-seq:chr14:77027859-77029044  \n",
       "\n",
       "[152801 rows x 13 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b487e865-8aa9-4ee8-845a-c13884bcbaa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_atac_uniq = MB_atac.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "36b1f4fa-0cfc-463c-baf0-fd21451af271",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>ATAC_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629232</td>\n",
       "      <td>630132</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>8.03966</td>\n",
       "      <td>5807.30420</td>\n",
       "      <td>5799.29053</td>\n",
       "      <td>471</td>\n",
       "      <td>chr1:629232-630132</td>\n",
       "      <td>ATAC-seq:chr1:629232-630132</td>\n",
       "      <td>ATAC-seq:chr1:629232-630132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633820</td>\n",
       "      <td>634277</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.97965</td>\n",
       "      <td>4538.55518</td>\n",
       "      <td>4531.56543</td>\n",
       "      <td>120</td>\n",
       "      <td>chr1:633820-634277</td>\n",
       "      <td>ATAC-seq:chr1:633820-634277</td>\n",
       "      <td>ATAC-seq:chr1:633820-634277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521223</td>\n",
       "      <td>22521670</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.38412</td>\n",
       "      <td>3410.55688</td>\n",
       "      <td>3403.83081</td>\n",
       "      <td>129</td>\n",
       "      <td>chr17:22521223-22521670</td>\n",
       "      <td>ATAC-seq:chr17:22521223-22521670</td>\n",
       "      <td>ATAC-seq:chr17:22521223-22521670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>631816</td>\n",
       "      <td>632360</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>3.19350</td>\n",
       "      <td>908.64136</td>\n",
       "      <td>902.35797</td>\n",
       "      <td>96</td>\n",
       "      <td>chr1:631816-632360</td>\n",
       "      <td>ATAC-seq:chr1:631816-632360</td>\n",
       "      <td>ATAC-seq:chr1:631816-632360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630706</td>\n",
       "      <td>631438</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>2.52313</td>\n",
       "      <td>488.17590</td>\n",
       "      <td>482.05035</td>\n",
       "      <td>91</td>\n",
       "      <td>chr1:630706-631438</td>\n",
       "      <td>ATAC-seq:chr1:630706-631438</td>\n",
       "      <td>ATAC-seq:chr1:630706-631438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152032</th>\n",
       "      <td>chr13</td>\n",
       "      <td>66933666</td>\n",
       "      <td>66933816</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>2.65957</td>\n",
       "      <td>3.22750</td>\n",
       "      <td>1.56448</td>\n",
       "      <td>55</td>\n",
       "      <td>chr13:66933666-66933816</td>\n",
       "      <td>ATAC-seq:chr13:66933666-66933816</td>\n",
       "      <td>ATAC-seq:chr13:66933666-66933816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152413</th>\n",
       "      <td>chr11</td>\n",
       "      <td>6357816</td>\n",
       "      <td>6358002</td>\n",
       "      <td>.</td>\n",
       "      <td>894</td>\n",
       "      <td>.</td>\n",
       "      <td>2.27445</td>\n",
       "      <td>3.14894</td>\n",
       "      <td>1.49114</td>\n",
       "      <td>76</td>\n",
       "      <td>chr11:6357816-6358002</td>\n",
       "      <td>ATAC-seq:chr11:6357816-6358002</td>\n",
       "      <td>ATAC-seq:chr11:6357816-6358002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152418</th>\n",
       "      <td>chr13</td>\n",
       "      <td>36965489</td>\n",
       "      <td>36965657</td>\n",
       "      <td>.</td>\n",
       "      <td>572</td>\n",
       "      <td>.</td>\n",
       "      <td>2.53165</td>\n",
       "      <td>3.14803</td>\n",
       "      <td>1.49029</td>\n",
       "      <td>97</td>\n",
       "      <td>chr13:36965489-36965657</td>\n",
       "      <td>ATAC-seq:chr13:36965489-36965657</td>\n",
       "      <td>ATAC-seq:chr13:36965489-36965657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152505</th>\n",
       "      <td>chr7</td>\n",
       "      <td>44382813</td>\n",
       "      <td>44383132</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>2.02117</td>\n",
       "      <td>3.12768</td>\n",
       "      <td>1.47136</td>\n",
       "      <td>50</td>\n",
       "      <td>chr7:44382813-44383132</td>\n",
       "      <td>ATAC-seq:chr7:44382813-44383132</td>\n",
       "      <td>ATAC-seq:chr7:44382813-44383132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152725</th>\n",
       "      <td>chr17</td>\n",
       "      <td>41836710</td>\n",
       "      <td>41836891</td>\n",
       "      <td>.</td>\n",
       "      <td>734</td>\n",
       "      <td>.</td>\n",
       "      <td>2.56739</td>\n",
       "      <td>3.07202</td>\n",
       "      <td>1.41915</td>\n",
       "      <td>83</td>\n",
       "      <td>chr17:41836710-41836891</td>\n",
       "      <td>ATAC-seq:chr17:41836710-41836891</td>\n",
       "      <td>ATAC-seq:chr17:41836710-41836891</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51939 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr     Start       End name  score strand  signalValue      pValue  \\\n",
       "0        chr1    629232    630132    .   1000      .      8.03966  5807.30420   \n",
       "2        chr1    633820    634277    .   1000      .      6.97965  4538.55518   \n",
       "4       chr17  22521223  22521670    .   1000      .     32.38412  3410.55688   \n",
       "6        chr1    631816    632360    .   1000      .      3.19350   908.64136   \n",
       "8        chr1    630706    631438    .   1000      .      2.52313   488.17590   \n",
       "...       ...       ...       ...  ...    ...    ...          ...         ...   \n",
       "152032  chr13  66933666  66933816    .   1000      .      2.65957     3.22750   \n",
       "152413  chr11   6357816   6358002    .    894      .      2.27445     3.14894   \n",
       "152418  chr13  36965489  36965657    .    572      .      2.53165     3.14803   \n",
       "152505   chr7  44382813  44383132    .   1000      .      2.02117     3.12768   \n",
       "152725  chr17  41836710  41836891    .    734      .      2.56739     3.07202   \n",
       "\n",
       "            qValue  peak                    index  \\\n",
       "0       5799.29053   471       chr1:629232-630132   \n",
       "2       4531.56543   120       chr1:633820-634277   \n",
       "4       3403.83081   129  chr17:22521223-22521670   \n",
       "6        902.35797    96       chr1:631816-632360   \n",
       "8        482.05035    91       chr1:630706-631438   \n",
       "...            ...   ...                      ...   \n",
       "152032     1.56448    55  chr13:66933666-66933816   \n",
       "152413     1.49114    76    chr11:6357816-6358002   \n",
       "152418     1.49029    97  chr13:36965489-36965657   \n",
       "152505     1.47136    50   chr7:44382813-44383132   \n",
       "152725     1.41915    83  chr17:41836710-41836891   \n",
       "\n",
       "                                 ATAC_MT                           ATAC_MB  \n",
       "0            ATAC-seq:chr1:629232-630132       ATAC-seq:chr1:629232-630132  \n",
       "2            ATAC-seq:chr1:633820-634277       ATAC-seq:chr1:633820-634277  \n",
       "4       ATAC-seq:chr17:22521223-22521670  ATAC-seq:chr17:22521223-22521670  \n",
       "6            ATAC-seq:chr1:631816-632360       ATAC-seq:chr1:631816-632360  \n",
       "8            ATAC-seq:chr1:630706-631438       ATAC-seq:chr1:630706-631438  \n",
       "...                                  ...                               ...  \n",
       "152032  ATAC-seq:chr13:66933666-66933816  ATAC-seq:chr13:66933666-66933816  \n",
       "152413    ATAC-seq:chr11:6357816-6358002    ATAC-seq:chr11:6357816-6358002  \n",
       "152418  ATAC-seq:chr13:36965489-36965657  ATAC-seq:chr13:36965489-36965657  \n",
       "152505   ATAC-seq:chr7:44382813-44383132   ATAC-seq:chr7:44382813-44383132  \n",
       "152725  ATAC-seq:chr17:41836710-41836891  ATAC-seq:chr17:41836710-41836891  \n",
       "\n",
       "[51939 rows x 13 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_atac_uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6537331d-3c8a-46c0-828f-d5cb5edcd06e",
   "metadata": {},
   "source": [
    "# MB DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a4a0ed79-75f5-4e2f-90ea-783256335851",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_DNase = pd.read_csv('/mnt/10T-backup-mingju/ifar/DNase_seq/myoblast/ENCFF647RNC_w_header.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d1431f3c-7a81-4fda-a912-8329de7365b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.039784</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.712660</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.072650</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.079569</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.472224</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373417</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26525100</td>\n",
       "      <td>26525284</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.081299</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373418</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26549406</td>\n",
       "      <td>26549574</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.076109</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373419</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663668</td>\n",
       "      <td>26663747</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.081299</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373420</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671160</td>\n",
       "      <td>26671440</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.306167</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373421</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671580</td>\n",
       "      <td>26671699</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.133191</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373422 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0       chr1    180791    180871    .      0      .     0.039784      -1   \n",
       "1       chr1    181400    181580    .      0      .     0.712660      -1   \n",
       "2       chr1    182681    182820    .      0      .     0.072650      -1   \n",
       "3       chr1    191400    191540    .      0      .     0.079569      -1   \n",
       "4       chr1    268011    268080    .      0      .     0.472224      -1   \n",
       "...      ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "373417  chrY  26525100  26525284    .      0      .     0.081299      -1   \n",
       "373418  chrY  26549406  26549574    .      0      .     0.076109      -1   \n",
       "373419  chrY  26663668  26663747    .      0      .     0.081299      -1   \n",
       "373420  chrY  26671160  26671440    .      0      .     0.306167      -1   \n",
       "373421  chrY  26671580  26671699    .      0      .     0.133191      -1   \n",
       "\n",
       "        qValue  peak  \n",
       "0           -1    75  \n",
       "1           -1    75  \n",
       "2           -1    75  \n",
       "3           -1    75  \n",
       "4           -1    75  \n",
       "...        ...   ...  \n",
       "373417      -1    75  \n",
       "373418      -1    75  \n",
       "373419      -1    75  \n",
       "373420      -1    75  \n",
       "373421      -1    75  \n",
       "\n",
       "[373422 rows x 10 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f7f2d258-eb5a-4f7b-a1fc-91482002c5a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_DNase['index'] = MB_DNase['Chr'] + \":\" + MB_DNase['Start'].astype(str) + \"-\" + MB_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a99223c6-f504-42ce-830e-62fdac815009",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_DNase['DNase_MB'] = \"DNase-seq:\" + MB_DNase['Chr'] + \":\" + MB_DNase['Start'].astype(str) + \"-\" + MB_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "390bd32f-5930-44f0-9a61-c518c16855b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>DNase_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.039784</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:180791-180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.712660</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181400-181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.072650</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:182681-182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.079569</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:191400-191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.472224</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:268011-268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373417</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26525100</td>\n",
       "      <td>26525284</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.081299</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26525100-26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373418</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26549406</td>\n",
       "      <td>26549574</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.076109</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26549406-26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373419</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663668</td>\n",
       "      <td>26663747</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.081299</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26663668-26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373420</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671160</td>\n",
       "      <td>26671440</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.306167</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26671160-26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373421</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671580</td>\n",
       "      <td>26671699</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.133191</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26671580-26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373422 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0       chr1    180791    180871    .      0      .     0.039784      -1   \n",
       "1       chr1    181400    181580    .      0      .     0.712660      -1   \n",
       "2       chr1    182681    182820    .      0      .     0.072650      -1   \n",
       "3       chr1    191400    191540    .      0      .     0.079569      -1   \n",
       "4       chr1    268011    268080    .      0      .     0.472224      -1   \n",
       "...      ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "373417  chrY  26525100  26525284    .      0      .     0.081299      -1   \n",
       "373418  chrY  26549406  26549574    .      0      .     0.076109      -1   \n",
       "373419  chrY  26663668  26663747    .      0      .     0.081299      -1   \n",
       "373420  chrY  26671160  26671440    .      0      .     0.306167      -1   \n",
       "373421  chrY  26671580  26671699    .      0      .     0.133191      -1   \n",
       "\n",
       "        qValue  peak                   index  \\\n",
       "0           -1    75      chr1:180791-180871   \n",
       "1           -1    75      chr1:181400-181580   \n",
       "2           -1    75      chr1:182681-182820   \n",
       "3           -1    75      chr1:191400-191540   \n",
       "4           -1    75      chr1:268011-268080   \n",
       "...        ...   ...                     ...   \n",
       "373417      -1    75  chrY:26525100-26525284   \n",
       "373418      -1    75  chrY:26549406-26549574   \n",
       "373419      -1    75  chrY:26663668-26663747   \n",
       "373420      -1    75  chrY:26671160-26671440   \n",
       "373421      -1    75  chrY:26671580-26671699   \n",
       "\n",
       "                                DNase_MT                          DNase_MB  \n",
       "0           DNase-seq:chr1:180791-180871      DNase-seq:chr1:180791-180871  \n",
       "1           DNase-seq:chr1:181400-181580      DNase-seq:chr1:181400-181580  \n",
       "2           DNase-seq:chr1:182681-182820      DNase-seq:chr1:182681-182820  \n",
       "3           DNase-seq:chr1:191400-191540      DNase-seq:chr1:191400-191540  \n",
       "4           DNase-seq:chr1:268011-268080      DNase-seq:chr1:268011-268080  \n",
       "...                                  ...                               ...  \n",
       "373417  DNase-seq:chrY:26525100-26525284  DNase-seq:chrY:26525100-26525284  \n",
       "373418  DNase-seq:chrY:26549406-26549574  DNase-seq:chrY:26549406-26549574  \n",
       "373419  DNase-seq:chrY:26663668-26663747  DNase-seq:chrY:26663668-26663747  \n",
       "373420  DNase-seq:chrY:26671160-26671440  DNase-seq:chrY:26671160-26671440  \n",
       "373421  DNase-seq:chrY:26671580-26671699  DNase-seq:chrY:26671580-26671699  \n",
       "\n",
       "[373422 rows x 13 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "118add04-3a86-4a9c-a938-e4e1c1923012",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_DNase_uniq = MB_DNase.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "bcd28bc9-2d1d-4792-9c5c-ce380db5e243",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>DNase_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.039784</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:180791-180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.712660</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181400-181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.072650</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:182681-182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.079569</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:191400-191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.472224</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:268011-268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373417</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26525100</td>\n",
       "      <td>26525284</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.081299</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26525100-26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373418</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26549406</td>\n",
       "      <td>26549574</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.076109</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26549406-26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373419</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663668</td>\n",
       "      <td>26663747</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.081299</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26663668-26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373420</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671160</td>\n",
       "      <td>26671440</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.306167</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26671160-26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373421</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671580</td>\n",
       "      <td>26671699</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.133191</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26671580-26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373422 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0       chr1    180791    180871    .      0      .     0.039784      -1   \n",
       "1       chr1    181400    181580    .      0      .     0.712660      -1   \n",
       "2       chr1    182681    182820    .      0      .     0.072650      -1   \n",
       "3       chr1    191400    191540    .      0      .     0.079569      -1   \n",
       "4       chr1    268011    268080    .      0      .     0.472224      -1   \n",
       "...      ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "373417  chrY  26525100  26525284    .      0      .     0.081299      -1   \n",
       "373418  chrY  26549406  26549574    .      0      .     0.076109      -1   \n",
       "373419  chrY  26663668  26663747    .      0      .     0.081299      -1   \n",
       "373420  chrY  26671160  26671440    .      0      .     0.306167      -1   \n",
       "373421  chrY  26671580  26671699    .      0      .     0.133191      -1   \n",
       "\n",
       "        qValue  peak                   index  \\\n",
       "0           -1    75      chr1:180791-180871   \n",
       "1           -1    75      chr1:181400-181580   \n",
       "2           -1    75      chr1:182681-182820   \n",
       "3           -1    75      chr1:191400-191540   \n",
       "4           -1    75      chr1:268011-268080   \n",
       "...        ...   ...                     ...   \n",
       "373417      -1    75  chrY:26525100-26525284   \n",
       "373418      -1    75  chrY:26549406-26549574   \n",
       "373419      -1    75  chrY:26663668-26663747   \n",
       "373420      -1    75  chrY:26671160-26671440   \n",
       "373421      -1    75  chrY:26671580-26671699   \n",
       "\n",
       "                                DNase_MT                          DNase_MB  \n",
       "0           DNase-seq:chr1:180791-180871      DNase-seq:chr1:180791-180871  \n",
       "1           DNase-seq:chr1:181400-181580      DNase-seq:chr1:181400-181580  \n",
       "2           DNase-seq:chr1:182681-182820      DNase-seq:chr1:182681-182820  \n",
       "3           DNase-seq:chr1:191400-191540      DNase-seq:chr1:191400-191540  \n",
       "4           DNase-seq:chr1:268011-268080      DNase-seq:chr1:268011-268080  \n",
       "...                                  ...                               ...  \n",
       "373417  DNase-seq:chrY:26525100-26525284  DNase-seq:chrY:26525100-26525284  \n",
       "373418  DNase-seq:chrY:26549406-26549574  DNase-seq:chrY:26549406-26549574  \n",
       "373419  DNase-seq:chrY:26663668-26663747  DNase-seq:chrY:26663668-26663747  \n",
       "373420  DNase-seq:chrY:26671160-26671440  DNase-seq:chrY:26671160-26671440  \n",
       "373421  DNase-seq:chrY:26671580-26671699  DNase-seq:chrY:26671580-26671699  \n",
       "\n",
       "[373422 rows x 13 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_DNase_uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "38dd58db-1c4f-4810-8458-037b15ea7ba7",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase = pd.merge(MB_atac_uniq[['index','Chr','Start','End','ATAC_MB']], MB_DNase_uniq[['index','DNase_MB']], on='index', how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "eed0c8d6-2f54-4b61-be73-65299bc61f30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MB</th>\n",
       "      <th>DNase_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006340-100006580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006340-100006580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100006347-100006849</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100006347.0</td>\n",
       "      <td>100006849.0</td>\n",
       "      <td>ATAC-seq:chr10:100006347-100006849</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100008320-100008500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100008320-100008500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100009206-100010573</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100009206.0</td>\n",
       "      <td>100010573.0</td>\n",
       "      <td>ATAC-seq:chr10:100009206-100010573</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100009351-100009460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100009351-100009460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425356</th>\n",
       "      <td>chrY:9321380-9321440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9321380-9321440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425357</th>\n",
       "      <td>chrY:9323340-9323540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9323340-9323540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425358</th>\n",
       "      <td>chrY:9334900-9335080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9334900-9335080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425359</th>\n",
       "      <td>chrY:9566200-9566420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9566200-9566420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425360</th>\n",
       "      <td>chrY:9570441-9570529</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9570441-9570529</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>425361 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index    Chr        Start          End  \\\n",
       "0       chr10:100006340-100006580    NaN          NaN          NaN   \n",
       "1       chr10:100006347-100006849  chr10  100006347.0  100006849.0   \n",
       "2       chr10:100008320-100008500    NaN          NaN          NaN   \n",
       "3       chr10:100009206-100010573  chr10  100009206.0  100010573.0   \n",
       "4       chr10:100009351-100009460    NaN          NaN          NaN   \n",
       "...                           ...    ...          ...          ...   \n",
       "425356       chrY:9321380-9321440    NaN          NaN          NaN   \n",
       "425357       chrY:9323340-9323540    NaN          NaN          NaN   \n",
       "425358       chrY:9334900-9335080    NaN          NaN          NaN   \n",
       "425359       chrY:9566200-9566420    NaN          NaN          NaN   \n",
       "425360       chrY:9570441-9570529    NaN          NaN          NaN   \n",
       "\n",
       "                                   ATAC_MB  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100006347-100006849   \n",
       "2                                      NaN   \n",
       "3       ATAC-seq:chr10:100009206-100010573   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "425356                                 NaN   \n",
       "425357                                 NaN   \n",
       "425358                                 NaN   \n",
       "425359                                 NaN   \n",
       "425360                                 NaN   \n",
       "\n",
       "                                   DNase_MB  \n",
       "0       DNase-seq:chr10:100006340-100006580  \n",
       "1                                       NaN  \n",
       "2       DNase-seq:chr10:100008320-100008500  \n",
       "3                                       NaN  \n",
       "4       DNase-seq:chr10:100009351-100009460  \n",
       "...                                     ...  \n",
       "425356       DNase-seq:chrY:9321380-9321440  \n",
       "425357       DNase-seq:chrY:9323340-9323540  \n",
       "425358       DNase-seq:chrY:9334900-9335080  \n",
       "425359       DNase-seq:chrY:9566200-9566420  \n",
       "425360       DNase-seq:chrY:9570441-9570529  \n",
       "\n",
       "[425361 rows x 6 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "a9c9b6be-5b84-4c50-9956-fc78965b93ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[\"OpenChromatin\"] = merge_atac_dnase.apply(\n",
    "    lambda row: f\"{row['ATAC_MB']};{row['DNase_MB']}\" if pd.notna(row['ATAC_MB']) and pd.notna(row['DNase_MB']) and row['DNase_MB'] != \"\"\n",
    "                else row['ATAC_MB'] if pd.notna(row['ATAC_MB'])\n",
    "                else row['DNase_MB'] if pd.notna(row['DNase_MB']) and row['DNase_MB'] != \"\"\n",
    "                else \"\", \n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f664675c-83a1-4733-b731-8cd4733396a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MB</th>\n",
       "      <th>DNase_MB</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006340-100006580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006340-100006580</td>\n",
       "      <td>DNase-seq:chr10:100006340-100006580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100006347-100006849</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100006347.0</td>\n",
       "      <td>100006849.0</td>\n",
       "      <td>ATAC-seq:chr10:100006347-100006849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100006347-100006849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100008320-100008500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100008320-100008500</td>\n",
       "      <td>DNase-seq:chr10:100008320-100008500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100009206-100010573</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100009206.0</td>\n",
       "      <td>100010573.0</td>\n",
       "      <td>ATAC-seq:chr10:100009206-100010573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100009206-100010573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100009351-100009460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100009351-100009460</td>\n",
       "      <td>DNase-seq:chr10:100009351-100009460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425356</th>\n",
       "      <td>chrY:9321380-9321440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9321380-9321440</td>\n",
       "      <td>DNase-seq:chrY:9321380-9321440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425357</th>\n",
       "      <td>chrY:9323340-9323540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9323340-9323540</td>\n",
       "      <td>DNase-seq:chrY:9323340-9323540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425358</th>\n",
       "      <td>chrY:9334900-9335080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9334900-9335080</td>\n",
       "      <td>DNase-seq:chrY:9334900-9335080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425359</th>\n",
       "      <td>chrY:9566200-9566420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9566200-9566420</td>\n",
       "      <td>DNase-seq:chrY:9566200-9566420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425360</th>\n",
       "      <td>chrY:9570441-9570529</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9570441-9570529</td>\n",
       "      <td>DNase-seq:chrY:9570441-9570529</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>425361 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index    Chr        Start          End  \\\n",
       "0       chr10:100006340-100006580    NaN          NaN          NaN   \n",
       "1       chr10:100006347-100006849  chr10  100006347.0  100006849.0   \n",
       "2       chr10:100008320-100008500    NaN          NaN          NaN   \n",
       "3       chr10:100009206-100010573  chr10  100009206.0  100010573.0   \n",
       "4       chr10:100009351-100009460    NaN          NaN          NaN   \n",
       "...                           ...    ...          ...          ...   \n",
       "425356       chrY:9321380-9321440    NaN          NaN          NaN   \n",
       "425357       chrY:9323340-9323540    NaN          NaN          NaN   \n",
       "425358       chrY:9334900-9335080    NaN          NaN          NaN   \n",
       "425359       chrY:9566200-9566420    NaN          NaN          NaN   \n",
       "425360       chrY:9570441-9570529    NaN          NaN          NaN   \n",
       "\n",
       "                                   ATAC_MB  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100006347-100006849   \n",
       "2                                      NaN   \n",
       "3       ATAC-seq:chr10:100009206-100010573   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "425356                                 NaN   \n",
       "425357                                 NaN   \n",
       "425358                                 NaN   \n",
       "425359                                 NaN   \n",
       "425360                                 NaN   \n",
       "\n",
       "                                   DNase_MB  \\\n",
       "0       DNase-seq:chr10:100006340-100006580   \n",
       "1                                       NaN   \n",
       "2       DNase-seq:chr10:100008320-100008500   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100009351-100009460   \n",
       "...                                     ...   \n",
       "425356       DNase-seq:chrY:9321380-9321440   \n",
       "425357       DNase-seq:chrY:9323340-9323540   \n",
       "425358       DNase-seq:chrY:9334900-9335080   \n",
       "425359       DNase-seq:chrY:9566200-9566420   \n",
       "425360       DNase-seq:chrY:9570441-9570529   \n",
       "\n",
       "                              OpenChromatin  \n",
       "0       DNase-seq:chr10:100006340-100006580  \n",
       "1        ATAC-seq:chr10:100006347-100006849  \n",
       "2       DNase-seq:chr10:100008320-100008500  \n",
       "3        ATAC-seq:chr10:100009206-100010573  \n",
       "4       DNase-seq:chr10:100009351-100009460  \n",
       "...                                     ...  \n",
       "425356       DNase-seq:chrY:9321380-9321440  \n",
       "425357       DNase-seq:chrY:9323340-9323540  \n",
       "425358       DNase-seq:chrY:9334900-9335080  \n",
       "425359       DNase-seq:chrY:9566200-9566420  \n",
       "425360       DNase-seq:chrY:9570441-9570529  \n",
       "\n",
       "[425361 rows x 7 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "a243595e-64d5-41d6-986d-716aa8b71758",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_index(index_str):\n",
    "    if pd.isna(index_str) or index_str == \"NaN\":\n",
    "        return pd.NA, pd.NA, pd.NA\n",
    "    chr_part, range_part = index_str.split(\":\")\n",
    "    chr_value = chr_part.replace(\"chr\", \"\")\n",
    "    start, end = range_part.split(\"-\")\n",
    "    return chr_value, pd.to_numeric(start), pd.to_numeric(end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "9367a17f-4410-4afa-9460-3a9d80d175ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[['Chr', 'Start', 'End']] = merge_atac_dnase['index'].apply(lambda x: pd.Series(parse_index(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ae373cf5-1cd5-4c34-a158-988fcc95b7af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MB</th>\n",
       "      <th>DNase_MB</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006340-100006580</td>\n",
       "      <td>10</td>\n",
       "      <td>100006340</td>\n",
       "      <td>100006580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006340-100006580</td>\n",
       "      <td>DNase-seq:chr10:100006340-100006580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100006347-100006849</td>\n",
       "      <td>10</td>\n",
       "      <td>100006347</td>\n",
       "      <td>100006849</td>\n",
       "      <td>ATAC-seq:chr10:100006347-100006849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100006347-100006849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100008320-100008500</td>\n",
       "      <td>10</td>\n",
       "      <td>100008320</td>\n",
       "      <td>100008500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100008320-100008500</td>\n",
       "      <td>DNase-seq:chr10:100008320-100008500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100009206-100010573</td>\n",
       "      <td>10</td>\n",
       "      <td>100009206</td>\n",
       "      <td>100010573</td>\n",
       "      <td>ATAC-seq:chr10:100009206-100010573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100009206-100010573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100009351-100009460</td>\n",
       "      <td>10</td>\n",
       "      <td>100009351</td>\n",
       "      <td>100009460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100009351-100009460</td>\n",
       "      <td>DNase-seq:chr10:100009351-100009460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425356</th>\n",
       "      <td>chrY:9321380-9321440</td>\n",
       "      <td>Y</td>\n",
       "      <td>9321380</td>\n",
       "      <td>9321440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9321380-9321440</td>\n",
       "      <td>DNase-seq:chrY:9321380-9321440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425357</th>\n",
       "      <td>chrY:9323340-9323540</td>\n",
       "      <td>Y</td>\n",
       "      <td>9323340</td>\n",
       "      <td>9323540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9323340-9323540</td>\n",
       "      <td>DNase-seq:chrY:9323340-9323540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425358</th>\n",
       "      <td>chrY:9334900-9335080</td>\n",
       "      <td>Y</td>\n",
       "      <td>9334900</td>\n",
       "      <td>9335080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9334900-9335080</td>\n",
       "      <td>DNase-seq:chrY:9334900-9335080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425359</th>\n",
       "      <td>chrY:9566200-9566420</td>\n",
       "      <td>Y</td>\n",
       "      <td>9566200</td>\n",
       "      <td>9566420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9566200-9566420</td>\n",
       "      <td>DNase-seq:chrY:9566200-9566420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425360</th>\n",
       "      <td>chrY:9570441-9570529</td>\n",
       "      <td>Y</td>\n",
       "      <td>9570441</td>\n",
       "      <td>9570529</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9570441-9570529</td>\n",
       "      <td>DNase-seq:chrY:9570441-9570529</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>425361 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index Chr      Start        End  \\\n",
       "0       chr10:100006340-100006580  10  100006340  100006580   \n",
       "1       chr10:100006347-100006849  10  100006347  100006849   \n",
       "2       chr10:100008320-100008500  10  100008320  100008500   \n",
       "3       chr10:100009206-100010573  10  100009206  100010573   \n",
       "4       chr10:100009351-100009460  10  100009351  100009460   \n",
       "...                           ...  ..        ...        ...   \n",
       "425356       chrY:9321380-9321440   Y    9321380    9321440   \n",
       "425357       chrY:9323340-9323540   Y    9323340    9323540   \n",
       "425358       chrY:9334900-9335080   Y    9334900    9335080   \n",
       "425359       chrY:9566200-9566420   Y    9566200    9566420   \n",
       "425360       chrY:9570441-9570529   Y    9570441    9570529   \n",
       "\n",
       "                                   ATAC_MB  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100006347-100006849   \n",
       "2                                      NaN   \n",
       "3       ATAC-seq:chr10:100009206-100010573   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "425356                                 NaN   \n",
       "425357                                 NaN   \n",
       "425358                                 NaN   \n",
       "425359                                 NaN   \n",
       "425360                                 NaN   \n",
       "\n",
       "                                   DNase_MB  \\\n",
       "0       DNase-seq:chr10:100006340-100006580   \n",
       "1                                       NaN   \n",
       "2       DNase-seq:chr10:100008320-100008500   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100009351-100009460   \n",
       "...                                     ...   \n",
       "425356       DNase-seq:chrY:9321380-9321440   \n",
       "425357       DNase-seq:chrY:9323340-9323540   \n",
       "425358       DNase-seq:chrY:9334900-9335080   \n",
       "425359       DNase-seq:chrY:9566200-9566420   \n",
       "425360       DNase-seq:chrY:9570441-9570529   \n",
       "\n",
       "                              OpenChromatin  \n",
       "0       DNase-seq:chr10:100006340-100006580  \n",
       "1        ATAC-seq:chr10:100006347-100006849  \n",
       "2       DNase-seq:chr10:100008320-100008500  \n",
       "3        ATAC-seq:chr10:100009206-100010573  \n",
       "4       DNase-seq:chr10:100009351-100009460  \n",
       "...                                     ...  \n",
       "425356       DNase-seq:chrY:9321380-9321440  \n",
       "425357       DNase-seq:chrY:9323340-9323540  \n",
       "425358       DNase-seq:chrY:9334900-9335080  \n",
       "425359       DNase-seq:chrY:9566200-9566420  \n",
       "425360       DNase-seq:chrY:9570441-9570529  \n",
       "\n",
       "[425361 rows x 7 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "9a58dab4-e981-4bd8-8f9d-7cfc07d5a5cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the chromosome order: 1 to 22, X, Y\n",
    "chr_order = [str(i) for i in range(1, 23)] + [\"X\", \"Y\"]\n",
    "\n",
    "# Convert Chr to categorical with the specified order\n",
    "merge_atac_dnase[\"Chr\"] = pd.Categorical(merge_atac_dnase[\"Chr\"], categories=chr_order, ordered=True)\n",
    "\n",
    "# Sort by Chr and Start\n",
    "merge_atac_dnase_sorted = merge_atac_dnase.sort_values(by=[\"Chr\", \"Start\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ea650c3c-46eb-401b-8b54-f7d0d60cedec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MB</th>\n",
       "      <th>DNase_MB</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:180791-180871</td>\n",
       "      <td>1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:181400-181580</td>\n",
       "      <td>1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:182681-182820</td>\n",
       "      <td>1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:191400-191540</td>\n",
       "      <td>1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:268011-268080</td>\n",
       "      <td>1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425356</th>\n",
       "      <td>chrY:26525100-26525284</td>\n",
       "      <td>Y</td>\n",
       "      <td>26525100</td>\n",
       "      <td>26525284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425357</th>\n",
       "      <td>chrY:26549406-26549574</td>\n",
       "      <td>Y</td>\n",
       "      <td>26549406</td>\n",
       "      <td>26549574</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425358</th>\n",
       "      <td>chrY:26663668-26663747</td>\n",
       "      <td>Y</td>\n",
       "      <td>26663668</td>\n",
       "      <td>26663747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425359</th>\n",
       "      <td>chrY:26671160-26671440</td>\n",
       "      <td>Y</td>\n",
       "      <td>26671160</td>\n",
       "      <td>26671440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425360</th>\n",
       "      <td>chrY:26671580-26671699</td>\n",
       "      <td>Y</td>\n",
       "      <td>26671580</td>\n",
       "      <td>26671699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>425361 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End ATAC_MB  \\\n",
       "0           chr1:180791-180871   1    180791    180871     NaN   \n",
       "1           chr1:181400-181580   1    181400    181580     NaN   \n",
       "2           chr1:182681-182820   1    182681    182820     NaN   \n",
       "3           chr1:191400-191540   1    191400    191540     NaN   \n",
       "4           chr1:268011-268080   1    268011    268080     NaN   \n",
       "...                        ...  ..       ...       ...     ...   \n",
       "425356  chrY:26525100-26525284   Y  26525100  26525284     NaN   \n",
       "425357  chrY:26549406-26549574   Y  26549406  26549574     NaN   \n",
       "425358  chrY:26663668-26663747   Y  26663668  26663747     NaN   \n",
       "425359  chrY:26671160-26671440   Y  26671160  26671440     NaN   \n",
       "425360  chrY:26671580-26671699   Y  26671580  26671699     NaN   \n",
       "\n",
       "                                DNase_MB                     OpenChromatin  \n",
       "0           DNase-seq:chr1:180791-180871      DNase-seq:chr1:180791-180871  \n",
       "1           DNase-seq:chr1:181400-181580      DNase-seq:chr1:181400-181580  \n",
       "2           DNase-seq:chr1:182681-182820      DNase-seq:chr1:182681-182820  \n",
       "3           DNase-seq:chr1:191400-191540      DNase-seq:chr1:191400-191540  \n",
       "4           DNase-seq:chr1:268011-268080      DNase-seq:chr1:268011-268080  \n",
       "...                                  ...                               ...  \n",
       "425356  DNase-seq:chrY:26525100-26525284  DNase-seq:chrY:26525100-26525284  \n",
       "425357  DNase-seq:chrY:26549406-26549574  DNase-seq:chrY:26549406-26549574  \n",
       "425358  DNase-seq:chrY:26663668-26663747  DNase-seq:chrY:26663668-26663747  \n",
       "425359  DNase-seq:chrY:26671160-26671440  DNase-seq:chrY:26671160-26671440  \n",
       "425360  DNase-seq:chrY:26671580-26671699  DNase-seq:chrY:26671580-26671699  \n",
       "\n",
       "[425361 rows x 7 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5743febe-5606-483e-a92f-025632971f88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MB</th>\n",
       "      <th>DNase_MB</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>160265</th>\n",
       "      <td>chr1:180791-180871</td>\n",
       "      <td>1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160414</th>\n",
       "      <td>chr1:181400-181580</td>\n",
       "      <td>1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160591</th>\n",
       "      <td>chr1:182681-182820</td>\n",
       "      <td>1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161651</th>\n",
       "      <td>chr1:191400-191540</td>\n",
       "      <td>1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174189</th>\n",
       "      <td>chr1:268011-268080</td>\n",
       "      <td>1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151365</th>\n",
       "      <td>chr1:1157480-1157680</td>\n",
       "      <td>1</td>\n",
       "      <td>1157480</td>\n",
       "      <td>1157680</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1157480-1157680</td>\n",
       "      <td>DNase-seq:chr1:1157480-1157680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151950</th>\n",
       "      <td>chr1:1181480-1181700</td>\n",
       "      <td>1</td>\n",
       "      <td>1181480</td>\n",
       "      <td>1181700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1181480-1181700</td>\n",
       "      <td>DNase-seq:chr1:1181480-1181700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152037</th>\n",
       "      <td>chr1:1188040-1188240</td>\n",
       "      <td>1</td>\n",
       "      <td>1188040</td>\n",
       "      <td>1188240</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1188040-1188240</td>\n",
       "      <td>DNase-seq:chr1:1188040-1188240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152354</th>\n",
       "      <td>chr1:1201457-1201597</td>\n",
       "      <td>1</td>\n",
       "      <td>1201457</td>\n",
       "      <td>1201597</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1201457-1201597</td>\n",
       "      <td>DNase-seq:chr1:1201457-1201597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152433</th>\n",
       "      <td>chr1:1208013-1209439</td>\n",
       "      <td>1</td>\n",
       "      <td>1208013</td>\n",
       "      <td>1209439</td>\n",
       "      <td>ATAC-seq:chr1:1208013-1209439</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1208013-1209439</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       index Chr    Start      End  \\\n",
       "160265    chr1:180791-180871   1   180791   180871   \n",
       "160414    chr1:181400-181580   1   181400   181580   \n",
       "160591    chr1:182681-182820   1   182681   182820   \n",
       "161651    chr1:191400-191540   1   191400   191540   \n",
       "174189    chr1:268011-268080   1   268011   268080   \n",
       "...                      ...  ..      ...      ...   \n",
       "151365  chr1:1157480-1157680   1  1157480  1157680   \n",
       "151950  chr1:1181480-1181700   1  1181480  1181700   \n",
       "152037  chr1:1188040-1188240   1  1188040  1188240   \n",
       "152354  chr1:1201457-1201597   1  1201457  1201597   \n",
       "152433  chr1:1208013-1209439   1  1208013  1209439   \n",
       "\n",
       "                              ATAC_MB                        DNase_MB  \\\n",
       "160265                            NaN    DNase-seq:chr1:180791-180871   \n",
       "160414                            NaN    DNase-seq:chr1:181400-181580   \n",
       "160591                            NaN    DNase-seq:chr1:182681-182820   \n",
       "161651                            NaN    DNase-seq:chr1:191400-191540   \n",
       "174189                            NaN    DNase-seq:chr1:268011-268080   \n",
       "...                               ...                             ...   \n",
       "151365                            NaN  DNase-seq:chr1:1157480-1157680   \n",
       "151950                            NaN  DNase-seq:chr1:1181480-1181700   \n",
       "152037                            NaN  DNase-seq:chr1:1188040-1188240   \n",
       "152354                            NaN  DNase-seq:chr1:1201457-1201597   \n",
       "152433  ATAC-seq:chr1:1208013-1209439                             NaN   \n",
       "\n",
       "                         OpenChromatin  \n",
       "160265    DNase-seq:chr1:180791-180871  \n",
       "160414    DNase-seq:chr1:181400-181580  \n",
       "160591    DNase-seq:chr1:182681-182820  \n",
       "161651    DNase-seq:chr1:191400-191540  \n",
       "174189    DNase-seq:chr1:268011-268080  \n",
       "...                                ...  \n",
       "151365  DNase-seq:chr1:1157480-1157680  \n",
       "151950  DNase-seq:chr1:1181480-1181700  \n",
       "152037  DNase-seq:chr1:1188040-1188240  \n",
       "152354  DNase-seq:chr1:1201457-1201597  \n",
       "152433   ATAC-seq:chr1:1208013-1209439  \n",
       "\n",
       "[100 rows x 7 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.head(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "2f08790f-b04c-4444-ad34-a91ca893ace2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MB</th>\n",
       "      <th>DNase_MB</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [index, Chr, Start, End, ATAC_MB, DNase_MB, OpenChromatin]\n",
       "Index: []"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted[merge_atac_dnase_sorted['ATAC_MB'].notna()&merge_atac_dnase_sorted['DNase_MB'].notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "6e472aa7-b594-416b-9ab3-c8ddad47c265",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MB</th>\n",
       "      <th>DNase_MB</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>160265</th>\n",
       "      <td>chr1:180791-180871</td>\n",
       "      <td>1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160414</th>\n",
       "      <td>chr1:181400-181580</td>\n",
       "      <td>1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160591</th>\n",
       "      <td>chr1:182681-182820</td>\n",
       "      <td>1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161651</th>\n",
       "      <td>chr1:191400-191540</td>\n",
       "      <td>1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174189</th>\n",
       "      <td>chr1:268011-268080</td>\n",
       "      <td>1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424962</th>\n",
       "      <td>chrY:26525100-26525284</td>\n",
       "      <td>Y</td>\n",
       "      <td>26525100</td>\n",
       "      <td>26525284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424963</th>\n",
       "      <td>chrY:26549406-26549574</td>\n",
       "      <td>Y</td>\n",
       "      <td>26549406</td>\n",
       "      <td>26549574</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424964</th>\n",
       "      <td>chrY:26663668-26663747</td>\n",
       "      <td>Y</td>\n",
       "      <td>26663668</td>\n",
       "      <td>26663747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424965</th>\n",
       "      <td>chrY:26671160-26671440</td>\n",
       "      <td>Y</td>\n",
       "      <td>26671160</td>\n",
       "      <td>26671440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424966</th>\n",
       "      <td>chrY:26671580-26671699</td>\n",
       "      <td>Y</td>\n",
       "      <td>26671580</td>\n",
       "      <td>26671699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>425361 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End ATAC_MB  \\\n",
       "160265      chr1:180791-180871   1    180791    180871     NaN   \n",
       "160414      chr1:181400-181580   1    181400    181580     NaN   \n",
       "160591      chr1:182681-182820   1    182681    182820     NaN   \n",
       "161651      chr1:191400-191540   1    191400    191540     NaN   \n",
       "174189      chr1:268011-268080   1    268011    268080     NaN   \n",
       "...                        ...  ..       ...       ...     ...   \n",
       "424962  chrY:26525100-26525284   Y  26525100  26525284     NaN   \n",
       "424963  chrY:26549406-26549574   Y  26549406  26549574     NaN   \n",
       "424964  chrY:26663668-26663747   Y  26663668  26663747     NaN   \n",
       "424965  chrY:26671160-26671440   Y  26671160  26671440     NaN   \n",
       "424966  chrY:26671580-26671699   Y  26671580  26671699     NaN   \n",
       "\n",
       "                                DNase_MB                     OpenChromatin  \n",
       "160265      DNase-seq:chr1:180791-180871      DNase-seq:chr1:180791-180871  \n",
       "160414      DNase-seq:chr1:181400-181580      DNase-seq:chr1:181400-181580  \n",
       "160591      DNase-seq:chr1:182681-182820      DNase-seq:chr1:182681-182820  \n",
       "161651      DNase-seq:chr1:191400-191540      DNase-seq:chr1:191400-191540  \n",
       "174189      DNase-seq:chr1:268011-268080      DNase-seq:chr1:268011-268080  \n",
       "...                                  ...                               ...  \n",
       "424962  DNase-seq:chrY:26525100-26525284  DNase-seq:chrY:26525100-26525284  \n",
       "424963  DNase-seq:chrY:26549406-26549574  DNase-seq:chrY:26549406-26549574  \n",
       "424964  DNase-seq:chrY:26663668-26663747  DNase-seq:chrY:26663668-26663747  \n",
       "424965  DNase-seq:chrY:26671160-26671440  DNase-seq:chrY:26671160-26671440  \n",
       "424966  DNase-seq:chrY:26671580-26671699  DNase-seq:chrY:26671580-26671699  \n",
       "\n",
       "[425361 rows x 7 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "70d2b8f7-5097-4a33-a80d-97ccf4b65369",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_merge_atac_dnase_sorted_bed = merge_atac_dnase_sorted[[\"Chr\",\"Start\",\"End\",\"OpenChromatin\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "cb43fd4d-d539-48af-9397-de97ee407b8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>160265</th>\n",
       "      <td>1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160414</th>\n",
       "      <td>1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160591</th>\n",
       "      <td>1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161651</th>\n",
       "      <td>1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174189</th>\n",
       "      <td>1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424962</th>\n",
       "      <td>Y</td>\n",
       "      <td>26525100</td>\n",
       "      <td>26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424963</th>\n",
       "      <td>Y</td>\n",
       "      <td>26549406</td>\n",
       "      <td>26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424964</th>\n",
       "      <td>Y</td>\n",
       "      <td>26663668</td>\n",
       "      <td>26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424965</th>\n",
       "      <td>Y</td>\n",
       "      <td>26671160</td>\n",
       "      <td>26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424966</th>\n",
       "      <td>Y</td>\n",
       "      <td>26671580</td>\n",
       "      <td>26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>425361 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End                     OpenChromatin\n",
       "160265   1    180791    180871      DNase-seq:chr1:180791-180871\n",
       "160414   1    181400    181580      DNase-seq:chr1:181400-181580\n",
       "160591   1    182681    182820      DNase-seq:chr1:182681-182820\n",
       "161651   1    191400    191540      DNase-seq:chr1:191400-191540\n",
       "174189   1    268011    268080      DNase-seq:chr1:268011-268080\n",
       "...     ..       ...       ...                               ...\n",
       "424962   Y  26525100  26525284  DNase-seq:chrY:26525100-26525284\n",
       "424963   Y  26549406  26549574  DNase-seq:chrY:26549406-26549574\n",
       "424964   Y  26663668  26663747  DNase-seq:chrY:26663668-26663747\n",
       "424965   Y  26671160  26671440  DNase-seq:chrY:26671160-26671440\n",
       "424966   Y  26671580  26671699  DNase-seq:chrY:26671580-26671699\n",
       "\n",
       "[425361 rows x 4 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "298a4b03-0774-44a3-9c4c-2d0a5e04a25c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2192758/3870062363.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  MB_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + MB_merge_atac_dnase_sorted_bed['Chr'].astype(str)\n"
     ]
    }
   ],
   "source": [
    "MB_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + MB_merge_atac_dnase_sorted_bed['Chr'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1ca0b26a-0d6c-473c-8b62-56054a166587",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>160265</th>\n",
       "      <td>chr1</td>\n",
       "      <td>180791</td>\n",
       "      <td>180871</td>\n",
       "      <td>DNase-seq:chr1:180791-180871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160414</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181400</td>\n",
       "      <td>181580</td>\n",
       "      <td>DNase-seq:chr1:181400-181580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160591</th>\n",
       "      <td>chr1</td>\n",
       "      <td>182681</td>\n",
       "      <td>182820</td>\n",
       "      <td>DNase-seq:chr1:182681-182820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161651</th>\n",
       "      <td>chr1</td>\n",
       "      <td>191400</td>\n",
       "      <td>191540</td>\n",
       "      <td>DNase-seq:chr1:191400-191540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174189</th>\n",
       "      <td>chr1</td>\n",
       "      <td>268011</td>\n",
       "      <td>268080</td>\n",
       "      <td>DNase-seq:chr1:268011-268080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424962</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26525100</td>\n",
       "      <td>26525284</td>\n",
       "      <td>DNase-seq:chrY:26525100-26525284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424963</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26549406</td>\n",
       "      <td>26549574</td>\n",
       "      <td>DNase-seq:chrY:26549406-26549574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424964</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26663668</td>\n",
       "      <td>26663747</td>\n",
       "      <td>DNase-seq:chrY:26663668-26663747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424965</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671160</td>\n",
       "      <td>26671440</td>\n",
       "      <td>DNase-seq:chrY:26671160-26671440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424966</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26671580</td>\n",
       "      <td>26671699</td>\n",
       "      <td>DNase-seq:chrY:26671580-26671699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>425361 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End                     OpenChromatin\n",
       "160265  chr1    180791    180871      DNase-seq:chr1:180791-180871\n",
       "160414  chr1    181400    181580      DNase-seq:chr1:181400-181580\n",
       "160591  chr1    182681    182820      DNase-seq:chr1:182681-182820\n",
       "161651  chr1    191400    191540      DNase-seq:chr1:191400-191540\n",
       "174189  chr1    268011    268080      DNase-seq:chr1:268011-268080\n",
       "...      ...       ...       ...                               ...\n",
       "424962  chrY  26525100  26525284  DNase-seq:chrY:26525100-26525284\n",
       "424963  chrY  26549406  26549574  DNase-seq:chrY:26549406-26549574\n",
       "424964  chrY  26663668  26663747  DNase-seq:chrY:26663668-26663747\n",
       "424965  chrY  26671160  26671440  DNase-seq:chrY:26671160-26671440\n",
       "424966  chrY  26671580  26671699  DNase-seq:chrY:26671580-26671699\n",
       "\n",
       "[425361 rows x 4 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "96f1f652-46bd-4e89-a9f6-6e7421b59c60",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_merge_atac_dnase_sorted_bed.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/myoblast/hg38_OpenChromatin_MB.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb553935-ac7f-4f70-9307-041aec35b3be",
   "metadata": {},
   "source": [
    "# promoter table overlap with openchromatin "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "3b9a02b2-ca3d-4686-9901-420de1720a8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/promoter_regions_pcg_250bp_sorted.bed -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/myoblast/hg38_OpenChromatin_MB.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3fd31f1-2423-41ea-b4f9-7b244cd80535",
   "metadata": {},
   "source": [
    "### remove duplicated rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "f53ae2b1-85be-4e39-86d1-5fa0dc43fb92",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'chr_open','start_open','end_open','openChromatin','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "5f46f4d7-96c6-4b1a-b74d-b29cb1593c1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "3666029e-908b-4ee6-802b-bd23f7877331",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267493</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267494</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267495</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267496</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267497</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>267498 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "267493  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "267494  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "267495  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "267496  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "267497  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923760    923940   \n",
       "4                            5.0     chr1      923760    923940   \n",
       "...                          ...      ...         ...       ...   \n",
       "267493                       5.0        .          -1        -1   \n",
       "267494                       1.0        .          -1        -1   \n",
       "267495                       2.0        .          -1        -1   \n",
       "267496                       1.0        .          -1        -1   \n",
       "267497                       2.0        .          -1        -1   \n",
       "\n",
       "                       openChromatin  overlap  \n",
       "0                                  .        0  \n",
       "1                                  .        0  \n",
       "2                                  .        0  \n",
       "3       DNase-seq:chr1:923760-923940      180  \n",
       "4       DNase-seq:chr1:923760-923940      180  \n",
       "...                              ...      ...  \n",
       "267493                             .        0  \n",
       "267494                             .        0  \n",
       "267495                             .        0  \n",
       "267496                             .        0  \n",
       "267497                             .        0  \n",
       "\n",
       "[267498 rows x 13 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "449c8781-8235-4b95-a974-533f204d70c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924809</td>\n",
       "      <td>924915</td>\n",
       "      <td>DNase-seq:chr1:924809-924915</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940300</td>\n",
       "      <td>940520</td>\n",
       "      <td>DNase-seq:chr1:940300-940520</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941680</td>\n",
       "      <td>941900</td>\n",
       "      <td>DNase-seq:chr1:941680-941900</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959000</td>\n",
       "      <td>959844</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959220</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959220-959440</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959000</td>\n",
       "      <td>959844</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959220</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959220-959440</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959000</td>\n",
       "      <td>959844</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959220</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959220-959440</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960480</td>\n",
       "      <td>960660</td>\n",
       "      <td>DNase-seq:chr1:960480-960660</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960481</td>\n",
       "      <td>961383</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960481</td>\n",
       "      <td>961383</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383</td>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>961331</td>\n",
       "      <td>961512</td>\n",
       "      <td>DNase-seq:chr1:961331-961512</td>\n",
       "      <td>181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966132</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966132</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966132</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>982100</td>\n",
       "      <td>982340</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>982100</td>\n",
       "      <td>982340</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1</td>\n",
       "      <td>982100</td>\n",
       "      <td>982340</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom   start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1   65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1  451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1  686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1  923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1  923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1  924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1  925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1  925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1  925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1  925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1  925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1  925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1  930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1  939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1  940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1  940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1  941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1  941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1  945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1  946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1  958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1  959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1  959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "23  chr1  959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "24  chr1  959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "25  chr1  959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "26  chr1  959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "27  chr1  960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "28  chr1  960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "29  chr1  961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "30  chr1  961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "31  chr1  962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "32  chr1  963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "33  chr1  966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "34  chr1  966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "35  chr1  966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "36  chr1  970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "37  chr1  973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "38  chr1  976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "39  chr1  981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "40  chr1  981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "41  chr1  981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "42  chr1  999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "43  chr1  999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "44  chr1  999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "45  chr1  999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "46  chr1  999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "47  chr1  999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "48  chr1  999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "49  chr1  999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923760    923940   \n",
       "4                        5.0     chr1      923760    923940   \n",
       "5                        5.0     chr1      924809    924915   \n",
       "6                        5.0        .          -1        -1   \n",
       "7                        5.0        .          -1        -1   \n",
       "8                        5.0        .          -1        -1   \n",
       "9                        5.0        .          -1        -1   \n",
       "10                       5.0        .          -1        -1   \n",
       "11                       5.0        .          -1        -1   \n",
       "12                       5.0        .          -1        -1   \n",
       "13                       2.0        .          -1        -1   \n",
       "14                       5.0     chr1      940300    940520   \n",
       "15                       5.0        .          -1        -1   \n",
       "16                       3.0     chr1      941680    941900   \n",
       "17                       2.0        .          -1        -1   \n",
       "18                       5.0        .          -1        -1   \n",
       "19                       2.0        .          -1        -1   \n",
       "20                       5.0        .          -1        -1   \n",
       "21                       1.0     chr1      959000    959844   \n",
       "22                       1.0     chr1      959220    959440   \n",
       "23                       2.0     chr1      959000    959844   \n",
       "24                       2.0     chr1      959220    959440   \n",
       "25                       3.0     chr1      959000    959844   \n",
       "26                       3.0     chr1      959220    959440   \n",
       "27                       1.0     chr1      960480    960660   \n",
       "28                       1.0     chr1      960481    961383   \n",
       "29                       1.0     chr1      960481    961383   \n",
       "30                       1.0     chr1      961331    961512   \n",
       "31                       3.0        .          -1        -1   \n",
       "32                       2.0        .          -1        -1   \n",
       "33                       1.0     chr1      966132    967323   \n",
       "34                       1.0     chr1      966132    967323   \n",
       "35                       2.0     chr1      966132    967323   \n",
       "36                       3.0        .          -1        -1   \n",
       "37                       3.0        .          -1        -1   \n",
       "38                       1.0        .          -1        -1   \n",
       "39                       2.0     chr1      982100    982340   \n",
       "40                       5.0     chr1      982100    982340   \n",
       "41                       NaN     chr1      982100    982340   \n",
       "42                       3.0     chr1      999917   1000904   \n",
       "43                       3.0     chr1     1000080   1000280   \n",
       "44                       2.0     chr1      999917   1000904   \n",
       "45                       2.0     chr1     1000080   1000280   \n",
       "46                       1.0     chr1      999917   1000904   \n",
       "47                       1.0     chr1     1000080   1000280   \n",
       "48                       2.0     chr1      999917   1000904   \n",
       "49                       2.0     chr1     1000080   1000280   \n",
       "\n",
       "                     openChromatin  overlap  \n",
       "0                                .        0  \n",
       "1                                .        0  \n",
       "2                                .        0  \n",
       "3     DNase-seq:chr1:923760-923940      180  \n",
       "4     DNase-seq:chr1:923760-923940      180  \n",
       "5     DNase-seq:chr1:924809-924915       15  \n",
       "6                                .        0  \n",
       "7                                .        0  \n",
       "8                                .        0  \n",
       "9                                .        0  \n",
       "10                               .        0  \n",
       "11                               .        0  \n",
       "12                               .        0  \n",
       "13                               .        0  \n",
       "14    DNase-seq:chr1:940300-940520      220  \n",
       "15                               .        0  \n",
       "16    DNase-seq:chr1:941680-941900       47  \n",
       "17                               .        0  \n",
       "18                               .        0  \n",
       "19                               .        0  \n",
       "20                               .        0  \n",
       "21     ATAC-seq:chr1:959000-959844      500  \n",
       "22    DNase-seq:chr1:959220-959440      220  \n",
       "23     ATAC-seq:chr1:959000-959844      500  \n",
       "24    DNase-seq:chr1:959220-959440      220  \n",
       "25     ATAC-seq:chr1:959000-959844      500  \n",
       "26    DNase-seq:chr1:959220-959440      220  \n",
       "27    DNase-seq:chr1:960480-960660      180  \n",
       "28     ATAC-seq:chr1:960481-961383      353  \n",
       "29     ATAC-seq:chr1:960481-961383      184  \n",
       "30    DNase-seq:chr1:961331-961512      181  \n",
       "31                               .        0  \n",
       "32                               .        0  \n",
       "33     ATAC-seq:chr1:966132-967323      500  \n",
       "34     ATAC-seq:chr1:966132-967323      500  \n",
       "35     ATAC-seq:chr1:966132-967323      500  \n",
       "36                               .        0  \n",
       "37                               .        0  \n",
       "38                               .        0  \n",
       "39    DNase-seq:chr1:982100-982340      240  \n",
       "40    DNase-seq:chr1:982100-982340      240  \n",
       "41    DNase-seq:chr1:982100-982340      240  \n",
       "42    ATAC-seq:chr1:999917-1000904      314  \n",
       "43  DNase-seq:chr1:1000080-1000280      151  \n",
       "44    ATAC-seq:chr1:999917-1000904      429  \n",
       "45  DNase-seq:chr1:1000080-1000280      200  \n",
       "46    ATAC-seq:chr1:999917-1000904      430  \n",
       "47  DNase-seq:chr1:1000080-1000280      200  \n",
       "48    ATAC-seq:chr1:999917-1000904      500  \n",
       "49  DNase-seq:chr1:1000080-1000280      200  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "c87fb1ef-12a8-43f4-b3fd-4af9a1b4982d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "ad368f03-a970-4982-b1ba-5a35b74079f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267493</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267494</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267495</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267496</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267497</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>267498 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "267493  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "267494  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "267495  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "267496  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "267497  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923760    923940   \n",
       "4                            5.0     chr1      923760    923940   \n",
       "...                          ...      ...         ...       ...   \n",
       "267493                       5.0        .          -1        -1   \n",
       "267494                       1.0        .          -1        -1   \n",
       "267495                       2.0        .          -1        -1   \n",
       "267496                       1.0        .          -1        -1   \n",
       "267497                       2.0        .          -1        -1   \n",
       "\n",
       "                       openChromatin  overlap  \n",
       "0                                  .        0  \n",
       "1                                  .        0  \n",
       "2                                  .        0  \n",
       "3       DNase-seq:chr1:923760-923940      180  \n",
       "4       DNase-seq:chr1:923760-923940      180  \n",
       "...                              ...      ...  \n",
       "267493                             .        0  \n",
       "267494                             .        0  \n",
       "267495                             .        0  \n",
       "267496                             .        0  \n",
       "267497                             .        0  \n",
       "\n",
       "[267498 rows x 13 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "b8e5b8f5-db77-4d35-baf9-772c5bd2fcc2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923760</td>\n",
       "      <td>923940</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>924809</td>\n",
       "      <td>924915</td>\n",
       "      <td>DNase-seq:chr1:924809-924915</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940300</td>\n",
       "      <td>940520</td>\n",
       "      <td>DNase-seq:chr1:940300-940520</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>941680</td>\n",
       "      <td>941900</td>\n",
       "      <td>DNase-seq:chr1:941680-941900</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959000</td>\n",
       "      <td>959844</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959220</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959220-959440</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959000</td>\n",
       "      <td>959844</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959220</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959220-959440</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959000</td>\n",
       "      <td>959844</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959220</td>\n",
       "      <td>959440</td>\n",
       "      <td>DNase-seq:chr1:959220-959440</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960480</td>\n",
       "      <td>960660</td>\n",
       "      <td>DNase-seq:chr1:960480-960660</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960481</td>\n",
       "      <td>961383</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960481</td>\n",
       "      <td>961383</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383</td>\n",
       "      <td>184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>961331</td>\n",
       "      <td>961512</td>\n",
       "      <td>DNase-seq:chr1:961331-961512</td>\n",
       "      <td>181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966132</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966132</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966132</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>982100</td>\n",
       "      <td>982340</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>982100</td>\n",
       "      <td>982340</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1</td>\n",
       "      <td>982100</td>\n",
       "      <td>982340</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999917</td>\n",
       "      <td>1000904</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000080</td>\n",
       "      <td>1000280</td>\n",
       "      <td>DNase-seq:chr1:1000080-1000280</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom   start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1   65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1  451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1  686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1  923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1  923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1  924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1  925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1  925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1  925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1  925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1  925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1  925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1  930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1  939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1  940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1  940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1  941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1  941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1  945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1  946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1  958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1  959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1  959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "23  chr1  959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "24  chr1  959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "25  chr1  959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "26  chr1  959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "27  chr1  960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "28  chr1  960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "29  chr1  961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "30  chr1  961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "31  chr1  962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "32  chr1  963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "33  chr1  966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "34  chr1  966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "35  chr1  966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "36  chr1  970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "37  chr1  973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "38  chr1  976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "39  chr1  981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "40  chr1  981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "41  chr1  981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "42  chr1  999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "43  chr1  999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "44  chr1  999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "45  chr1  999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "46  chr1  999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "47  chr1  999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "48  chr1  999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "49  chr1  999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923760    923940   \n",
       "4                        5.0     chr1      923760    923940   \n",
       "5                        5.0     chr1      924809    924915   \n",
       "6                        5.0        .          -1        -1   \n",
       "7                        5.0        .          -1        -1   \n",
       "8                        5.0        .          -1        -1   \n",
       "9                        5.0        .          -1        -1   \n",
       "10                       5.0        .          -1        -1   \n",
       "11                       5.0        .          -1        -1   \n",
       "12                       5.0        .          -1        -1   \n",
       "13                       2.0        .          -1        -1   \n",
       "14                       5.0     chr1      940300    940520   \n",
       "15                       5.0        .          -1        -1   \n",
       "16                       3.0     chr1      941680    941900   \n",
       "17                       2.0        .          -1        -1   \n",
       "18                       5.0        .          -1        -1   \n",
       "19                       2.0        .          -1        -1   \n",
       "20                       5.0        .          -1        -1   \n",
       "21                       1.0     chr1      959000    959844   \n",
       "22                       1.0     chr1      959220    959440   \n",
       "23                       2.0     chr1      959000    959844   \n",
       "24                       2.0     chr1      959220    959440   \n",
       "25                       3.0     chr1      959000    959844   \n",
       "26                       3.0     chr1      959220    959440   \n",
       "27                       1.0     chr1      960480    960660   \n",
       "28                       1.0     chr1      960481    961383   \n",
       "29                       1.0     chr1      960481    961383   \n",
       "30                       1.0     chr1      961331    961512   \n",
       "31                       3.0        .          -1        -1   \n",
       "32                       2.0        .          -1        -1   \n",
       "33                       1.0     chr1      966132    967323   \n",
       "34                       1.0     chr1      966132    967323   \n",
       "35                       2.0     chr1      966132    967323   \n",
       "36                       3.0        .          -1        -1   \n",
       "37                       3.0        .          -1        -1   \n",
       "38                       1.0        .          -1        -1   \n",
       "39                       2.0     chr1      982100    982340   \n",
       "40                       5.0     chr1      982100    982340   \n",
       "41                       NaN     chr1      982100    982340   \n",
       "42                       3.0     chr1      999917   1000904   \n",
       "43                       3.0     chr1     1000080   1000280   \n",
       "44                       2.0     chr1      999917   1000904   \n",
       "45                       2.0     chr1     1000080   1000280   \n",
       "46                       1.0     chr1      999917   1000904   \n",
       "47                       1.0     chr1     1000080   1000280   \n",
       "48                       2.0     chr1      999917   1000904   \n",
       "49                       2.0     chr1     1000080   1000280   \n",
       "\n",
       "                     openChromatin  overlap  \n",
       "0                                .        0  \n",
       "1                                .        0  \n",
       "2                                .        0  \n",
       "3     DNase-seq:chr1:923760-923940      180  \n",
       "4     DNase-seq:chr1:923760-923940      180  \n",
       "5     DNase-seq:chr1:924809-924915       15  \n",
       "6                                .        0  \n",
       "7                                .        0  \n",
       "8                                .        0  \n",
       "9                                .        0  \n",
       "10                               .        0  \n",
       "11                               .        0  \n",
       "12                               .        0  \n",
       "13                               .        0  \n",
       "14    DNase-seq:chr1:940300-940520      220  \n",
       "15                               .        0  \n",
       "16    DNase-seq:chr1:941680-941900       47  \n",
       "17                               .        0  \n",
       "18                               .        0  \n",
       "19                               .        0  \n",
       "20                               .        0  \n",
       "21     ATAC-seq:chr1:959000-959844      500  \n",
       "22    DNase-seq:chr1:959220-959440      220  \n",
       "23     ATAC-seq:chr1:959000-959844      500  \n",
       "24    DNase-seq:chr1:959220-959440      220  \n",
       "25     ATAC-seq:chr1:959000-959844      500  \n",
       "26    DNase-seq:chr1:959220-959440      220  \n",
       "27    DNase-seq:chr1:960480-960660      180  \n",
       "28     ATAC-seq:chr1:960481-961383      353  \n",
       "29     ATAC-seq:chr1:960481-961383      184  \n",
       "30    DNase-seq:chr1:961331-961512      181  \n",
       "31                               .        0  \n",
       "32                               .        0  \n",
       "33     ATAC-seq:chr1:966132-967323      500  \n",
       "34     ATAC-seq:chr1:966132-967323      500  \n",
       "35     ATAC-seq:chr1:966132-967323      500  \n",
       "36                               .        0  \n",
       "37                               .        0  \n",
       "38                               .        0  \n",
       "39    DNase-seq:chr1:982100-982340      240  \n",
       "40    DNase-seq:chr1:982100-982340      240  \n",
       "41    DNase-seq:chr1:982100-982340      240  \n",
       "42    ATAC-seq:chr1:999917-1000904      314  \n",
       "43  DNase-seq:chr1:1000080-1000280      151  \n",
       "44    ATAC-seq:chr1:999917-1000904      429  \n",
       "45  DNase-seq:chr1:1000080-1000280      200  \n",
       "46    ATAC-seq:chr1:999917-1000904      430  \n",
       "47  DNase-seq:chr1:1000080-1000280      200  \n",
       "48    ATAC-seq:chr1:999917-1000904      500  \n",
       "49  DNase-seq:chr1:1000080-1000280      200  "
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "eeaa6afc-3c44-4aa0-9562-fea240eb3ff8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to Type)\n",
    "group_cols = promoter_like_open.columns[:8]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "dccec6f3-8b09-454d-90a9-7dd12899d7e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open.groupby(list(group_cols), dropna=False)['openChromatin'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "4558aa29-a1bb-4237-a26a-ad8dec8070b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                 openChromatin  \n",
       "0                            NaN                             .  \n",
       "1                            NaN                             .  \n",
       "2                            NaN                             .  \n",
       "3                            5.0  DNase-seq:chr1:923760-923940  \n",
       "4                            5.0  DNase-seq:chr1:923760-923940  \n",
       "...                          ...                           ...  \n",
       "170552                       5.0                             .  \n",
       "170553                       1.0                             .  \n",
       "170554                       2.0                             .  \n",
       "170555                       1.0                             .  \n",
       "170556                       2.0                             .  \n",
       "\n",
       "[170557 rows x 9 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "f1a85a30-c934-46c4-acd6-62af217bd97e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:924809-924915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:940300-940520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:941680-941900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383;DNase-seq:chr1:960...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383;DNase-seq:chr1:961...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012922-1014091;DNase-seq:chr1:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019151-1020305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019151-1020305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                        openChromatin  \n",
       "0                                                   .  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3                        DNase-seq:chr1:923760-923940  \n",
       "4                        DNase-seq:chr1:923760-923940  \n",
       "5                        DNase-seq:chr1:924809-924915  \n",
       "6                                                   .  \n",
       "7                                                   .  \n",
       "8                                                   .  \n",
       "9                                                   .  \n",
       "10                                                  .  \n",
       "11                                                  .  \n",
       "12                                                  .  \n",
       "13                                                  .  \n",
       "14                       DNase-seq:chr1:940300-940520  \n",
       "15                                                  .  \n",
       "16                       DNase-seq:chr1:941680-941900  \n",
       "17                                                  .  \n",
       "18                                                  .  \n",
       "19                                                  .  \n",
       "20                                                  .  \n",
       "21  ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...  \n",
       "22  ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...  \n",
       "23  ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...  \n",
       "24  ATAC-seq:chr1:960481-961383;DNase-seq:chr1:960...  \n",
       "25  ATAC-seq:chr1:960481-961383;DNase-seq:chr1:961...  \n",
       "26                                                  .  \n",
       "27                                                  .  \n",
       "28                        ATAC-seq:chr1:966132-967323  \n",
       "29                        ATAC-seq:chr1:966132-967323  \n",
       "30                        ATAC-seq:chr1:966132-967323  \n",
       "31                                                  .  \n",
       "32                                                  .  \n",
       "33                                                  .  \n",
       "34                       DNase-seq:chr1:982100-982340  \n",
       "35                       DNase-seq:chr1:982100-982340  \n",
       "36                       DNase-seq:chr1:982100-982340  \n",
       "37  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...  \n",
       "38  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...  \n",
       "39  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...  \n",
       "40  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...  \n",
       "41  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...  \n",
       "42  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...  \n",
       "43  ATAC-seq:chr1:1012922-1014091;DNase-seq:chr1:1...  \n",
       "44                      ATAC-seq:chr1:1019151-1020305  \n",
       "45                      ATAC-seq:chr1:1019151-1020305  \n",
       "46                                                  .  \n",
       "47                                                  .  \n",
       "48                                                  .  \n",
       "49                                                  .  "
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "70759c9e-2a8a-470b-af36-c276ffb07909",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin.bed_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51e718c2-56ec-4903-8f6b-cf50fdaff699",
   "metadata": {},
   "source": [
    "## promoter table overlap with chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "06ae8b61-4fca-4708-b3c6-da6389dc21aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"ChromHMM_Myoblast\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "b5361981-a1cb-41ec-bdea-c4116e9226a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E120_25_myoblast/hg38_chromHMM_E120_25.txt',\n",
    "                          sep='\\t',\n",
    "                          header=None,\n",
    "                          names=cols\n",
    "                          )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "e87a09c4-0ed6-4cb7-98ea-6112acd69466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_Myoblast</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>96080</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>257849</td>\n",
       "      <td>267849</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267849</td>\n",
       "      <td>268049</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827550</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930136</td>\n",
       "      <td>155930336</td>\n",
       "      <td>21_Het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827551</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930336</td>\n",
       "      <td>156030135</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827552</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156030135</td>\n",
       "      <td>156030735</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>827555 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end ChromHMM_Myoblast\n",
       "0       chr1      10000      96080          25_Quies\n",
       "1       chr1      96276      96476          25_Quies\n",
       "2       chr1      97276     177200          25_Quies\n",
       "3       chr1     257849     267849          25_Quies\n",
       "4       chr1     267849     268049          19_DNase\n",
       "...      ...        ...        ...               ...\n",
       "827550  chrX  155930136  155930336            21_Het\n",
       "827551  chrX  155930336  156030135          25_Quies\n",
       "827552  chrX  156030135  156030735          19_DNase\n",
       "827553  chrY    1086142    1086334          25_Quies\n",
       "827554  chrY    1198191    1198391          25_Quies\n",
       "\n",
       "[827555 rows x 4 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "1fcc068f-3f0d-4f23-8378-fcc0e7ae08a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_chromHMM['ChromHMM_Myoblast'] = MB_chromHMM['Chrom'] + \":\" + MB_chromHMM['start'].astype('str') + \":\" + MB_chromHMM['end'].astype('str') + \":\" + MB_chromHMM['ChromHMM_Myoblast']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "b7e24ff8-e263-44c6-b30e-dce9e991e152",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_Myoblast</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>chr1:96276:96476:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>chr1:97276:177200:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>257849</td>\n",
       "      <td>267849</td>\n",
       "      <td>chr1:257849:267849:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267849</td>\n",
       "      <td>268049</td>\n",
       "      <td>chr1:267849:268049:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827550</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930136</td>\n",
       "      <td>155930336</td>\n",
       "      <td>chrX:155930136:155930336:21_Het</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827551</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155930336</td>\n",
       "      <td>156030135</td>\n",
       "      <td>chrX:155930336:156030135:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827552</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156030135</td>\n",
       "      <td>156030735</td>\n",
       "      <td>chrX:156030135:156030735:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>chrY:1086142:1086334:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>chrY:1198191:1198391:25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>827555 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end                  ChromHMM_Myoblast\n",
       "0       chr1      10000      96080          chr1:10000:96080:25_Quies\n",
       "1       chr1      96276      96476          chr1:96276:96476:25_Quies\n",
       "2       chr1      97276     177200         chr1:97276:177200:25_Quies\n",
       "3       chr1     257849     267849        chr1:257849:267849:25_Quies\n",
       "4       chr1     267849     268049        chr1:267849:268049:19_DNase\n",
       "...      ...        ...        ...                                ...\n",
       "827550  chrX  155930136  155930336    chrX:155930136:155930336:21_Het\n",
       "827551  chrX  155930336  156030135  chrX:155930336:156030135:25_Quies\n",
       "827552  chrX  156030135  156030735  chrX:156030135:156030735:19_DNase\n",
       "827553  chrY    1086142    1086334      chrY:1086142:1086334:25_Quies\n",
       "827554  chrY    1198191    1198391      chrY:1198191:1198391:25_Quies\n",
       "\n",
       "[827555 rows x 4 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MB_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "5efa4563-d041-49bd-95e8-72c56adcd6b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "MB_chromHMM.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E120_25_myoblast/hg38_chromHMM_myoblast.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "89b342b8-c2c6-4e67-bd2c-e4e88eef71d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin.bed_uniq -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E120_25_myoblast/hg38_chromHMM_myoblast.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin_chromHMM.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "c5e71ecf-dc77-4f08-9ec0-3e1c12adadef",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'openChromatin_MB','chr_HMM','start_HMM','end_HMM','chromHMM_MB','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "75f9f5e3-77f8-4174-bc66-67464437ffa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin_chromHMM.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "e1ddb9b9-2d80-4103-9525-b432d43ff309",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chr_HMM</th>\n",
       "      <th>start_HMM</th>\n",
       "      <th>end_HMM</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>348136</td>\n",
       "      <td>501536</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>634820</td>\n",
       "      <td>733820</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923620</td>\n",
       "      <td>923820</td>\n",
       "      <td>chr1:923620:923820:2_PromU</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923820</td>\n",
       "      <td>924620</td>\n",
       "      <td>chr1:923820:924620:22_PromP</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286669</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286670</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286671</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286672</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286673</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>286674 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "286669  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "286670  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "286671  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "286672  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "286673  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level              openChromatin_MB chr_HMM  \\\n",
       "0                            NaN                             .    chr1   \n",
       "1                            NaN                             .    chr1   \n",
       "2                            NaN                             .    chr1   \n",
       "3                            5.0  DNase-seq:chr1:923760-923940    chr1   \n",
       "4                            5.0  DNase-seq:chr1:923760-923940    chr1   \n",
       "...                          ...                           ...     ...   \n",
       "286669                       5.0                             .       .   \n",
       "286670                       1.0                             .       .   \n",
       "286671                       2.0                             .       .   \n",
       "286672                       1.0                             .       .   \n",
       "286673                       2.0                             .       .   \n",
       "\n",
       "        start_HMM  end_HMM                  chromHMM_MB  overlap  \n",
       "0           10000    96080    chr1:10000:96080:25_Quies      500  \n",
       "1          348136   501536  chr1:348136:501536:25_Quies      500  \n",
       "2          634820   733820  chr1:634820:733820:25_Quies      500  \n",
       "3          923620   923820   chr1:923620:923820:2_PromU      147  \n",
       "4          923820   924620  chr1:923820:924620:22_PromP      353  \n",
       "...           ...      ...                          ...      ...  \n",
       "286669         -1       -1                            .        0  \n",
       "286670         -1       -1                            .        0  \n",
       "286671         -1       -1                            .        0  \n",
       "286672         -1       -1                            .        0  \n",
       "286673         -1       -1                            .        0  \n",
       "\n",
       "[286674 rows x 14 columns]"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "3f9725c9-7ae4-4bd2-8bea-71be49de0499",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "07e9feff-7bb2-45f3-90fd-0548cc9c6104",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "286674"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoter_like_openChromatin_chromHMM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "3d639778-928d-473f-b09b-e4e9edbf362c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level', 'openChromatin_MB'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to openChromatin)\n",
    "group_cols = promoter_like_openChromatin_chromHMM.columns[:9]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "1ee55dbf-6dd4-41e9-9e4d-99d7bae31daf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "promoter_like_openChromatin_chromHMM_uniq = promoter_like_openChromatin_chromHMM.groupby(list(group_cols), dropna=False)['chromHMM_MB'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "24ff1d6d-f8cc-4742-b862-0fdb6cc0d39c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level              openChromatin_MB  \\\n",
       "0                            NaN                             .   \n",
       "1                            NaN                             .   \n",
       "2                            NaN                             .   \n",
       "3                            5.0  DNase-seq:chr1:923760-923940   \n",
       "4                            5.0  DNase-seq:chr1:923760-923940   \n",
       "...                          ...                           ...   \n",
       "170552                       5.0                             .   \n",
       "170553                       1.0                             .   \n",
       "170554                       2.0                             .   \n",
       "170555                       1.0                             .   \n",
       "170556                       2.0                             .   \n",
       "\n",
       "                                              chromHMM_MB  \n",
       "0                               chr1:10000:96080:25_Quies  \n",
       "1                             chr1:348136:501536:25_Quies  \n",
       "2                             chr1:634820:733820:25_Quies  \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 10 columns]"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "38e59480-fd89-428d-ac23-fcf67893c731",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:924809-924915</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:927020:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:927020:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:927020:930820:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938820:939620:8_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:940300-940520</td>\n",
       "      <td>chr1:939820:940420:16_EnhW1;chr1:940420:940820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:940820:941620:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>DNase-seq:chr1:941680-941900</td>\n",
       "      <td>chr1:941620:942220:22_PromP;chr1:942220:942420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942220:22_PromP;chr1:942220:942420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945020:945420:11_TxEnh3';chr1:945420:9500...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945420:950020:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:957820:958620:4_PromD2;chr1:958620:959220...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383;DNase-seq:chr1:960...</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961020:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960481-961383;DNase-seq:chr1:961...</td>\n",
       "      <td>chr1:961020:961220:22_PromP;chr1:961220:961820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962020:964420:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962020:964420:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>chr1:966220:967420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>chr1:966220:967420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966132-967323</td>\n",
       "      <td>chr1:966220:967420:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:970220:972420:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972420:975820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:976020:976620:16_EnhW1;chr1:976620:980020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>chr1:980020:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>chr1:980020:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:982100-982340</td>\n",
       "      <td>chr1:980020:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000220:1_TssA;chr1:1000220:10020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000220:1_TssA;chr1:1000220:10020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000220:1_TssA;chr1:1000220:10020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000220:1_TssA;chr1:1000220:10020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1002020:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000220:1002020:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012922-1014091;DNase-seq:chr1:1...</td>\n",
       "      <td>chr1:1013220:1013620:1_TssA;chr1:1013620:10144...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019151-1020305</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019151-1020305</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033820:1034020:23_PromBiv;chr1:1034020:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033820:1034020:23_PromBiv;chr1:1034020:1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1034020:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1044420:1047820:7_Tx3'</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                     openChromatin_MB  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3                        DNase-seq:chr1:923760-923940   \n",
       "4                        DNase-seq:chr1:923760-923940   \n",
       "5                        DNase-seq:chr1:924809-924915   \n",
       "6                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14                       DNase-seq:chr1:940300-940520   \n",
       "15                                                  .   \n",
       "16                       DNase-seq:chr1:941680-941900   \n",
       "17                                                  .   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                                                  .   \n",
       "21  ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:959000-959844;DNase-seq:chr1:959...   \n",
       "24  ATAC-seq:chr1:960481-961383;DNase-seq:chr1:960...   \n",
       "25  ATAC-seq:chr1:960481-961383;DNase-seq:chr1:961...   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28                        ATAC-seq:chr1:966132-967323   \n",
       "29                        ATAC-seq:chr1:966132-967323   \n",
       "30                        ATAC-seq:chr1:966132-967323   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                                                  .   \n",
       "34                       DNase-seq:chr1:982100-982340   \n",
       "35                       DNase-seq:chr1:982100-982340   \n",
       "36                       DNase-seq:chr1:982100-982340   \n",
       "37  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...   \n",
       "41  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...   \n",
       "42  ATAC-seq:chr1:999917-1000904;DNase-seq:chr1:10...   \n",
       "43  ATAC-seq:chr1:1012922-1014091;DNase-seq:chr1:1...   \n",
       "44                      ATAC-seq:chr1:1019151-1020305   \n",
       "45                      ATAC-seq:chr1:1019151-1020305   \n",
       "46                                                  .   \n",
       "47                                                  .   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                          chromHMM_MB  \n",
       "0                           chr1:10000:96080:25_Quies  \n",
       "1                         chr1:348136:501536:25_Quies  \n",
       "2                         chr1:634820:733820:25_Quies  \n",
       "3   chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "4   chr1:923620:923820:2_PromU;chr1:923820:924620:...  \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:927020:...  \n",
       "6                       chr1:925220:927020:23_PromBiv  \n",
       "7                       chr1:925220:927020:23_PromBiv  \n",
       "8                       chr1:925220:927020:23_PromBiv  \n",
       "9                       chr1:925220:927020:23_PromBiv  \n",
       "10                      chr1:925220:927020:23_PromBiv  \n",
       "11                      chr1:925220:927020:23_PromBiv  \n",
       "12                       chr1:927020:930820:24_ReprPC  \n",
       "13                          chr1:938820:939620:8_TxWk  \n",
       "14  chr1:939820:940420:16_EnhW1;chr1:940420:940820...  \n",
       "15                      chr1:940820:941620:23_PromBiv  \n",
       "16  chr1:941620:942220:22_PromP;chr1:942220:942420...  \n",
       "17  chr1:941620:942220:22_PromP;chr1:942220:942420...  \n",
       "18  chr1:945020:945420:11_TxEnh3';chr1:945420:9500...  \n",
       "19                          chr1:945420:950020:7_Tx3'  \n",
       "20  chr1:957820:958620:4_PromD2;chr1:958620:959220...  \n",
       "21  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "22  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "23  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961020:...  \n",
       "25  chr1:961020:961220:22_PromP;chr1:961220:961820...  \n",
       "26                          chr1:962020:964420:7_Tx3'  \n",
       "27                          chr1:962020:964420:7_Tx3'  \n",
       "28                         chr1:966220:967420:2_PromU  \n",
       "29                         chr1:966220:967420:2_PromU  \n",
       "30                         chr1:966220:967420:2_PromU  \n",
       "31                       chr1:970220:972420:24_ReprPC  \n",
       "32                        chr1:972420:975820:25_Quies  \n",
       "33  chr1:976020:976620:16_EnhW1;chr1:976620:980020...  \n",
       "34                       chr1:980020:984220:24_ReprPC  \n",
       "35                       chr1:980020:984220:24_ReprPC  \n",
       "36                       chr1:980020:984220:24_ReprPC  \n",
       "37  chr1:1000020:1000220:1_TssA;chr1:1000220:10020...  \n",
       "38  chr1:1000020:1000220:1_TssA;chr1:1000220:10020...  \n",
       "39  chr1:1000020:1000220:1_TssA;chr1:1000220:10020...  \n",
       "40  chr1:1000020:1000220:1_TssA;chr1:1000220:10020...  \n",
       "41                       chr1:1000220:1002020:2_PromU  \n",
       "42                       chr1:1000220:1002020:2_PromU  \n",
       "43  chr1:1013220:1013620:1_TssA;chr1:1013620:10144...  \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "46  chr1:1033820:1034020:23_PromBiv;chr1:1034020:1...  \n",
       "47  chr1:1033820:1034020:23_PromBiv;chr1:1034020:1...  \n",
       "48  chr1:1034020:1034620:24_ReprPC;chr1:1034620:10...  \n",
       "49                        chr1:1044420:1047820:7_Tx3'  "
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_uniq.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "4687dbce-a028-4937-a1f0-82ef46fe4945",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM_uniq.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin_chromHMM_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ee561cb-c25d-446b-b0d7-01d8e10f2eb8",
   "metadata": {},
   "source": [
    "# add TPM from RNA-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "0c65f7cf-9f47-4829-97e1-64af98af10fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/MB/rsem.merged.gene_tpm.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "f0cfa9a9-5011-4de3-b6e8-00b3ed995049",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78927</th>\n",
       "      <td>ENSG00000310553</td>\n",
       "      <td>ENST00000850732</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78928</th>\n",
       "      <td>ENSG00000310554</td>\n",
       "      <td>ENST00000850733</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78929</th>\n",
       "      <td>ENSG00000310555</td>\n",
       "      <td>ENST00000850734</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78930</th>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>ENST00000850735,ENST00000850736</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78931</th>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>ENST00000850818,ENST00000850819,ENST0000085082...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78932 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)   MB\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...  0.0\n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971  0.0\n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...  0.0\n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...  0.0\n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...  0.0\n",
       "...                ...                                                ...  ...\n",
       "78927  ENSG00000310553                                    ENST00000850732  0.0\n",
       "78928  ENSG00000310554                                    ENST00000850733  0.0\n",
       "78929  ENSG00000310555                                    ENST00000850734  0.0\n",
       "78930  ENSG00000310556                    ENST00000850735,ENST00000850736  0.0\n",
       "78931  ENSG00000310557  ENST00000850818,ENST00000850819,ENST0000085082...  0.0\n",
       "\n",
       "[78932 rows x 3 columns]"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "3778d24e-dfc7-47cb-8c4c-0784fd6a8b21",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map =pd.read_csv('./geneid2name.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "dcd4de1d-3254-4d36-937f-a967a206655c",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm.merge(\n",
    "    gene_map,\n",
    "    on=\"gene_id\",\n",
    "    how=\"left\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "52822e1a-ffea-442f-b133-2aa0ff8ee815",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>MB</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78927</th>\n",
       "      <td>ENSG00000310553</td>\n",
       "      <td>ENST00000850732</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78928</th>\n",
       "      <td>ENSG00000310554</td>\n",
       "      <td>ENST00000850733</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78929</th>\n",
       "      <td>ENSG00000310555</td>\n",
       "      <td>ENST00000850734</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78930</th>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>ENST00000850735,ENST00000850736</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78931</th>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>ENST00000850818,ENST00000850819,ENST0000085082...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78932 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)  \\\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...   \n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971   \n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...   \n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...   \n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...   \n",
       "...                ...                                                ...   \n",
       "78927  ENSG00000310553                                    ENST00000850732   \n",
       "78928  ENSG00000310554                                    ENST00000850733   \n",
       "78929  ENSG00000310555                                    ENST00000850734   \n",
       "78930  ENSG00000310556                    ENST00000850735,ENST00000850736   \n",
       "78931  ENSG00000310557  ENST00000850818,ENST00000850819,ENST0000085082...   \n",
       "\n",
       "        MB gene_name  \n",
       "0      0.0    TSPAN6  \n",
       "1      0.0      TNMD  \n",
       "2      0.0      DPM1  \n",
       "3      0.0     SCYL3  \n",
       "4      0.0     FIRRM  \n",
       "...    ...       ...  \n",
       "78927  0.0       NaN  \n",
       "78928  0.0       NaN  \n",
       "78929  0.0       NaN  \n",
       "78930  0.0       NaN  \n",
       "78931  0.0       NaN  \n",
       "\n",
       "[78932 rows x 4 columns]"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "8300f3e9-483f-41f0-bf94-d3888bde03da",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[gene_tpm[\"gene_name\"].notna()].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "dc7e948c-5a1c-48bb-8e07-0698a19fdfdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm.rename(columns={'MB':\"GeneTPM_MB\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "9d028528-2d13-47dd-bf9b-a18eba83b5dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[['GeneTPM_MB','gene_name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "f8e45637-5a5a-45ca-850c-77ef200599c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM = promoter_like_openChromatin_chromHMM_uniq.merge(\n",
    "    gene_tpm,\n",
    "    left_on=\"Gene\",\n",
    "    right_on=\"gene_name\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['gene_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "1b07062e-7a11-49da-9f65-f7b792aae635",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "      <th>GeneTPM_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level              openChromatin_MB  \\\n",
       "0                            NaN                             .   \n",
       "1                            NaN                             .   \n",
       "2                            NaN                             .   \n",
       "3                            5.0  DNase-seq:chr1:923760-923940   \n",
       "4                            5.0  DNase-seq:chr1:923760-923940   \n",
       "...                          ...                           ...   \n",
       "170934                       2.0                             .   \n",
       "170935                       1.0                             .   \n",
       "170936                       1.0                             .   \n",
       "170937                       2.0                             .   \n",
       "170938                       2.0                             .   \n",
       "\n",
       "                                              chromHMM_MB  GeneTPM_MB  \n",
       "0                               chr1:10000:96080:25_Quies         0.0  \n",
       "1                             chr1:348136:501536:25_Quies         0.0  \n",
       "2                             chr1:634820:733820:25_Quies         0.0  \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0  \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0  \n",
       "...                                                   ...         ...  \n",
       "170934                                                  .         0.0  \n",
       "170935                                                  .         0.0  \n",
       "170936                                                  .         0.0  \n",
       "170937                                                  .         0.0  \n",
       "170938                                                  .         0.0  \n",
       "\n",
       "[170939 rows x 11 columns]"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "98c4cb0e-7f26-4343-a361-3df00cad481d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneTPM_MB\n",
       "False    170939\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM['GeneTPM_MB'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "14a19835-bc9e-495c-aac0-ca3eca8e8552",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/MB/rsem.merged.transcript_tpm.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "7f6f0a62-2043-46e6-9aa7-89451119a2d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000373020</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000494424</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000496771</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000612152</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000373031</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387939</th>\n",
       "      <td>ENST00000850736</td>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387940</th>\n",
       "      <td>ENST00000850818</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387941</th>\n",
       "      <td>ENST00000850819</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387942</th>\n",
       "      <td>ENST00000850820</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387943</th>\n",
       "      <td>ENST00000850821</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>387944 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          transcript_id          gene_id   MB\n",
       "0       ENST00000373020  ENSG00000000003  0.0\n",
       "1       ENST00000494424  ENSG00000000003  0.0\n",
       "2       ENST00000496771  ENSG00000000003  0.0\n",
       "3       ENST00000612152  ENSG00000000003  0.0\n",
       "4       ENST00000373031  ENSG00000000005  0.0\n",
       "...                 ...              ...  ...\n",
       "387939  ENST00000850736  ENSG00000310556  0.0\n",
       "387940  ENST00000850818  ENSG00000310557  0.0\n",
       "387941  ENST00000850819  ENSG00000310557  0.0\n",
       "387942  ENST00000850820  ENSG00000310557  0.0\n",
       "387943  ENST00000850821  ENSG00000310557  0.0\n",
       "\n",
       "[387944 rows x 3 columns]"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tx_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "1adc7af1-e1f3-437d-9587-5279852c6d21",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = tx_tpm.rename(columns={'MB':\"TranscriptTPM_MB\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "3ae33442-5c03-47ae-bd4a-f838748254d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM['Transcript'] = promoter_like_openChromatin_chromHMM_GeneTPM['Transcript'].str.split(\".\").str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "b91d5f85-39f0-4c95-8e4a-eb9563d96e02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "      <th>GeneTPM_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level              openChromatin_MB  \\\n",
       "0                            NaN                             .   \n",
       "1                            NaN                             .   \n",
       "2                            NaN                             .   \n",
       "3                            5.0  DNase-seq:chr1:923760-923940   \n",
       "4                            5.0  DNase-seq:chr1:923760-923940   \n",
       "...                          ...                           ...   \n",
       "170934                       2.0                             .   \n",
       "170935                       1.0                             .   \n",
       "170936                       1.0                             .   \n",
       "170937                       2.0                             .   \n",
       "170938                       2.0                             .   \n",
       "\n",
       "                                              chromHMM_MB  GeneTPM_MB  \n",
       "0                               chr1:10000:96080:25_Quies         0.0  \n",
       "1                             chr1:348136:501536:25_Quies         0.0  \n",
       "2                             chr1:634820:733820:25_Quies         0.0  \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0  \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0  \n",
       "...                                                   ...         ...  \n",
       "170934                                                  .         0.0  \n",
       "170935                                                  .         0.0  \n",
       "170936                                                  .         0.0  \n",
       "170937                                                  .         0.0  \n",
       "170938                                                  .         0.0  \n",
       "\n",
       "[170939 rows x 11 columns]"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "ef08b377-167e-41e4-b6fa-4947a0c95e2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM = promoter_like_openChromatin_chromHMM_GeneTPM.merge(\n",
    "    tx_tpm,\n",
    "    left_on=\"Transcript\",\n",
    "    right_on=\"transcript_id\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['transcript_id','gene_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "107b3eae-9f76-451e-ad3b-77defdeaed39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "      <th>GeneTPM_MB</th>\n",
       "      <th>TranscriptTPM_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level              openChromatin_MB  \\\n",
       "0                            NaN                             .   \n",
       "1                            NaN                             .   \n",
       "2                            NaN                             .   \n",
       "3                            5.0  DNase-seq:chr1:923760-923940   \n",
       "4                            5.0  DNase-seq:chr1:923760-923940   \n",
       "...                          ...                           ...   \n",
       "170934                       2.0                             .   \n",
       "170935                       1.0                             .   \n",
       "170936                       1.0                             .   \n",
       "170937                       2.0                             .   \n",
       "170938                       2.0                             .   \n",
       "\n",
       "                                              chromHMM_MB  GeneTPM_MB  \\\n",
       "0                               chr1:10000:96080:25_Quies         0.0   \n",
       "1                             chr1:348136:501536:25_Quies         0.0   \n",
       "2                             chr1:634820:733820:25_Quies         0.0   \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0   \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0   \n",
       "...                                                   ...         ...   \n",
       "170934                                                  .         0.0   \n",
       "170935                                                  .         0.0   \n",
       "170936                                                  .         0.0   \n",
       "170937                                                  .         0.0   \n",
       "170938                                                  .         0.0   \n",
       "\n",
       "        TranscriptTPM_MB  \n",
       "0                    0.0  \n",
       "1                    0.0  \n",
       "2                    0.0  \n",
       "3                    0.0  \n",
       "4                    0.0  \n",
       "...                  ...  \n",
       "170934               0.0  \n",
       "170935               0.0  \n",
       "170936               0.0  \n",
       "170937               0.0  \n",
       "170938               0.0  \n",
       "\n",
       "[170939 rows x 12 columns]"
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "aa63ca69-6cb4-45af-856b-4bd639b25fda",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM[\"transcript_support_level\"] = (\n",
    "    promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM[\"transcript_support_level\"]\n",
    "        .fillna(0)\n",
    "        .astype(int)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "1554d93f-9e19-4bca-bce6-efe6eb8c3f1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "      <th>GeneTPM_MB</th>\n",
       "      <th>TranscriptTPM_MB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level              openChromatin_MB  \\\n",
       "0                              0                             .   \n",
       "1                              0                             .   \n",
       "2                              0                             .   \n",
       "3                              5  DNase-seq:chr1:923760-923940   \n",
       "4                              5  DNase-seq:chr1:923760-923940   \n",
       "...                          ...                           ...   \n",
       "170934                         2                             .   \n",
       "170935                         1                             .   \n",
       "170936                         1                             .   \n",
       "170937                         2                             .   \n",
       "170938                         2                             .   \n",
       "\n",
       "                                              chromHMM_MB  GeneTPM_MB  \\\n",
       "0                               chr1:10000:96080:25_Quies         0.0   \n",
       "1                             chr1:348136:501536:25_Quies         0.0   \n",
       "2                             chr1:634820:733820:25_Quies         0.0   \n",
       "3       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0   \n",
       "4       chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0   \n",
       "...                                                   ...         ...   \n",
       "170934                                                  .         0.0   \n",
       "170935                                                  .         0.0   \n",
       "170936                                                  .         0.0   \n",
       "170937                                                  .         0.0   \n",
       "170938                                                  .         0.0   \n",
       "\n",
       "        TranscriptTPM_MB  \n",
       "0                    0.0  \n",
       "1                    0.0  \n",
       "2                    0.0  \n",
       "3                    0.0  \n",
       "4                    0.0  \n",
       "...                  ...  \n",
       "170934               0.0  \n",
       "170935               0.0  \n",
       "170936               0.0  \n",
       "170937               0.0  \n",
       "170938               0.0  \n",
       "\n",
       "[170939 rows x 12 columns]"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "772c46be-a953-43f9-b0fe-46e7949816cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TranscriptTPM_MB\n",
       "False    170939\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM['TranscriptTPM_MB'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f80d806-da69-4112-a3c3-48b3194fa03e",
   "metadata": {},
   "source": [
    "# update the promoter evidence Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "85bf42df-3f7e-4f35-9387-a1f4fd0da2f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "def classify_promoter_type(row):\n",
    "    score = 0\n",
    "    sources = []  # track where score comes from\n",
    "\n",
    "    # Open chromatin evidence (Essential)\n",
    "    open_chrom = row['openChromatin_MB']\n",
    "    if isinstance(open_chrom, str) and open_chrom != \".\":\n",
    "        if 'ATAC' in open_chrom and 'DNase' in open_chrom:\n",
    "            score += 3\n",
    "            sources.append(\"OpenChromatin:ATAC+DNase(+3)\")\n",
    "        else:\n",
    "            score += 2\n",
    "            sources.append(\"OpenChromatin:ATAC/DNase(+2)\")\n",
    "\n",
    "    # ChromHMM evidence (Essential)\n",
    "    chromhmm = row['chromHMM_MB']\n",
    "    if isinstance(chromhmm, str) and chromhmm != \".\":\n",
    "        active_states = ['TssA', 'PromU', 'PromP', '1_Tss', '2_Prom']\n",
    "        if any(state in chromhmm for state in active_states):\n",
    "            score += 2\n",
    "            sources.append(\"ChromHMM:ActivePromoter(+2)\")\n",
    "\n",
    "    # Expression evidence\n",
    "    transcript_tpm = row.get('TranscriptTPM_MB', 0)\n",
    "    gene_tpm = row.get('GeneTPM_MB', 0)\n",
    "\n",
    "    if transcript_tpm > 1:\n",
    "        score += 2\n",
    "        sources.append(\"TranscriptTPM>1(+2)\")\n",
    "    elif transcript_tpm > 0:\n",
    "        score += 1\n",
    "        sources.append(\"TranscriptTPM>0(+1)\")\n",
    "    elif gene_tpm > 1:\n",
    "        score += 1\n",
    "        sources.append(\"GeneTPM>1(+1)\")\n",
    "\n",
    "    # Transcript support level\n",
    "    tsl = str(row['transcript_support_level'])\n",
    "    if tsl in ['1', '2']:\n",
    "        score += 1\n",
    "        sources.append(\"TSL1-2(+1)\")\n",
    "\n",
    "    # Optional Hi-C\n",
    "    # hic = row.get('HiC_hMSC')\n",
    "    # if isinstance(hic, str) and hic != '.':\n",
    "    #     score += 1\n",
    "    #     sources.append(\"HiC(+1)\")\n",
    "\n",
    "    # Final classification\n",
    "    if score >= 6:\n",
    "        promoter_type = 'very_high'\n",
    "    elif score >= 4:\n",
    "        promoter_type = 'high'\n",
    "    elif score >= 2:\n",
    "        promoter_type = 'moderate'\n",
    "    else:\n",
    "        promoter_type = 'low'\n",
    "\n",
    "    source_str = \";\".join(sources) if sources else \"NA\"\n",
    "    return promoter_type, score, source_str\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "70b02bca-4278-481d-83dd-8a830487b7ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM[[\"Type\", \"score\", \"score_source\"]] = (\n",
    "    promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM\n",
    "        .apply(classify_promoter_type, axis=1, result_type=\"expand\")\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "b2923628-e6f9-4986-8142-4a6fcb4fb7e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MB</th>\n",
       "      <th>chromHMM_MB</th>\n",
       "      <th>GeneTPM_MB</th>\n",
       "      <th>TranscriptTPM_MB</th>\n",
       "      <th>Type</th>\n",
       "      <th>score</th>\n",
       "      <th>score_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10000:96080:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>high</td>\n",
       "      <td>4</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2);ChromHMM:ActivePr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>DNase-seq:chr1:923760-923940</td>\n",
       "      <td>chr1:923620:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>high</td>\n",
       "      <td>4</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2);ChromHMM:ActivePr...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chrom   start     end       Transcript     TSS    Gene strand  \\\n",
       "0  chr1   65169   65669  ENST00000641515   65419   OR4F5      +   \n",
       "1  chr1  451428  451928  ENST00000426406  451678  OR4F29      -   \n",
       "2  chr1  686404  686904  ENST00000332831  686654  OR4F16      -   \n",
       "3  chr1  923673  924173  ENST00000616016  923923  SAMD11      +   \n",
       "4  chr1  923673  924173  ENST00000618323  923923  SAMD11      +   \n",
       "\n",
       "   transcript_support_level              openChromatin_MB  \\\n",
       "0                         0                             .   \n",
       "1                         0                             .   \n",
       "2                         0                             .   \n",
       "3                         5  DNase-seq:chr1:923760-923940   \n",
       "4                         5  DNase-seq:chr1:923760-923940   \n",
       "\n",
       "                                         chromHMM_MB  GeneTPM_MB  \\\n",
       "0                          chr1:10000:96080:25_Quies         0.0   \n",
       "1                        chr1:348136:501536:25_Quies         0.0   \n",
       "2                        chr1:634820:733820:25_Quies         0.0   \n",
       "3  chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0   \n",
       "4  chr1:923620:923820:2_PromU;chr1:923820:924620:...         0.0   \n",
       "\n",
       "   TranscriptTPM_MB  Type  score  \\\n",
       "0               0.0   low      0   \n",
       "1               0.0   low      0   \n",
       "2               0.0   low      0   \n",
       "3               0.0  high      4   \n",
       "4               0.0  high      4   \n",
       "\n",
       "                                        score_source  \n",
       "0                                                 NA  \n",
       "1                                                 NA  \n",
       "2                                                 NA  \n",
       "3  OpenChromatin:ATAC/DNase(+2);ChromHMM:ActivePr...  \n",
       "4  OpenChromatin:ATAC/DNase(+2);ChromHMM:ActivePr...  "
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "id": "1f6223fb-d6ca-4185-bee6-831ea759c33b",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin_chromHMM_Expression_level.tsv',sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "cbb225f2-fa98-49ac-b82e-d58f7874eaf6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Type\n",
      "low          38.382698\n",
      "high         29.395866\n",
      "very_high    17.315534\n",
      "moderate     14.905902\n",
      "Name: proportion, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "type_percentages = promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM['Type'].value_counts(normalize=True) * 100\n",
    "print(type_percentages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "fa24f6c9-e9e9-4e1f-854b-e921276e2d5c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAtIAAAIoCAYAAABEYwzVAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjgsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvwVt1zgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAX0RJREFUeJzt3Qd01FXe//EvIL2KICwoSBEURIorxQIIsriUR5AHF6SJlKWIuIKCUlYQKUpHiiC6SBFFhRUElZWqLO7zACLSlCJNpEnv7X8+939+88xMJiH8EpJJ8n6dkzPJzJ2SmcnkM9/53nvTXb169aoBAAAAuC7pr284AAAAAII0AAAA4BMVaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gASxaeffmqlS5cOfJUrV84efPBBa9Wqlb399tt25MiRGOcZN26cG3s9zp4968733XffXdf5Il1XrVq17K9//aslpvnz59s//vGPiKfp+nU7otm///1ve+KJJ6xChQru9v7rX/+KMUaPafBjHdtXNP6u//u//2vdu3e3hx9+2O655x677777rFmzZjZr1iw7c+ZMoj7e0Uh/N3psrvfvB0BkN8VyPAD4MmTIECtevLhdunTJhec1a9bYlClT7N1337VRo0bZAw88EBjbtGlTF2iuN0i/9dZb9uyzz1qVKlXifT4/1+XHggUL7Oeff7ann346xmkffvihFSxY0KLV1atX7fnnn7c77rjDJk6caFmzZrVixYrFGPf3v//dTp06Ffh52bJlbrz32Hui7XcdO3asjR8/3ipWrOjCdJEiRdzzad26de459csvv9grr7ySaI93NCpbtqx7HpYsWTK5bwqQKhCkASSqO++801WjPXXr1nUh46mnnnLh96uvvrJ8+fIFgtaNDlsKSgqESXFd16IqbzQ7ePCgHTt2zB599FGrVq1arOPCQ9iOHTsiPvbRZNGiRS5E//d//7cNGjTI0qVLFzitRo0a1r59e/v+++8ttbp48aL7nXPkyBH1z0MgJaG1A8ANV6hQIevVq5edPn3aZs+eHWe7hVoL1DqgavO9995rNWvWtG7durlAvHfv3kDAUwXRayHo3bt3yOVt3LjRnnvuObv//vutTp06sV6XZ/HixdawYUMXAmvXrm3vv/9+xLYVXX9cH5Prdqs6u2/fvpAWB0+kdoeffvrJOnfu7G6rrv/xxx+3uXPnRrweVT9V1X/ooYesUqVK7g2KF2Lj09LQpk0bV40tX768a2fQbQ1+LKpXr+6+Hz58uLs+tb74MW/ePHd+VXrD6XFTVfTAgQOB+6xBgwbu9j355JPuMdcnB6NHj7bLly+HnPfChQs2YcIEe+yxx1xbRtWqVe3ll1+233///Zq3SefLnTu39e3bNyREexQwdb96Zs6caS1atHDPNwVPPT/0yYoCqedaj3d8b6/GDR061LVC6bHR9f7444/u/vee236eL3ocdLm6PzV2165dsbZ2bNiwwTp16mSVK1d2Yxs1amQLFy4MGaO/wWHDhrnbpTEaqzYgPS+BtIqKNIAkoapfhgwZXGCKjYKqepb/+Mc/2uuvv265cuVygWvlypUuwNx66632zjvvuOqhKotq15C8efOGXI6Cd7169VxYvFbf6+bNm23w4MGuWq5KuXpedd26vnbt2l3X76iWh379+tmePXtcYLwWhWDdxltuucX69OljN998s3322WcuPB0+fNg6dOgQMn7kyJEuQOv2qbVCgVehSoFH921s/vOf/9gzzzxjpUqVcufNlCmTffDBBy446TJ1X+m+vOuuu9z94IVbjfNDl6fbpjCq4O5Ru4/aClTxLlCgQOD4Q4cO2d/+9jfr2LGjewPktYqcOHHC+vfv78ZcuXLFunTp4lqF9LjoflCA1RuAH374wT755BPLkiVLrJV2BVDdLn06ER+7d+9298Ftt91mGTNmtC1bttikSZPcY6YWlms93tdzexWu9Rjqea2wvW3bNvc4BLfP+H2+6E3AgAEDLH369O58Ghdu9erV7roV4l999VXLmTOnuz16TM6dO+fCsuj31vWp/efuu+92wVr3qz7FANIqgjSAJJEtWzb3j1+hJjaqJJ8/f95eeuklF+o8qgZ6VM0UtWnE9hG1qmkKZPGh26PKnXd9CvyqGKqSqHaU+AYvr+VB4V8BND4fnyt8KbCrAv6HP/whcP0KkGpDUGhSqAm+fAVUj8KRQo2qiXFd34gRI9ztmj59umXPnt0d98gjj7j7SRXGP//5z+7+VNAV3ZaEfPyv3/8vf/mLm2SqkKgAJ2rr0f3dsmXLkPEKYrq/9WmAqDKs54HCvgKePtFQa4beUCmI/ulPfwqcV4+b3lTpUwM9XpHs37/fHSoUx5dud3Ao1pu7PHnyuB5qBVdVt+N6vON7exWaVdFVCO7Ro4cbo8q03tS98MILCXq+qAdcfeHXoqCttpxp06bZTTf9/1igKvbRo0ddGNfzRM81fcKg2xbcD65PjIC0jNYOAEk6mS0uqnKp+qcqnz6uVqXPj+Dgci0KEMGhXVSJVDVQwf5GUiVQrQNeKPI0btw4MAkuWHirhddG8Ouvv8Z6HarIr1+/3vWqeyFaVMH+r//6L/vtt9/i3R5yPZo3b+4OP/roo8BxqlCrKq62hGC6XV6IDn4MFGD/53/+x/28dOlSF1r1BkCB3/vScyZ//vyu6p6YNm3a5Cr2ajHSdegNnNqT1G6iSYnXEt/b6x3qzUwwPV5eqPX7fInP34HaPfT4e29Wg2+rWn30acHOnTvdaWrnWLFihXszp9YQVauBtI6KNIAkoUCnyqOCVGxUQdNSYmrfGDhwoDvP7bff7loN1N8bX2oBiS9v4mOk4270R9a6fIWq2G5/+PWrIhrMa72IK9CoWqk3MNdzPYlB96FaKdTKoZYNrWyhth49rpHGxnacd9u0Aox+F/UaR6LqaWy84Bne4x4bvTFRn7JWLFEFunDhwpY5c2bXkqHbH58AGd/b6/1+4feBQnT44329z5dIY8N5rR76ZEJfcd1W9Zfrkwu1fahfXPeJPj3QJ0ha6QVIiwjSAJKE+l5VzdMEpbjoI3R9aawmXKkdQT3MChr169dP9NsVqWfUO84LMgoM3qSw+Ia3+NDlq+IXzmt/UStMQqkqqo/lb/T1RNK6dWv75z//aV9//bVrc9BtCW7TuZ7HQLdR3+tNViTB1fZIQVNv4L799tvAKi5x0drZehOntgyFaI/6pOMrvrfX+/30+wb3jasiHOmN1PU8jpEmVUa6naK5Cd7E3HDeEohqz1LLlL50e1WdVtuQKvdffPHFNa8LSI1o7QBww6nC98Ybb7j+TfVxxodaDzT5SRO6xGuziE8V9nqoUhoekNSzqqDj9WN7YWrr1q0h45YsWRLj8nT74nvb9DG9Pq73VrDwKHwq7CXGMmUKP7oftTJJ8O1S24QmjqnCGGmt6MSgaqwmG6p6qUmcakHQ7Qmn1VwUtsMfA70B8NpA1IurYKnbrRaD8K/g9asj0cS/48ePu6XvIrUY6TZ88803IQE0eLKlzhPcpnKtxzu+t9f7/cJXyPjyyy8DPes38vmi26Fqsv4GIt1OfWlFk3B6Y6tJiHpzq9YPvUEB0iIq0gASlYKpqskKAZq0p4/zNbFKwViTpcJX2AimyWUKCgoh+jheE860uoF4G7non7qCrYKXgoUmfamqdj0TycKrlVr5Qqsk6KNwhUtVLnv27BmoXCpMKGzqzYB+N1VWVbXUigzhVPnUpDrtlKcgqVAW29rKXbt2db20qtzqe/0uCpyq3r/44oshE8cSQpPWtGqHrkeH6kPX7dNjpclk8alc+qXr1OoPuo7YJgOq0qrVIjQpUKFu+fLlLrSqz1oTDUWBTfeN2kTU6qNl8vR7qMdb/brqsY6tour1IGuFCU1qVE+wJvx5G7KoZUPLMqoVRa0Keq7psnW/abKjPonQc1OtGvF9vON7e9Wjr37w9957z/2NaNUOPS76WY9/8GNzo54vmmyoyY5aXURvdlQZ15uO7du3uzew3oRFreyiv0315uu6dbpCvN4sXc+kXCA1IUgDSFTeagcKDQqcJUqUcP+k9U84rhAtmoilEKuP1PURtqqXCipaCi14jV8t4aZQqwCskKN//lov1w9dpypruk5NIlOw1u8QvDKBAo6WPnvttddchVxVSAUlLc2moBRMIUdBSOs9nzx50lUywyvZwdVABTiFWa/3VveXlhnzlhxLDGqnUe+5fkf9bqqSaoKl7ldNhruRtNSd7i9N2outj1ZvYHRfqkdXYVchTe0CWsYw+DHQ7dWKFQpvkydPdsepoq6qbly99x7tZqiQrHYhPT6qGKttR2G2bdu2bqUR0WOg+0prWes2KOgr7Oo5Eb7EXGyP9/XcXj3eug8+/vhj9zjpOanrVojX39CNfr4ovM+ZM8c9x9VGpTcM+p112cGTIDVOn8JodQ+9AVHg1ooeeqyAtCrd1WtNowcAwCcFL73hUZDUUm3hVK1VrzmbeoRau3atq8hrhYxIfeUAogMVaQBAotP6yNqARFVmVVi9XRMRkz6F0dJ1ag1RhVwVbb3xUAX/epZyBJD0CNIAgESnvltVVcuUKePabm5kH3ZKp75/hWm1gWjSo3r+9cZDPdreijEAohOtHQAAAIAPLH8HAAAA+ECQBgAAAHwgSAMAAAA+MNkwiWlmtlYc1Bq7AAAAiD4XL150k6S14VBcCNJJTCGapbsBAACiV3yzGkE6iXmV6Ni2DAYAAEDy2rBhQ7zG0SMNAAAA+ECQBgAAAAjSAAAAQNKgIg0AAAD4QJAGAAAAfCBIAwAAACkxSLdq1cpKly4d8evzzz8PjFu+fLk1atTILRtXp04dmzlzZsTLmzp1qtWqVcuNa9KkiX333Xcxxpw6dcr69+9vVapUcQttd+rUyfbt2xdj3M6dO61du3ZWoUIFq1atmg0aNMjOnTuXyPcAAAAAUqJkX0f673//uwu2waZNm2ZfffWVC6/eboBdunSxxx9/3Hr37m1r1651oTZTpkzWtGnTkBA9atQo+9vf/mZlypSxOXPmWIcOHdyhgrmnR48etnHjRuvXr5/lyJHDxo4da23btrXPPvvMsmTJ4sacOHHC2rRpY4UKFXKn//777zZkyBA7duyYDR8+PMnuHwAAAESnZA/SJUuWjHGcgu6DDz5oefPmdT+PHz/eBePBgwe7n6tWrWr79++3MWPGuKpz+vTp7cKFCzZx4kRr3bq1qyJL5cqVrWHDhjZp0iQXsGX9+vW2bNkymzx5stWoUcMdV6pUKVflnjt3rjVv3twdN3v2bBem582bF7gdGTJksJ49e1rnzp2tRIkSSXQPAQAAIBole2tHOFWb9+7d6wKwKCCvXr3a6tevHzJOpx86dMg2bdoUON/JkyetQYMGgTEKvvXq1XNtId5Wj/o+V65cVr169cA4VZ0rVarkTvOsWLHCVcS9EC1169Z1VfDgcQAAAEiboi5IL1iwwLJmzWq1a9d2P+/evdsuXrxoxYsXj1jJ3r59e8hh+DhVjk+fPm0HDhwIjCtWrJilS5cuxuV5l+GNC686K0QXKVIkZBwAAADSpmRv7Qh26dIl++KLL1yIzpYtmzvu+PHj7lBV5GDez97pasNQ0PV6nD25c+d2h+ptLliwoBuXM2fOGNety/Muy7u88OuMNM4PVcfPnDmToMsAAADAjaGsFl50jfog/e2339qRI0dC2jM8sf0ywcdHGuO1dFxrXFzHh19efMbFRRX2zZs3J+gyAAAAcOOoQJuigrTaOvLkyWMPPfRQjIpyeBVYFWPxqsY6PH/+vPvKnDlzjHHe5WicJiqGC69A63vvvMHUh53QiYYZM2aMOMkSAAAAyW/btm3xGhc1QVrrM3/99dduEqGCpkc9yfp5x44dIRMEvV/QC7XeofqXtcKHRz9nz57dChQoEBi3atWqGJVlXV5wQNb34b3Qmvionm2tFJIQul6vdQUAAADRJb7dB1Ez2XDJkiVuUqC3WkdwWV3L3S1atChG9Tp//vyB0KxVN9T7vHDhwsCYy5cvu/NpmTvvDtH3qjSvXLkyME4Vaq364S2HJwrtWi3k6NGjgeMWL17swnTwOAAAAKRNUVORnj9/vluG7r777otxWteuXa1ly5bWt29fF7QVerXJysCBA90a0l7g1vrOWi9aS9Z5G7Ls2bPHRo4cGbis8uXLW82aNa1Pnz5ucxdtyKL1qAsXLmyNGzcOjGvWrJnNmDHDbQSjL/VuDx061F0/a0gDAAAg3VVvNl4yUv+zNmDRToIvvvhixDFau1mBWO0WWn1DOxG2aNEiZIx+Fe1uqO3DDx8+7DZa0eWpoh1MOykOGzbMvvzySzfxT1uFa5dDhenwLcK1g+KaNWvcaiCaBKkNWcJXBrkeGzZscIfawhwAAADRJ755LSqCdFpCkAYAAEgdeS1qeqQBAACAlIQgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAIBkph2Xtfmc9r645557rHbt2jZkyBA7efJkYMylS5ds8uTJ9thjj7kN5mrVquX2u9COzddD5yldurTb2C6Y9uBo37692y26efPmtmvXrpDTjx07ZtWqVbONGzcm8LdNPQjSAAAAUbA5XcWKFe21115zm8tp47l58+ZZ9+7dA2PGjx/vdmNu1KiRvf3229auXTs35qWXXor39WzdutU++eQTt7NzuMGDB9vly5dt7Nixbsdo7QAdbPTo0fboo49a2bJlE/jbph5Rs0U4AABAWqXdk/Xl0a7LCrPaefnAgQNWoEABW7BggRvTqVMnN0bV6zNnzridn3WYLVu2a16PgroX0sN9++23NmXKFLv33nstZ86c9uSTTwYud8uWLbZo0SL3hf9DRRoAACAK5cmTJ9DS4R0q4AbLlSuXaZPq+GxU/dlnn9nevXutQ4cOEU+/ePGiZcmSxX3vHeo4ef31161r166WN2/eBP5WqQtBGgAAIEqoteL8+fOuD1mtHI888ogVLlzYnfaXv/zF/vnPf9qqVavs9OnTbhvrd9991xo3bmzZs2eP83JPnTplb7zxhmsDyZo1a8Qx2g571qxZrs1k5syZVqRIEcudO7d9/vnndvToUXvqqaduyO+cktHaAQAAECUUnNXKIQ8//LBr2/CopUNV6WeeeSZQgf7Tn/4UY9JgJG+99ZYVLVrU6tWrF+uYXr16WceOHe2DDz5wle9x48bZ2bNn7c0333QTH2+6idgYjnsEAAAgSmhVDvUlb9u2zSZMmODC83vvvWcZMmSwGTNm2D/+8Q83CVAT/nbu3OkmH/bt29eGDRsW62XqslRh/uijj+K87jJlytjSpUttz549rgqeOXNmGzVqlOuZ1modOm3o0KFudQ8F/gEDBlyzEp7aEaQBAACixF133eUOtQSdgm2TJk1s8eLFbvKhwvKLL75orVu3dmPuv/9+17Os3mUdF9tqGqoma8k8hWNvqbwrV664/mf9rBU80qf//92+GTNmtOLFi7vvFahVnZ47d64dOXLEXnjhBVf9fvDBB12f9cSJE61nz56WltEjDQAAEIXuvvtuV4nevXu3C7UXLlxwx4WPEY2JjSrXmmio4O197d+/31Wo9b1Ojy2At2rVygXw77//3q0i0rBhQxfe1Ze9atUqS+uoSAMAAEShdevWucmHt912mxUqVMgdp0mICr+eH3/80R16ExIjUZ+1JjAGU3W5QoUKrpLtXXb4UnibN28O6dFWBVu3R+Fe7ScgSAMAACS7Z5991u1oqB0HtfSc1m1+55133M/aBEXV4Lp167qeaE041NgdO3a4CYHayEU/e+rUqePC8bRp09zPCszh1P+stanVMhJOl6/l7jT50FsGT5ehdpDhw4db5cqV3eoe9eKYuJhWUJEGAABIZprQt3DhQjfZUCtyqMKsDVG0e6FCtLfzoPqS1ZKh3Qfz5cvnVu3Q7odej7OoaqzQ65cmNeqy1VftueWWW2zEiBFuCT1df82aNa1Lly6W1qW7Gp8VvJFotOajt1YjAAAAUm5eY7IhAAAA4ANBGgAAAPCBIA0AANKEK1foZk1LriTB481kQwAAkCakT5/Oxn/wre07eDy5bwpusMK35rauzR+80VdDkAYAAGmHQvQv+44m981AKkFrBwAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+BA1QXrOnDn2X//1X1auXDmrVq2aderUKeT05cuXW6NGjdzpderUsZkzZ0a8nKlTp1qtWrXcuCZNmth3330XY8ypU6esf//+VqVKFatYsaK7rn379sUYt3PnTmvXrp1VqFDB3aZBgwbZuXPnEvG3BgAAQEoVFUF63LhxNnToUGvYsKELwgMHDrRbb701cPq6deusS5cuVqZMGZsyZYo1btzYhVqF72A676hRo6xFixY2efJkK1q0qHXo0MG2bt0aMq5Hjx62ZMkS69evnxt/8OBBa9u2bUhIPnHihLVp08ZOnz5tY8eOtV69etn8+fOtb9++SXCPAAAAINrdlNw3YPv27TZx4kQXfB966KHA8ao6e8aPH+9C9ODBg93PVatWtf3799uYMWNc1Tl9+vR24cIFdzmtW7d2VWSpXLmyC+eTJk1ygVnWr19vy5Ytc9dXo0YNd1ypUqXc9c2dO9eaN2/ujps9e7YL0/PmzbO8efO64zJkyGA9e/a0zp07W4kSJZLwXgIAAEC0SfaK9Keffmq33357SIgOpoC8evVqq1+/fsjxCsiHDh2yTZs2uZ/Xrl1rJ0+etAYNGgTGKPjWq1fPtYVcvXrVHafvc+XKZdWrVw+MK1SokFWqVMmd5lmxYoVr5/BCtNStW9cyZcoUMg4AAABpU7IHaVWIVRFW1VnB9Z577rGWLVva5s2b3em7d++2ixcvWvHixUPOV7JkyUBFO/gwfJwqx2rPOHDgQGBcsWLFLF26dDEuz7sMb1x41VkhukiRIiHjAAAAkDYle2uHqsobN260n3/+2QYMGGAZM2a0t956y/Usf/XVV3b8+HE3TlXkYN7P3ulqw1DQzZIlS8i43Llzu8Njx45ZwYIF3bicOXPGuB26PO+yvMsLv85I4/xQdfzMmTMJugwAABB/KqBlzZqVuyyNOXv2bKAr4XroPOFF16gM0l6o1ITDO++80x1XtmxZq127tn344Yeu5UJi+2WCj480xrvzrjUuruPDLy8+4+KiCrtXcQcAADeeQrTmWyFt2blzpwvTfqhAG/VBWhXjfPnyBUK0aMUOtWhs27bNHnnkEXdceBVYFWPxqsY6PH/+vPvKnDlzjHFeZVrjNFExXHgFWt975w2mPuyETjRU1d1rTQEAADdeQotgSJmKFSvmqyKtDBofyR6kFUp//fXXGMfrl9ZqHOpJVvDcsWNHyARB7xf0Qq13qP7l4Hec+jl79uxWoECBwLhVq1bFqCzr8oIDsr4P74XWxEf1bGulkITQ9WbLli1BlwEAAIC4+W3nie8br2SfbFizZk07fPiw/fTTT4HjNDFQwbl06dKurK7l7hYtWhRyvgULFlj+/PkDoVktIOp9XrhwYWDM5cuX3fm0zJ13h+h7VZpXrlwZGKcKtVb98JbDE4V2rRZy9OjRwHGLFy92YTp4HAAAANKmZK9Ia/1m9UR369bNunfv7oKzVvDQsnNPPvmkG9O1a1e3koc2Q9Gydwq92oxFG7eoai06n9Z31nrROq8Ctsbs2bPHRo4cGbi+8uXLu/Dep08f6927t+XIkcOtR124cGG30YunWbNmNmPGDLcRjL6OHDkS2DSGNaQBAACQ7qqfxpFEppCqzVa0PvOlS5fs/vvvt5dffjlkKTudpkCsdgutvqFVPbSDYTD9KtrdUNuHq8qtZfVefPFFV9EO3yJ82LBh9uWXX7qJf9oqXLscKkyHN6hrB8U1a9a41UC0RrU2ZAlfGeR6bNiwwR1qC3MAAJC0Xhmz0H7Z93+fNiN1uqPwzTa4e70bnteiIkinJQRpAACSD0E6bbgjiYJ0svdIAwAAACkRQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGkCKs2jRIuvSpYvVqFHDKlSoYA0bNrRZs2bZlStXAmOuXr1qU6ZMsVq1atk999xjDRo0sIULF173dQ0aNMhKly5tAwcODDn+8OHD1r59e6tUqZI1b97cdu3aFXL6sWPHrFq1arZx48YE/KYAgGhGkAaQ4rz33nuWKVMme+mll2zSpEn26KOP2uuvv25vvvlmYMw777xjo0ePtieeeMLefvttq1y5sr3wwgu2ZMmSeF/P1q1b7ZNPPrEcOXLEOG3w4MF2+fJlGzt2rLstvXv3Djld163bVbZs2QT+tgCAaHVTct8AALheCs958+YN/Fy1alU7c+aMzZw50/72t7+54yZOnGitWrWyZ5991v384IMP2r59+1zAVZU6Pl577TVr27atzZs3L8Zp3377rat433vvvZYzZ0578skn3W3Ili2bbdmyxVXN9QUASL2oSANIcYJDtOfuu++28+fPu5aKPXv22OnTp+2hhx4KGfPwww+7KvOvv/56zev47LPPbO/evdahQ4eIp1+8eNGyZMnivvcOdZyoOt61a9eItxMAkHoQpAGkCmvWrLE8efLYLbfc4gK1ZMyYMWSMWjBk+/btcV7WqVOn7I033nCtI1mzZo04ply5cq4v+/jx464SXqRIEcudO7d9/vnndvToUXvqqacS7XcDAEQngjSAFG/Dhg326aefWps2bSxDhgwu1KZPn95++OGHkHHff/+9O1T4jctbb71lRYsWtXr16sU6plevXvavf/3L9V5rEqMmI549e9b1affp08duuonOOQBI7XilB5CiHTp0yJ577jlXIfbaMDQ58PHHH3cTDkuVKuVW9li6dKmrFotCdmy2bdvmKswfffRRnNdbpkwZd5lqIylcuLBlzpzZRo0a5XqmtVqHThs6dKhb3eORRx6xAQMGWPbs2RP5twcAJCcq0gBSrJMnT7rwrB5lTS4MbuXQKhpa9q5jx46uaqxQ2717d3davnz5Yr3MIUOG2GOPPebC8YkTJ9yXltVT/7P3vUfXV7x4cReiFag/+OADV6k+cuSIWyFEEx0XL15sO3fudLcPAJC6UJEGkCKpD7pz586u4vvhhx/azTffHHK6+qWnTp1qBw4ccK0cd9xxh3399dcu/KqaHBuF3m+++cZNNgymCrW+1MZRokSJiAFcq4QogOt61I+t9a2lcePGrvUEAJC6EKQBpDiXLl1y1WUtMzdjxgwXXmNToEAB96U1n1UxVt9zpHWhPSNHjgxMVvSouqz2kNatW1uhQoUiLoW3efNmd16PKti6TvVsa1k8AEDqk+xBWlWal19+Ocbx+ri2Z8+egZ+XL1/u+g81275gwYL29NNPW4sWLWKcTxUo9Teqb1K9kZp1X6VKlYgz8r/88ku7cOGCO71fv34x/hmrMqVdzbQagGbu169f390mb6krAMlDE/vUg/ziiy/auXPnApMIpWTJki4oq6KsQKyJhwcPHnRVay1nN3z48JDLqlOnjgvH06ZNcz8rMIdT64bCePhriRfqtdydWjq81wZdhlpAdF1qK9HqHnFNXAQApEzJHqQ9mhSkTQ08+qflWbdundsOWJOH1Pe4du1aF3D10WnTpk1DQrTCtjZk0Ee3c+bMcYFch9ri19OjRw+3ba/Cs/7hamcybbqgf7zeP0L1QmoFAP2D1em///67++hWa9SG/yMGkLTUeiHBOxl63n//fRd4tUX4u+++68KzNknRduL627311ltDxqtqHNz3fL1UEVfPtfqqPVqCb8SIEe4Nu9pBatas6V7DAACpS7qr+m8TBRXpf//737FuXtC+fXvX46hA7FEIVkVqxYoVbga+KssPPPCA211MVWjvH6R6FBWiFbBl/fr1bszkyZPdP1bR5gyqSvXt29eaN2/ujtPpEyZMcNsJe7dr/vz5riIdW49kfJfpEq0wAAAAktYrYxbaL/uOcrencncUvtkGd/f/SWB881rUr9qhgLx69WrXVhFMAVntG5s2bXI/q0qtGfwNGjQIjFFvoj5OVVuI935B3+fKlcuqV68eGKeqc6VKldxpHgV0LWEVHO7r1q3rquDB4wAAAJA2RU2QVgDWFr+1a9e2t99+21WTZffu3W7SjpaYCqY+yOAdyrzD8HGqHGurYM3c98YVK1bM0qVLF+Pygnc70/fhVWeFaPVbXmtXNCA1uHIlWT+sQhLj8QaAFNgjnT9/fuvWrZuVL1/ehVu1UowePdoF3/79+wd2IFMVOZj3s3e6epoVdMMnAmrLXlFvsyYpalxwL3bw5QXvdqZx4dcZaZwfqo4zix/RTH+LmmA7/oNvbd/BhD3fEf0K35rbujZ/0O3MmMzdfsANf11D2nLW5+uazhNedI3KIP3www+7L89DDz3kZshrBn2nTp0Cx8f2ywQfH2mMd+dda1xcx4dfXnzGxUUVdi2VBUQr/bPRhF2FaHoJ0w6tVKR/OkBqfl1D2rIzAa9rKtBGfZCO5M9//rObba+w6S1JF14FVsVYvKqxDrXUlb4UxMPHeZVpjdu/f3+M6wyvQOt777zB1Iftd6KhRxtCeK0pQDRK6JtFpExqe6MijdSK17W0qZjP17Vt27bFa1xUBulg6klW8NyxY0fIBEHvF/RCrXeo/uXgd5z6OXv27IHl9DRu1apVMSrLurzggKzvw3uhNfFRPdtNmjRJ0O+k69VyXAAQTfjYG0Bqk9VnO09833hFzWTDYFpeTituKBCrrF61alVbtGhRyJgFCxa4/movNGvVDfU+67weTVjU+bTMnXeH6HtVmleuXBkYpwq1Vv3wlsMThXatFnL06P8tkbN48WIXpoPHAQAAIG1K9op0u3btXFDWLoTy9ddfuw0MtBWvgrJ07drVWrZs6dZ51rJ3Cr1aU1q7m2kNaVHg7ty5s1svWkvWeRuy7NmzJ2TbXk1q1OYIffr0cZu7aEOWMWPGuBaSxo0bB8Y1a9bMbbSgTRT0deTIERs6dKi7/oS2dgAAACDluykaelc+/vhj++2339zuYnfccYe98sor1qpVq8CYihUrus1RFIjnzZvnVt9QqA7e1VCeeeYZ17Ixffp0O3z4sAvn2lgleFdD0Y5jw4YNswEDBriJf9oFbdy4cSErfqhHWhMetYOiVhXRaVqiL3jbcgAAAKRdyb6zYVrDzoZISdgBLG1I6A5gQErC61racAc7GwIAAADRKyonGwIAAADRjiANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAApPQgffr0aatevbqVLl3aNmzYEHLa8uXLrVGjRlauXDmrU6eOzZw5M+JlTJ061WrVquXGNWnSxL777rsYY06dOmX9+/e3KlWqWMWKFa1Tp062b9++GON27txp7dq1swoVKli1atVs0KBBdu7cuUT8jQEAAJBSRVWQnjBhgl2+fDnG8evWrbMuXbpYmTJlbMqUKda4cWMXaufMmRMjRI8aNcpatGhhkydPtqJFi1qHDh1s69atIeN69OhhS5YssX79+rnxBw8etLZt24aE5BMnTlibNm1cuB87dqz16tXL5s+fb3379r2B9wAAAABSipssSmzfvt1mzZrlAuvf//73kNPGjx/vQvTgwYPdz1WrVrX9+/fbmDFjXNU5ffr0duHCBZs4caK1bt3aVZGlcuXK1rBhQ5s0aZILzLJ+/XpbtmyZC9o1atRwx5UqVcpVuefOnWvNmzd3x82ePduF6Xnz5lnevHndcRkyZLCePXta586drUSJEkl6/wAAACC6RE1F+vXXX7dmzZpZsWLFQo5XQF69erXVr18/5HgF5EOHDtmmTZvcz2vXrrWTJ09agwYNAmMUfOvVq+faQq5eveqO0/e5cuVyLSSeQoUKWaVKldxpnhUrVrh2Di9ES926dS1Tpkwh4wAAAJA2RUWQ/uKLL2zLli3WtWvXGKft3r3bLl68aMWLFw85vmTJkoFKdvBh+DhVjtWeceDAgcA4hfV06dLFuDzvMrxx4VVnhegiRYqEjAMAAEDalOytHWfPnrWhQ4faCy+8YDly5Ihx+vHjx92hqsjBvJ+909WGoaCbJUuWkHG5c+d2h8eOHbOCBQu6cTlz5oxxPbo877K8ywu/zkjj/FB1/MyZMwm6DOBG0hvNrFmzcienMXo99j69A1IbXtfSprM+X9d0nvCia1QGafU133LLLfbEE0/EOS62Xyb4+EhjvDvvWuPiOj788uIzLi6qsG/evDlBlwHcSArRmpeAtEUrFemfDpAa8bqWNu1MwOuaCrRRHaS15Ny7777rJhNqSTrxKrU6VEuGV1EOrwKrYixe1ViH58+fd1+ZM2eOMc67HI3TRMVw4RVofe+dN5j6sBM60TBjxoyB1hQgGiX0zSJSJrW9UZFGasXrWtpUzOfr2rZt2+I1LlmD9N69e111tmPHjjFO0+ob5cuXtxkzZrjguWPHjpAJgt4v6IVa71D9y8GVNP2cPXt2K1CgQGDcqlWrYlSWdXnBAVnfh/dCa+Kjera1UkhC6HqzZcuWoMsAgMRGOw+A1CarzzbF+L7x8jXZ8O6777Yffvgh4mk//vijOz2+l/P++++HfL388svutAEDBrhl8FRW13J3ixYtCjnvggULLH/+/IHQrFU31Pu8cOHCwBitSa3zaZk77w7R96o0r1y5MjBOFWqt+uEthycK7Vot5OjRo4HjFi9e7MJ08DgAAACkTb4q0nGVyK9cuRLvFK/2Ce0uGEnZsmXdl2g1j5YtW7rNULTsnUKvNmMZOHCgW0NaFLi1vrPWi9aSdQrYGrNnzx4bOXJk4HJV5a5Zs6b16dPHevfu7SY4aj3qwoULu41ePFqKT9VwbQSjryNHjrhJkbp+1pAGAABAord2bNy4MeKqGAmhbby166ECsTZI0eobCtVNmzYNGffMM8+4kD99+nQ7fPiw22hFG69oy/FgI0aMsGHDhrmqt1pLFObHjRsXsuKHQv60adPcDordunVzp2mNam3IAgAAAKS7Gs8ObIVKtV7Ir7/+avny5Ysxm1ET/VS51cYlo0eP5t6NYMOGDe6wXLly3D+Ieq+MWWi/7Pu/9iakTncUvtkGd6+X3DcDSBK8rqUNdyTwdS2+eS3eFWktUXfnnXcGVtu4/fbbY6yzrGCtKrAmCgIAAACpWbyDtNoavO23W7VqZa+++iq9wgAAAEizfPVIqwcZAAAASMt8TzZUa7X6R9Tmod7ocI0aNUrobQMAAABSV5DWdotaam7Xrl0Rl8LT8ncEaQAAAKRmvoK01m/WxiRas1lLy8VnL3IAAADA0nqQ1q6Gr732mj322GOJf4sAAACAFMDXFuHZsmVzOwICAAAAaZWvIP3EE0/YggULEv/WAAAAAKm5tUObrnz++efWqVMnq1WrluXJkyfGmD/96U+JcfsAAACA1BOke/To4Q737t1ry5Yti7hqx+bNmxN+6wAAAIDUFKTff//9xL8lAAAAQGoP0pUrV078WwIAAACk9smGAAAAQFrnqyLdunXrOE9Xj/S0adP83iYAAAAgdQbpSNuCHzt2zG0dnjdvXrvjjjsS47YBAAAAqStIT58+PeLxCtJdunSxZ599NqG3CwAAAEg7PdLFihWzdu3a2ZtvvpmYFwsAAACk/smGhQsXtp9//jmxLxYAAABI3UH6q6++sltvvTWxLxYAAABI+T3SL7/8cozjLly4YD/99JNt27bNXnzxxcS4bQAAAEDqCtLfffddjOMyZ87s2jo6duxoDRs2TIzbBgAAAKSuIL1kyZLEvyUAAABACsLOhgAAAEBSVaS9DVj+8Y9/2OrVq+3o0aN288032wMPPGBt2rSx3Llz+71YAAAAIPVWpA8cOGBPPPGETZo0yU6ePGmFChVyhxMmTLDGjRu70wEAAIDUzFdFeuTIkXbu3Dn76KOP7N577w0c/8MPP1jnzp1t1KhRNnTo0MS8nQAAAEDKr0ivXLnSnn/++ZAQLfr5ueeesxUrViTW7QMAAABST5BWG4eWuovktttuc6cDAAAAqZmvIK2wvGzZsoinqRqt0wEAAIDUzFePtCYajhgxwq5evWqNGjWy/Pnz26FDh+yzzz6zGTNmWI8ePRL/lgIAAAApPUi3b9/e9uzZ40LzzJkzA8crWD/55JPWrl27xLyNAAAAQOoI0unSpbOBAwfa008/7bYL15rSefLksapVq1qxYsUS/1YCAAAAKbVH+vjx49atWzdbunRp4LjixYtb8+bN3ZJ3Ovzll1/cGG3QAgAAAKRm8Q7Sc+bMsS1bttjDDz8c6xid9tNPP4W0ewAAAABpOkgvXLjQmjZtajfdFHs3iE7TmCVLliTW7QMAAABSdpDeuXOnlStX7prjypYt61o8AAAAgNQs3kH68uXLcVajPRpz6dKlhN4uAAAAIHUEaa0VvW3btmuO+/nnny1fvnwJvV0AAABA6gjSlStXtlmzZtnFixdjHaPTPvjgA6tSpUpi3T4AAAAgZQfpNm3auD7pZ5991g4cOBDjdB3XtWtXN0brSwMAAACpWbw3ZLnrrrusf//+NmDAAKtdu7bdc889VrhwYXfavn377Mcff3Q7G7766qtWunTpG3mbAQAAgJS1s6G2/77zzjvt7bffdjsafv/99+74rFmzujWk//rXv1qFChVu1G0FAAAAUu4W4RUrVrRJkybZlStXAjsY3nzzzZY+fby7RAAAAIC0F6Q9Cs633HJL4t4aAAAAILUHaSDcrl27bOrUqbZ+/Xq3DGLx4sVtwYIFIWPi6p9fuXKl3XrrrRFPO3XqlL3yyiuuF//w4cOWLVs216f/3HPP2b333hsYp9N69+5ta9euddc1dOhQK1q0aOD0Y8eO2Z///Gd755133OZBAAAAfhGkkWgUnpcvX27ly5d3rT+afBruww8/jHFcr169XJ99bCHaW1oxc+bM1q1bN/vDH/5gJ0+etGnTprnVZD799FMrVqyYGzd48GC3edDYsWNtypQpLlRrSUbP6NGj7dFHHyVEAwCABCNII9HUqlXLhVRRgFX1OFz4ZNS9e/e6LeVffPHFOC9bffhvvvlmyHEPPPCAW7P8yy+/tE6dOrnjvv32WxegVaXOmTOnmyB75swZV8HesmWLLVq0yH0BAAAkFDMEkWj8TDhV60e6dOmsQYMG131ehWNVqYO3pFflOkuWLO5779DbROj11193a53nzZv3uq8LAAAgHEEayerzzz+3+++/3woWLBiv8WoZUXA+ePCg639WeH/88ccDp5crV87twHn8+HGbOXOmFSlSxHLnzu2uR6vMPPXUUzfwtwEAAGkJrR1INmq1+Omnn2zgwIHxPs+YMWPc8ouiVWMmT55st99+e0i/dceOHV1ftFo7xo0bZ2fPnnVtIUOGDLGbbuIpDwAAUklFWis1tGzZ0qpWrepWYdCuiQo8mkwWTJPYGjVq5CqOderUcdXGSLRqhHp1Na5JkyZu45hIK0Bol0b112pdbPXXanfGcNruvF27dq6vt1q1ajZo0CA7d+5cIv72adv8+fMtY8aMVrdu3XifRxXljz/+2CZOnOj6oBWaN27cGDi9TJkytnTpUtcHrX5pPW4K3hqr73Waru++++6znj172unTp2/QbwcAAFK7ZA/S+gheYfa1115zIbht27Y2b9486969e2DMunXrrEuXLi4kaSJZ48aNXaidM2dOyGXp/KNGjbIWLVq4SqWWPevQoYNt3bo1ZFyPHj1syZIl1q9fPzdebQK63uCQfOLECbcihIKWVoBQpVPBr2/fvklwr6R+WtFj4cKFbkfMPHnyxPt8BQoUcG+S9GZpwoQJVqhQIff4BFM419J76p/es2ePq07r8Tty5Ii98MIL9uyzz9rixYvdGyUFcgAAAD+S/XNuTTILnmimKnGmTJlcyD1w4IALTuPHj3chWkubiarX+/fvdx/zq+qsPtkLFy64UNS6dWtXRZbKlStbw4YNXUVSgVm0xvGyZctc0K5Ro4Y7rlSpUq7KPXfuXGvevLk7bvbs2S5MK9R7k9MyZMjgqpidO3e2EiVKJPl9lZqsWbPGfv3112uu1hEXPe533313YKv6SPTpRqtWraxw4cL29ddfu+eWnhOiN2RaOg8AACBFVqQj8SqUmlSmgLx69WqrX79+yBiFoUOHDtmmTZvcz9qAQ+0gwaFcwbdevXquLcRb01jf58qVy6pXrx4Yp6pmpUqV3GmeFStWuFaA4BUe1BKgIBY8Dv6ouq9VNx555BHfd6FW4/jhhx9CeqSDqbVj8+bN7lOJ4PNonWnRsngAAAAptiLtUbhRcN62bZurQCtgqYqonxV+9FF9sJIlS7rD7du3u95qHUr4OFWO1Z6h6rZWhtA4bd6hJdfCL++bb74J/KxxqnYHU4jWKhDedSGUJvV5bzLUc65e9C+++CLw6YD3pkSPs9Z+1prT2oglEn1CoDc42nTF28hFoVlrR+fPn9+9idKnBrt37444WVHXoeXu1NLhLYOnXnet+jF8+HB3e7S6h95oAQAApOggreCssCvqmx05cmSgh1pURQ7m/eydrjYMBV0vNHm09Jm3NbSCtMZpNYdwujzvsrzLC7/OSOP8UHU8NVZD1aoR3Nsu3s/qbf/jH/8YmGCqpegUlmO7H/TmSV/e6ao6K5SrN16fPGjFDm3xPX36dLcVePjlzJgxwwV3ffLgnabQrvYgtfl89NFH9tBDD7ne+NT4WCSU3mjG9iYHqfvNcKQdSYHUgNe1tOmsz9c1nSe86BrVQVo9ywo0qkBrEplW0njvvfcCp8f2ywQfH2mMd+dda1xcx4dfXnzGxUUBUS0HqZGqvLHxfud8+fIFxsV2P4wYMSLkdE0c1CTBcKowR7oMrcqhr/DT1HOv9ac9qmgjJoVozUtA2qIJuPqnA6RGvK6lTTsT8LqmAm2KCdJ33XWXO1Svsv6Bq61CKyt4LRzhVWBVjMWrGuvw/Pnz7kuhK3ycV5nWOE1UDBdegdb33nmDqRqa0ImGWlXC+72AaJTQN4tImdT2RkUaqRWva2lTMZ+vayrsxkfUBOlgWolBEwVVLdQyZwqeO3bsCJkg6P2CXqj1DtW/HFxJ08/Zs2d3lUhv3KpVq2JUlnV5wQFZ34f3Qmvio25TeO/09dL1aqIdAEQT2nkApDZZfbYpxveNV1Su2qF1ozX58LbbbnNldS13pw02gi1YsMBNOvNCsyrZ6n3W2sQeXYbOp2XuvDtE36vSrD5djyrUWvXDWw5PFNq1Woh6eT2qkCtMB49Laleu0L+YlvB4AwAQvZK9Iq2+V626oQljmiiobaPfeecd97NWdZCuXbu63Q+1GYqWvVPo1WYsWq1BawmLArfWd9ZEMk0yU8DWGG3I4U1clPLly1vNmjWtT58+1rt3b8uRI4dbj1orhGhdYU+zZs3chDVtBKMvbeah3lpdf3KuIZ0+fTob/8G3tu9gwiY8IvoVvjW3dW3+YHLfDAAAEK1BWls3q4qsyYZqt1CgffLJJ92mKl6Tt3Y+1AREBWJtkKLVNxSqmzZtGnJZzzzzjLsMreRw+PBht9GKLlehPHwi27Bhw2zAgAFu4p82gRk3blzIih/qkdbSa1ololu3bu40rVGtDVmSm0L0L/v+r1IOAACANBikO3bs6L6uRe0U12qpUPtG+/bt3VdcVIXWluT6ulaDurYdBwAAAFJEjzQAAAAQ7QjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDABCLXbt2Wf/+/e3xxx93O+ZqY664/Pjjj3b33Xe7jcTi48CBA/b888/bfffd587TqVMntyNvMG0wpv0RKlWqZM2bN3e3KdixY8esWrVqtnHjRh5HIIkRpAEAiMXPP/9sy5cvt6JFi1qJEiXivJ+0s642+sqbN2+87s/Lly+7gKzwPXDgQHvzzTftt99+szZt2tjp06cD4wYPHuzGjh071u3427t375DLGT16tD366KNWtmxZHkcgre1sCABAtKpVq5YLqaIAq9Abm08++cSOHj1qTZo0senTp1/zsr/44gv76aef7LPPPrPSpUu748qVK+eub86cOfb000+747799lubMmWK3XvvvZYzZ0578skn7cyZM5YtWzbbsmWLLVq0yH0BSHpUpAEAiO2fZPr4/Zs8ceKEjRgxwl5++WXLmDFjvM6zadMmy58/fyBES4ECBezOO++0JUuWBI67ePGiZcmSxX3vHeo4ef31161r167xroIDSFwEaQAAEkjtFWqteOSRR+J9nvPnz7tWjXA6bseOHYGfVaWeNWuWHT9+3GbOnGlFihSx3Llz2+eff+4q4E899RSPH5BMaO0AACABNm/ebB9//LHNnTv3us5XrFgx1xOtCYeqRIt6o7dt22bnzp0LjOvVq5d17NjRPvjgA9faMW7cODt79qzrqR4yZIjddBP/yoHkQkUaAACfNMFQEwVVFb7WZMRwWgFEwVjtILt373aBul+/fq7/ObilRKuFLF261PVBq19aK3RMmjTJ9Uzre51Wt25dt/JHz549QyYqArixCNIAAPi0cOFC2759u7Vq1cr1SetLLRsS/H0kas8YOXKkq0DXqVPHqlevbgcPHrRGjRpZvnz5Qsaq77p48eKWOXNmtzyeqtOqVB85csReeOEFe/bZZ23x4sW2c+dOmzhxIo8nkET4PAgAAJ/Uy6zeZa3uEe7++++3Dh06uCpxbB588EFXUf7ll19cb/Ttt9/u2jgqVKgQ63nUzqHgXrhwYfv666/d+Ro2bOhOa9y4sX366ac8nkASIUgDAOCTgmvlypVDjlOvtCrVWrKuUKFC17yMDBkyBNpCVN1etWqVO28kau1QT7Yq2R6t4KF1pnU5agsBkHQI0gAAxEKT+rQhi+zbt89OnTrl1n8WBejbbrvNfQX7z3/+40JtlSpVQo5X+4aC9bRp0wLHacKgqs85cuSwrVu3urYMtXao9zncpUuX3HJ3aunwlsHTea9cuWLDhw93t0ere9SrV4/HE0giBGkAAGKhHuTu3buHHOf9/P7778cIy3FR1VihN5hW7Xj11Vdde4haNf7617+6nQ0jmTFjhuudfuyxxwLH3XLLLW796jfeeMM++ugjq1mzpnXp0oXHE0giBGkAAGKharMqxdejW7du7itc8CYrHoXg+NJOh95uh8Fq167tvgAkPVbtAAAAAHwgSAMAAAA+EKQBAMnqypWrPAJpCI83UhN6pAEAySp9+nQ2/oNvbd/B4zwSqVzhW3Nb1+YPJvfNABINQRoAkOwUon/ZdzS5bwYAXBdaOwAAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAABASgzSixYtsi5duliNGjWsQoUK1rBhQ5s1a5ZduXIlZNzy5cutUaNGVq5cOatTp47NnDkz4uVNnTrVatWq5cY1adLEvvvuuxhjTp06Zf3797cqVapYxYoVrVOnTrZv374Y43bu3Gnt2rVzt6tatWo2aNAgO3fuXCL+9gAAAEipkj1Iv/fee5YpUyZ76aWXbNKkSfboo4/a66+/bm+++WZgzLp161zYLlOmjE2ZMsUaN27sQu2cOXNihOhRo0ZZixYtbPLkyVa0aFHr0KGDbd26NWRcjx49bMmSJdavXz83/uDBg9a2bduQkHzixAlr06aNnT592saOHWu9evWy+fPnW9++fZPgXgEAAEC0uym5b4DCc968eQM/V61a1c6cOeMqzn/7299cyB4/frwL0YMHDw6M2b9/v40ZM8ZVndOnT28XLlywiRMnWuvWrV0VWSpXruwq3LoOBWZZv369LVu2zAVtVcGlVKlSrso9d+5ca968uTtu9uzZLkzPmzcvcPsyZMhgPXv2tM6dO1uJEiWS/L4CAABA9Ej2inRwiPbcfffddv78eTt27JgLyKtXr7b69euHjFFAPnTokG3atMn9vHbtWjt58qQ1aNAgMEbBt169eq4t5OrVq+44fZ8rVy6rXr16YFyhQoWsUqVK7jTPihUrXDtH8O2rW7euC/bB4wAAAJA2JXuQjmTNmjWWJ08eu+WWW2z37t128eJFK168eMiYkiVLusPt27eHHIaPU+VY7RkHDhwIjCtWrJilS5cuxuV5l+GNC686K0QXKVIkZBwAAADSpmRv7Qi3YcMG+/TTT61r166uonz8+HF3vKrIwbyfvdPVhqGgmyVLlpBxuXPndoeqbhcsWNCNy5kzZ4zr1eV5l+VdXvh1Rhrnh6rjal+5Xgr/WbNmTdB1I+U5e/Zs4BOVpMJzLW3iuQaea0htzvr8H6rzhBddoz5Iq1XjueeecytuaJJgsNh+meDjI43x7rxrjYvr+PDLi8+4uKjCvnnz5us+n0K0esWRtmj1GL0QJCWea2kTzzXwXENqszMB/0NVoE0xQVr9zQrPqihr0mDGjBlDKsrhVWBVjMWrGutQfdX6ypw5c4xx3uVonCYqhguvQOt777zhtzOhEw31u3mtKdcjoQEeKZNakZKjIo20h+caeK4htSnm83/otm3b4jUuKoK0wq9Wwjh8+LB9+OGHdvPNNwdOU0+ygueOHTtCJgh6v6AXar1D9S8HV231c/bs2a1AgQKBcatWrYpRWdblBQdkfR/eC62Jj+rZ1kohCaHrzZYtW4IuA2kH7TzguYbUhtc1RPtzLb4FpWSfbHjp0iXr3r27bdmyxd555x0rXLhwjLK6lrvTxi3BFixYYPnz5w+EZq26od7nhQsXBsZcvnzZnU/L3Hl3iL5XpXnlypWBcapQa9UPbzk8UWjXaiFHjx4NHLd48WIXpoPHAQAAIG1K9or0wIEDbenSpfbiiy+6DVG+//77wGlqf8iRI4ebeNiyZUu3GYqWvVPo1WYsOq/WkPYCt6raWi9aS9YpYGvMnj17bOTIkYHLLF++vNWsWdP69OljvXv3dpev9agV4LXRi6dZs2Y2Y8YMtxGMvo4cOWJDhw51188a0gAAAEj2IP3NN9+4w+CdDD3vv/9+YBvvCRMmuECsDVK0+oZCddOmTUPGP/PMM65lY/r06a5NRButaOOV0qVLh4wbMWKEDRs2zAYMGOAm/uk6xo0bF7Lih3qkp02b5nZQ7NatmztNa1RrQxYAAAAg2YO0tuqOD7VTXKulQu0b7du3d19xURX6tddec1/XalDXtuMAAABA1PVIAwAAACkRQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAApMQgvWvXLuvfv789/vjjVqZMGWvQoEHEccuXL7dGjRpZuXLlrE6dOjZz5syI46ZOnWq1atVy45o0aWLfffddjDGnTp1y11mlShWrWLGiderUyfbt2xdj3M6dO61du3ZWoUIFq1atmg0aNMjOnTuXCL81AAAAUrpkD9I///yzC8lFixa1EiVKRByzbt0669KliwvaU6ZMscaNG7tQO2fOnBghetSoUdaiRQubPHmyu8wOHTrY1q1bQ8b16NHDlixZYv369XPjDx48aG3btg0JySdOnLA2bdrY6dOnbezYsdarVy+bP3++9e3b9wbdEwAAAEhJbkruG6Dq8aOPPuq+7927t/34448xxowfP96F6MGDB7ufq1atavv377cxY8a4qnP69OntwoULNnHiRGvdurWrIkvlypWtYcOGNmnSJBeYZf369bZs2TIXtGvUqOGOK1WqlKtyz50715o3b+6Omz17tgvT8+bNs7x587rjMmTIYD179rTOnTvHGvoBAACQNiR7RVohOC4KyKtXr7b69euHHK+AfOjQIdu0aZP7ee3atXby5MmQ1hAF33r16rmK99WrV91x+j5XrlxWvXr1wLhChQpZpUqV3GmeFStWuHYOL0RL3bp1LVOmTCHjAAAAkDYle5C+lt27d9vFixetePHiIceXLFnSHW7fvj3kMHycKsdqzzhw4EBgXLFixSxdunQxLs+7DG9ceNVZIbpIkSIh4wAAAJA2JXtrx7UcP37cHaqKHMz72TtdbRgKulmyZAkZlzt3bnd47NgxK1iwoBuXM2fOGNejy/Muy7u88OuMNM4PVcfPnDlz3edT+M+aNWuCrhspz9mzZwOfqCQVnmtpE8818FxDanPW5/9QnSe86Joig7Qntl8m+PhIY7w771rj4jo+/PLiMy4uqrBv3rz5us+nEK1ecaQtWj1GLwRJieda2sRzDTzXkNrsTMD/UBVoU3yQ9irK4VVgVYzFqxrr8Pz58+4rc+bMMcZ5l6NxmqgYLrwCre+98wZTH3ZCJxpmzJgx0JpyPRIa4JEyqRUpOSrSSHt4roHnGlKbYj7/h27bti1e46I+SKsnWcFzx44dIRMEvV/QC7XeofqXg6u2+jl79uxWoECBwLhVq1bFqCzr8oIDsr4P74XWxEf1bGulkITQ9WbLli1Bl4G0g3Ye8FxDasPrGqL9uRbfglLUTzZUWV3L3S1atCjk+AULFlj+/PkDoVmrbqj3eeHChYExly9fdufTMnfeHaLvVWleuXJlYJwq1Fr1w1sOTxTatVrI0aNHA8ctXrzYhengcQAAAEibkr0irb4Vbzk57S6oXQe/+OKLwDrQWn6ua9eu1rJlS7cZipa9U+jVZiwDBw4MLJ+nwK31nbVetM6jgK0xe/bssZEjRwaur3z58lazZk3r06ePW7c6R44cbj3qwoULu41ePM2aNbMZM2a4jWD0deTIERs6dKi7ftaQBgAAQLIHaQXU7t27hxzn/fz+++8HtvGeMGGCC8TaIEWrbyhUN23aNOR8zzzzjGvZmD59uh0+fNhttKKNV0qXLh0ybsSIETZs2DAbMGCAm/in6xg3blzIih/qkZ42bZrbQbFbt27uNK1RrQ1ZAAAAgGQP0rfddluMLbwjUTvFtVoq1L7Rvn179xUXVaFfe+0193WtBnVtOw4AAACkuB5pAAAAIBoRpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAAAgSAMAAABJg4o0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEhfw86dO61du3ZWoUIFq1atmg0aNMjOnTvn574GAABAKnJTct+AaHbixAlr06aNFSpUyMaOHWu///67DRkyxI4dO2bDhw9P7psHAACAZESQjsPs2bNdmJ43b57lzZvXHZchQwbr2bOnde7c2UqUKJFUjxMAAACiDK0dcVixYoVr5/BCtNStW9cyZcpky5cvT4rHBwAAAFGKIB2H7du3x6g6K0QXKVLEnQYAAIC0i9aOOKitI1euXDGO13HHjx/3dYdfvHjRrl69aj/88IOv86dLl87qV85vl6/c4uv8SDkypE9vGzZscM+X5MBzLe3guQaea0htMiTwf6jymv4PXgtB2gc9KPG5cyPxzuf3/JIrRxbf50XKk5DnSkLxXEtbeK6B5xpSm3QJyGsE6QRS5VlV6XAnT570PdGwYsWKCb1ZAAAAiAL0SMdBYTm8F/rChQu2e/duVuwAAABI4wjScahevbqtXr3ajh49Gjhu8eLFLkzXqFEjKR4fAAAARKl0V5NrJlMKoLaOBg0aWOHCha1Lly525MgRGzp0qD300ENsyAIAAJDGEaTjsUW4tgVfs2aNZcmSxQVrbcii7wEAAJB2EaQBAAAAH+iRBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgjRRp7969Vrp0afviiy/iHPfpp5+6cb///vt1Xb7f8yE69O7d222eFJdatWrZwIEDr/uy/Z4PaZOeh3o+JqXNmzfbuHHj7OzZs0l6vUg6enwrVqzIXR4FCNJI1WrWrGkffvih5cqVK7lvCqLMW2+9Zc8880xy3wzghgRpPb8J0sCNd1MSXAcQcO7cuSTdXj1v3rzuCwhXpkwZ7hSkGEn92gkgfqhII2Jbg0LG4cOHQ44/duyY3XPPPTZz5kz387p166x169ZWoUIFu++++6xHjx525MiRGO0Xury+fftalSpV7L//+7/t/fffd+c5depUyOXv3LnTjf/666/j/aicP3/efcx+//3320MPPWTDhg2zS5cuxdmi8dtvv9lf//pXu/fee+3hhx+2d955x12GPrIPt3//fmvfvr27vX/6059s3rx5PGNSkO+++84aNWrkHj8993788cc4WzRmz55tjzzyiJUvX97atGlj69evDzyHw82YMcON1XO/S5cutAGloJaflStXWsOGDd1rwFNPPWV79uxxr2/PP/+8VapUyR599FFbuHBhyHn1ydaf//xn9xqoT7pGjRoV8loja9eutSeeeMLKlSvnrmf58uURb4ff105ZtmyZtW3b1qpVq+Zua9OmTW3FihWB8+o8L7/8svteY3Q5wa9tev3r2bOnu0z9/i1atAj5u0DK9Ouvv9pzzz1nf/zjH93rl55fGzZsCJw+fvx497z1XL161T0H9KXvPTVq1LAJEyYk+e1PyQjSiEGB8aabborRf/zVV1+5Pzj9M9E/glatWlnOnDndP5TXXnvN/dF27tw5xuWNHDnS0qVLZyNGjLAXX3zRHn/8cbty5YotWLAgZNwnn3xi+fPnd3/I8TV69GhLnz69O/zLX/5i7777rs2ZMyfW8br9Cj366FMh6tVXX3XBPbbwrturgK4Xobvuusv9I962bRvPmhTg0KFDNmjQIGvXrp17jqqi9+yzz9rFixcjjtdz4O9//7s9+OCD7mNxHSrgRLJkyRJbunSp9e/f3/r06WP/+c9/3N8AUsbzYvjw4e61SocKrfo7f+GFF+zOO+90vadly5Z1x+3bt8+dZ/r06e6xrlq1qk2cONGaNWtmU6dOdccFX66ea5kyZXKvR/p+wIAB7vhgCXntFN1evYF744033G1VmO7YsaN70ygKS95lqUigNwB6Psvx48fdG4ctW7ZYv3793PmzZs3q3jQGB3mkLCpKtWzZ0r0h0nNSz5cLFy64ML19+3Y3RsUmFYb0plG2bt3qzqevn376yR2n0/RGS2MRf7R2IIYcOXK4MKugqz9Oj37WPxK1SuidryozeoHWC73on5CqPKrCBIdhVbfDQ0bdunVdcNY/JLl8+bKr9qp6qBAfX6qoqGIjCj6rVq2yL7/80po3bx5xvCo3GzdudFV1vXMXvSOvXr265cmTJ8Z4VWv0JXqXr2qQ3lCULFmSZ06UU2hQ1VjPS8mcObOr5KnK7D32wRSQ9PxW+BZ9WqFPPLwQEv6GTOMVmmTXrl0uWOkNot7YIbqfF7NmzbISJUq4nw8ePOhenzp06GBdu3Z1x6mivHjxYvvXv/7lXgP1Rvqxxx5zb7S854Ze9xSEFVpvv/12mzZtmjtu8uTJgTkZKgwoUAdTyEnIa2fwa7Keb3r90pv7jz76yH2v1+ciRYq40/WGILi1TbfxxIkTrthwyy23BKrWderUcc/fl156KZHvbSQFfQqhivT8+fMDr3d6XPVJxJQpU2zo0KHu/5der/7nf/7HPV//93//1z2/9Fqm4/TJhQ41RmMRf7ziI6L69evb999/7/44RVUV/ZHpxV4TWPQRpv6xKADr4019FStWzP3jCP44SSJVmJ988kn74Ycf7Oeffw4EXF1HkyZNrusRUbU4mAKu3lHHRrdN/+SCg5TeOOgf0LUuX+P+8Ic/xHn5iB633npr4J+KeMHpwIEDMcbqeaxPKcLbe2rXrh3xslWx8UK097xTpZuqXsp4XnjPBbnjjjvc4QMPPBA4Tq8RCqD6W9+xY4cdPXrU6tWrF+M1UiFkzZo17me9QdPrSPDEZr1+6HXDkxivnbpNvXr1cmFeQUhh+ZtvvnGtcdfy7bffutuYO3fuwHXrjZ9eD8OvGymHQrFe64Jf77Jnz+4+udBpXiFBbxD1f1x0qNcxfekTNe84FaeCX9twbVSkEZH+APWH+Pnnn7tKjfoFM2bM6HoHVdHQP4EhQ4a4r3D6+ChYpMl++uPVP4+PP/7Y9fPpUC/mOu566OPRYLqN+kgrNqo+Rbo9XnUmoZeP6BG+UoseO1GVOZx66BUqwp8bsT0vrueyEV1ie+zC/9YVJvR4qoIt+fLlCzldwVe801UIKFq0aIzrC34OJfS1UxVoVcBPnjzpPhXU9ak1Y+zYsTHOG4neEKhAovAdzqtiI+XR8yr8+Sk6znt+ev939T9d9AZQnwDrzeA///lPd5xCt94g4voQpBGR3r16E268IK3eO1VXVMHQR5KasKcx4W6++eaQn72PL8Npkox6+LQEmT7STIoeU1WjIq0NTSUxbVNgUUtR+HOD5wW8lq/w54LX+6zqrhesIz1fgo9TWE/Ia6daiDZt2uRaTYLPr/7/+NBtVSW7e/fuMU6jCply6XHVJyfhtGCA9/z0gvSkSZPs3//+t3teaqKr9wZLx+3evZv+aB8I0oiV3pmqb1kz3FXF0MQUyZYtm5ttrj9cfVTkV+PGjV2PoSZ0Kbjr484bTbdX7969j7VEky00USdSjzTShgwZMtjdd9/tJhxq4pVHPbJI2/Qpmd5oLVq0yE3E9qi4oKDrhRF9JP7BBx+4arFX3VbLRfDqRAl97fQ+8fCq6KIJkZrA6LWoBJ8e/umZ2lc+++wz19qi24LUQc9BzQ1Sr7w3f+fMmTNuQrQ+XfZoAxcVDLQqh3qivU9nSpUq5Y7TaWzycv0I0oiVXnT1D+SVV14JTED0aFKKAoeWi1Lg1h+kevc02U/LP8XWcxxMl60eVK0OohU39BHljaZJhfpYU+Fds/R1uzUZw6sUIe3SR+Za0UWTV/WmTpU/7yNPJhCm7TdZmoSoT8z0mqVgoueG2in0WqeJW6LXQ01i1Cd4+tIbdhUfwt+gJ+S1s3jx4lawYEE3YVFtHuq51u3QJ23BvB5wTapW5VrrTys4Pf30025CmiYsakWHQoUKuU9h1N9doEABdzpSHj1v/vGPf7hPOvS80pskfdqrN156LnrUrqmCgXqitXKMR0UlrUyjN4O8wbp+TDZErPTuVIFCfcWa1a2qsUdLLumfht71qsdZyy/pHa1esCP1CcZGlyveGqk3msKybqeWstMyQfrSP8bKlSvH6JFE2qI3dVoOUVVEBWp9EuOt0hA8YQxpj4KnlrLTZL1OnTq5gKrVOILXIVeY1ZtytVmodULf6/XF66VOjNdOtV8onOtQ1zFmzBh3e/T6FUyTELt16+aqz1oZyVsOT60jWg5PYUpL/6mtTr3aqmorRCFl0uuTVihSkUivYSoS6f+39mwInlgr3iexwUvced9HWs0I15buavBK3EASU3VGqyWoSpJc9PGn1sbWP6NIE4CQdmmZMFWo1fJx2223JffNAQBEGVo7kCy0GLwCtPoMvapfUlFFRh+LqvdRH7+qr1Ez3rVRAdIu7WyntX21lrQ+AtVyYJqYo0o1IRoAEAlBGslCHzWqN0/L74SvHa0PSbREVGzUr5qQnlW1qOhjV+0QJmrzePvttxM0cRIpnz4K1c5eWh5Kb7D0Mbh24dR2ygAAREJrB6KOVtDQRJi4VvvQTk0AAADJiSCNqKPlouLapUuVQj5qBwAAyY0gDQAAAPjA8ncAAACADwRpAAAAwAeCNAAAAOADy98BQBqkLaPjQ7ujxbVtNQCkZUw2BIA06Pvvvw/5WdtUa+nJadOmhRxfsmRJtkgHgFhQkQaANKhChQohP+fNm9dtdBR+PAAgdvRIAwBieOWVV6xy5cp29uzZGKdpw6T69euHtIkMHDjQZs+ebXXr1rV77rnH6tWr53aJDHfo0CHr37+/Va9e3Y2rVauW25r90qVLPAoAUhwq0gCAiGH5k08+sQULFljTpk0Dx2/bts21gCgMB1uyZIk7/rnnnrOsWbParFmz7IUXXrAMGTLYY489FgjRuixVvrt27WpFihSxdevW2cSJE23fvn02ZMgQHgkAKQpBGgAQw1133eUq0jNnzgwJ0jNmzHA9040aNQoZf/ToUfv4448tX7587ucaNWpYgwYNbOTIkYEgPW7cODt+/LirVBcqVMgdV61aNcuSJYsNGzbM2rVr53qyASCloLUDABBrVXrz5s22Zs0a9/OpU6fsn//8pzVu3NiyZ88eMlaB2AvRokq02jt27dplv/32mztu2bJlbgWQW2+91bVyeF9q85D//Oc/PBIAUhQq0gCAiGrXrm2FCxd2bRr33Xefffrpp65n+qmnnooxNjhEhx937NgxK1iwoB05csSWLl1qZcuWjXh9qmoDQEpCkAYARKRe5hYtWtioUaOsV69eLlCr8ly8ePEYYw8fPhzrcXny5HGHN998s5uY+Pzzz0e8PlWqASAlIUgDAGKl/mitqtGzZ0/buXOnO4zk3//+twvOXhX68uXLtnDhQjehUNVoqVmzpi1fvtwdlzt3bu51ACkeQRoAEKtcuXLZ448/bh988IFr89BydZGo2tymTRvr0qVLYNWOHTt2uGq2Ryt6rFq1ypo1a2atWrWyYsWK2YULF2zv3r22YsUKGzBgQCB0A0BKQJAGAMRJkwYVpBWA1e4RiQK2VtwYPXq07d+/326//XYbPny4O29w64ZW9tAuilOnTrUDBw64SYsK6A8//LAL7QCQkrBFOAAgTkOHDnVBWqtuqPIcTn3P6qUOX1saAFI7KtIAgIi+//57++WXX1ybxl/+8peIIRoA0jKCNAAgIoVn9TtrkmBsK20AQFpGawcAAADgAzsbAgAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAB2/f4f9kJ2FkrTR+MAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 800x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Set plot style\n",
    "sns.set(style=\"whitegrid\")\n",
    "\n",
    "# Calculate counts and percentages\n",
    "counts = promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM['Type'].value_counts()\n",
    "total = counts.sum()\n",
    "percentages = (counts / total * 100).round(2)\n",
    "\n",
    "# Desired order\n",
    "type_order = ['very_high', 'high', 'moderate', 'low']\n",
    "\n",
    "# Plot\n",
    "plt.figure(figsize=(8, 6))\n",
    "ax = sns.countplot(data=promoter_like_openChromatin_chromHMM_GeneTPM_TxTPM, x='Type', order=type_order)\n",
    "\n",
    "# Annotate each bar\n",
    "for p in ax.patches:\n",
    "    height = p.get_height()\n",
    "    label = p.get_x() + p.get_width() / 2\n",
    "    category = p.get_x() + p.get_width() / 2\n",
    "    type_name = p.get_x() + p.get_width() / 2\n",
    "    percentage = (height / total) * 100\n",
    "    ax.annotate(f'{percentage:.1f}%', (p.get_x() + p.get_width() / 2, height),\n",
    "                ha='center', va='bottom', fontsize=11)\n",
    "\n",
    "# Labels\n",
    "plt.title('Distribution of Type Categories')\n",
    "plt.xlabel('Type')\n",
    "plt.ylabel('Count')\n",
    "plt.ylim(0, counts.max() * 1.1)  # Add some space above bars\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45103ff0-df77-4f8c-9282-2cb881ef0cfc",
   "metadata": {},
   "source": [
    "# add Hi-C data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "258ee5ef-7830-442f-9374-7756ab4890ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin_chromHMM_Expression_level.tsv -b /mnt/10T-backup-mingju/ifar/Hi-C/MB/merge_results.txt_sorted_lexicographical_addchr -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin_chromHMM_Expression_Level_HiC.bed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "d2bb457a-11d1-4f5f-90a7-a0e55d4bf4ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "!perl ~/PreprocessDB/promoter_table/promoter_HiC_processing.pl /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MB/promoter_like_regions_MB_openchromatin_chromHMM_Expression_Level_HiC.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2c1cfe5-3833-4db0-88fb-abb2cc30f609",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "browser_data",
   "language": "python",
   "name": "browser_data"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
