{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "23a7f7d2-2e98-45be-b488-688489353f1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a29f3799-2118-4ac8-b80b-4c0ff7748a4e",
   "metadata": {},
   "source": [
    "# MT ATAC-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "c6ed78fb-9417-47a0-9a96-d7cf45fb998f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju\n"
     ]
    }
   ],
   "source": [
    "%cd ~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "8398baaa-d481-40ea-81b9-940aa7ee0027",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju/PreprocessDB/promoter_table\n"
     ]
    }
   ],
   "source": [
    "%cd ./PreprocessDB/promoter_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "3a776d02-bfde-46bb-8f33-96702b066d24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mingju/PreprocessDB/promoter_table'"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0444d201-284c-4343-bd23-243324430ab7",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac = pd.read_csv('/mnt/10T-backup-mingju/ifar/ATAC_seq/MT/atac_output/idr.optimal_peak.narrowPeak_w_header',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4d9d72a8-f8cc-465b-a0aa-4ee95e8a4f2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>10.18118</td>\n",
       "      <td>5255.37842</td>\n",
       "      <td>5247.06348</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>9.06746</td>\n",
       "      <td>4341.48682</td>\n",
       "      <td>4334.17773</td>\n",
       "      <td>321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.49633</td>\n",
       "      <td>3135.06812</td>\n",
       "      <td>3128.26001</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.74865</td>\n",
       "      <td>2599.60767</td>\n",
       "      <td>2593.02832</td>\n",
       "      <td>305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521226</td>\n",
       "      <td>22521666</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.02649</td>\n",
       "      <td>2324.03174</td>\n",
       "      <td>2317.51929</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151265</th>\n",
       "      <td>chr1</td>\n",
       "      <td>101174750</td>\n",
       "      <td>101175746</td>\n",
       "      <td>.</td>\n",
       "      <td>763</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151266</th>\n",
       "      <td>chr10</td>\n",
       "      <td>73076272</td>\n",
       "      <td>73077247</td>\n",
       "      <td>.</td>\n",
       "      <td>599</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151267</th>\n",
       "      <td>chr10</td>\n",
       "      <td>72637221</td>\n",
       "      <td>72638091</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151268</th>\n",
       "      <td>chr10</td>\n",
       "      <td>70007166</td>\n",
       "      <td>70007807</td>\n",
       "      <td>.</td>\n",
       "      <td>579</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151269</th>\n",
       "      <td>chr10</td>\n",
       "      <td>31871375</td>\n",
       "      <td>31872084</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151270 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629562     630131    .   1000      .     10.18118   \n",
       "1        chr1     629562     630131    .   1000      .      9.06746   \n",
       "2        chr1     633817     634262    .   1000      .      7.49633   \n",
       "3        chr1     633817     634262    .   1000      .      6.74865   \n",
       "4       chr17   22521226   22521666    .   1000      .     32.02649   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "151265   chr1  101174750  101175746    .    763      .      2.42782   \n",
       "151266  chr10   73076272   73077247    .    599      .      2.42782   \n",
       "151267  chr10   72637221   72638091    .    866      .      2.42782   \n",
       "151268  chr10   70007166   70007807    .    579      .      2.42782   \n",
       "151269  chr10   31871375   31872084    .    866      .      2.42782   \n",
       "\n",
       "            pValue      qValue  peak  \n",
       "0       5255.37842  5247.06348   139  \n",
       "1       4341.48682  4334.17773   321  \n",
       "2       3135.06812  3128.26001   119  \n",
       "3       2599.60767  2593.02832   305  \n",
       "4       2324.03174  2317.51929   132  \n",
       "...            ...         ...   ...  \n",
       "151265     3.35482     1.67457   572  \n",
       "151266     3.35482     1.67457    65  \n",
       "151267     3.35482     1.67457   710  \n",
       "151268     3.35482     1.67457   586  \n",
       "151269     3.35482     1.67457    58  \n",
       "\n",
       "[151270 rows x 10 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4ada0913-19b6-4530-9759-df5e068706c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac['index'] = MT_atac['Chr'] + \":\" + MT_atac['Start'].astype(str) + \"-\" + MT_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b47a673d-73d6-4efe-8a27-cbbea8e45e9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac['ATAC_MT'] = \"ATAC-seq:\" + MT_atac['Chr'] + \":\" + MT_atac['Start'].astype(str) + \"-\" + MT_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a959e645-7cd5-416a-8aa8-3a4030e4fa18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>10.18118</td>\n",
       "      <td>5255.37842</td>\n",
       "      <td>5247.06348</td>\n",
       "      <td>139</td>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>9.06746</td>\n",
       "      <td>4341.48682</td>\n",
       "      <td>4334.17773</td>\n",
       "      <td>321</td>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.49633</td>\n",
       "      <td>3135.06812</td>\n",
       "      <td>3128.26001</td>\n",
       "      <td>119</td>\n",
       "      <td>chr1:633817-634262</td>\n",
       "      <td>ATAC-seq:chr1:633817-634262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.74865</td>\n",
       "      <td>2599.60767</td>\n",
       "      <td>2593.02832</td>\n",
       "      <td>305</td>\n",
       "      <td>chr1:633817-634262</td>\n",
       "      <td>ATAC-seq:chr1:633817-634262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521226</td>\n",
       "      <td>22521666</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.02649</td>\n",
       "      <td>2324.03174</td>\n",
       "      <td>2317.51929</td>\n",
       "      <td>132</td>\n",
       "      <td>chr17:22521226-22521666</td>\n",
       "      <td>ATAC-seq:chr17:22521226-22521666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151265</th>\n",
       "      <td>chr1</td>\n",
       "      <td>101174750</td>\n",
       "      <td>101175746</td>\n",
       "      <td>.</td>\n",
       "      <td>763</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>572</td>\n",
       "      <td>chr1:101174750-101175746</td>\n",
       "      <td>ATAC-seq:chr1:101174750-101175746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151266</th>\n",
       "      <td>chr10</td>\n",
       "      <td>73076272</td>\n",
       "      <td>73077247</td>\n",
       "      <td>.</td>\n",
       "      <td>599</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>65</td>\n",
       "      <td>chr10:73076272-73077247</td>\n",
       "      <td>ATAC-seq:chr10:73076272-73077247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151267</th>\n",
       "      <td>chr10</td>\n",
       "      <td>72637221</td>\n",
       "      <td>72638091</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>710</td>\n",
       "      <td>chr10:72637221-72638091</td>\n",
       "      <td>ATAC-seq:chr10:72637221-72638091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151268</th>\n",
       "      <td>chr10</td>\n",
       "      <td>70007166</td>\n",
       "      <td>70007807</td>\n",
       "      <td>.</td>\n",
       "      <td>579</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>586</td>\n",
       "      <td>chr10:70007166-70007807</td>\n",
       "      <td>ATAC-seq:chr10:70007166-70007807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151269</th>\n",
       "      <td>chr10</td>\n",
       "      <td>31871375</td>\n",
       "      <td>31872084</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>58</td>\n",
       "      <td>chr10:31871375-31872084</td>\n",
       "      <td>ATAC-seq:chr10:31871375-31872084</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151270 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629562     630131    .   1000      .     10.18118   \n",
       "1        chr1     629562     630131    .   1000      .      9.06746   \n",
       "2        chr1     633817     634262    .   1000      .      7.49633   \n",
       "3        chr1     633817     634262    .   1000      .      6.74865   \n",
       "4       chr17   22521226   22521666    .   1000      .     32.02649   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "151265   chr1  101174750  101175746    .    763      .      2.42782   \n",
       "151266  chr10   73076272   73077247    .    599      .      2.42782   \n",
       "151267  chr10   72637221   72638091    .    866      .      2.42782   \n",
       "151268  chr10   70007166   70007807    .    579      .      2.42782   \n",
       "151269  chr10   31871375   31872084    .    866      .      2.42782   \n",
       "\n",
       "            pValue      qValue  peak                     index  \\\n",
       "0       5255.37842  5247.06348   139        chr1:629562-630131   \n",
       "1       4341.48682  4334.17773   321        chr1:629562-630131   \n",
       "2       3135.06812  3128.26001   119        chr1:633817-634262   \n",
       "3       2599.60767  2593.02832   305        chr1:633817-634262   \n",
       "4       2324.03174  2317.51929   132   chr17:22521226-22521666   \n",
       "...            ...         ...   ...                       ...   \n",
       "151265     3.35482     1.67457   572  chr1:101174750-101175746   \n",
       "151266     3.35482     1.67457    65   chr10:73076272-73077247   \n",
       "151267     3.35482     1.67457   710   chr10:72637221-72638091   \n",
       "151268     3.35482     1.67457   586   chr10:70007166-70007807   \n",
       "151269     3.35482     1.67457    58   chr10:31871375-31872084   \n",
       "\n",
       "                                  ATAC_MT  \n",
       "0             ATAC-seq:chr1:629562-630131  \n",
       "1             ATAC-seq:chr1:629562-630131  \n",
       "2             ATAC-seq:chr1:633817-634262  \n",
       "3             ATAC-seq:chr1:633817-634262  \n",
       "4        ATAC-seq:chr17:22521226-22521666  \n",
       "...                                   ...  \n",
       "151265  ATAC-seq:chr1:101174750-101175746  \n",
       "151266   ATAC-seq:chr10:73076272-73077247  \n",
       "151267   ATAC-seq:chr10:72637221-72638091  \n",
       "151268   ATAC-seq:chr10:70007166-70007807  \n",
       "151269   ATAC-seq:chr10:31871375-31872084  \n",
       "\n",
       "[151270 rows x 12 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b487e865-8aa9-4ee8-845a-c13884bcbaa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac_uniq = MT_atac.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "36b1f4fa-0cfc-463c-baf0-fd21451af271",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>10.18118</td>\n",
       "      <td>5255.37842</td>\n",
       "      <td>5247.06348</td>\n",
       "      <td>139</td>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.49633</td>\n",
       "      <td>3135.06812</td>\n",
       "      <td>3128.26001</td>\n",
       "      <td>119</td>\n",
       "      <td>chr1:633817-634262</td>\n",
       "      <td>ATAC-seq:chr1:633817-634262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521226</td>\n",
       "      <td>22521666</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.02649</td>\n",
       "      <td>2324.03174</td>\n",
       "      <td>2317.51929</td>\n",
       "      <td>132</td>\n",
       "      <td>chr17:22521226-22521666</td>\n",
       "      <td>ATAC-seq:chr17:22521226-22521666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr8</td>\n",
       "      <td>109333452</td>\n",
       "      <td>109334794</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>15.83920</td>\n",
       "      <td>633.10614</td>\n",
       "      <td>626.95770</td>\n",
       "      <td>851</td>\n",
       "      <td>chr8:109333452-109334794</td>\n",
       "      <td>ATAC-seq:chr8:109333452-109334794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169893202</td>\n",
       "      <td>169894709</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>17.79785</td>\n",
       "      <td>621.90320</td>\n",
       "      <td>615.77106</td>\n",
       "      <td>778</td>\n",
       "      <td>chr1:169893202-169894709</td>\n",
       "      <td>ATAC-seq:chr1:169893202-169894709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150395</th>\n",
       "      <td>chr2</td>\n",
       "      <td>221518222</td>\n",
       "      <td>221518503</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>2.01126</td>\n",
       "      <td>3.46056</td>\n",
       "      <td>1.77423</td>\n",
       "      <td>198</td>\n",
       "      <td>chr2:221518222-221518503</td>\n",
       "      <td>ATAC-seq:chr2:221518222-221518503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150671</th>\n",
       "      <td>chr1</td>\n",
       "      <td>177741234</td>\n",
       "      <td>177741384</td>\n",
       "      <td>.</td>\n",
       "      <td>603</td>\n",
       "      <td>.</td>\n",
       "      <td>2.15866</td>\n",
       "      <td>3.42075</td>\n",
       "      <td>1.73678</td>\n",
       "      <td>73</td>\n",
       "      <td>chr1:177741234-177741384</td>\n",
       "      <td>ATAC-seq:chr1:177741234-177741384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150742</th>\n",
       "      <td>chr11</td>\n",
       "      <td>108958462</td>\n",
       "      <td>108958678</td>\n",
       "      <td>.</td>\n",
       "      <td>741</td>\n",
       "      <td>.</td>\n",
       "      <td>2.33577</td>\n",
       "      <td>3.40943</td>\n",
       "      <td>1.72609</td>\n",
       "      <td>210</td>\n",
       "      <td>chr11:108958462-108958678</td>\n",
       "      <td>ATAC-seq:chr11:108958462-108958678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150764</th>\n",
       "      <td>chr11</td>\n",
       "      <td>64688141</td>\n",
       "      <td>64688293</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>1.79671</td>\n",
       "      <td>3.40386</td>\n",
       "      <td>1.72077</td>\n",
       "      <td>69</td>\n",
       "      <td>chr11:64688141-64688293</td>\n",
       "      <td>ATAC-seq:chr11:64688141-64688293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150802</th>\n",
       "      <td>chr11</td>\n",
       "      <td>109034161</td>\n",
       "      <td>109034311</td>\n",
       "      <td>.</td>\n",
       "      <td>549</td>\n",
       "      <td>.</td>\n",
       "      <td>2.33066</td>\n",
       "      <td>3.39760</td>\n",
       "      <td>1.71488</td>\n",
       "      <td>91</td>\n",
       "      <td>chr11:109034161-109034311</td>\n",
       "      <td>ATAC-seq:chr11:109034161-109034311</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52593 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629562     630131    .   1000      .     10.18118   \n",
       "2        chr1     633817     634262    .   1000      .      7.49633   \n",
       "4       chr17   22521226   22521666    .   1000      .     32.02649   \n",
       "6        chr8  109333452  109334794    .   1000      .     15.83920   \n",
       "7        chr1  169893202  169894709    .   1000      .     17.79785   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "150395   chr2  221518222  221518503    .   1000      .      2.01126   \n",
       "150671   chr1  177741234  177741384    .    603      .      2.15866   \n",
       "150742  chr11  108958462  108958678    .    741      .      2.33577   \n",
       "150764  chr11   64688141   64688293    .   1000      .      1.79671   \n",
       "150802  chr11  109034161  109034311    .    549      .      2.33066   \n",
       "\n",
       "            pValue      qValue  peak                      index  \\\n",
       "0       5255.37842  5247.06348   139         chr1:629562-630131   \n",
       "2       3135.06812  3128.26001   119         chr1:633817-634262   \n",
       "4       2324.03174  2317.51929   132    chr17:22521226-22521666   \n",
       "6        633.10614   626.95770   851   chr8:109333452-109334794   \n",
       "7        621.90320   615.77106   778   chr1:169893202-169894709   \n",
       "...            ...         ...   ...                        ...   \n",
       "150395     3.46056     1.77423   198   chr2:221518222-221518503   \n",
       "150671     3.42075     1.73678    73   chr1:177741234-177741384   \n",
       "150742     3.40943     1.72609   210  chr11:108958462-108958678   \n",
       "150764     3.40386     1.72077    69    chr11:64688141-64688293   \n",
       "150802     3.39760     1.71488    91  chr11:109034161-109034311   \n",
       "\n",
       "                                   ATAC_MT  \n",
       "0              ATAC-seq:chr1:629562-630131  \n",
       "2              ATAC-seq:chr1:633817-634262  \n",
       "4         ATAC-seq:chr17:22521226-22521666  \n",
       "6        ATAC-seq:chr8:109333452-109334794  \n",
       "7        ATAC-seq:chr1:169893202-169894709  \n",
       "...                                    ...  \n",
       "150395   ATAC-seq:chr2:221518222-221518503  \n",
       "150671   ATAC-seq:chr1:177741234-177741384  \n",
       "150742  ATAC-seq:chr11:108958462-108958678  \n",
       "150764    ATAC-seq:chr11:64688141-64688293  \n",
       "150802  ATAC-seq:chr11:109034161-109034311  \n",
       "\n",
       "[52593 rows x 12 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_atac_uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6537331d-3c8a-46c0-828f-d5cb5edcd06e",
   "metadata": {},
   "source": [
    "# MT DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "a4a0ed79-75f5-4e2f-90ea-783256335851",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase = pd.read_csv('/mnt/10T-backup-mingju/ifar/DNase_seq/myotube/ENCFF222PWD.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d1431f3c-7a81-4fda-a912-8329de7365b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>11484.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>34.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633528</td>\n",
       "      <td>633678</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>12252.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87053</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26320155</td>\n",
       "      <td>26320305</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87054</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>20.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87055</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87056</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>135.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87057</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>26.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87058 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0      chr1    181415    181565    .      0      .         30.0      -1   \n",
       "1      chr1    267990    268140    .      0      .         22.0      -1   \n",
       "2      chr1    629870    630020    .      0      .      11484.0      -1   \n",
       "3      chr1    630598    630748    .      0      .         34.0      -1   \n",
       "4      chr1    633528    633678    .      0      .      12252.0      -1   \n",
       "...     ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "87053  chrY  26320155  26320305    .      0      .         22.0      -1   \n",
       "87054  chrY  26359975  26360125    .      0      .         20.0      -1   \n",
       "87055  chrY  56727995  56728145    .      0      .         15.0      -1   \n",
       "87056  chrY  56734800  56734950    .      0      .        135.0      -1   \n",
       "87057  chrY  56872355  56872505    .      0      .         26.0      -1   \n",
       "\n",
       "       qValue  peak  \n",
       "0          -1    75  \n",
       "1          -1    75  \n",
       "2          -1    75  \n",
       "3          -1    75  \n",
       "4          -1    75  \n",
       "...       ...   ...  \n",
       "87053      -1    75  \n",
       "87054      -1    75  \n",
       "87055      -1    75  \n",
       "87056      -1    75  \n",
       "87057      -1    75  \n",
       "\n",
       "[87058 rows x 10 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f7f2d258-eb5a-4f7b-a1fc-91482002c5a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase['index'] = MT_DNase['Chr'] + \":\" + MT_DNase['Start'].astype(str) + \"-\" + MT_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a99223c6-f504-42ce-830e-62fdac815009",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase['DNase_MT'] = \"DNase-seq:\" + MT_DNase['Chr'] + \":\" + MT_DNase['Start'].astype(str) + \"-\" + MT_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "390bd32f-5930-44f0-9a61-c518c16855b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>11484.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>34.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633528</td>\n",
       "      <td>633678</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>12252.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:633528-633678</td>\n",
       "      <td>DNase-seq:chr1:633528-633678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87053</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26320155</td>\n",
       "      <td>26320305</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26320155-26320305</td>\n",
       "      <td>DNase-seq:chrY:26320155-26320305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87054</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>20.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87055</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87056</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>135.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87057</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>26.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87058 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0      chr1    181415    181565    .      0      .         30.0      -1   \n",
       "1      chr1    267990    268140    .      0      .         22.0      -1   \n",
       "2      chr1    629870    630020    .      0      .      11484.0      -1   \n",
       "3      chr1    630598    630748    .      0      .         34.0      -1   \n",
       "4      chr1    633528    633678    .      0      .      12252.0      -1   \n",
       "...     ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "87053  chrY  26320155  26320305    .      0      .         22.0      -1   \n",
       "87054  chrY  26359975  26360125    .      0      .         20.0      -1   \n",
       "87055  chrY  56727995  56728145    .      0      .         15.0      -1   \n",
       "87056  chrY  56734800  56734950    .      0      .        135.0      -1   \n",
       "87057  chrY  56872355  56872505    .      0      .         26.0      -1   \n",
       "\n",
       "       qValue  peak                   index                          DNase_MT  \n",
       "0          -1    75      chr1:181415-181565      DNase-seq:chr1:181415-181565  \n",
       "1          -1    75      chr1:267990-268140      DNase-seq:chr1:267990-268140  \n",
       "2          -1    75      chr1:629870-630020      DNase-seq:chr1:629870-630020  \n",
       "3          -1    75      chr1:630598-630748      DNase-seq:chr1:630598-630748  \n",
       "4          -1    75      chr1:633528-633678      DNase-seq:chr1:633528-633678  \n",
       "...       ...   ...                     ...                               ...  \n",
       "87053      -1    75  chrY:26320155-26320305  DNase-seq:chrY:26320155-26320305  \n",
       "87054      -1    75  chrY:26359975-26360125  DNase-seq:chrY:26359975-26360125  \n",
       "87055      -1    75  chrY:56727995-56728145  DNase-seq:chrY:56727995-56728145  \n",
       "87056      -1    75  chrY:56734800-56734950  DNase-seq:chrY:56734800-56734950  \n",
       "87057      -1    75  chrY:56872355-56872505  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[87058 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "118add04-3a86-4a9c-a938-e4e1c1923012",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase_uniq = MT_DNase.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "bcd28bc9-2d1d-4792-9c5c-ce380db5e243",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>11484.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>34.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633528</td>\n",
       "      <td>633678</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>12252.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:633528-633678</td>\n",
       "      <td>DNase-seq:chr1:633528-633678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87053</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26320155</td>\n",
       "      <td>26320305</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26320155-26320305</td>\n",
       "      <td>DNase-seq:chrY:26320155-26320305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87054</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>20.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87055</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87056</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>135.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87057</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>26.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87058 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0      chr1    181415    181565    .      0      .         30.0      -1   \n",
       "1      chr1    267990    268140    .      0      .         22.0      -1   \n",
       "2      chr1    629870    630020    .      0      .      11484.0      -1   \n",
       "3      chr1    630598    630748    .      0      .         34.0      -1   \n",
       "4      chr1    633528    633678    .      0      .      12252.0      -1   \n",
       "...     ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "87053  chrY  26320155  26320305    .      0      .         22.0      -1   \n",
       "87054  chrY  26359975  26360125    .      0      .         20.0      -1   \n",
       "87055  chrY  56727995  56728145    .      0      .         15.0      -1   \n",
       "87056  chrY  56734800  56734950    .      0      .        135.0      -1   \n",
       "87057  chrY  56872355  56872505    .      0      .         26.0      -1   \n",
       "\n",
       "       qValue  peak                   index                          DNase_MT  \n",
       "0          -1    75      chr1:181415-181565      DNase-seq:chr1:181415-181565  \n",
       "1          -1    75      chr1:267990-268140      DNase-seq:chr1:267990-268140  \n",
       "2          -1    75      chr1:629870-630020      DNase-seq:chr1:629870-630020  \n",
       "3          -1    75      chr1:630598-630748      DNase-seq:chr1:630598-630748  \n",
       "4          -1    75      chr1:633528-633678      DNase-seq:chr1:633528-633678  \n",
       "...       ...   ...                     ...                               ...  \n",
       "87053      -1    75  chrY:26320155-26320305  DNase-seq:chrY:26320155-26320305  \n",
       "87054      -1    75  chrY:26359975-26360125  DNase-seq:chrY:26359975-26360125  \n",
       "87055      -1    75  chrY:56727995-56728145  DNase-seq:chrY:56727995-56728145  \n",
       "87056      -1    75  chrY:56734800-56734950  DNase-seq:chrY:56734800-56734950  \n",
       "87057      -1    75  chrY:56872355-56872505  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[87058 rows x 12 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_DNase_uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "38dd58db-1c4f-4810-8458-037b15ea7ba7",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase = pd.merge(MT_atac_uniq[['index','Chr','Start','End','ATAC_MT']], MT_DNase_uniq[['index','DNase_MT']], on='index', how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "eed0c8d6-2f54-4b61-be73-65299bc61f30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006395-100006545</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100009208-100010573</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100009208.0</td>\n",
       "      <td>100010573.0</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100025224-100026486</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100025224.0</td>\n",
       "      <td>100026486.0</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100043466-100044717</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100043466.0</td>\n",
       "      <td>100044717.0</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100043915-100044065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:9028075-9028225</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:9160678-9161527</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9160678.0</td>\n",
       "      <td>9161527.0</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:9177772-9177922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:9259495-9259645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:9278995-9279145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index    Chr        Start          End  \\\n",
       "0       chr10:100006395-100006545    NaN          NaN          NaN   \n",
       "1       chr10:100009208-100010573  chr10  100009208.0  100010573.0   \n",
       "2       chr10:100025224-100026486  chr10  100025224.0  100026486.0   \n",
       "3       chr10:100043466-100044717  chr10  100043466.0  100044717.0   \n",
       "4       chr10:100043915-100044065    NaN          NaN          NaN   \n",
       "...                           ...    ...          ...          ...   \n",
       "139646       chrY:9028075-9028225    NaN          NaN          NaN   \n",
       "139647       chrY:9160678-9161527   chrY    9160678.0    9161527.0   \n",
       "139648       chrY:9177772-9177922    NaN          NaN          NaN   \n",
       "139649       chrY:9259495-9259645    NaN          NaN          NaN   \n",
       "139650       chrY:9278995-9279145    NaN          NaN          NaN   \n",
       "\n",
       "                                   ATAC_MT  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100009208-100010573   \n",
       "2       ATAC-seq:chr10:100025224-100026486   \n",
       "3       ATAC-seq:chr10:100043466-100044717   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "139646                                 NaN   \n",
       "139647       ATAC-seq:chrY:9160678-9161527   \n",
       "139648                                 NaN   \n",
       "139649                                 NaN   \n",
       "139650                                 NaN   \n",
       "\n",
       "                                   DNase_MT  \n",
       "0       DNase-seq:chr10:100006395-100006545  \n",
       "1                                       NaN  \n",
       "2                                       NaN  \n",
       "3                                       NaN  \n",
       "4       DNase-seq:chr10:100043915-100044065  \n",
       "...                                     ...  \n",
       "139646       DNase-seq:chrY:9028075-9028225  \n",
       "139647                                  NaN  \n",
       "139648       DNase-seq:chrY:9177772-9177922  \n",
       "139649       DNase-seq:chrY:9259495-9259645  \n",
       "139650       DNase-seq:chrY:9278995-9279145  \n",
       "\n",
       "[139651 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a9c9b6be-5b84-4c50-9956-fc78965b93ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[\"OpenChromatin\"] = merge_atac_dnase.apply(\n",
    "    lambda row: f\"{row['ATAC_MT']};{row['DNase_MT']}\" if pd.notna(row['ATAC_MT']) and pd.notna(row['DNase_MT']) and row['DNase_MT'] != \"\"\n",
    "                else row['ATAC_MT'] if pd.notna(row['ATAC_MT'])\n",
    "                else row['DNase_MT'] if pd.notna(row['DNase_MT']) and row['DNase_MT'] != \"\"\n",
    "                else \"\", \n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f664675c-83a1-4733-b731-8cd4733396a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006395-100006545</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100009208-100010573</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100009208.0</td>\n",
       "      <td>100010573.0</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100025224-100026486</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100025224.0</td>\n",
       "      <td>100026486.0</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100043466-100044717</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100043466.0</td>\n",
       "      <td>100044717.0</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100043915-100044065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:9028075-9028225</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:9160678-9161527</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9160678.0</td>\n",
       "      <td>9161527.0</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:9177772-9177922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:9259495-9259645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:9278995-9279145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index    Chr        Start          End  \\\n",
       "0       chr10:100006395-100006545    NaN          NaN          NaN   \n",
       "1       chr10:100009208-100010573  chr10  100009208.0  100010573.0   \n",
       "2       chr10:100025224-100026486  chr10  100025224.0  100026486.0   \n",
       "3       chr10:100043466-100044717  chr10  100043466.0  100044717.0   \n",
       "4       chr10:100043915-100044065    NaN          NaN          NaN   \n",
       "...                           ...    ...          ...          ...   \n",
       "139646       chrY:9028075-9028225    NaN          NaN          NaN   \n",
       "139647       chrY:9160678-9161527   chrY    9160678.0    9161527.0   \n",
       "139648       chrY:9177772-9177922    NaN          NaN          NaN   \n",
       "139649       chrY:9259495-9259645    NaN          NaN          NaN   \n",
       "139650       chrY:9278995-9279145    NaN          NaN          NaN   \n",
       "\n",
       "                                   ATAC_MT  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100009208-100010573   \n",
       "2       ATAC-seq:chr10:100025224-100026486   \n",
       "3       ATAC-seq:chr10:100043466-100044717   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "139646                                 NaN   \n",
       "139647       ATAC-seq:chrY:9160678-9161527   \n",
       "139648                                 NaN   \n",
       "139649                                 NaN   \n",
       "139650                                 NaN   \n",
       "\n",
       "                                   DNase_MT  \\\n",
       "0       DNase-seq:chr10:100006395-100006545   \n",
       "1                                       NaN   \n",
       "2                                       NaN   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100043915-100044065   \n",
       "...                                     ...   \n",
       "139646       DNase-seq:chrY:9028075-9028225   \n",
       "139647                                  NaN   \n",
       "139648       DNase-seq:chrY:9177772-9177922   \n",
       "139649       DNase-seq:chrY:9259495-9259645   \n",
       "139650       DNase-seq:chrY:9278995-9279145   \n",
       "\n",
       "                              OpenChromatin  \n",
       "0       DNase-seq:chr10:100006395-100006545  \n",
       "1        ATAC-seq:chr10:100009208-100010573  \n",
       "2        ATAC-seq:chr10:100025224-100026486  \n",
       "3        ATAC-seq:chr10:100043466-100044717  \n",
       "4       DNase-seq:chr10:100043915-100044065  \n",
       "...                                     ...  \n",
       "139646       DNase-seq:chrY:9028075-9028225  \n",
       "139647        ATAC-seq:chrY:9160678-9161527  \n",
       "139648       DNase-seq:chrY:9177772-9177922  \n",
       "139649       DNase-seq:chrY:9259495-9259645  \n",
       "139650       DNase-seq:chrY:9278995-9279145  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a243595e-64d5-41d6-986d-716aa8b71758",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_index(index_str):\n",
    "    if pd.isna(index_str) or index_str == \"NaN\":\n",
    "        return pd.NA, pd.NA, pd.NA\n",
    "    chr_part, range_part = index_str.split(\":\")\n",
    "    chr_value = chr_part.replace(\"chr\", \"\")\n",
    "    start, end = range_part.split(\"-\")\n",
    "    return chr_value, pd.to_numeric(start), pd.to_numeric(end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9367a17f-4410-4afa-9460-3a9d80d175ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[['Chr', 'Start', 'End']] = merge_atac_dnase['index'].apply(lambda x: pd.Series(parse_index(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "ae373cf5-1cd5-4c34-a158-988fcc95b7af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006395-100006545</td>\n",
       "      <td>10</td>\n",
       "      <td>100006395</td>\n",
       "      <td>100006545</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100009208-100010573</td>\n",
       "      <td>10</td>\n",
       "      <td>100009208</td>\n",
       "      <td>100010573</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100025224-100026486</td>\n",
       "      <td>10</td>\n",
       "      <td>100025224</td>\n",
       "      <td>100026486</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100043466-100044717</td>\n",
       "      <td>10</td>\n",
       "      <td>100043466</td>\n",
       "      <td>100044717</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100043915-100044065</td>\n",
       "      <td>10</td>\n",
       "      <td>100043915</td>\n",
       "      <td>100044065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:9028075-9028225</td>\n",
       "      <td>Y</td>\n",
       "      <td>9028075</td>\n",
       "      <td>9028225</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:9160678-9161527</td>\n",
       "      <td>Y</td>\n",
       "      <td>9160678</td>\n",
       "      <td>9161527</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:9177772-9177922</td>\n",
       "      <td>Y</td>\n",
       "      <td>9177772</td>\n",
       "      <td>9177922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:9259495-9259645</td>\n",
       "      <td>Y</td>\n",
       "      <td>9259495</td>\n",
       "      <td>9259645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:9278995-9279145</td>\n",
       "      <td>Y</td>\n",
       "      <td>9278995</td>\n",
       "      <td>9279145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index Chr      Start        End  \\\n",
       "0       chr10:100006395-100006545  10  100006395  100006545   \n",
       "1       chr10:100009208-100010573  10  100009208  100010573   \n",
       "2       chr10:100025224-100026486  10  100025224  100026486   \n",
       "3       chr10:100043466-100044717  10  100043466  100044717   \n",
       "4       chr10:100043915-100044065  10  100043915  100044065   \n",
       "...                           ...  ..        ...        ...   \n",
       "139646       chrY:9028075-9028225   Y    9028075    9028225   \n",
       "139647       chrY:9160678-9161527   Y    9160678    9161527   \n",
       "139648       chrY:9177772-9177922   Y    9177772    9177922   \n",
       "139649       chrY:9259495-9259645   Y    9259495    9259645   \n",
       "139650       chrY:9278995-9279145   Y    9278995    9279145   \n",
       "\n",
       "                                   ATAC_MT  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100009208-100010573   \n",
       "2       ATAC-seq:chr10:100025224-100026486   \n",
       "3       ATAC-seq:chr10:100043466-100044717   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "139646                                 NaN   \n",
       "139647       ATAC-seq:chrY:9160678-9161527   \n",
       "139648                                 NaN   \n",
       "139649                                 NaN   \n",
       "139650                                 NaN   \n",
       "\n",
       "                                   DNase_MT  \\\n",
       "0       DNase-seq:chr10:100006395-100006545   \n",
       "1                                       NaN   \n",
       "2                                       NaN   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100043915-100044065   \n",
       "...                                     ...   \n",
       "139646       DNase-seq:chrY:9028075-9028225   \n",
       "139647                                  NaN   \n",
       "139648       DNase-seq:chrY:9177772-9177922   \n",
       "139649       DNase-seq:chrY:9259495-9259645   \n",
       "139650       DNase-seq:chrY:9278995-9279145   \n",
       "\n",
       "                              OpenChromatin  \n",
       "0       DNase-seq:chr10:100006395-100006545  \n",
       "1        ATAC-seq:chr10:100009208-100010573  \n",
       "2        ATAC-seq:chr10:100025224-100026486  \n",
       "3        ATAC-seq:chr10:100043466-100044717  \n",
       "4       DNase-seq:chr10:100043915-100044065  \n",
       "...                                     ...  \n",
       "139646       DNase-seq:chrY:9028075-9028225  \n",
       "139647        ATAC-seq:chrY:9160678-9161527  \n",
       "139648       DNase-seq:chrY:9177772-9177922  \n",
       "139649       DNase-seq:chrY:9259495-9259645  \n",
       "139650       DNase-seq:chrY:9278995-9279145  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "9a58dab4-e981-4bd8-8f9d-7cfc07d5a5cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the chromosome order: 1 to 22, X, Y\n",
    "chr_order = [str(i) for i in range(1, 23)] + [\"X\", \"Y\"]\n",
    "\n",
    "# Convert Chr to categorical with the specified order\n",
    "merge_atac_dnase[\"Chr\"] = pd.Categorical(merge_atac_dnase[\"Chr\"], categories=chr_order, ordered=True)\n",
    "\n",
    "# Sort by Chr and Start\n",
    "merge_atac_dnase_sorted = merge_atac_dnase.sort_values(by=[\"Chr\", \"Start\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ea650c3c-46eb-401b-8b54-f7d0d60cedec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>Y</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:26365821-26366654</td>\n",
       "      <td>Y</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>Y</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>Y</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>Y</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End  \\\n",
       "0           chr1:181415-181565   1    181415    181565   \n",
       "1           chr1:267990-268140   1    267990    268140   \n",
       "2           chr1:629562-630131   1    629562    630131   \n",
       "3           chr1:629870-630020   1    629870    630020   \n",
       "4           chr1:630598-630748   1    630598    630748   \n",
       "...                        ...  ..       ...       ...   \n",
       "139646  chrY:26359975-26360125   Y  26359975  26360125   \n",
       "139647  chrY:26365821-26366654   Y  26365821  26366654   \n",
       "139648  chrY:56727995-56728145   Y  56727995  56728145   \n",
       "139649  chrY:56734800-56734950   Y  56734800  56734950   \n",
       "139650  chrY:56872355-56872505   Y  56872355  56872505   \n",
       "\n",
       "                                ATAC_MT                          DNase_MT  \\\n",
       "0                                   NaN      DNase-seq:chr1:181415-181565   \n",
       "1                                   NaN      DNase-seq:chr1:267990-268140   \n",
       "2           ATAC-seq:chr1:629562-630131                               NaN   \n",
       "3                                   NaN      DNase-seq:chr1:629870-630020   \n",
       "4                                   NaN      DNase-seq:chr1:630598-630748   \n",
       "...                                 ...                               ...   \n",
       "139646                              NaN  DNase-seq:chrY:26359975-26360125   \n",
       "139647  ATAC-seq:chrY:26365821-26366654                               NaN   \n",
       "139648                              NaN  DNase-seq:chrY:56727995-56728145   \n",
       "139649                              NaN  DNase-seq:chrY:56734800-56734950   \n",
       "139650                              NaN  DNase-seq:chrY:56872355-56872505   \n",
       "\n",
       "                           OpenChromatin  \n",
       "0           DNase-seq:chr1:181415-181565  \n",
       "1           DNase-seq:chr1:267990-268140  \n",
       "2            ATAC-seq:chr1:629562-630131  \n",
       "3           DNase-seq:chr1:629870-630020  \n",
       "4           DNase-seq:chr1:630598-630748  \n",
       "...                                  ...  \n",
       "139646  DNase-seq:chrY:26359975-26360125  \n",
       "139647   ATAC-seq:chrY:26365821-26366654  \n",
       "139648  DNase-seq:chrY:56727995-56728145  \n",
       "139649  DNase-seq:chrY:56734800-56734950  \n",
       "139650  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5743febe-5606-483e-a92f-025632971f88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52686</th>\n",
       "      <td>chr1:1511955-1512105</td>\n",
       "      <td>1</td>\n",
       "      <td>1511955</td>\n",
       "      <td>1512105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1511955-1512105</td>\n",
       "      <td>DNase-seq:chr1:1511955-1512105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52920</th>\n",
       "      <td>chr1:1540281-1541159</td>\n",
       "      <td>1</td>\n",
       "      <td>1540281</td>\n",
       "      <td>1541159</td>\n",
       "      <td>ATAC-seq:chr1:1540281-1541159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1540281-1541159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53046</th>\n",
       "      <td>chr1:1551073-1551223</td>\n",
       "      <td>1</td>\n",
       "      <td>1551073</td>\n",
       "      <td>1551223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1551073-1551223</td>\n",
       "      <td>DNase-seq:chr1:1551073-1551223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53100</th>\n",
       "      <td>chr1:1554491-1555433</td>\n",
       "      <td>1</td>\n",
       "      <td>1554491</td>\n",
       "      <td>1555433</td>\n",
       "      <td>ATAC-seq:chr1:1554491-1555433</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1554491-1555433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53257</th>\n",
       "      <td>chr1:1567475-1567625</td>\n",
       "      <td>1</td>\n",
       "      <td>1567475</td>\n",
       "      <td>1567625</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1567475-1567625</td>\n",
       "      <td>DNase-seq:chr1:1567475-1567625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      index Chr    Start      End  \\\n",
       "54899    chr1:181415-181565   1   181415   181565   \n",
       "59412    chr1:267990-268140   1   267990   268140   \n",
       "62036    chr1:629562-630131   1   629562   630131   \n",
       "62038    chr1:629870-630020   1   629870   630020   \n",
       "62041    chr1:630598-630748   1   630598   630748   \n",
       "...                     ...  ..      ...      ...   \n",
       "52686  chr1:1511955-1512105   1  1511955  1512105   \n",
       "52920  chr1:1540281-1541159   1  1540281  1541159   \n",
       "53046  chr1:1551073-1551223   1  1551073  1551223   \n",
       "53100  chr1:1554491-1555433   1  1554491  1555433   \n",
       "53257  chr1:1567475-1567625   1  1567475  1567625   \n",
       "\n",
       "                             ATAC_MT                        DNase_MT  \\\n",
       "54899                            NaN    DNase-seq:chr1:181415-181565   \n",
       "59412                            NaN    DNase-seq:chr1:267990-268140   \n",
       "62036    ATAC-seq:chr1:629562-630131                             NaN   \n",
       "62038                            NaN    DNase-seq:chr1:629870-630020   \n",
       "62041                            NaN    DNase-seq:chr1:630598-630748   \n",
       "...                              ...                             ...   \n",
       "52686                            NaN  DNase-seq:chr1:1511955-1512105   \n",
       "52920  ATAC-seq:chr1:1540281-1541159                             NaN   \n",
       "53046                            NaN  DNase-seq:chr1:1551073-1551223   \n",
       "53100  ATAC-seq:chr1:1554491-1555433                             NaN   \n",
       "53257                            NaN  DNase-seq:chr1:1567475-1567625   \n",
       "\n",
       "                        OpenChromatin  \n",
       "54899    DNase-seq:chr1:181415-181565  \n",
       "59412    DNase-seq:chr1:267990-268140  \n",
       "62036     ATAC-seq:chr1:629562-630131  \n",
       "62038    DNase-seq:chr1:629870-630020  \n",
       "62041    DNase-seq:chr1:630598-630748  \n",
       "...                               ...  \n",
       "52686  DNase-seq:chr1:1511955-1512105  \n",
       "52920   ATAC-seq:chr1:1540281-1541159  \n",
       "53046  DNase-seq:chr1:1551073-1551223  \n",
       "53100   ATAC-seq:chr1:1554491-1555433  \n",
       "53257  DNase-seq:chr1:1567475-1567625  \n",
       "\n",
       "[100 rows x 7 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.head(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2f08790f-b04c-4444-ad34-a91ca893ace2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [index, Chr, Start, End, ATAC_MT, DNase_MT, OpenChromatin]\n",
       "Index: []"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted[merge_atac_dnase_sorted['ATAC_MT'].notna()&merge_atac_dnase_sorted['DNase_MT'].notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6e472aa7-b594-416b-9ab3-c8ddad47c265",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139576</th>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>Y</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139577</th>\n",
       "      <td>chrY:26365821-26366654</td>\n",
       "      <td>Y</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139588</th>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>Y</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139589</th>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>Y</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139590</th>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>Y</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End  \\\n",
       "54899       chr1:181415-181565   1    181415    181565   \n",
       "59412       chr1:267990-268140   1    267990    268140   \n",
       "62036       chr1:629562-630131   1    629562    630131   \n",
       "62038       chr1:629870-630020   1    629870    630020   \n",
       "62041       chr1:630598-630748   1    630598    630748   \n",
       "...                        ...  ..       ...       ...   \n",
       "139576  chrY:26359975-26360125   Y  26359975  26360125   \n",
       "139577  chrY:26365821-26366654   Y  26365821  26366654   \n",
       "139588  chrY:56727995-56728145   Y  56727995  56728145   \n",
       "139589  chrY:56734800-56734950   Y  56734800  56734950   \n",
       "139590  chrY:56872355-56872505   Y  56872355  56872505   \n",
       "\n",
       "                                ATAC_MT                          DNase_MT  \\\n",
       "54899                               NaN      DNase-seq:chr1:181415-181565   \n",
       "59412                               NaN      DNase-seq:chr1:267990-268140   \n",
       "62036       ATAC-seq:chr1:629562-630131                               NaN   \n",
       "62038                               NaN      DNase-seq:chr1:629870-630020   \n",
       "62041                               NaN      DNase-seq:chr1:630598-630748   \n",
       "...                                 ...                               ...   \n",
       "139576                              NaN  DNase-seq:chrY:26359975-26360125   \n",
       "139577  ATAC-seq:chrY:26365821-26366654                               NaN   \n",
       "139588                              NaN  DNase-seq:chrY:56727995-56728145   \n",
       "139589                              NaN  DNase-seq:chrY:56734800-56734950   \n",
       "139590                              NaN  DNase-seq:chrY:56872355-56872505   \n",
       "\n",
       "                           OpenChromatin  \n",
       "54899       DNase-seq:chr1:181415-181565  \n",
       "59412       DNase-seq:chr1:267990-268140  \n",
       "62036        ATAC-seq:chr1:629562-630131  \n",
       "62038       DNase-seq:chr1:629870-630020  \n",
       "62041       DNase-seq:chr1:630598-630748  \n",
       "...                                  ...  \n",
       "139576  DNase-seq:chrY:26359975-26360125  \n",
       "139577   ATAC-seq:chrY:26365821-26366654  \n",
       "139588  DNase-seq:chrY:56727995-56728145  \n",
       "139589  DNase-seq:chrY:56734800-56734950  \n",
       "139590  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "70d2b8f7-5097-4a33-a80d-97ccf4b65369",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_merge_atac_dnase_sorted_bed = merge_atac_dnase_sorted[[\"Chr\",\"Start\",\"End\",\"OpenChromatin\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "cb43fd4d-d539-48af-9397-de97ee407b8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139576</th>\n",
       "      <td>Y</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139577</th>\n",
       "      <td>Y</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139588</th>\n",
       "      <td>Y</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139589</th>\n",
       "      <td>Y</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139590</th>\n",
       "      <td>Y</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End                     OpenChromatin\n",
       "54899    1    181415    181565      DNase-seq:chr1:181415-181565\n",
       "59412    1    267990    268140      DNase-seq:chr1:267990-268140\n",
       "62036    1    629562    630131       ATAC-seq:chr1:629562-630131\n",
       "62038    1    629870    630020      DNase-seq:chr1:629870-630020\n",
       "62041    1    630598    630748      DNase-seq:chr1:630598-630748\n",
       "...     ..       ...       ...                               ...\n",
       "139576   Y  26359975  26360125  DNase-seq:chrY:26359975-26360125\n",
       "139577   Y  26365821  26366654   ATAC-seq:chrY:26365821-26366654\n",
       "139588   Y  56727995  56728145  DNase-seq:chrY:56727995-56728145\n",
       "139589   Y  56734800  56734950  DNase-seq:chrY:56734800-56734950\n",
       "139590   Y  56872355  56872505  DNase-seq:chrY:56872355-56872505\n",
       "\n",
       "[139651 rows x 4 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "298a4b03-0774-44a3-9c4c-2d0a5e04a25c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_4105959/3441541605.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  MT_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + MT_merge_atac_dnase_sorted_bed['Chr'].astype(str)\n"
     ]
    }
   ],
   "source": [
    "MT_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + MT_merge_atac_dnase_sorted_bed['Chr'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "1ca0b26a-0d6c-473c-8b62-56054a166587",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139576</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139577</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139588</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139589</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139590</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End                     OpenChromatin\n",
       "54899   chr1    181415    181565      DNase-seq:chr1:181415-181565\n",
       "59412   chr1    267990    268140      DNase-seq:chr1:267990-268140\n",
       "62036   chr1    629562    630131       ATAC-seq:chr1:629562-630131\n",
       "62038   chr1    629870    630020      DNase-seq:chr1:629870-630020\n",
       "62041   chr1    630598    630748      DNase-seq:chr1:630598-630748\n",
       "...      ...       ...       ...                               ...\n",
       "139576  chrY  26359975  26360125  DNase-seq:chrY:26359975-26360125\n",
       "139577  chrY  26365821  26366654   ATAC-seq:chrY:26365821-26366654\n",
       "139588  chrY  56727995  56728145  DNase-seq:chrY:56727995-56728145\n",
       "139589  chrY  56734800  56734950  DNase-seq:chrY:56734800-56734950\n",
       "139590  chrY  56872355  56872505  DNase-seq:chrY:56872355-56872505\n",
       "\n",
       "[139651 rows x 4 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "96f1f652-46bd-4e89-a9f6-6e7421b59c60",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_merge_atac_dnase_sorted_bed.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/myotube/hg38_OpenChromatin_MT.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb553935-ac7f-4f70-9307-041aec35b3be",
   "metadata": {},
   "source": [
    "### promoter table overlap with openchromatin "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "3b9a02b2-ca3d-4686-9901-420de1720a8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/promoter_regions_pcg_250bp_sorted.bed -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/myotube/hg38_OpenChromatin_MT.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3fd31f1-2423-41ea-b4f9-7b244cd80535",
   "metadata": {},
   "source": [
    "### remove duplicated rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "f53ae2b1-85be-4e39-86d1-5fa0dc43fb92",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'chr_open','start_open','end_open','openChromatin','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "5f46f4d7-96c6-4b1a-b74d-b29cb1593c1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "3666029e-908b-4ee6-802b-bd23f7877331",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241883</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241884</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241885</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241886</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241887</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>241888 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "241883  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "241884  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "241885  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "241886  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "241887  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923647    924887   \n",
       "4                            5.0     chr1      923647    924887   \n",
       "...                          ...      ...         ...       ...   \n",
       "241883                       5.0        .          -1        -1   \n",
       "241884                       1.0        .          -1        -1   \n",
       "241885                       2.0        .          -1        -1   \n",
       "241886                       1.0        .          -1        -1   \n",
       "241887                       2.0        .          -1        -1   \n",
       "\n",
       "                      openChromatin  overlap  \n",
       "0                                 .        0  \n",
       "1                                 .        0  \n",
       "2                                 .        0  \n",
       "3       ATAC-seq:chr1:923647-924887      500  \n",
       "4       ATAC-seq:chr1:923647-924887      500  \n",
       "...                             ...      ...  \n",
       "241883                            .        0  \n",
       "241884                            .        0  \n",
       "241885                            .        0  \n",
       "241886                            .        0  \n",
       "241887                            .        0  \n",
       "\n",
       "[241888 rows x 13 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "449c8781-8235-4b95-a974-533f204d70c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>976006</td>\n",
       "      <td>976687</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "23  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "24  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "25  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "26  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "27  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "28  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "29  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "30  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "31  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "32  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "33  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "34  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "35  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "36  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "37  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "38  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "39  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "40  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "41  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "42  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "43  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "44  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "45  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "46  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "47  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "48  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "49  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923647    924887   \n",
       "4                        5.0     chr1      923647    924887   \n",
       "5                        5.0        .          -1        -1   \n",
       "6                        5.0        .          -1        -1   \n",
       "7                        5.0        .          -1        -1   \n",
       "8                        5.0        .          -1        -1   \n",
       "9                        5.0        .          -1        -1   \n",
       "10                       5.0        .          -1        -1   \n",
       "11                       5.0        .          -1        -1   \n",
       "12                       5.0        .          -1        -1   \n",
       "13                       2.0        .          -1        -1   \n",
       "14                       5.0     chr1      940001    941223   \n",
       "15                       5.0     chr1      940001    941223   \n",
       "16                       3.0        .          -1        -1   \n",
       "17                       2.0        .          -1        -1   \n",
       "18                       5.0        .          -1        -1   \n",
       "19                       2.0        .          -1        -1   \n",
       "20                       5.0        .          -1        -1   \n",
       "21                       1.0     chr1      958785    959675   \n",
       "22                       1.0     chr1      959223    959373   \n",
       "23                       2.0     chr1      958785    959675   \n",
       "24                       2.0     chr1      959223    959373   \n",
       "25                       3.0     chr1      958785    959675   \n",
       "26                       3.0     chr1      959223    959373   \n",
       "27                       1.0     chr1      960473    961422   \n",
       "28                       1.0     chr1      960473    961422   \n",
       "29                       3.0        .          -1        -1   \n",
       "30                       2.0        .          -1        -1   \n",
       "31                       1.0     chr1      966276    967323   \n",
       "32                       1.0     chr1      966276    967323   \n",
       "33                       2.0     chr1      966276    967323   \n",
       "34                       3.0        .          -1        -1   \n",
       "35                       3.0        .          -1        -1   \n",
       "36                       1.0     chr1      976006    976687   \n",
       "37                       2.0        .          -1        -1   \n",
       "38                       5.0        .          -1        -1   \n",
       "39                       NaN        .          -1        -1   \n",
       "40                       3.0     chr1      999640   1000919   \n",
       "41                       3.0     chr1     1000105   1000255   \n",
       "42                       2.0     chr1      999640   1000919   \n",
       "43                       2.0     chr1     1000105   1000255   \n",
       "44                       1.0     chr1      999640   1000919   \n",
       "45                       1.0     chr1     1000105   1000255   \n",
       "46                       2.0     chr1      999640   1000919   \n",
       "47                       2.0     chr1     1000105   1000255   \n",
       "48                       3.0     chr1      999640   1000919   \n",
       "49                       3.0     chr1      999640   1000919   \n",
       "\n",
       "                     openChromatin  overlap  \n",
       "0                                .        0  \n",
       "1                                .        0  \n",
       "2                                .        0  \n",
       "3      ATAC-seq:chr1:923647-924887      500  \n",
       "4      ATAC-seq:chr1:923647-924887      500  \n",
       "5                                .        0  \n",
       "6                                .        0  \n",
       "7                                .        0  \n",
       "8                                .        0  \n",
       "9                                .        0  \n",
       "10                               .        0  \n",
       "11                               .        0  \n",
       "12                               .        0  \n",
       "13                               .        0  \n",
       "14     ATAC-seq:chr1:940001-941223      500  \n",
       "15     ATAC-seq:chr1:940001-941223      397  \n",
       "16                               .        0  \n",
       "17                               .        0  \n",
       "18                               .        0  \n",
       "19                               .        0  \n",
       "20                               .        0  \n",
       "21     ATAC-seq:chr1:958785-959675      500  \n",
       "22    DNase-seq:chr1:959223-959373      150  \n",
       "23     ATAC-seq:chr1:958785-959675      500  \n",
       "24    DNase-seq:chr1:959223-959373      150  \n",
       "25     ATAC-seq:chr1:958785-959675      500  \n",
       "26    DNase-seq:chr1:959223-959373      150  \n",
       "27     ATAC-seq:chr1:960473-961422      361  \n",
       "28     ATAC-seq:chr1:960473-961422      223  \n",
       "29                               .        0  \n",
       "30                               .        0  \n",
       "31     ATAC-seq:chr1:966276-967323      456  \n",
       "32     ATAC-seq:chr1:966276-967323      476  \n",
       "33     ATAC-seq:chr1:966276-967323      476  \n",
       "34                               .        0  \n",
       "35                               .        0  \n",
       "36     ATAC-seq:chr1:976006-976687      296  \n",
       "37                               .        0  \n",
       "38                               .        0  \n",
       "39                               .        0  \n",
       "40    ATAC-seq:chr1:999640-1000919      500  \n",
       "41  DNase-seq:chr1:1000105-1000255      126  \n",
       "42    ATAC-seq:chr1:999640-1000919      500  \n",
       "43  DNase-seq:chr1:1000105-1000255      150  \n",
       "44    ATAC-seq:chr1:999640-1000919      500  \n",
       "45  DNase-seq:chr1:1000105-1000255      150  \n",
       "46    ATAC-seq:chr1:999640-1000919      500  \n",
       "47  DNase-seq:chr1:1000105-1000255      150  \n",
       "48    ATAC-seq:chr1:999640-1000919       31  \n",
       "49    ATAC-seq:chr1:999640-1000919       24  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "c87fb1ef-12a8-43f4-b3fd-4af9a1b4982d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "ad368f03-a970-4982-b1ba-5a35b74079f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241883</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241884</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241885</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241886</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241887</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>241888 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "241883  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "241884  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "241885  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "241886  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "241887  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923647    924887   \n",
       "4                            5.0     chr1      923647    924887   \n",
       "...                          ...      ...         ...       ...   \n",
       "241883                       5.0        .          -1        -1   \n",
       "241884                       1.0        .          -1        -1   \n",
       "241885                       2.0        .          -1        -1   \n",
       "241886                       1.0        .          -1        -1   \n",
       "241887                       2.0        .          -1        -1   \n",
       "\n",
       "                      openChromatin  overlap  \n",
       "0                                 .        0  \n",
       "1                                 .        0  \n",
       "2                                 .        0  \n",
       "3       ATAC-seq:chr1:923647-924887      500  \n",
       "4       ATAC-seq:chr1:923647-924887      500  \n",
       "...                             ...      ...  \n",
       "241883                            .        0  \n",
       "241884                            .        0  \n",
       "241885                            .        0  \n",
       "241886                            .        0  \n",
       "241887                            .        0  \n",
       "\n",
       "[241888 rows x 13 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "b8e5b8f5-db77-4d35-baf9-772c5bd2fcc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>976006</td>\n",
       "      <td>976687</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "23  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "24  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "25  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "26  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "27  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "28  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "29  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "30  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "31  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "32  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "33  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "34  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "35  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "36  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "37  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "38  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "39  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "40  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "41  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "42  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "43  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "44  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "45  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "46  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "47  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "48  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "49  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923647    924887   \n",
       "4                        5.0     chr1      923647    924887   \n",
       "5                        5.0        .          -1        -1   \n",
       "6                        5.0        .          -1        -1   \n",
       "7                        5.0        .          -1        -1   \n",
       "8                        5.0        .          -1        -1   \n",
       "9                        5.0        .          -1        -1   \n",
       "10                       5.0        .          -1        -1   \n",
       "11                       5.0        .          -1        -1   \n",
       "12                       5.0        .          -1        -1   \n",
       "13                       2.0        .          -1        -1   \n",
       "14                       5.0     chr1      940001    941223   \n",
       "15                       5.0     chr1      940001    941223   \n",
       "16                       3.0        .          -1        -1   \n",
       "17                       2.0        .          -1        -1   \n",
       "18                       5.0        .          -1        -1   \n",
       "19                       2.0        .          -1        -1   \n",
       "20                       5.0        .          -1        -1   \n",
       "21                       1.0     chr1      958785    959675   \n",
       "22                       1.0     chr1      959223    959373   \n",
       "23                       2.0     chr1      958785    959675   \n",
       "24                       2.0     chr1      959223    959373   \n",
       "25                       3.0     chr1      958785    959675   \n",
       "26                       3.0     chr1      959223    959373   \n",
       "27                       1.0     chr1      960473    961422   \n",
       "28                       1.0     chr1      960473    961422   \n",
       "29                       3.0        .          -1        -1   \n",
       "30                       2.0        .          -1        -1   \n",
       "31                       1.0     chr1      966276    967323   \n",
       "32                       1.0     chr1      966276    967323   \n",
       "33                       2.0     chr1      966276    967323   \n",
       "34                       3.0        .          -1        -1   \n",
       "35                       3.0        .          -1        -1   \n",
       "36                       1.0     chr1      976006    976687   \n",
       "37                       2.0        .          -1        -1   \n",
       "38                       5.0        .          -1        -1   \n",
       "39                       NaN        .          -1        -1   \n",
       "40                       3.0     chr1      999640   1000919   \n",
       "41                       3.0     chr1     1000105   1000255   \n",
       "42                       2.0     chr1      999640   1000919   \n",
       "43                       2.0     chr1     1000105   1000255   \n",
       "44                       1.0     chr1      999640   1000919   \n",
       "45                       1.0     chr1     1000105   1000255   \n",
       "46                       2.0     chr1      999640   1000919   \n",
       "47                       2.0     chr1     1000105   1000255   \n",
       "48                       3.0     chr1      999640   1000919   \n",
       "49                       3.0     chr1      999640   1000919   \n",
       "\n",
       "                     openChromatin  overlap  \n",
       "0                                .        0  \n",
       "1                                .        0  \n",
       "2                                .        0  \n",
       "3      ATAC-seq:chr1:923647-924887      500  \n",
       "4      ATAC-seq:chr1:923647-924887      500  \n",
       "5                                .        0  \n",
       "6                                .        0  \n",
       "7                                .        0  \n",
       "8                                .        0  \n",
       "9                                .        0  \n",
       "10                               .        0  \n",
       "11                               .        0  \n",
       "12                               .        0  \n",
       "13                               .        0  \n",
       "14     ATAC-seq:chr1:940001-941223      500  \n",
       "15     ATAC-seq:chr1:940001-941223      397  \n",
       "16                               .        0  \n",
       "17                               .        0  \n",
       "18                               .        0  \n",
       "19                               .        0  \n",
       "20                               .        0  \n",
       "21     ATAC-seq:chr1:958785-959675      500  \n",
       "22    DNase-seq:chr1:959223-959373      150  \n",
       "23     ATAC-seq:chr1:958785-959675      500  \n",
       "24    DNase-seq:chr1:959223-959373      150  \n",
       "25     ATAC-seq:chr1:958785-959675      500  \n",
       "26    DNase-seq:chr1:959223-959373      150  \n",
       "27     ATAC-seq:chr1:960473-961422      361  \n",
       "28     ATAC-seq:chr1:960473-961422      223  \n",
       "29                               .        0  \n",
       "30                               .        0  \n",
       "31     ATAC-seq:chr1:966276-967323      456  \n",
       "32     ATAC-seq:chr1:966276-967323      476  \n",
       "33     ATAC-seq:chr1:966276-967323      476  \n",
       "34                               .        0  \n",
       "35                               .        0  \n",
       "36     ATAC-seq:chr1:976006-976687      296  \n",
       "37                               .        0  \n",
       "38                               .        0  \n",
       "39                               .        0  \n",
       "40    ATAC-seq:chr1:999640-1000919      500  \n",
       "41  DNase-seq:chr1:1000105-1000255      126  \n",
       "42    ATAC-seq:chr1:999640-1000919      500  \n",
       "43  DNase-seq:chr1:1000105-1000255      150  \n",
       "44    ATAC-seq:chr1:999640-1000919      500  \n",
       "45  DNase-seq:chr1:1000105-1000255      150  \n",
       "46    ATAC-seq:chr1:999640-1000919      500  \n",
       "47  DNase-seq:chr1:1000105-1000255      150  \n",
       "48    ATAC-seq:chr1:999640-1000919       31  \n",
       "49    ATAC-seq:chr1:999640-1000919       24  "
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "61d50afc-ba20-4f1c-ac8d-ba760187b092",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to Type)\n",
    "group_cols = promoter_like_open.columns[:8]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "dccec6f3-8b09-454d-90a9-7dd12899d7e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open.groupby(list(group_cols), dropna=False)['openChromatin'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "4558aa29-a1bb-4237-a26a-ad8dec8070b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin  \n",
       "0                            NaN                            .  \n",
       "1                            NaN                            .  \n",
       "2                            NaN                            .  \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887  \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887  \n",
       "...                          ...                          ...  \n",
       "170552                       5.0                            .  \n",
       "170553                       1.0                            .  \n",
       "170554                       2.0                            .  \n",
       "170555                       1.0                            .  \n",
       "170556                       2.0                            .  \n",
       "\n",
       "[170557 rows x 9 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "f1a85a30-c934-46c4-acd6-62af217bd97e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                        openChromatin  \n",
       "0                                                   .  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3                         ATAC-seq:chr1:923647-924887  \n",
       "4                         ATAC-seq:chr1:923647-924887  \n",
       "5                                                   .  \n",
       "6                                                   .  \n",
       "7                                                   .  \n",
       "8                                                   .  \n",
       "9                                                   .  \n",
       "10                                                  .  \n",
       "11                                                  .  \n",
       "12                                                  .  \n",
       "13                                                  .  \n",
       "14                        ATAC-seq:chr1:940001-941223  \n",
       "15                        ATAC-seq:chr1:940001-941223  \n",
       "16                                                  .  \n",
       "17                                                  .  \n",
       "18                                                  .  \n",
       "19                                                  .  \n",
       "20                                                  .  \n",
       "21  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...  \n",
       "22  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...  \n",
       "23  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...  \n",
       "24                        ATAC-seq:chr1:960473-961422  \n",
       "25                        ATAC-seq:chr1:960473-961422  \n",
       "26                                                  .  \n",
       "27                                                  .  \n",
       "28                        ATAC-seq:chr1:966276-967323  \n",
       "29                        ATAC-seq:chr1:966276-967323  \n",
       "30                        ATAC-seq:chr1:966276-967323  \n",
       "31                                                  .  \n",
       "32                                                  .  \n",
       "33                        ATAC-seq:chr1:976006-976687  \n",
       "34                                                  .  \n",
       "35                                                  .  \n",
       "36                                                  .  \n",
       "37  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "38  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "39  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "40  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "41                       ATAC-seq:chr1:999640-1000919  \n",
       "42                       ATAC-seq:chr1:999640-1000919  \n",
       "43                      ATAC-seq:chr1:1012923-1014210  \n",
       "44                      ATAC-seq:chr1:1019149-1020697  \n",
       "45                      ATAC-seq:chr1:1019149-1020697  \n",
       "46                                                  .  \n",
       "47                                                  .  \n",
       "48                                                  .  \n",
       "49                                                  .  "
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "70759c9e-2a8a-470b-af36-c276ffb07909",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51e718c2-56ec-4903-8f6b-cf50fdaff699",
   "metadata": {},
   "source": [
    "## promoter table overlap with chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "06ae8b61-4fca-4708-b3c6-da6389dc21aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"ChromHMM_Myotube\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "b5361981-a1cb-41ec-bdea-c4116e9226a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E121_25_myotube/hg38_chromHMM_E121_25.txt',\n",
    "                          sep='\\t',\n",
    "                          header=None,\n",
    "                          names=cols\n",
    "                          )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "e87a09c4-0ed6-4cb7-98ea-6112acd69466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_Myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>10200</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10200</td>\n",
       "      <td>10600</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891173</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155966935</td>\n",
       "      <td>155967135</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891174</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155967135</td>\n",
       "      <td>156030135</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891175</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156030135</td>\n",
       "      <td>156030735</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891176</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891177</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>891178 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end ChromHMM_Myotube\n",
       "0       chr1      10000      10200         25_Quies\n",
       "1       chr1      10200      10600         19_DNase\n",
       "2       chr1      10600      96080         25_Quies\n",
       "3       chr1      96276      96476         25_Quies\n",
       "4       chr1      97276     177200         25_Quies\n",
       "...      ...        ...        ...              ...\n",
       "891173  chrX  155966935  155967135         19_DNase\n",
       "891174  chrX  155967135  156030135         25_Quies\n",
       "891175  chrX  156030135  156030735         19_DNase\n",
       "891176  chrY    1086142    1086334         25_Quies\n",
       "891177  chrY    1198191    1198391         25_Quies\n",
       "\n",
       "[891178 rows x 4 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1fcc068f-3f0d-4f23-8378-fcc0e7ae08a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_chromHMM['ChromHMM_Myotube'] = MT_chromHMM['Chrom'] + \":\" + MT_chromHMM['start'].astype('str') + \":\" + MT_chromHMM['end'].astype('str') + \":\" + MT_chromHMM['ChromHMM_Myotube']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "b7e24ff8-e263-44c6-b30e-dce9e991e152",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_Myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>10200</td>\n",
       "      <td>chr1:10000:10200:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10200</td>\n",
       "      <td>10600</td>\n",
       "      <td>chr1:10200:10600:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>chr1:96276:96476:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>chr1:97276:177200:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891173</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155966935</td>\n",
       "      <td>155967135</td>\n",
       "      <td>chrX:155966935:155967135:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891174</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155967135</td>\n",
       "      <td>156030135</td>\n",
       "      <td>chrX:155967135:156030135:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891175</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156030135</td>\n",
       "      <td>156030735</td>\n",
       "      <td>chrX:156030135:156030735:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891176</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>chrY:1086142:1086334:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891177</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>chrY:1198191:1198391:25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>891178 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end                   ChromHMM_Myotube\n",
       "0       chr1      10000      10200          chr1:10000:10200:25_Quies\n",
       "1       chr1      10200      10600          chr1:10200:10600:19_DNase\n",
       "2       chr1      10600      96080          chr1:10600:96080:25_Quies\n",
       "3       chr1      96276      96476          chr1:96276:96476:25_Quies\n",
       "4       chr1      97276     177200         chr1:97276:177200:25_Quies\n",
       "...      ...        ...        ...                                ...\n",
       "891173  chrX  155966935  155967135  chrX:155966935:155967135:19_DNase\n",
       "891174  chrX  155967135  156030135  chrX:155967135:156030135:25_Quies\n",
       "891175  chrX  156030135  156030735  chrX:156030135:156030735:19_DNase\n",
       "891176  chrY    1086142    1086334      chrY:1086142:1086334:25_Quies\n",
       "891177  chrY    1198191    1198391      chrY:1198191:1198391:25_Quies\n",
       "\n",
       "[891178 rows x 4 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "5efa4563-d041-49bd-95e8-72c56adcd6b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_chromHMM.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E121_25_myotube/hg38_chromHMM_myotube.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "89b342b8-c2c6-4e67-bd2c-e4e88eef71d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed_uniq -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E121_25_myotube/hg38_chromHMM_myotube.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "c5e71ecf-dc77-4f08-9ec0-3e1c12adadef",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'openChromatin','chr_HMM','start_HMM','end_HMM','chromHMM_myotube','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "75f9f5e3-77f8-4174-bc66-67464437ffa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "e1ddb9b9-2d80-4103-9525-b432d43ff309",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chr_HMM</th>\n",
       "      <th>start_HMM</th>\n",
       "      <th>end_HMM</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>348136</td>\n",
       "      <td>501536</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>634820</td>\n",
       "      <td>733820</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923420</td>\n",
       "      <td>923820</td>\n",
       "      <td>chr1:923420:923820:2_PromU</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923820</td>\n",
       "      <td>924620</td>\n",
       "      <td>chr1:923820:924620:22_PromP</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290239</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290240</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290241</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290242</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290243</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>290244 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "290239  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "290240  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "290241  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "290242  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "290243  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin chr_HMM  \\\n",
       "0                            NaN                            .    chr1   \n",
       "1                            NaN                            .    chr1   \n",
       "2                            NaN                            .    chr1   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "...                          ...                          ...     ...   \n",
       "290239                       5.0                            .       .   \n",
       "290240                       1.0                            .       .   \n",
       "290241                       2.0                            .       .   \n",
       "290242                       1.0                            .       .   \n",
       "290243                       2.0                            .       .   \n",
       "\n",
       "        start_HMM  end_HMM             chromHMM_myotube  overlap  \n",
       "0           10600    96080    chr1:10600:96080:25_Quies      500  \n",
       "1          348136   501536  chr1:348136:501536:25_Quies      500  \n",
       "2          634820   733820  chr1:634820:733820:25_Quies      500  \n",
       "3          923420   923820   chr1:923420:923820:2_PromU      147  \n",
       "4          923820   924620  chr1:923820:924620:22_PromP      353  \n",
       "...           ...      ...                          ...      ...  \n",
       "290239         -1       -1                            .        0  \n",
       "290240         -1       -1                            .        0  \n",
       "290241         -1       -1                            .        0  \n",
       "290242         -1       -1                            .        0  \n",
       "290243         -1       -1                            .        0  \n",
       "\n",
       "[290244 rows x 14 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "3f9725c9-7ae4-4bd2-8bea-71be49de0499",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "07e9feff-7bb2-45f3-90fd-0548cc9c6104",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chr_HMM</th>\n",
       "      <th>start_HMM</th>\n",
       "      <th>end_HMM</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>348136</td>\n",
       "      <td>501536</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>634820</td>\n",
       "      <td>733820</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923420</td>\n",
       "      <td>923820</td>\n",
       "      <td>chr1:923420:923820:2_PromU</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923820</td>\n",
       "      <td>924620</td>\n",
       "      <td>chr1:923820:924620:22_PromP</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290239</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290240</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290241</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290242</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290243</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>290244 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "290239  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "290240  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "290241  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "290242  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "290243  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin chr_HMM  \\\n",
       "0                            NaN                            .    chr1   \n",
       "1                            NaN                            .    chr1   \n",
       "2                            NaN                            .    chr1   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "...                          ...                          ...     ...   \n",
       "290239                       5.0                            .       .   \n",
       "290240                       1.0                            .       .   \n",
       "290241                       2.0                            .       .   \n",
       "290242                       1.0                            .       .   \n",
       "290243                       2.0                            .       .   \n",
       "\n",
       "        start_HMM  end_HMM             chromHMM_myotube  overlap  \n",
       "0           10600    96080    chr1:10600:96080:25_Quies      500  \n",
       "1          348136   501536  chr1:348136:501536:25_Quies      500  \n",
       "2          634820   733820  chr1:634820:733820:25_Quies      500  \n",
       "3          923420   923820   chr1:923420:923820:2_PromU      147  \n",
       "4          923820   924620  chr1:923820:924620:22_PromP      353  \n",
       "...           ...      ...                          ...      ...  \n",
       "290239         -1       -1                            .        0  \n",
       "290240         -1       -1                            .        0  \n",
       "290241         -1       -1                            .        0  \n",
       "290242         -1       -1                            .        0  \n",
       "290243         -1       -1                            .        0  \n",
       "\n",
       "[290244 rows x 14 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "05252c80-26d5-443e-a025-3efeeb851b89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level', 'openChromatin'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to openChromatin)\n",
    "group_cols = promoter_like_open.columns[:9]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "1ee55dbf-6dd4-41e9-9e4d-99d7bae31daf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open.groupby(list(group_cols), dropna=False)['chromHMM_myotube'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "24ff1d6d-f8cc-4742-b862-0fdb6cc0d39c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170552                       5.0                            .   \n",
       "170553                       1.0                            .   \n",
       "170554                       2.0                            .   \n",
       "170555                       1.0                            .   \n",
       "170556                       2.0                            .   \n",
       "\n",
       "                                         chromHMM_myotube  \n",
       "0                               chr1:10600:96080:25_Quies  \n",
       "1                             chr1:348136:501536:25_Quies  \n",
       "2                             chr1:634820:733820:25_Quies  \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 10 columns]"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "38e59480-fd89-428d-ac23-fcf67893c731",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:928020:933420:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938220:939220:17_EnhW2;chr1:939220:939420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:939820:940220:23_PromBiv;chr1:940220:9404...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:940420:941020:2_PromU;chr1:941020:941420:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:957820:958620:4_PromD2;chr1:958620:959220...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961220:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960420:961220:1_TssA;chr1:961220:961420:2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:962820:7_Tx3';chr1:962820:965820:8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962820:965820:8_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:969220:972420:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972420:975820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>chr1:976020:976620:16_EnhW1;chr1:976620:980820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:1000420:10020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10142...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1041020:1052220:8_TxWk</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                        openChromatin  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3                         ATAC-seq:chr1:923647-924887   \n",
       "4                         ATAC-seq:chr1:923647-924887   \n",
       "5                                                   .   \n",
       "6                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14                        ATAC-seq:chr1:940001-941223   \n",
       "15                        ATAC-seq:chr1:940001-941223   \n",
       "16                                                  .   \n",
       "17                                                  .   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                                                  .   \n",
       "21  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "24                        ATAC-seq:chr1:960473-961422   \n",
       "25                        ATAC-seq:chr1:960473-961422   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28                        ATAC-seq:chr1:966276-967323   \n",
       "29                        ATAC-seq:chr1:966276-967323   \n",
       "30                        ATAC-seq:chr1:966276-967323   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                        ATAC-seq:chr1:976006-976687   \n",
       "34                                                  .   \n",
       "35                                                  .   \n",
       "36                                                  .   \n",
       "37  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "41                       ATAC-seq:chr1:999640-1000919   \n",
       "42                       ATAC-seq:chr1:999640-1000919   \n",
       "43                      ATAC-seq:chr1:1012923-1014210   \n",
       "44                      ATAC-seq:chr1:1019149-1020697   \n",
       "45                      ATAC-seq:chr1:1019149-1020697   \n",
       "46                                                  .   \n",
       "47                                                  .   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                     chromHMM_myotube  \n",
       "0                           chr1:10600:96080:25_Quies  \n",
       "1                         chr1:348136:501536:25_Quies  \n",
       "2                         chr1:634820:733820:25_Quies  \n",
       "3   chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "4   chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...  \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:926820...  \n",
       "7                       chr1:925620:926820:23_PromBiv  \n",
       "8                       chr1:925620:926820:23_PromBiv  \n",
       "9                       chr1:925620:926820:23_PromBiv  \n",
       "10                      chr1:925620:926820:23_PromBiv  \n",
       "11                      chr1:925620:926820:23_PromBiv  \n",
       "12                        chr1:928020:933420:25_Quies  \n",
       "13  chr1:938220:939220:17_EnhW2;chr1:939220:939420...  \n",
       "14  chr1:939820:940220:23_PromBiv;chr1:940220:9404...  \n",
       "15  chr1:940420:941020:2_PromU;chr1:941020:941420:...  \n",
       "16                        chr1:941620:942620:22_PromP  \n",
       "17                        chr1:941620:942620:22_PromP  \n",
       "18                          chr1:945220:950020:7_Tx3'  \n",
       "19                          chr1:945220:950020:7_Tx3'  \n",
       "20  chr1:957820:958620:4_PromD2;chr1:958620:959220...  \n",
       "21  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "22  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "23  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961220:...  \n",
       "25  chr1:960420:961220:1_TssA;chr1:961220:961420:2...  \n",
       "26  chr1:962420:962820:7_Tx3';chr1:962820:965820:8...  \n",
       "27                          chr1:962820:965820:8_TxWk  \n",
       "28                         chr1:966220:967220:2_PromU  \n",
       "29                         chr1:966220:967220:2_PromU  \n",
       "30                         chr1:966220:967220:2_PromU  \n",
       "31                       chr1:969220:972420:24_ReprPC  \n",
       "32                        chr1:972420:975820:25_Quies  \n",
       "33  chr1:976020:976620:16_EnhW1;chr1:976620:980820...  \n",
       "34                       chr1:980820:984220:24_ReprPC  \n",
       "35                       chr1:980820:984220:24_ReprPC  \n",
       "36                       chr1:980820:984220:24_ReprPC  \n",
       "37  chr1:1000020:1000420:1_TssA;chr1:999020:100002...  \n",
       "38  chr1:1000020:1000420:1_TssA;chr1:999020:100002...  \n",
       "39  chr1:1000020:1000420:1_TssA;chr1:999020:100002...  \n",
       "40  chr1:1000020:1000420:1_TssA;chr1:1000420:10020...  \n",
       "41                       chr1:1000420:1002020:2_PromU  \n",
       "42                       chr1:1000420:1002020:2_PromU  \n",
       "43  chr1:1013220:1013420:1_TssA;chr1:1013420:10142...  \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "46                     chr1:1033420:1034620:24_ReprPC  \n",
       "47                     chr1:1033420:1034620:24_ReprPC  \n",
       "48  chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...  \n",
       "49                        chr1:1041020:1052220:8_TxWk  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "4687dbce-a028-4937-a1f0-82ef46fe4945",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48e46e4c-4a02-4286-b18d-b9843cbce955",
   "metadata": {},
   "source": [
    "## promoter overlap with Hi-C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "c641047c-e227-441b-a240-966c2e6f4463",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_uniq -b /mnt/10T-backup-mingju/ifar/Hi-C/MT/merge_results.txt_sorted_lexicographical_addchr -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_HiC.bed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "37b3a092-3e46-41aa-97ff-18431186071b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'openChromatin_MT','chromHMM_MT','chr_hic','start_hic','end_hic','HiC_MT','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "d89d3ca3-a38f-41f4-9b86-844ee9a210e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_HiC.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "162b7fbc-df46-4875-991c-10a094cd253b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>chr_hic</th>\n",
       "      <th>start_hic</th>\n",
       "      <th>end_hic</th>\n",
       "      <th>HiC_MT</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>922000</td>\n",
       "      <td>924000</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>922000</td>\n",
       "      <td>924000</td>\n",
       "      <td>P:1:922000:924000;D:1:900000:902000;GOTHIC:rea...</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172726</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172727</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172728</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172729</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172730</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5172731 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0        chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1        chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2        chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3        chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4        chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...       ...       ...       ...              ...       ...     ...    ...   \n",
       "5172726  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "5172727  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "5172728  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "5172729  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "5172730  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "         transcript_support_level             openChromatin_MT  \\\n",
       "0                             NaN                            .   \n",
       "1                             NaN                            .   \n",
       "2                             NaN                            .   \n",
       "3                             5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                             5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                           ...                          ...   \n",
       "5172726                       5.0                            .   \n",
       "5172727                       1.0                            .   \n",
       "5172728                       2.0                            .   \n",
       "5172729                       1.0                            .   \n",
       "5172730                       2.0                            .   \n",
       "\n",
       "                                               chromHMM_MT chr_hic  start_hic  \\\n",
       "0                                chr1:10600:96080:25_Quies       .         -1   \n",
       "1                              chr1:348136:501536:25_Quies       .         -1   \n",
       "2                              chr1:634820:733820:25_Quies       .         -1   \n",
       "3        chr1:923420:923820:2_PromU;chr1:923820:924620:...    chr1     922000   \n",
       "4        chr1:923420:923820:2_PromU;chr1:923820:924620:...    chr1     922000   \n",
       "...                                                    ...     ...        ...   \n",
       "5172726                                                  .       .         -1   \n",
       "5172727                                                  .       .         -1   \n",
       "5172728                                                  .       .         -1   \n",
       "5172729                                                  .       .         -1   \n",
       "5172730                                                  .       .         -1   \n",
       "\n",
       "         end_hic                                             HiC_MT  overlap  \n",
       "0             -1                                                  .        0  \n",
       "1             -1                                                  .        0  \n",
       "2             -1                                                  .        0  \n",
       "3         924000  P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...      327  \n",
       "4         924000  P:1:922000:924000;D:1:900000:902000;GOTHIC:rea...      327  \n",
       "...          ...                                                ...      ...  \n",
       "5172726       -1                                                  .        0  \n",
       "5172727       -1                                                  .        0  \n",
       "5172728       -1                                                  .        0  \n",
       "5172729       -1                                                  .        0  \n",
       "5172730       -1                                                  .        0  \n",
       "\n",
       "[5172731 rows x 15 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "cf2857fe-7286-4625-9f30-b1b9788f701e",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open_chromHMM.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "956e233f-312f-4751-91bb-fafbd06a580b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5172731"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoter_like_open_chromHMM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "b99edd05-1aba-4cc4-9b51-361b6ffa3dcd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level', 'openChromatin_MT', 'chromHMM_MT'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to chromHMM_myotube)\n",
    "group_cols = promoter_like_open_chromHMM.columns[:10]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "8d8e279f-5444-4bfb-9eb0-89011bf8a0fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open_chromHMM.groupby(list(group_cols), dropna=False)['HiC_MT'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "e1617945-fc63-47a7-887b-dafce9f38cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level             openChromatin_MT  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170552                       5.0                            .   \n",
       "170553                       1.0                            .   \n",
       "170554                       2.0                            .   \n",
       "170555                       1.0                            .   \n",
       "170556                       2.0                            .   \n",
       "\n",
       "                                              chromHMM_MT  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170552                                                  .   \n",
       "170553                                                  .   \n",
       "170554                                                  .   \n",
       "170555                                                  .   \n",
       "170556                                                  .   \n",
       "\n",
       "                                                   HiC_MT  \n",
       "0                                                       .  \n",
       "1                                                       .  \n",
       "2                                                       .  \n",
       "3       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "4       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 11 columns]"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "4123bfc4-2464-4621-a25a-f9bfc4e5eab0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:928020:933420:25_Quies</td>\n",
       "      <td>P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938220:939220:17_EnhW2;chr1:939220:939420...</td>\n",
       "      <td>P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:939820:940220:23_PromBiv;chr1:940220:9404...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:940420:941020:2_PromU;chr1:941020:941420:...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:957820:958620:4_PromD2;chr1:958620:959220...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961220:...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960420:961220:1_TssA;chr1:961220:961420:2...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:962820:7_Tx3';chr1:962820:965820:8...</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962820:965820:8_TxWk</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:969220:972420:24_ReprPC</td>\n",
       "      <td>P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972420:975820:25_Quies</td>\n",
       "      <td>P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>chr1:976020:976620:16_EnhW1;chr1:976620:980820...</td>\n",
       "      <td>P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:1000420:10020...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10142...</td>\n",
       "      <td>P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "      <td>P:1:1034000:1036000;D:1:1016000:1018000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1041020:1052220:8_TxWk</td>\n",
       "      <td>P:1:1044000:1046000;D:1:1018000:1020000;GOTHIC...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                     openChromatin_MT  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3                         ATAC-seq:chr1:923647-924887   \n",
       "4                         ATAC-seq:chr1:923647-924887   \n",
       "5                                                   .   \n",
       "6                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14                        ATAC-seq:chr1:940001-941223   \n",
       "15                        ATAC-seq:chr1:940001-941223   \n",
       "16                                                  .   \n",
       "17                                                  .   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                                                  .   \n",
       "21  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "24                        ATAC-seq:chr1:960473-961422   \n",
       "25                        ATAC-seq:chr1:960473-961422   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28                        ATAC-seq:chr1:966276-967323   \n",
       "29                        ATAC-seq:chr1:966276-967323   \n",
       "30                        ATAC-seq:chr1:966276-967323   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                        ATAC-seq:chr1:976006-976687   \n",
       "34                                                  .   \n",
       "35                                                  .   \n",
       "36                                                  .   \n",
       "37  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "41                       ATAC-seq:chr1:999640-1000919   \n",
       "42                       ATAC-seq:chr1:999640-1000919   \n",
       "43                      ATAC-seq:chr1:1012923-1014210   \n",
       "44                      ATAC-seq:chr1:1019149-1020697   \n",
       "45                      ATAC-seq:chr1:1019149-1020697   \n",
       "46                                                  .   \n",
       "47                                                  .   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                          chromHMM_MT  \\\n",
       "0                           chr1:10600:96080:25_Quies   \n",
       "1                         chr1:348136:501536:25_Quies   \n",
       "2                         chr1:634820:733820:25_Quies   \n",
       "3   chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4   chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...   \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "7                       chr1:925620:926820:23_PromBiv   \n",
       "8                       chr1:925620:926820:23_PromBiv   \n",
       "9                       chr1:925620:926820:23_PromBiv   \n",
       "10                      chr1:925620:926820:23_PromBiv   \n",
       "11                      chr1:925620:926820:23_PromBiv   \n",
       "12                        chr1:928020:933420:25_Quies   \n",
       "13  chr1:938220:939220:17_EnhW2;chr1:939220:939420...   \n",
       "14  chr1:939820:940220:23_PromBiv;chr1:940220:9404...   \n",
       "15  chr1:940420:941020:2_PromU;chr1:941020:941420:...   \n",
       "16                        chr1:941620:942620:22_PromP   \n",
       "17                        chr1:941620:942620:22_PromP   \n",
       "18                          chr1:945220:950020:7_Tx3'   \n",
       "19                          chr1:945220:950020:7_Tx3'   \n",
       "20  chr1:957820:958620:4_PromD2;chr1:958620:959220...   \n",
       "21  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "22  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "23  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961220:...   \n",
       "25  chr1:960420:961220:1_TssA;chr1:961220:961420:2...   \n",
       "26  chr1:962420:962820:7_Tx3';chr1:962820:965820:8...   \n",
       "27                          chr1:962820:965820:8_TxWk   \n",
       "28                         chr1:966220:967220:2_PromU   \n",
       "29                         chr1:966220:967220:2_PromU   \n",
       "30                         chr1:966220:967220:2_PromU   \n",
       "31                       chr1:969220:972420:24_ReprPC   \n",
       "32                        chr1:972420:975820:25_Quies   \n",
       "33  chr1:976020:976620:16_EnhW1;chr1:976620:980820...   \n",
       "34                       chr1:980820:984220:24_ReprPC   \n",
       "35                       chr1:980820:984220:24_ReprPC   \n",
       "36                       chr1:980820:984220:24_ReprPC   \n",
       "37  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "38  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "39  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "40  chr1:1000020:1000420:1_TssA;chr1:1000420:10020...   \n",
       "41                       chr1:1000420:1002020:2_PromU   \n",
       "42                       chr1:1000420:1002020:2_PromU   \n",
       "43  chr1:1013220:1013420:1_TssA;chr1:1013420:10142...   \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "46                     chr1:1033420:1034620:24_ReprPC   \n",
       "47                     chr1:1033420:1034620:24_ReprPC   \n",
       "48  chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...   \n",
       "49                        chr1:1041020:1052220:8_TxWk   \n",
       "\n",
       "                                               HiC_MT  \n",
       "0                                                   .  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3   P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "4   P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "5   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "6   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "7   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "8   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "9   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "10  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "11  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "12  P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...  \n",
       "13  P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...  \n",
       "14  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "15  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "16  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "17  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "18  P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...  \n",
       "19  P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...  \n",
       "20  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "21  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "22  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "23  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "24  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...  \n",
       "25  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...  \n",
       "26  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...  \n",
       "27  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...  \n",
       "28  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "29  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "30  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "31  P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...  \n",
       "32  P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...  \n",
       "33  P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...  \n",
       "34  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "35  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "36  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "37  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "38  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "39  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "40  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "41  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "42  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "43  P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...  \n",
       "44  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "45  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "46  P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...  \n",
       "47  P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...  \n",
       "48  P:1:1034000:1036000;D:1:1016000:1018000;GOTHIC...  \n",
       "49  P:1:1044000:1046000;D:1:1018000:1020000;GOTHIC...  "
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19c6cd9d-9fde-4449-aeec-712b0c9beb97",
   "metadata": {},
   "source": [
    "# add TPM from RNA-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "4b56c1b9-d383-402b-ba60-4082c5f9ac67",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/MT/ENCFF082PRM_gene_quantifications.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "e770c00f-12d2-433a-b1fa-af70fabf4e05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>length</th>\n",
       "      <th>effective_length</th>\n",
       "      <th>expected_count</th>\n",
       "      <th>TPM</th>\n",
       "      <th>FPKM</th>\n",
       "      <th>posterior_mean_count</th>\n",
       "      <th>posterior_standard_deviation_of_count</th>\n",
       "      <th>pme_TPM</th>\n",
       "      <th>pme_FPKM</th>\n",
       "      <th>TPM_ci_lower_bound</th>\n",
       "      <th>TPM_ci_upper_bound</th>\n",
       "      <th>TPM_coefficient_of_quartile_variation</th>\n",
       "      <th>FPKM_ci_lower_bound</th>\n",
       "      <th>FPKM_ci_upper_bound</th>\n",
       "      <th>FPKM_coefficient_of_quartile_variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10904</td>\n",
       "      <td>10904</td>\n",
       "      <td>93.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12954</td>\n",
       "      <td>12954</td>\n",
       "      <td>94.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12956</td>\n",
       "      <td>12956</td>\n",
       "      <td>72.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12958</td>\n",
       "      <td>12958</td>\n",
       "      <td>82.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12960</td>\n",
       "      <td>12960</td>\n",
       "      <td>73.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59521</th>\n",
       "      <td>gSpikein_ERCC-00165</td>\n",
       "      <td>tSpikein_ERCC-00165</td>\n",
       "      <td>872.0</td>\n",
       "      <td>660.31</td>\n",
       "      <td>394.0</td>\n",
       "      <td>2.35</td>\n",
       "      <td>8.09</td>\n",
       "      <td>394.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.34</td>\n",
       "      <td>8.09</td>\n",
       "      <td>2.113210e+00</td>\n",
       "      <td>2.574600</td>\n",
       "      <td>0.033787</td>\n",
       "      <td>7.297980e+00</td>\n",
       "      <td>8.890880</td>\n",
       "      <td>0.033765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59522</th>\n",
       "      <td>gSpikein_ERCC-00168</td>\n",
       "      <td>tSpikein_ERCC-00168</td>\n",
       "      <td>1024.0</td>\n",
       "      <td>812.31</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>8.780920e-08</td>\n",
       "      <td>0.014472</td>\n",
       "      <td>0.654832</td>\n",
       "      <td>3.031480e-07</td>\n",
       "      <td>0.049961</td>\n",
       "      <td>0.654813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59523</th>\n",
       "      <td>gSpikein_ERCC-00170</td>\n",
       "      <td>tSpikein_ERCC-00170</td>\n",
       "      <td>1023.0</td>\n",
       "      <td>811.31</td>\n",
       "      <td>41.0</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.69</td>\n",
       "      <td>41.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.70</td>\n",
       "      <td>1.425040e-01</td>\n",
       "      <td>0.263636</td>\n",
       "      <td>0.104314</td>\n",
       "      <td>4.918400e-01</td>\n",
       "      <td>0.909995</td>\n",
       "      <td>0.104317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59524</th>\n",
       "      <td>gSpikein_ERCC-00171</td>\n",
       "      <td>tSpikein_ERCC-00171</td>\n",
       "      <td>505.0</td>\n",
       "      <td>293.35</td>\n",
       "      <td>11035.0</td>\n",
       "      <td>148.00</td>\n",
       "      <td>510.17</td>\n",
       "      <td>11035.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>147.47</td>\n",
       "      <td>508.85</td>\n",
       "      <td>1.447460e+02</td>\n",
       "      <td>150.261000</td>\n",
       "      <td>0.006505</td>\n",
       "      <td>4.997160e+02</td>\n",
       "      <td>518.761000</td>\n",
       "      <td>0.006507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59525</th>\n",
       "      <td>gSpikein_phiX174</td>\n",
       "      <td>tSpikein_phiX174</td>\n",
       "      <td>5386.0</td>\n",
       "      <td>5174.31</td>\n",
       "      <td>795.0</td>\n",
       "      <td>0.60</td>\n",
       "      <td>2.08</td>\n",
       "      <td>795.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.60</td>\n",
       "      <td>2.08</td>\n",
       "      <td>5.613070e-01</td>\n",
       "      <td>0.645292</td>\n",
       "      <td>0.023993</td>\n",
       "      <td>1.938520e+00</td>\n",
       "      <td>2.228480</td>\n",
       "      <td>0.023977</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59526 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   gene_id     transcript_id(s)  length  effective_length  \\\n",
       "0                    10904                10904    93.0              0.00   \n",
       "1                    12954                12954    94.0              0.00   \n",
       "2                    12956                12956    72.0              0.00   \n",
       "3                    12958                12958    82.0              0.00   \n",
       "4                    12960                12960    73.0              0.00   \n",
       "...                    ...                  ...     ...               ...   \n",
       "59521  gSpikein_ERCC-00165  tSpikein_ERCC-00165   872.0            660.31   \n",
       "59522  gSpikein_ERCC-00168  tSpikein_ERCC-00168  1024.0            812.31   \n",
       "59523  gSpikein_ERCC-00170  tSpikein_ERCC-00170  1023.0            811.31   \n",
       "59524  gSpikein_ERCC-00171  tSpikein_ERCC-00171   505.0            293.35   \n",
       "59525     gSpikein_phiX174     tSpikein_phiX174  5386.0           5174.31   \n",
       "\n",
       "       expected_count     TPM    FPKM  posterior_mean_count  \\\n",
       "0                 0.0    0.00    0.00                   0.0   \n",
       "1                 0.0    0.00    0.00                   0.0   \n",
       "2                 0.0    0.00    0.00                   0.0   \n",
       "3                 0.0    0.00    0.00                   0.0   \n",
       "4                 0.0    0.00    0.00                   0.0   \n",
       "...               ...     ...     ...                   ...   \n",
       "59521           394.0    2.35    8.09                 394.0   \n",
       "59522             0.0    0.00    0.00                   0.0   \n",
       "59523            41.0    0.20    0.69                  41.0   \n",
       "59524         11035.0  148.00  510.17               11035.0   \n",
       "59525           795.0    0.60    2.08                 795.0   \n",
       "\n",
       "       posterior_standard_deviation_of_count  pme_TPM  pme_FPKM  \\\n",
       "0                                        0.0     0.00      0.00   \n",
       "1                                        0.0     0.00      0.00   \n",
       "2                                        0.0     0.00      0.00   \n",
       "3                                        0.0     0.00      0.00   \n",
       "4                                        0.0     0.00      0.00   \n",
       "...                                      ...      ...       ...   \n",
       "59521                                    0.0     2.34      8.09   \n",
       "59522                                    0.0     0.00      0.02   \n",
       "59523                                    0.0     0.20      0.70   \n",
       "59524                                    0.0   147.47    508.85   \n",
       "59525                                    0.0     0.60      2.08   \n",
       "\n",
       "       TPM_ci_lower_bound  TPM_ci_upper_bound  \\\n",
       "0            0.000000e+00            0.000000   \n",
       "1            0.000000e+00            0.000000   \n",
       "2            0.000000e+00            0.000000   \n",
       "3            0.000000e+00            0.000000   \n",
       "4            0.000000e+00            0.000000   \n",
       "...                   ...                 ...   \n",
       "59521        2.113210e+00            2.574600   \n",
       "59522        8.780920e-08            0.014472   \n",
       "59523        1.425040e-01            0.263636   \n",
       "59524        1.447460e+02          150.261000   \n",
       "59525        5.613070e-01            0.645292   \n",
       "\n",
       "       TPM_coefficient_of_quartile_variation  FPKM_ci_lower_bound  \\\n",
       "0                                   0.000000         0.000000e+00   \n",
       "1                                   0.000000         0.000000e+00   \n",
       "2                                   0.000000         0.000000e+00   \n",
       "3                                   0.000000         0.000000e+00   \n",
       "4                                   0.000000         0.000000e+00   \n",
       "...                                      ...                  ...   \n",
       "59521                               0.033787         7.297980e+00   \n",
       "59522                               0.654832         3.031480e-07   \n",
       "59523                               0.104314         4.918400e-01   \n",
       "59524                               0.006505         4.997160e+02   \n",
       "59525                               0.023993         1.938520e+00   \n",
       "\n",
       "       FPKM_ci_upper_bound  FPKM_coefficient_of_quartile_variation  \n",
       "0                 0.000000                                0.000000  \n",
       "1                 0.000000                                0.000000  \n",
       "2                 0.000000                                0.000000  \n",
       "3                 0.000000                                0.000000  \n",
       "4                 0.000000                                0.000000  \n",
       "...                    ...                                     ...  \n",
       "59521             8.890880                                0.033765  \n",
       "59522             0.049961                                0.654813  \n",
       "59523             0.909995                                0.104317  \n",
       "59524           518.761000                                0.006507  \n",
       "59525             2.228480                                0.023977  \n",
       "\n",
       "[59526 rows x 17 columns]"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "edf0f486-52bc-48bc-9f8b-bd52ca6e86bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[['gene_id','TPM']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "003aef71-d14c-46aa-8310-242a4e3830bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 1. keep rows where gene_id starts with 'ENSG'\n",
    "gene_tpm = gene_tpm[gene_tpm[\"gene_id\"].str.startswith(\"ENSG\", na=False)].copy()\n",
    "\n",
    "# 2. remove version suffix (e.g. .8)\n",
    "gene_tpm[\"gene_id\"] = gene_tpm[\"gene_id\"].str.replace(r\"\\.\\d+$\", \"\", regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "bc3a875c-48e4-41b6-b1e2-03d408273c1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>TPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>6.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>1.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>0.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59424</th>\n",
       "      <td>ENSG00000285990</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59425</th>\n",
       "      <td>ENSG00000285991</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>ENSG00000285992</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>ENSG00000285993</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>ENSG00000285994</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58780 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id   TPM\n",
       "649    ENSG00000000003  2.92\n",
       "650    ENSG00000000005  0.00\n",
       "651    ENSG00000000419  6.24\n",
       "652    ENSG00000000457  1.14\n",
       "653    ENSG00000000460  0.49\n",
       "...                ...   ...\n",
       "59424  ENSG00000285990  0.00\n",
       "59425  ENSG00000285991  0.00\n",
       "59426  ENSG00000285992  0.00\n",
       "59427  ENSG00000285993  0.00\n",
       "59428  ENSG00000285994  0.00\n",
       "\n",
       "[58780 rows x 2 columns]"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "70745fa0-dd06-4c13-bc4b-74f60e35c7ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map =pd.read_csv('./geneid2name.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "6dd4c7a8-2336-4319-a66d-a1beca807a4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000142611</td>\n",
       "      <td>PRDM16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000157911</td>\n",
       "      <td>PEX10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000142655</td>\n",
       "      <td>PEX14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000149527</td>\n",
       "      <td>PLCH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000171621</td>\n",
       "      <td>SPSB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19469</th>\n",
       "      <td>ENSG00000198886</td>\n",
       "      <td>MT-ND4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19470</th>\n",
       "      <td>ENSG00000198786</td>\n",
       "      <td>MT-ND5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19471</th>\n",
       "      <td>ENSG00000198695</td>\n",
       "      <td>MT-ND6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19472</th>\n",
       "      <td>ENSG00000198727</td>\n",
       "      <td>MT-CYB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19473</th>\n",
       "      <td>ENSG00000274847</td>\n",
       "      <td>MAFIP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19474 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id gene_name\n",
       "0      ENSG00000142611    PRDM16\n",
       "1      ENSG00000157911     PEX10\n",
       "2      ENSG00000142655     PEX14\n",
       "3      ENSG00000149527     PLCH2\n",
       "4      ENSG00000171621     SPSB1\n",
       "...                ...       ...\n",
       "19469  ENSG00000198886    MT-ND4\n",
       "19470  ENSG00000198786    MT-ND5\n",
       "19471  ENSG00000198695    MT-ND6\n",
       "19472  ENSG00000198727    MT-CYB\n",
       "19473  ENSG00000274847     MAFIP\n",
       "\n",
       "[19474 rows x 2 columns]"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "036b3871-80db-4c4a-bde8-1a931eea7005",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm.merge(\n",
    "    gene_map,\n",
    "    on=\"gene_id\",\n",
    "    how=\"left\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "bd038c81-8eb5-4dcc-bbe2-fa7062f5c351",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>TPM</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2.92</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>6.24</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>1.14</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>0.49</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58775</th>\n",
       "      <td>ENSG00000285990</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58776</th>\n",
       "      <td>ENSG00000285991</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58777</th>\n",
       "      <td>ENSG00000285992</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58778</th>\n",
       "      <td>ENSG00000285993</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58779</th>\n",
       "      <td>ENSG00000285994</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58780 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id   TPM gene_name\n",
       "0      ENSG00000000003  2.92    TSPAN6\n",
       "1      ENSG00000000005  0.00      TNMD\n",
       "2      ENSG00000000419  6.24      DPM1\n",
       "3      ENSG00000000457  1.14     SCYL3\n",
       "4      ENSG00000000460  0.49     FIRRM\n",
       "...                ...   ...       ...\n",
       "58775  ENSG00000285990  0.00       NaN\n",
       "58776  ENSG00000285991  0.00       NaN\n",
       "58777  ENSG00000285992  0.00       NaN\n",
       "58778  ENSG00000285993  0.00       NaN\n",
       "58779  ENSG00000285994  0.00       NaN\n",
       "\n",
       "[58780 rows x 3 columns]"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "c3f21298-a993-4fdc-831b-f5ef3ad4fec5",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[gene_tpm[\"gene_name\"].notna()].reset_index(drop=True)\n",
    "gene_tpm = gene_tpm.rename(columns={'TPM':\"GeneTPM_MT\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "bd97b2bd-b912-4d05-8fb9-dbfdc5c3bbbf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2.92</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>6.24</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>1.14</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>0.49</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19389</th>\n",
       "      <td>ENSG00000285913</td>\n",
       "      <td>0.00</td>\n",
       "      <td>FAM90A18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19390</th>\n",
       "      <td>ENSG00000285937</td>\n",
       "      <td>0.00</td>\n",
       "      <td>FAM90A8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19391</th>\n",
       "      <td>ENSG00000285946</td>\n",
       "      <td>0.00</td>\n",
       "      <td>LCE7A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19392</th>\n",
       "      <td>ENSG00000285950</td>\n",
       "      <td>0.00</td>\n",
       "      <td>FAM90A10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19393</th>\n",
       "      <td>ENSG00000285975</td>\n",
       "      <td>0.00</td>\n",
       "      <td>FAM90A7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19394 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id  GeneTPM_MT gene_name\n",
       "0      ENSG00000000003        2.92    TSPAN6\n",
       "1      ENSG00000000005        0.00      TNMD\n",
       "2      ENSG00000000419        6.24      DPM1\n",
       "3      ENSG00000000457        1.14     SCYL3\n",
       "4      ENSG00000000460        0.49     FIRRM\n",
       "...                ...         ...       ...\n",
       "19389  ENSG00000285913        0.00  FAM90A18\n",
       "19390  ENSG00000285937        0.00   FAM90A8\n",
       "19391  ENSG00000285946        0.00     LCE7A\n",
       "19392  ENSG00000285950        0.00  FAM90A10\n",
       "19393  ENSG00000285975        0.00   FAM90A7\n",
       "\n",
       "[19394 rows x 3 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "8624f0b0-90c6-45ef-b83b-03b2943b80a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df = collapsed_df.merge(\n",
    "    gene_tpm,\n",
    "    left_on=\"Gene\",\n",
    "    right_on=\"gene_name\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['gene_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "43d51dac-9f53-4c7a-bb56-9b4b325a8eee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000284733</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000284662</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>ENSG00000187634</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>ENSG00000187634</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170652</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>2.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170653</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>2.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170654</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>2.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170655</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>2.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170656</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>2.30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170657 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170652  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170653  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170654  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170655  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170656  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level             openChromatin_MT  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170652                       5.0                            .   \n",
       "170653                       1.0                            .   \n",
       "170654                       2.0                            .   \n",
       "170655                       1.0                            .   \n",
       "170656                       2.0                            .   \n",
       "\n",
       "                                              chromHMM_MT  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170652                                                  .   \n",
       "170653                                                  .   \n",
       "170654                                                  .   \n",
       "170655                                                  .   \n",
       "170656                                                  .   \n",
       "\n",
       "                                                   HiC_MT          gene_id  \\\n",
       "0                                                       .  ENSG00000186092   \n",
       "1                                                       .  ENSG00000284733   \n",
       "2                                                       .  ENSG00000284662   \n",
       "3       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  ENSG00000187634   \n",
       "4       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  ENSG00000187634   \n",
       "...                                                   ...              ...   \n",
       "170652                                                  .  ENSG00000182484   \n",
       "170653                                                  .  ENSG00000182484   \n",
       "170654                                                  .  ENSG00000182484   \n",
       "170655                                                  .  ENSG00000182484   \n",
       "170656                                                  .  ENSG00000182484   \n",
       "\n",
       "        GeneTPM_MT  \n",
       "0             0.00  \n",
       "1             0.00  \n",
       "2             0.00  \n",
       "3             0.25  \n",
       "4             0.25  \n",
       "...            ...  \n",
       "170652        2.30  \n",
       "170653        2.30  \n",
       "170654        2.30  \n",
       "170655        2.30  \n",
       "170656        2.30  \n",
       "\n",
       "[170657 rows x 13 columns]"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "56c41029-6283-432c-9f4d-e2b717e44d44",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneTPM_MT\n",
       "False    170451\n",
       "True        206\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df['GeneTPM_MT'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "1b3f4836-dcb2-4c7e-9da7-f0409ca0da92",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/MT/ENCFF163LMY_transcript_quantifications.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "c0075f08-1f08-4cb6-8c5d-5e811ae8818e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>length</th>\n",
       "      <th>effective_length</th>\n",
       "      <th>expected_count</th>\n",
       "      <th>TPM</th>\n",
       "      <th>FPKM</th>\n",
       "      <th>IsoPct</th>\n",
       "      <th>posterior_mean_count</th>\n",
       "      <th>posterior_standard_deviation_of_count</th>\n",
       "      <th>pme_TPM</th>\n",
       "      <th>pme_FPKM</th>\n",
       "      <th>IsoPct_from_pme_TPM</th>\n",
       "      <th>TPM_ci_lower_bound</th>\n",
       "      <th>TPM_ci_upper_bound</th>\n",
       "      <th>TPM_coefficient_of_quartile_variation</th>\n",
       "      <th>FPKM_ci_lower_bound</th>\n",
       "      <th>FPKM_ci_upper_bound</th>\n",
       "      <th>FPKM_coefficient_of_quartile_variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10904</td>\n",
       "      <td>10904</td>\n",
       "      <td>93</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12954</td>\n",
       "      <td>12954</td>\n",
       "      <td>94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12956</td>\n",
       "      <td>12956</td>\n",
       "      <td>72</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12958</td>\n",
       "      <td>12958</td>\n",
       "      <td>82</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12960</td>\n",
       "      <td>12960</td>\n",
       "      <td>73</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207502</th>\n",
       "      <td>tSpikein_ERCC-00165</td>\n",
       "      <td>gSpikein_ERCC-00165</td>\n",
       "      <td>872</td>\n",
       "      <td>686.14</td>\n",
       "      <td>301.0</td>\n",
       "      <td>2.50</td>\n",
       "      <td>6.36</td>\n",
       "      <td>100.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.50</td>\n",
       "      <td>6.36</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2.219180</td>\n",
       "      <td>2.781900</td>\n",
       "      <td>0.038790</td>\n",
       "      <td>5.631410</td>\n",
       "      <td>7.065010</td>\n",
       "      <td>0.038766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207503</th>\n",
       "      <td>tSpikein_ERCC-00168</td>\n",
       "      <td>gSpikein_ERCC-00168</td>\n",
       "      <td>1024</td>\n",
       "      <td>838.14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.03</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.05</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.001853</td>\n",
       "      <td>0.042867</td>\n",
       "      <td>0.386214</td>\n",
       "      <td>0.004742</td>\n",
       "      <td>0.109185</td>\n",
       "      <td>0.386213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207504</th>\n",
       "      <td>tSpikein_ERCC-00170</td>\n",
       "      <td>gSpikein_ERCC-00170</td>\n",
       "      <td>1023</td>\n",
       "      <td>837.14</td>\n",
       "      <td>51.0</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.88</td>\n",
       "      <td>100.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.90</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.258597</td>\n",
       "      <td>0.449118</td>\n",
       "      <td>0.093611</td>\n",
       "      <td>0.658386</td>\n",
       "      <td>1.143530</td>\n",
       "      <td>0.093571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207505</th>\n",
       "      <td>tSpikein_ERCC-00171</td>\n",
       "      <td>gSpikein_ERCC-00171</td>\n",
       "      <td>505</td>\n",
       "      <td>319.16</td>\n",
       "      <td>7538.0</td>\n",
       "      <td>134.73</td>\n",
       "      <td>342.20</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7538.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>134.07</td>\n",
       "      <td>341.25</td>\n",
       "      <td>100.0</td>\n",
       "      <td>131.062000</td>\n",
       "      <td>137.150000</td>\n",
       "      <td>0.007806</td>\n",
       "      <td>333.603000</td>\n",
       "      <td>349.115000</td>\n",
       "      <td>0.007803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207506</th>\n",
       "      <td>tSpikein_phiX174</td>\n",
       "      <td>gSpikein_phiX174</td>\n",
       "      <td>5386</td>\n",
       "      <td>5200.14</td>\n",
       "      <td>1663.0</td>\n",
       "      <td>1.82</td>\n",
       "      <td>4.63</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1663.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.82</td>\n",
       "      <td>4.62</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1.729570</td>\n",
       "      <td>1.903670</td>\n",
       "      <td>0.016646</td>\n",
       "      <td>4.404830</td>\n",
       "      <td>4.848080</td>\n",
       "      <td>0.016642</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>207507 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              transcript_id              gene_id  length  effective_length  \\\n",
       "0                     10904                10904      93              0.00   \n",
       "1                     12954                12954      94              0.00   \n",
       "2                     12956                12956      72              0.00   \n",
       "3                     12958                12958      82              0.00   \n",
       "4                     12960                12960      73              0.00   \n",
       "...                     ...                  ...     ...               ...   \n",
       "207502  tSpikein_ERCC-00165  gSpikein_ERCC-00165     872            686.14   \n",
       "207503  tSpikein_ERCC-00168  gSpikein_ERCC-00168    1024            838.14   \n",
       "207504  tSpikein_ERCC-00170  gSpikein_ERCC-00170    1023            837.14   \n",
       "207505  tSpikein_ERCC-00171  gSpikein_ERCC-00171     505            319.16   \n",
       "207506     tSpikein_phiX174     gSpikein_phiX174    5386           5200.14   \n",
       "\n",
       "        expected_count     TPM    FPKM  IsoPct  posterior_mean_count  \\\n",
       "0                  0.0    0.00    0.00     0.0                   0.0   \n",
       "1                  0.0    0.00    0.00     0.0                   0.0   \n",
       "2                  0.0    0.00    0.00     0.0                   0.0   \n",
       "3                  0.0    0.00    0.00     0.0                   0.0   \n",
       "4                  0.0    0.00    0.00     0.0                   0.0   \n",
       "...                ...     ...     ...     ...                   ...   \n",
       "207502           301.0    2.50    6.36   100.0                 301.0   \n",
       "207503             2.0    0.01    0.03   100.0                   2.0   \n",
       "207504            51.0    0.35    0.88   100.0                  51.0   \n",
       "207505          7538.0  134.73  342.20   100.0                7538.0   \n",
       "207506          1663.0    1.82    4.63   100.0                1663.0   \n",
       "\n",
       "        posterior_standard_deviation_of_count  pme_TPM  pme_FPKM  \\\n",
       "0                                         0.0     0.00      0.00   \n",
       "1                                         0.0     0.00      0.00   \n",
       "2                                         0.0     0.00      0.00   \n",
       "3                                         0.0     0.00      0.00   \n",
       "4                                         0.0     0.00      0.00   \n",
       "...                                       ...      ...       ...   \n",
       "207502                                    0.0     2.50      6.36   \n",
       "207503                                    0.0     0.02      0.05   \n",
       "207504                                    0.0     0.35      0.90   \n",
       "207505                                    0.0   134.07    341.25   \n",
       "207506                                    0.0     1.82      4.62   \n",
       "\n",
       "        IsoPct_from_pme_TPM  TPM_ci_lower_bound  TPM_ci_upper_bound  \\\n",
       "0                       0.0            0.000000            0.000000   \n",
       "1                       0.0            0.000000            0.000000   \n",
       "2                       0.0            0.000000            0.000000   \n",
       "3                       0.0            0.000000            0.000000   \n",
       "4                       0.0            0.000000            0.000000   \n",
       "...                     ...                 ...                 ...   \n",
       "207502                100.0            2.219180            2.781900   \n",
       "207503                100.0            0.001853            0.042867   \n",
       "207504                100.0            0.258597            0.449118   \n",
       "207505                100.0          131.062000          137.150000   \n",
       "207506                100.0            1.729570            1.903670   \n",
       "\n",
       "        TPM_coefficient_of_quartile_variation  FPKM_ci_lower_bound  \\\n",
       "0                                    0.000000             0.000000   \n",
       "1                                    0.000000             0.000000   \n",
       "2                                    0.000000             0.000000   \n",
       "3                                    0.000000             0.000000   \n",
       "4                                    0.000000             0.000000   \n",
       "...                                       ...                  ...   \n",
       "207502                               0.038790             5.631410   \n",
       "207503                               0.386214             0.004742   \n",
       "207504                               0.093611             0.658386   \n",
       "207505                               0.007806           333.603000   \n",
       "207506                               0.016646             4.404830   \n",
       "\n",
       "        FPKM_ci_upper_bound  FPKM_coefficient_of_quartile_variation  \n",
       "0                  0.000000                                0.000000  \n",
       "1                  0.000000                                0.000000  \n",
       "2                  0.000000                                0.000000  \n",
       "3                  0.000000                                0.000000  \n",
       "4                  0.000000                                0.000000  \n",
       "...                     ...                                     ...  \n",
       "207502             7.065010                                0.038766  \n",
       "207503             0.109185                                0.386213  \n",
       "207504             1.143530                                0.093571  \n",
       "207505           349.115000                                0.007803  \n",
       "207506             4.848080                                0.016642  \n",
       "\n",
       "[207507 rows x 19 columns]"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tx_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "9db23975-bef9-4db4-be4f-8b5746b69c76",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = tx_tpm[['transcript_id','TPM']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "50c58e31-e41d-4253-8907-0c5427d57775",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 1. keep rows where gene_id starts with 'ENST'\n",
    "tx_tpm = tx_tpm[tx_tpm[\"transcript_id\"].str.startswith(\"ENST\", na=False)].copy()\n",
    "\n",
    "# 2. remove version suffix (e.g. .8)\n",
    "tx_tpm[\"transcript_id\"] = tx_tpm[\"transcript_id\"].str.replace(r\"\\.\\d+$\", \"\", regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "6f8308a9-d993-43fc-aa52-9d8821e6606d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>TPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>ENST00000373020</td>\n",
       "      <td>2.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>ENST00000494424</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>ENST00000496771</td>\n",
       "      <td>0.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>ENST00000612152</td>\n",
       "      <td>0.07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>ENST00000614008</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207405</th>\n",
       "      <td>ENST00000649331</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207406</th>\n",
       "      <td>ENST00000647612</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207407</th>\n",
       "      <td>ENST00000648949</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207408</th>\n",
       "      <td>ENST00000650266</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207409</th>\n",
       "      <td>ENST00000648650</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>206761 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          transcript_id   TPM\n",
       "649     ENST00000373020  2.92\n",
       "650     ENST00000494424  0.00\n",
       "651     ENST00000496771  0.54\n",
       "652     ENST00000612152  0.07\n",
       "653     ENST00000614008  0.00\n",
       "...                 ...   ...\n",
       "207405  ENST00000649331  0.00\n",
       "207406  ENST00000647612  0.00\n",
       "207407  ENST00000648949  0.00\n",
       "207408  ENST00000650266  0.00\n",
       "207409  ENST00000648650  0.00\n",
       "\n",
       "[206761 rows x 2 columns]"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tx_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "6e27d4f1-a2fd-4e08-bc8c-8595e4b1a1f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = tx_tpm.rename(columns={'TPM':\"TranscriptTPM_MT\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "48d8aa62-d137-4876-aa0d-52e334b36cfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df['Transcript'] = collapsed_df['Transcript'].str.split(\".\").str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "a580f019-333b-449d-9307-37d4248fb171",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df = collapsed_df.merge(\n",
    "    tx_tpm,\n",
    "    left_on=\"Transcript\",\n",
    "    right_on=\"transcript_id\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['transcript_id','gene_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "ce2c97df-8c1a-4b23-960f-91209a2c078d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "      <th>TranscriptTPM_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170652</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.30</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170653</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.30</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170654</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.30</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170655</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.30</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170656</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.30</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170657 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170652  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170653  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170654  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170655  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170656  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level             openChromatin_MT  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170652                       5.0                            .   \n",
       "170653                       1.0                            .   \n",
       "170654                       2.0                            .   \n",
       "170655                       1.0                            .   \n",
       "170656                       2.0                            .   \n",
       "\n",
       "                                              chromHMM_MT  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170652                                                  .   \n",
       "170653                                                  .   \n",
       "170654                                                  .   \n",
       "170655                                                  .   \n",
       "170656                                                  .   \n",
       "\n",
       "                                                   HiC_MT  GeneTPM_MT  \\\n",
       "0                                                       .        0.00   \n",
       "1                                                       .        0.00   \n",
       "2                                                       .        0.00   \n",
       "3       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.25   \n",
       "4       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.25   \n",
       "...                                                   ...         ...   \n",
       "170652                                                  .        2.30   \n",
       "170653                                                  .        2.30   \n",
       "170654                                                  .        2.30   \n",
       "170655                                                  .        2.30   \n",
       "170656                                                  .        2.30   \n",
       "\n",
       "        TranscriptTPM_MT  \n",
       "0                    0.0  \n",
       "1                    0.0  \n",
       "2                    0.0  \n",
       "3                    0.0  \n",
       "4                    0.0  \n",
       "...                  ...  \n",
       "170652               NaN  \n",
       "170653               NaN  \n",
       "170654               NaN  \n",
       "170655               NaN  \n",
       "170656               NaN  \n",
       "\n",
       "[170657 rows x 13 columns]"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "2650e004-85c0-41d5-b584-350c3a41e6c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TranscriptTPM_MT\n",
       "False    146277\n",
       "True      24380\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df['TranscriptTPM_MT'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f80d806-da69-4112-a3c3-48b3194fa03e",
   "metadata": {},
   "source": [
    "# update the promoter evidence Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "eb83069e-df7b-40c2-b1d1-029b1151cdc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define your conditions\n",
    "# very high: chromHMM is Tss/Prom and within openChromatin\n",
    "# high: chromHMM is Tss/Prom\n",
    "# moderate: RefTSS or CCRE\n",
    "# others: low\n",
    "cond1 = collapsed_df['chromHMM_myotube'].str.contains('Tss|Prom', na=False) & collapsed_df['openChromatin_myotube'].notna() & (collapsed_df['openChromatin_myotube'] != '.')\n",
    "cond2 = collapsed_df['chromHMM_myotube'].str.contains('Tss|Prom', na=False)\n",
    "cond3 = collapsed_df['RefTSS'].notna() | collapsed_df['CCRE'].notna()\n",
    "\n",
    "# Define corresponding values\n",
    "choices = ['very high', 'high', 'moderate']\n",
    "\n",
    "# Apply conditions to create a new 'Type' column\n",
    "collapsed_df['Type'] = np.select([cond1, cond2, cond3], choices, default='low')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "70b02bca-4278-481d-83dd-8a830487b7ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>RefTSS</th>\n",
       "      <th>CCRE</th>\n",
       "      <th>Type</th>\n",
       "      <th>openChromatin_myotube</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "      <th>HiC_myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65219</td>\n",
       "      <td>65619</td>\n",
       "      <td>ENST00000641515.2</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>68855</td>\n",
       "      <td>69255</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>69055</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451497</td>\n",
       "      <td>451897</td>\n",
       "      <td>ENST00000426406.3</td>\n",
       "      <td>451697</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686473</td>\n",
       "      <td>686873</td>\n",
       "      <td>ENST00000332831.4</td>\n",
       "      <td>686673</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923728</td>\n",
       "      <td>924128</td>\n",
       "      <td>ENST00000420190.6</td>\n",
       "      <td>923928</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_24.1:SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924950</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000437963.5</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000616016.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000616125.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000617307.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000618181.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000618323.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000618779.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000620200.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925583</td>\n",
       "      <td>925767</td>\n",
       "      <td>ENST00000622503.4</td>\n",
       "      <td>925741</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_28.1:SAMD11</td>\n",
       "      <td>chr1:925583:925767-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930112</td>\n",
       "      <td>930512</td>\n",
       "      <td>ENST00000341065.8</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:928020:933420:25_Quies</td>\n",
       "      <td>P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939075</td>\n",
       "      <td>939475</td>\n",
       "      <td>ENST00000455979.1</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_34.1:SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938220:939220:17_EnhW2;chr1:939220:939420...</td>\n",
       "      <td>P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940146</td>\n",
       "      <td>940546</td>\n",
       "      <td>ENST00000478729.1</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:939820:940220:23_PromBiv;chr1:940220:9404...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940876</td>\n",
       "      <td>941276</td>\n",
       "      <td>ENST00000474461.1</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:940420:941020:2_PromU;chr1:941020:941420:...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941903</td>\n",
       "      <td>942303</td>\n",
       "      <td>ENST00000466827.1</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_36.1:SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941966</td>\n",
       "      <td>942366</td>\n",
       "      <td>ENST00000464948.1</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_36.1:SAMD11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945362</td>\n",
       "      <td>945762</td>\n",
       "      <td>ENST00000496938.1</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946860</td>\n",
       "      <td>947260</td>\n",
       "      <td>ENST00000483767.5</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958258</td>\n",
       "      <td>958658</td>\n",
       "      <td>ENST00000477976.5</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:957820:958620:4_PromD2;chr1:958620:959220...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959147</td>\n",
       "      <td>959495</td>\n",
       "      <td>ENST00000327044.7</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9670.1:NOC2L</td>\n",
       "      <td>chr1:959147:959495-PLS,CTCF-bound</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959147</td>\n",
       "      <td>959495</td>\n",
       "      <td>ENST00000469563.1</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9670.1:NOC2L</td>\n",
       "      <td>chr1:959147:959495-PLS,CTCF-bound</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959147</td>\n",
       "      <td>959495</td>\n",
       "      <td>ENST00000487214.1</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9670.1:NOC2L</td>\n",
       "      <td>chr1:959147:959495-PLS,CTCF-bound</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960392</td>\n",
       "      <td>960733</td>\n",
       "      <td>ENST00000338591.8</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_41.1:KLHL17</td>\n",
       "      <td>chr1:960392:960733-PLS,CTCF-bound</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961220:...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960392</td>\n",
       "      <td>960733</td>\n",
       "      <td>ENST00000622660.1</td>\n",
       "      <td>960639</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_41.1:KLHL17</td>\n",
       "      <td>chr1:960392:960733-PLS,CTCF-bound</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961220:...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961249</td>\n",
       "      <td>961649</td>\n",
       "      <td>ENST00000463212.1</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:961220:961420:2_PromU;chr1:961420:961820:...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962527</td>\n",
       "      <td>962927</td>\n",
       "      <td>ENST00000466300.1</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:962820:7_Tx3';chr1:962820:965820:8...</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963352</td>\n",
       "      <td>963752</td>\n",
       "      <td>ENST00000481067.1</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962820:965820:8_TxWk</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966292</td>\n",
       "      <td>966617</td>\n",
       "      <td>ENST00000379407.7</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_42.1:PLEKHN1</td>\n",
       "      <td>chr1:966292:966617-PLS</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966292</td>\n",
       "      <td>966617</td>\n",
       "      <td>ENST00000379409.6</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_42.1:PLEKHN1</td>\n",
       "      <td>chr1:966292:966617-PLS</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966292</td>\n",
       "      <td>966617</td>\n",
       "      <td>ENST00000379410.8</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_42.1:PLEKHN1</td>\n",
       "      <td>chr1:966292:966617-PLS</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970675</td>\n",
       "      <td>971075</td>\n",
       "      <td>ENST00000480267.1</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:969220:972420:24_ReprPC</td>\n",
       "      <td>P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973312</td>\n",
       "      <td>973712</td>\n",
       "      <td>ENST00000491024.1</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972420:975820:25_Quies</td>\n",
       "      <td>P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976441</td>\n",
       "      <td>976841</td>\n",
       "      <td>ENST00000479361.1</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>chr1:976020:976620:16_EnhW1;chr1:976620:980820...</td>\n",
       "      <td>P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>980829</td>\n",
       "      <td>981229</td>\n",
       "      <td>ENST00000433179.3</td>\n",
       "      <td>981029</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>low</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981893</td>\n",
       "      <td>982293</td>\n",
       "      <td>ENST00000341290.6</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9675.1:PERM1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999856</td>\n",
       "      <td>1000409</td>\n",
       "      <td>ENST00000304952.11</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9681.1:HES4</td>\n",
       "      <td>chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999856</td>\n",
       "      <td>1000409</td>\n",
       "      <td>ENST00000428771.6</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9682.1:HES4</td>\n",
       "      <td>chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999856</td>\n",
       "      <td>1000409</td>\n",
       "      <td>ENST00000481869.1</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9681.1:HES4</td>\n",
       "      <td>chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999856</td>\n",
       "      <td>1000409</td>\n",
       "      <td>ENST00000484667.2</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>hg_9681.1:HES4</td>\n",
       "      <td>chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1001102</td>\n",
       "      <td>1001426</td>\n",
       "      <td>ENST00000624652.1</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_44.1:ISG15</td>\n",
       "      <td>chr1:1001102:1001426-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1001102</td>\n",
       "      <td>1001426</td>\n",
       "      <td>ENST00000624697.4</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_44.1:ISG15</td>\n",
       "      <td>chr1:1001102:1001426-PLS,CTCF-bound</td>\n",
       "      <td>high</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013312</td>\n",
       "      <td>1013662</td>\n",
       "      <td>ENST00000649529.1</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_232494.1:NA</td>\n",
       "      <td>chr1:1013312:1013662-PLS,CTCF-bound</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10142...</td>\n",
       "      <td>P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019920</td>\n",
       "      <td>1020401</td>\n",
       "      <td>ENST00000379370.7</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_50.1:AGRN</td>\n",
       "      <td>chr1:1020056:1020215-PLS,CTCF-bound;chr1:10202...</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019923</td>\n",
       "      <td>1020401</td>\n",
       "      <td>ENST00000620552.4</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_50.1:AGRN</td>\n",
       "      <td>chr1:1020056:1020215-PLS,CTCF-bound;chr1:10202...</td>\n",
       "      <td>very high</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033906</td>\n",
       "      <td>1034306</td>\n",
       "      <td>ENST00000651234.1</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>hg_223441.1:AGRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end          Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65219    65619   ENST00000641515.2    65419    OR4F5      +   \n",
       "1   chr1    68855    69255   ENST00000335137.4    69055    OR4F5      +   \n",
       "2   chr1   451497   451897   ENST00000426406.3   451697   OR4F29      -   \n",
       "3   chr1   686473   686873   ENST00000332831.4   686673   OR4F16      -   \n",
       "4   chr1   923728   924128   ENST00000420190.6   923928   SAMD11      +   \n",
       "5   chr1   924950   925767   ENST00000437963.5   925150   SAMD11      +   \n",
       "6   chr1   925583   925767   ENST00000342066.8   925731   SAMD11      +   \n",
       "7   chr1   925583   925767   ENST00000616016.4   925741   SAMD11      +   \n",
       "8   chr1   925583   925767   ENST00000616125.4   925741   SAMD11      +   \n",
       "9   chr1   925583   925767   ENST00000617307.4   925741   SAMD11      +   \n",
       "10  chr1   925583   925767   ENST00000618181.4   925741   SAMD11      +   \n",
       "11  chr1   925583   925767   ENST00000618323.4   925741   SAMD11      +   \n",
       "12  chr1   925583   925767   ENST00000618779.4   925741   SAMD11      +   \n",
       "13  chr1   925583   925767   ENST00000620200.4   925741   SAMD11      +   \n",
       "14  chr1   925583   925767   ENST00000622503.4   925741   SAMD11      +   \n",
       "15  chr1   930112   930512   ENST00000341065.8   930312   SAMD11      +   \n",
       "16  chr1   939075   939475   ENST00000455979.1   939275   SAMD11      +   \n",
       "17  chr1   940146   940546   ENST00000478729.1   940346   SAMD11      +   \n",
       "18  chr1   940876   941276   ENST00000474461.1   941076   SAMD11      +   \n",
       "19  chr1   941903   942303   ENST00000466827.1   942103   SAMD11      +   \n",
       "20  chr1   941966   942366   ENST00000464948.1   942166   SAMD11      +   \n",
       "21  chr1   945362   945762   ENST00000496938.1   945562    NOC2L      -   \n",
       "22  chr1   946860   947260   ENST00000483767.5   947060    NOC2L      -   \n",
       "23  chr1   958258   958658   ENST00000477976.5   958458    NOC2L      -   \n",
       "24  chr1   959147   959495   ENST00000327044.7   959256    NOC2L      -   \n",
       "25  chr1   959147   959495   ENST00000469563.1   959256    NOC2L      -   \n",
       "26  chr1   959147   959495   ENST00000487214.1   959309    NOC2L      -   \n",
       "27  chr1   960392   960733   ENST00000338591.8   960584   KLHL17      +   \n",
       "28  chr1   960392   960733   ENST00000622660.1   960639   KLHL17      +   \n",
       "29  chr1   961249   961649   ENST00000463212.1   961449   KLHL17      +   \n",
       "30  chr1   962527   962927   ENST00000466300.1   962727   KLHL17      +   \n",
       "31  chr1   963352   963752   ENST00000481067.1   963552   KLHL17      +   \n",
       "32  chr1   966292   966617   ENST00000379407.7   966502  PLEKHN1      +   \n",
       "33  chr1   966292   966617   ENST00000379409.6   966502  PLEKHN1      +   \n",
       "34  chr1   966292   966617   ENST00000379410.8   966482  PLEKHN1      +   \n",
       "35  chr1   970675   971075   ENST00000480267.1   970875  PLEKHN1      +   \n",
       "36  chr1   973312   973712   ENST00000491024.1   973512  PLEKHN1      +   \n",
       "37  chr1   976441   976841   ENST00000479361.1   976641    PERM1      -   \n",
       "38  chr1   980829   981229   ENST00000433179.3   981029    PERM1      -   \n",
       "39  chr1   981893   982293   ENST00000341290.6   982093    PERM1      -   \n",
       "40  chr1   999856  1000409  ENST00000304952.11  1000097     HES4      -   \n",
       "41  chr1   999856  1000409   ENST00000428771.6  1000172     HES4      -   \n",
       "42  chr1   999856  1000409   ENST00000481869.1  1000096     HES4      -   \n",
       "43  chr1   999856  1000409   ENST00000484667.2   999981     HES4      -   \n",
       "44  chr1  1001102  1001426   ENST00000624652.1  1001145    ISG15      +   \n",
       "45  chr1  1001102  1001426   ENST00000624697.4  1001138    ISG15      +   \n",
       "46  chr1  1013312  1013662   ENST00000649529.1  1013497    ISG15      +   \n",
       "47  chr1  1019920  1020401   ENST00000379370.7  1020120     AGRN      +   \n",
       "48  chr1  1019923  1020401   ENST00000620552.4  1020123     AGRN      +   \n",
       "49  chr1  1033906  1034306   ENST00000651234.1  1034106     AGRN      +   \n",
       "\n",
       "              RefTSS                                               CCRE  \\\n",
       "0                NaN                                                NaN   \n",
       "1                NaN                                                NaN   \n",
       "2                NaN                                                NaN   \n",
       "3                NaN                                                NaN   \n",
       "4     hg_24.1:SAMD11                                                NaN   \n",
       "5                NaN                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "6     hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "7     hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "8     hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "9     hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "10    hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "11    hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "12    hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "13    hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "14    hg_28.1:SAMD11                  chr1:925583:925767-PLS,CTCF-bound   \n",
       "15               NaN                                                NaN   \n",
       "16    hg_34.1:SAMD11                                                NaN   \n",
       "17               NaN                                                NaN   \n",
       "18               NaN                                                NaN   \n",
       "19    hg_36.1:SAMD11                                                NaN   \n",
       "20    hg_36.1:SAMD11                                                NaN   \n",
       "21               NaN                                                NaN   \n",
       "22               NaN                                                NaN   \n",
       "23               NaN                                                NaN   \n",
       "24   hg_9670.1:NOC2L                  chr1:959147:959495-PLS,CTCF-bound   \n",
       "25   hg_9670.1:NOC2L                  chr1:959147:959495-PLS,CTCF-bound   \n",
       "26   hg_9670.1:NOC2L                  chr1:959147:959495-PLS,CTCF-bound   \n",
       "27    hg_41.1:KLHL17                  chr1:960392:960733-PLS,CTCF-bound   \n",
       "28    hg_41.1:KLHL17                  chr1:960392:960733-PLS,CTCF-bound   \n",
       "29               NaN                                                NaN   \n",
       "30               NaN                                                NaN   \n",
       "31               NaN                                                NaN   \n",
       "32   hg_42.1:PLEKHN1                             chr1:966292:966617-PLS   \n",
       "33   hg_42.1:PLEKHN1                             chr1:966292:966617-PLS   \n",
       "34   hg_42.1:PLEKHN1                             chr1:966292:966617-PLS   \n",
       "35               NaN                                                NaN   \n",
       "36               NaN                                                NaN   \n",
       "37               NaN                                                NaN   \n",
       "38               NaN                                                NaN   \n",
       "39   hg_9675.1:PERM1                                                NaN   \n",
       "40    hg_9681.1:HES4  chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...   \n",
       "41    hg_9682.1:HES4  chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...   \n",
       "42    hg_9681.1:HES4  chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...   \n",
       "43    hg_9681.1:HES4  chr1:999856:1000009-PLS,CTCF-bound;chr1:100006...   \n",
       "44     hg_44.1:ISG15                chr1:1001102:1001426-PLS,CTCF-bound   \n",
       "45     hg_44.1:ISG15                chr1:1001102:1001426-PLS,CTCF-bound   \n",
       "46    hg_232494.1:NA                chr1:1013312:1013662-PLS,CTCF-bound   \n",
       "47      hg_50.1:AGRN  chr1:1020056:1020215-PLS,CTCF-bound;chr1:10202...   \n",
       "48      hg_50.1:AGRN  chr1:1020056:1020215-PLS,CTCF-bound;chr1:10202...   \n",
       "49  hg_223441.1:AGRN                                                NaN   \n",
       "\n",
       "         Type                              openChromatin_myotube  \\\n",
       "0         low                                                  .   \n",
       "1         low                                                  .   \n",
       "2         low                                                  .   \n",
       "3         low                                                  .   \n",
       "4   very high                        ATAC-seq:chr1:923647-924887   \n",
       "5        high                                                  .   \n",
       "6        high                                                  .   \n",
       "7        high                                                  .   \n",
       "8        high                                                  .   \n",
       "9        high                                                  .   \n",
       "10       high                                                  .   \n",
       "11       high                                                  .   \n",
       "12       high                                                  .   \n",
       "13       high                                                  .   \n",
       "14       high                                                  .   \n",
       "15        low                                                  .   \n",
       "16   moderate                                                  .   \n",
       "17  very high                        ATAC-seq:chr1:940001-941223   \n",
       "18  very high                        ATAC-seq:chr1:940001-941223   \n",
       "19       high                                                  .   \n",
       "20       high                                                  .   \n",
       "21        low                                                  .   \n",
       "22        low                                                  .   \n",
       "23       high                                                  .   \n",
       "24  very high  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "25  very high  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "26  very high  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "27  very high                        ATAC-seq:chr1:960473-961422   \n",
       "28  very high                        ATAC-seq:chr1:960473-961422   \n",
       "29  very high                        ATAC-seq:chr1:960473-961422   \n",
       "30        low                                                  .   \n",
       "31        low                                                  .   \n",
       "32  very high                        ATAC-seq:chr1:966276-967323   \n",
       "33  very high                        ATAC-seq:chr1:966276-967323   \n",
       "34  very high                        ATAC-seq:chr1:966276-967323   \n",
       "35        low                                                  .   \n",
       "36        low                                                  .   \n",
       "37        low                        ATAC-seq:chr1:976006-976687   \n",
       "38        low                                                  .   \n",
       "39   moderate                                                  .   \n",
       "40  very high  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "41  very high  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "42  very high  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "43  very high  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "44       high                                                  .   \n",
       "45       high                                                  .   \n",
       "46  very high                      ATAC-seq:chr1:1012923-1014210   \n",
       "47  very high                      ATAC-seq:chr1:1019149-1020697   \n",
       "48  very high                      ATAC-seq:chr1:1019149-1020697   \n",
       "49   moderate                                                  .   \n",
       "\n",
       "                                     chromHMM_myotube  \\\n",
       "0                           chr1:10600:96080:25_Quies   \n",
       "1                           chr1:10600:96080:25_Quies   \n",
       "2                         chr1:348136:501536:25_Quies   \n",
       "3                         chr1:634820:733820:25_Quies   \n",
       "4   chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...   \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "7   chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "8   chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "9   chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "10  chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "11  chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "12  chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "13  chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "14  chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "15                        chr1:928020:933420:25_Quies   \n",
       "16  chr1:938220:939220:17_EnhW2;chr1:939220:939420...   \n",
       "17  chr1:939820:940220:23_PromBiv;chr1:940220:9404...   \n",
       "18  chr1:940420:941020:2_PromU;chr1:941020:941420:...   \n",
       "19                        chr1:941620:942620:22_PromP   \n",
       "20                        chr1:941620:942620:22_PromP   \n",
       "21                          chr1:945220:950020:7_Tx3'   \n",
       "22                          chr1:945220:950020:7_Tx3'   \n",
       "23  chr1:957820:958620:4_PromD2;chr1:958620:959220...   \n",
       "24  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "25  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "26  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "27  chr1:960220:960420:2_PromU;chr1:960420:961220:...   \n",
       "28  chr1:960220:960420:2_PromU;chr1:960420:961220:...   \n",
       "29  chr1:961220:961420:2_PromU;chr1:961420:961820:...   \n",
       "30  chr1:962420:962820:7_Tx3';chr1:962820:965820:8...   \n",
       "31                          chr1:962820:965820:8_TxWk   \n",
       "32                         chr1:966220:967220:2_PromU   \n",
       "33                         chr1:966220:967220:2_PromU   \n",
       "34                         chr1:966220:967220:2_PromU   \n",
       "35                       chr1:969220:972420:24_ReprPC   \n",
       "36                        chr1:972420:975820:25_Quies   \n",
       "37  chr1:976020:976620:16_EnhW1;chr1:976620:980820...   \n",
       "38                       chr1:980820:984220:24_ReprPC   \n",
       "39                       chr1:980820:984220:24_ReprPC   \n",
       "40  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "41  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "42  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "43  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "44                       chr1:1000420:1002020:2_PromU   \n",
       "45                       chr1:1000420:1002020:2_PromU   \n",
       "46  chr1:1013220:1013420:1_TssA;chr1:1013420:10142...   \n",
       "47  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "48  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "49                     chr1:1033420:1034620:24_ReprPC   \n",
       "\n",
       "                                          HiC_myotube  \n",
       "0                                                   .  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3                                                   .  \n",
       "4   P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "5   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "6   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "7   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "8   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "9   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "10  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "11  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "12  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "13  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "14  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "15  P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...  \n",
       "16  P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...  \n",
       "17  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "18  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "19  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "20  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "21  P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...  \n",
       "22  P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...  \n",
       "23  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "24  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "25  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "26  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "27  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...  \n",
       "28  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...  \n",
       "29  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...  \n",
       "30  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...  \n",
       "31  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...  \n",
       "32  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "33  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "34  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "35  P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...  \n",
       "36  P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...  \n",
       "37  P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...  \n",
       "38  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "39  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "40  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "41  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "42  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "43  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "44  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "45  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "46  P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...  \n",
       "47  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "48  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "49  P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "1f6223fb-d6ca-4185-bee6-831ea759c33b",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_HiC_level.tsv',sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "cbb225f2-fa98-49ac-b82e-d58f7874eaf6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Type\n",
      "very high    47.615969\n",
      "low          27.003284\n",
      "moderate     18.210555\n",
      "high          7.170192\n",
      "Name: proportion, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "type_percentages = collapsed_df['Type'].value_counts(normalize=True) * 100\n",
    "print(type_percentages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "fa24f6c9-e9e9-4e1f-854b-e921276e2d5c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAtIAAAInCAYAAAC1Nb4AAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjgsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvwVt1zgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAYyxJREFUeJzt3QeYU2X6//+b3qtSfoyiFEFApLhSVimCiEtZYFlckCYCSlFRAUEpCiLFpQiIYMFdOqgrrCKorEhRhN0vTRQsFKXI0qRKL//r83z/J98kk2lnBqbk/bquuTJz8uQkk8kkn9y5z/NkunLlyhUDAAAAkCSZkzYcAAAAAEEaAAAA8ImKNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAp4v3337fy5csHvipXrmx33XWXdezY0V5//XU7cuRIrMtMnjzZjU2KM2fOuMutW7cuSZeLdF0NGjSwRx991FLShx9+aH//+98jnqfr1+1Iy7766iv705/+ZFWrVnW391//+lesMfqbBv+t4/pKi7/r//zP/1ifPn2sTp06dtttt9kdd9xhbdu2tblz59rp06dT9O+dFun/Rn+bpP7/AIgsaxzbAcCXUaNGWenSpe3ixYsuPK9fv97efPNNe/vtt23ChAn2+9//PjC2TZs2LtAkNUi/+uqr9thjj1nNmjUTfTk/1+XH4sWL7ccff7SHHnoo1nkLFiyw4sWLW1p15coVe/LJJ+3mm2+2qVOnWq5cuaxUqVKxxj3//PN26tSpwM8rVqxw472/vSet/a6TJk2yKVOmWLVq1VyYLlmypHs8bdy40T2mfvrpJ3vuuedS7O+dFlWqVMk9DsuWLZvaNwXIEAjSAFLULbfc4qrRnsaNG7uQ8eCDD7rw++mnn9r1118fCFpXO2wpKCkQXovrSoiqvGnZwYMH7dixY3bvvfda7dq14xwXHsJ27twZ8W+flixdutSF6D//+c82YsQIy5QpU+C8evXqWbdu3WzTpk2WUV24cMH9znnz5k3zj0MgPaG1A8BVV6JECRswYID99ttvNn/+/HjbLdRaoNYBVZtvv/12q1+/vj3++OMuEO/duzcQ8FRB9FoIBg4cGLK/b7/91p544gm78847rVGjRnFel2fZsmXWvHlzFwIbNmxoM2fOjNi2ouuP72Ny3W5VZ/ft2xfS4uCJ1O7www8/WM+ePd1t1fW3aNHCFi5cGPF6VP1UVf/uu++26tWruzcoXohNTEtD586dXTW2SpUqrp1BtzX4b1G3bl33/dixY931qfXFj0WLFrnLq9IbTn83VUUPHDgQuM+aNWvmbt8DDzzg/ub65OCVV16xS5cuhVz2/Pnz9tprr9n999/v2jJq1aplzz77rP36668J3iZdrkCBAjZ48OCQEO1RwNT96pkzZ461b9/ePd4UPPX40CcrCqSehP7eib29Gjd69GjXCqW/ja73m2++cfe/99j283jR30H71f2psT///HOcrR1btmyxHj16WI0aNdzYli1b2pIlS0LG6H9wzJgx7nZpjMaqDUiPSyBaUZEGcE2o6pclSxYXmOKioKqe5d/97nf20ksvWf78+V3gWr16tQswRYsWtbfeestVD1VZVLuGFC5cOGQ/Ct5NmjRxYTGhvtdt27bZyJEjXbVclXL1vOq6dX1du3ZN0u+olochQ4bYnj17XGBMiEKwbuN1111ngwYNskKFCtkHH3zgwtPhw4ete/fuIePHjx/vArRun1orFHgVqhR4dN/G5d///rc9/PDDVq5cOXfZ7Nmz27x581xw0j51X+m+vPXWW9394IVbjfND+9NtUxhVcPeo3UdtBap4FytWLLD90KFD9tRTT9kjjzzi3gB5rSInTpywoUOHujGXL1+2Xr16uVYh/V10PyjA6g3A119/bf/4xz8sZ86ccVbaFUB1u/TpRGLs3r3b3Qc33HCDZcuWzb777jubNm2a+5uphSWhv3dSbq/Ctf6GelwrbG/fvt39HYLbZ/w+XvQmYNiwYZY5c2Z3OY0Lt3btWnfdCvEvvPCC5cuXz90e/U3Onj3rwrLo99b1qf2nQoUKLljrftWnGEC0IkgDuCZy587tXvgVauKiSvK5c+fsmWeecaHOo2qgR9VMUZtGXB9Rq5qmQJYYuj2q3HnXp8CviqEqiWpHSWzw8loeFP4VQBPz8bnClwK7KuD/7//9v8D1K0CqDUGhSaEmeP8KqB6FI4UaVRPju75x48a52zVr1izLkyeP23bPPfe4+0kVxj/84Q/u/lTQFd2W5Hz8r9//L3/5izvIVCFRAU7U1qP7u0OHDiHjFcR0f+vTAFFlWI8DhX0FPH2iodYMvaFSEL3vvvsCl9XfTW+q9KmB/l6R7N+/350qFCeWbndwKNabu4IFC7oeagVXVbfj+3sn9vYqNKuiqxDct29fN0aVab2pe/rpp5P1eFEPuPrCE6KgrbacGTNmWNas/xsLVMU+evSoC+N6nOixpk8YdNuC+8H1iREQzWjtAHBND2aLj6pcqv6pyqePq1Xp8yM4uCREASI4tIsqkaoGKthfTaoEqnXAC0WeVq1aBQ6CCxbeauG1Efzyyy9xXocq8ps3b3a96l6IFlWw//jHP9p///vfRLeHJEW7du3c6TvvvBPYpgq1quJqSwim2+WF6OC/gQLsf/7zH/fz559/7kKr3gAo8HtfeswUKVLEVd1T0tatW13FXi1Gug69gVN7ktpNdFBiQhJ7e71TvZkJpr+XF2r9Pl4S83+gdg/9/b03q8G3Va0++rRg165d7jy1c6xatcq9mVNriKrVQLSjIg3gmlCgU+VRQSouqqBpKjG1bwwfPtxd5sYbb3StBurvTSy1gCSWd+BjpG1X+yNr7V+hKq7bH379qogG81ov4gs0qlbqDUxSricl6D5UK4VaOdSyoZkt1Najv2uksXFt826bZoDR76Je40hUPY2LFzzDe9zjojcm6lPWjCWqQMfExFiOHDlcS4Zuf2ICZGJvr/f7hd8HCtHhf++kPl4ijQ3ntXrokwl9xXdb1V+uTy7U9qF+cd0n+vRAnyBpphcgGhGkAVwT6ntVNU8HKMVHH6HrS2N1wJXaEdTDrKDRtGnTFL9dkXpGvW1ekFFg8A4KS2x4SwztXxW/cF77i1phkktVUX0sf7WvJ5JOnTrZP//5T/vss89cm4NuS3CbTlL+BrqN+l5vsiIJrrZHCpp6A/fll18GZnGJj+bO1ps4tWUoRHvUJ51Yib293u+n3ze4b1wV4UhvpJLyd4x0UGWk2yk6NsE7MDecNwWi2rPUMqUv3V5Vp9U2pMr9xx9/nOB1ARkRrR0ArjpV+F5++WXXv6k+zsRQ64EOftIBXeK1WSSmCpsUqpSGByT1rCroeP3YXpj6/vvvQ8YtX7481v50+xJ72/QxvT6u92aw8Ch8KuylxDRlCj+6HzUzSfDtUtuEDhxThTHSXNEpQdVYHWyo6qUO4lQLgm5POM3morAd/jfQGwCvDUS9uAqWut1qMQj/Cp6/OhId+Hf8+HE39V2kFiPdhi+++CIkgAYfbKnLBLepJPT3Tuzt9X6/8BkyPvnkk0DP+tV8vOh2qJqs/4FIt1NfmtEknN7Y6iBEvblV64feoADRiIo0gBSlYKpqskKADtrTx/k6sErBWAdLhc+wEUwHlykoKITo43gdcKbZDcRbyEUv6gq2Cl4KFjroS1W1pBxIFl6t1MwXmiVBH4UrXKpy2a9fv0DlUmFCYVNvBvS7qbKqqqVmZAinyqcOqtNKeQqSCmVxza3cu3dv10uryq2+1++iwKnqff/+/UMOHEsOHbSmWTt0PTpVH7pun/5WOpgsMZVLv3Sdmv1B1xHXwYCqtGq2CB0UqFC3cuVKF1rVZ60DDUWBTfeN2kTU6qNp8vR7qMdb/brqsY6rour1IGuGCR3UqJ5gHfDnLciilg1Ny6hWFLUq6LGmfet+08GO+iRCj021aiT2753Y26seffWD/+1vf3P/I5q1Q38X/ay/f/Df5mo9XnSwoQ521OwierOjyrjedOzYscO9gfUOWNTMLvrfVG++rlvnK8TrzVJSDsoFMhKCNIAU5c12oNCgwFmmTBn3Iq0X4fhCtOhALIVYfaSuj7BVvVRQ0VRowXP8ago3hVoFYIUcvfhrvlw/dJ2qrOk6dRCZgrV+h+CZCRRwNPXZiy++6CrkqkIqKGlqNgWlYAo5CkKa7/nkyZOukhleyQ6uBirAKcx6vbe6vzTNmDflWEpQO416z/U76ndTlVQHWOp+1cFwV5OmutP9pYP24uqj1RsY3Zfq0VXYVUhTu4CmMQz+G+j2asYKhbc33njDbVNFXVXd+HrvPVrNUCFZ7UL6+6hirLYdhdkuXbq4mUZEfwPdV5rLWrdBQV9hV4+J8Cnm4vp7J+X26u+t++C9995zfyc9JnXdCvH6H7rajxeF93fffdc9xtVGpTcM+p217+CDIDVOn8Jodg+9AVHg1owe+lsB0SrTlYQOowcAwCcFL73hUZDUVG3hVK1VrzmLeoTasGGDq8hrhoxIfeUA0gYq0gCAFKf5kbUAiarMqrB6qyYiNn0Ko6nr1BqiCrkq2nrjoQp+UqZyBHDtEaQBAClOfbeqqlasWNG13VzNPuz0Tn3/CtNqA9FBj+r51xsP9Wh7M8YASJto7QAAAAB8YPo7AAAAwAeCNAAAAOADQRoAAADwgYMNrzEdma0ZBzXHLgAAANKeCxcuuIOkteBQmq9Ia4UwLdZQvXp1N1m+VhjTylPhtNqVJn/XqlFaEWrOnDkR9zd9+nRr0KCBG9e6dWu3ilS4U6dOuQUAtEiA7iRNKK+pmsJp6VOt9qSlV7WKmpaXTc7SxArRTN0NAACQdiU2r6X6rB1r1qxxS9b+8Y9/tBYtWrgVlbSMsILuRx995KYF8iq5HTp0cGM0VtMqaeUpre6kEB4corXClJak1bRLWq1JQV2nWtbU8+ijj7qlTwcOHOiuQ0ug6jq1PHDOnDndGN0WrWalJWp79erlljvWClJ16tRxk+T7sWXLFnca15LBAAAASF2JzWup3tqhsKygqkn7vXlGY2JiXDhev359YCWsKVOmuGCs5Uu9pUr3799vEydOdFXnzJkzu6WCtSSrlmxVFdlbGlerQmnpUwVs2bx5s61YsSJkpS0t16oq98KFC91qUqKlWBWmFy1aFFjaWEu89uvXz63UpeVTAQAAEJ1SvbXj4sWLlidPnpDJ+vPlyxcyRgF57dq11rRp05DtCsiHDh2yrVu3up9VpT558qSrInsUfJs0aeLaQrziu77Pnz9/yEpbCvNqLdF5nlWrVrl2Di9ES+PGjS179uwh4wAAABB9Uj1I//nPf3b90LNmzXLV371797rqtKq9CrGye/du1/RdunTpkMuWLVvWne7YsSPkNHyc9qXVog4cOBAYV6pUqVgrbWl/3j68ceFVZ4XokiVLhowDAABA9En11o4777zT9UT37dvXHcjnBdq3337bhVY5fvy4O1UVOZj3s3e+grgu4/U4ewoUKOBOjx07ZsWLF3fjwqve3v68fXn7C7/OSOOSSpXx06dP+748AAAArh5ltfCCa5oM0mrH6N+/v+tz1kwbOuBP/czdu3e3efPmBQ42lLh+oeDtkcZ4LR0JjYtve/j+EjMuLqqub9u2zfflAQAAcHV5Bd00HaRVhdaBg4MGDQpsu+OOO1z/smba6NKlS6CiHF4FVsVYvKqxTs+dO+e+cuTIEWuctx+N04GK4cIr0Preu2ww9WEn50BDzSHttaUAAAAgbdm+fXuixqV6kFavsSrRwXRwX9GiRV1vtKgnWeFTvdTBBwh6v6QXar1T7VMzfARfhw5oLFasWGCcpt0Lryxrf8EBWd+H90LrwEfdLlXQ/dJ15s6d2/flAQAAcPUktvMg1Q821GwZms85mGbiOHjwoJsGzyutq2q9dOnSkHGLFy+2IkWKBEKzZt1Q7/OSJUsCYy5duuQup2nuvDtF36vSvHr16sA4VajVZuJNhycK7Zot5OjRo4Fty5Ytc2E6eBwAAACiT6pXpNu3b28vvviiW1ilYcOGLuC+/vrrrmKrhVc8vXv3dguyDB482E17p9Cr1g9dTnNIe4Fb8ztrvmhVtb0FWfbs2WPjx48P7KtKlSpWv359107iLcii+agV3Fu1ahUY17ZtW5s9e7ZbjEVfR44csdGjR7vrZw5pAACA6JbqKxvq6t955x2bO3eua5lQgNYqMlqZMHglQtHczQrEarfQ7Bvqn1YQD9+fVjfU8uGHDx92C63oYEZVtIPpoEZNs/fJJ5+4g/+0VPiQIUMCVfDgJcLVx63FYTQbiOao1oIs4TODJBYrGwIAAKRtic1rqR6kow1BGgAAIGPktVTvkQYAAADSI4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKRxVfz2229Wt25dK1++vG3ZsiWwXT/H9XXw4MEE9/vjjz9ajx497I477rBq1arZn/70J9uwYUPg/MOHD1u3bt2sevXq1q5dO/v5559DLn/s2DGrXbu2ffvttyn8GwMAgGiTNbVvADKm1157zS5duhRr+4IFC2JtGzBggOXKlcuKFi0a7z6/++47a9++vdWvX9/Gjx9vWbNmdYH47NmzgTEjR4501ztp0iR78803beDAgTZv3rzA+a+88orde++9VqlSpWT/jgAAILoRpJHiduzYYXPnznUB+fnnnw85r2rVqiE/792713766Sfr379/gvt94YUXXIgeN25cYNtdd90VMubLL790Afr222+3fPny2QMPPGCnT5+23LlzuyC+dOlS9wUAAJBctHYgxb300kvWtm1bK1WqVIJjFy9ebJkyZbJmzZolGM43btxoHTp0iHfchQsXLGfOnO5771TbvNvVu3dvK1y4cBJ+GwAAgMgI0khRH3/8sav8KrAmxkcffWR33nmnFS9ePN5xmzZtcqcnT560Fi1aWMWKFa1BgwY2a9askHGVK1d21fDjx4/bnDlzrGTJklagQAF3PUePHrUHH3wwGb8dAADA/yFII8WcOXPGRo8ebU8//bTlzZs3wfEK3D/88EOC1WjvIEJRC0iTJk3s7bffdr3OI0aMsA8++CAwTu0k//rXv6xGjRq2ZMkSGz58uLtdf/3rX23QoEGurxoAACAlkCqQYqZOnWrXXXedm0kjMT788EPLli2bNW7cOMGxly9fdqetW7e2Rx991H1fq1Yt2717t02bNs3++Mc/um2qVH/++ee2Z88ei4mJsRw5ctiECRNcz7Rm69B5CvsK5vfcc48NGzbM8uTJk6zfGwAARCcq0kgR+/btc1XiJ554wk6dOmUnTpxwB/mJTjUdXrArV664inGdOnWsYMGCCe5f7RleeA6mn3WwotcHLQrnpUuXdiFagVqzdqhSfeTIEVctf+yxx2zZsmW2a9cuF/4BAAD8oCKNFKHZNxRmH3nkkVjnderUyapUqWLvvPNOYNv69evtl19+SdRsHVKmTJk4z8ucObM7YDGSUaNGWceOHV11+rPPPrPs2bNb8+bN3XmtWrWy999/P1HXDwAAEI4gjRRRoUIFmzlzZsi2bdu2uSCr9gkdBBje1qEp6dRekRhafEVV6a+++sot9OLRzwrZkXqfNRWeboPmnPYo7Gue6SxZsgQq5gAAAH4QpJEi8ufPbzVr1ox4nhY/CV4A5eLFi/bJJ5+4gwW1EEskjRo1shIlStiMGTPcz6ok9+rVy8aOHevmh1aFW/3OK1assClTpsS6vK5D092ppcObBk9zWKvXWvvQwYia3UMHLgIAAPhBkMY198UXX7ip6OKbrUNVY+8AQ89DDz3kWjhU+dbKiTfeeKONGTPGBfJws2fPtuuvv97uv//+wDYdCKnFXF5++WXXZqLFXRTOAQAA/Mh0RUd94ZrZsmWLOw1vdQAAAED6ymvM2gEAAAD4QJAGAAAAfCBIpzOXL9OJE034ewMAkHZxsGE6kzlzJpsy70vbd/B4at8UXGUxRQtY73Z3cT8DAJBGEaTTIYXon/YdTe2bAQAAENVo7QAAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAABAegzSHTt2tPLly0f8+uijjwLjVq5caS1btrTKlStbo0aNbM6cORH3N336dGvQoIEb17p1a1u3bl2sMadOnbKhQ4dazZo1rVq1atajRw/bt29frHG7du2yrl27WtWqVa127do2YsQIO3v2bArfAwAAAEiPsqb2DXj++eddsA02Y8YM+/TTT114lY0bN1qvXr2sRYsWNnDgQNuwYYMLtdmzZ7c2bdqEhOgJEybYU089ZRUrVrR3333Xunfv7k4VzD19+/a1b7/91oYMGWJ58+a1SZMmWZcuXeyDDz6wnDlzujEnTpywzp07W4kSJdz5v/76q40aNcqOHTtmY8eOvWb3DwAAANKmVA/SZcuWjbVNQfeuu+6ywoULu5+nTJnigvHIkSPdz7Vq1bL9+/fbxIkTXdU5c+bMdv78eZs6dap16tTJVZGlRo0a1rx5c5s2bZoL2LJ582ZbsWKFvfHGG1avXj23rVy5cq7KvXDhQmvXrp3bNn/+fBemFy1aFLgdWbJksX79+lnPnj2tTJky1+geAgAAQFqU6q0d4VRt3rt3rwvAooC8du1aa9q0acg4nX/o0CHbunVr4HInT560Zs2aBcYo+DZp0sS1hVy5csVt0/f58+e3unXrBsap6ly9enV3nmfVqlWuIu6FaGncuLGrggePAwAAQHRKc0F68eLFlitXLmvYsKH7effu3XbhwgUrXbp0xEr2jh07Qk7Dx6ly/Ntvv9mBAwcC40qVKmWZMmWKtT9vH9648KqzQnTJkiVDxgEAACA6pXprR7CLFy/axx9/7EJ07ty53bbjx4+7U1WRg3k/e+erDUNB1+tx9hQoUMCdqre5ePHibly+fPliXbf25+3L21/4dUYa54eq46dPn07y5RT+9SYD0eXMmTOBT1QAAMDVp9fd8KJrmg/SX375pR05ciSkPcMT1y8TvD3SGC+AJDQuvu3h+0vMuPiowr5t27YkX04hWr3iiC6aPUZhGgAAXDsq0KarIK22joIFC9rdd98dq6IcXgVWxVi8qrFOz507575y5MgRa5y3H43TgYrhwivQ+t67bDD1YSf3QMNs2bJFPMgyIckN8Eif1IpERRoAgGtn+/btiRqXZoK05mf+7LPP3EGECpoe9STr5507d4YcIOj9gl6o9U7VvxxctdXPefLksWLFigXGrVmzJlZlWfsLDsj6PrwXWgc+qmdbM4Ukh67Xa10BEkI7DwAA11Zii5dp5mDD5cuXu4MCvdk6gsvqmu5u6dKlsarXRYoUCYRmzbqh3uclS5YExly6dMldTtPceXeIvlelefXq1YFxqlBr1g9vOjxRaNdsIUePHg1sW7ZsmQvTweMAAAAQndJMRfrDDz9009Ddcccdsc7r3bu3dejQwQYPHuyCtkKvFlkZPny4m0PaC9ya31nzRWvKOm9Blj179tj48eMD+6pSpYrVr1/fBg0a5BZ30YIsmo86JibGWrVqFRjXtm1bmz17tlsIRl/q3R49erS7fuaQBgAAQJoI0up/VoVYKwlGKqVrGe/XXnvNBWItkKLZNxSqg1c1lIcffti1bMyaNcsOHz7sFlrRwivBqxrKuHHjbMyYMTZs2DB34J+WCp88eXLIjB/qkdYKi1pB8fHHH3fn6SBILcgCAAAAZLrCUUzX1JYtW9xp5cqVfe/juYlL7Kd9/9dygozp5phCNrJPk9S+GQAARJ0ticxraaZHGgAAAEhPCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAAkJ6D9Lvvvmt//OMfrXLlyla7dm3r0aNHyPkrV660li1buvMbNWpkc+bMibif6dOnW4MGDdy41q1b27p162KNOXXqlA0dOtRq1qxp1apVc9e1b9++WON27dplXbt2tapVq7rbNGLECDt79mwK/tYAAABIr9JEkJ48ebKNHj3amjdv7oLw8OHDrWjRooHzN27caL169bKKFSvam2++aa1atXKhVuE7mC47YcIEa9++vb3xxht20003Wffu3e37778PGde3b19bvny5DRkyxI0/ePCgdenSJSQknzhxwjp37my//fabTZo0yQYMGGAffvihDR48+BrcIwAAAEjrsqb2DdixY4dNnTrVBd+77747sF1VZ8+UKVNciB45cqT7uVatWrZ//36bOHGiqzpnzpzZzp8/7/bTqVMnV0WWGjVquHA+bdo0F5hl8+bNtmLFCnd99erVc9vKlSvnrm/hwoXWrl07t23+/PkuTC9atMgKFy7stmXJksX69etnPXv2tDJlylzDewkAAABpTapXpN9//3278cYbQ0J0MAXktWvXWtOmTUO2KyAfOnTItm7d6n7esGGDnTx50po1axYYo+DbpEkT1xZy5coVt03f58+f3+rWrRsYV6JECatevbo7z7Nq1SrXzuGFaGncuLFlz549ZBwAAACiU6oHaVWIVRFW1VnB9bbbbrMOHTrYtm3b3Pm7d++2CxcuWOnSpUMuV7Zs2UBFO/g0fJwqx2rPOHDgQGBcqVKlLFOmTLH25+3DGxdedVaILlmyZMg4AAAARKdUb+1QVfnbb7+1H3/80YYNG2bZsmWzV1991fUsf/rpp3b8+HE3TlXkYN7P3vlqw1DQzZkzZ8i4AgUKuNNjx45Z8eLF3bh8+fLFuh3an7cvb3/h1xlpnB+qjp8+fTrJl1P4z5UrV7KuG+nPmTNnAp+oAACAq0+vu+FF1zQZpL1QqQMOb7nlFretUqVK1rBhQ1uwYIFruZC4fpng7ZHGeAEkoXHxbQ/fX2LGxUcVdq/inhQK0eoVR3TR7DEK0wAA4NpRgTbNB2lVjK+//vpAiBbN2KEWje3bt9s999zjtoVXgVUxFq9qrNNz5865rxw5csQa51WmNU4HKoYLr0Dre++ywdSHndwDDVV191pTkiK5AR7pk1qRqEgDAHDtKIMmRqoHaYXSX375JdZ2BQfNxqGeZAXPnTt3hhwg6P2CXqj1TtW/HFy11c958uSxYsWKBcatWbMmVmVZ+wsOyPo+vBdaBz6qZ1szhSSHrjd37tzJ2geiB+08AABcW4ktXqb6wYb169e3w4cP2w8//BDYpgMDFZzLly/vyuqa7m7p0qUhl1u8eLEVKVIkEJrVAqLe5yVLlgTGXLp0yV1O09x5d4i+V6V59erVgXGqUGvWD286PFFo12whR48eDWxbtmyZC9PB4wAAABCdUr0irfmb1RP9+OOPW58+fVxw1gwemnbugQcecGN69+7tZvLQYiia9k6hV4uxaOEWVa1Fl9P8zpovWpdVwNaYPXv22Pjx4wPXV6VKFRfeBw0aZAMHDrS8efO6+ahjYmLcQi+etm3b2uzZs91CMPo6cuRIYNEY5pAGAABApitpoPlSIVWLrWh+5osXL9qdd95pzz77bMhUdjpPgVjtFpp9Q7N6aAXDYPpVtLqhlg9XlVvT6vXv399VtMOXCB8zZox98skn7sA/LRWuVQ4VpsMP8tIKiuvXr3ezgWiOai3IEj4zSFJs2bLFnWoJc7+em7jEftr3f5VyZEw3xxSykX2apPbNAAAg6mxJZF5LE0E6mhCkkVgEaQAA0nZeS/UeaQAAACA9IkgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAQHoM0u+//76VL18+1tfYsWNDxq1cudJatmxplStXtkaNGtmcOXMi7m/69OnWoEEDN65169a2bt26WGNOnTplQ4cOtZo1a1q1atWsR48etm/fvljjdu3aZV27drWqVata7dq1bcSIEXb27NkU/O0BAACQXmW1NOKtt96yfPnyBX4uVqxY4PuNGzdar169rEWLFjZw4EDbsGGDC7XZs2e3Nm3ahIToCRMm2FNPPWUVK1a0d99917p37+5OFc49ffv2tW+//daGDBliefPmtUmTJlmXLl3sgw8+sJw5c7oxJ06csM6dO1uJEiXc+b/++quNGjXKjh07FivkAwAAIPqkmSBdqVIlK1y4cMTzpkyZ4oLxyJEj3c+1atWy/fv328SJE13VOXPmzHb+/HmbOnWqderUyVWRpUaNGta8eXObNm2aC9iyefNmW7Fihb3xxhtWr149t61cuXKuyr1w4UJr166d2zZ//nwXphctWhS4XVmyZLF+/fpZz549rUyZMtfkfgEAAEDalOqtHQlRQF67dq01bdo0ZLsC8qFDh2zr1q3uZ1WpT548ac2aNQuMUfBt0qSJawu5cuWK26bv8+fPb3Xr1g2MU9W5evXq7jzPqlWrXDtHcLhv3Lixq4IHjwMAAEB0SjNBWgG4QoUK1rBhQ3v99dft0qVLbvvu3bvtwoULVrp06ZDxZcuWdac7duwIOQ0fp8rxb7/9ZgcOHAiMK1WqlGXKlCnW/rx9eOPCq84K0SVLlgwZBwAAgOiU6q0dRYoUsccff9yqVKniwu3y5cvtlVdeccFXBwQeP37cjVMVOZj3s3e+2jAUdL0eZ0+BAgXcqXqbixcv7sYF92IH78/bl7e/8OuMNM4PVcdPnz6d5Mvp/smVK1eyrhvpz5kzZwKfqAAAgKtPr7vhRdc0GaTr1Knjvjx333235ciRw2bMmOFm0/DE9csEb480xgsgCY2Lb3v4/hIzLj6qsG/bti3Jl1OIVq84ootmj1GYBgAA144KtGk+SEfyhz/8wd5++20XNmNiYty28CqwKsbiVY11eu7cOfelIB4+zqtMa5wOVAwXXoHW995lg6kPO7kHGmbLli3QmpIUyQ3wSJ/UikRFGgCAa2f79u2JGpcmg3Qw9SQreO7cuTPkAEHvF/RCrXeq/uXgqq1+zpMnT2A6PY1bs2ZNrMqy9hcckPV9eC+0DnxUz7ZmCkkOXW/u3LmTtQ9ED9p5AAC4thJbvEwzBxsGW7JkiZtxQ4FYZXVNd7d06dKQMYsXL3b91V5o1qwb6n3WZT06YFGX0zR33h2i71VpXr16dWCcKtSa9cObDk8U2jVbyNGjRwPbli1b5sJ08DgAAABEp1SvSGvOZwVlzeUsn332mb3zzjtuPmgFZendu7d16NDBBg8e7Ka9U+jVIivDhw93c0iLArfmd9Z80ZqyzluQZc+ePTZ+/PjA9emgxvr169ugQYPc4i5akEXzUauFpFWrVoFxbdu2tdmzZ7uFYPR15MgRGz16tLt+5pAGAABA1rTQ//nee+/Zf//7X7t8+bLdfPPN9txzz1nHjh0DY7SM92uvveYCsRZI0ewbCtXBqxrKww8/7Fo2Zs2aZYcPH3bhXAuvBK9qKOPGjbMxY8bYsGHD3IF/Wip88uTJITN+qEdaBzxqBUXNKqLzNEWfFmQBAAAAMl3hKKZrasuWLe60cuXKvvfx3MQl9tO+/2s5QcZ0c0whG9mnSWrfDAAAos6WROa1NNkjDQAAAKR1BGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAMC1CtIVKlSwr7/+OuJ533zzjTsfAAAAibN06VLr1auX1atXz6pWrWrNmze3uXPn2uXLlwNjypcvH+fXwYMH493/hQsXbNy4cXb33XdblSpVrGPHjvbdd9+FjDl8+LB169bNqlevbu3atbOff/455Pxjx45Z7dq17dtvv+XP+v/Laj5cuXIlzvP0B8+UKZOf3QIAAESlv/3tb1aiRAl75pln7LrrrrN169bZSy+9ZHv27LEBAwa4MQsWLIh1OZ2XK1cuK1q0aLz7HzVqlC1atMgGDhxoMTEx9tZbb9lDDz1kH374oRUpUsSNGTlypF26dMkmTZpkb775phs7b968wD5eeeUVu/fee61SpUop/vtHVZCOj96l5MuXL6V3CwAAkGFNmzbNChcuHPi5Vq1advr0aZszZ4499dRTlj17dlepDrZ371776aefrH///vHu+8CBAzZ//nwbNGiQPfDAA26bqtINGza0GTNmWL9+/dy2L7/80gXo22+/3WU5jdVtyJ07t6teq2quL/gI0rqjZ86c6b5Xxbl3797ujxrs3LlzduTIEWvcuHFidwsAABD1gkO0R62yylZqqYhUcV68eLHLZM2aNYv3/vviiy9cpblp06aBbXnz5rUGDRrYypUrA0Fa7R85c+Z033un2iaqjiv7Rbqd0SzRQVofM9xyyy3u+3379tmNN95o+fPnDxmjYF2uXDnr1KlTyt9SAACAKLJ+/XorWLCgy2CRfPTRR3bnnXda8eLF493Pjh077Prrr3f7ClamTBnX2qG23MyZM1vlypVdX7Yq4KqElyxZ0goUKOCu5+jRo/bggw+m6O8XVUFa73a8dzxqUH/hhRfcHwAAAAApa8uWLfb++++7KnCWLFlina9Wix9++MGGDx+e4L5OnDgRse1WIVkVZ7VvqEKtfutHHnnE9UVr/OTJk+3MmTP217/+1fVYZ82a4h3B6Z6ve2TWrFkpf0sAAABghw4dsieeeMJViLt37x7xHlElOVu2bIlup400EUT45BEVK1a0zz//3B3gqAMSc+TIYRMmTHA905qtQ+eNHj3aze5xzz332LBhwyxPnjxR/Rfz/dZCd77eLanNQ/074Vq2bJnc2wYAABBVTp486cKzepSnTp3qwnKkDLZkyRKrU6dOrHaNSNSKq6p0OG3T/nUwoUc/ly5d2n2vQK3q9MKFC90xcE8//bSrgN91113uNk6dOjXQXx2tfAXpXbt2Wc+ePd38gpGmwtO7HoI0AABA4qkwqXyliq+muitUqFCcvdO//PJLgrN1eNSKqyCsgxaDg7d6p0uVKuX6oyNRO4faeVWd/uyzz9yxcJrfWlq1auVaT6KdryCtdyPnz5935X5NAh4+ewcAAAAS7+LFi9anTx/X+zx79mwXXuOitg5VkdVekRhahEVhWVPXaaEV+e2332z58uXWpk2biJfRVHjbtm2z8ePHB7apn1qzf6hnW33V8Bmktarhiy++aPfffz/3IQAAQDKpSKkeZFWZz549a5s2bQqcV7ZsWXcwoBe4P/nkE7cwihZiiaRRo0ZucRdNXSzFihWztm3b2tixY90Bgzrv7bffdud17tw51uV1HZruTgcfetPgaQ5rze6hfdSoUcPN7tGkSZOo/7v7CtJ6F+T9QQEAAJA8mutZNENGOK3jUbNmzcA4TUUX39zRqhoHLy0uWqVQ+U2rE6oPWwuyKGh7qxoGU0Vc0+UFF0w1BZ+WGH/55ZftnXfesfr167slzaNdpivxrfcdB7V07N+/392ZSBodoCk6Etev5yYusZ/2HeWuz+BujilkI/vwbh8AgLSa13xVpLXoiibn7tGjh1sVJ9IRo/fdd5+fXQMAAADpgq8g3bdv38Aa7ytWrIg4a4ca1AEAAICMyleQVq8OAABAenL58hXLnDn2wiTImC5fg7+3ryCtozUBAADSE4WqKfO+tH0Hj6f2TcFVFlO0gPVud1faXdkQAAAgvVGI5oB9pGqQ7tSpU7znq0fam7sQAAAAyIgirwmZAM2YF/6lOQ03bNhgP/30U8RlwxNDq+zUrVvXrZboTTviWblypVt2XNOQaKLxOXPmRNzH9OnT3UwiGte6dWtbt25drDGnTp2yoUOHujkZq1Wr5mYf2bdvX8Sl0Lt27eomIa9du7aNGDHCTZIOAAAA+KpIz5o1K+J2BU9Nzv3YY4/5umdfe+01N4l4uI0bN7r9tmjRwk0orsCuUKulyYOXtlSI1hzXTz31lFWsWNHeffdd6969uztVOA+edeTbb7+1IUOGuIVlJk2aZF26dLEPPvggsILPiRMn3Go/Wv1H5//6669uzXmtU69VfQAAABDdfFWk41KqVClXwY20Kk9CduzY4ZabfPzxx2OdN2XKFBeMR44cabVq1XKh+s9//rNNnDgxsHLP+fPnberUqa7tRLdBFWTdjhtuuMGmTZsW2NfmzZvdlH1a+lKrAmllnldffdVVpBcuXBgYN3/+fBemFe5VJVc1fPDgwW59e91WAAAARLcUDdISExNjP/74Y5Ivp2CrdeAVxoMpIK9du9aaNm0asr158+Z26NAh27p1q/tZVWoteRm8ZGaWLFncOvBqC/HaTfR9/vz5XTj2qOpcvXp1d55n1apVLowXLlw4sK1x48auCh48DgAAANEpxYP0p59+akWLFk3SZT7++GP77rvvrHfv3rHO2717t124cMFKly4dsr1s2bLu1KsOe6fh48qUKeN6rw8cOBAYp7CuAyLD9xdcadb3umwwheiSJUtSkQYAAIC/Hulnn3021jZVjn/44Qfbvn279e/fP9H7OnPmjI0ePdqefvpp168c7vjx/53rUVXkYN7P3vlqw1DQ9XqcPQUKFHCn6m0uXry4G5cvX75Y16P9efvy9hd+nZHG+aHq+OnTp5N8OYX/XLlyJeu6kf7of8TvAbwAgP/Fa2h0OuPzNVSXCS+6pliQjjQTRo4cOVxbxyOPPOLaLhJLfc3XXXed/elPf4p3XFy/TPD2SGO8Oy+hcfFtD99fYsbFRxV2P0uoK0SrVxzRRQfx6okAAOAfr6HRaVcyXkNVoL0qQXr58uWWEnSA39tvv+0OJtSUdOJVanWqlgyvohxeBVbFWLyqsU7PnTvnvhTqw8d5+9G4/fv3x7ot4RVofe9dNpj6sMNbPpIqW7ZsgdaUpEhugEf6pFYkKtIAkDy8hkanUj5fQ9VhkeZXNty7d6+rzqqKHU6zb1SpUsVmz57tgufOnTtDDhD0fkEv1Hqn6m0Ortrq5zx58lixYsUC49asWROrsqz9BQdkfR8+O4faV9Szrfmpk0PXmzt37mTtA9GDdh4AAK7ta2hi33j5DtLqOf773//uZtTQYiyFChWy3//+927uZa/6m5AKFSrYzJkzQ7ap5UHzNQ8bNswtqqKyuqa8W7p0qT300EOBcYsXL7YiRYoEQrNm3VDv85IlSwLbNCe1LlevXr3AHaLvVQFfvXp1IJirQq1ZPzS9nUfnqe3E+91k2bJlLkxrHwAAAIhuvoK0ZsBo166d/fLLL65yq+njDh486OZcXrRokc2bNy9QAY6P2ie0umAklSpVcl+i2Tw6dOjggq76rxV6tcjK8OHDLXPm/514RIG7Z8+ebkEWTVnnLciyZ88eGz9+fGC/qnJr7uhBgwa5xV10gKPmo1Z/d6tWrQLjNBWfquGas1pfR44ccQdF6vqT29oBAACAKA3SCqZaKvudd96x22+/PbD966+/DoRZhc6UomW8FdJ1vQrqmn1DoTp4VUN5+OGHXcuGVl48fPiwlStXzt54442QVQ1l3LhxNmbMGFf1VmuJwvzkyZNDZvxQyJ8xY4ZbQVGLxOg8zVHdr1+/FPu9AAAAkH5luuKjA1stHE8++aQ98MADsc5bsGCBq/CqDxmxbdmyxZ2qbcWv5yYusZ/2HeXuzeBujilkI/s0Se2bAQAZCq+h0eHmZL6GJjav+VqQRTNXqBUiEi3JrfMBAACAjMxXkFZYXrFiRcTztLS2zgcAAAAyMl890lo8RX3G6gpp2bKlmz3j0KFD9sEHH7gD9Pr27ZvytxQAAABI70G6W7dubjYMheY5c+YEtitYq2+6a9euKXkbAQAAgIwRpDUns6ae07zOWi5cc0oXLFjQzfesFWQAAACAjC7RPdJaolvTwH3++eeBbaVLl3bzSWvKO53+9NNPbowWMQEAAAAyskQHaS1u8t1331mdOnXiHKPzfvjhh5B2DwAAACCqg7SW3tYCKFmzxt0NovM0Zvny5Sl1+wAAAID0HaR37dqVqEVEtKy3WjwAAACAjCzRQfrSpUvxVqM9GnPx4sXk3i4AAAAgYwRpzRW9ffv2BMf9+OOPdv311yf3dgEAAAAZI0jXqFHD5s6daxcuXIhzjM6bN2+e1axZM6VuHwAAAJC+g3Tnzp1dn/Rjjz1mBw4ciHW+tvXu3duN0fzSAAAAQEaW6AVZbr31Vhs6dKgNGzbMGjZsaLfddpvFxMS48/bt22fffPONW9nwhRdesPLly1/N2wwAAACkr5UNtfz3LbfcYq+//rpb0XDTpk1ue65cudwc0o8++qhVrVr1at1WAAAAIP0uEV6tWjWbNm2aXb58ObCCYaFChSxz5kR3iQAAAADRF6Q9Cs7XXXddyt4aAAAAIJ2gjAwAAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAOkxSK9evdo6dOhgtWrVsttuu80aNmxoo0aNspMnT4aMW7lypbVs2dIqV65sjRo1sjlz5kTc3/Tp061BgwZuXOvWrW3dunWxxpw6dcqGDh1qNWvWtGrVqlmPHj1s3759scbt2rXLunbtalWrVrXatWvbiBEj7OzZsyn42wMAACC9SvUgffz4cRdmX3zxRReCu3TpYosWLbI+ffoExmzcuNF69eplFStWtDfffNNatWrlQu27774bsi9dfsKECda+fXt744037KabbrLu3bvb999/HzKub9++tnz5chsyZIgbf/DgQXe9wSH5xIkT1rlzZ/vtt99s0qRJNmDAAPvwww9t8ODB1+BeAQAAQFqXNbVvQLNmzdyXR1Xi7Nmzu5B74MABK1asmE2ZMsWF6JEjR7oxql7v37/fJk6c6KrOmTNntvPnz9vUqVOtU6dOroosNWrUsObNm9u0adNcYJbNmzfbihUrXNCuV6+e21auXDlX5V64cKG1a9fObZs/f74L0wr1hQsXdtuyZMli/fr1s549e1qZMmWu+X0FALi2fv75Z1ek0WvHjz/+aKVLl7bFixeHjLl48aK9/fbb9v7777vXpuuuu859MvrEE09Y/vz549y3PvWcPXu2ffXVV/bLL79YoUKF3KefTz31lBUpUiQw7vDhwzZw4EDbsGGDlS9f3kaPHu0KRZ5jx47ZH/7wB3vrrbesUqVKV+meAJAmK9KRFCxYMPDkpIC8du1aa9q0acgYBeRDhw7Z1q1b3c96glE7SHAoV/Bt0qSJawu5cuWK26bv9cRWt27dwLgSJUpY9erV3XmeVatWuSc0L0RL48aNXcgPHgcAyLgUnvWcr+AaVwFFxR4VdtR++Prrr7tijoowzzzzTLz7/vLLL+3f//63PfDAA+5yTz75pP3nP/+xv/zlL+7TUI+KSJcuXXKfjuo1SKE62CuvvGL33nsvIRqIxoq0R08SCs7bt293T0r33HOPxcTEuJ8vXLjgqgDBypYt60537Njheqt1KuHj9MSnJyRVt4sXL+7GlSpVyjJlyhRrf1988UXgZ41TtTuYnsBKliwZuC4AQMamyrJCqijAfvPNN7HGqEKtIo6Ot/E+NT19+rSNHz/enebOnTvivlXoUSti8OuRKs4tWrSwTz/91LUxeoFbbY2333675cuXzwVvb7/fffedLV261H0BiOIgreCssCt16tRxT0BeD7WEfzzm/eydrzYMBd2cOXOGjCtQoEDgoy8FaY3TE1E47c/bl7e/SB/JhY/zQ9VxPQkmlZ5sc+XKlazrRvpz5syZwCcqAFKPij2XL1+O9fytT0712hO8XT8n9FyvMfr/DnbjjTe6T1P37t0buKyKSRK8L71GyfDhw92xQOHXj9h4DY1OZ3y+huoy4UXXNB2k1bOsJwFVoF977TX3zv5vf/tb4Py4fpng7ZHGeHdeQuPi2x6+v8SMi4+eFLdt25bkyylEq1cc0UV9lOEvtgCuPRVRzp07F+v5W62CH3zwgfu085ZbbnH9zupX1nb1WCfFDz/84D6hzZYtW+B6br75Ztf6oUr0ggUL3LFDmmlKB9yrAKVKtZ/XlGjDa2h02pWM11AVaNNNkL711lvdqXqVFRbVVrFs2bJAC0d4Fdh7N+5VjXWqJzh95ciRI9Y4rzKtcToYJFx4BVrfe5cNpj7s5B5oqCdI7/dKiuQGeKRPenGmIg2kPr2O6PWlQoUKIdv1s46n0UGA3v+qWkL0s57vk1Jkeemll1xwfvDBBy1r1v99idZsUY8//rj961//srx589rYsWPdmKefftqGDRvm2huRMF5Do1Mpn6+hKuwmRpoJ0uFPSvpoa/fu3e7JSE9EO3fuDDlA0PsFvVDrnap/Obhqq5/z5Mnj3sF749asWROrsqz9BQdkfR/eC62P73Sbwnunk0rXG1fPHBCOdh4gbVCw1SxR4c/fmnlDaxuoh1qzZqgCpoMPdZDgmDFjEr1/rW+g1x3tL7iwowKTZpvas2ePO3ZIYV4zUVWpUsXq169vn3/+uQvtmt1DbZIK13rdA2C+X0MT+8YrTc7aoXmj9dHWDTfc4MrqOnAj/EAKHdyh6YG80KwnGvU+L1myJDBG+9DlNM2dd4foe1WatRCMRxVqzfrhTYcnCu2aLeTo0aOBbaqQK0wHjwMARC+9Rigsa6q7hx56yO68807XgqEwq5k7vv3220Tt59VXX7X33nvPHR+kBcXCqaCkg+kVohWo582b59Y3OHLkiKtMP/bYY+41SiFeU8ECuDZSvSKtf359LKUjlXWwhI5AVm+ZfvaOlO7du7db/VAfb2naO4Ve9YbpIAtVB0SBW/M76126PmJTwNYYPeF4By6K9w5+0KBBrnqgj8lUOdC7fO8IaWnbtq2rCmghGH3pyUrv+HX9zCENABC9xqjAEqndQ/QpZkJzO6uaPXnyZPeaptV9E6LVfzt27Ohetz777DP3+qfXJtHrmOazBhAlQVoHSaiKrIMN1W6hJwa9m9c8nF6Tt1Y+1AGICsR6h6/ZNxSq27RpE7Kvhx9+2O1j1qxZ7iMuLbSi/SqUBxs3bpyrIKhioJ40LQKjJ7HgGT/0sdqMGTPcCorqTdN5mt5IC7IAAOCtQyCqPKsa7fGmydNrWnw++ugj9zqjirbmj06IpsLTgYXBBSK9jukTWLVEMnMHEGVB+pFHHnFfCVE7RUItFWrf6Natm/uKj6rQWpJcXwk1qGtFKwBAdNLR/t4iXJop49SpU/bxxx8HVs+9/vrr3WJd+mRT0+PpE1Yd06PijIpAwQcCagVdBW8VaUSLsag943e/+53dddddtmnTpsBYfbKqdQuCaf86GFGX8Qo/VatWdVPy6QBE3Z65c+e6+akBREmQBgAgrVJbX58+fUK2eT/PnDnTfaKpgwrVl/zOO++41QcVru+77z43zms/FFWNFXo969atc9VkBerwarRaNNROGEzthtr3/fffH9im5cj1KevLL7/srl+ti2pHBHBtZLrCvFrX1JYtW9xppINJEuu5iUvsp33/dxAkMqabYwrZyD5UlgAgJfEaGh1uTuZraGLzWpqctQMAAABI6wjSAAAAgA8EaQBAqrp8OemrjiH94u+NjISDDQEAqSpz5kw2Zd6Xtu/gcf4SGVxM0QLWu91dqX0zgBRDkAYApDqFaA6iBpDe0NoBAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAEB6DNJLly61Xr16Wb169axq1arWvHlzmzt3rl2+fDlk3MqVK61ly5ZWuXJla9Sokc2ZMyfi/qZPn24NGjRw41q3bm3r1q2LNebUqVM2dOhQq1mzplWrVs169Ohh+/btizVu165d1rVrV3e7ateubSNGjLCzZ8+m4G8PAACA9CrVg/Tf/vY3y549uz3zzDM2bdo0u/fee+2ll16yv/71r4ExGzdudGG7YsWK9uabb1qrVq1cqH333XdjhegJEyZY+/bt7Y033rCbbrrJunfvbt9//33IuL59+9ry5cttyJAhbvzBgwetS5cuISH5xIkT1rlzZ/vtt99s0qRJNmDAAPvwww9t8ODB1+BeAQAAQFqXNbVvgMJz4cKFAz/XqlXLTp8+7SrOTz31lAvZU6ZMcSF65MiRgTH79++3iRMnuqpz5syZ7fz58zZ16lTr1KmTqyJLjRo1XIVb16HALJs3b7YVK1a4oK0quJQrV85VuRcuXGjt2rVz2+bPn+/C9KJFiwK3L0uWLNavXz/r2bOnlSlT5prfVwAAAEg7Ur0iHRyiPRUqVLBz587ZsWPHXEBeu3atNW3aNGSMAvKhQ4ds69at7ucNGzbYyZMnrVmzZoExCr5NmjRxbSFXrlxx2/R9/vz5rW7duoFxJUqUsOrVq7vzPKtWrXLtHMG3r3Hjxi7YB48DAABAdEr1IB3J+vXrrWDBgnbdddfZ7t277cKFC1a6dOmQMWXLlnWnO3bsCDkNH6fKsdozDhw4EBhXqlQpy5QpU6z9efvwxoVXnRWiS5YsGTIOAAAA0SnVWzvCbdmyxd5//33r3bu3qygfP37cbVcVOZj3s3e+2jAUdHPmzBkyrkCBAu5U1e3ixYu7cfny5Yt1vdqfty9vf+HXGWmcH6qOq30lqRT+c+XKlazrRvpz5syZwCcqQEbD81p0So3nNR5r0emMz8eaLhNedE3zQVqtGk888YSbcUMHCQaL65cJ3h5pjHfnJTQuvu3h+0vMuPiowr5t27YkX04hWr3iiC6aPUZPBEBGxPNadEqN5zUea9FpVzIeayrQppsgrf5mhWdVlHXQYLZs2UIqyuFVYFWMxasa61R91frKkSNHrHHefjROByqGC69A63vvsuG3M7kHGup381pTkiK5AR7pk1qRqEgjo+J5LTqlxvMaj7XoVMrnY2379u2JGpcmgrTCr2bCOHz4sC1YsMAKFSoUOE89yQqeO3fuDDlA0PsFvVDrnap/Obhqq5/z5MljxYoVC4xbs2ZNrMqy9hcckPV9eC+0DnxUz7ZmCkkOXW/u3LmTtQ9ED9p5AGQ0PK8hrT/WEvvGK9UPNrx48aL16dPHvvvuO3vrrbcsJiYmVlld091p4ZZgixcvtiJFigRCs2bdUO/zkiVLAmMuXbrkLqdp7rw7RN+r0rx69erAOFWoNeuHNx2eKLRrtpCjR48Gti1btsyF6eBxAAAAiE6pXpEePny4ff7559a/f3+3IMqmTZsC56n9IW/evO7Aww4dOrjFUDTtnUKvFmPRZTWHtBe4VdXWfNGask4BW2P27Nlj48ePD+yzSpUqVr9+fRs0aJANHDjQ7V/zUSvAa6EXT9u2bW327NluIRh9HTlyxEaPHu2unzmkAQAAkOpB+osvvnCnwSsZembOnBlYxvu1115zgVgLpGj2DYXqNm3ahIx/+OGHXcvGrFmzXJuIFlrRwivly5cPGTdu3DgbM2aMDRs2zB34p+uYPHlyyIwf6pGeMWOGW0Hx8ccfd+dpjmotyAIAAACkepDWUt2JoXaKhFoq1L7RrVs39xUfVaFffPFF95VQg7qWHQcAAADSXI80AAAAkB4RpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRpAhtCxY0crX758xK+PPvoo4mVOnTplkydPtjZt2tjvfvc7q1WrlnXt2tW+/fbbkHEXL160F1980WrUqGH33XefrVy5MuL1//3vf79qvx8AIO3Jmto3AABSwvPPP++CcbAZM2bYp59+arVr1454mV9++cUWLFhgrVu3tieeeMIF5pkzZ1rbtm1t/vz5VqlSJTfuvffes+XLl9uYMWNszZo19vTTT9tnn31mBQsWdOcvWbLEfv31V+vQoQN/TACIIgRpABlC2bJlY23r27ev3XXXXVa4cOGIl7nhhhts2bJllitXrsC23//+99awYUObPXu2jRo1ym1TeG7fvr3dc889VqdOHResN2/ebPXq1bOzZ8/aX//6V3vppZcsa1aeUgEgmtDaASBD2rBhg+3du9eaN28e55jcuXOHhGjJkSOHlSlTxg4ePBjYdv78ecuZM6f7XmE5e/bsbpu8/vrrrnKtAA4AiC4EaQAZ0uLFi11IVnU5KU6fPm3btm2z0qVLB7ZVrlzZ/vnPf9rhw4dt0aJFdvLkSatQoYIL6nPmzLEBAwZchd8AAJDW8TkkgAxHvc4ff/yxC9GqOifFK6+8YmfOnAnpd+7UqZM7wFBtIpkyZXItI2oLeeyxx1zLx4033ngVfgsAQFpHkAaQ4Xz55Zd25MgRa9asWZIu9+GHH7oDFIcOHWo33XRTYHu+fPncQYmqQOt7HWT41Vdf2TfffOP6o3fu3GmDBw92lWxVqkeOHGk333zzVfjNAABpCa0dADJkW4fC7t13352k8P3ss8+66e9UZQ6nSrQqz9qvKt46uPCZZ55x7SP9+/e3W2+91VavXm233HKL2w4AyPgI0gAyFM2ioanp7r//fsuWLVuiLvP111+7Ng1dRqE4IZrRo1ChQtakSRM35Z4q05oyL2/evC6Ea0aP8Kn4AAAZD60dADIUzff822+/xTtbR7AdO3ZY9+7drXr16m66O1We46P5oqdOnepaQMIDvKi/GgAQHahIA8hQ1OdcokQJu+OOO2Kd99xzz1nFihUDP6uPWq0cqlx369bNrWi4adMm97V169aI+x83bpw1bdrUtXKIqtCa/m7ixImuPUQHK2qWD20HAGRsVKQBZBjHjx93fcqdO3eOWFm+fPmyXbp0KfDz9u3bbf/+/e77hx56KGRsTEyMq24HUwuHti1dujRkuw441MGGag/RkuQvv/xyCv9mAIC0iCANIMMoUKCAC7txGT16tPvy1KxZ077//vtE7/+2225zs3WE0wIu8+bN83GLAQDpGa0dAAAAgA8EaQAAACA9Bumff/7ZLX7QokULdxBQXAsoaFWxli1buoN4GjVq5JbljWT69OnWoEEDN65169a2bt26WGM0LZWuUx/rVqtWzXr06GH79u2LNW7Xrl3uQKSqVata7dq1bcSIEYEj84GM7vLlK6l9E3AN8fcGgHTYI/3jjz+6kFylShV3INCVK7FfvDdu3Gi9evVyYXvgwIG2YcMGF2qzZ89ubdq0CQnREyZMsKeeesqF8nfffddNa6VTHQDk0fK+Ojp/yJAh7sj6SZMmWZcuXeyDDz6wnDlzujEnTpxwByzp6H+drymvNDXWsWPHbOzYsdfo3gFST+bMmWzKvC9t38Hj/BkyuJiiBax3u7tS+2YAQLqT6kFa1eN7773Xfa+QHOlAoSlTprhgrGV3pVatWu5Ie003papz5syZ7fz5825u106dOrkqstSoUcPNJTtt2jQXsEULJaxYscLeeOMNq1evnttWrlw5V+VeuHChtWvXzm2bP3++C9OLFi2ywoULu21ZsmSxfv36Wc+ePd3BRUBGpxD9076jqX0zAABIk1K9tUMhOD4KyGvXrnXztgZTQD506FBgrldVqU+ePBnSGqLgq5XHVPH2Kt36Pn/+/Fa3bt3AOFWdtRiDzvOsWrXKtXN4IVoaN27squDB4wAAABCdUj1IJ2T37t124cIFK126dMj2smXLBlYlCz4NH6fKsVY5O3DgQGBcqVKlYs0xq/15+/DGhVedFaJLliwZMg4AAADRKdVbOxKzwIKoihzM+9k7X20YCrpej3PwvLKi3ubixYu7cfny5Yt1Pdqfty9vf+HXGWmcH6qOnz59OsmXU/jPlStXsq4b6Y+WnI507MDVxGMtOvFYA481ZDRnfL6G6jKRFvZKd0HaE9cvE7w90hjvzktoXHzbw/eXmHHxUYV927ZtSb6cQnTw8saIDpo9Rk8E1xKPtejEYw081pDR7ErGa6gKtOk+SHsV5fAqsCrG4lWNdXru3Dn3lSNHjljjvP1onLckcPj+givQ+t67bDD1YSf3QMNs2bIFWlOSIrkBHumTWpFSoyKN6MNjDTzWkNGU8vkaun379kSNS/NBWj3JCp47d+4MOUDQ+wW9UOudqn85uGqrn/PkyWPFihULjFuzZk2syrL2FxyQ9X14L7QOfFTPtmYKSQ5db+7cuZO1D0QP2nnAYw0ZDc9rSOuPtcQWlNL8wYYqq2u6u6VLl4ZsX7x4sRUpUiQQmjXrhnqflyxZEhhz6dIldzlNc+fdIfpelebVq1cHxqlCrVk/vOnwRKFds4UcPfp/U38tW7bMhengcQAAAIhOqV6RVt+KN52cVhfUqoMff/xxYB5oTT/Xu3dv69Chgw0ePNhNe6fQq0VWhg8fHpg+T4Fb8ztrvmhdxluQZc+ePTZ+/PjA9Wnhl/r169ugQYPcvNVakEXzUcfExFirVq0C49q2bWuzZ892C8Ho68iRIzZ69Gh3/cwhDQAAgFQP0gqoffr0Cdnm/Txz5szAMt6vvfaaC8RaIEWzbyhUB69qKA8//LBr2Zg1a5YdPnzYLbSihVeCVzWUcePG2ZgxY2zYsGHuwD9dx+TJk0Nm/FCP9IwZM9wKio8//rg7T3NUa0EWAAAAINWD9A033GDff/99guPUTpFQS4XaN7p16+a+4qMq9Isvvui+EmpQ17LjAAAAQLrrkQYAAADSIoI0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAPhCkAQAAAII0AAAAcG1QkQYAAAB8IEgDAAAAPhCkAQAAAB8I0gAAAIAPBGkAAADAB4I0AAAA4ANBGgAAAPCBIA0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRoAAADwgSANAAAA+ECQBgAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA8E6QTs2rXLunbtalWrVrXatWvbiBEj7OzZs37uawAAAGQgWVP7BqRlJ06csM6dO1uJEiVs0qRJ9uuvv9qoUaPs2LFjNnbs2NS+eQAAAEhFBOl4zJ8/34XpRYsWWeHChd22LFmyWL9+/axnz55WpkyZa/V3AgAAQBpDa0c8Vq1a5do5vBAtjRs3tuzZs9vKlSuvxd8HAAAAaRRBOh47duyIVXVWiC5ZsqQ7DwAAANGL1o54qK0jf/78sbZr2/Hjx33d4RcuXLArV67Y119/7evymTJlsqY1itily9f5ujzSjyyZM9uWLVvc4yU18FiLHjzWwGMNGU2WZL6GKq/pdTAhBGkf9EdJzJ0biXc5v5eX/Hlz+r4s0p/kPFaSi8dadOGxBh5ryGgyJSOvEaSTSZVnVaXDnTx50veBhtWqVUvuzQIAAEAaQI90PBSWw3uhz58/b7t372bGDgAAgChHkI5H3bp1be3atXb06NHAtmXLlrkwXa9evWvx9wEAAEAalelKah3JlA6oraNZs2YWExNjvXr1siNHjtjo0aPt7rvvZkEWAACAKEeQTsQS4VoWfP369ZYzZ04XrLUgi74HAABA9CJIAwAAAD7QIw0AAAD4QJAGAAAAfCBIAwAAAD4QpAEAAAAfCNIAAACADwRpAAAAwAeCNAAAAOADQRrpUoMGDWz48OHxjtm7d6+VL1/ePv744yTt2+/lkHYMHDjQLZ6U3MdQSl4O0UmPQz0er6Vt27bZ5MmT7cyZM9f0enHt6O9brVo17vI0IGtq3wDgailatKgtWLDAbr75Zu5kxPLqq69a/vz5uWeQ4ShI6/Hdvn17y5UrV2rfHCBDI0gjVZw9e/aqL7OePXt2q1q16lW9DqRfFStWTO2bAKSp50wASUdrByJ6//33XdA4fPhwyPZjx47ZbbfdZnPmzAls27hxo3Xq1MmF1jvuuMP69u1rR44cidUqoX0OHjzYatasaX/+859t5syZ7jKnTp0KuY5du3a58Z999lmCf53Zs2fbPffc4663V69e9uuvv8bbonH+/HkbMWKE1ahRw13mueees4ULF7pxGh/s3Llz7iP8O++80+6++24bM2aMXbx4kUdMOrJu3Tpr2bKle5zpMffNN9/E26Ixf/5893iqUqWKde7c2TZv3hx47CblsYe03fKzevVqa968ud1+++324IMP2p49e9xz25NPPmnVq1e3e++915YsWRJyWX269Yc//ME9/9WvX98mTJgQ6/lgw4YN9qc//ckqV67srmflypURb4ff50xZsWKFdenSxWrXru1ua5s2bWzVqlWBy+oyzz77rPteY7QfPdY9//3vf61fv35un/r9VbUO/r9A+vTLL7/YE088Yb/73e/c85ceX1u2bAmcP2XKFPe49Vy5csU9BvSl7z316tWz11577Zrf/vSMII2I7rvvPsuaNWusPuFPP/3U/dPpBcV7QejYsaPly5fPvbC8+OKL7p+3Z8+esfY5fvx4y5Qpk40bN8769+9vLVq0sMuXL9vixYtDxv3jH/+wIkWKuH/o+Cxfvtw+//xzGzp0qA0aNMj+/e9/u+uPj65bYalbt272yiuvuG3eaThtz5w5szv9y1/+Ym+//ba9++67PGLSiUOHDrk3TV27dnWPTVX0HnvsMbtw4ULE8Xrj9vzzz9tdd93lPhbXqQJOSj32kHYeF2PHjnXPUTpVaNXz0dNPP2233HKL6z2tVKmS27Zv3z53mVmzZrm/da1atWzq1KnWtm1bmz59utsWvF891vRJmJ4z9P2wYcPc9mDJec4U3V69gXv55ZfdbVWYfuSRR9ybRlFY8vb11ltvuTcAejzL8ePH3RuH7777zoYMGeIur9YPvWkMDvJIX1SM6tChg3tDpMekHi8qGilM79ixw41RQWj//v3uTaN8//337nL6+uGHH9w2nac3WhqLxKO1AxHlzZvXBVmFXP2DevSzXkwKFy7sftY/rCo0eqLWE77oxUjVHlVjgsOwKtzhYaNx48YuOOuFSS5dumSLFi1yVUQF+fgo0OtFTS9c8vPPP7sXN4VzBeBwqjjNmzfPvcjohUfq1KnjXtT05BFO1RpVg0Shas2aNfbJJ59Yu3bteNSkAwoNqhrr8Sg5cuRwlTxVmVW1CafHkh7bCt/eY0OfSnghJDmPPaStx8XcuXOtTJky7ueDBw+656Xu3btb79693TZVlJctW2b/+te/3POfqnn333+/e6PlPTb0fKcgrOeTG2+80WbMmOG2vfHGG4HeexUEFKiDJfc5M/j5WI83VRS3b99u77zzjvtez80lS5Z05+sNgfdcLbqNJ06ccAWB6667LlC1btSokXv8PvPMMyl8b+Na0KcQqkh/+OGHgec7/V31ScSbb75po0ePdlVqPV/95z//cY/X//mf/3GPLz2XaZs+udCpxmgsEo9nfMSpadOmtmnTJvcPKqqs6B9NT/iiI8L1UaZeYBSA9TGnvkqVKuVeQII/VpJIFeYHHnjAvv76a/vxxx/dz/qIUtfTunXrBP8yetfsBRkpW7asqzbGVVnRu24Fo4YNG4ZsD//Zo3aOYNp/pMCNtHuwqfeiIl5wOnDgQKyxevzqAK3gj8Dje2wk9bGHtPW48B4L4h2M/Pvf/z6wTUFYAVT/7zt37rSjR49akyZNYj0/KoSsX7/e/aw3aAqywQew6jlERQlPSjxn6jYNGDDAhXkFIYXlL774wrXEJeTLL790t7FAgQKB69YbP72xDL9upB8KxXquC36+y5Mnj/vkQud5hQS9QdRruOhUz2P60idq3jYVkIKf25AwKtKIk/4J9c/40UcfuWqNegazZcvm+gdFlQ29GIwaNcp9hdPHSMGCKyMe/RPrReS9995zfX061ZO6tiUkfMYF3TZRWI7E+4i1UKFCCd4u0Uev4fvXx2VIH5Ly+FB/s0JF+GPBq9olZ99IW+L624X/vytM6O+pCrZcf/31Iecr+Ip3vp5fbrrppljXF/wYSu5zpirQqoCfPHnS9cPq+tSaMWnSpFiXjURvCFQcUfgO51Wxkf7ocRX++BRt8x6f3uutXs9FbwD1ya/eDP7zn/902xS69QYRSUOQRpz0DtY76MYL0uq/8yoseuHRR5OPPvpoIFwHCw+s3seY4XSwjHr5Hn74YffR5tXqNfVe+PRiUqxYscB2DhKDAotaicIfC1SYUbBgwYiPBe+Nuaq73vNLpMdL8LbkPmeqhWjr1q2u1ST48ur/TwzdVlWy+/TpE+s8qpDpl/6u+uQknCYL8B6fXpCeNm2affXVV+5xqQNdvddEbdu9ezf90T4QpBEvvTtVz7KOclclQweneHLnzu2OOtc/sD4y8qtVq1au11AHdim862PPq6FcuXJu/+p7vPXWWwPb9TOiW5YsWaxChQrugEMdeOXhsQF9OqY3WkuXLnUHYXtUWFDQ9cKIPhLXMRiqFnvVbbVcBM9KlNznTO8TD6+KLjogUgcwBs+X750f/gma2lc++OAD19qi24KMQY9BHb+jXnm1mcnp06fdAdH6ZNmjBVxUMNCsHOqJ9j6d0Wujtuk8FnlJOoI04qUnXr2IaJo47wDEYDo4RcFD00YpdOsfUz18OjBP00CpHy8h2r96UTVDiGbHuFoLCKiypAMF9Y5cgVrBSS+G3lHMHCQW3fSRuaax0wGmejOnyp/3kSePjeh+k6WDEPVJmZ6rFEz02FA7hZ7jdOCW6HlQBzHq0zt96eN2FR68inZKPGeWLl3aihcv7g5YVJuHeq51O9T3HczrAdc0papca/5pBaeHHnrIHZCmAxY1o0OJEiXcpzDq79andDof6Y8eN3//+9/dJx16XOlNkj7l1RsvPRY9atXU6556onWQfXClWjPT6M0gb7CSjoMNES+9Q1Wo0JHtOrJbATSYpl7Si4fe/arHWbNh6J2tnrgj9QvGRfsWb67Uq0VVb4V1HVmvjzf1YuQdVR/eI4noojdzL7zwgqsiKlDrUxhvlobgA8YQfRQ8NZWdDtbr0aOHC6h63gieh1xhVjMkqM1Czy36XlOReS1lKfGcqfYLhXOd6jomTpzobo/mxQ+mgxAff/xxV33WjEjedHhqHdF0eApTmvpP7XTq1VZVWyEK6ZOenzRDkXrf9RymqRz12q21GoIPrBVvarvgKe687yPNZoSEZboSPBM3kEpUpdGsCaqWXGtanEBH0mtuYCCYpglThVotHzfccAN3DgAgBK0dSFWaFF4BWi0WXvXvatJHWgrNeueuarRWCdPc2FrxDNFN84xrbl/NJa2PQDUdmNqAVKkmRAMAIiFII1XpI0f16GkansTMHZ1c6v9SeFb/mD6CjYmJcSGa3kDoo1D1y2t6KPW36mNwrb6pTywAAIiE1g4AAADABw42BAAAAHwgSAMAAAA+EKQBAAAAHwjSAAAAgA/M2gEAUUgr3SWGFnVIzAqlABCNmLUDAKLQpk2bQn7W6nrr1q2zGTNmhGwvW7YsKzsCQByoSANAFKpatWrIz4ULF7bMmTPH2g4AiBs90gCAWJ577jmrUaOGnTlzJtZ5nTp1sqZNm4a0iQwfPtzmz59vjRs3tttuu82aNGniFrcJd+jQIRs6dKjVrVvXjWvQoIFbUfLixYv8FQCkO1SkAQARw/I//vEPW7x4sbVp0yawffv27a4FRGE42PLly932J554wnLlymVz5861p59+2rJkyWL3339/IERrX6p89+7d20qWLGkbN260qVOn2r59+2zUqFH8JQCkKwRpAEAst956q6tIz5kzJyRIz5492/VMt2zZMmT80aNH7b333rPrr7/e/VyvXj1r1qyZjR8/PhCkJ0+ebMePH3eV6hIlSrhttWvXtpw5c9qYMWOsa9euricbANILWjsAAHFWpbdt22br1693P586dcr++c9/WqtWrSxPnjwhYxWIvRAtqkSrvePnn3+2//73v27bihUr3AwgRYsWda0c3pfaPOTf//43fwkA6QoVaQBARA0bNrSYmBjXpnHHHXfY+++/73qmH3zwwVhjg0N0+LZjx45Z8eLF7ciRI/b5559bpUqVIl6fqtoAkJ4QpAEAEamXuX379jZhwgQbMGCAC9SqPJcuXTrW2MOHD8e5rWDBgu60UKFC7sDEJ598MuL1qVINAOkJQRoAECf1R2tWjX79+tmuXbvcaSRfffWVC85eFfrSpUu2ZMkSd0ChqtFSv359W7lypdtWoEAB7nUA6R5BGgAQp/z581uLFi1s3rx5rs1D09VFompz586drVevXoFZO3bu3Omq2R7N6LFmzRpr27atdezY0UqVKmXnz5+3vXv32qpVq2zYsGGB0A0A6QFBGgAQLx00qCCtAKx2j0gUsDXjxiuvvGL79++3G2+80caOHesuG9y6oZk9tIri9OnT7cCBA+6gRQX0OnXquNAOAOkJS4QDAOI1evRoF6Q164Yqz+HU96xe6vC5pQEgo6MiDQCIaNOmTfbTTz+5No2//OUvEUM0AEQzgjQAICKFZ/U76yDBuGbaAIBoRmsHAAAA4AMrGwIAAAA+EKQBAAAAHwjSAAAAgA8EaQAAAMAHgjQAAADgA0EaAAAA8IEgDQAAAPhAkAYAAAB8IEgDAAAAlnT/HwRtLFCJ+HVbAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 800x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Set plot style\n",
    "sns.set(style=\"whitegrid\")\n",
    "\n",
    "# Calculate counts and percentages\n",
    "counts = collapsed_df['Type'].value_counts()\n",
    "total = counts.sum()\n",
    "percentages = (counts / total * 100).round(2)\n",
    "\n",
    "# Desired order\n",
    "type_order = ['very high', 'high', 'moderate', 'low']\n",
    "\n",
    "# Plot\n",
    "plt.figure(figsize=(8, 6))\n",
    "ax = sns.countplot(data=collapsed_df, x='Type', order=type_order)\n",
    "\n",
    "# Annotate each bar\n",
    "for p in ax.patches:\n",
    "    height = p.get_height()\n",
    "    label = p.get_x() + p.get_width() / 2\n",
    "    category = p.get_x() + p.get_width() / 2\n",
    "    type_name = p.get_x() + p.get_width() / 2\n",
    "    percentage = (height / total) * 100\n",
    "    ax.annotate(f'{percentage:.1f}%', (p.get_x() + p.get_width() / 2, height),\n",
    "                ha='center', va='bottom', fontsize=11)\n",
    "\n",
    "# Labels\n",
    "plt.title('Distribution of Type Categories')\n",
    "plt.xlabel('Type')\n",
    "plt.ylabel('Count')\n",
    "plt.ylim(0, counts.max() * 1.1)  # Add some space above bars\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65e1acea-88b7-40f9-9330-e39fa4843240",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "browser_data",
   "language": "python",
   "name": "browser_data"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
