{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "23a7f7d2-2e98-45be-b488-688489353f1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a29f3799-2118-4ac8-b80b-4c0ff7748a4e",
   "metadata": {},
   "source": [
    "# MT ATAC-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "c6ed78fb-9417-47a0-9a96-d7cf45fb998f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju\n"
     ]
    }
   ],
   "source": [
    "%cd ~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "8398baaa-d481-40ea-81b9-940aa7ee0027",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mingju/PreprocessDB/promoter_table\n"
     ]
    }
   ],
   "source": [
    "%cd ./PreprocessDB/promoter_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "3a776d02-bfde-46bb-8f33-96702b066d24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/mingju/PreprocessDB/promoter_table'"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0444d201-284c-4343-bd23-243324430ab7",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac = pd.read_csv('/mnt/10T-backup-mingju/ifar/ATAC_seq/MT/atac_output/idr.optimal_peak.narrowPeak_w_header',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4d9d72a8-f8cc-465b-a0aa-4ee95e8a4f2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>10.18118</td>\n",
       "      <td>5255.37842</td>\n",
       "      <td>5247.06348</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>9.06746</td>\n",
       "      <td>4341.48682</td>\n",
       "      <td>4334.17773</td>\n",
       "      <td>321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.49633</td>\n",
       "      <td>3135.06812</td>\n",
       "      <td>3128.26001</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.74865</td>\n",
       "      <td>2599.60767</td>\n",
       "      <td>2593.02832</td>\n",
       "      <td>305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521226</td>\n",
       "      <td>22521666</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.02649</td>\n",
       "      <td>2324.03174</td>\n",
       "      <td>2317.51929</td>\n",
       "      <td>132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151265</th>\n",
       "      <td>chr1</td>\n",
       "      <td>101174750</td>\n",
       "      <td>101175746</td>\n",
       "      <td>.</td>\n",
       "      <td>763</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151266</th>\n",
       "      <td>chr10</td>\n",
       "      <td>73076272</td>\n",
       "      <td>73077247</td>\n",
       "      <td>.</td>\n",
       "      <td>599</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151267</th>\n",
       "      <td>chr10</td>\n",
       "      <td>72637221</td>\n",
       "      <td>72638091</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>710</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151268</th>\n",
       "      <td>chr10</td>\n",
       "      <td>70007166</td>\n",
       "      <td>70007807</td>\n",
       "      <td>.</td>\n",
       "      <td>579</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151269</th>\n",
       "      <td>chr10</td>\n",
       "      <td>31871375</td>\n",
       "      <td>31872084</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151270 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629562     630131    .   1000      .     10.18118   \n",
       "1        chr1     629562     630131    .   1000      .      9.06746   \n",
       "2        chr1     633817     634262    .   1000      .      7.49633   \n",
       "3        chr1     633817     634262    .   1000      .      6.74865   \n",
       "4       chr17   22521226   22521666    .   1000      .     32.02649   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "151265   chr1  101174750  101175746    .    763      .      2.42782   \n",
       "151266  chr10   73076272   73077247    .    599      .      2.42782   \n",
       "151267  chr10   72637221   72638091    .    866      .      2.42782   \n",
       "151268  chr10   70007166   70007807    .    579      .      2.42782   \n",
       "151269  chr10   31871375   31872084    .    866      .      2.42782   \n",
       "\n",
       "            pValue      qValue  peak  \n",
       "0       5255.37842  5247.06348   139  \n",
       "1       4341.48682  4334.17773   321  \n",
       "2       3135.06812  3128.26001   119  \n",
       "3       2599.60767  2593.02832   305  \n",
       "4       2324.03174  2317.51929   132  \n",
       "...            ...         ...   ...  \n",
       "151265     3.35482     1.67457   572  \n",
       "151266     3.35482     1.67457    65  \n",
       "151267     3.35482     1.67457   710  \n",
       "151268     3.35482     1.67457   586  \n",
       "151269     3.35482     1.67457    58  \n",
       "\n",
       "[151270 rows x 10 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4ada0913-19b6-4530-9759-df5e068706c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac['index'] = MT_atac['Chr'] + \":\" + MT_atac['Start'].astype(str) + \"-\" + MT_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b47a673d-73d6-4efe-8a27-cbbea8e45e9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac['ATAC_MT'] = \"ATAC-seq:\" + MT_atac['Chr'] + \":\" + MT_atac['Start'].astype(str) + \"-\" + MT_atac['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a959e645-7cd5-416a-8aa8-3a4030e4fa18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>10.18118</td>\n",
       "      <td>5255.37842</td>\n",
       "      <td>5247.06348</td>\n",
       "      <td>139</td>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>9.06746</td>\n",
       "      <td>4341.48682</td>\n",
       "      <td>4334.17773</td>\n",
       "      <td>321</td>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.49633</td>\n",
       "      <td>3135.06812</td>\n",
       "      <td>3128.26001</td>\n",
       "      <td>119</td>\n",
       "      <td>chr1:633817-634262</td>\n",
       "      <td>ATAC-seq:chr1:633817-634262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>6.74865</td>\n",
       "      <td>2599.60767</td>\n",
       "      <td>2593.02832</td>\n",
       "      <td>305</td>\n",
       "      <td>chr1:633817-634262</td>\n",
       "      <td>ATAC-seq:chr1:633817-634262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521226</td>\n",
       "      <td>22521666</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.02649</td>\n",
       "      <td>2324.03174</td>\n",
       "      <td>2317.51929</td>\n",
       "      <td>132</td>\n",
       "      <td>chr17:22521226-22521666</td>\n",
       "      <td>ATAC-seq:chr17:22521226-22521666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151265</th>\n",
       "      <td>chr1</td>\n",
       "      <td>101174750</td>\n",
       "      <td>101175746</td>\n",
       "      <td>.</td>\n",
       "      <td>763</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>572</td>\n",
       "      <td>chr1:101174750-101175746</td>\n",
       "      <td>ATAC-seq:chr1:101174750-101175746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151266</th>\n",
       "      <td>chr10</td>\n",
       "      <td>73076272</td>\n",
       "      <td>73077247</td>\n",
       "      <td>.</td>\n",
       "      <td>599</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>65</td>\n",
       "      <td>chr10:73076272-73077247</td>\n",
       "      <td>ATAC-seq:chr10:73076272-73077247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151267</th>\n",
       "      <td>chr10</td>\n",
       "      <td>72637221</td>\n",
       "      <td>72638091</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>710</td>\n",
       "      <td>chr10:72637221-72638091</td>\n",
       "      <td>ATAC-seq:chr10:72637221-72638091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151268</th>\n",
       "      <td>chr10</td>\n",
       "      <td>70007166</td>\n",
       "      <td>70007807</td>\n",
       "      <td>.</td>\n",
       "      <td>579</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>586</td>\n",
       "      <td>chr10:70007166-70007807</td>\n",
       "      <td>ATAC-seq:chr10:70007166-70007807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151269</th>\n",
       "      <td>chr10</td>\n",
       "      <td>31871375</td>\n",
       "      <td>31872084</td>\n",
       "      <td>.</td>\n",
       "      <td>866</td>\n",
       "      <td>.</td>\n",
       "      <td>2.42782</td>\n",
       "      <td>3.35482</td>\n",
       "      <td>1.67457</td>\n",
       "      <td>58</td>\n",
       "      <td>chr10:31871375-31872084</td>\n",
       "      <td>ATAC-seq:chr10:31871375-31872084</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151270 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629562     630131    .   1000      .     10.18118   \n",
       "1        chr1     629562     630131    .   1000      .      9.06746   \n",
       "2        chr1     633817     634262    .   1000      .      7.49633   \n",
       "3        chr1     633817     634262    .   1000      .      6.74865   \n",
       "4       chr17   22521226   22521666    .   1000      .     32.02649   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "151265   chr1  101174750  101175746    .    763      .      2.42782   \n",
       "151266  chr10   73076272   73077247    .    599      .      2.42782   \n",
       "151267  chr10   72637221   72638091    .    866      .      2.42782   \n",
       "151268  chr10   70007166   70007807    .    579      .      2.42782   \n",
       "151269  chr10   31871375   31872084    .    866      .      2.42782   \n",
       "\n",
       "            pValue      qValue  peak                     index  \\\n",
       "0       5255.37842  5247.06348   139        chr1:629562-630131   \n",
       "1       4341.48682  4334.17773   321        chr1:629562-630131   \n",
       "2       3135.06812  3128.26001   119        chr1:633817-634262   \n",
       "3       2599.60767  2593.02832   305        chr1:633817-634262   \n",
       "4       2324.03174  2317.51929   132   chr17:22521226-22521666   \n",
       "...            ...         ...   ...                       ...   \n",
       "151265     3.35482     1.67457   572  chr1:101174750-101175746   \n",
       "151266     3.35482     1.67457    65   chr10:73076272-73077247   \n",
       "151267     3.35482     1.67457   710   chr10:72637221-72638091   \n",
       "151268     3.35482     1.67457   586   chr10:70007166-70007807   \n",
       "151269     3.35482     1.67457    58   chr10:31871375-31872084   \n",
       "\n",
       "                                  ATAC_MT  \n",
       "0             ATAC-seq:chr1:629562-630131  \n",
       "1             ATAC-seq:chr1:629562-630131  \n",
       "2             ATAC-seq:chr1:633817-634262  \n",
       "3             ATAC-seq:chr1:633817-634262  \n",
       "4        ATAC-seq:chr17:22521226-22521666  \n",
       "...                                   ...  \n",
       "151265  ATAC-seq:chr1:101174750-101175746  \n",
       "151266   ATAC-seq:chr10:73076272-73077247  \n",
       "151267   ATAC-seq:chr10:72637221-72638091  \n",
       "151268   ATAC-seq:chr10:70007166-70007807  \n",
       "151269   ATAC-seq:chr10:31871375-31872084  \n",
       "\n",
       "[151270 rows x 12 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_atac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b487e865-8aa9-4ee8-845a-c13884bcbaa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_atac_uniq = MT_atac.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "36b1f4fa-0cfc-463c-baf0-fd21451af271",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>ATAC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>10.18118</td>\n",
       "      <td>5255.37842</td>\n",
       "      <td>5247.06348</td>\n",
       "      <td>139</td>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633817</td>\n",
       "      <td>634262</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>7.49633</td>\n",
       "      <td>3135.06812</td>\n",
       "      <td>3128.26001</td>\n",
       "      <td>119</td>\n",
       "      <td>chr1:633817-634262</td>\n",
       "      <td>ATAC-seq:chr1:633817-634262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17</td>\n",
       "      <td>22521226</td>\n",
       "      <td>22521666</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>32.02649</td>\n",
       "      <td>2324.03174</td>\n",
       "      <td>2317.51929</td>\n",
       "      <td>132</td>\n",
       "      <td>chr17:22521226-22521666</td>\n",
       "      <td>ATAC-seq:chr17:22521226-22521666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr8</td>\n",
       "      <td>109333452</td>\n",
       "      <td>109334794</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>15.83920</td>\n",
       "      <td>633.10614</td>\n",
       "      <td>626.95770</td>\n",
       "      <td>851</td>\n",
       "      <td>chr8:109333452-109334794</td>\n",
       "      <td>ATAC-seq:chr8:109333452-109334794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169893202</td>\n",
       "      <td>169894709</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>17.79785</td>\n",
       "      <td>621.90320</td>\n",
       "      <td>615.77106</td>\n",
       "      <td>778</td>\n",
       "      <td>chr1:169893202-169894709</td>\n",
       "      <td>ATAC-seq:chr1:169893202-169894709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150395</th>\n",
       "      <td>chr2</td>\n",
       "      <td>221518222</td>\n",
       "      <td>221518503</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>2.01126</td>\n",
       "      <td>3.46056</td>\n",
       "      <td>1.77423</td>\n",
       "      <td>198</td>\n",
       "      <td>chr2:221518222-221518503</td>\n",
       "      <td>ATAC-seq:chr2:221518222-221518503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150671</th>\n",
       "      <td>chr1</td>\n",
       "      <td>177741234</td>\n",
       "      <td>177741384</td>\n",
       "      <td>.</td>\n",
       "      <td>603</td>\n",
       "      <td>.</td>\n",
       "      <td>2.15866</td>\n",
       "      <td>3.42075</td>\n",
       "      <td>1.73678</td>\n",
       "      <td>73</td>\n",
       "      <td>chr1:177741234-177741384</td>\n",
       "      <td>ATAC-seq:chr1:177741234-177741384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150742</th>\n",
       "      <td>chr11</td>\n",
       "      <td>108958462</td>\n",
       "      <td>108958678</td>\n",
       "      <td>.</td>\n",
       "      <td>741</td>\n",
       "      <td>.</td>\n",
       "      <td>2.33577</td>\n",
       "      <td>3.40943</td>\n",
       "      <td>1.72609</td>\n",
       "      <td>210</td>\n",
       "      <td>chr11:108958462-108958678</td>\n",
       "      <td>ATAC-seq:chr11:108958462-108958678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150764</th>\n",
       "      <td>chr11</td>\n",
       "      <td>64688141</td>\n",
       "      <td>64688293</td>\n",
       "      <td>.</td>\n",
       "      <td>1000</td>\n",
       "      <td>.</td>\n",
       "      <td>1.79671</td>\n",
       "      <td>3.40386</td>\n",
       "      <td>1.72077</td>\n",
       "      <td>69</td>\n",
       "      <td>chr11:64688141-64688293</td>\n",
       "      <td>ATAC-seq:chr11:64688141-64688293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150802</th>\n",
       "      <td>chr11</td>\n",
       "      <td>109034161</td>\n",
       "      <td>109034311</td>\n",
       "      <td>.</td>\n",
       "      <td>549</td>\n",
       "      <td>.</td>\n",
       "      <td>2.33066</td>\n",
       "      <td>3.39760</td>\n",
       "      <td>1.71488</td>\n",
       "      <td>91</td>\n",
       "      <td>chr11:109034161-109034311</td>\n",
       "      <td>ATAC-seq:chr11:109034161-109034311</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52593 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End name  score strand  signalValue  \\\n",
       "0        chr1     629562     630131    .   1000      .     10.18118   \n",
       "2        chr1     633817     634262    .   1000      .      7.49633   \n",
       "4       chr17   22521226   22521666    .   1000      .     32.02649   \n",
       "6        chr8  109333452  109334794    .   1000      .     15.83920   \n",
       "7        chr1  169893202  169894709    .   1000      .     17.79785   \n",
       "...       ...        ...        ...  ...    ...    ...          ...   \n",
       "150395   chr2  221518222  221518503    .   1000      .      2.01126   \n",
       "150671   chr1  177741234  177741384    .    603      .      2.15866   \n",
       "150742  chr11  108958462  108958678    .    741      .      2.33577   \n",
       "150764  chr11   64688141   64688293    .   1000      .      1.79671   \n",
       "150802  chr11  109034161  109034311    .    549      .      2.33066   \n",
       "\n",
       "            pValue      qValue  peak                      index  \\\n",
       "0       5255.37842  5247.06348   139         chr1:629562-630131   \n",
       "2       3135.06812  3128.26001   119         chr1:633817-634262   \n",
       "4       2324.03174  2317.51929   132    chr17:22521226-22521666   \n",
       "6        633.10614   626.95770   851   chr8:109333452-109334794   \n",
       "7        621.90320   615.77106   778   chr1:169893202-169894709   \n",
       "...            ...         ...   ...                        ...   \n",
       "150395     3.46056     1.77423   198   chr2:221518222-221518503   \n",
       "150671     3.42075     1.73678    73   chr1:177741234-177741384   \n",
       "150742     3.40943     1.72609   210  chr11:108958462-108958678   \n",
       "150764     3.40386     1.72077    69    chr11:64688141-64688293   \n",
       "150802     3.39760     1.71488    91  chr11:109034161-109034311   \n",
       "\n",
       "                                   ATAC_MT  \n",
       "0              ATAC-seq:chr1:629562-630131  \n",
       "2              ATAC-seq:chr1:633817-634262  \n",
       "4         ATAC-seq:chr17:22521226-22521666  \n",
       "6        ATAC-seq:chr8:109333452-109334794  \n",
       "7        ATAC-seq:chr1:169893202-169894709  \n",
       "...                                    ...  \n",
       "150395   ATAC-seq:chr2:221518222-221518503  \n",
       "150671   ATAC-seq:chr1:177741234-177741384  \n",
       "150742  ATAC-seq:chr11:108958462-108958678  \n",
       "150764    ATAC-seq:chr11:64688141-64688293  \n",
       "150802  ATAC-seq:chr11:109034161-109034311  \n",
       "\n",
       "[52593 rows x 12 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_atac_uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6537331d-3c8a-46c0-828f-d5cb5edcd06e",
   "metadata": {},
   "source": [
    "# MT DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "a4a0ed79-75f5-4e2f-90ea-783256335851",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase = pd.read_csv('/mnt/10T-backup-mingju/ifar/DNase_seq/myotube/ENCFF222PWD.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d1431f3c-7a81-4fda-a912-8329de7365b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>11484.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>34.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633528</td>\n",
       "      <td>633678</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>12252.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87053</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26320155</td>\n",
       "      <td>26320305</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87054</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>20.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87055</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87056</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>135.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87057</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>26.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87058 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0      chr1    181415    181565    .      0      .         30.0      -1   \n",
       "1      chr1    267990    268140    .      0      .         22.0      -1   \n",
       "2      chr1    629870    630020    .      0      .      11484.0      -1   \n",
       "3      chr1    630598    630748    .      0      .         34.0      -1   \n",
       "4      chr1    633528    633678    .      0      .      12252.0      -1   \n",
       "...     ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "87053  chrY  26320155  26320305    .      0      .         22.0      -1   \n",
       "87054  chrY  26359975  26360125    .      0      .         20.0      -1   \n",
       "87055  chrY  56727995  56728145    .      0      .         15.0      -1   \n",
       "87056  chrY  56734800  56734950    .      0      .        135.0      -1   \n",
       "87057  chrY  56872355  56872505    .      0      .         26.0      -1   \n",
       "\n",
       "       qValue  peak  \n",
       "0          -1    75  \n",
       "1          -1    75  \n",
       "2          -1    75  \n",
       "3          -1    75  \n",
       "4          -1    75  \n",
       "...       ...   ...  \n",
       "87053      -1    75  \n",
       "87054      -1    75  \n",
       "87055      -1    75  \n",
       "87056      -1    75  \n",
       "87057      -1    75  \n",
       "\n",
       "[87058 rows x 10 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f7f2d258-eb5a-4f7b-a1fc-91482002c5a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase['index'] = MT_DNase['Chr'] + \":\" + MT_DNase['Start'].astype(str) + \"-\" + MT_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a99223c6-f504-42ce-830e-62fdac815009",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase['DNase_MT'] = \"DNase-seq:\" + MT_DNase['Chr'] + \":\" + MT_DNase['Start'].astype(str) + \"-\" + MT_DNase['End'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "390bd32f-5930-44f0-9a61-c518c16855b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>11484.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>34.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633528</td>\n",
       "      <td>633678</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>12252.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:633528-633678</td>\n",
       "      <td>DNase-seq:chr1:633528-633678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87053</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26320155</td>\n",
       "      <td>26320305</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26320155-26320305</td>\n",
       "      <td>DNase-seq:chrY:26320155-26320305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87054</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>20.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87055</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87056</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>135.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87057</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>26.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87058 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0      chr1    181415    181565    .      0      .         30.0      -1   \n",
       "1      chr1    267990    268140    .      0      .         22.0      -1   \n",
       "2      chr1    629870    630020    .      0      .      11484.0      -1   \n",
       "3      chr1    630598    630748    .      0      .         34.0      -1   \n",
       "4      chr1    633528    633678    .      0      .      12252.0      -1   \n",
       "...     ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "87053  chrY  26320155  26320305    .      0      .         22.0      -1   \n",
       "87054  chrY  26359975  26360125    .      0      .         20.0      -1   \n",
       "87055  chrY  56727995  56728145    .      0      .         15.0      -1   \n",
       "87056  chrY  56734800  56734950    .      0      .        135.0      -1   \n",
       "87057  chrY  56872355  56872505    .      0      .         26.0      -1   \n",
       "\n",
       "       qValue  peak                   index                          DNase_MT  \n",
       "0          -1    75      chr1:181415-181565      DNase-seq:chr1:181415-181565  \n",
       "1          -1    75      chr1:267990-268140      DNase-seq:chr1:267990-268140  \n",
       "2          -1    75      chr1:629870-630020      DNase-seq:chr1:629870-630020  \n",
       "3          -1    75      chr1:630598-630748      DNase-seq:chr1:630598-630748  \n",
       "4          -1    75      chr1:633528-633678      DNase-seq:chr1:633528-633678  \n",
       "...       ...   ...                     ...                               ...  \n",
       "87053      -1    75  chrY:26320155-26320305  DNase-seq:chrY:26320155-26320305  \n",
       "87054      -1    75  chrY:26359975-26360125  DNase-seq:chrY:26359975-26360125  \n",
       "87055      -1    75  chrY:56727995-56728145  DNase-seq:chrY:56727995-56728145  \n",
       "87056      -1    75  chrY:56734800-56734950  DNase-seq:chrY:56734800-56734950  \n",
       "87057      -1    75  chrY:56872355-56872505  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[87058 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_DNase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "118add04-3a86-4a9c-a938-e4e1c1923012",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_DNase_uniq = MT_DNase.drop_duplicates(subset=\"index\", keep='first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "bcd28bc9-2d1d-4792-9c5c-ce380db5e243",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>signalValue</th>\n",
       "      <th>pValue</th>\n",
       "      <th>qValue</th>\n",
       "      <th>peak</th>\n",
       "      <th>index</th>\n",
       "      <th>DNase_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>30.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>11484.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>34.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>633528</td>\n",
       "      <td>633678</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>12252.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chr1:633528-633678</td>\n",
       "      <td>DNase-seq:chr1:633528-633678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87053</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26320155</td>\n",
       "      <td>26320305</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26320155-26320305</td>\n",
       "      <td>DNase-seq:chrY:26320155-26320305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87054</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>20.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87055</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87056</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>135.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87057</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>26.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>75</td>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87058 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End name  score strand  signalValue  pValue  \\\n",
       "0      chr1    181415    181565    .      0      .         30.0      -1   \n",
       "1      chr1    267990    268140    .      0      .         22.0      -1   \n",
       "2      chr1    629870    630020    .      0      .      11484.0      -1   \n",
       "3      chr1    630598    630748    .      0      .         34.0      -1   \n",
       "4      chr1    633528    633678    .      0      .      12252.0      -1   \n",
       "...     ...       ...       ...  ...    ...    ...          ...     ...   \n",
       "87053  chrY  26320155  26320305    .      0      .         22.0      -1   \n",
       "87054  chrY  26359975  26360125    .      0      .         20.0      -1   \n",
       "87055  chrY  56727995  56728145    .      0      .         15.0      -1   \n",
       "87056  chrY  56734800  56734950    .      0      .        135.0      -1   \n",
       "87057  chrY  56872355  56872505    .      0      .         26.0      -1   \n",
       "\n",
       "       qValue  peak                   index                          DNase_MT  \n",
       "0          -1    75      chr1:181415-181565      DNase-seq:chr1:181415-181565  \n",
       "1          -1    75      chr1:267990-268140      DNase-seq:chr1:267990-268140  \n",
       "2          -1    75      chr1:629870-630020      DNase-seq:chr1:629870-630020  \n",
       "3          -1    75      chr1:630598-630748      DNase-seq:chr1:630598-630748  \n",
       "4          -1    75      chr1:633528-633678      DNase-seq:chr1:633528-633678  \n",
       "...       ...   ...                     ...                               ...  \n",
       "87053      -1    75  chrY:26320155-26320305  DNase-seq:chrY:26320155-26320305  \n",
       "87054      -1    75  chrY:26359975-26360125  DNase-seq:chrY:26359975-26360125  \n",
       "87055      -1    75  chrY:56727995-56728145  DNase-seq:chrY:56727995-56728145  \n",
       "87056      -1    75  chrY:56734800-56734950  DNase-seq:chrY:56734800-56734950  \n",
       "87057      -1    75  chrY:56872355-56872505  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[87058 rows x 12 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_DNase_uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "38dd58db-1c4f-4810-8458-037b15ea7ba7",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase = pd.merge(MT_atac_uniq[['index','Chr','Start','End','ATAC_MT']], MT_DNase_uniq[['index','DNase_MT']], on='index', how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "eed0c8d6-2f54-4b61-be73-65299bc61f30",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006395-100006545</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100009208-100010573</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100009208.0</td>\n",
       "      <td>100010573.0</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100025224-100026486</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100025224.0</td>\n",
       "      <td>100026486.0</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100043466-100044717</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100043466.0</td>\n",
       "      <td>100044717.0</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100043915-100044065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:9028075-9028225</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:9160678-9161527</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9160678.0</td>\n",
       "      <td>9161527.0</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:9177772-9177922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:9259495-9259645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:9278995-9279145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index    Chr        Start          End  \\\n",
       "0       chr10:100006395-100006545    NaN          NaN          NaN   \n",
       "1       chr10:100009208-100010573  chr10  100009208.0  100010573.0   \n",
       "2       chr10:100025224-100026486  chr10  100025224.0  100026486.0   \n",
       "3       chr10:100043466-100044717  chr10  100043466.0  100044717.0   \n",
       "4       chr10:100043915-100044065    NaN          NaN          NaN   \n",
       "...                           ...    ...          ...          ...   \n",
       "139646       chrY:9028075-9028225    NaN          NaN          NaN   \n",
       "139647       chrY:9160678-9161527   chrY    9160678.0    9161527.0   \n",
       "139648       chrY:9177772-9177922    NaN          NaN          NaN   \n",
       "139649       chrY:9259495-9259645    NaN          NaN          NaN   \n",
       "139650       chrY:9278995-9279145    NaN          NaN          NaN   \n",
       "\n",
       "                                   ATAC_MT  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100009208-100010573   \n",
       "2       ATAC-seq:chr10:100025224-100026486   \n",
       "3       ATAC-seq:chr10:100043466-100044717   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "139646                                 NaN   \n",
       "139647       ATAC-seq:chrY:9160678-9161527   \n",
       "139648                                 NaN   \n",
       "139649                                 NaN   \n",
       "139650                                 NaN   \n",
       "\n",
       "                                   DNase_MT  \n",
       "0       DNase-seq:chr10:100006395-100006545  \n",
       "1                                       NaN  \n",
       "2                                       NaN  \n",
       "3                                       NaN  \n",
       "4       DNase-seq:chr10:100043915-100044065  \n",
       "...                                     ...  \n",
       "139646       DNase-seq:chrY:9028075-9028225  \n",
       "139647                                  NaN  \n",
       "139648       DNase-seq:chrY:9177772-9177922  \n",
       "139649       DNase-seq:chrY:9259495-9259645  \n",
       "139650       DNase-seq:chrY:9278995-9279145  \n",
       "\n",
       "[139651 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a9c9b6be-5b84-4c50-9956-fc78965b93ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[\"OpenChromatin\"] = merge_atac_dnase.apply(\n",
    "    lambda row: f\"{row['ATAC_MT']};{row['DNase_MT']}\" if pd.notna(row['ATAC_MT']) and pd.notna(row['DNase_MT']) and row['DNase_MT'] != \"\"\n",
    "                else row['ATAC_MT'] if pd.notna(row['ATAC_MT'])\n",
    "                else row['DNase_MT'] if pd.notna(row['DNase_MT']) and row['DNase_MT'] != \"\"\n",
    "                else \"\", \n",
    "    axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f664675c-83a1-4733-b731-8cd4733396a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006395-100006545</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100009208-100010573</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100009208.0</td>\n",
       "      <td>100010573.0</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100025224-100026486</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100025224.0</td>\n",
       "      <td>100026486.0</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100043466-100044717</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100043466.0</td>\n",
       "      <td>100044717.0</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100043915-100044065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:9028075-9028225</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:9160678-9161527</td>\n",
       "      <td>chrY</td>\n",
       "      <td>9160678.0</td>\n",
       "      <td>9161527.0</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:9177772-9177922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:9259495-9259645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:9278995-9279145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index    Chr        Start          End  \\\n",
       "0       chr10:100006395-100006545    NaN          NaN          NaN   \n",
       "1       chr10:100009208-100010573  chr10  100009208.0  100010573.0   \n",
       "2       chr10:100025224-100026486  chr10  100025224.0  100026486.0   \n",
       "3       chr10:100043466-100044717  chr10  100043466.0  100044717.0   \n",
       "4       chr10:100043915-100044065    NaN          NaN          NaN   \n",
       "...                           ...    ...          ...          ...   \n",
       "139646       chrY:9028075-9028225    NaN          NaN          NaN   \n",
       "139647       chrY:9160678-9161527   chrY    9160678.0    9161527.0   \n",
       "139648       chrY:9177772-9177922    NaN          NaN          NaN   \n",
       "139649       chrY:9259495-9259645    NaN          NaN          NaN   \n",
       "139650       chrY:9278995-9279145    NaN          NaN          NaN   \n",
       "\n",
       "                                   ATAC_MT  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100009208-100010573   \n",
       "2       ATAC-seq:chr10:100025224-100026486   \n",
       "3       ATAC-seq:chr10:100043466-100044717   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "139646                                 NaN   \n",
       "139647       ATAC-seq:chrY:9160678-9161527   \n",
       "139648                                 NaN   \n",
       "139649                                 NaN   \n",
       "139650                                 NaN   \n",
       "\n",
       "                                   DNase_MT  \\\n",
       "0       DNase-seq:chr10:100006395-100006545   \n",
       "1                                       NaN   \n",
       "2                                       NaN   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100043915-100044065   \n",
       "...                                     ...   \n",
       "139646       DNase-seq:chrY:9028075-9028225   \n",
       "139647                                  NaN   \n",
       "139648       DNase-seq:chrY:9177772-9177922   \n",
       "139649       DNase-seq:chrY:9259495-9259645   \n",
       "139650       DNase-seq:chrY:9278995-9279145   \n",
       "\n",
       "                              OpenChromatin  \n",
       "0       DNase-seq:chr10:100006395-100006545  \n",
       "1        ATAC-seq:chr10:100009208-100010573  \n",
       "2        ATAC-seq:chr10:100025224-100026486  \n",
       "3        ATAC-seq:chr10:100043466-100044717  \n",
       "4       DNase-seq:chr10:100043915-100044065  \n",
       "...                                     ...  \n",
       "139646       DNase-seq:chrY:9028075-9028225  \n",
       "139647        ATAC-seq:chrY:9160678-9161527  \n",
       "139648       DNase-seq:chrY:9177772-9177922  \n",
       "139649       DNase-seq:chrY:9259495-9259645  \n",
       "139650       DNase-seq:chrY:9278995-9279145  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a243595e-64d5-41d6-986d-716aa8b71758",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_index(index_str):\n",
    "    if pd.isna(index_str) or index_str == \"NaN\":\n",
    "        return pd.NA, pd.NA, pd.NA\n",
    "    chr_part, range_part = index_str.split(\":\")\n",
    "    chr_value = chr_part.replace(\"chr\", \"\")\n",
    "    start, end = range_part.split(\"-\")\n",
    "    return chr_value, pd.to_numeric(start), pd.to_numeric(end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9367a17f-4410-4afa-9460-3a9d80d175ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_atac_dnase[['Chr', 'Start', 'End']] = merge_atac_dnase['index'].apply(lambda x: pd.Series(parse_index(x)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "ae373cf5-1cd5-4c34-a158-988fcc95b7af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100006395-100006545</td>\n",
       "      <td>10</td>\n",
       "      <td>100006395</td>\n",
       "      <td>100006545</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "      <td>DNase-seq:chr10:100006395-100006545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100009208-100010573</td>\n",
       "      <td>10</td>\n",
       "      <td>100009208</td>\n",
       "      <td>100010573</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100009208-100010573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100025224-100026486</td>\n",
       "      <td>10</td>\n",
       "      <td>100025224</td>\n",
       "      <td>100026486</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100025224-100026486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100043466-100044717</td>\n",
       "      <td>10</td>\n",
       "      <td>100043466</td>\n",
       "      <td>100044717</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr10:100043466-100044717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100043915-100044065</td>\n",
       "      <td>10</td>\n",
       "      <td>100043915</td>\n",
       "      <td>100044065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "      <td>DNase-seq:chr10:100043915-100044065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:9028075-9028225</td>\n",
       "      <td>Y</td>\n",
       "      <td>9028075</td>\n",
       "      <td>9028225</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "      <td>DNase-seq:chrY:9028075-9028225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:9160678-9161527</td>\n",
       "      <td>Y</td>\n",
       "      <td>9160678</td>\n",
       "      <td>9161527</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:9160678-9161527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:9177772-9177922</td>\n",
       "      <td>Y</td>\n",
       "      <td>9177772</td>\n",
       "      <td>9177922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "      <td>DNase-seq:chrY:9177772-9177922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:9259495-9259645</td>\n",
       "      <td>Y</td>\n",
       "      <td>9259495</td>\n",
       "      <td>9259645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "      <td>DNase-seq:chrY:9259495-9259645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:9278995-9279145</td>\n",
       "      <td>Y</td>\n",
       "      <td>9278995</td>\n",
       "      <td>9279145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "      <td>DNase-seq:chrY:9278995-9279145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            index Chr      Start        End  \\\n",
       "0       chr10:100006395-100006545  10  100006395  100006545   \n",
       "1       chr10:100009208-100010573  10  100009208  100010573   \n",
       "2       chr10:100025224-100026486  10  100025224  100026486   \n",
       "3       chr10:100043466-100044717  10  100043466  100044717   \n",
       "4       chr10:100043915-100044065  10  100043915  100044065   \n",
       "...                           ...  ..        ...        ...   \n",
       "139646       chrY:9028075-9028225   Y    9028075    9028225   \n",
       "139647       chrY:9160678-9161527   Y    9160678    9161527   \n",
       "139648       chrY:9177772-9177922   Y    9177772    9177922   \n",
       "139649       chrY:9259495-9259645   Y    9259495    9259645   \n",
       "139650       chrY:9278995-9279145   Y    9278995    9279145   \n",
       "\n",
       "                                   ATAC_MT  \\\n",
       "0                                      NaN   \n",
       "1       ATAC-seq:chr10:100009208-100010573   \n",
       "2       ATAC-seq:chr10:100025224-100026486   \n",
       "3       ATAC-seq:chr10:100043466-100044717   \n",
       "4                                      NaN   \n",
       "...                                    ...   \n",
       "139646                                 NaN   \n",
       "139647       ATAC-seq:chrY:9160678-9161527   \n",
       "139648                                 NaN   \n",
       "139649                                 NaN   \n",
       "139650                                 NaN   \n",
       "\n",
       "                                   DNase_MT  \\\n",
       "0       DNase-seq:chr10:100006395-100006545   \n",
       "1                                       NaN   \n",
       "2                                       NaN   \n",
       "3                                       NaN   \n",
       "4       DNase-seq:chr10:100043915-100044065   \n",
       "...                                     ...   \n",
       "139646       DNase-seq:chrY:9028075-9028225   \n",
       "139647                                  NaN   \n",
       "139648       DNase-seq:chrY:9177772-9177922   \n",
       "139649       DNase-seq:chrY:9259495-9259645   \n",
       "139650       DNase-seq:chrY:9278995-9279145   \n",
       "\n",
       "                              OpenChromatin  \n",
       "0       DNase-seq:chr10:100006395-100006545  \n",
       "1        ATAC-seq:chr10:100009208-100010573  \n",
       "2        ATAC-seq:chr10:100025224-100026486  \n",
       "3        ATAC-seq:chr10:100043466-100044717  \n",
       "4       DNase-seq:chr10:100043915-100044065  \n",
       "...                                     ...  \n",
       "139646       DNase-seq:chrY:9028075-9028225  \n",
       "139647        ATAC-seq:chrY:9160678-9161527  \n",
       "139648       DNase-seq:chrY:9177772-9177922  \n",
       "139649       DNase-seq:chrY:9259495-9259645  \n",
       "139650       DNase-seq:chrY:9278995-9279145  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "9a58dab4-e981-4bd8-8f9d-7cfc07d5a5cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the chromosome order: 1 to 22, X, Y\n",
    "chr_order = [str(i) for i in range(1, 23)] + [\"X\", \"Y\"]\n",
    "\n",
    "# Convert Chr to categorical with the specified order\n",
    "merge_atac_dnase[\"Chr\"] = pd.Categorical(merge_atac_dnase[\"Chr\"], categories=chr_order, ordered=True)\n",
    "\n",
    "# Sort by Chr and Start\n",
    "merge_atac_dnase_sorted = merge_atac_dnase.sort_values(by=[\"Chr\", \"Start\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ea650c3c-46eb-401b-8b54-f7d0d60cedec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139646</th>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>Y</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139647</th>\n",
       "      <td>chrY:26365821-26366654</td>\n",
       "      <td>Y</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139648</th>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>Y</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139649</th>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>Y</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139650</th>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>Y</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End  \\\n",
       "0           chr1:181415-181565   1    181415    181565   \n",
       "1           chr1:267990-268140   1    267990    268140   \n",
       "2           chr1:629562-630131   1    629562    630131   \n",
       "3           chr1:629870-630020   1    629870    630020   \n",
       "4           chr1:630598-630748   1    630598    630748   \n",
       "...                        ...  ..       ...       ...   \n",
       "139646  chrY:26359975-26360125   Y  26359975  26360125   \n",
       "139647  chrY:26365821-26366654   Y  26365821  26366654   \n",
       "139648  chrY:56727995-56728145   Y  56727995  56728145   \n",
       "139649  chrY:56734800-56734950   Y  56734800  56734950   \n",
       "139650  chrY:56872355-56872505   Y  56872355  56872505   \n",
       "\n",
       "                                ATAC_MT                          DNase_MT  \\\n",
       "0                                   NaN      DNase-seq:chr1:181415-181565   \n",
       "1                                   NaN      DNase-seq:chr1:267990-268140   \n",
       "2           ATAC-seq:chr1:629562-630131                               NaN   \n",
       "3                                   NaN      DNase-seq:chr1:629870-630020   \n",
       "4                                   NaN      DNase-seq:chr1:630598-630748   \n",
       "...                                 ...                               ...   \n",
       "139646                              NaN  DNase-seq:chrY:26359975-26360125   \n",
       "139647  ATAC-seq:chrY:26365821-26366654                               NaN   \n",
       "139648                              NaN  DNase-seq:chrY:56727995-56728145   \n",
       "139649                              NaN  DNase-seq:chrY:56734800-56734950   \n",
       "139650                              NaN  DNase-seq:chrY:56872355-56872505   \n",
       "\n",
       "                           OpenChromatin  \n",
       "0           DNase-seq:chr1:181415-181565  \n",
       "1           DNase-seq:chr1:267990-268140  \n",
       "2            ATAC-seq:chr1:629562-630131  \n",
       "3           DNase-seq:chr1:629870-630020  \n",
       "4           DNase-seq:chr1:630598-630748  \n",
       "...                                  ...  \n",
       "139646  DNase-seq:chrY:26359975-26360125  \n",
       "139647   ATAC-seq:chrY:26365821-26366654  \n",
       "139648  DNase-seq:chrY:56727995-56728145  \n",
       "139649  DNase-seq:chrY:56734800-56734950  \n",
       "139650  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5743febe-5606-483e-a92f-025632971f88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52686</th>\n",
       "      <td>chr1:1511955-1512105</td>\n",
       "      <td>1</td>\n",
       "      <td>1511955</td>\n",
       "      <td>1512105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1511955-1512105</td>\n",
       "      <td>DNase-seq:chr1:1511955-1512105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52920</th>\n",
       "      <td>chr1:1540281-1541159</td>\n",
       "      <td>1</td>\n",
       "      <td>1540281</td>\n",
       "      <td>1541159</td>\n",
       "      <td>ATAC-seq:chr1:1540281-1541159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1540281-1541159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53046</th>\n",
       "      <td>chr1:1551073-1551223</td>\n",
       "      <td>1</td>\n",
       "      <td>1551073</td>\n",
       "      <td>1551223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1551073-1551223</td>\n",
       "      <td>DNase-seq:chr1:1551073-1551223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53100</th>\n",
       "      <td>chr1:1554491-1555433</td>\n",
       "      <td>1</td>\n",
       "      <td>1554491</td>\n",
       "      <td>1555433</td>\n",
       "      <td>ATAC-seq:chr1:1554491-1555433</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1554491-1555433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53257</th>\n",
       "      <td>chr1:1567475-1567625</td>\n",
       "      <td>1</td>\n",
       "      <td>1567475</td>\n",
       "      <td>1567625</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:1567475-1567625</td>\n",
       "      <td>DNase-seq:chr1:1567475-1567625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      index Chr    Start      End  \\\n",
       "54899    chr1:181415-181565   1   181415   181565   \n",
       "59412    chr1:267990-268140   1   267990   268140   \n",
       "62036    chr1:629562-630131   1   629562   630131   \n",
       "62038    chr1:629870-630020   1   629870   630020   \n",
       "62041    chr1:630598-630748   1   630598   630748   \n",
       "...                     ...  ..      ...      ...   \n",
       "52686  chr1:1511955-1512105   1  1511955  1512105   \n",
       "52920  chr1:1540281-1541159   1  1540281  1541159   \n",
       "53046  chr1:1551073-1551223   1  1551073  1551223   \n",
       "53100  chr1:1554491-1555433   1  1554491  1555433   \n",
       "53257  chr1:1567475-1567625   1  1567475  1567625   \n",
       "\n",
       "                             ATAC_MT                        DNase_MT  \\\n",
       "54899                            NaN    DNase-seq:chr1:181415-181565   \n",
       "59412                            NaN    DNase-seq:chr1:267990-268140   \n",
       "62036    ATAC-seq:chr1:629562-630131                             NaN   \n",
       "62038                            NaN    DNase-seq:chr1:629870-630020   \n",
       "62041                            NaN    DNase-seq:chr1:630598-630748   \n",
       "...                              ...                             ...   \n",
       "52686                            NaN  DNase-seq:chr1:1511955-1512105   \n",
       "52920  ATAC-seq:chr1:1540281-1541159                             NaN   \n",
       "53046                            NaN  DNase-seq:chr1:1551073-1551223   \n",
       "53100  ATAC-seq:chr1:1554491-1555433                             NaN   \n",
       "53257                            NaN  DNase-seq:chr1:1567475-1567625   \n",
       "\n",
       "                        OpenChromatin  \n",
       "54899    DNase-seq:chr1:181415-181565  \n",
       "59412    DNase-seq:chr1:267990-268140  \n",
       "62036     ATAC-seq:chr1:629562-630131  \n",
       "62038    DNase-seq:chr1:629870-630020  \n",
       "62041    DNase-seq:chr1:630598-630748  \n",
       "...                               ...  \n",
       "52686  DNase-seq:chr1:1511955-1512105  \n",
       "52920   ATAC-seq:chr1:1540281-1541159  \n",
       "53046  DNase-seq:chr1:1551073-1551223  \n",
       "53100   ATAC-seq:chr1:1554491-1555433  \n",
       "53257  DNase-seq:chr1:1567475-1567625  \n",
       "\n",
       "[100 rows x 7 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted.head(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2f08790f-b04c-4444-ad34-a91ca893ace2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [index, Chr, Start, End, ATAC_MT, DNase_MT, OpenChromatin]\n",
       "Index: []"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted[merge_atac_dnase_sorted['ATAC_MT'].notna()&merge_atac_dnase_sorted['DNase_MT'].notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6e472aa7-b594-416b-9ab3-c8ddad47c265",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ATAC_MT</th>\n",
       "      <th>DNase_MT</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>chr1:181415-181565</td>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>chr1:267990-268140</td>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>chr1:629562-630131</td>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>chr1:629870-630020</td>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>chr1:630598-630748</td>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139576</th>\n",
       "      <td>chrY:26359975-26360125</td>\n",
       "      <td>Y</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139577</th>\n",
       "      <td>chrY:26365821-26366654</td>\n",
       "      <td>Y</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139588</th>\n",
       "      <td>chrY:56727995-56728145</td>\n",
       "      <td>Y</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139589</th>\n",
       "      <td>chrY:56734800-56734950</td>\n",
       "      <td>Y</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139590</th>\n",
       "      <td>chrY:56872355-56872505</td>\n",
       "      <td>Y</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index Chr     Start       End  \\\n",
       "54899       chr1:181415-181565   1    181415    181565   \n",
       "59412       chr1:267990-268140   1    267990    268140   \n",
       "62036       chr1:629562-630131   1    629562    630131   \n",
       "62038       chr1:629870-630020   1    629870    630020   \n",
       "62041       chr1:630598-630748   1    630598    630748   \n",
       "...                        ...  ..       ...       ...   \n",
       "139576  chrY:26359975-26360125   Y  26359975  26360125   \n",
       "139577  chrY:26365821-26366654   Y  26365821  26366654   \n",
       "139588  chrY:56727995-56728145   Y  56727995  56728145   \n",
       "139589  chrY:56734800-56734950   Y  56734800  56734950   \n",
       "139590  chrY:56872355-56872505   Y  56872355  56872505   \n",
       "\n",
       "                                ATAC_MT                          DNase_MT  \\\n",
       "54899                               NaN      DNase-seq:chr1:181415-181565   \n",
       "59412                               NaN      DNase-seq:chr1:267990-268140   \n",
       "62036       ATAC-seq:chr1:629562-630131                               NaN   \n",
       "62038                               NaN      DNase-seq:chr1:629870-630020   \n",
       "62041                               NaN      DNase-seq:chr1:630598-630748   \n",
       "...                                 ...                               ...   \n",
       "139576                              NaN  DNase-seq:chrY:26359975-26360125   \n",
       "139577  ATAC-seq:chrY:26365821-26366654                               NaN   \n",
       "139588                              NaN  DNase-seq:chrY:56727995-56728145   \n",
       "139589                              NaN  DNase-seq:chrY:56734800-56734950   \n",
       "139590                              NaN  DNase-seq:chrY:56872355-56872505   \n",
       "\n",
       "                           OpenChromatin  \n",
       "54899       DNase-seq:chr1:181415-181565  \n",
       "59412       DNase-seq:chr1:267990-268140  \n",
       "62036        ATAC-seq:chr1:629562-630131  \n",
       "62038       DNase-seq:chr1:629870-630020  \n",
       "62041       DNase-seq:chr1:630598-630748  \n",
       "...                                  ...  \n",
       "139576  DNase-seq:chrY:26359975-26360125  \n",
       "139577   ATAC-seq:chrY:26365821-26366654  \n",
       "139588  DNase-seq:chrY:56727995-56728145  \n",
       "139589  DNase-seq:chrY:56734800-56734950  \n",
       "139590  DNase-seq:chrY:56872355-56872505  \n",
       "\n",
       "[139651 rows x 7 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_atac_dnase_sorted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "70d2b8f7-5097-4a33-a80d-97ccf4b65369",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_merge_atac_dnase_sorted_bed = merge_atac_dnase_sorted[[\"Chr\",\"Start\",\"End\",\"OpenChromatin\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "cb43fd4d-d539-48af-9397-de97ee407b8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139576</th>\n",
       "      <td>Y</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139577</th>\n",
       "      <td>Y</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139588</th>\n",
       "      <td>Y</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139589</th>\n",
       "      <td>Y</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139590</th>\n",
       "      <td>Y</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End                     OpenChromatin\n",
       "54899    1    181415    181565      DNase-seq:chr1:181415-181565\n",
       "59412    1    267990    268140      DNase-seq:chr1:267990-268140\n",
       "62036    1    629562    630131       ATAC-seq:chr1:629562-630131\n",
       "62038    1    629870    630020      DNase-seq:chr1:629870-630020\n",
       "62041    1    630598    630748      DNase-seq:chr1:630598-630748\n",
       "...     ..       ...       ...                               ...\n",
       "139576   Y  26359975  26360125  DNase-seq:chrY:26359975-26360125\n",
       "139577   Y  26365821  26366654   ATAC-seq:chrY:26365821-26366654\n",
       "139588   Y  56727995  56728145  DNase-seq:chrY:56727995-56728145\n",
       "139589   Y  56734800  56734950  DNase-seq:chrY:56734800-56734950\n",
       "139590   Y  56872355  56872505  DNase-seq:chrY:56872355-56872505\n",
       "\n",
       "[139651 rows x 4 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "298a4b03-0774-44a3-9c4c-2d0a5e04a25c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_4105959/3441541605.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  MT_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + MT_merge_atac_dnase_sorted_bed['Chr'].astype(str)\n"
     ]
    }
   ],
   "source": [
    "MT_merge_atac_dnase_sorted_bed['Chr'] = 'chr' + MT_merge_atac_dnase_sorted_bed['Chr'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "1ca0b26a-0d6c-473c-8b62-56054a166587",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>OpenChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>54899</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181565</td>\n",
       "      <td>DNase-seq:chr1:181415-181565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59412</th>\n",
       "      <td>chr1</td>\n",
       "      <td>267990</td>\n",
       "      <td>268140</td>\n",
       "      <td>DNase-seq:chr1:267990-268140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62036</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629562</td>\n",
       "      <td>630131</td>\n",
       "      <td>ATAC-seq:chr1:629562-630131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62038</th>\n",
       "      <td>chr1</td>\n",
       "      <td>629870</td>\n",
       "      <td>630020</td>\n",
       "      <td>DNase-seq:chr1:629870-630020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62041</th>\n",
       "      <td>chr1</td>\n",
       "      <td>630598</td>\n",
       "      <td>630748</td>\n",
       "      <td>DNase-seq:chr1:630598-630748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139576</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26359975</td>\n",
       "      <td>26360125</td>\n",
       "      <td>DNase-seq:chrY:26359975-26360125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139577</th>\n",
       "      <td>chrY</td>\n",
       "      <td>26365821</td>\n",
       "      <td>26366654</td>\n",
       "      <td>ATAC-seq:chrY:26365821-26366654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139588</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56727995</td>\n",
       "      <td>56728145</td>\n",
       "      <td>DNase-seq:chrY:56727995-56728145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139589</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56734800</td>\n",
       "      <td>56734950</td>\n",
       "      <td>DNase-seq:chrY:56734800-56734950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139590</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56872355</td>\n",
       "      <td>56872505</td>\n",
       "      <td>DNase-seq:chrY:56872355-56872505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139651 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr     Start       End                     OpenChromatin\n",
       "54899   chr1    181415    181565      DNase-seq:chr1:181415-181565\n",
       "59412   chr1    267990    268140      DNase-seq:chr1:267990-268140\n",
       "62036   chr1    629562    630131       ATAC-seq:chr1:629562-630131\n",
       "62038   chr1    629870    630020      DNase-seq:chr1:629870-630020\n",
       "62041   chr1    630598    630748      DNase-seq:chr1:630598-630748\n",
       "...      ...       ...       ...                               ...\n",
       "139576  chrY  26359975  26360125  DNase-seq:chrY:26359975-26360125\n",
       "139577  chrY  26365821  26366654   ATAC-seq:chrY:26365821-26366654\n",
       "139588  chrY  56727995  56728145  DNase-seq:chrY:56727995-56728145\n",
       "139589  chrY  56734800  56734950  DNase-seq:chrY:56734800-56734950\n",
       "139590  chrY  56872355  56872505  DNase-seq:chrY:56872355-56872505\n",
       "\n",
       "[139651 rows x 4 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_merge_atac_dnase_sorted_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "96f1f652-46bd-4e89-a9f6-6e7421b59c60",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_merge_atac_dnase_sorted_bed.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/myotube/hg38_OpenChromatin_MT.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb553935-ac7f-4f70-9307-041aec35b3be",
   "metadata": {},
   "source": [
    "### promoter table overlap with openchromatin "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "3b9a02b2-ca3d-4686-9901-420de1720a8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/promoter_regions_pcg_250bp_sorted.bed -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/openChromatin/myotube/hg38_OpenChromatin_MT.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3fd31f1-2423-41ea-b4f9-7b244cd80535",
   "metadata": {},
   "source": [
    "### remove duplicated rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "f53ae2b1-85be-4e39-86d1-5fa0dc43fb92",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'chr_open','start_open','end_open','openChromatin','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "5f46f4d7-96c6-4b1a-b74d-b29cb1593c1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "3666029e-908b-4ee6-802b-bd23f7877331",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241883</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241884</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241885</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241886</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241887</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>241888 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "241883  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "241884  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "241885  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "241886  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "241887  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923647    924887   \n",
       "4                            5.0     chr1      923647    924887   \n",
       "...                          ...      ...         ...       ...   \n",
       "241883                       5.0        .          -1        -1   \n",
       "241884                       1.0        .          -1        -1   \n",
       "241885                       2.0        .          -1        -1   \n",
       "241886                       1.0        .          -1        -1   \n",
       "241887                       2.0        .          -1        -1   \n",
       "\n",
       "                      openChromatin  overlap  \n",
       "0                                 .        0  \n",
       "1                                 .        0  \n",
       "2                                 .        0  \n",
       "3       ATAC-seq:chr1:923647-924887      500  \n",
       "4       ATAC-seq:chr1:923647-924887      500  \n",
       "...                             ...      ...  \n",
       "241883                            .        0  \n",
       "241884                            .        0  \n",
       "241885                            .        0  \n",
       "241886                            .        0  \n",
       "241887                            .        0  \n",
       "\n",
       "[241888 rows x 13 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "449c8781-8235-4b95-a974-533f204d70c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>976006</td>\n",
       "      <td>976687</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "23  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "24  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "25  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "26  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "27  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "28  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "29  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "30  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "31  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "32  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "33  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "34  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "35  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "36  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "37  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "38  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "39  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "40  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "41  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "42  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "43  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "44  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "45  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "46  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "47  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "48  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "49  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923647    924887   \n",
       "4                        5.0     chr1      923647    924887   \n",
       "5                        5.0        .          -1        -1   \n",
       "6                        5.0        .          -1        -1   \n",
       "7                        5.0        .          -1        -1   \n",
       "8                        5.0        .          -1        -1   \n",
       "9                        5.0        .          -1        -1   \n",
       "10                       5.0        .          -1        -1   \n",
       "11                       5.0        .          -1        -1   \n",
       "12                       5.0        .          -1        -1   \n",
       "13                       2.0        .          -1        -1   \n",
       "14                       5.0     chr1      940001    941223   \n",
       "15                       5.0     chr1      940001    941223   \n",
       "16                       3.0        .          -1        -1   \n",
       "17                       2.0        .          -1        -1   \n",
       "18                       5.0        .          -1        -1   \n",
       "19                       2.0        .          -1        -1   \n",
       "20                       5.0        .          -1        -1   \n",
       "21                       1.0     chr1      958785    959675   \n",
       "22                       1.0     chr1      959223    959373   \n",
       "23                       2.0     chr1      958785    959675   \n",
       "24                       2.0     chr1      959223    959373   \n",
       "25                       3.0     chr1      958785    959675   \n",
       "26                       3.0     chr1      959223    959373   \n",
       "27                       1.0     chr1      960473    961422   \n",
       "28                       1.0     chr1      960473    961422   \n",
       "29                       3.0        .          -1        -1   \n",
       "30                       2.0        .          -1        -1   \n",
       "31                       1.0     chr1      966276    967323   \n",
       "32                       1.0     chr1      966276    967323   \n",
       "33                       2.0     chr1      966276    967323   \n",
       "34                       3.0        .          -1        -1   \n",
       "35                       3.0        .          -1        -1   \n",
       "36                       1.0     chr1      976006    976687   \n",
       "37                       2.0        .          -1        -1   \n",
       "38                       5.0        .          -1        -1   \n",
       "39                       NaN        .          -1        -1   \n",
       "40                       3.0     chr1      999640   1000919   \n",
       "41                       3.0     chr1     1000105   1000255   \n",
       "42                       2.0     chr1      999640   1000919   \n",
       "43                       2.0     chr1     1000105   1000255   \n",
       "44                       1.0     chr1      999640   1000919   \n",
       "45                       1.0     chr1     1000105   1000255   \n",
       "46                       2.0     chr1      999640   1000919   \n",
       "47                       2.0     chr1     1000105   1000255   \n",
       "48                       3.0     chr1      999640   1000919   \n",
       "49                       3.0     chr1      999640   1000919   \n",
       "\n",
       "                     openChromatin  overlap  \n",
       "0                                .        0  \n",
       "1                                .        0  \n",
       "2                                .        0  \n",
       "3      ATAC-seq:chr1:923647-924887      500  \n",
       "4      ATAC-seq:chr1:923647-924887      500  \n",
       "5                                .        0  \n",
       "6                                .        0  \n",
       "7                                .        0  \n",
       "8                                .        0  \n",
       "9                                .        0  \n",
       "10                               .        0  \n",
       "11                               .        0  \n",
       "12                               .        0  \n",
       "13                               .        0  \n",
       "14     ATAC-seq:chr1:940001-941223      500  \n",
       "15     ATAC-seq:chr1:940001-941223      397  \n",
       "16                               .        0  \n",
       "17                               .        0  \n",
       "18                               .        0  \n",
       "19                               .        0  \n",
       "20                               .        0  \n",
       "21     ATAC-seq:chr1:958785-959675      500  \n",
       "22    DNase-seq:chr1:959223-959373      150  \n",
       "23     ATAC-seq:chr1:958785-959675      500  \n",
       "24    DNase-seq:chr1:959223-959373      150  \n",
       "25     ATAC-seq:chr1:958785-959675      500  \n",
       "26    DNase-seq:chr1:959223-959373      150  \n",
       "27     ATAC-seq:chr1:960473-961422      361  \n",
       "28     ATAC-seq:chr1:960473-961422      223  \n",
       "29                               .        0  \n",
       "30                               .        0  \n",
       "31     ATAC-seq:chr1:966276-967323      456  \n",
       "32     ATAC-seq:chr1:966276-967323      476  \n",
       "33     ATAC-seq:chr1:966276-967323      476  \n",
       "34                               .        0  \n",
       "35                               .        0  \n",
       "36     ATAC-seq:chr1:976006-976687      296  \n",
       "37                               .        0  \n",
       "38                               .        0  \n",
       "39                               .        0  \n",
       "40    ATAC-seq:chr1:999640-1000919      500  \n",
       "41  DNase-seq:chr1:1000105-1000255      126  \n",
       "42    ATAC-seq:chr1:999640-1000919      500  \n",
       "43  DNase-seq:chr1:1000105-1000255      150  \n",
       "44    ATAC-seq:chr1:999640-1000919      500  \n",
       "45  DNase-seq:chr1:1000105-1000255      150  \n",
       "46    ATAC-seq:chr1:999640-1000919      500  \n",
       "47  DNase-seq:chr1:1000105-1000255      150  \n",
       "48    ATAC-seq:chr1:999640-1000919       31  \n",
       "49    ATAC-seq:chr1:999640-1000919       24  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "c87fb1ef-12a8-43f4-b3fd-4af9a1b4982d",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "ad368f03-a970-4982-b1ba-5a35b74079f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241883</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241884</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241885</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241886</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241887</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>241888 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "241883  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "241884  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "241885  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "241886  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "241887  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                            NaN        .          -1        -1   \n",
       "1                            NaN        .          -1        -1   \n",
       "2                            NaN        .          -1        -1   \n",
       "3                            5.0     chr1      923647    924887   \n",
       "4                            5.0     chr1      923647    924887   \n",
       "...                          ...      ...         ...       ...   \n",
       "241883                       5.0        .          -1        -1   \n",
       "241884                       1.0        .          -1        -1   \n",
       "241885                       2.0        .          -1        -1   \n",
       "241886                       1.0        .          -1        -1   \n",
       "241887                       2.0        .          -1        -1   \n",
       "\n",
       "                      openChromatin  overlap  \n",
       "0                                 .        0  \n",
       "1                                 .        0  \n",
       "2                                 .        0  \n",
       "3       ATAC-seq:chr1:923647-924887      500  \n",
       "4       ATAC-seq:chr1:923647-924887      500  \n",
       "...                             ...      ...  \n",
       "241883                            .        0  \n",
       "241884                            .        0  \n",
       "241885                            .        0  \n",
       "241886                            .        0  \n",
       "241887                            .        0  \n",
       "\n",
       "[241888 rows x 13 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "b8e5b8f5-db77-4d35-baf9-772c5bd2fcc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>chr_open</th>\n",
       "      <th>start_open</th>\n",
       "      <th>end_open</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923647</td>\n",
       "      <td>924887</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>940001</td>\n",
       "      <td>941223</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>958785</td>\n",
       "      <td>959675</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>959223</td>\n",
       "      <td>959373</td>\n",
       "      <td>DNase-seq:chr1:959223-959373</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>960473</td>\n",
       "      <td>961422</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>966276</td>\n",
       "      <td>967323</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>976006</td>\n",
       "      <td>976687</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1000105</td>\n",
       "      <td>1000255</td>\n",
       "      <td>DNase-seq:chr1:1000105-1000255</td>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>999640</td>\n",
       "      <td>1000919</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "23  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "24  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "25  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "26  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "27  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "28  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "29  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "30  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "31  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "32  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "33  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "34  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "35  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "36  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "37  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "38  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "39  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "40  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "41  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "42  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "43  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "44  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "45  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "46  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "47  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "48  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "49  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "\n",
       "    transcript_support_level chr_open  start_open  end_open  \\\n",
       "0                        NaN        .          -1        -1   \n",
       "1                        NaN        .          -1        -1   \n",
       "2                        NaN        .          -1        -1   \n",
       "3                        5.0     chr1      923647    924887   \n",
       "4                        5.0     chr1      923647    924887   \n",
       "5                        5.0        .          -1        -1   \n",
       "6                        5.0        .          -1        -1   \n",
       "7                        5.0        .          -1        -1   \n",
       "8                        5.0        .          -1        -1   \n",
       "9                        5.0        .          -1        -1   \n",
       "10                       5.0        .          -1        -1   \n",
       "11                       5.0        .          -1        -1   \n",
       "12                       5.0        .          -1        -1   \n",
       "13                       2.0        .          -1        -1   \n",
       "14                       5.0     chr1      940001    941223   \n",
       "15                       5.0     chr1      940001    941223   \n",
       "16                       3.0        .          -1        -1   \n",
       "17                       2.0        .          -1        -1   \n",
       "18                       5.0        .          -1        -1   \n",
       "19                       2.0        .          -1        -1   \n",
       "20                       5.0        .          -1        -1   \n",
       "21                       1.0     chr1      958785    959675   \n",
       "22                       1.0     chr1      959223    959373   \n",
       "23                       2.0     chr1      958785    959675   \n",
       "24                       2.0     chr1      959223    959373   \n",
       "25                       3.0     chr1      958785    959675   \n",
       "26                       3.0     chr1      959223    959373   \n",
       "27                       1.0     chr1      960473    961422   \n",
       "28                       1.0     chr1      960473    961422   \n",
       "29                       3.0        .          -1        -1   \n",
       "30                       2.0        .          -1        -1   \n",
       "31                       1.0     chr1      966276    967323   \n",
       "32                       1.0     chr1      966276    967323   \n",
       "33                       2.0     chr1      966276    967323   \n",
       "34                       3.0        .          -1        -1   \n",
       "35                       3.0        .          -1        -1   \n",
       "36                       1.0     chr1      976006    976687   \n",
       "37                       2.0        .          -1        -1   \n",
       "38                       5.0        .          -1        -1   \n",
       "39                       NaN        .          -1        -1   \n",
       "40                       3.0     chr1      999640   1000919   \n",
       "41                       3.0     chr1     1000105   1000255   \n",
       "42                       2.0     chr1      999640   1000919   \n",
       "43                       2.0     chr1     1000105   1000255   \n",
       "44                       1.0     chr1      999640   1000919   \n",
       "45                       1.0     chr1     1000105   1000255   \n",
       "46                       2.0     chr1      999640   1000919   \n",
       "47                       2.0     chr1     1000105   1000255   \n",
       "48                       3.0     chr1      999640   1000919   \n",
       "49                       3.0     chr1      999640   1000919   \n",
       "\n",
       "                     openChromatin  overlap  \n",
       "0                                .        0  \n",
       "1                                .        0  \n",
       "2                                .        0  \n",
       "3      ATAC-seq:chr1:923647-924887      500  \n",
       "4      ATAC-seq:chr1:923647-924887      500  \n",
       "5                                .        0  \n",
       "6                                .        0  \n",
       "7                                .        0  \n",
       "8                                .        0  \n",
       "9                                .        0  \n",
       "10                               .        0  \n",
       "11                               .        0  \n",
       "12                               .        0  \n",
       "13                               .        0  \n",
       "14     ATAC-seq:chr1:940001-941223      500  \n",
       "15     ATAC-seq:chr1:940001-941223      397  \n",
       "16                               .        0  \n",
       "17                               .        0  \n",
       "18                               .        0  \n",
       "19                               .        0  \n",
       "20                               .        0  \n",
       "21     ATAC-seq:chr1:958785-959675      500  \n",
       "22    DNase-seq:chr1:959223-959373      150  \n",
       "23     ATAC-seq:chr1:958785-959675      500  \n",
       "24    DNase-seq:chr1:959223-959373      150  \n",
       "25     ATAC-seq:chr1:958785-959675      500  \n",
       "26    DNase-seq:chr1:959223-959373      150  \n",
       "27     ATAC-seq:chr1:960473-961422      361  \n",
       "28     ATAC-seq:chr1:960473-961422      223  \n",
       "29                               .        0  \n",
       "30                               .        0  \n",
       "31     ATAC-seq:chr1:966276-967323      456  \n",
       "32     ATAC-seq:chr1:966276-967323      476  \n",
       "33     ATAC-seq:chr1:966276-967323      476  \n",
       "34                               .        0  \n",
       "35                               .        0  \n",
       "36     ATAC-seq:chr1:976006-976687      296  \n",
       "37                               .        0  \n",
       "38                               .        0  \n",
       "39                               .        0  \n",
       "40    ATAC-seq:chr1:999640-1000919      500  \n",
       "41  DNase-seq:chr1:1000105-1000255      126  \n",
       "42    ATAC-seq:chr1:999640-1000919      500  \n",
       "43  DNase-seq:chr1:1000105-1000255      150  \n",
       "44    ATAC-seq:chr1:999640-1000919      500  \n",
       "45  DNase-seq:chr1:1000105-1000255      150  \n",
       "46    ATAC-seq:chr1:999640-1000919      500  \n",
       "47  DNase-seq:chr1:1000105-1000255      150  \n",
       "48    ATAC-seq:chr1:999640-1000919       31  \n",
       "49    ATAC-seq:chr1:999640-1000919       24  "
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "61d50afc-ba20-4f1c-ac8d-ba760187b092",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to Type)\n",
    "group_cols = promoter_like_open.columns[:8]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "dccec6f3-8b09-454d-90a9-7dd12899d7e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open.groupby(list(group_cols), dropna=False)['openChromatin'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "4558aa29-a1bb-4237-a26a-ad8dec8070b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin  \n",
       "0                            NaN                            .  \n",
       "1                            NaN                            .  \n",
       "2                            NaN                            .  \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887  \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887  \n",
       "...                          ...                          ...  \n",
       "170552                       5.0                            .  \n",
       "170553                       1.0                            .  \n",
       "170554                       2.0                            .  \n",
       "170555                       1.0                            .  \n",
       "170556                       2.0                            .  \n",
       "\n",
       "[170557 rows x 9 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "f1a85a30-c934-46c4-acd6-62af217bd97e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                        openChromatin  \n",
       "0                                                   .  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3                         ATAC-seq:chr1:923647-924887  \n",
       "4                         ATAC-seq:chr1:923647-924887  \n",
       "5                                                   .  \n",
       "6                                                   .  \n",
       "7                                                   .  \n",
       "8                                                   .  \n",
       "9                                                   .  \n",
       "10                                                  .  \n",
       "11                                                  .  \n",
       "12                                                  .  \n",
       "13                                                  .  \n",
       "14                        ATAC-seq:chr1:940001-941223  \n",
       "15                        ATAC-seq:chr1:940001-941223  \n",
       "16                                                  .  \n",
       "17                                                  .  \n",
       "18                                                  .  \n",
       "19                                                  .  \n",
       "20                                                  .  \n",
       "21  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...  \n",
       "22  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...  \n",
       "23  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...  \n",
       "24                        ATAC-seq:chr1:960473-961422  \n",
       "25                        ATAC-seq:chr1:960473-961422  \n",
       "26                                                  .  \n",
       "27                                                  .  \n",
       "28                        ATAC-seq:chr1:966276-967323  \n",
       "29                        ATAC-seq:chr1:966276-967323  \n",
       "30                        ATAC-seq:chr1:966276-967323  \n",
       "31                                                  .  \n",
       "32                                                  .  \n",
       "33                        ATAC-seq:chr1:976006-976687  \n",
       "34                                                  .  \n",
       "35                                                  .  \n",
       "36                                                  .  \n",
       "37  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "38  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "39  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "40  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...  \n",
       "41                       ATAC-seq:chr1:999640-1000919  \n",
       "42                       ATAC-seq:chr1:999640-1000919  \n",
       "43                      ATAC-seq:chr1:1012923-1014210  \n",
       "44                      ATAC-seq:chr1:1019149-1020697  \n",
       "45                      ATAC-seq:chr1:1019149-1020697  \n",
       "46                                                  .  \n",
       "47                                                  .  \n",
       "48                                                  .  \n",
       "49                                                  .  "
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "70759c9e-2a8a-470b-af36-c276ffb07909",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51e718c2-56ec-4903-8f6b-cf50fdaff699",
   "metadata": {},
   "source": [
    "## promoter table overlap with chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "06ae8b61-4fca-4708-b3c6-da6389dc21aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"ChromHMM_Myotube\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "b5361981-a1cb-41ec-bdea-c4116e9226a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E121_25_myotube/hg38_chromHMM_E121_25.txt',\n",
    "                          sep='\\t',\n",
    "                          header=None,\n",
    "                          names=cols\n",
    "                          )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "e87a09c4-0ed6-4cb7-98ea-6112acd69466",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_Myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>10200</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10200</td>\n",
       "      <td>10600</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891173</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155966935</td>\n",
       "      <td>155967135</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891174</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155967135</td>\n",
       "      <td>156030135</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891175</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156030135</td>\n",
       "      <td>156030735</td>\n",
       "      <td>19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891176</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891177</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>891178 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end ChromHMM_Myotube\n",
       "0       chr1      10000      10200         25_Quies\n",
       "1       chr1      10200      10600         19_DNase\n",
       "2       chr1      10600      96080         25_Quies\n",
       "3       chr1      96276      96476         25_Quies\n",
       "4       chr1      97276     177200         25_Quies\n",
       "...      ...        ...        ...              ...\n",
       "891173  chrX  155966935  155967135         19_DNase\n",
       "891174  chrX  155967135  156030135         25_Quies\n",
       "891175  chrX  156030135  156030735         19_DNase\n",
       "891176  chrY    1086142    1086334         25_Quies\n",
       "891177  chrY    1198191    1198391         25_Quies\n",
       "\n",
       "[891178 rows x 4 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1fcc068f-3f0d-4f23-8378-fcc0e7ae08a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_chromHMM['ChromHMM_Myotube'] = MT_chromHMM['Chrom'] + \":\" + MT_chromHMM['start'].astype('str') + \":\" + MT_chromHMM['end'].astype('str') + \":\" + MT_chromHMM['ChromHMM_Myotube']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "b7e24ff8-e263-44c6-b30e-dce9e991e152",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ChromHMM_Myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10000</td>\n",
       "      <td>10200</td>\n",
       "      <td>chr1:10000:10200:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10200</td>\n",
       "      <td>10600</td>\n",
       "      <td>chr1:10200:10600:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>96276</td>\n",
       "      <td>96476</td>\n",
       "      <td>chr1:96276:96476:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>97276</td>\n",
       "      <td>177200</td>\n",
       "      <td>chr1:97276:177200:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891173</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155966935</td>\n",
       "      <td>155967135</td>\n",
       "      <td>chrX:155966935:155967135:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891174</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155967135</td>\n",
       "      <td>156030135</td>\n",
       "      <td>chrX:155967135:156030135:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891175</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156030135</td>\n",
       "      <td>156030735</td>\n",
       "      <td>chrX:156030135:156030735:19_DNase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891176</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1086142</td>\n",
       "      <td>1086334</td>\n",
       "      <td>chrY:1086142:1086334:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891177</th>\n",
       "      <td>chrY</td>\n",
       "      <td>1198191</td>\n",
       "      <td>1198391</td>\n",
       "      <td>chrY:1198191:1198391:25_Quies</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>891178 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom      start        end                   ChromHMM_Myotube\n",
       "0       chr1      10000      10200          chr1:10000:10200:25_Quies\n",
       "1       chr1      10200      10600          chr1:10200:10600:19_DNase\n",
       "2       chr1      10600      96080          chr1:10600:96080:25_Quies\n",
       "3       chr1      96276      96476          chr1:96276:96476:25_Quies\n",
       "4       chr1      97276     177200         chr1:97276:177200:25_Quies\n",
       "...      ...        ...        ...                                ...\n",
       "891173  chrX  155966935  155967135  chrX:155966935:155967135:19_DNase\n",
       "891174  chrX  155967135  156030135  chrX:155967135:156030135:25_Quies\n",
       "891175  chrX  156030135  156030735  chrX:156030135:156030735:19_DNase\n",
       "891176  chrY    1086142    1086334      chrY:1086142:1086334:25_Quies\n",
       "891177  chrY    1198191    1198391      chrY:1198191:1198391:25_Quies\n",
       "\n",
       "[891178 rows x 4 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MT_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "5efa4563-d041-49bd-95e8-72c56adcd6b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "MT_chromHMM.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E121_25_myotube/hg38_chromHMM_myotube.bed',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "89b342b8-c2c6-4e67-bd2c-e4e88eef71d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin.bed_uniq -b /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/chromHMM_E121_25_myotube/hg38_chromHMM_myotube.bed -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "c5e71ecf-dc77-4f08-9ec0-3e1c12adadef",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'openChromatin','chr_HMM','start_HMM','end_HMM','chromHMM_myotube','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "75f9f5e3-77f8-4174-bc66-67464437ffa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "e1ddb9b9-2d80-4103-9525-b432d43ff309",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chr_HMM</th>\n",
       "      <th>start_HMM</th>\n",
       "      <th>end_HMM</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>348136</td>\n",
       "      <td>501536</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>634820</td>\n",
       "      <td>733820</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923420</td>\n",
       "      <td>923820</td>\n",
       "      <td>chr1:923420:923820:2_PromU</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923820</td>\n",
       "      <td>924620</td>\n",
       "      <td>chr1:923820:924620:22_PromP</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290239</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290240</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290241</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290242</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290243</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>290244 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "290239  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "290240  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "290241  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "290242  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "290243  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin chr_HMM  \\\n",
       "0                            NaN                            .    chr1   \n",
       "1                            NaN                            .    chr1   \n",
       "2                            NaN                            .    chr1   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "...                          ...                          ...     ...   \n",
       "290239                       5.0                            .       .   \n",
       "290240                       1.0                            .       .   \n",
       "290241                       2.0                            .       .   \n",
       "290242                       1.0                            .       .   \n",
       "290243                       2.0                            .       .   \n",
       "\n",
       "        start_HMM  end_HMM             chromHMM_myotube  overlap  \n",
       "0           10600    96080    chr1:10600:96080:25_Quies      500  \n",
       "1          348136   501536  chr1:348136:501536:25_Quies      500  \n",
       "2          634820   733820  chr1:634820:733820:25_Quies      500  \n",
       "3          923420   923820   chr1:923420:923820:2_PromU      147  \n",
       "4          923820   924620  chr1:923820:924620:22_PromP      353  \n",
       "...           ...      ...                          ...      ...  \n",
       "290239         -1       -1                            .        0  \n",
       "290240         -1       -1                            .        0  \n",
       "290241         -1       -1                            .        0  \n",
       "290242         -1       -1                            .        0  \n",
       "290243         -1       -1                            .        0  \n",
       "\n",
       "[290244 rows x 14 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "3f9725c9-7ae4-4bd2-8bea-71be49de0499",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "07e9feff-7bb2-45f3-90fd-0548cc9c6104",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chr_HMM</th>\n",
       "      <th>start_HMM</th>\n",
       "      <th>end_HMM</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10600</td>\n",
       "      <td>96080</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>348136</td>\n",
       "      <td>501536</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1</td>\n",
       "      <td>634820</td>\n",
       "      <td>733820</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923420</td>\n",
       "      <td>923820</td>\n",
       "      <td>chr1:923420:923820:2_PromU</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1</td>\n",
       "      <td>923820</td>\n",
       "      <td>924620</td>\n",
       "      <td>chr1:923820:924620:22_PromP</td>\n",
       "      <td>353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290239</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290240</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290241</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290242</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290243</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>290244 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "290239  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "290240  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "290241  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "290242  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "290243  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin chr_HMM  \\\n",
       "0                            NaN                            .    chr1   \n",
       "1                            NaN                            .    chr1   \n",
       "2                            NaN                            .    chr1   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887    chr1   \n",
       "...                          ...                          ...     ...   \n",
       "290239                       5.0                            .       .   \n",
       "290240                       1.0                            .       .   \n",
       "290241                       2.0                            .       .   \n",
       "290242                       1.0                            .       .   \n",
       "290243                       2.0                            .       .   \n",
       "\n",
       "        start_HMM  end_HMM             chromHMM_myotube  overlap  \n",
       "0           10600    96080    chr1:10600:96080:25_Quies      500  \n",
       "1          348136   501536  chr1:348136:501536:25_Quies      500  \n",
       "2          634820   733820  chr1:634820:733820:25_Quies      500  \n",
       "3          923420   923820   chr1:923420:923820:2_PromU      147  \n",
       "4          923820   924620  chr1:923820:924620:22_PromP      353  \n",
       "...           ...      ...                          ...      ...  \n",
       "290239         -1       -1                            .        0  \n",
       "290240         -1       -1                            .        0  \n",
       "290241         -1       -1                            .        0  \n",
       "290242         -1       -1                            .        0  \n",
       "290243         -1       -1                            .        0  \n",
       "\n",
       "[290244 rows x 14 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "05252c80-26d5-443e-a025-3efeeb851b89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level', 'openChromatin'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to openChromatin)\n",
    "group_cols = promoter_like_open.columns[:9]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "1ee55dbf-6dd4-41e9-9e4d-99d7bae31daf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open.groupby(list(group_cols), dropna=False)['chromHMM_myotube'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "24ff1d6d-f8cc-4742-b862-0fdb6cc0d39c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level                openChromatin  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170552                       5.0                            .   \n",
       "170553                       1.0                            .   \n",
       "170554                       2.0                            .   \n",
       "170555                       1.0                            .   \n",
       "170556                       2.0                            .   \n",
       "\n",
       "                                         chromHMM_myotube  \n",
       "0                               chr1:10600:96080:25_Quies  \n",
       "1                             chr1:348136:501536:25_Quies  \n",
       "2                             chr1:634820:733820:25_Quies  \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 10 columns]"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "38e59480-fd89-428d-ac23-fcf67893c731",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin</th>\n",
       "      <th>chromHMM_myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:928020:933420:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938220:939220:17_EnhW2;chr1:939220:939420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:939820:940220:23_PromBiv;chr1:940220:9404...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:940420:941020:2_PromU;chr1:941020:941420:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:957820:958620:4_PromD2;chr1:958620:959220...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961220:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960420:961220:1_TssA;chr1:961220:961420:2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:962820:7_Tx3';chr1:962820:965820:8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962820:965820:8_TxWk</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:969220:972420:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972420:975820:25_Quies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>chr1:976020:976620:16_EnhW1;chr1:976620:980820...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:1000420:10020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10142...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1041020:1052220:8_TxWk</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                        openChromatin  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3                         ATAC-seq:chr1:923647-924887   \n",
       "4                         ATAC-seq:chr1:923647-924887   \n",
       "5                                                   .   \n",
       "6                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14                        ATAC-seq:chr1:940001-941223   \n",
       "15                        ATAC-seq:chr1:940001-941223   \n",
       "16                                                  .   \n",
       "17                                                  .   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                                                  .   \n",
       "21  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "24                        ATAC-seq:chr1:960473-961422   \n",
       "25                        ATAC-seq:chr1:960473-961422   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28                        ATAC-seq:chr1:966276-967323   \n",
       "29                        ATAC-seq:chr1:966276-967323   \n",
       "30                        ATAC-seq:chr1:966276-967323   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                        ATAC-seq:chr1:976006-976687   \n",
       "34                                                  .   \n",
       "35                                                  .   \n",
       "36                                                  .   \n",
       "37  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "41                       ATAC-seq:chr1:999640-1000919   \n",
       "42                       ATAC-seq:chr1:999640-1000919   \n",
       "43                      ATAC-seq:chr1:1012923-1014210   \n",
       "44                      ATAC-seq:chr1:1019149-1020697   \n",
       "45                      ATAC-seq:chr1:1019149-1020697   \n",
       "46                                                  .   \n",
       "47                                                  .   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                     chromHMM_myotube  \n",
       "0                           chr1:10600:96080:25_Quies  \n",
       "1                         chr1:348136:501536:25_Quies  \n",
       "2                         chr1:634820:733820:25_Quies  \n",
       "3   chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "4   chr1:923420:923820:2_PromU;chr1:923820:924620:...  \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...  \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:926820...  \n",
       "7                       chr1:925620:926820:23_PromBiv  \n",
       "8                       chr1:925620:926820:23_PromBiv  \n",
       "9                       chr1:925620:926820:23_PromBiv  \n",
       "10                      chr1:925620:926820:23_PromBiv  \n",
       "11                      chr1:925620:926820:23_PromBiv  \n",
       "12                        chr1:928020:933420:25_Quies  \n",
       "13  chr1:938220:939220:17_EnhW2;chr1:939220:939420...  \n",
       "14  chr1:939820:940220:23_PromBiv;chr1:940220:9404...  \n",
       "15  chr1:940420:941020:2_PromU;chr1:941020:941420:...  \n",
       "16                        chr1:941620:942620:22_PromP  \n",
       "17                        chr1:941620:942620:22_PromP  \n",
       "18                          chr1:945220:950020:7_Tx3'  \n",
       "19                          chr1:945220:950020:7_Tx3'  \n",
       "20  chr1:957820:958620:4_PromD2;chr1:958620:959220...  \n",
       "21  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "22  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "23  chr1:958620:959220:3_PromD1;chr1:959220:959420...  \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961220:...  \n",
       "25  chr1:960420:961220:1_TssA;chr1:961220:961420:2...  \n",
       "26  chr1:962420:962820:7_Tx3';chr1:962820:965820:8...  \n",
       "27                          chr1:962820:965820:8_TxWk  \n",
       "28                         chr1:966220:967220:2_PromU  \n",
       "29                         chr1:966220:967220:2_PromU  \n",
       "30                         chr1:966220:967220:2_PromU  \n",
       "31                       chr1:969220:972420:24_ReprPC  \n",
       "32                        chr1:972420:975820:25_Quies  \n",
       "33  chr1:976020:976620:16_EnhW1;chr1:976620:980820...  \n",
       "34                       chr1:980820:984220:24_ReprPC  \n",
       "35                       chr1:980820:984220:24_ReprPC  \n",
       "36                       chr1:980820:984220:24_ReprPC  \n",
       "37  chr1:1000020:1000420:1_TssA;chr1:999020:100002...  \n",
       "38  chr1:1000020:1000420:1_TssA;chr1:999020:100002...  \n",
       "39  chr1:1000020:1000420:1_TssA;chr1:999020:100002...  \n",
       "40  chr1:1000020:1000420:1_TssA;chr1:1000420:10020...  \n",
       "41                       chr1:1000420:1002020:2_PromU  \n",
       "42                       chr1:1000420:1002020:2_PromU  \n",
       "43  chr1:1013220:1013420:1_TssA;chr1:1013420:10142...  \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...  \n",
       "46                     chr1:1033420:1034620:24_ReprPC  \n",
       "47                     chr1:1033420:1034620:24_ReprPC  \n",
       "48  chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...  \n",
       "49                        chr1:1041020:1052220:8_TxWk  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "4687dbce-a028-4937-a1f0-82ef46fe4945",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_uniq',sep='\\t',header=False,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48e46e4c-4a02-4286-b18d-b9843cbce955",
   "metadata": {},
   "source": [
    "## promoter overlap with Hi-C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "c641047c-e227-441b-a240-966c2e6f4463",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bedtools intersect -a /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_uniq -b /mnt/10T-backup-mingju/ifar/Hi-C/MT/merge_results.txt_sorted_lexicographical_addchr -wao > /mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_HiC.bed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "37b3a092-3e46-41aa-97ff-18431186071b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = [\"Chrom\",\"start\",\"end\",\"Transcript\",\"TSS\",\"Gene\",\"strand\",\"transcript_support_level\",'openChromatin_MT','chromHMM_MT','chr_hic','start_hic','end_hic','HiC_MT','overlap']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "d89d3ca3-a38f-41f4-9b86-844ee9a210e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open_chromHMM = pd.read_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_HiC.bed',\n",
    "                                 sep='\\t',\n",
    "                                 header=None,\n",
    "                                 names=cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "162b7fbc-df46-4875-991c-10a094cd253b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>chr_hic</th>\n",
       "      <th>start_hic</th>\n",
       "      <th>end_hic</th>\n",
       "      <th>HiC_MT</th>\n",
       "      <th>overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>922000</td>\n",
       "      <td>924000</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>chr1</td>\n",
       "      <td>922000</td>\n",
       "      <td>924000</td>\n",
       "      <td>P:1:922000:924000;D:1:900000:902000;GOTHIC:rea...</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172726</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172727</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172728</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172729</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5172730</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5172731 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0        chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1        chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2        chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3        chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4        chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "...       ...       ...       ...              ...       ...     ...    ...   \n",
       "5172726  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "5172727  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "5172728  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "5172729  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "5172730  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "         transcript_support_level             openChromatin_MT  \\\n",
       "0                             NaN                            .   \n",
       "1                             NaN                            .   \n",
       "2                             NaN                            .   \n",
       "3                             5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                             5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                           ...                          ...   \n",
       "5172726                       5.0                            .   \n",
       "5172727                       1.0                            .   \n",
       "5172728                       2.0                            .   \n",
       "5172729                       1.0                            .   \n",
       "5172730                       2.0                            .   \n",
       "\n",
       "                                               chromHMM_MT chr_hic  start_hic  \\\n",
       "0                                chr1:10600:96080:25_Quies       .         -1   \n",
       "1                              chr1:348136:501536:25_Quies       .         -1   \n",
       "2                              chr1:634820:733820:25_Quies       .         -1   \n",
       "3        chr1:923420:923820:2_PromU;chr1:923820:924620:...    chr1     922000   \n",
       "4        chr1:923420:923820:2_PromU;chr1:923820:924620:...    chr1     922000   \n",
       "...                                                    ...     ...        ...   \n",
       "5172726                                                  .       .         -1   \n",
       "5172727                                                  .       .         -1   \n",
       "5172728                                                  .       .         -1   \n",
       "5172729                                                  .       .         -1   \n",
       "5172730                                                  .       .         -1   \n",
       "\n",
       "         end_hic                                             HiC_MT  overlap  \n",
       "0             -1                                                  .        0  \n",
       "1             -1                                                  .        0  \n",
       "2             -1                                                  .        0  \n",
       "3         924000  P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...      327  \n",
       "4         924000  P:1:922000:924000;D:1:900000:902000;GOTHIC:rea...      327  \n",
       "...          ...                                                ...      ...  \n",
       "5172726       -1                                                  .        0  \n",
       "5172727       -1                                                  .        0  \n",
       "5172728       -1                                                  .        0  \n",
       "5172729       -1                                                  .        0  \n",
       "5172730       -1                                                  .        0  \n",
       "\n",
       "[5172731 rows x 15 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "promoter_like_open_chromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "cf2857fe-7286-4625-9f30-b1b9788f701e",
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_like_open_chromHMM.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "956e233f-312f-4751-91bb-fafbd06a580b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5172731"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(promoter_like_open_chromHMM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "b99edd05-1aba-4cc4-9b51-361b6ffa3dcd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Chrom', 'start', 'end', 'Transcript', 'TSS', 'Gene', 'strand',\n",
      "       'transcript_support_level', 'openChromatin_MT', 'chromHMM_MT'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# Define the columns to group by (Chrom to chromHMM_myotube)\n",
    "group_cols = promoter_like_open_chromHMM.columns[:10]  # Adjust if needed\n",
    "print(group_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "8d8e279f-5444-4bfb-9eb0-89011bf8a0fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by Chrom to Type, and aggregate openChromatin by joining with \";\"\n",
    "collapsed_df = promoter_like_open_chromHMM.groupby(list(group_cols), dropna=False)['HiC_MT'] \\\n",
    "    .agg(lambda x: ';'.join(sorted(set(x.dropna())))) \\\n",
    "    .reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "e1617945-fc63-47a7-887b-dafce9f38cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170552</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57208269</td>\n",
       "      <td>57208769</td>\n",
       "      <td>ENST00000711281</td>\n",
       "      <td>57208519</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170553</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209056</td>\n",
       "      <td>57209556</td>\n",
       "      <td>ENST00000711282</td>\n",
       "      <td>57209306</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170554</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170555</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170556</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170557 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170552  chrY  57208269  57208769  ENST00000711281  57208519  WASH6P      +   \n",
       "170553  chrY  57209056  57209556  ENST00000711282  57209306  WASH6P      +   \n",
       "170554  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170555  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170556  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level             openChromatin_MT  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170552                       5.0                            .   \n",
       "170553                       1.0                            .   \n",
       "170554                       2.0                            .   \n",
       "170555                       1.0                            .   \n",
       "170556                       2.0                            .   \n",
       "\n",
       "                                              chromHMM_MT  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170552                                                  .   \n",
       "170553                                                  .   \n",
       "170554                                                  .   \n",
       "170555                                                  .   \n",
       "170556                                                  .   \n",
       "\n",
       "                                                   HiC_MT  \n",
       "0                                                       .  \n",
       "1                                                       .  \n",
       "2                                                       .  \n",
       "3       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "4       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "...                                                   ...  \n",
       "170552                                                  .  \n",
       "170553                                                  .  \n",
       "170554                                                  .  \n",
       "170555                                                  .  \n",
       "170556                                                  .  \n",
       "\n",
       "[170557 rows x 11 columns]"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "4123bfc4-2464-4621-a25a-f9bfc4e5eab0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:928020:933420:25_Quies</td>\n",
       "      <td>P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938220:939220:17_EnhW2;chr1:939220:939420...</td>\n",
       "      <td>P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:939820:940220:23_PromBiv;chr1:940220:9404...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:940420:941020:2_PromU;chr1:941020:941420:...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:957820:958620:4_PromD2;chr1:958620:959220...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961220:...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960420:961220:1_TssA;chr1:961220:961420:2...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:962820:7_Tx3';chr1:962820:965820:8...</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962820:965820:8_TxWk</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:969220:972420:24_ReprPC</td>\n",
       "      <td>P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972420:975820:25_Quies</td>\n",
       "      <td>P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>chr1:976020:976620:16_EnhW1;chr1:976620:980820...</td>\n",
       "      <td>P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:1000420:10020...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10142...</td>\n",
       "      <td>P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "      <td>P:1:1034000:1036000;D:1:1016000:1018000;GOTHIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3.0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1041020:1052220:8_TxWk</td>\n",
       "      <td>P:1:1044000:1046000;D:1:1018000:1020000;GOTHIC...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                        NaN   \n",
       "1                        NaN   \n",
       "2                        NaN   \n",
       "3                        5.0   \n",
       "4                        5.0   \n",
       "5                        5.0   \n",
       "6                        5.0   \n",
       "7                        5.0   \n",
       "8                        5.0   \n",
       "9                        5.0   \n",
       "10                       5.0   \n",
       "11                       5.0   \n",
       "12                       5.0   \n",
       "13                       2.0   \n",
       "14                       5.0   \n",
       "15                       5.0   \n",
       "16                       3.0   \n",
       "17                       2.0   \n",
       "18                       5.0   \n",
       "19                       2.0   \n",
       "20                       5.0   \n",
       "21                       1.0   \n",
       "22                       2.0   \n",
       "23                       3.0   \n",
       "24                       1.0   \n",
       "25                       1.0   \n",
       "26                       3.0   \n",
       "27                       2.0   \n",
       "28                       1.0   \n",
       "29                       1.0   \n",
       "30                       2.0   \n",
       "31                       3.0   \n",
       "32                       3.0   \n",
       "33                       1.0   \n",
       "34                       2.0   \n",
       "35                       5.0   \n",
       "36                       NaN   \n",
       "37                       3.0   \n",
       "38                       2.0   \n",
       "39                       1.0   \n",
       "40                       2.0   \n",
       "41                       3.0   \n",
       "42                       3.0   \n",
       "43                       NaN   \n",
       "44                       1.0   \n",
       "45                       5.0   \n",
       "46                       NaN   \n",
       "47                       NaN   \n",
       "48                       3.0   \n",
       "49                       3.0   \n",
       "\n",
       "                                     openChromatin_MT  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3                         ATAC-seq:chr1:923647-924887   \n",
       "4                         ATAC-seq:chr1:923647-924887   \n",
       "5                                                   .   \n",
       "6                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14                        ATAC-seq:chr1:940001-941223   \n",
       "15                        ATAC-seq:chr1:940001-941223   \n",
       "16                                                  .   \n",
       "17                                                  .   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                                                  .   \n",
       "21  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "24                        ATAC-seq:chr1:960473-961422   \n",
       "25                        ATAC-seq:chr1:960473-961422   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28                        ATAC-seq:chr1:966276-967323   \n",
       "29                        ATAC-seq:chr1:966276-967323   \n",
       "30                        ATAC-seq:chr1:966276-967323   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                        ATAC-seq:chr1:976006-976687   \n",
       "34                                                  .   \n",
       "35                                                  .   \n",
       "36                                                  .   \n",
       "37  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "41                       ATAC-seq:chr1:999640-1000919   \n",
       "42                       ATAC-seq:chr1:999640-1000919   \n",
       "43                      ATAC-seq:chr1:1012923-1014210   \n",
       "44                      ATAC-seq:chr1:1019149-1020697   \n",
       "45                      ATAC-seq:chr1:1019149-1020697   \n",
       "46                                                  .   \n",
       "47                                                  .   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                          chromHMM_MT  \\\n",
       "0                           chr1:10600:96080:25_Quies   \n",
       "1                         chr1:348136:501536:25_Quies   \n",
       "2                         chr1:634820:733820:25_Quies   \n",
       "3   chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4   chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...   \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "7                       chr1:925620:926820:23_PromBiv   \n",
       "8                       chr1:925620:926820:23_PromBiv   \n",
       "9                       chr1:925620:926820:23_PromBiv   \n",
       "10                      chr1:925620:926820:23_PromBiv   \n",
       "11                      chr1:925620:926820:23_PromBiv   \n",
       "12                        chr1:928020:933420:25_Quies   \n",
       "13  chr1:938220:939220:17_EnhW2;chr1:939220:939420...   \n",
       "14  chr1:939820:940220:23_PromBiv;chr1:940220:9404...   \n",
       "15  chr1:940420:941020:2_PromU;chr1:941020:941420:...   \n",
       "16                        chr1:941620:942620:22_PromP   \n",
       "17                        chr1:941620:942620:22_PromP   \n",
       "18                          chr1:945220:950020:7_Tx3'   \n",
       "19                          chr1:945220:950020:7_Tx3'   \n",
       "20  chr1:957820:958620:4_PromD2;chr1:958620:959220...   \n",
       "21  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "22  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "23  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961220:...   \n",
       "25  chr1:960420:961220:1_TssA;chr1:961220:961420:2...   \n",
       "26  chr1:962420:962820:7_Tx3';chr1:962820:965820:8...   \n",
       "27                          chr1:962820:965820:8_TxWk   \n",
       "28                         chr1:966220:967220:2_PromU   \n",
       "29                         chr1:966220:967220:2_PromU   \n",
       "30                         chr1:966220:967220:2_PromU   \n",
       "31                       chr1:969220:972420:24_ReprPC   \n",
       "32                        chr1:972420:975820:25_Quies   \n",
       "33  chr1:976020:976620:16_EnhW1;chr1:976620:980820...   \n",
       "34                       chr1:980820:984220:24_ReprPC   \n",
       "35                       chr1:980820:984220:24_ReprPC   \n",
       "36                       chr1:980820:984220:24_ReprPC   \n",
       "37  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "38  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "39  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "40  chr1:1000020:1000420:1_TssA;chr1:1000420:10020...   \n",
       "41                       chr1:1000420:1002020:2_PromU   \n",
       "42                       chr1:1000420:1002020:2_PromU   \n",
       "43  chr1:1013220:1013420:1_TssA;chr1:1013420:10142...   \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "46                     chr1:1033420:1034620:24_ReprPC   \n",
       "47                     chr1:1033420:1034620:24_ReprPC   \n",
       "48  chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...   \n",
       "49                        chr1:1041020:1052220:8_TxWk   \n",
       "\n",
       "                                               HiC_MT  \n",
       "0                                                   .  \n",
       "1                                                   .  \n",
       "2                                                   .  \n",
       "3   P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "4   P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...  \n",
       "5   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "6   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "7   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "8   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "9   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "10  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "11  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...  \n",
       "12  P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...  \n",
       "13  P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...  \n",
       "14  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "15  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "16  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "17  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...  \n",
       "18  P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...  \n",
       "19  P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...  \n",
       "20  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "21  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "22  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "23  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...  \n",
       "24  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...  \n",
       "25  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...  \n",
       "26  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...  \n",
       "27  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...  \n",
       "28  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "29  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "30  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...  \n",
       "31  P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...  \n",
       "32  P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...  \n",
       "33  P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...  \n",
       "34  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "35  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "36  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...  \n",
       "37  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "38  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "39  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "40  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "41  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "42  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...  \n",
       "43  P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...  \n",
       "44  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "45  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...  \n",
       "46  P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...  \n",
       "47  P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...  \n",
       "48  P:1:1034000:1036000;D:1:1016000:1018000;GOTHIC...  \n",
       "49  P:1:1044000:1046000;D:1:1018000:1020000;GOTHIC...  "
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19c6cd9d-9fde-4449-aeec-712b0c9beb97",
   "metadata": {},
   "source": [
    "# add TPM from RNA-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "4b56c1b9-d383-402b-ba60-4082c5f9ac67",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/MT/nf_results/star_rsem/rsem.merged.gene_tpm.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "e770c00f-12d2-433a-b1fa-af70fabf4e05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>Myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>2.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>8.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>1.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>0.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78927</th>\n",
       "      <td>ENSG00000310553</td>\n",
       "      <td>ENST00000850732</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78928</th>\n",
       "      <td>ENSG00000310554</td>\n",
       "      <td>ENST00000850733</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78929</th>\n",
       "      <td>ENSG00000310555</td>\n",
       "      <td>ENST00000850734</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78930</th>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>ENST00000850735,ENST00000850736</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78931</th>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>ENST00000850818,ENST00000850819,ENST0000085082...</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78932 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)  \\\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...   \n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971   \n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...   \n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...   \n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...   \n",
       "...                ...                                                ...   \n",
       "78927  ENSG00000310553                                    ENST00000850732   \n",
       "78928  ENSG00000310554                                    ENST00000850733   \n",
       "78929  ENSG00000310555                                    ENST00000850734   \n",
       "78930  ENSG00000310556                    ENST00000850735,ENST00000850736   \n",
       "78931  ENSG00000310557  ENST00000850818,ENST00000850819,ENST0000085082...   \n",
       "\n",
       "       Myotube  \n",
       "0         2.48  \n",
       "1         0.00  \n",
       "2         8.80  \n",
       "3         1.23  \n",
       "4         0.66  \n",
       "...        ...  \n",
       "78927     0.00  \n",
       "78928     0.00  \n",
       "78929     0.00  \n",
       "78930     0.00  \n",
       "78931     0.00  \n",
       "\n",
       "[78932 rows x 3 columns]"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "edf0f486-52bc-48bc-9f8b-bd52ca6e86bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_map =pd.read_csv('./geneid2name.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "003aef71-d14c-46aa-8310-242a4e3830bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm.merge(\n",
    "    gene_map,\n",
    "    on=\"gene_id\",\n",
    "    how=\"left\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "bc3a875c-48e4-41b6-b1e2-03d408273c1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>Myotube</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>2.48</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>8.80</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>1.23</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>0.66</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78927</th>\n",
       "      <td>ENSG00000310553</td>\n",
       "      <td>ENST00000850732</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78928</th>\n",
       "      <td>ENSG00000310554</td>\n",
       "      <td>ENST00000850733</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78929</th>\n",
       "      <td>ENSG00000310555</td>\n",
       "      <td>ENST00000850734</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78930</th>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>ENST00000850735,ENST00000850736</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78931</th>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>ENST00000850818,ENST00000850819,ENST0000085082...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78932 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)  \\\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...   \n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971   \n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...   \n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...   \n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...   \n",
       "...                ...                                                ...   \n",
       "78927  ENSG00000310553                                    ENST00000850732   \n",
       "78928  ENSG00000310554                                    ENST00000850733   \n",
       "78929  ENSG00000310555                                    ENST00000850734   \n",
       "78930  ENSG00000310556                    ENST00000850735,ENST00000850736   \n",
       "78931  ENSG00000310557  ENST00000850818,ENST00000850819,ENST0000085082...   \n",
       "\n",
       "       Myotube gene_name  \n",
       "0         2.48    TSPAN6  \n",
       "1         0.00      TNMD  \n",
       "2         8.80      DPM1  \n",
       "3         1.23     SCYL3  \n",
       "4         0.66     FIRRM  \n",
       "...        ...       ...  \n",
       "78927     0.00       NaN  \n",
       "78928     0.00       NaN  \n",
       "78929     0.00       NaN  \n",
       "78930     0.00       NaN  \n",
       "78931     0.00       NaN  \n",
       "\n",
       "[78932 rows x 4 columns]"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "70745fa0-dd06-4c13-bc4b-74f60e35c7ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[gene_tpm[\"gene_name\"].notna()].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "6dd4c7a8-2336-4319-a66d-a1beca807a4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>Myotube</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>2.48</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>8.80</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>1.23</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>0.66</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19469</th>\n",
       "      <td>ENSG00000292373</td>\n",
       "      <td>ENST00000711286,ENST00000711288,ENST0000071128...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19470</th>\n",
       "      <td>ENSG00000293137</td>\n",
       "      <td>ENST00000438684</td>\n",
       "      <td>0.70</td>\n",
       "      <td>MSL3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19471</th>\n",
       "      <td>ENSG00000293542</td>\n",
       "      <td>ENST00000473072,ENST00000713561</td>\n",
       "      <td>1.68</td>\n",
       "      <td>DUSP13B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19472</th>\n",
       "      <td>ENSG00000293543</td>\n",
       "      <td>ENST00000308475,ENST00000372702,ENST0000047988...</td>\n",
       "      <td>4.43</td>\n",
       "      <td>DUSP13A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19473</th>\n",
       "      <td>ENSG00000310517</td>\n",
       "      <td>ENST00000309190,ENST00000325674,ENST0000033825...</td>\n",
       "      <td>10.94</td>\n",
       "      <td>CAST</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19474 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)  \\\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...   \n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971   \n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...   \n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...   \n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...   \n",
       "...                ...                                                ...   \n",
       "19469  ENSG00000292373  ENST00000711286,ENST00000711288,ENST0000071128...   \n",
       "19470  ENSG00000293137                                    ENST00000438684   \n",
       "19471  ENSG00000293542                    ENST00000473072,ENST00000713561   \n",
       "19472  ENSG00000293543  ENST00000308475,ENST00000372702,ENST0000047988...   \n",
       "19473  ENSG00000310517  ENST00000309190,ENST00000325674,ENST0000033825...   \n",
       "\n",
       "       Myotube gene_name  \n",
       "0         2.48    TSPAN6  \n",
       "1         0.00      TNMD  \n",
       "2         8.80      DPM1  \n",
       "3         1.23     SCYL3  \n",
       "4         0.66     FIRRM  \n",
       "...        ...       ...  \n",
       "19469     0.00      IL9R  \n",
       "19470     0.70     MSL3B  \n",
       "19471     1.68   DUSP13B  \n",
       "19472     4.43   DUSP13A  \n",
       "19473    10.94      CAST  \n",
       "\n",
       "[19474 rows x 4 columns]"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "036b3871-80db-4c4a-bde8-1a931eea7005",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm.rename(columns={'Myotube':\"GeneTPM_MT\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "bd038c81-8eb5-4dcc-bbe2-fa7062f5c351",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id(s)</th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>ENST00000373020,ENST00000494424,ENST0000049677...</td>\n",
       "      <td>2.48</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>ENST00000373031,ENST00000485971</td>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>ENST00000371582,ENST00000371584,ENST0000037158...</td>\n",
       "      <td>8.80</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>ENST00000367770,ENST00000367771,ENST0000036777...</td>\n",
       "      <td>1.23</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>ENST00000286031,ENST00000359326,ENST0000041381...</td>\n",
       "      <td>0.66</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19469</th>\n",
       "      <td>ENSG00000292373</td>\n",
       "      <td>ENST00000711286,ENST00000711288,ENST0000071128...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19470</th>\n",
       "      <td>ENSG00000293137</td>\n",
       "      <td>ENST00000438684</td>\n",
       "      <td>0.70</td>\n",
       "      <td>MSL3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19471</th>\n",
       "      <td>ENSG00000293542</td>\n",
       "      <td>ENST00000473072,ENST00000713561</td>\n",
       "      <td>1.68</td>\n",
       "      <td>DUSP13B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19472</th>\n",
       "      <td>ENSG00000293543</td>\n",
       "      <td>ENST00000308475,ENST00000372702,ENST0000047988...</td>\n",
       "      <td>4.43</td>\n",
       "      <td>DUSP13A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19473</th>\n",
       "      <td>ENSG00000310517</td>\n",
       "      <td>ENST00000309190,ENST00000325674,ENST0000033825...</td>\n",
       "      <td>10.94</td>\n",
       "      <td>CAST</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19474 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               gene_id                                   transcript_id(s)  \\\n",
       "0      ENSG00000000003  ENST00000373020,ENST00000494424,ENST0000049677...   \n",
       "1      ENSG00000000005                    ENST00000373031,ENST00000485971   \n",
       "2      ENSG00000000419  ENST00000371582,ENST00000371584,ENST0000037158...   \n",
       "3      ENSG00000000457  ENST00000367770,ENST00000367771,ENST0000036777...   \n",
       "4      ENSG00000000460  ENST00000286031,ENST00000359326,ENST0000041381...   \n",
       "...                ...                                                ...   \n",
       "19469  ENSG00000292373  ENST00000711286,ENST00000711288,ENST0000071128...   \n",
       "19470  ENSG00000293137                                    ENST00000438684   \n",
       "19471  ENSG00000293542                    ENST00000473072,ENST00000713561   \n",
       "19472  ENSG00000293543  ENST00000308475,ENST00000372702,ENST0000047988...   \n",
       "19473  ENSG00000310517  ENST00000309190,ENST00000325674,ENST0000033825...   \n",
       "\n",
       "       GeneTPM_MT gene_name  \n",
       "0            2.48    TSPAN6  \n",
       "1            0.00      TNMD  \n",
       "2            8.80      DPM1  \n",
       "3            1.23     SCYL3  \n",
       "4            0.66     FIRRM  \n",
       "...           ...       ...  \n",
       "19469        0.00      IL9R  \n",
       "19470        0.70     MSL3B  \n",
       "19471        1.68   DUSP13B  \n",
       "19472        4.43   DUSP13A  \n",
       "19473       10.94      CAST  \n",
       "\n",
       "[19474 rows x 4 columns]"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "c3f21298-a993-4fdc-831b-f5ef3ad4fec5",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_tpm = gene_tpm[['GeneTPM_MT','gene_name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "bd97b2bd-b912-4d05-8fb9-dbfdc5c3bbbf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2.48</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.00</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8.80</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.23</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.66</td>\n",
       "      <td>FIRRM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19469</th>\n",
       "      <td>0.00</td>\n",
       "      <td>IL9R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19470</th>\n",
       "      <td>0.70</td>\n",
       "      <td>MSL3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19471</th>\n",
       "      <td>1.68</td>\n",
       "      <td>DUSP13B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19472</th>\n",
       "      <td>4.43</td>\n",
       "      <td>DUSP13A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19473</th>\n",
       "      <td>10.94</td>\n",
       "      <td>CAST</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19474 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GeneTPM_MT gene_name\n",
       "0            2.48    TSPAN6\n",
       "1            0.00      TNMD\n",
       "2            8.80      DPM1\n",
       "3            1.23     SCYL3\n",
       "4            0.66     FIRRM\n",
       "...           ...       ...\n",
       "19469        0.00      IL9R\n",
       "19470        0.70     MSL3B\n",
       "19471        1.68   DUSP13B\n",
       "19472        4.43   DUSP13A\n",
       "19473       10.94      CAST\n",
       "\n",
       "[19474 rows x 2 columns]"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "8624f0b0-90c6-45ef-b83b-03b2943b80a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df = collapsed_df.merge(\n",
    "    gene_tpm,\n",
    "    left_on=\"Gene\",\n",
    "    right_on=\"gene_name\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['gene_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "43d51dac-9f53-4c7a-bb56-9b4b325a8eee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level             openChromatin_MT  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170934                       2.0                            .   \n",
       "170935                       1.0                            .   \n",
       "170936                       1.0                            .   \n",
       "170937                       2.0                            .   \n",
       "170938                       2.0                            .   \n",
       "\n",
       "                                              chromHMM_MT  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                                   HiC_MT  GeneTPM_MT  \n",
       "0                                                       .        0.00  \n",
       "1                                                       .        0.00  \n",
       "2                                                       .        0.00  \n",
       "3       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.34  \n",
       "4       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.34  \n",
       "...                                                   ...         ...  \n",
       "170934                                                  .        1.43  \n",
       "170935                                                  .        1.43  \n",
       "170936                                                  .        1.43  \n",
       "170937                                                  .        1.43  \n",
       "170938                                                  .        1.43  \n",
       "\n",
       "[170939 rows x 12 columns]"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "56c41029-6283-432c-9f4d-e2b717e44d44",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneTPM_MT\n",
       "False    170939\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df['GeneTPM_MT'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "1b3f4836-dcb2-4c7e-9da7-f0409ca0da92",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = pd.read_csv('/mnt/10T-backup-mingju/ifar/RNA_seq/MT/nf_results/star_rsem/rsem.merged.transcript_tpm.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "c0075f08-1f08-4cb6-8c5d-5e811ae8818e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000373020</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>1.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000494424</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000496771</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000612152</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000373031</td>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387939</th>\n",
       "      <td>ENST00000850736</td>\n",
       "      <td>ENSG00000310556</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387940</th>\n",
       "      <td>ENST00000850818</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387941</th>\n",
       "      <td>ENST00000850819</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387942</th>\n",
       "      <td>ENST00000850820</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387943</th>\n",
       "      <td>ENST00000850821</td>\n",
       "      <td>ENSG00000310557</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>387944 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          transcript_id          gene_id  Myotube\n",
       "0       ENST00000373020  ENSG00000000003     1.76\n",
       "1       ENST00000494424  ENSG00000000003     0.10\n",
       "2       ENST00000496771  ENSG00000000003     0.62\n",
       "3       ENST00000612152  ENSG00000000003     0.00\n",
       "4       ENST00000373031  ENSG00000000005     0.00\n",
       "...                 ...              ...      ...\n",
       "387939  ENST00000850736  ENSG00000310556     0.00\n",
       "387940  ENST00000850818  ENSG00000310557     0.00\n",
       "387941  ENST00000850819  ENSG00000310557     0.00\n",
       "387942  ENST00000850820  ENSG00000310557     0.00\n",
       "387943  ENST00000850821  ENSG00000310557     0.00\n",
       "\n",
       "[387944 rows x 3 columns]"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tx_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "9db23975-bef9-4db4-be4f-8b5746b69c76",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = tx_tpm[['transcript_id','Myotube']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "6f8308a9-d993-43fc-aa52-9d8821e6606d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>Myotube</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000373020</td>\n",
       "      <td>1.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000494424</td>\n",
       "      <td>0.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000496771</td>\n",
       "      <td>0.62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000612152</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000373031</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387939</th>\n",
       "      <td>ENST00000850736</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387940</th>\n",
       "      <td>ENST00000850818</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387941</th>\n",
       "      <td>ENST00000850819</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387942</th>\n",
       "      <td>ENST00000850820</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>387943</th>\n",
       "      <td>ENST00000850821</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>387944 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          transcript_id  Myotube\n",
       "0       ENST00000373020     1.76\n",
       "1       ENST00000494424     0.10\n",
       "2       ENST00000496771     0.62\n",
       "3       ENST00000612152     0.00\n",
       "4       ENST00000373031     0.00\n",
       "...                 ...      ...\n",
       "387939  ENST00000850736     0.00\n",
       "387940  ENST00000850818     0.00\n",
       "387941  ENST00000850819     0.00\n",
       "387942  ENST00000850820     0.00\n",
       "387943  ENST00000850821     0.00\n",
       "\n",
       "[387944 rows x 2 columns]"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tx_tpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "6e27d4f1-a2fd-4e08-bc8c-8595e4b1a1f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "tx_tpm = tx_tpm.rename(columns={'Myotube':\"TranscriptTPM_MT\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "48d8aa62-d137-4876-aa0d-52e334b36cfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df['Transcript'] = collapsed_df['Transcript'].str.split(\".\").str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "a580f019-333b-449d-9307-37d4248fb171",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df = collapsed_df.merge(\n",
    "    tx_tpm,\n",
    "    left_on=\"Transcript\",\n",
    "    right_on=\"transcript_id\",\n",
    "    how=\"left\"\n",
    ").drop(columns=['transcript_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "ce2c97df-8c1a-4b23-960f-91209a2c078d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "      <th>TranscriptTPM_MT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170934</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57209637</td>\n",
       "      <td>57210137</td>\n",
       "      <td>ENST00000711283</td>\n",
       "      <td>57209887</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170935</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170936</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210094</td>\n",
       "      <td>57210594</td>\n",
       "      <td>ENST00000711284</td>\n",
       "      <td>57210344</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170937</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170938</th>\n",
       "      <td>chrY</td>\n",
       "      <td>57210341</td>\n",
       "      <td>57210841</td>\n",
       "      <td>ENST00000711285</td>\n",
       "      <td>57210591</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>+</td>\n",
       "      <td>2.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.43</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>170939 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chrom     start       end       Transcript       TSS    Gene strand  \\\n",
       "0       chr1     65169     65669  ENST00000641515     65419   OR4F5      +   \n",
       "1       chr1    451428    451928  ENST00000426406    451678  OR4F29      -   \n",
       "2       chr1    686404    686904  ENST00000332831    686654  OR4F16      -   \n",
       "3       chr1    923673    924173  ENST00000616016    923923  SAMD11      +   \n",
       "4       chr1    923673    924173  ENST00000618323    923923  SAMD11      +   \n",
       "...      ...       ...       ...              ...       ...     ...    ...   \n",
       "170934  chrY  57209637  57210137  ENST00000711283  57209887  WASH6P      +   \n",
       "170935  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170936  chrY  57210094  57210594  ENST00000711284  57210344  WASH6P      +   \n",
       "170937  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "170938  chrY  57210341  57210841  ENST00000711285  57210591  WASH6P      +   \n",
       "\n",
       "        transcript_support_level             openChromatin_MT  \\\n",
       "0                            NaN                            .   \n",
       "1                            NaN                            .   \n",
       "2                            NaN                            .   \n",
       "3                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "4                            5.0  ATAC-seq:chr1:923647-924887   \n",
       "...                          ...                          ...   \n",
       "170934                       2.0                            .   \n",
       "170935                       1.0                            .   \n",
       "170936                       1.0                            .   \n",
       "170937                       2.0                            .   \n",
       "170938                       2.0                            .   \n",
       "\n",
       "                                              chromHMM_MT  \\\n",
       "0                               chr1:10600:96080:25_Quies   \n",
       "1                             chr1:348136:501536:25_Quies   \n",
       "2                             chr1:634820:733820:25_Quies   \n",
       "3       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4       chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "...                                                   ...   \n",
       "170934                                                  .   \n",
       "170935                                                  .   \n",
       "170936                                                  .   \n",
       "170937                                                  .   \n",
       "170938                                                  .   \n",
       "\n",
       "                                                   HiC_MT  GeneTPM_MT  \\\n",
       "0                                                       .        0.00   \n",
       "1                                                       .        0.00   \n",
       "2                                                       .        0.00   \n",
       "3       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.34   \n",
       "4       P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.34   \n",
       "...                                                   ...         ...   \n",
       "170934                                                  .        1.43   \n",
       "170935                                                  .        1.43   \n",
       "170936                                                  .        1.43   \n",
       "170937                                                  .        1.43   \n",
       "170938                                                  .        1.43   \n",
       "\n",
       "        TranscriptTPM_MT  \n",
       "0                   0.00  \n",
       "1                   0.00  \n",
       "2                   0.00  \n",
       "3                   0.00  \n",
       "4                   0.06  \n",
       "...                  ...  \n",
       "170934              0.00  \n",
       "170935              0.00  \n",
       "170936              0.00  \n",
       "170937              0.00  \n",
       "170938              0.00  \n",
       "\n",
       "[170939 rows x 13 columns]"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "2650e004-85c0-41d5-b584-350c3a41e6c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TranscriptTPM_MT\n",
       "False    170939\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df['TranscriptTPM_MT'].isna().value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f80d806-da69-4112-a3c3-48b3194fa03e",
   "metadata": {},
   "source": [
    "# update the promoter evidence Type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "6afe607b-cf35-42ce-8b07-9e5596b8f178",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df[\"transcript_support_level\"] = (\n",
    "    collapsed_df[\"transcript_support_level\"]\n",
    "        .fillna(0)\n",
    "        .astype(int)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "eb83069e-df7b-40c2-b1d1-029b1151cdc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def classify_promoter_type(row):\n",
    "    score = 0\n",
    "    sources = []  # track where score comes from\n",
    "\n",
    "    # Open chromatin evidence (Essential)\n",
    "    open_chrom = row['openChromatin_MT']\n",
    "    if isinstance(open_chrom, str) and open_chrom != \".\":\n",
    "        if 'ATAC' in open_chrom and 'DNase' in open_chrom:\n",
    "            score += 3\n",
    "            sources.append(\"OpenChromatin:ATAC+DNase(+3)\")\n",
    "        else:\n",
    "            score += 2\n",
    "            sources.append(\"OpenChromatin:ATAC/DNase(+2)\")\n",
    "\n",
    "    # ChromHMM evidence (Essential)\n",
    "    chromhmm = row['chromHMM_MT']\n",
    "    if isinstance(chromhmm, str) and chromhmm != \".\":\n",
    "        active_states = ['TssA', 'PromU', 'PromP', '1_Tss', '2_Prom']\n",
    "        if any(state in chromhmm for state in active_states):\n",
    "            score += 2\n",
    "            sources.append(\"ChromHMM:ActivePromoter(+2)\")\n",
    "\n",
    "    # Expression evidence\n",
    "    transcript_tpm = row.get('TranscriptTPM_MT', 0)\n",
    "    gene_tpm = row.get('GeneTPM_MT', 0)\n",
    "\n",
    "    if transcript_tpm > 1:\n",
    "        score += 2\n",
    "        sources.append(\"TranscriptTPM>1(+2)\")\n",
    "    elif transcript_tpm > 0:\n",
    "        score += 1\n",
    "        sources.append(\"TranscriptTPM>0(+1)\")\n",
    "    elif gene_tpm > 1:\n",
    "        score += 1\n",
    "        sources.append(\"GeneTPM>1(+1)\")\n",
    "\n",
    "    # Transcript support level\n",
    "    tsl = str(row['transcript_support_level'])\n",
    "    if tsl in ['1', '2']:\n",
    "        score += 1\n",
    "        sources.append(\"TSL1-2(+1)\")\n",
    "\n",
    "    # Optional Hi-C\n",
    "    # hic = row.get('HiC_hMSC')\n",
    "    # if isinstance(hic, str) and hic != '.':\n",
    "    #     score += 1\n",
    "    #     sources.append(\"HiC(+1)\")\n",
    "\n",
    "    # Final classification\n",
    "    if score >= 6:\n",
    "        promoter_type = 'very_high'\n",
    "    elif score >= 4:\n",
    "        promoter_type = 'high'\n",
    "    elif score >= 2:\n",
    "        promoter_type = 'moderate'\n",
    "    else:\n",
    "        promoter_type = 'low'\n",
    "\n",
    "    return promoter_type, score, \"; \".join(sources)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "cd3c7c94-00f2-4a12-b5ae-95ac70e78316",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df[[\"Type\", \"score\", \"score_source\"]] = (\n",
    "    collapsed_df\n",
    "        .apply(classify_promoter_type, axis=1, result_type=\"expand\")\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "70b02bca-4278-481d-83dd-8a830487b7ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>TSS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>openChromatin_MT</th>\n",
       "      <th>chromHMM_MT</th>\n",
       "      <th>HiC_MT</th>\n",
       "      <th>GeneTPM_MT</th>\n",
       "      <th>TranscriptTPM_MT</th>\n",
       "      <th>Type</th>\n",
       "      <th>score</th>\n",
       "      <th>score_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65169</td>\n",
       "      <td>65669</td>\n",
       "      <td>ENST00000641515</td>\n",
       "      <td>65419</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:10600:96080:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>451428</td>\n",
       "      <td>451928</td>\n",
       "      <td>ENST00000426406</td>\n",
       "      <td>451678</td>\n",
       "      <td>OR4F29</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:348136:501536:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>686404</td>\n",
       "      <td>686904</td>\n",
       "      <td>ENST00000332831</td>\n",
       "      <td>686654</td>\n",
       "      <td>OR4F16</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:634820:733820:25_Quies</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000616016</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>4</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923673</td>\n",
       "      <td>924173</td>\n",
       "      <td>ENST00000618323</td>\n",
       "      <td>923923</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>ATAC-seq:chr1:923647-924887</td>\n",
       "      <td>chr1:923420:923820:2_PromU;chr1:923820:924620:...</td>\n",
       "      <td>P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.06</td>\n",
       "      <td>high</td>\n",
       "      <td>5</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>924900</td>\n",
       "      <td>925400</td>\n",
       "      <td>ENST00000437963</td>\n",
       "      <td>925150</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:924620:925220:2_PromU;chr1:925220:925620:...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>ChromHMM:ActivePromoter(+2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925481</td>\n",
       "      <td>925981</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>925731</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925220:925620:22_PromP;chr1:925620:926820...</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>ChromHMM:ActivePromoter(+2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000616125</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000617307</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618181</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000618779</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>925692</td>\n",
       "      <td>926192</td>\n",
       "      <td>ENST00000622503</td>\n",
       "      <td>925942</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:925620:926820:23_PromBiv</td>\n",
       "      <td>P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>930062</td>\n",
       "      <td>930562</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>930312</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:928020:933420:25_Quies</td>\n",
       "      <td>P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.26</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>TranscriptTPM&gt;0(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>939025</td>\n",
       "      <td>939525</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>939275</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:938220:939220:17_EnhW2;chr1:939220:939420...</td>\n",
       "      <td>P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.01</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>TranscriptTPM&gt;0(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940096</td>\n",
       "      <td>940596</td>\n",
       "      <td>ENST00000478729</td>\n",
       "      <td>940346</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:939820:940220:23_PromBiv;chr1:940220:9404...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>4</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>940826</td>\n",
       "      <td>941326</td>\n",
       "      <td>ENST00000474461</td>\n",
       "      <td>941076</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>ATAC-seq:chr1:940001-941223</td>\n",
       "      <td>chr1:940420:941020:2_PromU;chr1:941020:941420:...</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>4</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941853</td>\n",
       "      <td>942353</td>\n",
       "      <td>ENST00000466827</td>\n",
       "      <td>942103</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>ChromHMM:ActivePromoter(+2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>941916</td>\n",
       "      <td>942416</td>\n",
       "      <td>ENST00000464948</td>\n",
       "      <td>942166</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:941620:942620:22_PromP</td>\n",
       "      <td>P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>moderate</td>\n",
       "      <td>3</td>\n",
       "      <td>ChromHMM:ActivePromoter(+2); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>945312</td>\n",
       "      <td>945812</td>\n",
       "      <td>ENST00000496938</td>\n",
       "      <td>945562</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...</td>\n",
       "      <td>11.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>GeneTPM&gt;1(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>946810</td>\n",
       "      <td>947310</td>\n",
       "      <td>ENST00000483767</td>\n",
       "      <td>947060</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:945220:950020:7_Tx3'</td>\n",
       "      <td>P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...</td>\n",
       "      <td>11.08</td>\n",
       "      <td>0.40</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>TranscriptTPM&gt;0(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>958208</td>\n",
       "      <td>958708</td>\n",
       "      <td>ENST00000477976</td>\n",
       "      <td>958458</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:957820:958620:4_PromD2;chr1:958620:959220...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "      <td>11.08</td>\n",
       "      <td>0.02</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>TranscriptTPM&gt;0(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "      <td>11.08</td>\n",
       "      <td>8.90</td>\n",
       "      <td>very_high</td>\n",
       "      <td>8</td>\n",
       "      <td>OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959006</td>\n",
       "      <td>959506</td>\n",
       "      <td>ENST00000469563</td>\n",
       "      <td>959256</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "      <td>11.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>very_high</td>\n",
       "      <td>7</td>\n",
       "      <td>OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>959059</td>\n",
       "      <td>959559</td>\n",
       "      <td>ENST00000487214</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...</td>\n",
       "      <td>chr1:958620:959220:3_PromD1;chr1:959220:959420...</td>\n",
       "      <td>P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...</td>\n",
       "      <td>11.08</td>\n",
       "      <td>1.76</td>\n",
       "      <td>very_high</td>\n",
       "      <td>7</td>\n",
       "      <td>OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>960334</td>\n",
       "      <td>960834</td>\n",
       "      <td>ENST00000338591</td>\n",
       "      <td>960584</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960220:960420:2_PromU;chr1:960420:961220:...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.61</td>\n",
       "      <td>very_high</td>\n",
       "      <td>6</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>961199</td>\n",
       "      <td>961699</td>\n",
       "      <td>ENST00000463212</td>\n",
       "      <td>961449</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:960473-961422</td>\n",
       "      <td>chr1:960420:961220:1_TssA;chr1:961220:961420:2...</td>\n",
       "      <td>P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.08</td>\n",
       "      <td>very_high</td>\n",
       "      <td>6</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>962477</td>\n",
       "      <td>962977</td>\n",
       "      <td>ENST00000466300</td>\n",
       "      <td>962727</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962420:962820:7_Tx3';chr1:962820:965820:8...</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr1</td>\n",
       "      <td>963302</td>\n",
       "      <td>963802</td>\n",
       "      <td>ENST00000481067</td>\n",
       "      <td>963552</td>\n",
       "      <td>KLHL17</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:962820:965820:8_TxWk</td>\n",
       "      <td>P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.05</td>\n",
       "      <td>moderate</td>\n",
       "      <td>2</td>\n",
       "      <td>TranscriptTPM&gt;0(+1); TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966232</td>\n",
       "      <td>966732</td>\n",
       "      <td>ENST00000379410</td>\n",
       "      <td>966482</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>5</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379407</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "      <td>very_high</td>\n",
       "      <td>6</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr1</td>\n",
       "      <td>966252</td>\n",
       "      <td>966752</td>\n",
       "      <td>ENST00000379409</td>\n",
       "      <td>966502</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>ATAC-seq:chr1:966276-967323</td>\n",
       "      <td>chr1:966220:967220:2_PromU</td>\n",
       "      <td>P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>5</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr1</td>\n",
       "      <td>970625</td>\n",
       "      <td>971125</td>\n",
       "      <td>ENST00000480267</td>\n",
       "      <td>970875</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:969220:972420:24_ReprPC</td>\n",
       "      <td>P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr1</td>\n",
       "      <td>973262</td>\n",
       "      <td>973762</td>\n",
       "      <td>ENST00000491024</td>\n",
       "      <td>973512</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:972420:975820:25_Quies</td>\n",
       "      <td>P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr1</td>\n",
       "      <td>976391</td>\n",
       "      <td>976891</td>\n",
       "      <td>ENST00000479361</td>\n",
       "      <td>976641</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:976006-976687</td>\n",
       "      <td>chr1:976020:976620:16_EnhW1;chr1:976620:980820...</td>\n",
       "      <td>P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>high</td>\n",
       "      <td>4</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); TranscriptTPM&gt;0(...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000341290</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>TSL1-2(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981843</td>\n",
       "      <td>982343</td>\n",
       "      <td>ENST00000433179</td>\n",
       "      <td>982093</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>981867</td>\n",
       "      <td>982367</td>\n",
       "      <td>ENST00000694917</td>\n",
       "      <td>982117</td>\n",
       "      <td>PERM1</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:980820:984220:24_ReprPC</td>\n",
       "      <td>P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999731</td>\n",
       "      <td>1000231</td>\n",
       "      <td>ENST00000484667</td>\n",
       "      <td>999981</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "      <td>3.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>very_high</td>\n",
       "      <td>6</td>\n",
       "      <td>OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999846</td>\n",
       "      <td>1000346</td>\n",
       "      <td>ENST00000481869</td>\n",
       "      <td>1000096</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "      <td>3.55</td>\n",
       "      <td>0.59</td>\n",
       "      <td>very_high</td>\n",
       "      <td>7</td>\n",
       "      <td>OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999847</td>\n",
       "      <td>1000347</td>\n",
       "      <td>ENST00000304952</td>\n",
       "      <td>1000097</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:999020:100002...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "      <td>3.55</td>\n",
       "      <td>2.01</td>\n",
       "      <td>very_high</td>\n",
       "      <td>8</td>\n",
       "      <td>OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1</td>\n",
       "      <td>999922</td>\n",
       "      <td>1000422</td>\n",
       "      <td>ENST00000428771</td>\n",
       "      <td>1000172</td>\n",
       "      <td>HES4</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...</td>\n",
       "      <td>chr1:1000020:1000420:1_TssA;chr1:1000420:10020...</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "      <td>3.55</td>\n",
       "      <td>0.94</td>\n",
       "      <td>very_high</td>\n",
       "      <td>7</td>\n",
       "      <td>OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000888</td>\n",
       "      <td>1001388</td>\n",
       "      <td>ENST00000624697</td>\n",
       "      <td>1001138</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "      <td>17.92</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>5</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1000895</td>\n",
       "      <td>1001395</td>\n",
       "      <td>ENST00000624652</td>\n",
       "      <td>1001145</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>ATAC-seq:chr1:999640-1000919</td>\n",
       "      <td>chr1:1000420:1002020:2_PromU</td>\n",
       "      <td>P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...</td>\n",
       "      <td>17.92</td>\n",
       "      <td>0.00</td>\n",
       "      <td>high</td>\n",
       "      <td>5</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1013247</td>\n",
       "      <td>1013747</td>\n",
       "      <td>ENST00000649529</td>\n",
       "      <td>1013497</td>\n",
       "      <td>ISG15</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ATAC-seq:chr1:1012923-1014210</td>\n",
       "      <td>chr1:1013220:1013420:1_TssA;chr1:1013420:10142...</td>\n",
       "      <td>P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "      <td>17.92</td>\n",
       "      <td>17.92</td>\n",
       "      <td>very_high</td>\n",
       "      <td>6</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019870</td>\n",
       "      <td>1020370</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>1020120</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "      <td>3.68</td>\n",
       "      <td>2.46</td>\n",
       "      <td>very_high</td>\n",
       "      <td>7</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1019873</td>\n",
       "      <td>1020373</td>\n",
       "      <td>ENST00000620552</td>\n",
       "      <td>1020123</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>5</td>\n",
       "      <td>ATAC-seq:chr1:1019149-1020697</td>\n",
       "      <td>chr1:1019820:1020020:22_PromP;chr1:1020020:102...</td>\n",
       "      <td>P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.85</td>\n",
       "      <td>high</td>\n",
       "      <td>5</td>\n",
       "      <td>OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000651234</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>GeneTPM&gt;1(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1033856</td>\n",
       "      <td>1034356</td>\n",
       "      <td>ENST00000652369</td>\n",
       "      <td>1034106</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC</td>\n",
       "      <td>P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>GeneTPM&gt;1(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1034617</td>\n",
       "      <td>1035117</td>\n",
       "      <td>ENST00000469403</td>\n",
       "      <td>1034867</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...</td>\n",
       "      <td>P:1:1034000:1036000;D:1:1016000:1018000;GOTHIC...</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>GeneTPM&gt;1(+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1045149</td>\n",
       "      <td>1045649</td>\n",
       "      <td>ENST00000479707</td>\n",
       "      <td>1045399</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>+</td>\n",
       "      <td>3</td>\n",
       "      <td>.</td>\n",
       "      <td>chr1:1041020:1052220:8_TxWk</td>\n",
       "      <td>P:1:1044000:1046000;D:1:1018000:1020000;GOTHIC...</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.00</td>\n",
       "      <td>low</td>\n",
       "      <td>1</td>\n",
       "      <td>GeneTPM&gt;1(+1)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom    start      end       Transcript      TSS     Gene strand  \\\n",
       "0   chr1    65169    65669  ENST00000641515    65419    OR4F5      +   \n",
       "1   chr1   451428   451928  ENST00000426406   451678   OR4F29      -   \n",
       "2   chr1   686404   686904  ENST00000332831   686654   OR4F16      -   \n",
       "3   chr1   923673   924173  ENST00000616016   923923   SAMD11      +   \n",
       "4   chr1   923673   924173  ENST00000618323   923923   SAMD11      +   \n",
       "5   chr1   924900   925400  ENST00000437963   925150   SAMD11      +   \n",
       "6   chr1   925481   925981  ENST00000342066   925731   SAMD11      +   \n",
       "7   chr1   925692   926192  ENST00000616125   925942   SAMD11      +   \n",
       "8   chr1   925692   926192  ENST00000617307   925942   SAMD11      +   \n",
       "9   chr1   925692   926192  ENST00000618181   925942   SAMD11      +   \n",
       "10  chr1   925692   926192  ENST00000618779   925942   SAMD11      +   \n",
       "11  chr1   925692   926192  ENST00000622503   925942   SAMD11      +   \n",
       "12  chr1   930062   930562  ENST00000341065   930312   SAMD11      +   \n",
       "13  chr1   939025   939525  ENST00000455979   939275   SAMD11      +   \n",
       "14  chr1   940096   940596  ENST00000478729   940346   SAMD11      +   \n",
       "15  chr1   940826   941326  ENST00000474461   941076   SAMD11      +   \n",
       "16  chr1   941853   942353  ENST00000466827   942103   SAMD11      +   \n",
       "17  chr1   941916   942416  ENST00000464948   942166   SAMD11      +   \n",
       "18  chr1   945312   945812  ENST00000496938   945562    NOC2L      -   \n",
       "19  chr1   946810   947310  ENST00000483767   947060    NOC2L      -   \n",
       "20  chr1   958208   958708  ENST00000477976   958458    NOC2L      -   \n",
       "21  chr1   959006   959506  ENST00000327044   959256    NOC2L      -   \n",
       "22  chr1   959006   959506  ENST00000469563   959256    NOC2L      -   \n",
       "23  chr1   959059   959559  ENST00000487214   959309    NOC2L      -   \n",
       "24  chr1   960334   960834  ENST00000338591   960584   KLHL17      +   \n",
       "25  chr1   961199   961699  ENST00000463212   961449   KLHL17      +   \n",
       "26  chr1   962477   962977  ENST00000466300   962727   KLHL17      +   \n",
       "27  chr1   963302   963802  ENST00000481067   963552   KLHL17      +   \n",
       "28  chr1   966232   966732  ENST00000379410   966482  PLEKHN1      +   \n",
       "29  chr1   966252   966752  ENST00000379407   966502  PLEKHN1      +   \n",
       "30  chr1   966252   966752  ENST00000379409   966502  PLEKHN1      +   \n",
       "31  chr1   970625   971125  ENST00000480267   970875  PLEKHN1      +   \n",
       "32  chr1   973262   973762  ENST00000491024   973512  PLEKHN1      +   \n",
       "33  chr1   976391   976891  ENST00000479361   976641    PERM1      -   \n",
       "34  chr1   981843   982343  ENST00000341290   982093    PERM1      -   \n",
       "35  chr1   981843   982343  ENST00000433179   982093    PERM1      -   \n",
       "36  chr1   981867   982367  ENST00000694917   982117    PERM1      -   \n",
       "37  chr1   999731  1000231  ENST00000484667   999981     HES4      -   \n",
       "38  chr1   999846  1000346  ENST00000481869  1000096     HES4      -   \n",
       "39  chr1   999847  1000347  ENST00000304952  1000097     HES4      -   \n",
       "40  chr1   999922  1000422  ENST00000428771  1000172     HES4      -   \n",
       "41  chr1  1000888  1001388  ENST00000624697  1001138    ISG15      +   \n",
       "42  chr1  1000895  1001395  ENST00000624652  1001145    ISG15      +   \n",
       "43  chr1  1013247  1013747  ENST00000649529  1013497    ISG15      +   \n",
       "44  chr1  1019870  1020370  ENST00000379370  1020120     AGRN      +   \n",
       "45  chr1  1019873  1020373  ENST00000620552  1020123     AGRN      +   \n",
       "46  chr1  1033856  1034356  ENST00000651234  1034106     AGRN      +   \n",
       "47  chr1  1033856  1034356  ENST00000652369  1034106     AGRN      +   \n",
       "48  chr1  1034617  1035117  ENST00000469403  1034867     AGRN      +   \n",
       "49  chr1  1045149  1045649  ENST00000479707  1045399     AGRN      +   \n",
       "\n",
       "    transcript_support_level  \\\n",
       "0                          0   \n",
       "1                          0   \n",
       "2                          0   \n",
       "3                          5   \n",
       "4                          5   \n",
       "5                          5   \n",
       "6                          5   \n",
       "7                          5   \n",
       "8                          5   \n",
       "9                          5   \n",
       "10                         5   \n",
       "11                         5   \n",
       "12                         5   \n",
       "13                         2   \n",
       "14                         5   \n",
       "15                         5   \n",
       "16                         3   \n",
       "17                         2   \n",
       "18                         5   \n",
       "19                         2   \n",
       "20                         5   \n",
       "21                         1   \n",
       "22                         2   \n",
       "23                         3   \n",
       "24                         1   \n",
       "25                         1   \n",
       "26                         3   \n",
       "27                         2   \n",
       "28                         1   \n",
       "29                         1   \n",
       "30                         2   \n",
       "31                         3   \n",
       "32                         3   \n",
       "33                         1   \n",
       "34                         2   \n",
       "35                         5   \n",
       "36                         0   \n",
       "37                         3   \n",
       "38                         2   \n",
       "39                         1   \n",
       "40                         2   \n",
       "41                         3   \n",
       "42                         3   \n",
       "43                         0   \n",
       "44                         1   \n",
       "45                         5   \n",
       "46                         0   \n",
       "47                         0   \n",
       "48                         3   \n",
       "49                         3   \n",
       "\n",
       "                                     openChromatin_MT  \\\n",
       "0                                                   .   \n",
       "1                                                   .   \n",
       "2                                                   .   \n",
       "3                         ATAC-seq:chr1:923647-924887   \n",
       "4                         ATAC-seq:chr1:923647-924887   \n",
       "5                                                   .   \n",
       "6                                                   .   \n",
       "7                                                   .   \n",
       "8                                                   .   \n",
       "9                                                   .   \n",
       "10                                                  .   \n",
       "11                                                  .   \n",
       "12                                                  .   \n",
       "13                                                  .   \n",
       "14                        ATAC-seq:chr1:940001-941223   \n",
       "15                        ATAC-seq:chr1:940001-941223   \n",
       "16                                                  .   \n",
       "17                                                  .   \n",
       "18                                                  .   \n",
       "19                                                  .   \n",
       "20                                                  .   \n",
       "21  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "22  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "23  ATAC-seq:chr1:958785-959675;DNase-seq:chr1:959...   \n",
       "24                        ATAC-seq:chr1:960473-961422   \n",
       "25                        ATAC-seq:chr1:960473-961422   \n",
       "26                                                  .   \n",
       "27                                                  .   \n",
       "28                        ATAC-seq:chr1:966276-967323   \n",
       "29                        ATAC-seq:chr1:966276-967323   \n",
       "30                        ATAC-seq:chr1:966276-967323   \n",
       "31                                                  .   \n",
       "32                                                  .   \n",
       "33                        ATAC-seq:chr1:976006-976687   \n",
       "34                                                  .   \n",
       "35                                                  .   \n",
       "36                                                  .   \n",
       "37  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "38  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "39  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "40  ATAC-seq:chr1:999640-1000919;DNase-seq:chr1:10...   \n",
       "41                       ATAC-seq:chr1:999640-1000919   \n",
       "42                       ATAC-seq:chr1:999640-1000919   \n",
       "43                      ATAC-seq:chr1:1012923-1014210   \n",
       "44                      ATAC-seq:chr1:1019149-1020697   \n",
       "45                      ATAC-seq:chr1:1019149-1020697   \n",
       "46                                                  .   \n",
       "47                                                  .   \n",
       "48                                                  .   \n",
       "49                                                  .   \n",
       "\n",
       "                                          chromHMM_MT  \\\n",
       "0                           chr1:10600:96080:25_Quies   \n",
       "1                         chr1:348136:501536:25_Quies   \n",
       "2                         chr1:634820:733820:25_Quies   \n",
       "3   chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "4   chr1:923420:923820:2_PromU;chr1:923820:924620:...   \n",
       "5   chr1:924620:925220:2_PromU;chr1:925220:925620:...   \n",
       "6   chr1:925220:925620:22_PromP;chr1:925620:926820...   \n",
       "7                       chr1:925620:926820:23_PromBiv   \n",
       "8                       chr1:925620:926820:23_PromBiv   \n",
       "9                       chr1:925620:926820:23_PromBiv   \n",
       "10                      chr1:925620:926820:23_PromBiv   \n",
       "11                      chr1:925620:926820:23_PromBiv   \n",
       "12                        chr1:928020:933420:25_Quies   \n",
       "13  chr1:938220:939220:17_EnhW2;chr1:939220:939420...   \n",
       "14  chr1:939820:940220:23_PromBiv;chr1:940220:9404...   \n",
       "15  chr1:940420:941020:2_PromU;chr1:941020:941420:...   \n",
       "16                        chr1:941620:942620:22_PromP   \n",
       "17                        chr1:941620:942620:22_PromP   \n",
       "18                          chr1:945220:950020:7_Tx3'   \n",
       "19                          chr1:945220:950020:7_Tx3'   \n",
       "20  chr1:957820:958620:4_PromD2;chr1:958620:959220...   \n",
       "21  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "22  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "23  chr1:958620:959220:3_PromD1;chr1:959220:959420...   \n",
       "24  chr1:960220:960420:2_PromU;chr1:960420:961220:...   \n",
       "25  chr1:960420:961220:1_TssA;chr1:961220:961420:2...   \n",
       "26  chr1:962420:962820:7_Tx3';chr1:962820:965820:8...   \n",
       "27                          chr1:962820:965820:8_TxWk   \n",
       "28                         chr1:966220:967220:2_PromU   \n",
       "29                         chr1:966220:967220:2_PromU   \n",
       "30                         chr1:966220:967220:2_PromU   \n",
       "31                       chr1:969220:972420:24_ReprPC   \n",
       "32                        chr1:972420:975820:25_Quies   \n",
       "33  chr1:976020:976620:16_EnhW1;chr1:976620:980820...   \n",
       "34                       chr1:980820:984220:24_ReprPC   \n",
       "35                       chr1:980820:984220:24_ReprPC   \n",
       "36                       chr1:980820:984220:24_ReprPC   \n",
       "37  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "38  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "39  chr1:1000020:1000420:1_TssA;chr1:999020:100002...   \n",
       "40  chr1:1000020:1000420:1_TssA;chr1:1000420:10020...   \n",
       "41                       chr1:1000420:1002020:2_PromU   \n",
       "42                       chr1:1000420:1002020:2_PromU   \n",
       "43  chr1:1013220:1013420:1_TssA;chr1:1013420:10142...   \n",
       "44  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "45  chr1:1019820:1020020:22_PromP;chr1:1020020:102...   \n",
       "46                     chr1:1033420:1034620:24_ReprPC   \n",
       "47                     chr1:1033420:1034620:24_ReprPC   \n",
       "48  chr1:1033420:1034620:24_ReprPC;chr1:1034620:10...   \n",
       "49                        chr1:1041020:1052220:8_TxWk   \n",
       "\n",
       "                                               HiC_MT  GeneTPM_MT  \\\n",
       "0                                                   .        0.00   \n",
       "1                                                   .        0.00   \n",
       "2                                                   .        0.00   \n",
       "3   P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.34   \n",
       "4   P:1:922000:924000;D:1:894000:896000;GOTHIC:rea...        0.34   \n",
       "5   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...        0.34   \n",
       "6   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...        0.34   \n",
       "7   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...        0.34   \n",
       "8   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...        0.34   \n",
       "9   P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...        0.34   \n",
       "10  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...        0.34   \n",
       "11  P:1:924000:926000;D:1:890000:892000;GOTHIC:rea...        0.34   \n",
       "12  P:1:930000:932000;D:1:912000:914000;GOTHIC:rea...        0.34   \n",
       "13  P:1:938000:940000;D:1:860000:862000;GOTHIC:rea...        0.34   \n",
       "14  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...        0.34   \n",
       "15  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...        0.34   \n",
       "16  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...        0.34   \n",
       "17  P:1:940000:942000;D:1:892000:894000;GOTHIC:rea...        0.34   \n",
       "18  P:1:944000:946000;D:1:892000:894000;GOTHIC:rea...       11.08   \n",
       "19  P:1:946000:948000;D:1:884000:886000;GOTHIC:rea...       11.08   \n",
       "20  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...       11.08   \n",
       "21  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...       11.08   \n",
       "22  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...       11.08   \n",
       "23  P:1:958000:960000;D:1:1010000:1012000;GOTHIC:r...       11.08   \n",
       "24  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...        0.74   \n",
       "25  P:1:960000:962000;D:1:898000:900000;GOTHIC:rea...        0.74   \n",
       "26  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...        0.74   \n",
       "27  P:1:962000:964000;D:1:1012000:1014000;GOTHIC:r...        0.74   \n",
       "28  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...        0.02   \n",
       "29  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...        0.02   \n",
       "30  P:1:966000:968000;D:1:912000:914000;GOTHIC:rea...        0.02   \n",
       "31  P:1:970000:972000;D:1:1284000:1286000;GOTHIC:r...        0.02   \n",
       "32  P:1:972000:974000;D:1:822000:824000;GOTHIC:rea...        0.02   \n",
       "33  P:1:976000:978000;D:1:1002000:1004000;GOTHIC:r...        0.01   \n",
       "34  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...        0.01   \n",
       "35  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...        0.01   \n",
       "36  P:1:980000:982000;D:1:1000000:1002000;GOTHIC:r...        0.01   \n",
       "37  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...        3.55   \n",
       "38  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...        3.55   \n",
       "39  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...        3.55   \n",
       "40  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...        3.55   \n",
       "41  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...       17.92   \n",
       "42  P:1:1000000:1002000;D:1:1002000:1004000;HOMER:...       17.92   \n",
       "43  P:1:1012000:1014000;D:1:1000000:1002000;GOTHIC...       17.92   \n",
       "44  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...        3.68   \n",
       "45  P:1:1018000:1020000;D:1:1000000:1002000;GOTHIC...        3.68   \n",
       "46  P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...        3.68   \n",
       "47  P:1:1032000:1034000;D:1:1020000:1022000;GOTHIC...        3.68   \n",
       "48  P:1:1034000:1036000;D:1:1016000:1018000;GOTHIC...        3.68   \n",
       "49  P:1:1044000:1046000;D:1:1018000:1020000;GOTHIC...        3.68   \n",
       "\n",
       "    TranscriptTPM_MT       Type  score  \\\n",
       "0               0.00        low      0   \n",
       "1               0.00        low      0   \n",
       "2               0.00        low      0   \n",
       "3               0.00       high      4   \n",
       "4               0.06       high      5   \n",
       "5               0.00   moderate      2   \n",
       "6               0.00   moderate      2   \n",
       "7               0.00        low      0   \n",
       "8               0.00        low      0   \n",
       "9               0.00        low      0   \n",
       "10              0.00        low      0   \n",
       "11              0.00        low      0   \n",
       "12              0.26        low      1   \n",
       "13              0.01   moderate      2   \n",
       "14              0.00       high      4   \n",
       "15              0.00       high      4   \n",
       "16              0.00   moderate      2   \n",
       "17              0.00   moderate      3   \n",
       "18              0.00        low      1   \n",
       "19              0.40   moderate      2   \n",
       "20              0.02        low      1   \n",
       "21              8.90  very_high      8   \n",
       "22              0.00  very_high      7   \n",
       "23              1.76  very_high      7   \n",
       "24              0.61  very_high      6   \n",
       "25              0.08  very_high      6   \n",
       "26              0.00        low      0   \n",
       "27              0.05   moderate      2   \n",
       "28              0.00       high      5   \n",
       "29              0.02  very_high      6   \n",
       "30              0.00       high      5   \n",
       "31              0.00        low      0   \n",
       "32              0.00        low      0   \n",
       "33              0.01       high      4   \n",
       "34              0.00        low      1   \n",
       "35              0.00        low      0   \n",
       "36              0.00        low      0   \n",
       "37              0.00  very_high      6   \n",
       "38              0.59  very_high      7   \n",
       "39              2.01  very_high      8   \n",
       "40              0.94  very_high      7   \n",
       "41              0.00       high      5   \n",
       "42              0.00       high      5   \n",
       "43             17.92  very_high      6   \n",
       "44              2.46  very_high      7   \n",
       "45              0.85       high      5   \n",
       "46              0.00        low      1   \n",
       "47              0.00        low      1   \n",
       "48              0.00        low      1   \n",
       "49              0.00        low      1   \n",
       "\n",
       "                                         score_source  \n",
       "0                                                      \n",
       "1                                                      \n",
       "2                                                      \n",
       "3   OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "4   OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "5                         ChromHMM:ActivePromoter(+2)  \n",
       "6                         ChromHMM:ActivePromoter(+2)  \n",
       "7                                                      \n",
       "8                                                      \n",
       "9                                                      \n",
       "10                                                     \n",
       "11                                                     \n",
       "12                                TranscriptTPM>0(+1)  \n",
       "13                    TranscriptTPM>0(+1); TSL1-2(+1)  \n",
       "14  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "15  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "16                        ChromHMM:ActivePromoter(+2)  \n",
       "17            ChromHMM:ActivePromoter(+2); TSL1-2(+1)  \n",
       "18                                      GeneTPM>1(+1)  \n",
       "19                    TranscriptTPM>0(+1); TSL1-2(+1)  \n",
       "20                                TranscriptTPM>0(+1)  \n",
       "21  OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...  \n",
       "22  OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...  \n",
       "23  OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...  \n",
       "24  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "25  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "26                                                     \n",
       "27                    TranscriptTPM>0(+1); TSL1-2(+1)  \n",
       "28  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "29  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "30  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "31                                                     \n",
       "32                                                     \n",
       "33  OpenChromatin:ATAC/DNase(+2); TranscriptTPM>0(...  \n",
       "34                                         TSL1-2(+1)  \n",
       "35                                                     \n",
       "36                                                     \n",
       "37  OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...  \n",
       "38  OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...  \n",
       "39  OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...  \n",
       "40  OpenChromatin:ATAC+DNase(+3); ChromHMM:ActiveP...  \n",
       "41  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "42  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "43  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "44  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "45  OpenChromatin:ATAC/DNase(+2); ChromHMM:ActiveP...  \n",
       "46                                      GeneTPM>1(+1)  \n",
       "47                                      GeneTPM>1(+1)  \n",
       "48                                      GeneTPM>1(+1)  \n",
       "49                                      GeneTPM>1(+1)  "
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "collapsed_df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "1f6223fb-d6ca-4185-bee6-831ea759c33b",
   "metadata": {},
   "outputs": [],
   "source": [
    "collapsed_df.to_csv('/mnt/10T-backup-mingju/ifar/databases/hg38/RegionBased/promoter/MT/promoter_like_regions_MT_openchromatin_chromHMM_HiC_level.tsv',sep='\\t',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "cbb225f2-fa98-49ac-b82e-d58f7874eaf6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Type\n",
      "very_high    34.629312\n",
      "low          33.767016\n",
      "moderate     16.373677\n",
      "high         15.229994\n",
      "Name: proportion, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "type_percentages = collapsed_df['Type'].value_counts(normalize=True) * 100\n",
    "print(type_percentages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "fa24f6c9-e9e9-4e1f-854b-e921276e2d5c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAtIAAAIoCAYAAABEYwzVAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjgsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvwVt1zgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAWfpJREFUeJzt3Qm8lHP///FP+14q0V1Ei6KkBZGoRKLlrhBFCxItshWFREkLLSqVnVZR1E0K3dJCyn1ThCwtlO6k0kqbOv/H+/t7XPOfmTPndM73nM42r+fjcR5zzsw111znzJyZ9/W5Pt/vlSshISHBAAAAAKRK7tQtDgAAAIAgDQAAAHiiIg0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNIB08fbbb1u1atVCXzVr1rQGDRpYp06d7Pnnn7edO3cmus/48ePdsqlx4MABd7+VK1em6n6xHqtJkyZ25513Wnp699137bXXXot5mx5f25GVffbZZ3bttdda7dq13fb++9//TrSMntPw5zqpr6z4u/73v/+1e+65xy677DI799xz7fzzz7f27dvbjBkz7K+//krX5zsr0v+NnpvU/v8AiC1vEtcDgJdhw4ZZpUqV7O+//3bh+YsvvrAXX3zRXnnlFRszZoxdcskloWXbtWvnAk1qg/Szzz5rd911l1100UUpvp/PY/mYN2+e/fTTT3bLLbckuu2NN96wsmXLWlaVkJBg9957r5155pk2adIkK1SokFWsWDHRco899pjt378/9PPixYvd8sFzH8hqv+u4ceNswoQJVqdOHRemK1So4F5Pq1atcq+pn3/+2R5++OF0e76zoho1arjXYZUqVTJ7U4AcgSANIF2dddZZrhodaNasmQsZN910kwu/H374oZ188smhoHWiw5aCkgJhRjzW8ajKm5X9/vvvtnv3brvyyiutfv36SS4XHcI2bNgQ87nPShYsWOBC9PXXX29DhgyxXLlyhW5r1KiR3X777bZ69WrLqY4cOeJ+56JFi2b51yGQndDaAeCEK1eunPXr18/+/PNPmzlzZrLtFmotUOuAqs3nnXeeNW7c2Hr37u0C8a+//hoKeKogBi0E/fv3j1jft99+a3fffbddeOGF1rRp0yQfK7Bw4UJr1aqVC4FXXHGFTZkyJWbbih4/ucPk2m5VZ7ds2RLR4hCI1e7w448/Wo8ePdy26vFbt25tc+bMifk4qn6qqn/ppZda3bp13Q5KEGJT0tLQpUsXV42tVauWa2fQtoY/Fw0bNnTfjxw50j2eWl98zJ07191fld5oet5UFd22bVvob9ayZUu3fTfccIN7znXk4JlnnrGjR49G3Pfw4cM2ceJEu/rqq11bxsUXX2wPPfSQ/fHHH8fdJt2vRIkSNmDAgIgQHVDA1N81MH36dLv55pvd603BU68PHVlRIA0c7/lO6fZqueHDh7tWKD03etxvvvnG/f2D17bP60XPg9arv6eW/eWXX5Js7VizZo11797d6tWr55Zt06aNzZ8/P2IZ/Q+OGDHCbZeW0bJqA9LrEohXVKQBZAhV/fLkyeMCU1IUVNWzfMEFF9iTTz5pxYsXd4Fr2bJlLsCccsop9tJLL7nqoSqLateQUqVKRaxHwbt58+YuLB6v73Xt2rU2dOhQVy1XpVw9r3psPV7Xrl1T9Tuq5eHRRx+1zZs3u8B4PArB2sbSpUvbI488YiVLlrR33nnHhacdO3ZYt27dIpYfPXq0C9DaPrVWKPAqVCnw6G+blM8//9xuu+02q1q1qrtv/vz57fXXX3fBSevU30p/y7PPPtv9HYJwq+V8aH3aNoVRBfeA2n3UVqCK96mnnhq6fvv27XbffffZHXfc4XaAglaRvXv32sCBA90yx44ds549e7pWIT0v+jsowGoH4Ouvv7a33nrLChYsmGSlXQFU26WjEymxadMm9zc47bTTLF++fPb999/bc889554ztbAc7/lOzfYqXOs51OtaYXvdunXueQhvn/F9vWgnYNCgQZY7d253Py0XbcWKFe6xFeIff/xxK1asmNsePScHDx50YVn0e+vx1P5zzjnnuGCtv6uOYgDxiiANIEMULlzYffAr1CRFleRDhw7Zgw8+6EJdQNXAgKqZojaNpA5Rq5qmQJYS2h5V7oLHU+BXxVCVRLWjpDR4BS0PCv8KoCk5fK7wpcCuCvg//vGP0OMrQKoNQaFJoSZ8/QqoAYUjhRpVE5N7vFGjRrntmjp1qhUpUsRdd/nll7u/kyqM11xzjft7KuiKtiUth//1+994441ukKlCogKcqK1Hf++OHTtGLK8gpr+3jgaIKsN6HSjsK+DpiIZaM7RDpSB61VVXhe6r5007VTpqoOcrlq1bt7pLheKU0naHh2Lt3J100kmuh1rBVdXt5J7vlG6vQrMqugrBffr0ccuoMq2duvvvvz9Nrxf1gKsv/HgUtNWWM3nyZMub9/9igarYu3btcmFcrxO91nSEQdsW3g+uI0ZAPKO1A0CGDmZLjqpcqv6pyqfD1ar0+QgPLsejABEe2kWVSFUDFexPJFUC1ToQhKJA27ZtQ4PgwkW3WgRtBP/73/+SfAxV5L/66ivXqx6EaFEF+5///Kf99ttvKW4PSY0OHTq4yzfffDN0nSrUqoqrLSGctisI0eHPgQLsf/7zH/fzxx9/7EKrdgAU+IMvvWbKlCnjqu7p6bvvvnMVe7UY6TG0A6f2JLWbaFDi8aR0e4NL7cyE0/MVhFrf10tK/g/U7qHnP9hZDd9WtfroaMHGjRvdbWrnWLp0qduZU2uIqtVAvKMiDSBDKNCp8qgglRRV0DSVmNo3Bg8e7O5z+umnu1YD9femlFpAUioY+BjruhN9yFrrV6hKavujH18V0XBB60VygUbVSu3ApOZx0oP+hmqlUCuHWjY0s4XaevS8xlo2qeuCbdMMMPpd1Gsci6qnSQmCZ3SPe1K0Y6I+Zc1Yogp0+fLlrUCBAq4lQ9ufkgCZ0u0Nfr/ov4FCdPTzndrXS6xlowWtHjoyoa/ktlX95TpyobYP9Yvrb6KjBzqCpJlegHhEkAaQIdT3qmqeBiglR4fQ9aVlNeBK7QjqYVbQaNGiRbpvV6ye0eC6IMgoMASDwlIa3lJC61fFL1rQ/qJWmLRSVVSH5U/048TSuXNn+9e//mUfffSRa3PQtoS36aTmOdA26nvtZMUSXm2PFTS1A/fpp5+GZnFJjubO1k6c2jIUogPqk06plG5v8Pvp9w3vG1dFONaOVGqex1iDKmNtp2hsQjAwN1owBaLas9QypS9tr6rTahtS5f79998/7mMBORGtHQBOOFX4nnrqKde/qT7OlFDrgQY/aUCXBG0WKanCpoYqpdEBST2rCjpBP3YQpn744YeI5RYtWpRofdq+lG6bDtPrcH0wg0VA4VNhLz2mKVP40d9RM5OEb5faJjRwTBXGWHNFpwdVYzXYUNVLDeJUC4K2J5pmc1HYjn4OtAMQtIGoF1fBUtutFoPor/D5q2PRwL89e/a4qe9itRhpGz755JOIABo+2FL3CW9TOd7zndLtDX6/6BkyPvjgg1DP+ol8vWg7VE3W/0Cs7dSXZjSJph1bDULUzq1aP7SDAsQjKtIA0pWCqarJCgEatKfD+RpYpWCswVLRM2yE0+AyBQWFEB2O14AzzW4gwYlc9KGuYKvgpWChQV+qqqVmIFl0tVIzX2iWBB0KV7hU5bJv376hyqXChMKmdgb0u6myqqqlZmSIpsqnBtXpTHkKkgplSc2t3KtXL9dLq8qtvtfvosCp6v0DDzwQMXAsLTRoTbN26HF0qT50bZ+eKw0mS0nl0pceU7M/6DGSGgyoSqtmi9CgQIW6JUuWuNCqPmsNNBQFNv1t1CaiVh9Nk6ffQz3e6tdVj3VSFdWgB1kzTGhQo3qCNeAvOCGLWjY0LaNaUdSqoNea1q2/mwY76kiEXptq1Ujp853S7VWPvvrBX331Vfc/olk79LzoZz3/4c/NiXq9aLChBjtqdhHt7Kgyrp2O9evXux3YYMCiZnbR/6Z68/XYul0hXjtLqRmUC+QkBGkA6SqY7UChQYGzcuXK7kNaH8LJhWjRQCyFWB1S1yFsVS8VVDQVWvgcv5rCTaFWAVghRx/+mi/Xhx5TlTU9pgaRKVjrdwifmUABR1OfPfHEE65CriqkgpKmZlNQCqeQoyCk+Z737dvnKpnRlezwaqACnMJs0Hurv5emGQumHEsPaqdR77l+R/1uqpJqgKX+rhoMdyJpqjv9vTRoL6k+Wu3A6G+pHl2FXYU0tQtoGsPw50DbqxkrFN5eeOEFd50q6qrqJtd7H9DZDBWS1S6k50cVY7XtKMzeeuutbqYR0XOgv5XmstY2KOgr7Oo1ET3FXFLPd2q2V8+3/gazZ892z5Nek3pshXj9D53o14vC+6xZs9xrXG1U2mHQ76x1hw+C1HI6CqPZPbQDosCtGT30XAHxKlfC8YbRAwDgScFLOzwKkpqqLZqqteo156Qekb788ktXkdcMGbH6ygFkDVSkAQDpTvMj6wQkqjKrwhqcNRGJ6SiMpq5Ta4gq5Kpoa8dDFfzUTOUIIOMRpAEA6U59t6qqVq9e3bXdnMg+7OxOff8K02oD0aBH9fxrx0M92sGMMQCyJlo7AAAAAA9MfwcAAAB4IEgDAAAAHgjSAAAAgAcGG2YwjczWjIOaYxcAAABZz5EjR9wgaZ1wKDkE6QymEM3U3QAAAFlXSrMaQTqDBZXopE4ZDAAAgMy1Zs2aFC1HjzQAAADggSANAAAAEKQBAACAjEFFGgAAAPBAkAYAAAA8EKQBAAAADwRpAAAAwANBGgAAAPBAkAYAAAA8EKQBAAAADwRpAAAAwANBGgAAAPBAkAYAAAA8EKQBAAAADwRpAAAAwANBGgAAAPBAkAYAAAA8EKQBAAAADwRpAAAAwANBGgAAAPBAkAYAAAA8EKQBAAAADwRpAAAAwANBGgAAAPBAkAYAAAA8EKSRbpYtW2YdO3a0iy++2M4991y74oorbNiwYbZv376Yy3/zzTd2zjnnWJ06dVL8GD/99JN1797dzj//fHe/a6+91r788svQ7Tt27LDbb7/d6tatax06dLBffvkl4v67d++2+vXr27fffpuG3xQAAMAsL38EpJc9e/a4cNulSxcrXry4C73jx493l6+88krEsgkJCfbEE09YqVKl7K+//krR+r///nu7+eabrXHjxjZ69GjLmzevC8QHDx4MLTN06FA7evSojRs3zl588UXr37+/vf7666Hbn3nmGbvyyiutRo0aPPEAACBNCNJINy1btnRfgYsuusjy589vjz76qG3bts1OPfXU0G1vvfWW7dq1y6677jqbOnVqitb/+OOPuxA9atSo0HUNGjSIWObTTz91Afq8886zYsWK2Q033OCCeuHChV0QX7BggfsCAABIK1o7cEKddNJJ7vLvv/8OXbd3714Xhh966CHLly9fitazfv16W7VqlWsdSc6RI0esYMGC7vvgUtfJk08+ab169XJVcAAAgBwTpGfNmmX//Oc/rWbNmq6HVX2w4ZYsWWJt2rRxtzdt2tSmT58ecz0vv/yyNWnSxC2naufKlSsTLbN//34bOHCgq5iqFUGPtWXLlkTLbdy40bp27Wq1a9d22zRkyJCINgLEptaKQ4cOubaLCRMm2OWXX27ly5ePaK9Qa4WuT6nVq1e7S/Vbt27d2qpXr+6e5+hqtp73GTNmuDYTvUYqVKhgJUqUsPfee89VwG+66SaeNgAAkHNaO9RH+9prr7lAW6tWLReCNHAtoEpkz549XYBSz6sGlynUqm2gXbt2ESF6zJgxdt9997mgpXDerVs3d1mtWrXQcn369HEhTy0HRYsWdf20t956q73zzjuhKqaqpur1LVeunLv9jz/+cAPnNFht5MiRGfwXyl4UkNXKIZdddpnrZw6sXbvWZs+ebXPmzEnVOjWIUB544AG77bbb3Otk0aJF7nWgoKydMOnXr5/dcccdri9arR16bR04cMCefvpp9/yprxoAACBdJGSydevWJZxzzjkJy5YtS3KZrl27Jlx//fUR1w0YMCChQYMGCUePHnU/Hzp0KOH8889PGDFiRGiZv//+O+Gaa65JuPfee0PXrV69OqFq1aoJixcvDl23ZcuWhOrVqyfMmDEjdN3zzz+fUKtWrYSdO3eGrnvnnXfcfbXNvr7++mv3lZOtXbs24Ysvvkh44403Eho1apTQqVMn91wcO3YsoX379gnDhg0LLTtu3LiE2rVrH3edEydOdH/78OdX7rzzTvcchzt8+HDC+vXrEw4ePOh+Hj16dELv3r3d94sWLUq46qqrEurWrZvQp0+fhP3796fTbw0AAHKKlOa1TG/tePvtt+3000+3Sy+9NObthw8fthUrVliLFi0irm/VqpVt377dvvvuO/ezqtQ67B8+2C1PnjzWvHlz1xaiWSJE32tGiYYNG4aWU9VZ06XptsDSpUtdO0d4P22zZs1cFTx8OSR29tlnu7+nBvo9++yzrr1m4cKFNn/+fNfr3KlTJ1fx15daQCT8+1hUdRZNrRdOP//888+hPmhR33WlSpWsQIECtnnzZledVqV6586ddv/999tdd93ltketO5MmTeIpBAAAXjL9OPdXX31lVatWdb2006ZNc2FYPcmPPPKIm2N406ZNLiQpGIWrUqWKu1Qw05zFupTo5SpXrmx//vmnazUoW7asW65ixYqWK1euROv75JNPQj9rOfVYh1OIVs9t8Fi+FOpTOuVbdnfGGWe4HZp169a5nSK17ai3OdqFF15ot9xyi91zzz0x13Paaae5S4Xt8L+d1qnnUu0b4WE6oNaP9u3bW8mSJW3x4sXuOdT81qKdM7XzqG0IAAAgPKtFZ8UsGaRVVVa/suYaHjRokKsmqoqpnuUPP/zQBS9RFTlc8HNwuyqaCklBj3N0JVO9zQrSWk69s9G0vmBdwfqiHzPWcj4U+NQrHA805ZwGH+rFqL71AQMGRNyuyv9nn33mKsalS5dO8u+i6nKRIkXc1HXhRwnUJ62BjHr9RFuzZo076YsCutb766+/hgZB5s6d252sRQE8Xp4LAACQcsqVWT5IB9VZDQo766yz3HWa0UFVwzfeeMO1CEhSewXh18daJmjpON5yyV0fvb6ULJcc7SwEFfWcRIM4FZb1PCr4/vjjjzZ58mT3s6atizXVnXak/vOf/ySq/mvw4D/+8Q97/vnnQ9f16NHDzfihKrdm51CLjQaiajCjjl6E03R7Oqrx4IMPuoGJonmsdeTjgw8+sAsuuMBVqNWuE31fAAAQ33QkPSUyPUirYnzyySeHQrSccsoprkVDv0QwRVp0FVgVYwmqxrpUtVFfCnHRywWVaS23devWRNsRXYHW98F9w6n1RO0iaaEgrhOE5DSaSlB90K+++qrb4VCl+MYbb3RTCGp2lFgUrmP9PY4dO+Yuw6/XDCzaO5wyZYo76Yp660eMGJGof140C4xeR5rpJaB1KXQ/9dRTbtYQndzl7rvvzpHPBQAge9FsZSoeKftoml4Vf3QmXo3rCY6k6yzBaknUEVYVjPQ5qM9ZnfX3eEU+3UefgZ9//rlreT3zzDPd0f9g1qtghqxgdjTNdjZ8+HBXvAro6P4111xjL730Uo4/Q3CuFBZNMz1IK5T+73//S3S9gpgOv6snWWFrw4YNEQMEgz2FINQGl+pfVlU0oJ/VEhCcVU/LLV++PFFlWesLD8j6ProXWv246tmOrp7i/2jaOX2lRu/evd1XNLVsxKIpCfV1PGrn0Fc0HekIeqQBAMgqVDBUQUqfcSrmqWVRR+t1qQAtwaQKKj4qG6k1UmOBFLyjz78RTkVGFbXk4YcfdidLmzdvnptSVi2xV111lbtt6NChrh1T0/6qYKVQrQH7AR0VVrjP6SE6NTI9SKsqqOqg2gA06FA0MFDB+dprr3UVSM3MoN7Y8GCkF0CZMmVCoVktINpjU0U0uE4vBt2vUaNGodCs73V4X3t+QTBXhVp7X+H9u7pNMzroJB4aqCaa6UFhWusAAABILwrI4TOP6aRxykA654VykQqC0QPyL7nkEleMVI5KLkhrzJBmuFK7ZTD7le6rCR+Um4Ig/emnn7oAfd5557lMpdm31H6rI7ca86RMpS9koSCtsxRqz0ZVSb1A9KJR0NWAMj2BotM6q8dWQVfT3in06iQrgwcPdlVr0f3UQ6sTsui+wQlZNP1Z+AlB1C+r8K7+We1pqeVg7Nixrg2hbdu2oeU004NmEdGMDvrS1Gk6xKHHT2trBwAAwPGocixq40iKin2xZq0KF9w/erIF/RyMJROtJ5i0IbgM1v3kk0+6PBY+4B9Z4BThmhpNez+awk6n7e7bt6/rmVaPa9C7qkMdEydOdHtUOjTx1ltvuVAdflZD0Rnv7r33XnfaaPXTau/rhRdeiDiroYwaNcqFac0SouXVS6u+3vAZP3RYRXtu2gaFfIVo7SnqEEpmOnbs/7/gkfPxfANAfNHR9GCGKRUWNVZMxb7oYKw+Zw2anzt3rnXu3DnZdZ5//vlukgMVFlVgVIuIJnTQzFYqHAY0kH/GjBmuzWT69OmuvVZjzN577z13hP6mm246Yb93dpVLZ2XJ7I2IJ9oZCF6svia8/qlt+T1tU/Ah6yt/Sgnr1aFBZm8GACADqbVUrRxy2WWXuX7l8EHxmro1aMUQHY1XUfB4dGRdy6qdQ9RjHRQJAzrJncY6aUYtVavVo61ze2iA4bBhw9yJ6uLFmhTmtUxv7UDqKUT/vGUXfzoAAHIYHUlXX7ImQdDRePU+66i5juCLpoadPXu2W0bTx+qovtpcNQtVUg4ePOhuV7Vb5+pQW6sG9T/00EMRZ3tWW+zHH3/sqtaqgmsWNLXMqmdaIVq3KXxrdg9VygcNGuQmdIhnBGkAAIAs4uyzzw5NoqBgq5nCNNnB1VdfHRoTFlRJNSBR1eqRI0dahw4d3CQMsSh4qxKtk6AFPc4Kxhqo+PTTT0fMiqZKdXCWaAVqzdqhwYyqaN9///1ufFqDBg1cC+2kSZNcS248y/QeaQAAACSmE4apEq2pd5OiCRtUad6yZUuSy6i6rVk/ogcKav0Ky0lRO0enTp1cdXr16tUuxGvSBa1HEzQsX7487p82gjQAAEAWpLP3KiSfdtppSS7zxRdfuCl+k1umXLlyru9aVeVwGmwYPZAxoKnw1q5d6yrPAc3goe0RtZaA1g4AAIBMpzMYagYzzTSmWcQ0b7POIKifdRIUzbShUKszEepsg5q5Y8WKFW6mMp3dUDOehU8trPCs2cdE91Hvte6vL/VFq11EPc+PP/54om3RujXdXb9+/UIzmmnQoc46rDaSevXqudk9mjdvbvGOHmkAAIBMpgF9OjmKAq8mVFOlWOfT0LS/aqmQihUruumBVV1WwNX0dBrw16ZNm4h1qWqs0BsoW7asC9w6M6Gm8VU1WWFc319//fWJtkXn0VAwD/qypXTp0m764KeeesrefPNNN41wz549Ld4x/V02nP7u4bHzmbUjDpxZvqQNvYe9fQAAsmpeo0caAAAA8ECQBgAAADwQpAEAQFw4doyTOceTYxnwfDPYEAAAxIXcuXPZhNc/dWcIRs5W/pQS1qtDgxP+OARpAAAQNxSif96yK7M3AzkErR0AAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAA4IEgDQAAAHggSAMAAAAeCNIAAACAB4I0AAAAkB2D9Ntvv23VqlVL9DVy5MiI5ZYsWWJt2rSxmjVrWtOmTW369Okx1/fyyy9bkyZN3HLXXXedrVy5MtEy+/fvt4EDB9pFF11kderUse7du9uWLVsSLbdx40br2rWr1a5d2+rXr29DhgyxgwcPpuNvDwAAgOwqr2URL730khUrViz086mnnhr6ftWqVdazZ09r3bq19e/f37788ksXavPnz2/t2rWLCNFjxoyx++67z6pXr26zZs2ybt26uUuF80CfPn3s22+/tUcffdSKFi1q48aNs1tvvdXeeecdK1iwoFtm79691qVLFytXrpy7/Y8//rBhw4bZ7t27E4V8AAAAxJ8sE6Rr1KhhpUqVinnbhAkTXDAeOnSo+/niiy+2rVu32tixY13VOXfu3Hb48GGbNGmSde7c2VWRpV69etaqVSt77rnnXMCWr776yhYvXmwvvPCCNWrUyF1XtWpVV+WeM2eOdejQwV03c+ZMF6bnzp0b2q48efJY3759rUePHla5cuUM+bsAAAAga8r01o7jUUBesWKFtWjRIuJ6BeTt27fbd999535WlXrfvn3WsmXL0DIKvs2bN3dtIQkJCe46fV+8eHFr2LBhaDlVnevWretuCyxdutS1c4SH+2bNmrkqePhyAAAAiE9ZpiKtALxr1y4Xam+44Qa7/fbbXRDetGmTHTlyxCpVqhSxfJUqVdzl+vXr7dxzz3WXEr2cKsd//vmnbdu2zcqWLeuWq1ixouXKlSvR+j755JPQz1pO1e5wCtEVKlQIPZYvhfq//vor1ffTNhcqVChNj43s58CBA6EdQQCAHz5D49MBz89Q3Sc6K2bJIF2mTBnr3bu31apVy23wokWL7JlnnnHBVwMC9+zZ45ZTFTlc8HNwu9owFHSDHudAiRIl3KV6mxWktVx4L3b4+oJ1BeuLfsxYy/nQjsHatWtTfT+FaLW4IL5o0KveCAAA/vgMjU8b0/AZqlyZ5YP0ZZdd5r4Cl156qRUoUMAmT57sZtMIJLVXEH59rGWCvZDjLZfc9dHrS8lyycmXL1+oop4aaX1cZE86gkJFGgDShs/Q+FTR8zN03bp1KVou04N0LNdcc4298sorrmpbvnx5d110FVgVYwmqxro8dOiQ+1IQj14uqExrOQ1UjBZdgdb3wX3DqQ87rQMN9c9cuHDhNK0D8YN2HgAAMvYzNKU7Xll+sKF6klXB3bBhQ8w9hSDUBpfR/cv6uUiRIqHp9LScyvzReydaX3hA1vfR69LAR/VsM2MHAAAAsmSQnj9/vhtoqH5g9adoursFCxZELDNv3jzXXx30DGvWDfU+676Bo0ePuvtpmrtgz0Lfq9K8bNmy0HKqUGvWj2A6PNGsHpotRAMgAwsXLnRhOnw5AAAAxKdMb+3QnM8KyprLWT766CN788033XzQCsrSq1cv69ixow0YMMBNe6fQq5OsDB482M0hLQrcmt9Z80VryrrghCybN2+20aNHhx5PgxobN25sjzzyiDu5i07Iovmo1ULStm3b0HLt27e3adOmuRPB6Gvnzp02fPhw9/hUpAEAAJA3KzSBz54923777Tc7duyYnXnmmfbwww9bp06dQsvoNN4TJ050gVgnSNHsGwrV4Wc1lNtuu821bEydOtV27NjhwrlOvBJ+VkMZNWqUjRgxwgYNGuRm0NCpwsePHx8x44d6pDXgUWdQ1Kwiuk1T9OmELAAAAECuBKYDyFBr1qxxlzVr1vRex8Nj59vPW/5/ywlypjPLl7Sh9zTP7M0AgByFz9D4cGYaP0NTmteyZI80AAAAkNURpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAIDsHqT//PNPa9iwoVWrVs3WrFkTcduSJUusTZs2VrNmTWvatKlNnz495jpefvlla9KkiVvuuuuus5UrVyZaZv/+/TZw4EC76KKLrE6dOta9e3fbsmVLouU2btxoXbt2tdq1a1v9+vVtyJAhdvDgwXT8jQEAAJBdZakgPXHiRDt69Gii61etWmU9e/a06tWr24svvmht27Z1oXbWrFmJQvSYMWPs5ptvthdeeMHOOOMM69atm/3www8Ry/Xp08cWLVpkjz76qFv+999/t1tvvTUiJO/du9e6dOniwv24ceOsX79+9u6779qAAQNO4F8AAAAA2UVeyyLWr19vM2bMcIH1sccei7htwoQJLkQPHTrU/XzxxRfb1q1bbezYsa7qnDt3bjt8+LBNmjTJOnfu7KrIUq9ePWvVqpU999xzLjDLV199ZYsXL3ZBu1GjRu66qlWruir3nDlzrEOHDu66mTNnujA9d+5cK1WqlLsuT5481rdvX+vRo4dVrlw5Q/8+AAAAyFqyTEX6ySeftPbt21vFihUjrldAXrFihbVo0SLiegXk7du323fffed+/vLLL23fvn3WsmXL0DIKvs2bN3dtIQkJCe46fV+8eHHXQhIoV66c1a1b190WWLp0qWvnCEK0NGvWzPLnzx+xHAAAAOJTlqhIv//++/b999+7Fopvv/024rZNmzbZkSNHrFKlShHXV6lSJVTJPvfcc92lRC+nyrHaM7Zt22Zly5Z1yyms58qVK9H6Pvnkk9DPWk7V7nAK0RUqVAg9li+F+r/++ivV99M2FypUKE2PjeznwIEDoR1BAIAfPkPj0wHPz1DdJzorZskgrV9w+PDhdv/991vRokUT3b5nzx53qSpyuODn4Ha1YSjoFixYMGK5EiVKuMvdu3e7IK3lihUrluhxtL5gXcH6oh8z1nI+tGOwdu3aVN9PIVotLogvGvSq/xMAgD8+Q+PTxjR8hipXZvkgrb7m0qVL27XXXpvsckntFYRfH2uZYC/keMsld330+lKyXHLy5csXqqinRlofF9mTjqBQkQaAtOEzND5V9PwMXbduXYqWy9QgrSnnXnnlFTeYUFPSSdDyoEu1ZAQV5egqsCrGElSNdXno0CH3VaBAgUTLBevRchqoGC26Aq3vg/uGUx92Wgca6p+5cOHCaVoH4gftPAAAZOxnaEp3vDI1SP/666+uzeGOO+5IdJtm36hVq5ZNmzbNVXA3bNgQMUAw2FMIQm1wqf7l8PYH/VykSBE79dRTQ8stX748UWVZ6wsPyPo+uhdaAx/Vsx3dOw0AAID4k6mzdpxzzjk2ZcqUiK+HHnrI3TZo0CA3DZ76UzTd3YIFCyLuO2/ePCtTpkwoNGvWDfU+z58/P7SM5qTW/TTNXRCa9b0qzcuWLQstpwq1Zv0IpsMThXbNFrJr167QdQsXLnRhOnw5AAAAxKdMrUirfUJnF4ylRo0a7kt69eplHTt2dCdD0bR3Cr06GcvgwYPdHNKiwK35nTVftKasU8DWMps3b7bRo0eH1qsqd+PGje2RRx6x/v37uwGOmo+6fPny7kQvAU3Fp2q4TgSjr507d7pBkXp85pAGAABApg82TAmdxltnPVQg1glSNPuGQnW7du0ilrvttttcy8bUqVNtx44d7kQrOvGKTjkebtSoUTZixAhX9VZricL8+PHjI2b8UMifPHmyO4Ni79693W2ao1onZAEAAAByJTAdQIZas2aNu6xZs6b3Oh4eO99+3vL/W06QM51ZvqQNvad5Zm8GAOQofIbGhzPT+Bma0ryWZc5sCAAAAGQnBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAAyKggfc4559jXX38d87ZvvvnG3Q4AAADkZF5BOiEhIcnbjh07Zrly5UrLNgEAAADx19rx7bffWrFixdJ7tQAAAECWkjelC06ePNmmTJnivlfFuVevXpY/f/6IZQ4dOmQ7d+60Zs2apf+WAgAAANkxSJcuXdrOOuss9/2WLVvs9NNPt+LFi0cso2BdtWpV69y5c/pvKQAAAJAdg3TLli3dl3Tq1Mkef/xxq1y58oncNgAAACD7B+lwU6dOTf8tAQAAAHJ6kA5m7lizZo1r81BvdLQ2bdqkddsAAACAnBWkN27caD169LBffvkl5lR4GoxIkAYAAEBO5hWkBw8ebIcPH7YxY8ZYtWrVEs3eAQAAAOR0XkFaZzV84okn7Oqrr07/LQIAAABy6glZChcubEWLFk3/rQEAAABycpC+9tprbd68eem/NQAAAEBObu3QSVfee+896969uzVp0sROOumkRMtcddVV6bF9AAAAQM4J0n369HGXv/76qy1evDjmrB1r165N+9YBAAAAOSlIT5kyJf23BAAAAMjpQbpevXrpvyUAAABATh9sCAAAAMQ7r4p0586dk71dPdKTJ0/23SYAAAAgZwbpWKcF3717tzt1eKlSpezMM89Mj20DAAAAclaQnjp1aszrFaR79uxpd911V1q3CwAAAIifHumKFSta165d7emnn07P1QIAAAA5f7Bh+fLl7aeffkrv1QIAAAA5O0h/+OGHdsopp6T3agEAAIDs3yP90EMPJbru8OHD9uOPP9q6devsgQceSI9tAwAAAHJWkF65cmWi6woUKODaOu644w5r1apVemwbAAAAkLOC9KJFi9J/SwAAAIBshDMbAgAAABlVkQ5OwPLaa6/ZihUrbNeuXVayZEm75JJLrEuXLlaiRAnf1QIAAAA5tyK9bds2u/baa+25556zffv2Wbly5dzlxIkTrW3btu52AAAAICfzqkiPHj3aDh48aG+++aadd955oeu//vpr69Gjh40ZM8aGDx+entsJAAAAZP+K9LJly+zee++NCNGin++++25bunRpem0fAAAAkHOCtNo4NNVdLKeddpq7HQAAAMjJvIK0wvLixYtj3qZqtG4HAAAAcjKvIK2BhlOnTrUhQ4bYN9984wYX6nLo0KHu+uuvvz5VbSIdO3a0iy++2M4991y74oorbNiwYYmq2kuWLLE2bdpYzZo1rWnTpjZ9+vSY63v55ZetSZMmbrnrrrsu5slj9u/fbwMHDrSLLrrI6tSpY927d7ctW7YkWm7jxo3WtWtXq127ttWvX9/9vuoNBwAAALwGG95+++22efNmmzZtWkSgTUhIsBtuuMGFz5Tas2ePC7OaNq948eL2008/2fjx493lK6+84pZZtWqV9ezZ01q3bm39+/e3L7/80oXa/PnzW7t27SJCtAY63nfffVa9enWbNWuWdevWzV1Wq1YttFyfPn3s22+/tUcffdSKFi1q48aNs1tvvdXeeecdK1iwoFtm7969bps0I4lu/+OPP1zA17R/I0eO5JUDAAAQ57yCdK5cuWzw4MF2yy23uIqvwuVJJ53kqsoVK1ZM1bpatmzpvgKqEisgK+Sq0n3qqafahAkTXDBWxVv0OFu3brWxY8e6qnPu3Lnt8OHDNmnSJOvcuXMoyNerV8+drlzT9Clgy1dffeXaUl544QVr1KiRu65q1aquyj1nzhzr0KGDu27mzJkuTM+dO9dKlSrlrsuTJ4/17dvXzUxSuXJlnz8dAAAA4q21Q5Xj3r1728cffxy6rlKlSi54Kljq8ueff3bL6AQtaaFQLn///bcLyDrpS4sWLSKWUUDevn27fffdd+5nVanVDhIeyhV8mzdv7tpCVC0Xfa/Kd8OGDUPLqepct25dd1t4r7faOYIQLc2aNXMhP3w5AAAAxKcUB2m1R3z//fd22WWXJbmMbvvxxx+T7F9OztGjR+3QoUOu5UIV6Msvv9zNDLJp0yY7cuSIC+3hqlSp4i7Xr18fcRm9nCrHf/75Z+gkMVpOVXNV1aPXF6wjWC666qwQXaFChYjlAAAAEJ9S3Noxf/5814+cN2/Sd9FtWkbL3nXXXanaEAXnIOwqkOukL0ElXFRFDhf8HNyuNgwF3aDHORCcrlztJ2XLlnXLFStWLNHja33BuoL1RT9mrOV8qDr+119/pfp+Cv+FChVK02Mj+zlw4EDoiAoAwA+fofHpgOdnqO4TXXRNU5DWDBaaCeN4atSo4U4VnlrqWVa4XLdunbu/ZtJ49dVXQ7cn9cuEXx9rmeCPd7zlkrs+en0pWS45qrCvXbs21fdTiFavOOKL/vf0RgAg4/3222/23nvvuc8mDbJXK+BTTz2VaDm1Ib799tu2fPny0LghjcPROJ6Umjx5sn3wwQduzI4GwAdUvNFYHx3xPf30093nowpD4TNRafxOv379Uj1OKZ7wGRqfNqbhM1QF2nQL0mq9SK4aHVph3ryutzm1zj77bHepXmWFRb35LFy4MNTCEV0FVsVYgqqxLtUaoq8CBQokWi6oTGs5DVSMFl2B1vfBfcOpDzutAw3z5csX+r1SI60BHtmTPhipSAOZF6Q1vaumZ9WH6rFjx+ycc85J9PmomaV0VFXjhBS2dT99RS+bFM1UpelgNZOUxuaE30+zVelo66hRo+y1116zKVOmRBSaNBD/yiuvdGOCkDQ+Q+NTRc/PUO08p0SKg3SZMmXcSi+88MLjvhmcfPLJlhZ6A9FAQfVHa05oBc8NGzZEDBAMfsEg1AaX6l8Or9rq5yJFirjZP4LlVDGIrixrfeEBWd9H90Kr4qBtSk2FIRY9buHChdO0DsQP2nmAzHPNNdeEBrsr0CpUR79/v/HGG24M0YIFC7w//55++mlXhdZMUSpIhT+GZsd68cUX7bzzznPr1zSzomX0uCo66bH5XAHS7zM0pTteKR5sqKnkZsyY4doSkqLbXn/9dTeFXVpo3mjt4esMiaoAaLo7vUmEmzdvngv3QWhWJVu9z+rPDmgdup8OrwV/EH2vSrP2/AOqUGvWj2A6PFFo12wh4TOQ6M1KYTp8OQBAzqXpVY9n9uzZLnD7hmidw+DXX3915z1I6rM1GP8TXAafxU8++aT16tUrYoYpABknxUFaJydRn4kGEQaDAsPpOv0zaxnNL51SWp96vzSt3meffeYOV91zzz3uBCo6VCVar6oAAwYMcHvmmi9as4houeBNToFb0/DpsJdO5KIQ/OCDD7qeNvWTBWrVqmWNGze2Rx55xPW9aSo7rV8zhLRt2za0XPv27V0w1+E6hW5VCZ544gk37R5zSAMARMUVTcOqnuUHHnjAfcaosKMTf6VkKlj1N6vnWp9XSVXOND5JhSy1OGpWLM0epXZFfYbpMW666SaeDCCT5E1ND7NOqz1o0CB3Gm/1iyl8ik6vraCrdonHH3884iyCx6NDVaoia7Ch7q91BmdHDJq8deZDDUDUTB4KtHrDUqgOP6uh3HbbbW4dOk35jh073IlWtN7o7VGf2YgRI9zvor16VdB1NsXwGT/UI62BHzqDonredJvmqNaADgAARAMLNS5IrRf6LNH0rTrHgcLx/fffH9HLHMuzzz5rZ5xxRrL9zRpEeMcdd7gjvirw6PNKg6fUDqIz7qZk/BKAEyNV/30KuGeddZY9//zzrjK8evVqd732ojVl3Z133mm1a9dO1QbozUFfx6N2iuO1VKh9Q6cv11dyNJhD1WV9Ha9BXacdBwAglmAQk4ov48aNCxWANDZHRZivv/7aFYxi0dgcVZjffPPNZP+4amHUUVsdYVWxSQPqdbZerVcnDtNtw4cPdwUkTSWrIpEeH8CJl+rdWFWH1YqhkcvBYauSJUumqI8MAICcJJjtSe0c4VNlaWxPMAA/qSCtavLVV1/twnEwS5Q+W3WkVD+r6BN8tmrQfXDCMQVqVafnzJljO3fudJXvwYMHW4MGDVyftdofOXoKZAzv40H65y5dunT6bg0AANmIjsgGbY6xJFdk0piiTz75xA02DKcKtb7U9hhrTI4CeKdOndzjfvTRRy7Aa/yOaKyP5rMGkDForAIAIA00gP3f//63G3gYVKU1zWr4ORJi0bgfnfsgnKrLapHs3Lmzm4862qeffupO6BWc/VdUwdYsVZo21uesuQD8EaQBAEiCBvVpdqdgYL1m2Xj//fdD08Jq2jkNjldVWTNAdezY0X7//Xc3qF0zT4WfWEVnLFQ41kB2iTWmSP3POu9BrGlkNahR091p8GEwOF7rUDvIyJEjQ9PUcmIWIOMQpAEASIJ6kDXVarjgZ51hUIFXLRaaelUD/jTAUO0ezZo1c1PahVPVWKHX17Rp09xc1eqrDqjFUqFds4SoHUTVcU3bCiBjEKQBAEiCTgz2ww8/HPfvoylhFXSTs2jRouOuJ7lldI6GWOdp0JS0+gKQ8ZhqAwAAAPBAkAaQ7fzyyy/uBFGtW7d2c+zqZEnR+vfv707GFP21dOnSZNetmRQ0x7z6TNV/qnl5H374YXeSjXCas1dz1mvasw4dOrhtij5Rh+b4/fbbb9Ppt865jh37v7mYER94vpGT0NoBINvR3LwaAKbTMavnNDgpRrTTTz/dDcIKF2s6sehZET7//HN3AioNFPvtt9/c2eduvPFGe/fdd0Mnuhg6dKjredVJOHRWOwV3ze0beOaZZ9xgsxo1aqTL75yT5c6dyya8/qlt+X1PZm8KTrDyp5SwXh0a8HdGjkGQBpDtNGnSxIVUUYD95ptvYi6nmQ1Se7ZVVaJvvvlmd6bUgCrZqn5/+OGHbp7eIHArQOtkGzpts4K3ph4rXLiwff/997ZgwQL3hZRRiP55y/+d5AsAsguCNIBs50SeSVXTmUVTkNYcvZrWLHzu3mAKsuBS14mmKNNUaLHWBQDIOeiRBpBjbdq0yS644AI3o8K1117rTprhY9WqVa6NI7wtpGbNmm7O3j179tj06dOtQoUKVqJECXvvvfds165ddtNNN6XjbwIAyIqoSAPIkdTfrLBbpUoV27dvn+tfVpV47NixEfPwHo+qzOqHrlixopujN6CTYtxxxx1uvWrtGD9+vDt5x9NPP+1O4Zw3L2+vAJDT8U4PIEfq0qVLor7q9u3bu8GBqQnSmsFDgxs1R3B4ONZsIR9//LFt3rzZnZBDZ6QbM2aM65nWbB26TSfo0Owemvlj0KBBoYGKAICcgdYOAHHTV33VVVfZ+vXr7eDBgym6j2brmD17to0ePdpVt6Ply5fPKlWq5EK0ArWq06pU62x4999/v9111122cOFCN6XepEmTTsBvBQDITARpAHEjqWnyYlHfs9o1HnvssRSdNU7tHJ06dXLV6dWrV1v+/PmtVatWbsChZvpYvnx5GrceAJDVEKQBxAXNN/3BBx/YWWedFZplIykaMDhkyBC7++673fzRx6Op8NauXWvdunWL6K3WAEXRtHgAgJyHHmkA2Y4G9emELLJlyxbbv3+/vf/+++7nevXquds1v7TOeKjZNDSzhtouNN+0qszhmjZtauXKlbPJkye7n3UyFrVnaLaPBg0auOpyQNVlrS/c33//7aa7032CgK65qxXcdTIYbY9m99D81ACAnIUgDSDbUQ/yPffcE3Fd8POUKVPcvM9Fixa1CRMm2B9//OF6mTUFnk6gctlll0XcT1Vjhd7AypUrXTVZgTq6Gq0WDQ0gDKdBiCeffHLEAMbSpUvbqFGj7KmnnrI333zTzfbRs2fPdP0bAAAyH0EaQLZz2mmn2Q8//JDsMikd3Ldo0aKIn3v37u2+UuqWW25xX9HUV52S3moAQPZFjzQAAADggSANIKZjx1I+wwWyP55vAEg9WjsAxJQ7dy6b8PqntuX3PfyFcrjyp5SwXh0aZPZmAEC2Q5AGkCSF6J+37OIvBABADLR2AAAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAIAHgjQAAADggSANAAAAeCBIAwAAAB4I0gAAAEB2DNILFiywnj17WqNGjax27drWqlUrmzFjhh07dixiuSVLllibNm2sZs2a1rRpU5s+fXrM9b388svWpEkTt9x1111nK1euTLTM/v37beDAgXbRRRdZnTp1rHv37rZly5ZEy23cuNG6du3qtqt+/fo2ZMgQO3jwYDr+9gAAAMiuMj1Iv/rqq5Y/f3578MEH7bnnnrMrr7zSnnzySXv66adDy6xatcqF7erVq9uLL75obdu2daF21qxZiUL0mDFj7Oabb7YXXnjBzjjjDOvWrZv98MMPEcv16dPHFi1aZI8++qhb/vfff7dbb701IiTv3bvXunTpYn/++aeNGzfO+vXrZ++++64NGDAgA/4qAAAAyOryZvYGKDyXKlUq9PPFF19sf/31l6s433fffS5kT5gwwYXooUOHhpbZunWrjR071lWdc+fObYcPH7ZJkyZZ586dXRVZ6tWr5yrcegwFZvnqq69s8eLFLmirCi5Vq1Z1Ve45c+ZYhw4d3HUzZ850YXru3Lmh7cuTJ4/17dvXevToYZUrV87wvxUAAACyjkyvSIeH6MA555xjhw4dst27d7uAvGLFCmvRokXEMgrI27dvt++++879/OWXX9q+ffusZcuWoWUUfJs3b+7aQhISEtx1+r548eLWsGHD0HLlypWzunXrutsCS5cude0c4dvXrFkzF+zDlwMAAEB8yvQgHcsXX3xhJ510kpUuXdo2bdpkR44csUqVKkUsU6VKFXe5fv36iMvo5VQ5VnvGtm3bQstVrFjRcuXKlWh9wTqC5aKrzgrRFSpUiFgOAAAA8SnTWzuirVmzxt5++23r1auXqyjv2bPHXa8qcrjg5+B2tWEo6BYsWDBiuRIlSrhLVbfLli3rlitWrFiix9X6gnUF64t+zFjL+VB1XO0rqaXwX6hQoTQ9NrKfAwcOhI6oZBRea/GJ1xp4rSGnOeD5Gar7RBdds3yQVqvG3Xff7Wbc0CDBcEn9MuHXx1om+OMdb7nkro9eX0qWS44q7GvXrk31/RSi1SuO+KLZY/RGkJF4rcUnXmvgtYacZmMaPkNVoM02QVr9zQrPqihr0GC+fPkiKsrRVWBVjCWoGutSfdX6KlCgQKLlgvVoOQ1UjBZdgdb3wX2jtzOtAw31uwWtKamR1gCP7EmtSJlRkUb84bUGXmvIaSp6foauW7cuRctliSCt8KuZMHbs2GFvvPGGlSxZMnSbepIVPDds2BAxQDD4BYNQG1yqfzm8aqufixQpYqeeempoueXLlyeqLGt94QFZ30f3Qmvgo3q2NVNIWuhxCxcunKZ1IH7QzgNea8hpeF9DVn+tpbSglOmDDf/++2+755577Pvvv7eXXnrJypcvn6isrunudOKWcPPmzbMyZcqEQrNm3VDv8/z580PLHD161N1P09wFfxB9r0rzsmXLQsupQq1ZP4Lp8EShXbOF7Nq1K3TdwoULXZgOXw4AAADxKdMr0oMHD7aPP/7YHnjgAXdClNWrV4duU/tD0aJF3cDDjh07upOhaNo7hV6djEX31RzSQeBWVVvzRWvKOgVsLbN582YbPXp0aJ21atWyxo0b2yOPPGL9+/d369d81ArwOtFLoH379jZt2jR3Ihh97dy504YPH+4enzmkAQAAkOlB+pNPPnGX4WcyDEyZMiV0Gu+JEye6QKwTpGj2DYXqdu3aRSx/2223uZaNqVOnujYRnWhFJ16pVq1axHKjRo2yESNG2KBBg9zAPz3G+PHjI2b8UI/05MmT3RkUe/fu7W7THNU6IQsAAACQ6UFap+pOCbVTHK+lQu0bt99+u/tKjqrQTzzxhPs6XoO6TjsOAAAAZLkeaQAAACA7IkgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAA2TFI//LLLzZw4EBr3bq1Va9e3Vq2bBlzuSVLllibNm2sZs2a1rRpU5s+fXrM5V5++WVr0qSJW+66666zlStXJlpm//797jEvuugiq1OnjnXv3t22bNmSaLmNGzda165drXbt2la/fn0bMmSIHTx4MB1+awAAAGR3mR6kf/rpJxeSzzjjDKtcuXLMZVatWmU9e/Z0QfvFF1+0tm3bulA7a9asRCF6zJgxdvPNN9sLL7zg1tmtWzf74YcfIpbr06ePLVq0yB599FG3/O+//2633nprREjeu3evdenSxf78808bN26c9evXz959910bMGDACfpLAAAAIDvJm9kboOrxlVde6b7v37+/ffPNN4mWmTBhggvRQ4cOdT9ffPHFtnXrVhs7dqyrOufOndsOHz5skyZNss6dO7sqstSrV89atWplzz33nAvM8tVXX9nixYtd0G7UqJG7rmrVqq7KPWfOHOvQoYO7bubMmS5Mz50710qVKuWuy5Mnj/Xt29d69OiRZOgHAABAfMj0irRCcHIUkFesWGEtWrSIuF4Befv27fbdd9+5n7/88kvbt29fRGuIgm/z5s1dxTshIcFdp++LFy9uDRs2DC1Xrlw5q1u3rrstsHTpUtfOEYRoadasmeXPnz9iOQAAAMSnTA/Sx7Np0yY7cuSIVapUKeL6KlWquMv169dHXEYvp8qx2jO2bdsWWq5ixYqWK1euROsL1hEsF111VoiuUKFCxHIAAACIT5ne2nE8e/bscZeqIocLfg5uVxuGgm7BggUjlitRooS73L17t5UtW9YtV6xYsUSPo/UF6wrWF/2YsZbzoer4X3/9ler7KfwXKlQoTY+N7OfAgQOhIyoZhddafOK1Bl5ryGkOeH6G6j7RRddsGaQDSf0y4dfHWib44x1vueSuj15fSpZLjirsa9euTfX9FKLVK474otlj9EaQkXitxSdea+C1hpxmYxo+Q1WgzfZBOqgoR1eBVTGWoGqsy0OHDrmvAgUKJFouWI+W00DFaNEVaH0f3Dec+rDTOtAwX758odaU1EhrgEf2pFakzKhII/7wWgOvNeQ0FT0/Q9etW5ei5bJ8kFZPsoLnhg0bIgYIBr9gEGqDS/Uvh1dt9XORIkXs1FNPDS23fPnyRJVlrS88IOv76F5oDXxUz7ZmCkkLPW7hwoXTtA7ED9p5wGsNOQ3va8jqr7WUFpSy/GBDldU13d2CBQsirp83b56VKVMmFJo164Z6n+fPnx9a5ujRo+5+muYu+IPoe1Waly1bFlpOFWrN+hFMhycK7ZotZNeuXaHrFi5c6MJ0+HIAAACIT5lekVbfSjCdnM4uqLMOvv/++6F5oDX9XK9evaxjx47uZCia9k6hVydjGTx4cGj6PAVuze+s+aJ1HwVsLbN582YbPXp06PFq1apljRs3tkceecTNW120aFE3H3X58uXdiV4C7du3t2nTprkTwehr586dNnz4cPf4zCENAACATA/SCqj33HNPxHXBz1OmTAmdxnvixIkuEOsEKZp9Q6G6Xbt2Efe77bbbXMvG1KlTbceOHe5EKzrxSrVq1SKWGzVqlI0YMcIGDRrkBv7pMcaPHx8x44d6pCdPnuzOoNi7d293m+ao1glZAAAAgEwP0qeddlqiU3jHonaK47VUqH3j9ttvd1/JURX6iSeecF/Ha1DXaccBAACAbNcjDQAAAGRFBGkAAADAA0EaAAAAIEgDAAAAGYOKNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAOCBIA0AAAB4IEgDAAAAHgjSAAAAgAeCNAAAAECQBgAAADIGFWkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBOnj2Lhxo3Xt2tVq165t9evXtyFDhtjBgwd9/tYAAADIQfJm9gZkZXv37rUuXbpYuXLlbNy4cfbHH3/YsGHDbPfu3TZy5MjM3jwAAABkIoJ0MmbOnOnC9Ny5c61UqVLuujx58ljfvn2tR48eVrly5Yx6ngAAAJDF0NqRjKVLl7p2jiBES7NmzSx//vy2ZMmSjHh+AAAAkEURpJOxfv36RFVnhegKFSq42wAAABC/aO1Ihto6ihcvnuh6Xbdnzx6vP/iRI0csISHBvv76a6/758qVy1rUK2NHj5X2uj+yjzy5c9uaNWvc6yUz8FqLH7zWwGsNOU2eNH6GKq/pc/B4CNIe9KSk5I8bS3A/3/tL8aIFve+L7Cctr5W04rUWX3itgdcacppcachrBOk0UuVZVelo+/bt8x5oWKdOnbRuFgAAALIAeqSTobAc3Qt9+PBh27RpEzN2AAAAxDmCdDIaNmxoK1assF27doWuW7hwoQvTjRo1yojnBwAAAFlUroTMGsmUDaito2XLlla+fHnr2bOn7dy504YPH26XXnopJ2QBAACIcwTpFJwiXKcF/+KLL6xgwYIuWOuELPoeAAAA8YsgDQAAAHigRxoAAADwQJAGAAAAPBCkAQAAAA8EaQAAAMADQRoAAADwQJAGAAAAPBCkAQAAAA8EaWRLv/76q1WrVs3ef//9ZJd7++233XJ//PFHqtbvez9kDf3793cnT0pOkyZNbPDgwalet+/9EJ/0OtTrMSOtXbvWxo8fbwcOHMjQx0XG0fNbp04d/uRZAEEaOVrjxo3tjTfesOLFi2f2piCLefbZZ+22227L7M0ATkiQ1uubIA2ceHkz4DGAkIMHD2bo6dVLlSrlvoBo1atX54+CbCOj3zsBpAwVacRsa1DI2LFjR8T1u3fvtnPPPdemT5/ufl61apV17tzZateubeeff7716dPHdu7cmaj9QusbMGCAXXTRRXb99dfblClT3H32798fsf6NGze65T/66KMUPyuHDh1yh9kvvPBCu/TSS23EiBH2999/J9ui8dtvv9mdd95p5513nl122WX20ksvuXXokH20rVu32u233+6296qrrrK5c+fyislGVq5caW3atHHPn15733zzTbItGjNnzrTLL7/catWqZV26dLGvvvoq9BqONm3aNLesXvs9e/akDSgbtfwsW7bMWrVq5d4DbrrpJtu8ebN7f7v33nutbt26duWVV9r8+fMj7qsjW9dcc417D9SRrjFjxkS818iXX35p1157rdWsWdM9zpIlS2Juh+97pyxevNhuvfVWq1+/vtvWdu3a2dKlS0P31X0eeugh972W0XrC39v0/te3b1+3Tv3+N998c8T/BbKn//3vf3b33XfbBRdc4N6/9Ppas2ZN6PYJEya4120gISHBvQb0pe8DjRo1sokTJ2b49mdnBGkkosCYN2/eRP3HH374ofuH04eJPgg6depkxYoVcx8oTzzxhPun7dGjR6L1jR492nLlymWjRo2yBx54wFq3bm3Hjh2zefPmRSz31ltvWZkyZdw/cko988wzljt3bnd544032iuvvGKzZs1Kcnltv0KPDn0qRD3++OMuuCcV3rW9Cuh6Ezr77LPdB/G6det41WQD27dvtyFDhljXrl3da1QVvbvuusuOHDkSc3m9Bh577DFr0KCBOyyuSwWcWBYtWmQff/yxDRw40B555BH7/PPP3f8AssfrYuTIke69SpcKrfo/v//+++2ss85yvac1atRw123ZssXdZ+rUqe65vvjii23SpEnWvn17e/nll9114evVay1//vzu/UjfDxo0yF0fLi3vnaLt1Q7cU0895bZVYfqOO+5wO42isBSsS0UC7QDo9Sx79uxxOw7ff/+9Pfroo+7+hQoVcjuN4UEe2YuKUh07dnQ7RHpN6vVy+PBhF6bXr1/vllGxSYUh7TTKDz/84O6nrx9//NFdp9u0o6VlkXK0diCRokWLujCroKt/zoB+1geJWiW056vKjN6g9UYv+hBSlUdVmPAwrOp2dMho1qyZC876QJKjR4+6aq+qhwrxKaWKiio2ouCzfPly++CDD6xDhw4xl1fl5ttvv3VVde25i/bIGzZsaCeddFKi5VWt0ZdoL1/VIO1QVKlShVdOFqfQoKqxXpdSoEABV8lTlTl47sMpIOn1rfAtOlqhIx5BCIneIdPyCk3yyy+/uGClHUTt2CFrvy5mzJhhlStXdj///vvv7v2pW7du1qtXL3edKsoLFy60f//73+49UDvSV199tdvRCl4bet9TEFZoPf30023y5MnuuhdeeCE0JkOFAQXqcAo5aXnvDH9P1utN71/auX/zzTfd93p/rlChgrtdOwThrW3axr1797piQ+nSpUNV66ZNm7rX74MPPpjOf21kBB2FUEX63XffDb3f6XnVkYgXX3zRhg8f7j6/9H71n//8x71e//vf/7rXl97LdJ2OXOhSy2hZpBzv+IipRYsWtnr1avfPKaqq6J9Mb/YawKJDmPpgUQDW4U19VaxY0X1whB9OklgV5htuuMG+/vpr++mnn0IBV49x3XXXpeoZUbU4nAKu9qiTom3Th1x4kNKOgz6Ajrd+LfePf/wj2fUj6zjllFNCHyoSBKdt27YlWlavYx2liG7vueKKK2KuWxWbIEQHrztVuqnqZY/XRfBakDPPPNNdXnLJJaHr9B6hAKr/9Q0bNtiuXbusefPmid4jFUK++OIL97N20PQ+Ej6wWe8fet8IpMd7p7apX79+LswrCCksf/LJJ6417ng+/fRTt40lSpQIPbZ2/PR+GP3YyD4UivVeF/5+V6RIEXfkQrcFhQTtIOpzXHSp9zF96YhacJ2KU+HvbTg+KtKISf+A+kd87733XKVG/YL58uVzvYOqaOhDYNiwYe4rmg4fhYs12E//vPrwmD17tuvn06XezHVdaujwaDhtow5pJUXVp1jbE1Rn0rp+ZB3RM7XouRNVmaOph16hIvq1kdTrIjXrRtaS1HMX/b+uMKHnUxVsOfnkkyNuV/CV4HYVAs4444xEjxf+Gkrre6cq0KqA79u3zx0V1OOpNWPcuHGJ7huLdghUIFH4jhZUsZH96HUV/foUXRe8PoPPXX2mi3YAdQRYO4P/+te/3HUK3dpBROoQpBGT9l6DATdBkFbvnaorqmDokKQG7GmZaCVLloz4OTh8GU2DZNTDpynIdEgzI3pMVY2KNTc0lcT4psCilqLo1wavCwQtX9GvhaD3WdXdIFjHer2EX6ewnpb3TrUQfffdd67VJPz+6v9PCW2rKtn33HNPotuoQmZfel515CSaJgwIXp9BkH7uuefss88+c69LDXQNdrB03aZNm+iP9kCQRpK0Z6q+ZY1wVxVDA1OkcOHCbrS5/nF1qMhX27ZtXY+hBnQpuOtw54mm7dXee3BYSzTYQgN1YvVIIz7kyZPHzjnnHDfgUAOvAuqRRXzTUTLtaC1YsMANxA6ouKCgG4QRHRJ//fXXXbU4qG6r5SJ8dqK0vncGRzyCKrpoQKQGMAYtKuG3Rx89U/vKO++841pbtC3IGfQa1Ngg9coH43f++usvNyBaR5cDOoGLCgaalUM90cHRmapVq7rrdBsneUk9gjSSpDddfYA8/PDDoQGIAQ1KUeDQdFEK3PqHVO+eBvtp+qekeo7Dad3qQdXsIJpxQ4coTzQNKtRhTYV3jdLXdmswRlApQvzSIXPN6KLBq9qpU+UvOOTJAML43snSIEQdMdN7loKJXhtqp9B7nQZuid4PNYhRR/D0pR12FR+id9DT8t5ZqVIlK1u2rBuwqDYP9VxrO3SkLVzQA65B1apca/5pBadbbrnFDUjTgEXN6FCuXDl3FEb93aeeeqq7HdmPXjevvfaaO9Kh15V2knS0Vzteei0G1K6pgoF6ojVzTEBFJc1Mo51BdrBSj8GGSJL2ThUo1FesUd2qGgc05ZI+NLTXqx5nTb+kPVq9YcfqE0yK1ivBHKknmsKytlNT2WmaIH3pg7FevXqJeiQRX7RTp+kQVUVUoNaRmGCWhvABY4g/Cp6ayk6D9bp37+4CqmbjCJ+HXGFWO+Vqs1DrhL7X+0vQS50e751qv1A416UeY+zYsW579P4VToMQe/fu7arPmhkpmA5PrSOaDk9hSlP/qa1OvdqqaitEIXvS+5NmKFKRSO9hKhLp81vnbAgfWCvBkdjwKe6C72PNZoTjy5UQPhM3kMFUndFsCaqSZBYd/tTc2PowijUACPFL04SpQq2Wj9NOOy2zNwcAkMXQ2oFMocngFaDVZxhU/TKKKjI6LKreRx1+VV+jRrzrRAWIXzqzneb21VzSOgSq6cA0MEeVakI0ACAWgjQyhQ41qjdP0+9Ezx2tgySaIiop6ldNS8+qWlR02FVnCBO1eTz//PNpGjiJ7E+HQnVmL00PpR0sHQbXWTh1OmUAAGKhtQNZjmbQ0ECY5Gb70JmaAAAAMhNBGlmOpotK7ixdqhRyqB0AAGQ2gjQAAADggenvAAAAAA8EaQAAAMADQRoAAADwwPR3ABCHdMrolNDZ0ZI7bTUAxDMGGwJAHFq9enXEzzpNtaaenDx5csT1VapU4RTpAJAEKtIAEIdq164d8XOpUqXciY6irwcAJI0eaQBAIg8//LDVq1fPDhw4kOg2nTCpRYsWEW0igwcPtpkzZ1qzZs3s3HPPtebNm7uzREbbvn27DRw40Bo2bOiWa9KkiTs1+99//82zACDboSINAIgZlt966y2bN2+etWvXLnT9unXrXAuIwnC4RYsWuevvvvtuK1SokM2YMcPuv/9+y5Mnj1199dWhEK11qfLdq1cvq1Chgq1atcomTZpkW7ZssWHDhvFMAMhWCNIAgETOPvtsV5GePn16RJCeNm2a65lu06ZNxPK7du2y2bNn28knn+x+btSokbVs2dJGjx4dCtLjx4+3PXv2uEp1uXLl3HX169e3ggUL2ogRI6xr166uJxsAsgtaOwAASVal165da1988YX7ef/+/favf/3L2rZta0WKFIlYVoE4CNGiSrTaO3755Rf77bff3HWLFy92M4CccsoprpUj+FKbh3z++ec8EwCyFSrSAICYrrjiCitfvrxr0zj//PPt7bffdj3TN910U6Jlw0N09HW7d++2smXL2s6dO+3jjz+2GjVqxHw8VbUBIDshSAMAYlIv880332xjxoyxfv36uUCtynOlSpUSLbtjx44krzvppJPcZcmSJd3AxHvvvTfm46lSDQDZCUEaAJAk9UdrVo2+ffvaxo0b3WUsn332mQvOQRX66NGjNn/+fDegUNVoady4sS1ZssRdV6JECf7qALI9gjQAIEnFixe31q1b2+uvv+7aPDRdXSyqNnfp0sV69uwZmrVjw4YNrpod0Iwey5cvt/bt21unTp2sYsWKdvjwYfv1119t6dKlNmjQoFDoBoDsgCANAEiWBg0qSCsAq90jFgVszbjxzDPP2NatW+3000+3kSNHuvuGt25oZg+dRfHll1+2bdu2uUGLCuiXXXaZC+0AkJ1winAAQLKGDx/ugrRm3VDlOZr6ntVLHT23NADkdFSkAQAxrV692n7++WfXpnHjjTfGDNEAEM8I0gCAmBSe1e+sQYJJzbQBAPGM1g4AAADAA2c2BAAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAADwQpAEAAAAPBGkAAADAA0EaAAAA8ECQBgAAACz1/h8RmQOVclrgpQAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 800x600 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Set plot style\n",
    "sns.set(style=\"whitegrid\")\n",
    "\n",
    "# Calculate counts and percentages\n",
    "counts = collapsed_df['Type'].value_counts()\n",
    "total = counts.sum()\n",
    "percentages = (counts / total * 100).round(2)\n",
    "\n",
    "# Desired order\n",
    "type_order = ['very_high', 'high', 'moderate', 'low']\n",
    "\n",
    "# Plot\n",
    "plt.figure(figsize=(8, 6))\n",
    "ax = sns.countplot(data=collapsed_df, x='Type', order=type_order)\n",
    "\n",
    "# Annotate each bar\n",
    "for p in ax.patches:\n",
    "    height = p.get_height()\n",
    "    label = p.get_x() + p.get_width() / 2\n",
    "    category = p.get_x() + p.get_width() / 2\n",
    "    type_name = p.get_x() + p.get_width() / 2\n",
    "    percentage = (height / total) * 100\n",
    "    ax.annotate(f'{percentage:.1f}%', (p.get_x() + p.get_width() / 2, height),\n",
    "                ha='center', va='bottom', fontsize=11)\n",
    "\n",
    "# Labels\n",
    "plt.title('Distribution of Type Categories')\n",
    "plt.xlabel('Type')\n",
    "plt.ylabel('Count')\n",
    "plt.ylim(0, counts.max() * 1.1)  # Add some space above bars\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65e1acea-88b7-40f9-9330-e39fa4843240",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "browser_data",
   "language": "python",
   "name": "browser_data"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
